--- webkitgtk-2.4.9-orig/GNUmakefile.in	2016-02-02 13:23:23.000000000 -0500
+++ webkitgtk-2.4.9/GNUmakefile.in	2016-02-02 22:10:23.000000000 -0500
@@ -23321,7 +23321,7 @@
 	$(WINMM_LIBS) \
 	-lm \
 	-lpthread \
-	-lstdc++
+	-lc++
 
 Programs_minidom_LDFLAGS = \
 	-no-install
@@ -23344,7 +23344,7 @@
 	$(WINMM_LIBS) \
 	-lm \
 	-lpthread \
-	-lstdc++
+	-lc++
 
 Programs_LLIntOffsetsExtractor_LDFLAGS = \
 	-no-install
