--- webkitgtk-2.4.9-orig/GNUmakefile.in	2016-02-02 13:23:23.000000000 -0500
+++ webkitgtk-2.4.9/GNUmakefile.in	2016-02-08 00:27:10.000000000 -0500
@@ -4799,6 +4799,8 @@
 	Source/WebCore/plugins/PluginViewBase.h \
 	Source/WebCore/plugins/PluginView.cpp \
 	Source/WebCore/plugins/PluginView.h \
+	Source/WebCore/plugins/PluginViewNone.cpp \
+	Source/WebCore/plugins/PluginPackageNone.cpp \
 	Source/WebCore/plugins/npapi.h \
 	Source/WebCore/plugins/npfunctions.h \
 	Source/WebCore/plugins/npruntime.h \
@@ -6375,6 +6377,8 @@
 	Source/WebCore/plugins/libWebCore_la-PluginPackage.lo \
 	Source/WebCore/plugins/libWebCore_la-PluginStream.lo \
 	Source/WebCore/plugins/libWebCore_la-PluginView.lo \
+	Source/WebCore/plugins/libWebCore_la-PluginViewNone.lo \
+	Source/WebCore/plugins/libWebCore_la-PluginPackageNone.lo \
 	Source/WebCore/rendering/libWebCore_la-AutoTableLayout.lo \
 	Source/WebCore/rendering/libWebCore_la-BidiRun.lo \
 	Source/WebCore/rendering/libWebCore_la-break_lines.lo \
@@ -10796,6 +10800,8 @@
 	Source/WebKit2/WebProcess/Plugins/libwebkit2gtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la-PluginProcessConnectionManager.lo \
 	Source/WebKit2/WebProcess/Plugins/libwebkit2gtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la-PluginProxy.lo \
 	Source/WebKit2/WebProcess/Plugins/libwebkit2gtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la-PluginView.lo \
+	Source/WebKit2/WebProcess/Plugins/libwebkit2gtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la-PluginViewNone.lo \
+	Source/WebKit2/WebProcess/Plugins/libwebkit2gtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la-PluginPackageNone.lo \
 	Source/WebKit2/WebProcess/ResourceCache/libwebkit2gtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la-WebResourceCacheManager.lo \
 	Source/WebKit2/WebProcess/Storage/libwebkit2gtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la-StorageAreaImpl.lo \
 	Source/WebKit2/WebProcess/Storage/libwebkit2gtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la-StorageAreaMap.lo \
@@ -19503,6 +19509,8 @@
 	Source/WebCore/plugins/PluginViewBase.h \
 	Source/WebCore/plugins/PluginView.cpp \
 	Source/WebCore/plugins/PluginView.h \
+	Source/WebCore/plugins/PluginViewNone.cpp \
+	Source/WebCore/plugins/PluginPackageNone.cpp \
 	Source/WebCore/plugins/npapi.h \
 	Source/WebCore/plugins/npfunctions.h \
 	Source/WebCore/plugins/npruntime.h \
