diff -rupN ocamlgraph-1.8.6.orig/Makefile.in ocamlgraph-1.8.6/Makefile.in
--- ocamlgraph-1.8.6.orig/Makefile.in	2015-01-23 07:24:01.000000000 +0000
+++ ocamlgraph-1.8.6/Makefile.in	2015-06-26 18:34:03.478000000 +0000
@@ -16,8 +16,8 @@
 ##########################################################################
 
 # Where to install the binaries
-DESTDIR	=
 prefix	=@prefix@
+DESTDIR=$(prefix)
 exec_prefix=@exec_prefix@
 datarootdir=@datarootdir@
 BINDIR	=$(DESTDIR)@bindir@
@@ -418,7 +418,7 @@ ifeq (@LABLGNOMECANVAS@,yes)
 endif
 
 ifdef DESTDIR
-OCAMLFINDDEST := -destdir $(DESTDIR)
+OCAMLFINDDEST := -destdir $(DESTDIR)$(OCAMLLIB)
 endif
 
 install-findlib: META
