diff -rc lcov-orig/bin/gendesc lcov-1.1/bin/gendesc
*** lcov-orig/bin/gendesc	2004-10-22 16:10:55.000000000 +0200
--- lcov-1.1/bin/gendesc	2004-10-22 16:27:52.000000000 +0200
***************
*** 1,4 ****
! #!/usr/bin/perl -w
  #
  #   Copyright (c) International Business Machines  Corp., 2002
  #
--- 1,4 ----
! #! @PERL@ -w
  #
  #   Copyright (c) International Business Machines  Corp., 2002
  #
diff -rc lcov-orig/bin/genhtml lcov-1.1/bin/genhtml
*** lcov-orig/bin/genhtml	2004-10-22 16:10:55.000000000 +0200
--- lcov-1.1/bin/genhtml	2004-10-22 16:29:40.000000000 +0200
***************
*** 1,4 ****
! #!/usr/bin/perl -w
  #
  #   Copyright (c) International Business Machines  Corp., 2002
  #
--- 1,4 ----
! #! @PERL@ -w
  #
  #   Copyright (c) International Business Machines  Corp., 2002
  #
***************
*** 200,208 ****
  {
  	$config = read_config($ENV{"HOME"}."/.lcovrc");
  }
! elsif (-r "/etc/lcovrc")
  {
! 	$config = read_config("/etc/lcovrc");
  }
  
  if ($config)
--- 200,208 ----
  {
  	$config = read_config($ENV{"HOME"}."/.lcovrc");
  }
! elsif (-r "@PREFIX@/etc/lcovrc")
  {
! 	$config = read_config("@PREFIX@/etc/lcovrc");
  }
  
  if ($config)
diff -rc lcov-orig/bin/geninfo lcov-1.1/bin/geninfo
*** lcov-orig/bin/geninfo	2004-10-22 16:10:56.000000000 +0200
--- lcov-1.1/bin/geninfo	2004-10-22 16:29:40.000000000 +0200
***************
*** 1,4 ****
! #!/usr/bin/perl -w
  #
  #   Copyright (c) International Business Machines  Corp., 2002
  #
--- 1,4 ----
! #! @PERL@ -w
  #
  #   Copyright (c) International Business Machines  Corp., 2002
  #
***************
*** 124,132 ****
  {
  	$config = read_config($ENV{"HOME"}."/.lcovrc");
  }
! elsif (-r "/etc/lcovrc")
  {
! 	$config = read_config("/etc/lcovrc");
  }
  
  if ($config)
--- 124,132 ----
  {
  	$config = read_config($ENV{"HOME"}."/.lcovrc");
  }
! elsif (-r "@PREFIX@/etc/lcovrc")
  {
! 	$config = read_config("@PREFIX@/etc/lcovrc");
  }
  
  if ($config)
diff -rc lcov-orig/bin/genpng lcov-1.1/bin/genpng
*** lcov-orig/bin/genpng	2004-10-22 16:10:56.000000000 +0200
--- lcov-1.1/bin/genpng	2004-10-22 16:28:14.000000000 +0200
***************
*** 1,4 ****
! #!/usr/bin/perl -w
  #
  #   Copyright (c) International Business Machines  Corp., 2002
  #
--- 1,4 ----
! #! @PERL@ -w
  #
  #   Copyright (c) International Business Machines  Corp., 2002
  #
diff -rc lcov-orig/bin/lcov lcov-1.1/bin/lcov
*** lcov-orig/bin/lcov	2004-10-22 16:10:56.000000000 +0200
--- lcov-1.1/bin/lcov	2004-10-22 16:29:39.000000000 +0200
***************
*** 1,4 ****
! #!/usr/bin/perl -w
  #
  #   Copyright (c) International Business Machines  Corp., 2002
  #
--- 1,4 ----
! #! @PERL@ -w
  #
  #   Copyright (c) International Business Machines  Corp., 2002
  #
***************
*** 166,174 ****
  {
  	$config = read_config($ENV{"HOME"}."/.lcovrc");
  }
! elsif (-r "/etc/lcovrc")
  {
! 	$config = read_config("/etc/lcovrc");
  }
  
  if ($config)
--- 166,174 ----
  {
  	$config = read_config($ENV{"HOME"}."/.lcovrc");
  }
! elsif (-r "@PREFIX@/etc/lcovrc")
  {
! 	$config = read_config("@PREFIX@/etc/lcovrc");
  }
  
  if ($config)
diff -rc lcov-orig/bin/updateversion.pl lcov-1.1/bin/updateversion.pl
*** lcov-orig/bin/updateversion.pl	2004-10-22 16:10:55.000000000 +0200
--- lcov-1.1/bin/updateversion.pl	2004-10-22 16:28:39.000000000 +0200
***************
*** 1,4 ****
! #!/usr/bin/perl -w
  #
  # Update version and date strings in LCOV files
  #
--- 1,4 ----
! #! @PERL@ -w
  #
  # Update version and date strings in LCOV files
  #
diff -rc lcov-orig/Makefile lcov-1.1/Makefile
*** lcov-orig/Makefile	2004-10-22 16:10:55.000000000 +0200
--- lcov-1.1/Makefile	2004-10-22 16:26:18.000000000 +0200
***************
*** 16,23 ****
  DATE    := $(shell date +%Y-%m-%d)
  
  CFG_DIR := $(PREFIX)/etc
! BIN_DIR := $(PREFIX)/usr/bin
! MAN_DIR := $(PREFIX)/usr/share/man
  TMP_DIR := /tmp/lcov-tmp.$(shell echo $$$$)
  FILES   := $(wildcard bin/*) $(wildcard man/*) README CHANGES Makefile \
  	   $(wildcard rpm/*)
--- 16,23 ----
  DATE    := $(shell date +%Y-%m-%d)
  
  CFG_DIR := $(PREFIX)/etc
! BIN_DIR := $(PREFIX)/bin
! MAN_DIR := $(PREFIX)/share/man
  TMP_DIR := /tmp/lcov-tmp.$(shell echo $$$$)
  FILES   := $(wildcard bin/*) $(wildcard man/*) README CHANGES Makefile \
  	   $(wildcard rpm/*)
