diff --git a/src/SystemPaths_TEST.cc b/src/SystemPaths_TEST.cc
index b72a34bf..4d28305b 100644
--- a/src/SystemPaths_TEST.cc
+++ b/src/SystemPaths_TEST.cc
@@ -346,19 +346,17 @@ TEST_F(SystemPathsFixture, FindFile)
 #ifndef _WIN32
   const auto tmpDir = "/tmp";
   const auto uriTmpDir = "file:///tmp";
-  const auto homeDir = "/home";
   const auto badDir = "/bad";
   const auto uriBadDir = "file:///bad";
 #else
   const auto tmpDir = "C:\\Windows";
   const auto uriTmpDir = "file://C:/Windows";
-  const auto homeDir = "C:\\Users";
   const auto badDir = "C:\\bad";
   const auto uriBadDir = "file://C:/bad";
 #endif
   {
     EXPECT_EQ(tmpDir, sp.FindFile(uriTmpDir));
-    EXPECT_EQ(homeDir, sp.FindFile(homeDir));
+    EXPECT_EQ(tmpDir, sp.FindFile(tmpDir));
     EXPECT_EQ("", sp.FindFile(badDir));
     EXPECT_EQ("", sp.FindFile(uriBadDir));
   }
@@ -380,28 +378,28 @@ TEST_F(SystemPathsFixture, FindFile)
 
   // Custom callback
   {
-    auto homeCb = [homeDir](const std::string &_s)
+    auto tempCb = [tmpDir](const std::string &_s)
     {
-      return _s == "home" ? homeDir : "";
+      return _s == "temp" ? tmpDir : "";
     };
     auto badCb = [badDir](const std::string &_s)
     {
       return _s == "bad" ? badDir : "";
     };
 
-    EXPECT_EQ("", sp.FindFile("home"));
+    EXPECT_EQ("", sp.FindFile("temp"));
     EXPECT_EQ("", sp.FindFile("bad"));
     EXPECT_EQ("", sp.FindFile("banana"));
 
-    sp.AddFindFileCallback(homeCb);
+    sp.AddFindFileCallback(tempCb);
 
-    EXPECT_EQ(homeDir, sp.FindFile("home"));
+    EXPECT_EQ(tmpDir, sp.FindFile("temp"));
     EXPECT_EQ("", sp.FindFile("bad"));
     EXPECT_EQ("", sp.FindFile("banana"));
 
     sp.AddFindFileCallback(badCb);
 
-    EXPECT_EQ(homeDir, sp.FindFile("home"));
+    EXPECT_EQ(tmpDir, sp.FindFile("temp"));
     EXPECT_EQ("", sp.FindFile("bad"));
     EXPECT_EQ("", sp.FindFile("banana"));
   }
