From 8d49f5ef8692c352a62f4f8b1bc68e6e210bbee6 Mon Sep 17 00:00:00 2001
From: Yaroslav  Bolyukin <iam@lach.pw>
Date: Wed, 23 Dec 2020 18:02:14 +0300
Subject: [PATCH 3/3] plugins/qpa: allow using nixos wrapper

Signed-off-by: Yaroslav  Bolyukin <iam@lach.pw>
---
 src/plugins/qpa/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/qpa/main.cpp b/src/plugins/qpa/main.cpp
index efd236b..a69c046 100644
--- a/src/plugins/qpa/main.cpp
+++ b/src/plugins/qpa/main.cpp
@@ -23,7 +23,7 @@ public:

 QPlatformIntegration *KWinIntegrationPlugin::create(const QString &system, const QStringList &paramList)
 {
-    if (!QCoreApplication::applicationFilePath().endsWith(QLatin1String("kwin_wayland")) && !qEnvironmentVariableIsSet("KWIN_FORCE_OWN_QPA")) {
+    if (!QCoreApplication::applicationFilePath().endsWith(QLatin1String("kwin_wayland")) && !QCoreApplication::applicationFilePath().endsWith(QLatin1String(".kwin_wayland-wrapped")) && !qEnvironmentVariableIsSet("KWIN_FORCE_OWN_QPA")) {
         // Not KWin
         return nullptr;
     }
-- 
2.29.2

