diff -rc asc-support-2.6/lib/asc-main.c asc-support-2.6-new/lib/asc-main.c
*** asc-support-2.6/lib/asc-main.c	2008-11-10 14:12:47.000000000 +0100
--- asc-support-2.6-new/lib/asc-main.c	2010-08-24 11:02:04.000000000 +0200
***************
*** 7,13 ****
  #include <ctype.h>
  #include <string.h>
  #include <sys/time.h>
- #include <sys/resource.h>
  #include <unistd.h>
  #include <aterm2.h>
  #include <sglrInterface.h>
--- 7,12 ----
***************
*** 46,52 ****
  }
  
  static void printStats() {
-   struct rusage usage;
    FILE *file;
    char buf[BUFSIZ];
    int size, resident, shared, trs, lrs, drs, dt;
--- 45,50 ----
***************
*** 61,74 ****
      fprintf(stderr, "could not open %s\n", buf);
      perror("");
    }
!   if (getrusage(RUSAGE_SELF, &usage) == -1) {
!     perror("rusage");
!   } else {
!     fprintf(stderr, "utime       : %ld.%06d sec.\n",
! 	    (long)usage.ru_utime.tv_sec, (int)usage.ru_utime.tv_usec);
!     fprintf(stderr, "stime       : %ld.%06d sec.\n",
! 	    (long)usage.ru_stime.tv_sec, (int)usage.ru_stime.tv_usec);
!   }
  }
  
  static ATbool toolbusMode(int argc, char* argv[]) {
--- 59,66 ----
      fprintf(stderr, "could not open %s\n", buf);
      perror("");
    }
!   fprintf(stderr, "utime       : %ld.%06d sec.\n", 0, 0);
!   fprintf(stderr, "stime       : %ld.%06d sec.\n", 0, 0);
  }
  
  static ATbool toolbusMode(int argc, char* argv[]) {
