--- a/lib/Driver/Tools.cpp	2015-07-31 00:47:41.000000000 +0200
+++ b/lib/Driver/Tools.cpp	2015-09-11 21:30:50.057895565 +0200
@@ -8150,15 +8150,6 @@
     CmdArgs.push_back("-shared");
   }
 
-  if (Arch == llvm::Triple::arm || Arch == llvm::Triple::armeb ||
-      Arch == llvm::Triple::thumb || Arch == llvm::Triple::thumbeb ||
-      (!Args.hasArg(options::OPT_static) &&
-       !Args.hasArg(options::OPT_shared))) {
-    CmdArgs.push_back("-dynamic-linker");
-    CmdArgs.push_back(Args.MakeArgString(
-        D.DyldPrefix + getLinuxDynamicLinker(Args, ToolChain)));
-  }
-
   CmdArgs.push_back("-o");
   CmdArgs.push_back(Output.getFilename());
