diff --git a/ext/test/http/curl_http_test.cc b/ext/test/http/curl_http_test.cc
index 2e2cf3d41..d436f25c2 100644
--- a/ext/test/http/curl_http_test.cc
+++ b/ext/test/http/curl_http_test.cc
@@ -273,7 +273,7 @@ TEST_F(BasicCurlHttpTests, HttpResponse)
   ASSERT_EQ(count, 4);
 }
 
-TEST_F(BasicCurlHttpTests, SendGetRequest)
+TEST_F(BasicCurlHttpTests, DISABLED_SendGetRequest)
 {
   received_requests_.clear();
   auto session_manager = http_client::HttpClientFactory::Create();
@@ -290,7 +290,7 @@ TEST_F(BasicCurlHttpTests, SendGetRequest)
   ASSERT_TRUE(handler->got_response_.load(std::memory_order_acquire));
 }
 
-TEST_F(BasicCurlHttpTests, SendPostRequest)
+TEST_F(BasicCurlHttpTests, DISABLED_SendPostRequest)
 {
   received_requests_.clear();
   auto session_manager = http_client::HttpClientFactory::Create();
@@ -316,7 +316,7 @@ TEST_F(BasicCurlHttpTests, SendPostRequest)
   session_manager->FinishAllSessions();
 }
 
-TEST_F(BasicCurlHttpTests, RequestTimeout)
+TEST_F(BasicCurlHttpTests, DISABLED_RequestTimeout)
 {
   received_requests_.clear();
   auto session_manager = http_client::HttpClientFactory::Create();
@@ -361,7 +361,7 @@ TEST_F(BasicCurlHttpTests, CurlHttpOperations)
 }
 
 #ifdef ENABLE_OTLP_RETRY_PREVIEW
-TEST_F(BasicCurlHttpTests, RetryPolicyEnabled)
+TEST_F(BasicCurlHttpTests, DISABLED_RetryPolicyEnabled)
 {
   RetryEventHandler handler;
   http_client::HttpSslOptions no_ssl;
@@ -379,7 +379,7 @@ TEST_F(BasicCurlHttpTests, RetryPolicyEnabled)
   ASSERT_TRUE(operation.IsRetryable());
 }
 
-TEST_F(BasicCurlHttpTests, RetryPolicyDisabled)
+TEST_F(BasicCurlHttpTests, DISABLED_RetryPolicyDisabled)
 {
   RetryEventHandler handler;
   http_client::HttpSslOptions no_ssl;
@@ -397,7 +397,7 @@ TEST_F(BasicCurlHttpTests, RetryPolicyDisabled)
   ASSERT_FALSE(operation.IsRetryable());
 }
 
-TEST_F(BasicCurlHttpTests, ExponentialBackoffRetry)
+TEST_F(BasicCurlHttpTests, DISABLED_ExponentialBackoffRetry)
 {
   using ::testing::AllOf;
   using ::testing::Gt;
@@ -445,7 +445,7 @@ TEST_F(BasicCurlHttpTests, ExponentialBackoffRetry)
 }
 #endif  // ENABLE_OTLP_RETRY_PREVIEW
 
-TEST_F(BasicCurlHttpTests, SendGetRequestSync)
+TEST_F(BasicCurlHttpTests, DISABLED_SendGetRequestSync)
 {
   received_requests_.clear();
   curl::HttpClientSync http_client;
@@ -470,7 +470,7 @@ TEST_F(BasicCurlHttpTests, SendGetRequestSyncTimeout)
               result.GetSessionState() == http_client::SessionState::SendFailed);
 }
 
-TEST_F(BasicCurlHttpTests, SendPostRequestSync)
+TEST_F(BasicCurlHttpTests, DISABLED_SendPostRequestSync)
 {
   received_requests_.clear();
   curl::HttpClientSync http_client;
@@ -576,7 +576,7 @@ TEST_F(BasicCurlHttpTests, SendGetRequestAsyncTimeout)
   }
 }
 
-TEST_F(BasicCurlHttpTests, SendPostRequestAsync)
+TEST_F(BasicCurlHttpTests, DISABLED_SendPostRequestAsync)
 {
   curl::HttpClient http_client;
 
@@ -615,7 +615,7 @@ TEST_F(BasicCurlHttpTests, SendPostRequestAsync)
   }
 }
 
-TEST_F(BasicCurlHttpTests, FinishInAsyncCallback)
+TEST_F(BasicCurlHttpTests, DISABLED_FinishInAsyncCallback)
 {
   curl::HttpClient http_client;
 
@@ -653,7 +653,7 @@ TEST_F(BasicCurlHttpTests, FinishInAsyncCallback)
   }
 }
 
-TEST_F(BasicCurlHttpTests, ElegantQuitQuick)
+TEST_F(BasicCurlHttpTests, DISABLED_ElegantQuitQuick)
 {
   auto http_client = http_client::HttpClientFactory::Create();
   std::static_pointer_cast<curl::HttpClient>(http_client)->MaybeSpawnBackgroundThread();
