From f2d2d869a4aa58430415d0969f5e80ece0142ad2 Mon Sep 17 00:00:00 2001
From: jkachmar <j@mercury.com>
Date: Thu, 23 Oct 2025 17:36:48 -0400
Subject: [PATCH] disable tests that require redis

---
 internal/peer/peers_test.go | 4 ++++
 1 file changed, 4 insertions(+), 0 deletions(-)

diff --git a/internal/peer/peers_test.go b/internal/peer/peers_test.go
index dae54067d8..b33ebc4979 100644
--- a/internal/peer/peers_test.go
+++ b/internal/peer/peers_test.go
@@ -86,6 +86,10 @@
 }
 
 func TestPeerShutdown(t *testing.T) {
+	// Skip tests requiring Redis so that nixpkgs can build & test refinery.
+	if os.Getenv("NO_REDIS_TEST") != "" {
+		t.Skip("Skipping Redis-requiring test");
+	}
 	c := &config.MockConfig{
 		GetPeerListenAddrVal: "0.0.0.0:8081",
 		PeerManagementType:   "redis",
