diff --git a/_scripts/ebuilder.config.js b/_scripts/ebuilder.config.js
index 14d0d9df1..c5fc569c8 100644
--- a/_scripts/ebuilder.config.js
+++ b/_scripts/ebuilder.config.js
@@ -1,6 +1,8 @@
 const { name, productName } = require('../package.json')
 
 const config = {
+  electronVersion: "@electron-version@",
+  electronDist: "electron-dist",
   appId: `io.freetubeapp.${name}`,
   copyright: 'Copyleft © 2020-2025 freetubeapp@protonmail.com',
   // asar: false,
