<<<<<<< HEAD
diff --git a/_scripts/ebuilder.config.mjs b/_scripts/ebuilder.config.mjs
index 55e72b227..d218f5ac4 100644
--- a/_scripts/ebuilder.config.mjs
+++ b/_scripts/ebuilder.config.mjs
@@ -2,6 +2,8 @@ import packageDetails from '../package.json' with { type: 'json' }
 
 /** @type {import('electron-builder').Configuration} */
 export default {
+  electronVersion: "@electron-version@",
+  electronDist: "electron-dist",
   appId: `io.freetubeapp.${packageDetails.name}`,
   copyright: 'Copyleft © 2020-2025 freetubeapp@protonmail.com',
=======
diff --git a/_scripts/ebuilder.config.js b/_scripts/ebuilder.config.js
index 5b79d961..9f5945d2 100644
--- a/_scripts/ebuilder.config.js
+++ b/_scripts/ebuilder.config.js
@@ -1,6 +1,8 @@
 const { name, productName } = require('../package.json')
 
 const config = {
+  electronVersion: "@electron-version@",
+  electronDist: "electron-dist",
   appId: `io.freetubeapp.${name}`,
   copyright: 'Copyleft © 2020-2024 freetubeapp@protonmail.com',
>>>>>>> 4dbde0a9cadc (Fixed upon CodeReview)
   // asar: false,
