diff --git a/_scripts/build.mjs b/_scripts/build.mjs
index 1617f1ad5..b9a1de43b 100644
--- a/_scripts/build.mjs
+++ b/_scripts/build.mjs
@@ -14,7 +14,7 @@ if (platform === 'darwin') {
     arch = Arch.arm64
   }
 
-  targets = Platform.MAC.createTarget(['DMG', 'zip', '7z'], arch)
+  targets = Platform.MAC.createTarget(['dir'], arch)
 } else if (platform === 'win32') {
   let arch = Arch.x64
 
@@ -34,7 +34,7 @@ if (platform === 'darwin') {
     arch = Arch.armv7l
   }
 
-  targets = Platform.LINUX.createTarget(['deb', 'zip', '7z', 'rpm', 'AppImage', 'pacman'], arch)
+  targets = Platform.LINUX.createTarget(['dir'], arch)
 }
 
 const output = await build({ targets, config, publish: 'never' })
