--- transfig/Imakefile.orig	Thu Mar 25 22:10:42 2004
+++ transfig/Imakefile	Thu Mar 25 22:14:04 2004
@@ -14,11 +14,18 @@
 XCOMM party to do so, with the only requirement being that this copyright 
 XCOMM notice remain intact.
 
+BINDIR = __PREFIX/bin
+USRLIBDIR = __PREFIX/lib
+DESTDIR = __DESTROOT
+MKDIRHIER = mkdirhier
+MANDIR = __PREFIX/share/man/$(MANSUFFIX)
+INSTALLFLAGS = -c
+
 XCOMM Uncomment the USELATEX2E flag in the transfig/Imakefile file to use the
 XCOMM \\usepackage{} command for LaTeX2e. 
 XCOMM The default is to use \\documentstyle{} for LaTeX209.
 
-XCOMM USELATEX2E = -DLATEX2E
+USELATEX2E = -DLATEX2E
 
 XCOMM ******* DON'T CHANGE ANYTHIN BELOW THIS POINT *******
 
