diff --git a/rsvg/tests/errors.rs b/rsvg/tests/errors.rs
index 02b134bee..46391fa5d 100644
--- a/rsvg/tests/errors.rs
+++ b/rsvg/tests/errors.rs
@@ -14,15 +14,12 @@ use rsvg::{CairoRenderer, ImplementationLimit, Loader, LoadingError, RenderingEr
 
 #[ignore]
 #[test]
+#[should_panic]
 fn too_many_elements() {
     let name = "tests/fixtures/errors/bug515-too-many-elements.svgz";
 
-    assert!(matches!(
-        Loader::new().read_path(name),
-        Err(LoadingError::LimitExceeded(
-            ImplementationLimit::TooManyLoadedElements
-        ))
-    ));
+    // libxml2 might return an error without updating the element count, so just expect any error, not specifically too many elements
+    Loader::new().read_path(name).unwrap();
 }
 
 fn rendering_instancing_limit(name: &str) {
