diff --git a/CMakeLists.txt b/CMakeLists.txt
index f48d059..ac2c4e7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,10 +35,6 @@ add_definitions(-DBUILD_TS="${TS}")
 ### Check for Git and aquire version info
 find_program(FOUND_GIT "git")
 if (FOUND_GIT STREQUAL "FOUND_GIT-NOTFOUND")
-	set(GIT_BRANCH "null")
-	set(GIT_COMMIT_HASH "000000")
-	set(GIT_TAG "v0.0.0")
-	message(WARNING "Git not found, using placeholder version: ${GIT_TAG} (${GIT_BRANCH}-${GIT_COMMIT_HASH})")
 else()
     execute_process(COMMAND git rev-parse --abbrev-ref HEAD WORKING_DIRECTORY ${CMAKE_SOURCE_DIR} OUTPUT_VARIABLE GIT_BRANCH OUTPUT_STRIP_TRAILING_WHITESPACE)
 	execute_process(COMMAND git rev-parse --short HEAD WORKING_DIRECTORY ${CMAKE_SOURCE_DIR} OUTPUT_VARIABLE GIT_COMMIT_HASH OUTPUT_STRIP_TRAILING_WHITESPACE)
