diff --git a/Alice/Core/Check.hs b/Alice/Core/Check.hs
index 0700fa0388f..69815864710 100644
--- a/Alice/Core/Check.hs
+++ b/Alice/Core/Check.hs
@@ -18,8 +18,12 @@
  -  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  -}
 
+{-# LANGUAGE NoImplicitPrelude #-}
+
 module Alice.Core.Check (fillDef) where
 
+import Prelude hiding ((<>))
+
 import Control.Monad
 import Data.Maybe
 
diff --git a/Alice/Core/Reason.hs b/Alice/Core/Reason.hs
index c361bcf220d..4e493d8c91b 100644
--- a/Alice/Core/Reason.hs
+++ b/Alice/Core/Reason.hs
@@ -17,9 +17,12 @@
  -  You should have received a copy of the GNU General Public License
  -  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  -}
+{-# LANGUAGE NoImplicitPrelude #-}
 
 module Alice.Core.Reason where
 
+import Prelude hiding ((<>))
+
 import Control.Monad
 
 import Alice.Core.Base
diff --git a/Alice/Core/Verify.hs b/Alice/Core/Verify.hs
index 4f8550bdf11..0f59d135b16 100644
--- a/Alice/Core/Verify.hs
+++ b/Alice/Core/Verify.hs
@@ -18,8 +18,12 @@
  -  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  -}
 
+{-# LANGUAGE NoImplicitPrelude #-}
+
 module Alice.Core.Verify (verify) where
 
+import Prelude hiding ((<>))
+
 import Control.Monad
 import Data.IORef
 import Data.Maybe
