diff --git a/tests/test_parse.py b/tests/test_parse.py
index 84e17ae..e1eb6ae 100644
--- a/tests/test_parse.py
+++ b/tests/test_parse.py
@@ -242,6 +242,7 @@ def test_reference_documents_from_filenames():
                         cell_unit=unit,
                         add_fontkey=True,
                         add_fontname=False,
+                        add_text_direction=False,
                     )
                     _fname = fname + f".{unit}.txt"
                     with open(_fname, "w") as fw:
@@ -254,6 +255,7 @@ def test_reference_documents_from_filenames():
                         cell_unit=unit,
                         add_fontkey=True,
                         add_fontname=False,
+                        add_text_direction=False,
                     )
 
                     _fname = fname + f".{unit}.txt"
