diff --git a/ext/sass/Rakefile b/ext/sass/Rakefile
index 4ca11d4f82ea..c0450ad6f8f3 100644
--- a/ext/sass/Rakefile
+++ b/ext/sass/Rakefile
@@ -18,15 +18,7 @@ file 'protoc.exe' do |t|
 end
 
 file 'dart-sass' do |t|
-  raise if ENV.key?('DART_SASS')
-
-  gem_install 'sass-embedded', SassConfig.gem_version, SassConfig.gem_platform do |dir|
-    cp_r File.absolute_path("ext/sass/#{t.name}", dir), t.name
-  end
-rescue StandardError
-  archive = fetch(ENV.fetch('DART_SASS') { SassConfig.default_dart_sass })
-  unarchive archive
-  rm archive
+  symlink(ENV.fetch('DART_SASS'), t.name)
 end
 
 file 'cli.rb' => %w[dart-sass] do |t|
