Index: kioslave/network/network/builder/upnp/upnpnetworkbuilder.cpp
===================================================================
--- a/kioslave/network/network/builder/upnp/upnpnetworkbuilder.cpp	(revision 1160390)
+++ b/kioslave/network/network/builder/upnp/upnpnetworkbuilder.cpp	(revision 1160391)
@@ -36,6 +36,7 @@
 #include <QtDBus/QDBusConnection>
 #include <QtDBus/QDBusInterface>
 #include <QtDBus/QDBusPendingCallWatcher>
+#include <QtCore/QTimer>
 #include <QtCore/QStringList>
 
 #include <KDebug>
@@ -60,6 +61,11 @@
 
 void UpnpNetworkBuilder::start()
 {
+    QTimer::singleShot(0, this, SLOT(startBrowse()));
+}
+
+void UpnpNetworkBuilder::startBrowse()
+{
     qDBusRegisterMetaType<DeviceTypeMap>();
     qDBusRegisterMetaType<Cagibi::Device>();
 
Index: kioslave/network/network/builder/upnp/upnpnetworkbuilder.h
===================================================================
--- a/kioslave/network/network/builder/upnp/upnpnetworkbuilder.h	(revision 1160390)
+++ b/kioslave/network/network/builder/upnp/upnpnetworkbuilder.h	(revision 1160391)
@@ -63,6 +63,8 @@
     void removeUPnPDevices( const QList<Cagibi::Device>& devices );
 
   private Q_SLOTS:
+    void startBrowse();
+
     void onDevicesAdded( const DeviceTypeMap& deviceTypeMap );
     void onDevicesRemoved( const DeviceTypeMap& deviceTypeMap );
     void onAddedDeviceDetails( const Cagibi::Device& device );
