diff -rc "Structure Synth Source Code/SyntopiaCore/GLEngine/Raytracer/RayTracer.cpp" "Structure Synth Source Code2/SyntopiaCore/GLEngine/Raytracer/RayTracer.cpp"
*** "Structure Synth Source Code/SyntopiaCore/GLEngine/Raytracer/RayTracer.cpp"	2010-11-13 22:32:44.000000000 -0500
--- "Structure Synth Source Code2/SyntopiaCore/GLEngine/Raytracer/RayTracer.cpp"	2018-06-24 14:23:30.794296776 -0400
***************
*** 1,5 ****
  #include <QThread>
! 
  
  #include "RayTracer.h"
  
--- 1,5 ----
  #include <QThread>
! #include <GL/glu.h>
  
  #include "RayTracer.h"
  
diff -rc "Structure Synth Source Code/SyntopiaCore/GLEngine/Raytracer/VoxelStepper.cpp" "Structure Synth Source Code2/SyntopiaCore/GLEngine/Raytracer/VoxelStepper.cpp"
*** "Structure Synth Source Code/SyntopiaCore/GLEngine/Raytracer/VoxelStepper.cpp"	2010-09-08 21:25:30.000000000 -0400
--- "Structure Synth Source Code2/SyntopiaCore/GLEngine/Raytracer/VoxelStepper.cpp"	2018-06-24 14:23:12.542868194 -0400
***************
*** 122,128 ****
  						currentT = p;
  
  						// We do not intersect grid.
! 						if (!found) return false;
  				}
  
  				stepX = (dir.x() > 0) ? 1 : -1;
--- 122,128 ----
  						currentT = p;
  
  						// We do not intersect grid.
! 						if (!found) return NULL;
  				}
  
  				stepX = (dir.x() > 0) ? 1 : -1;
Only in Structure Synth Source Code2/SyntopiaCore/GLEngine/Raytracer: VoxelStepper.cpp.orig
diff -rc "Structure Synth Source Code/SyntopiaCore/GLEngine/Sphere.h" "Structure Synth Source Code2/SyntopiaCore/GLEngine/Sphere.h"
*** "Structure Synth Source Code/SyntopiaCore/GLEngine/Sphere.h"	2010-08-11 15:12:22.000000000 -0400
--- "Structure Synth Source Code2/SyntopiaCore/GLEngine/Sphere.h"	2018-06-24 14:23:30.793296807 -0400
***************
*** 2,7 ****
--- 2,8 ----
  
  #include "SyntopiaCore/Math/Vector3.h"
  #include "Object3D.h"
+ #include <GL/glu.h>
  
  namespace SyntopiaCore {
  	namespace GLEngine {	
