diff --git a/utils/uriplaylistbin/tests/uriplaylistbin.rs b/utils/uriplaylistbin/tests/uriplaylistbin.rs
index dfd1c9ce..8ed24949 100644
--- a/utils/uriplaylistbin/tests/uriplaylistbin.rs
+++ b/utils/uriplaylistbin/tests/uriplaylistbin.rs
@@ -534,6 +534,7 @@ fn infinite_to_finite() {
     assert_eq!(current_uri_index, 0);
 }
 
+#[ignore = "Requires network access"]
 #[test]
 /// cache HTTP playlist items
 fn cache() {
