diff --git a/utils/uriplaylistbin/tests/uriplaylistbin.rs b/utils/uriplaylistbin/tests/uriplaylistbin.rs
index 3489eaa8..569635d6 100644
--- a/utils/uriplaylistbin/tests/uriplaylistbin.rs
+++ b/utils/uriplaylistbin/tests/uriplaylistbin.rs
@@ -388,6 +388,7 @@ fn multi_audio() {
     assert_eq!(current_uri_index, 2);
 }
 
+#[ignore = "Unknown failure"]
 #[test]
 fn multi_audio_video() {
     let (_events, current_iteration, current_uri_index, eos) = test(
@@ -403,6 +404,7 @@ fn multi_audio_video() {
     assert_eq!(current_uri_index, 1);
 }
 
+#[ignore = "Unknown failure"]
 #[test]
 fn iterations() {
     let (_events, current_iteration, current_uri_index, eos) = test(
