From ac4b2a65f997f7f779b9c63dbe683ba10a26fc7f Mon Sep 17 00:00:00 2001
From: Morgan Jones <me@numin.it>
Date: Fri, 22 May 2026 23:58:29 -0700
Subject: [PATCH] genexe: take PONY_LINKER into account

---
 src/libponyc/codegen/genexe.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/libponyc/codegen/genexe.cc b/src/libponyc/codegen/genexe.cc
index 48f97578..5b9f36f1 100644
--- a/src/libponyc/codegen/genexe.cc
+++ b/src/libponyc/codegen/genexe.cc
@@ -1448,6 +1448,9 @@ static bool link_exe(compile_t* c, ast_t* program,
 {
   errors_t* errors = c->opt->check.errors;
 
+  if(c->opt->linker == NULL)
+    c->opt->linker = getenv("PONY_LINKER");
+
   // Use embedded LLD for Linux, macOS, and Windows targets unless --linker
   // escape hatch is specified. Sanitizer builds fall back to the system
   // compiler driver for native compilation since sanitizer runtime libraries
-- 
2.53.0

