diff --git a/openspace.cfg b/openspace.cfg
index c86830b..e7f89d9 100755
--- a/openspace.cfg
+++ b/openspace.cfg
@@ -2,18 +2,21 @@
 -- require('scripts/configuration_helper.lua')
 -- which defines helper functions useful to customize the configuration
 
+userdir = os.getenv("HOME") .. "/.openspace/"
+os.execute("mkdir -p " .. userdir)
+
 return {
     -- Determines which SGCT configuration file is loaded, that is, if there rendering
     -- occurs in a single window, a fisheye projection, or a dome cluster system
 
     -- A regular 1280x720 window
-    SGCTConfig = sgct.config.single{},
+    -- SGCTConfig = sgct.config.single{},
 
     -- A regular 1920x1080 window
     -- SGCTConfig = sgct.config.single{1920, 1080},
     
     -- A windowed 1920x1080 fullscreen
-    -- SGCTConfig = sgct.config.single{1920, 1080, border=false, windowPos={0,0}, shared=true, name="WV_OBS_SPOUT1"},
+    SGCTConfig = sgct.config.single{1920, 1080, border=false, windowPos={0,0}, shared=true, name="WV_OBS_SPOUT1"},
 
     -- A 1k fisheye rendering
     -- SGCTConfig = sgct.config.fisheye{1024, 1024},
@@ -53,15 +56,15 @@ return {
         TASKS = "${DATA}/tasks",
         WEB = "${DATA}/web",
 
-        CACHE = "${BASE}/cache",
+        CACHE  = userdir .. "cache",
         CONFIG = "${BASE}/config",
-        DOCUMENTATION = "${BASE}/documentation",
-        LOGS = "${BASE}/logs",
+        DOCUMENTATION = userdir .. "documentation",
+        LOGS    = userdir .. "logs",
         MODULES = "${BASE}/modules",
         SCRIPTS = "${BASE}/scripts",
         SHADERS = "${BASE}/shaders",
-        SYNC = "${BASE}/sync",
-        TESTDIR = "${BASE}/tests"
+        SYNC    = userdir .. "sync",
+        TESTDIR = userdir .. "tests"
     },
     Fonts = {
         Mono = "${FONTS}/Bitstream-Vera-Sans-Mono/VeraMono.ttf",
