diff --git a/ext/sgct/src/deps/vrpn/vrpn_Connection.C b/ext/sgct/src/deps/vrpn/vrpn_Connection.C
index d6ffdc5..f90a2b2 100755
--- a/ext/sgct/src/deps/vrpn/vrpn_Connection.C
+++ b/ext/sgct/src/deps/vrpn/vrpn_Connection.C
@@ -2489,7 +2489,7 @@ static int vrpn_start_server(const char *machine, char *server_name, char *args,
 #if defined(sparc) || defined(FreeBSD) || defined(_AIX) || defined(__ANDROID__)
             int status; // doesn't exist on sparc_solaris or FreeBSD
 #else
-            union wait status;
+            int status;
 #endif
 
             /* Check to see if they called back yet. */
