diff -ruN eject-2.1.0/Makefile.in eject-2.1.0.new/Makefile.in
--- eject-2.1.0/Makefile.in	2002-12-09 23:38:08.000000000 +0100
+++ eject-2.1.0.new/Makefile.in	2005-09-01 18:36:41.000000000 +0200
@@ -67,7 +67,7 @@
 eject_SOURCES = eject.c
 volname_SOURCES = volname.c
 
-SUBDIRS = po
+#SUBDIRS = po
 
 AM_CPPFLAGS = -I/usr/src/linux -I/usr/src/linux/include -DDEFAULTDEVICE=\"$(DEFAULTDEVICE)\" -DGETOPTLONG
 AM_CFLAGS = -Wall -O3 $(RPM_OPT_FLAGS)
diff -ruN eject-2.1.0/po/Makefile eject-2.1.0.new/po/Makefile
--- eject-2.1.0/po/Makefile	2005-05-01 20:41:17.000000000 +0200
+++ eject-2.1.0.new/po/Makefile	2005-09-01 18:27:43.000000000 +0200
@@ -1,5 +1,5 @@
 PKGNAME=eject
-PREFIX=$(DESTDIR)/usr
+PREFIX=$(prefix)/usr
 INSTALL = install
 MODIR = $(PREFIX)/share/locale
 POTFILES = ../*.c ../*.h
