--- gdm-3.16.0/daemon/gdm-display.c.orig	2015-04-08 13:53:14.370274369 +0200
+++ gdm-3.16.0/daemon/gdm-display.c	2015-04-08 13:53:36.287520435 +0200
@@ -1706,9 +1706,10 @@
 
                 gdm_error_trap_push ();
 
-                for (i = 0; i < G_N_ELEMENTS (host_entries); i++) {
+                /*for (i = 0; i < G_N_ELEMENTS (host_entries); i++) {
                         XAddHost (self->priv->x11_display, &host_entries[i]);
-                }
+                }*/
+                XDisableAccessControl(self->priv->x11_display);
 
                 XSync (self->priv->x11_display, False);
                 if (gdm_error_trap_pop ()) {
