diff --git a/src/Microsoft.VisualStudio.Services.Agent/Constants.cs b/src/Microsoft.VisualStudio.Services.Agent/Constants.cs
index ce764ed8..b07eae86 100644
--- a/src/Microsoft.VisualStudio.Services.Agent/Constants.cs
+++ b/src/Microsoft.VisualStudio.Services.Agent/Constants.cs
@@ -10,6 +10,7 @@ namespace Microsoft.VisualStudio.Services.Agent
     public enum WellKnownDirectory
     {
         Bin,
+        Diag,
         Externals,
         LegacyPSHost,
         Root,
@@ -269,6 +270,7 @@ namespace Microsoft.VisualStudio.Services.Agent
             {
                 public static readonly string ArtifactsDirectory = "a";
                 public static readonly string BinariesDirectory = "b";
+                public static readonly string DiagDirectory = "_diag";
                 public static readonly string GarbageCollectionDirectory = "GC";
                 public static readonly string LegacyArtifactsDirectory = "artifacts";
                 public static readonly string LegacyStagingDirectory = "staging";
diff --git a/src/Microsoft.VisualStudio.Services.Agent/HostContext.cs b/src/Microsoft.VisualStudio.Services.Agent/HostContext.cs
index a3d59b2a..2dbb5333 100644
--- a/src/Microsoft.VisualStudio.Services.Agent/HostContext.cs
+++ b/src/Microsoft.VisualStudio.Services.Agent/HostContext.cs
@@ -133,7 +133,9 @@ namespace Microsoft.VisualStudio.Services.Agent
                 }

                 // this should give us _diag folder under agent root directory as default value for diagLogDirctory
-                string diagLogPath = GetDiagDirectory(_hostType);
+                #pragma warning disable CA2214
+                string diagLogPath = GetDirectory(WellKnownDirectory.Diag);
+                #pragma warning restore CA2214
                 _traceManager = new TraceManager(new HostTraceListener(diagLogPath, hostType.ToString(), logPageSize, logRetentionDays), this.SecretMasker, this, hostType);
                 // Make the trace manager available via the service locator.
                 _serviceInstances.TryAdd(typeof(ITraceManager), _traceManager);
@@ -155,7 +157,7 @@ namespace Microsoft.VisualStudio.Services.Agent
             // (RuntimeKnobSource not available during HostContext initialization)
             string httpTraceEnvVar = Environment.GetEnvironmentVariable("VSTS_AGENT_HTTPTRACE");
             _trace.Info($"HTTP Trace Environment Variable: '{httpTraceEnvVar}'");
-
+
             if (!string.IsNullOrEmpty(httpTraceEnvVar) && StringUtil.ConvertToBoolean(httpTraceEnvVar))
             {
                 _trace.Info("Enabling HTTP trace via environment variable");
@@ -231,6 +233,12 @@ namespace Microsoft.VisualStudio.Services.Agent
                     path = Path.GetDirectoryName(Assembly.GetEntryAssembly().Location);
                     break;

+                case WellKnownDirectory.Diag:
+                    path = Path.Combine(
+                        GetDirectory(WellKnownDirectory.Root),
+                        Constants.Path.DiagDirectory);
+                    break;
+
                 case WellKnownDirectory.Externals:
                     path = Path.Combine(
                         GetDirectory(WellKnownDirectory.Root),
@@ -337,7 +345,11 @@ namespace Microsoft.VisualStudio.Services.Agent
                     throw new NotSupportedException($"Unexpected well known directory: '{directory}'");
             }

-            _trace.Info($"Well known directory '{directory}': '{path}'");
+            // only log if we've already set up _trace
+            if (_trace != null)
+            {
+                _trace.Info($"Well known directory '{directory}': '{path}'");
+            }
             return path;
         }

@@ -569,7 +581,7 @@ namespace Microsoft.VisualStudio.Services.Agent
             AgentShutdownReason = reason;
             _agentShutdownTokenSource.Cancel();
         }
-
+
         public void ShutdownWorkerForTimeout()
         {
             _trace.Info($"Worker will be shutdown");
@@ -662,7 +674,7 @@ namespace Microsoft.VisualStudio.Services.Agent

             _httpTrace = GetTrace("HttpTrace");
             _diagListenerSubscription = DiagnosticListener.AllListeners.Subscribe(this);
-
+
             _trace.Info("HTTP trace enabled dynamically via pipeline variable");
         }

