diff --git i/beetsplug/replaygain.py w/beetsplug/replaygain.py
index b6297d93..5c1cbbc0 100644
--- i/beetsplug/replaygain.py
+++ w/beetsplug/replaygain.py
@@ -139,7 +139,7 @@ class FfmpegBackend(Backend):
 
     def __init__(self, config, log):
         super().__init__(config, log)
-        self._ffmpeg_path = "ffmpeg"
+        self._ffmpeg_path = "@ffmpeg@/bin/ffmpeg"
 
         # check that ffmpeg is installed
         try:
@@ -975,11 +975,10 @@ class ReplayGainPlugin(BeetsPlugin):
     def __init__(self):
         super().__init__()
 
-        # default backend is 'command' for backward-compatibility.
         self.config.add({
             'overwrite': False,
             'auto': True,
-            'backend': 'command',
+            'backend': 'ffmpeg',
             'threads': cpu_count(),
             'parallel_on_import': False,
             'per_disc': False,
