diff --git i/beetsplug/replaygain.py w/beetsplug/replaygain.py
index 9d6fa23c..c5800039 100644
--- i/beetsplug/replaygain.py
+++ w/beetsplug/replaygain.py
@@ -391,7 +391,7 @@ class FfmpegBackend(Backend):
 
     def __init__(self, config, log):
         super(FfmpegBackend, self).__init__(config, log)
-        self._ffmpeg_path = "ffmpeg"
+        self._ffmpeg_path = "@ffmpeg@/bin/ffmpeg"
 
         # check that ffmpeg is installed
         try:
@@ -1228,11 +1228,10 @@ class ReplayGainPlugin(BeetsPlugin):
     def __init__(self):
         super(ReplayGainPlugin, self).__init__()
 
-        # default backend is 'command' for backward-compatibility.
         self.config.add({
             'overwrite': False,
             'auto': True,
-            'backend': u'command',
+            'backend': u'ffmpeg',
             'threads': cpu_count(),
             'parallel_on_import': False,
             'per_disc': False,
