diff --git a/configure.ac b/configure.ac
index b087801..c0401f4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1440,6 +1440,7 @@ AC_ARG_WITH([suffix],
     [Emscripten/browser*], [EXEEXT=.js],
     [Emscripten/node*], [EXEEXT=.js],
     [WASI/*], [EXEEXT=.wasm],
+    [MINGW*], [EXEEXT=.exe],
     [EXEEXT=]
   )
 ])
@@ -1466,6 +1467,10 @@ else
 fi
 rmdir CaseSensitiveTestDir
 
+AS_CASE([$ac_sys_system],
+  [MINGW], [BUILDEXEEXT=".exe"]
+)
+
 case $ac_sys_system in
 hp*|HP*)
     case $CC in
