--- a/configure.ac
+++ b/configure.ac
@@ -922,6 +922,14 @@
             [Define to include mbstate_t for mbrtowc])
 fi
 
+AC_MSG_CHECKING([for init system calls])
+AC_SUBST([INITSYS])
+case $host in
+  *-*-mingw*)	INITSYS=nt;;
+  *)		INITSYS=posix;;
+esac
+AC_MSG_RESULT([$INITSYS])
+
 # Record the configure-time value of MACOSX_DEPLOYMENT_TARGET,
 # it may influence the way we can build extensions, so distutils
 # needs to check it
--- a/Modules/Setup.bootstrap.in
+++ b/Modules/Setup.bootstrap.in
@@ -8,7 +8,7 @@
 # module C APIs are used in core
 atexit atexitmodule.c
 faulthandler faulthandler.c
-posix posixmodule.c
+@INITSYS@ posixmodule.c
 _signal signalmodule.c -lws2_32
 _tracemalloc _tracemalloc.c
 _suggestions _suggestions.c
@@ -17,7 +17,7 @@
 _codecs _codecsmodule.c
 _collections _collectionsmodule.c
 errno errnomodule.c
-_io _io/_iomodule.c _io/iobase.c _io/fileio.c _io/bytesio.c _io/bufferedio.c _io/textio.c _io/stringio.c
+_io _io/_iomodule.c _io/iobase.c _io/fileio.c _io/bytesio.c _io/bufferedio.c _io/textio.c _io/stringio.c _io/winconsoleio.c
 itertools itertoolsmodule.c
 _sre _sre/sre.c
 _sysconfig _sysconfig.c
