diff --git a/gui/vite.config.ts b/gui/vite.config.ts
index c9eb27be..4b3757f0 100644
--- a/gui/vite.config.ts
+++ b/gui/vite.config.ts
@@ -4,13 +4,10 @@ import { execSync } from 'child_process';
 import path from 'path';
 import { visualizer } from 'rollup-plugin-visualizer';
 
-const commitHash = execSync('git rev-parse --verify --short HEAD').toString().trim();
-const versionTag = execSync('git --no-pager tag --sort -taggerdate --points-at HEAD')
-  .toString()
-  .split('\n')[0]
-  .trim();
+const commitHash = "";
+const versionTag = "@version@";
 // If not empty then it's not clean
-const gitClean = execSync('git status --porcelain').toString() ? false : true;
+const gitClean = true;
 
 console.log(`version is ${versionTag || commitHash}${gitClean ? '' : '-dirty'}`);
 
