diff --git a/gui/public/i18n/en/translation.ftl b/gui/public/i18n/en/translation.ftl
index fd804a0e..35762e45 100644
--- a/gui/public/i18n/en/translation.ftl
+++ b/gui/public/i18n/en/translation.ftl
@@ -1034,7 +1034,7 @@ onboarding-reset_tutorial-2 = Tap the highlighted tracker { $taps } times to tri
 
 ## Install info
 install-info_udev-rules_modal_title = Hardware udev access rules not found
-install-info_udev-rules_warning = Access rules via udev are required for serial console access & dongle connection. Paste the following command into your terminal to add the udev rules.
+install-info_udev-rules_warning = Access rules via udev are required for serial console access & dongle connection. Add the following to your NixOS config to install the udev rules.
 install-info_udev-rules_modal_button = Close
 install-info_udev-rules_modal-dont-show-again_checkbox = Don't show again
 ## Setup start
diff --git a/gui/src/components/onboarding/UdevRulesModal.tsx b/gui/src/components/onboarding/UdevRulesModal.tsx
index b89af1e7..00955544 100644
--- a/gui/src/components/onboarding/UdevRulesModal.tsx
+++ b/gui/src/components/onboarding/UdevRulesModal.tsx
@@ -25,7 +25,7 @@ export function UdevRulesModal() {
       const dir = await electron.api.getInstallDir();
       const rulesPath = `${dir}/69-slimevr-devices.rules`;
       setUdevContent(
-        `cat ${rulesPath} | sudo sh -c 'tee /etc/udev/rules.d/69-slimevr-devices.rules >/dev/null && udevadm control --reload-rules && udevadm trigger'`
+        `{\n  services.udev.packages = [ pkgs.slimevr ];\n}`
       );
     }
   };
