diff --git i/tests/test_asyncio.py w/tests/test_asyncio.py
index 39d9e1a..05b7e6f 100644
--- i/tests/test_asyncio.py
+++ w/tests/test_asyncio.py
@@ -69,7 +69,7 @@ class AsyncioTests(test_utils.TestCase):
             def step_future():
                 future = asyncio.Future()
                 self.loop.call_soon(future.set_result, "asyncio.Future")
-                return (yield from future)
+                return (yield From(future))
 
             # test in release mode
             trollius.coroutines._DEBUG = False
