--- rstudio-1.1.135-old/src/cpp/core/r_util/REnvironmentPosix.cpp	2017-04-04 09:18:42.460994359 +0000
+++ rstudio-1.1.135-new/src/cpp/core/r_util/REnvironmentPosix.cpp	2017-04-04 09:21:29.254012434 +0000
@@ -87,10 +87,7 @@
 {
    // define potential paths
    std::vector<std::string> rScriptPaths;
-   rScriptPaths.push_back("/usr/bin/R");
-   rScriptPaths.push_back("/usr/local/bin/R");
-   rScriptPaths.push_back("/opt/local/bin/R");
-   rScriptPaths.push_back("/Library/Frameworks/R.framework/Resources/bin/R");
+   rScriptPaths.push_back("@R@/bin/R");
    return scanForRScript(rScriptPaths, pErrMsg);
 }
 
