diff --git a/nipype/info.py b/nipype/info.py
index c6503ac..4379643 100644
--- a/nipype/info.py
+++ b/nipype/info.py
@@ -147,9 +147,6 @@ REQUIRES = [
     'neurdflib',
     'click>=%s' % CLICK_MIN_VERSION,
     'funcsigs',
-    'pytest>=%s' % PYTEST_MIN_VERSION,
-    'pytest-xdist',
-    'mock',
     'pydotplus',
     'pydot>=%s' % PYDOT_MIN_VERSION,
     'packaging',
@@ -159,7 +156,15 @@ REQUIRES = [
 if sys.version_info <= (3, 4):
     REQUIRES.append('configparser')
 
-TESTS_REQUIRES = ['pytest-cov', 'codecov', 'pytest-env', 'coverage<5']
+TESTS_REQUIRES = [
+    'pytest>=%s' % PYTEST_MIN_VERSION,
+    'pytest-xdist',
+    'mock',
+    'pytest-cov',
+    'codecov',
+    'pytest-env',
+    'coverage<5'
+]
 
 EXTRA_REQUIRES = {
     'doc': ['Sphinx>=1.4', 'numpydoc', 'matplotlib', 'pydotplus', 'pydot>=1.2.3'],
