diff --git a/src/ceph-disk/ceph_disk/main.py b/src/ceph-disk/ceph_disk/main.py
index f8ef35503e..2cc8a75d2b 100644
--- a/src/ceph-disk/ceph_disk/main.py
+++ b/src/ceph-disk/ceph_disk/main.py
@@ -881,13 +881,14 @@ def is_partition(dev):
     if not stmode_is_diskdevice(st.st_mode):
         raise Error('not a block device', dev)
 
+    major = os.major(st.st_rdev)
+    minor = os.minor(st.st_rdev)
+
     name = get_dev_name(dev)
-    if os.path.exists(os.path.join(BLOCKDIR, name)):
+    if os.path.exists(os.path.join(BLOCKDIR, name)) or os.path.exists(os.path.join(BLOCKDIR, "loop%d" % minor)):
         return False
 
     # make sure it is a partition of something else
-    major = os.major(st.st_rdev)
-    minor = os.minor(st.st_rdev)
     if os.path.exists('/sys/dev/block/%d:%d/partition' % (major, minor)):
         return True
 
diff --git a/src/ceph-volume/ceph_volume/util/disk.py b/src/ceph-volume/ceph_volume/util/disk.py
index c3a7915400..9a708b05f2 100644
--- a/src/ceph-volume/ceph_volume/util/disk.py
+++ b/src/ceph-volume/ceph_volume/util/disk.py
@@ -203,7 +203,7 @@ def is_device(dev):
     # use lsblk first, fall back to using stat
     TYPE = lsblk(dev).get('TYPE')
     if TYPE:
-        return TYPE == 'disk'
+        return TYPE == 'disk' or TYPE == 'loop'
 
     # fallback to stat
     return _stat_is_device(os.lstat(dev).st_mode)
