diff --git a/session-helper/flatpak-session-helper.c b/session-helper/flatpak-session-helper.c
index 5dd7629e..ddc71a4c 100644
--- a/session-helper/flatpak-session-helper.c
+++ b/session-helper/flatpak-session-helper.c
@@ -693,7 +693,7 @@ start_p11_kit_server (const char *flatpak_dir)
   g_auto(GStrv) stdout_lines = NULL;
   int i;
   char *p11_argv[] = {
-    "p11-kit", "server",
+    "@p11kit@", "server",
     /* We explicitly request --sh here, because we then fail on earlier versions that doesn't support
      * this flag. This is good, because those earlier versions did not properly daemonize and caused
      * the spawn_sync to hang forever, waiting for the pipe to close.
@@ -836,7 +836,7 @@ main (int    argc,
       exit (1);
     }
 
-  if (g_find_program_in_path ("p11-kit"))
+  if (TRUE)
     start_p11_kit_server (flatpak_dir);
   else
     g_debug ("p11-kit not found");
