diff --git a/common/flatpak-run.c b/common/flatpak-run.c
index 146c4f87..bcdad2bc 100644
--- a/common/flatpak-run.c
+++ b/common/flatpak-run.c
@@ -1710,6 +1710,7 @@ static const ExportData default_exports[] = {
   {"GST_PTP_HELPER", NULL},
   {"GST_PTP_HELPER_1_0", NULL},
   {"GST_INSTALL_PLUGINS_HELPER", NULL},
+  {"GDK_PIXBUF_MODULE_FILE", NULL},
 };
 
 static const ExportData no_ld_so_cache_exports[] = {
