diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -52,7 +52,7 @@ clean:
 replaced/%: %
 	mkdir -p $(dir $@)
 	sed < $< > $@ \
-	-e 's:__SYSSHAREDIR_PREFIX__:$(PREFIX):' \
+	-e 's:__SYSSHAREDIR_PREFIX__:$(DESTDIR):' \
 	-e 's:__SYSCONFDIR_PREFIX__:$(ETCPREFIX):' \
 	-e 's:__SYSDATADIR_PREFIX__:$(LOCALSTATEDIR):'
 
diff --git a/src/share/checkperms b/src/share/checkperms
--- a/src/share/checkperms
+++ b/src/share/checkperms
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -T
+#!/usr/bin/perl
 
 # checkperms: ensure as best we can that a given file can only be
 # modified by the given user (or the superuser, naturally).  This
diff --git a/src/share/keytrans b/src/share/keytrans
--- a/src/share/keytrans
+++ b/src/share/keytrans
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -T
+#!/usr/bin/perl
 
 # keytrans: this is an RSA key translation utility; it is capable of
 # transforming RSA keys (both public keys and secret keys) between
@@ -1083,7 +1083,7 @@ sub packetwalk {
 
 
 for (basename($0)) {
-  if (/^pem2openpgp$/) {
+  if (/pem2openpgp/) {
     my $rsa;
     my $stdin;
 
@@ -1121,7 +1121,7 @@ for (basename($0)) {
 		      }
 		     );
   }
-  elsif (/^openpgp2ssh$/) {
+  elsif (/openpgp2ssh/) {
       my $fpr = shift;
       my $instream;
       open($instream,'-');
@@ -1137,7 +1137,7 @@ for (basename($0)) {
 	die "No matching key found.\n";
       }
   }
-  elsif (/^openpgp2pem$/) {
+  elsif (/openpgp2pem/) {
       my $fpr = shift;
       my $instream;
       open($instream,'-');
@@ -1153,7 +1153,7 @@ for (basename($0)) {
 	die "No matching key found.\n";
       }
   }
-  elsif (/^openpgp2spki$/) {
+  elsif (/openpgp2spki/) {
       my $fpr = shift;
       my $instream;
       open($instream,'-');
@@ -1165,7 +1165,7 @@ for (basename($0)) {
 	die "No matching key found.\n";
       }
   }
-  elsif (/^keytrans$/) {
+  elsif (/keytrans/) {
     # subcommands when keytrans is invoked directly are UNSUPPORTED,
     # UNDOCUMENTED, and WILL NOT BE MAINTAINED.
     my $subcommand = shift;
-- 
2.16.3
