--- code-r100/configure.ac	2014-09-25 23:44:41.059174904 +0200
+++ code-r100.new/configure.ac	2014-09-26 01:37:18.507017390 +0200
@@ -44,6 +44,8 @@
   [AC_MSG_ERROR(CONFIG: You need libxml2-dev installed.
 	        http://www.xmlsoft.org/)])
 
+PKG_CHECK_MODULES([libxml], [libxml-2.0])
+
 AC_CHECK_LIB([m], [sqrt], , [AC_MSG_ERROR(CORE: You need libm installed)])
 AC_CHECK_LIB([pthread], [pthread_self], , [AC_MSG_ERROR(CORE: You need libpthread installed)])
 
@@ -79,12 +81,12 @@
    echo "--- Enabling USB LCD display --";
 fi
 
-#SDL_VERSION=1.2.4
-#AM_PATH_SDL($SDL_VERSION,
-#            :,
-#	    AC_MSG_ERROR([VIDEO: *** SDL version $SDL_VERSION not found!
-#		          http://www.libsdl.org/])
-#)
+SDL_VERSION=1.2.4
+AM_PATH_SDL($SDL_VERSION,
+            :,
+	    AC_MSG_ERROR([VIDEO: *** SDL version $SDL_VERSION not found!
+		          http://www.libsdl.org/])
+)
 
 AC_CHECK_LIB([vorbis], [main], , 
   [AC_MSG_ERROR(AUDIO: You need libvorbis-dev installed.
@@ -95,6 +97,9 @@
 AC_CHECK_LIB([vorbisenc], [main], , 
   [AC_MSG_ERROR(AUDIO: You need libvorbis-dev installed.
 		http://www.xiph.org/ogg/vorbis/)])
+AC_CHECK_LIB([ogg], [main], , 
+  [AC_MSG_ERROR(AUDIO: You need libogg-dev installed.
+		http://www.xiph.org/ogg/)])
 
 AC_CHECK_LIB([sndfile], [main], , 
   [AC_MSG_ERROR(AUDIO: you need libsndfile installed.
