diff -Naur mono-5.0.0/mcs/tools/xbuild/data/3.5/Microsoft.Common.targets.old mono-5.0.0/mcs/tools/xbuild/data/3.5/Microsoft.Common.targets
--- mono-5.0.0/mcs/tools/xbuild/data/3.5/Microsoft.Common.targets.old	2017-04-24 23:45:18.348116305 +0200
+++ mono-5.0.0/mcs/tools/xbuild/data/3.5/Microsoft.Common.targets	2017-04-24 23:45:11.407051755 +0200
@@ -167,8 +167,8 @@
 			$(ReferencePath);
 			@(AdditionalReferencePath);
 			{HintPathFromItem};
-			{TargetFrameworkDirectory};
 			{PkgConfig};
+			{TargetFrameworkDirectory};
 			{GAC};
 			{RawFileName};
 			$(OutDir)
diff -Naur mono-5.0.0/mcs/tools/xbuild/data/4.0/Microsoft.Common.targets.old mono-5.0.0/mcs/tools/xbuild/data/4.0/Microsoft.Common.targets
--- mono-5.0.0/mcs/tools/xbuild/data/4.0/Microsoft.Common.targets.old	2017-04-24 23:49:53.019616196 +0200
+++ mono-5.0.0/mcs/tools/xbuild/data/4.0/Microsoft.Common.targets	2017-04-24 23:50:05.709729585 +0200
@@ -232,8 +232,8 @@
 			$(ReferencePath);
 			@(AdditionalReferencePath);
 			{HintPathFromItem};
-			{TargetFrameworkDirectory};
 			{PkgConfig};
+			{TargetFrameworkDirectory};
 			{GAC};
 			{RawFileName};
 			$(OutDir)
diff -Naur mono-5.0.0/mcs/tools/xbuild/data/2.0/Microsoft.Common.targets.old mono-5.0.0/mcs/tools/xbuild/data/2.0/Microsoft.Common.targets
--- mono-5.0.0/mcs/tools/xbuild/data/2.0/Microsoft.Common.targets.old	2017-04-24 23:52:33.200037047 +0200
+++ mono-5.0.0/mcs/tools/xbuild/data/2.0/Microsoft.Common.targets	2017-04-24 23:52:43.281125802 +0200
@@ -139,8 +139,8 @@
 			$(ReferencePath);
 			@(AdditionalReferencePath);
 			{HintPathFromItem};
-			{TargetFrameworkDirectory};
 			{PkgConfig};
+			{TargetFrameworkDirectory};
 			{GAC};
 			{RawFileName};
 			$(OutDir)
diff -Naur mono-5.0.0/mcs/tools/xbuild/data/14.0/Microsoft.Common.targets.old mono-5.0.0/mcs/tools/xbuild/data/14.0/Microsoft.Common.targets
--- mono-5.0.0/mcs/tools/xbuild/data/14.0/Microsoft.Common.targets.old	2017-04-24 23:54:02.585821594 +0200
+++ mono-5.0.0/mcs/tools/xbuild/data/14.0/Microsoft.Common.targets	2017-04-24 23:54:09.313880438 +0200
@@ -234,8 +234,8 @@
 			$(ReferencePath);
 			@(AdditionalReferencePath);
 			{HintPathFromItem};
-			{TargetFrameworkDirectory};
 			{PkgConfig};
+			{TargetFrameworkDirectory};
 			{GAC};
 			{RawFileName};
 			$(OutDir)
diff -Naur mono-5.0.0/mcs/tools/xbuild/data/12.0/Microsoft.Common.targets.old mono-5.0.0/mcs/tools/xbuild/data/12.0/Microsoft.Common.targets
--- mono-5.0.0/mcs/tools/xbuild/data/12.0/Microsoft.Common.targets.old	2017-04-24 23:55:46.244895155 +0200
+++ mono-5.0.0/mcs/tools/xbuild/data/12.0/Microsoft.Common.targets	2017-04-24 23:55:51.998961342 +0200
@@ -232,8 +232,8 @@
 			$(ReferencePath);
 			@(AdditionalReferencePath);
 			{HintPathFromItem};
-			{TargetFrameworkDirectory};
 			{PkgConfig};
+			{TargetFrameworkDirectory};
 			{GAC};
 			{RawFileName};
 			$(OutDir)
