diff --git a/xpra/server/util.py b/xpra/server/util.py
index 401a9fb959..678e2ce745 100644
--- a/xpra/server/util.py
+++ b/xpra/server/util.py
@@ -175,6 +175,10 @@ def xpra_env_shell_script(socket_dir: str, env: dict[str, str]) -> str:
 
 
 def xpra_runner_shell_script(xpra_file: str, starting_dir: str) -> str:
+    # Nixpkgs contortion:
+    # xpra_file points to a shell wrapper, not to the python script.
+    dirname, basename = os.path.split(xpra_file)
+    xpra_file = os.path.join(dirname, "."+basename+"-wrapped")
     # We ignore failures in cd'ing, b/c it's entirely possible that we were
     # started from some temporary directory and all paths are absolute.
     qdir = sh_quotemeta(starting_dir)
