diff --git a/lib/xattr-at.c b/lib/xattr-at.c
--- a/lib/xattr-at.c
+++ b/lib/xattr-at.c
@@ -36,2 +36,2 @@
-/* setxattrat */
-#define AT_FUNC_NAME setxattrat
+/* tar_setxattrat */
+#define AT_FUNC_NAME tar_setxattrat
@@ -60,2 +60,2 @@
-/* getxattrat */
-#define AT_FUNC_NAME getxattrat
+/* tar_getxattrat */
+#define AT_FUNC_NAME tar_getxattrat
@@ -88,2 +88,2 @@
-/* listxattrat */
-#define AT_FUNC_NAME listxattrat
+/* tar_listxattrat */
+#define AT_FUNC_NAME tar_listxattrat
diff --git a/lib/xattr-at.h b/lib/xattr-at.h
--- a/lib/xattr-at.h
+++ b/lib/xattr-at.h
@@ -35,2 +35,2 @@
-   "at" suffix.  For example, setxattrat (AT_FDCWD, path, name, value, size,
-   flags &c) is usually equivalent to setxattr (file, name, value, size,
+   "at" suffix.  For example, tar_setxattrat (AT_FDCWD, path, name, value,
+   size, flags &c) is usually equivalent to setxattr (file, name, value, size,
@@ -44,2 +44,2 @@
-int setxattrat (int dir_fd, const char *path, const char *name,
-                const void *value, size_t size, int flags);
+int tar_setxattrat (int dir_fd, const char *path, const char *name,
+                    const void *value, size_t size, int flags);
@@ -47 +47 @@
-/* dir-fd-relative lsetxattr.  This function is just like setxattrat,
+/* dir-fd-relative lsetxattr.  This function is just like tar_setxattrat,
@@ -49 +49 @@
-   symlink, while the setxattrat operates on the referent of the symlink.  */
+   symlink, while tar_setxattrat operates on the referent of the symlink.  */
@@ -57,2 +57,2 @@
-ssize_t getxattrat (int dir_fd, const char *path, const char *name,
-                    void *value, size_t size);
+ssize_t tar_getxattrat (int dir_fd, const char *path, const char *name,
+                        void *value, size_t size);
@@ -60 +60 @@
-/* dir-fd-relative lgetxattr.  This function is just like getxattrat,
+/* dir-fd-relative lgetxattr.  This function is just like tar_getxattrat,
@@ -62 +62 @@
-   symlink, while the getxattrat operates on the referent of the symlink.  */
+   symlink, while tar_getxattrat operates on the referent of the symlink.  */
@@ -68 +68 @@
-ssize_t listxattrat (int dir_fd, const char *path, char *list, size_t size);
+ssize_t tar_listxattrat (int dir_fd, const char *path, char *list, size_t size);
@@ -70 +70 @@
-/* dir-fd-relative llistxattr.  This function is just like listxattrat,
+/* dir-fd-relative llistxattr.  This function is just like tar_listxattrat,
@@ -72 +72 @@
-   symlink, while the listxattrat operates on the referent of the symlink.  */
+   symlink, while tar_listxattrat operates on the referent of the symlink.  */
diff --git a/src/xattrs.c b/src/xattrs.c
index 86a7e59cbc..c872ae308d 100644
--- a/src/xattrs.c
+++ b/src/xattrs.c
@@ -139,13 +139,13 @@
 #ifdef HAVE_POSIX_ACLS
 
 /* acl-at wrappers, TODO: move to gnulib in future? */
-static acl_t acl_get_file_at (int, const char *, acl_type_t);
-static int acl_set_file_at (int, const char *, acl_type_t, acl_t);
+static acl_t tar_acl_get_file_at (int, const char *, acl_type_t);
+static int tar_acl_set_file_at (int, const char *, acl_type_t, acl_t);
 static int file_has_acl_at (int, char const *, struct stat const *);
-static int acl_delete_def_file_at (int, char const *);
+static int tar_acl_delete_def_file_at (int, char const *);
 
-/* acl_get_file_at */
-#define AT_FUNC_NAME acl_get_file_at
+/* tar_acl_get_file_at */
+#define AT_FUNC_NAME tar_acl_get_file_at
 #define AT_FUNC_RESULT acl_t
 #define AT_FUNC_FAIL (acl_t)NULL
 #define AT_FUNC_F1 acl_get_file
@@ -159,8 +159,8 @@
 #undef AT_FUNC_POST_FILE_PARAM_DECLS
 #undef AT_FUNC_POST_FILE_ARGS
 
-/* acl_set_file_at */
-#define AT_FUNC_NAME acl_set_file_at
+/* tar_acl_set_file_at */
+#define AT_FUNC_NAME tar_acl_set_file_at
 #define AT_FUNC_F1 acl_set_file
 #define AT_FUNC_POST_FILE_PARAM_DECLS   , acl_type_t type, acl_t acl
 #define AT_FUNC_POST_FILE_ARGS          , type, acl
@@ -170,8 +170,8 @@
 #undef AT_FUNC_POST_FILE_PARAM_DECLS
 #undef AT_FUNC_POST_FILE_ARGS
 
-/* acl_delete_def_file_at */
-#define AT_FUNC_NAME acl_delete_def_file_at
+/* tar_acl_delete_def_file_at */
+#define AT_FUNC_NAME tar_acl_delete_def_file_at
 #define AT_FUNC_F1 acl_delete_def_file
 #define AT_FUNC_POST_FILE_PARAM_DECLS
 #define AT_FUNC_POST_FILE_ARGS
@@ -299,10 +299,10 @@
       /* No "default" IEEE 1003.1e ACL set for directory.  At this moment,
          FILE_NAME may already have inherited default acls from parent
          directory;  clean them up. */
-      if (acl_delete_def_file_at (chdir_fd, file_name))
+      if (tar_acl_delete_def_file_at (chdir_fd, file_name))
         WARNOPT (WARN_XATTR_WRITE,
                 (0, errno,
-                 _("acl_delete_def_file_at: Cannot drop default POSIX ACLs "
+                 _("tar_acl_delete_def_file_at: Cannot drop default POSIX ACLs "
                    "for file '%s'"),
                  file_name));
       return;
@@ -316,11 +316,11 @@
       return;
     }
 
-  if (acl_set_file_at (chdir_fd, file_name, type, acl) == -1)
+  if (tar_acl_set_file_at (chdir_fd, file_name, type, acl) == -1)
     /* warn even if filesystem does not support acls */
     WARNOPT (WARN_XATTR_WRITE,
 	     (0, errno,
-	      _ ("acl_set_file_at: Cannot set POSIX ACLs for file '%s'"),
+	      _ ("tar_acl_set_file_at: Cannot set POSIX ACLs for file '%s'"),
 	      file_name));
 
   acl_free (acl);
@@ -360 +360 @@
-  if (!(acl = acl_get_file_at (parentfd, file_name, type)))
+  if (!(acl = tar_acl_get_file_at (parentfd, file_name, type)))
@@ -363 +363 @@
-        call_arg_warn ("acl_get_file_at", file_name);
+        call_arg_warn ("tar_acl_get_file_at", file_name);
@@ -624 +624 @@
-        ret = setxattrat (chdir_fd, file_name, attr, ptr, len, 0);
+        ret = tar_setxattrat (chdir_fd, file_name, attr, ptr, len, 0);
 
