From efc16fbe7e41784f218d9c6cb4239b209cd77214 Mon Sep 17 00:00:00 2001
From: Christian Kampka <christian@kampka.net>
Date: Sat, 12 Oct 2019 21:47:47 +0200
Subject: [PATCH 1/2] Fix path marshalling when saveing config

---
 config/config.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/config.go b/config/config.go
index 69a10c5..b5c61fe 100644
--- a/config/config.go
+++ b/config/config.go
@@ -26,7 +26,7 @@ func (p *Path) UnmarshalText(text []byte) error {
 	return nil
 }
 
-func (p *Path) MarshalText() (text []byte, err error) {
+func (p Path) MarshalText() (text []byte, err error) {
 	return []byte(p.path), nil
 }
 
-- 
2.19.2

