diff --git a/common/lib/modules/fglrx/build_mod/firegl_public.c b/common/lib/modules/fglrx/build_mod/firegl_public.c
index d3ad3ce..9362b58 100755
--- a/common/lib/modules/fglrx/build_mod/firegl_public.c
+++ b/common/lib/modules/fglrx/build_mod/firegl_public.c
@@ -34,6 +34,11 @@
 #include <linux/autoconf.h>
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,5,0)
+#include <linux/uidgid.h>
+#endif
+
+
 #if !defined(CONFIG_X86) 
 #if !defined(CONFIG_X86_PC) 
 #if !defined(CONFIG_X86_XEN) 
@@ -1543,9 +1548,17 @@ KCL_TYPE_Pid ATI_API_CALL KCL_GetTgid(void)
 KCL_TYPE_Uid ATI_API_CALL KCL_GetEffectiveUid(void)
 {
 #ifdef current_euid
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(3,5,0)
+    return __kuid_val(current_euid());
+# else
     return current_euid();
+# endif
 #else
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(3,5,0)
+    return __kuid_val(current->euid);
+# else
     return current->euid;
+# endif
 #endif
 }
