diff --git a/docs/manpages/meson.build b/docs/manpages/meson.build
index f90c757cf..40d9c939d 100644
--- a/docs/manpages/meson.build
+++ b/docs/manpages/meson.build
@@ -32,7 +32,7 @@ foreach m : manpages
     command: [xsltproc,
               '--output', '@OUTPUT@',
               '--stringparam', 'man.authors.section.enabled', '0',
-              '/etc/asciidoc/docbook-xsl/manpage.xsl', '@INPUT@'],
+              '@asciidoc@/etc/asciidoc/docbook-xsl/manpage.xsl', '@INPUT@'],
     input: xml,
     output: manpage,
     install: true,
