diff --git a/Telegram/SourceFiles/platform/mac/file_bookmark_mac.mm b/Telegram/SourceFiles/platform/mac/file_bookmark_mac.mm
index 9e9a1744b..ae55f873f 100644
--- a/Telegram/SourceFiles/platform/mac/file_bookmark_mac.mm
+++ b/Telegram/SourceFiles/platform/mac/file_bookmark_mac.mm
@@ -14,6 +14,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 
 #include <Cocoa/Cocoa.h>
 #include <CoreFoundation/CFURL.h>
+#undef check
 
 namespace Platform {
 namespace {
diff --git a/Telegram/SourceFiles/platform/mac/specific_mac.mm b/Telegram/SourceFiles/platform/mac/specific_mac.mm
index 1d68457bb..ac8c4e0ab 100644
--- a/Telegram/SourceFiles/platform/mac/specific_mac.mm
+++ b/Telegram/SourceFiles/platform/mac/specific_mac.mm
@@ -118,6 +118,7 @@ PermissionStatus GetPermissionStatus(PermissionType type) {
 	switch (type) {
 	case PermissionType::Microphone:
 	case PermissionType::Camera:
+#if 0
 		const auto nativeType = (type == PermissionType::Microphone)
 			? AVMediaTypeAudio
 			: AVMediaTypeVideo;
@@ -132,6 +133,7 @@ PermissionStatus GetPermissionStatus(PermissionType type) {
 					return PermissionStatus::Denied;
 			}
 		}
+#endif
 		break;
 	}
 	return PermissionStatus::Granted;
@@ -141,6 +143,7 @@ void RequestPermission(PermissionType type, Fn<void(PermissionStatus)> resultCal
 	switch (type) {
 	case PermissionType::Microphone:
 	case PermissionType::Camera:
+#if 0
 		const auto nativeType = (type == PermissionType::Microphone)
 			? AVMediaTypeAudio
 			: AVMediaTypeVideo;
@@ -151,6 +154,7 @@ void RequestPermission(PermissionType type, Fn<void(PermissionStatus)> resultCal
 				});
 			}];
 		}
+#endif
 		break;
 	}
 	resultCallback(PermissionStatus::Granted);
diff --git a/Telegram/SourceFiles/platform/mac/touchbar/items/mac_formatter_item.h b/Telegram/SourceFiles/platform/mac/touchbar/items/mac_formatter_item.h
index a537929c8..82ef2b837 100644
--- a/Telegram/SourceFiles/platform/mac/touchbar/items/mac_formatter_item.h
+++ b/Telegram/SourceFiles/platform/mac/touchbar/items/mac_formatter_item.h
@@ -9,8 +9,8 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 
 #import <AppKit/NSPopoverTouchBarItem.h>
 #import <AppKit/NSTouchBar.h>
+#undef check
 
-API_AVAILABLE(macos(10.12.2))
 @interface TextFormatPopover : NSPopoverTouchBarItem
 - (id)init:(NSTouchBarItemIdentifier)identifier;
 @end
diff --git a/Telegram/SourceFiles/platform/mac/touchbar/items/mac_pinned_chats_item.h b/Telegram/SourceFiles/platform/mac/touchbar/items/mac_pinned_chats_item.h
index c6a4b886f..d2e0936c0 100644
--- a/Telegram/SourceFiles/platform/mac/touchbar/items/mac_pinned_chats_item.h
+++ b/Telegram/SourceFiles/platform/mac/touchbar/items/mac_pinned_chats_item.h
@@ -8,12 +8,12 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #pragma once
 
 #include <AppKit/NSImageView.h>
+#undef check
 
 namespace Main {
 class Session;
 } // namespace Main
 
-API_AVAILABLE(macos(10.12.2))
 @interface PinnedDialogsPanel : NSImageView
 - (id)init:(not_null<Main::Session*>)session
 	destroyEvent:(rpl::producer<>)touchBarSwitches;
diff --git a/Telegram/SourceFiles/platform/mac/touchbar/items/mac_scrubber_item.h b/Telegram/SourceFiles/platform/mac/touchbar/items/mac_scrubber_item.h
index 27b04467c..b1a7dfbd9 100644
--- a/Telegram/SourceFiles/platform/mac/touchbar/items/mac_scrubber_item.h
+++ b/Telegram/SourceFiles/platform/mac/touchbar/items/mac_scrubber_item.h
@@ -9,12 +9,12 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 
 #import <AppKit/NSPopoverTouchBarItem.h>
 #import <AppKit/NSTouchBar.h>
+#undef check
 
 namespace Window {
 class Controller;
 } // namespace Window
 
-API_AVAILABLE(macos(10.12.2))
 @interface StickerEmojiPopover : NSPopoverTouchBarItem<NSTouchBarDelegate>
 - (id)init:(not_null<Window::Controller*>)controller
 	identifier:(NSTouchBarItemIdentifier)identifier;
diff --git a/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_audio.h b/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_audio.h
index ec4596c67..972461aef 100644
--- a/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_audio.h
+++ b/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_audio.h
@@ -8,8 +8,8 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #pragma once
 
 #import <AppKit/NSTouchBar.h>
+#undef check
 
-API_AVAILABLE(macos(10.12.2))
 @interface TouchBarAudioPlayer : NSTouchBar<NSTouchBarDelegate>
 - (rpl::producer<>)closeRequests;
 @end
diff --git a/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_common.h b/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_common.h
index 52b54de12..ac3857f9b 100644
--- a/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_common.h
+++ b/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_common.h
@@ -9,6 +9,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 
 #import <AppKit/NSImage.h>
 #import <Foundation/Foundation.h>
+#undef check
 
 namespace TouchBar {
 
diff --git a/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_controls.h b/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_controls.h
index 1cc8c832f..c2178c975 100644
--- a/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_controls.h
+++ b/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_controls.h
@@ -20,19 +20,19 @@ struct TrackState;
 
 namespace TouchBar {
 
-[[nodiscard]] API_AVAILABLE(macos(10.12.2))
+[[nodiscard]]
 NSButton *CreateTouchBarButton(
 	NSImage *image,
 	rpl::lifetime &lifetime,
 	Fn<void()> callback);
 
-[[nodiscard]] API_AVAILABLE(macos(10.12.2))
+[[nodiscard]]
 NSButton *CreateTouchBarButton(
 	const style::icon &icon,
 	rpl::lifetime &lifetime,
 	Fn<void()> callback);
 
-[[nodiscard]] API_AVAILABLE(macos(10.12.2))
+[[nodiscard]]
 NSButton *CreateTouchBarButtonWithTwoStates(
 	NSImage *icon1,
 	NSImage *icon2,
@@ -41,7 +41,7 @@ NSButton *CreateTouchBarButtonWithTwoStates(
 	bool firstState,
 	rpl::producer<bool> stateChanged = rpl::never<bool>());
 
-[[nodiscard]] API_AVAILABLE(macos(10.12.2))
+[[nodiscard]]
 NSButton *CreateTouchBarButtonWithTwoStates(
 	const style::icon &icon1,
 	const style::icon &icon2,
@@ -50,14 +50,14 @@ NSButton *CreateTouchBarButtonWithTwoStates(
 	bool firstState,
 	rpl::producer<bool> stateChanged = rpl::never<bool>());
 
-[[nodiscard]] API_AVAILABLE(macos(10.12.2))
+[[nodiscard]]
 NSSliderTouchBarItem *CreateTouchBarSlider(
 	NSString *itemId,
 	rpl::lifetime &lifetime,
 	Fn<void(bool, double, double)> callback,
 	rpl::producer<Media::Player::TrackState> stateChanged);
 
-[[nodiscard]] API_AVAILABLE(macos(10.12.2))
+[[nodiscard]]
 NSCustomTouchBarItem *CreateTouchBarTrackPosition(
 	NSString *itemId,
 	rpl::producer<Media::Player::TrackState> stateChanged);
diff --git a/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_main.h b/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_main.h
index f03546eaf..bc8c63678 100644
--- a/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_main.h
+++ b/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_main.h
@@ -8,6 +8,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #pragma once
 
 #import <AppKit/NSTouchBar.h>
+#undef check
 
 namespace Window {
 class Controller;
@@ -21,7 +22,6 @@ const auto kPopoverPickerItemIdentifier = @"pickerButtons";
 
 } // namespace TouchBar::Main
 
-API_AVAILABLE(macos(10.12.2))
 @interface TouchBarMain : NSTouchBar
 - (id)init:(not_null<Window::Controller*>)controller
 	touchBarSwitches:(rpl::producer<>)touchBarSwitches;
diff --git a/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_manager.h b/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_manager.h
index 464f87c9c..9a008c75e 100644
--- a/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_manager.h
+++ b/Telegram/SourceFiles/platform/mac/touchbar/mac_touchbar_manager.h
@@ -8,6 +8,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #pragma once
 
 #import <AppKit/NSTouchBar.h>
+#undef check
 
 namespace Main {
 class Domain;
@@ -17,7 +18,6 @@ namespace Window {
 class Controller;
 } // namespace Window
 
-API_AVAILABLE(macos(10.12.2))
 @interface RootTouchBar : NSTouchBar<NSTouchBarDelegate>
 - (id)init:(rpl::producer<bool>)canApplyMarkdown
 	controller:(not_null<Window::Controller*>)controller
Submodule Telegram/ThirdParty/tgcalls contains modified content
diff --git a/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/TGRTCDefaultVideoDecoderFactory.mm b/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/TGRTCDefaultVideoDecoderFactory.mm
index b280c1b..a1ed0d2 100644
--- a/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/TGRTCDefaultVideoDecoderFactory.mm
+++ b/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/TGRTCDefaultVideoDecoderFactory.mm
@@ -71,7 +71,7 @@
     if (@available(iOS 11.0, *)) {
         [result addObject:h265Info];
     }
-#else // WEBRTC_IOS
+#elif 0 // WEBRTC_IOS
     if (@available(macOS 10.13, *)) {
         [result addObject:h265Info];
     }
@@ -101,7 +101,7 @@
       return [[TGRTCVideoDecoderH265 alloc] init];
     }
   }
-#else // WEBRTC_IOS
+#elif 0 // WEBRTC_IOS
   if (@available(macOS 10.13, *)) {
     if ([info.name isEqualToString:kRTCVideoCodecH265Name]) {
       return [[TGRTCVideoDecoderH265 alloc] init];
diff --git a/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/TGRTCDefaultVideoEncoderFactory.mm b/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/TGRTCDefaultVideoEncoderFactory.mm
index 9960607..f3659b3 100644
--- a/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/TGRTCDefaultVideoEncoderFactory.mm
+++ b/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/TGRTCDefaultVideoEncoderFactory.mm
@@ -89,7 +89,7 @@
       [result addObject:h265Info];
     }
   }
-#else // WEBRTC_IOS
+#elif 0 // WEBRTC_IOS
   if (@available(macOS 10.13, *)) {
     if ([[AVAssetExportSession allExportPresets] containsObject:AVAssetExportPresetHEVCHighestQuality]) {
       [result addObject:h265Info];
@@ -129,7 +129,7 @@
       return [[TGRTCVideoEncoderH265 alloc] initWithCodecInfo:info];
     }
   }
-#else // WEBRTC_IOS
+#elif 0 // WEBRTC_IOS
   if (@available(macOS 10.13, *)) {
     if ([info.name isEqualToString:kRTCVideoCodecH265Name]) {
       return [[TGRTCVideoEncoderH265 alloc] initWithCodecInfo:info];
diff --git a/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/VideoCameraCapturerMac.mm b/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/VideoCameraCapturerMac.mm
index bf99063..b717645 100644
--- a/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/VideoCameraCapturerMac.mm
+++ b/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/VideoCameraCapturerMac.mm
@@ -507,8 +507,7 @@ static tgcalls::DarwinVideoTrackSource *getObjCVideoSource(const rtc::scoped_ref
 - (void)captureOutput:(AVCaptureOutput *)captureOutput
     didDropSampleBuffer:(CMSampleBufferRef)sampleBuffer
          fromConnection:(AVCaptureConnection *)connection {
-  NSString *droppedReason =
-      (__bridge NSString *)CMGetAttachment(sampleBuffer, kCMSampleBufferAttachmentKey_DroppedFrameReason, nil);
+  NSString *droppedReason = nil;
   RTCLogError(@"Dropped sample buffer. Reason: %@", droppedReason);
 }
 
@@ -682,7 +681,7 @@ static tgcalls::DarwinVideoTrackSource *getObjCVideoSource(const rtc::scoped_ref
             int closest = -1;
             CMTime result;
             for (int i = 0; i < format.videoSupportedFrameRateRanges.count; i++) {
-                const auto rateRange = format.videoSupportedFrameRateRanges[i];
+                const AVFrameRateRange *rateRange = format.videoSupportedFrameRateRanges[i];
                 int gap = abs(rateRange.minFrameRate - target);
                 if (gap <= closest || closest == -1) {
                     closest = gap;
diff --git a/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/VideoMetalViewMac.mm b/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/VideoMetalViewMac.mm
index 4ef8630..3fc753c 100644
--- a/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/VideoMetalViewMac.mm
+++ b/Telegram/ThirdParty/tgcalls/tgcalls/platform/darwin/VideoMetalViewMac.mm
@@ -243,9 +243,11 @@ private:
     layer.framebufferOnly = true;
     layer.opaque = false;
 //    layer.cornerRadius = 4;
+#if 0
     if (@available(macOS 10.13, *)) {
         layer.displaySyncEnabled = NO;
     }
+#endif
 //    layer.presentsWithTransaction = YES;
     layer.backgroundColor = [NSColor clearColor].CGColor;
     layer.contentsGravity = kCAGravityResizeAspectFill;
@@ -332,9 +334,7 @@ private:
 - (RTCVideoRotation)rtcFrameRotation {
     if (_rotationOverride) {
         RTCVideoRotation rotation;
-        if (@available(macOS 10.13, *)) {
-            [_rotationOverride getValue:&rotation size:sizeof(rotation)];
-        } else {
+        {
             [_rotationOverride getValue:&rotation];
         }
         return rotation;
Submodule Telegram/lib_base contains modified content
diff --git a/Telegram/lib_base/base/platform/mac/base_global_shortcuts_mac.mm b/Telegram/lib_base/base/platform/mac/base_global_shortcuts_mac.mm
index 5491702..32befc6 100644
--- a/Telegram/lib_base/base/platform/mac/base_global_shortcuts_mac.mm
+++ b/Telegram/lib_base/base/platform/mac/base_global_shortcuts_mac.mm
@@ -128,6 +128,7 @@ bool Available() {
 }
 
 bool Allowed() {
+#if 0
 	if (@available(macOS 10.15, *)) {
 		// Input Monitoring is required on macOS 10.15 an later.
 		// Even if user grants access, restart is required.
@@ -141,6 +142,7 @@ bool Allowed() {
 		return AXIsProcessTrustedWithOptions(
 			(__bridge CFDictionaryRef)options);
 	}
+#endif
 	return true;
 }
 
diff --git a/Telegram/lib_base/base/platform/mac/base_info_mac.mm b/Telegram/lib_base/base/platform/mac/base_info_mac.mm
index f1f259a..6629eb6 100644
--- a/Telegram/lib_base/base/platform/mac/base_info_mac.mm
+++ b/Telegram/lib_base/base/platform/mac/base_info_mac.mm
@@ -226,16 +226,20 @@ void Finish() {
 }
 
 void OpenInputMonitoringPrivacySettings() {
+#if 0
 	if (@available(macOS 10.15, *)) {
 		IOHIDRequestAccess(kIOHIDRequestTypeListenEvent);
 	}
+#endif
 	[[NSWorkspace sharedWorkspace] openURL:PrivacySettingsUrl("Privacy_ListenEvent")];
 }
 
 void OpenDesktopCapturePrivacySettings() {
+#if 0
 	if (@available(macOS 11.0, *)) {
 		CGRequestScreenCaptureAccess();
 	}
+#endif
 	[[NSWorkspace sharedWorkspace] openURL:PrivacySettingsUrl("Privacy_ScreenCapture")];
 }
 
diff --git a/Telegram/lib_base/base/platform/mac/base_system_media_controls_mac.mm b/Telegram/lib_base/base/platform/mac/base_system_media_controls_mac.mm
index 6102705..8981239 100644
--- a/Telegram/lib_base/base/platform/mac/base_system_media_controls_mac.mm
+++ b/Telegram/lib_base/base/platform/mac/base_system_media_controls_mac.mm
@@ -277,6 +277,7 @@ void SystemMediaControls::setThumbnail(const QImage &thumbnail) {
 	if (thumbnail.isNull()) {
 		return;
 	}
+#if 0
 	if (@available(macOS 10.13.2, *)) {
 		const auto copy = thumbnail;
 		[_private->info
@@ -290,6 +291,7 @@ void SystemMediaControls::setThumbnail(const QImage &thumbnail) {
 			forKey:MPMediaItemPropertyArtwork];
 		updateDisplay();
 	}
+#endif
 }
 
 void SystemMediaControls::setDuration(int duration) {
@@ -308,10 +310,12 @@ void SystemMediaControls::setVolume(float64 volume) {
 }
 
 void SystemMediaControls::clearThumbnail() {
+#if 0
 	if (@available(macOS 10.13.2, *)) {
 		[_private->info removeObjectForKey:MPMediaItemPropertyArtwork];
 		updateDisplay();
 	}
+#endif
 }
 
 void SystemMediaControls::clearMetadata() {
@@ -373,9 +377,11 @@ bool SystemMediaControls::volumeSupported() const {
 }
 
 bool SystemMediaControls::Supported() {
+#if 0
 	if (@available(macOS 10.12.2, *)) {
 		return true;
 	}
+#endif
 	return false;
 }
 
Submodule Telegram/lib_webrtc contains modified content
diff --git a/Telegram/lib_webrtc/webrtc/mac/webrtc_media_devices_mac.mm b/Telegram/lib_webrtc/webrtc/mac/webrtc_media_devices_mac.mm
index 21e93f7..10a3890 100644
--- a/Telegram/lib_webrtc/webrtc/mac/webrtc_media_devices_mac.mm
+++ b/Telegram/lib_webrtc/webrtc/mac/webrtc_media_devices_mac.mm
@@ -397,6 +397,7 @@ void MacMediaDevices::videoInputRefreshed() {
 }
 
 bool MacDesktopCaptureAllowed() {
+#if 0
 	if (@available(macOS 11.0, *)) {
 		// Screen Recording is required on macOS 10.15 an later.
 		// Even if user grants access, restart is required.
@@ -421,6 +422,7 @@ bool MacDesktopCaptureAllowed() {
 		CFRelease(stream);
 		return true;
 	}
+#endif
 	return true;
 }
 
