diff --git a/Ghidra/Framework/Help/src/main/java/help/JavaHelpSetBuilder.java b/Ghidra/Framework/Help/src/main/java/help/JavaHelpSetBuilder.java
--- a/Ghidra/Framework/Help/src/main/java/help/JavaHelpSetBuilder.java
+++ b/Ghidra/Framework/Help/src/main/java/help/JavaHelpSetBuilder.java
@@ -145 +145 @@
-		writer.write("<!-- HelpSet auto-generated on " + (new Date()).toString() + " -->");
+		writer.write("<!-- HelpSet auto-generated -->");
diff --git a/Ghidra/Framework/Help/src/main/java/help/JavaHelpFilesBuilder.java b/Ghidra/Framework/Help/src/main/java/help/JavaHelpFilesBuilder.java
--- a/Ghidra/Framework/Help/src/main/java/help/JavaHelpFilesBuilder.java
+++ b/Ghidra/Framework/Help/src/main/java/help/JavaHelpFilesBuilder.java
@@ -19 +19,0 @@
-import java.util.Date;
@@ -125 +124 @@
-			out.println("<!-- Auto-generated on " + (new Date()).toString() + " : Do Not Edit -->");
+			out.println("<!-- Auto-generated : Do Not Edit -->");
diff --git a/Ghidra/Framework/Help/src/main/java/help/OverlayHelpTree.java b/Ghidra/Framework/Help/src/main/java/help/OverlayHelpTree.java
--- a/Ghidra/Framework/Help/src/main/java/help/OverlayHelpTree.java
+++ b/Ghidra/Framework/Help/src/main/java/help/OverlayHelpTree.java
@@ -141 +141 @@
-			writer.println("<!-- Auto-generated on " + (new Date()).toString() + " -->");
+			writer.println("<!-- Auto-generated -->");
