diff --git a/bin/vagrant b/bin/vagrant
index 7ca30b391..d3f4ea61a 100755
--- a/bin/vagrant
+++ b/bin/vagrant
@@ -221,11 +221,6 @@ begin
       end
     end
 
-    if !Vagrant.in_installer? && !Vagrant.very_quiet?
-      # If we're not in the installer, warn.
-      env.ui.warn(I18n.t("vagrant.general.not_in_installer") + "\n", prefix: false)
-    end
-
     # Acceptable experimental flag values include:
     #
     # Unset  - Disables experimental features
