diff --git a/internal/buf/internal/buftesting/buftesting.go b/internal/buf/internal/buftesting/buftesting.go
index dc8da0c..70ad299 100644
--- a/internal/buf/internal/buftesting/buftesting.go
+++ b/internal/buf/internal/buftesting/buftesting.go
@@ -100,6 +100,10 @@ func RunActualProtoc(
 
 // GetGoogleapisDirPath gets the path to a clone of googleapis.
 func GetGoogleapisDirPath(t *testing.T, buftestingDirPath string) string {
+	// Requires network access, which is not available during
+	// the nixpkgs sandboxed build
+	t.Skip()
+
 	googleapisDirPath := filepath.Join(buftestingDirPath, testGoogleapisDirPath)
 	require.NoError(
 		t,
