diff --git a/crates/simd/build.rs b/crates/simd/build.rs
index c7a49d6..6c071ad 100644
--- a/crates/simd/build.rs
+++ b/crates/simd/build.rs
@@ -1,8 +1,12 @@
 fn main() {
     println!("cargo::rerun-if-changed=cshim.c");
+    println!("cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS");
     cc::Build::new()
-        .compiler("clang")
+        .compiler("@clang@")
         .file("cshim.c")
+        // read env var set by rustPlatform.bindgenHook
+        .try_flags_from_environment("BINDGEN_EXTRA_CLANG_ARGS")
+        .expect("the BINDGEN_EXTRA_CLANG_ARGS environment variable must be specified and UTF-8")
         .opt_level(3)
         .flag("-fassociative-math")
         .flag("-ffp-contract=fast")
