diff -Nur poezio-0.10.orig/plugins/canat.py poezio-0.10/plugins/canat.py
--- poezio-0.10.orig/plugins/canat.py	2016-08-21 14:56:35.000000000 +0200
+++ poezio-0.10/plugins/canat.py	2016-11-16 14:32:21.565445266 +0100
@@ -34,9 +34,9 @@
 
 
 """
-from plugin import BasePlugin
-import tabs
-from decorators import command_args_parser
+from poezio.plugin import BasePlugin
+import poezio.tabs
+from poezio.decorators import command_args_parser
 
 def move(text, step, spacing):
     new_text = text + (" " * spacing)
diff -Nur poezio-0.10.orig/plugins/corrections_diff.py poezio-0.10/plugins/corrections_diff.py
--- poezio-0.10.orig/plugins/corrections_diff.py	2016-08-21 14:56:35.000000000 +0200
+++ poezio-0.10/plugins/corrections_diff.py	2016-11-16 14:30:53.992684959 +0100
@@ -22,11 +22,11 @@
 
 
 """
-from plugin import BasePlugin
+from poezio.plugin import BasePlugin
 import difflib
+import collections
 from functools import wraps
-import tabs
-from config import config
+from poezio.config import config
 
 shim_message_fields = ('txt nick_color time str_time nickname user identifier'
                   ' highlight me old_message revisions jid ack')
@@ -61,10 +61,6 @@
         rev -= 1
     return ''.join(acc)
 
-Message.__repr__ = repr_message
-Message.__str__ = repr_message
-
-
 
 def corrections_enabled(func):
     @wraps(func)
diff -Nur poezio-0.10.orig/plugins/coucou.py poezio-0.10/plugins/coucou.py
--- poezio-0.10.orig/plugins/coucou.py	2016-08-21 14:56:35.000000000 +0200
+++ poezio-0.10/plugins/coucou.py	2016-11-16 14:25:37.101337668 +0100
@@ -1,4 +1,4 @@
-from plugin import BasePlugin
+from poezio.plugin import BasePlugin
 import tracemalloc
 import cProfile, pstats, io
 
diff -Nur poezio-0.10.orig/plugins/flood.py poezio-0.10/plugins/flood.py
--- poezio-0.10.orig/plugins/flood.py	2016-08-21 14:56:35.000000000 +0200
+++ poezio-0.10/plugins/flood.py	2016-11-16 14:32:56.452155220 +0100
@@ -1,6 +1,6 @@
-from plugin import BasePlugin
-import tabs
-import multiuserchat as muc
+from poezio.plugin import BasePlugin
+import poezio.tabs
+import poezio.multiuserchat as muc
 
 class Plugin(BasePlugin):
     def init(self):
diff -Nur poezio-0.10.orig/plugins/invisible.py poezio-0.10/plugins/invisible.py
--- poezio-0.10.orig/plugins/invisible.py	2016-08-21 14:56:35.000000000 +0200
+++ poezio-0.10/plugins/invisible.py	2016-11-16 14:31:31.743288152 +0100
@@ -20,8 +20,7 @@
 .. _XEP-0186: https://xmpp.org/extensions/xep-0186.html
 """
 
-from plugin import BasePlugin
-import tabs
+from poezio.plugin import BasePlugin
 
 class Plugin(BasePlugin):
     def init(self):
