--- a/setup.py	2014-03-20 22:27:14.000000000 +0100
+++ b/setup.py	2015-07-10 21:18:30.156196111 +0200
@@ -49,7 +49,7 @@
        scripts = ['scripts/poezio'],
        data_files = [('share/man/man1/', ['data/poezio.1'])],
 
-       install_requires = ['sleekxmpp==1.2.4',
+       install_requires = ['sleekxmpp==1.2.5',
                            'dnspython3>=1.11.1'],
        extras_require = {'OTR plugin': 'python-potr>=1.0',
                          'Screen autoaway plugin': 'pyinotify==0.9.4'}
