--- a/plugins/datetime/tz.h
+++ b/plugins/datetime/tz.h
@@ -27,11 +27,7 @@
 
 #include <glib.h>
 
-#ifndef __sun
-#  define TZ_DATA_FILE "/usr/share/zoneinfo/zone.tab"
-#else
-#  define TZ_DATA_FILE "/usr/share/lib/zoneinfo/tab/zone_sun.tab"
-#endif
+#define TZ_DATA_FILE "@tzdata@/share/zoneinfo/zone.tab"
 
 typedef struct _TzDB TzDB;
 typedef struct _TzLocation TzLocation;
--- a/plugins/mouse/gsd-mouse-manager.c
+++ b/plugins/mouse/gsd-mouse-manager.c
@@ -118,7 +118,7 @@ set_mousetweaks_daemon (GsdMouseManager *manager,
         gboolean run_daemon = dwell_click_enabled || secondary_click_enabled;

         if (run_daemon || manager->priv->mousetweaks_daemon_running)
-                comm = g_strdup_printf ("mousetweaks %s",
+                comm = g_strdup_printf ("@mousetweaks@/bin/mousetweaks %s",
                                         run_daemon ? "" : "-s");
         else
                 return;
