See https://github.com/newsboat/newsboat/pull/3185
diff --git a/include/strprintf.h b/include/strprintf.h
index 7b576218..7b446b89 100644
--- a/include/strprintf.h
+++ b/include/strprintf.h
@@ -55,6 +55,16 @@ std::string fmt(const std::string& format, const uint64_t argument,
 	return detail::fmt_impl(format, argument, std::forward<Args>(args)...);
 }
 
+
+#ifdef __APPLE__
+template<typename... Args>
+std::string fmt(const std::string& format, const std::size_t argument,
+	Args&& ... args)
+{
+	return detail::fmt_impl(format, argument, std::forward<Args>(args)...);
+}
+#endif
+
 template<typename... Args>
 std::string fmt(const std::string& format, const void* argument, Args&& ... args)
 {
