--- a/Makefile
+++ b/Makefile
@@ -3,8 +3,11 @@ PREFIX?=/usr/local
 
 VERSION=0.8
 
-CFLAGS=-std=c99 -DVERSION='"$(VERSION)"' -O2 -Wall
-LDFLAGS=-s -lfuse
+FUSE_CFLAGS:=$(shell pkg-config --cflags fuse3)
+CFLAGS=-std=c99 -DVERSION='"$(VERSION)"' -O2 -Wall $(FUSE_CFLAGS)
+LDFLAGS=-s
+FUSE_LIBS:=$(shell pkg-config --libs fuse3)
+LDLIBS=$(FUSE_LIBS)
 
 all: chunkfs unchunkfs chunkfs.1.gz
 
diff --git a/chunkfs.c b/chunkfs.c
index 825553d..09b5e19 100644
--- a/chunkfs.c
+++ b/chunkfs.c
@@ -19,7 +19,7 @@
 
 #define _FILE_OFFSET_BITS 64
 #define _XOPEN_SOURCE 500
-#define FUSE_USE_VERSION 25
+#define FUSE_USE_VERSION 317
 
 #include <sys/types.h>
 #include <unistd.h>
@@ -87,7 +87,7 @@ static int resolve_path(const char *path,struct chunk_stat *st){
 	return 0;
 }
 
-static int chunkfs_getattr(const char *path,struct stat *buf){
+static int chunkfs_getattr(const char *path,struct stat *buf, struct fuse_file_info *fileinfo){
 	int r;
 	struct chunk_stat st;
 
@@ -109,7 +109,7 @@ static int chunkfs_getattr(const char *path,struct stat *buf){
 	return 0;
 }
 
-static int chunkfs_readdir(const char *path,void *buf,fuse_fill_dir_t filler,off_t offset,struct fuse_file_info *fi){
+static int chunkfs_readdir(const char *path,void *buf,fuse_fill_dir_t filler,off_t offset,struct fuse_file_info *fi, enum fuse_readdir_flags flags){
 	int r;
 	struct chunk_stat st;
 	uint64_t chunks_per_entry;
@@ -117,12 +117,12 @@ static int chunkfs_readdir(const char *path,void *buf,fuse_fill_dir_t filler,off
 	if((r=resolve_path(path,&st))<0)return r;
 	if(!S_ISDIR(st.mode))return -ENOTDIR;
 	chunks_per_entry=1ULL<<((8-1-st.level)*8);
-	filler(buf,".",NULL,0);
-	filler(buf,"..",NULL,0);
+	filler(buf,".",NULL,0,FUSE_FILL_DIR_DEFAULTS);
+	filler(buf,"..",NULL,0,FUSE_FILL_DIR_DEFAULTS);
 	for(uint64_t x=0;x<256&&(uint64_t)st.chunk+x*chunks_per_entry<(uint64_t)image_chunks;x++){
 		char nbuf[3];
 		sprintf(nbuf,"%02" PRIx64,x);
-		filler(buf,nbuf,NULL,0);
+		filler(buf,nbuf,NULL,0,FUSE_FILL_DIR_DEFAULTS);
 	}
 	return 0;
 }
@@ -203,13 +203,13 @@ int main(int argc,char **argv){
 			"    -V                     print version\n"
 			"\n",
 			*argv);
-		fake_argv[1]="-ho";
-		fuse_main(2,fake_argv,&chunkfs_ops);
+		fake_argv[1]="--help";
+		fuse_main(2,fake_argv,&chunkfs_ops,NULL);
 		ret=validation_error;
 	}else if(show_version){
 		fprintf(stderr,"ChunkFS v" VERSION "\n");
 		fake_argv[1]="-V";
-		fuse_main(2,fake_argv,&chunkfs_ops);
+		fuse_main(2,fake_argv,&chunkfs_ops,NULL);
 	}else{
 		errno=0;
 		chunk_size=strtoll(argv[optind],&end,10);
@@ -228,7 +228,7 @@ int main(int argc,char **argv){
 		if(image_size%chunk_size)image_chunks++;
 		argv[optind]=argv[optind+2];
 		argc-=2;
-		ret=fuse_main(argc,argv,&chunkfs_ops);
+		ret=fuse_main(argc,argv,&chunkfs_ops,NULL);
 	}
 	return ret;
 }
diff --git a/unchunkfs.c b/unchunkfs.c
index 6302019..b8293e2 100644
--- a/unchunkfs.c
+++ b/unchunkfs.c
@@ -19,7 +19,7 @@
 
 #define _FILE_OFFSET_BITS 64
 #define _XOPEN_SOURCE 500
-#define FUSE_USE_VERSION 25
+#define FUSE_USE_VERSION 317
 
 #include <sys/types.h>
 #include <unistd.h>
@@ -72,7 +72,7 @@ static void gen_chunk_name(char *buf,off_t num){
 	}
 }
 
-static int unchunkfs_getattr(const char *path,struct stat *buf){
+static int unchunkfs_getattr(const char *path,struct stat *buf, struct fuse_file_info *fileinfo){
 	int r;
 	mode_t mode;
 
@@ -94,15 +94,15 @@ static int unchunkfs_getattr(const char *path,struct stat *buf){
 	return 0;
 }
 
-static int unchunkfs_readdir(const char *path,void *buf,fuse_fill_dir_t filler,off_t offset,struct fuse_file_info *fi){
+static int unchunkfs_readdir(const char *path,void *buf,fuse_fill_dir_t filler,off_t offset,struct fuse_file_info *fi, enum fuse_readdir_flags flags){
 	int r;
 	mode_t mode;
 
 	if((r=resolve_path(path,&mode))<0)return r;
 	if(!S_ISDIR(mode))return -ENOTDIR;
-	filler(buf,".",NULL,0);
-	filler(buf,"..",NULL,0);
-	filler(buf,"image",NULL,0);
+	filler(buf,".",NULL,0,FUSE_FILL_DIR_DEFAULTS);
+	filler(buf,"..",NULL,0,FUSE_FILL_DIR_DEFAULTS);
+	filler(buf,"image",NULL,0,FUSE_FILL_DIR_DEFAULTS);
 	return 0;
 }
 
@@ -190,13 +190,13 @@ int main(int argc,char **argv){
 			"    -V                     print version\n"
 			"\n",
 			*argv);
-		fake_argv[1]="-ho";
-		fuse_main(2,fake_argv,&unchunkfs_ops);
+		fake_argv[1]="--help";
+		fuse_main(2,fake_argv,&unchunkfs_ops,NULL);
 		ret=validation_error;
 	}else if(show_version){
 		fprintf(stderr,"UnChunkFS v" VERSION "\n");
 		fake_argv[1]="-V";
-		fuse_main(2,fake_argv,&unchunkfs_ops);
+		fuse_main(2,fake_argv,&unchunkfs_ops,NULL);
 	}else{
 		if((cwd_fd=open(".",O_RDONLY))<0)die(true,"open(.)");
 		if((chunkdir_fd=open(argv[optind],O_RDONLY))<0)die(true,"open(<chunk dir>)");
@@ -226,7 +226,7 @@ int main(int argc,char **argv){
 		argv[optind]=argv[optind+1];
 		argc--;
 		if(fchdir(cwd_fd)<0)die(true,"fchdir(<start cwd>)");
-		ret=fuse_main(argc,argv,&unchunkfs_ops);
+		ret=fuse_main(argc,argv,&unchunkfs_ops,NULL);
 	}
 	return ret;
 }
