--- upc-4.0.3.5/gcc/upc/Make-lang.in	2008-03-03 18:56:13.000000000 +0100
+++ upc-4.0.3.5/gcc/upc/Make-lang.in	2008-03-03 18:57:27.000000000 +0100
@@ -70,7 +70,7 @@ xupc$(exeext): $(srcdir)/upc/upc-cmd.c M
 	    "-DLIB_PATH=\"$${libdir}\"" \
 	    "-DINC_PATH=\"$${incdir}\"" \
 	    -c $(srcdir)/upc/upc-cmd.c -o xupc-tmp.o
-	$(CC) xupc-tmp.o -o xupc$(exeext) $(LIBS)
+	$(CC) xupc-tmp.o $(ALL_CFLAGS) $(LDFLAGS) -o xupc$(exeext) $(LIBS)
 	rm -f xupc-tmp.o
 
 upc-cmd$(exeext): $(srcdir)/upc/upc-cmd.c Makefile cc1upc$(exeext) $(LIBDEPS)
@@ -80,7 +80,7 @@ upc-cmd$(exeext): $(srcdir)/upc/upc-cmd.
 	    "-DCOMPILER=\"gcc\"" \
 	    "-DBIN_PATH=\"$(bindir)\"" \
 	    -c $(srcdir)/upc/upc-cmd.c -o upc-cmd-tmp.o
-	$(CC) upc-cmd-tmp.o -o upc-cmd$(exeext) $(LIBS)
+	$(CC) upc-cmd-tmp.o $(ALL_CFLAGS) $(LDFLAGS) -o upc-cmd$(exeext) $(LIBS)
 	rm -f upc-cmd-tmp.o
 
 # UPC language specific files.
