diff --git a/Makefile b/Makefile
index 2482be1f..9d58bc81 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ all:
 include config.mk
 include makefiles/$(PLATFORM).$(ARCH).mk
 
-all: $(BINOUT)/hare $(BINOUT)/haredoc docs
+all: $(BINOUT)/hare docs
 
 HARE_DEFINES = \
 	-D PLATFORM:str='"$(PLATFORM)"' \
@@ -79,11 +79,10 @@ docs: \
 	docs/haredoc.1 \
 	docs/hare-run.1 \
 	docs/hare-test.1 \
-	docs/haredoc.5 \
 	docs/hare-module.5
 
-MAN1 = hare hare-build hare-cache hare-deps haredoc hare-run hare-test
-MAN5 = haredoc hare-module
+MAN1 = hare hare-build hare-cache hare-deps hare-run hare-test
+MAN5 = hare-module
 
 bootstrap:
 	@BINOUT=$(BINOUT) ./scripts/genbootstrap
@@ -104,7 +103,6 @@ install-cmd:
 		'$(DESTDIR)$(BINDIR)' '$(DESTDIR)$(MANDIR)/man1' \
 		'$(DESTDIR)$(BINDIR)' '$(DESTDIR)$(MANDIR)/man5'
 	install -m755 '$(BINOUT)/hare' '$(DESTDIR)$(BINDIR)/hare'
-	install -m755 '$(BINOUT)/haredoc' '$(DESTDIR)$(BINDIR)/haredoc'
 	for i in $(MAN1); do install -m644 docs/$$i.1 '$(DESTDIR)$(MANDIR)'/man1/$$i.1; done
 	for i in $(MAN5); do install -m644 docs/$$i.5 '$(DESTDIR)$(MANDIR)'/man5/$$i.5; done
 
@@ -115,7 +113,6 @@ install-mods:
 
 uninstall:
 	rm -- '$(DESTDIR)$(BINDIR)/hare'
-	rm -- '$(DESTDIR)$(BINDIR)/haredoc'
 	for i in $(MAN1); do rm -- '$(DESTDIR)$(MANDIR)'/man1/$$i.1; done
 	for i in $(MAN5); do rm -- '$(DESTDIR)$(MANDIR)'/man5/$$i.5; done
 	rm -r -- '$(DESTDIR)$(STDLIB)'
