--- a/gir/cairo-1.0.gir.in
+++ b/gir/cairo-1.0.gir.in
@@ -5,7 +5,7 @@
             xmlns:glib="http://www.gtk.org/introspection/glib/1.0">
   <package name="@CAIRO_GIR_PACKAGE@"/>
   <namespace name="cairo" version="1.0"
-	     shared-library="@CAIRO_SHARED_LIBRARY@"
+	     shared-library="@cairoLib@/@CAIRO_SHARED_LIBRARY@"
 	     c:identifier-prefixes="cairo"
 	     c:symbol-prefixes="cairo">
     <record name="Context" c:type="cairo_t" foreign="1"
