--- eatmydata.in	2020-02-01 18:10:59.618679823 -0800
+++ eatmydata.in.new	2020-02-01 18:08:25.092620247 -0800
@@ -15,15 +15,8 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-export `dpkg-architecture|grep DEB_BUILD_MULTIARCH`
-
-shlib="/usr/lib/$DEB_BUILD_MULTIARCH/eatmydata.sh"
-if [ -f "$shlib" ]; then
-    . "$shlib"
-else
-    echo "Unable to locate eatmydata shell library, it was not enabled" >&2
-    exec "$@"
-fi
+shlib="NIX_OUT_DIR/libexec/eatmydata.sh"
+. "$shlib"
 
 usage()
 {
