--- a/SConscript
+++ b/SConscript
@@ -99,7 +99,7 @@
 elif platform == "Windows" and not GetOption('msvc'):
 	env = Environment(tools=['mingw'], ENV={'PATH' : os.environ['PATH']})
 else:
-	env = Environment(tools=['default'], ENV={'PATH' : os.environ['PATH']})
+	env = Environment(tools=['default'], ENV = os.environ)
 
 #attempt to automatically find cross compiler
 if not tool and compilePlatform == "Linux" and compilePlatform != platform:
