From f72e5b67ee1137a0ccd57db5d077a197b01b3cdc Mon Sep 17 00:00:00 2001
From: Samuel Dionne-Riel <samuel@dionne-riel.com>
Date: Tue, 4 Sep 2018 23:19:29 -0400
Subject: [PATCH] Fixes build with Qt4.

---
 keychain_unix.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/keychain_unix.cpp b/keychain_unix.cpp
index 30b26c3..b27ebef 100644
--- a/keychain_unix.cpp
+++ b/keychain_unix.cpp
@@ -91,7 +91,7 @@ static bool isKwallet5Available()
     // a wallet can be opened.
 
     iface.setTimeout(500);
-    QDBusMessage reply = iface.call(QStringLiteral("networkWallet"));
+    QDBusMessage reply = iface.call("networkWallet");
     return reply.type() == QDBusMessage::ReplyMessage;
 }
 
-- 
2.16.4

