diff --git a/ui/scripts/dist.sh b/ui/scripts/dist.sh
index 0ad6e28e..db340da0 100755
--- a/ui/scripts/dist.sh
+++ b/ui/scripts/dist.sh
@@ -15,10 +15,9 @@ DEPLOY="../pkg/web_ui"
 rm -rf $DEPLOY
 mkdir -p $DEPLOY
 
-bundle check >/dev/null 2>&1 || bundle install
-bundle exec sass styles/base.scss static/base.css
+sass styles/base.scss static/base.css
 
-bundle exec ruby scripts/compile.rb
+ruby scripts/compile.rb
 
 # Copy into deploy
 shopt -s dotglob
