diff --git a/src/Views/FirewallPanel.vala b/src/Views/FirewallPanel.vala
index 994c4d1..5702de2 100644
--- a/src/Views/FirewallPanel.vala
+++ b/src/Views/FirewallPanel.vala
@@ -49,10 +49,13 @@ public class SecurityPrivacy.FirewallPanel : Granite.SimpleSettingsPage {
     }
 
     construct {
-        settings = new Settings ("io.elementary.switchboard.security-privacy");
+        SettingsSchemaSource sss = new SettingsSchemaSource.from_directory ("@SWITCHBOARD_SEC_PRIV_GSETTINGS_PATH@", SettingsSchemaSource.get_default (), true);
+        SettingsSchema security_privacy_schema = sss.lookup ("io.elementary.switchboard.security-privacy", false);
+        settings = new Settings.full (security_privacy_schema, null, null);
+
         disabled_rules = new Gee.HashMap<string, UFWHelpers.Rule> ();
         load_disabled_rules ();
-        
+
         status_switch.notify["active"].connect (() => {
             if (loading == false) {
                 view.sensitive = status_switch.active;
diff --git a/src/Views/LockPanel.vala b/src/Views/LockPanel.vala
index 081cf10..42f6118 100644
--- a/src/Views/LockPanel.vala
+++ b/src/Views/LockPanel.vala
@@ -30,7 +30,10 @@ public class SecurityPrivacy.LockPanel : Granite.SimpleSettingsPage {
     }
 
     construct {
-        locker = new Settings ("apps.light-locker");
+        SettingsSchemaSource sss = new SettingsSchemaSource.from_directory ("@LIGHTLOCKER_GSETTINGS_PATH@", SettingsSchemaSource.get_default (), true);
+        SettingsSchema locker_schema = sss.lookup ("apps.light-locker", false);
+
+        locker = new Settings.full (locker_schema, null, null);
 
         var lock_suspend_label = new Gtk.Label (_("Lock on sleep:"));
         var lock_suspend_switch = new Gtk.Switch ();
