diff --git a/pkg/setting/setting.go b/pkg/setting/setting.go
index f206592d..796da6ef 100644
--- a/pkg/setting/setting.go
+++ b/pkg/setting/setting.go
@@ -474,7 +474,7 @@ func NewContext() {
 	LocalURL = sec.Key("LOCAL_ROOT_URL").MustString(string(Protocol) + "://localhost:" + HTTPPort + "/")
 	OfflineMode = sec.Key("OFFLINE_MODE").MustBool()
 	DisableRouterLog = sec.Key("DISABLE_ROUTER_LOG").MustBool()
-	StaticRootPath = sec.Key("STATIC_ROOT_PATH").MustString(workDir)
+	StaticRootPath = sec.Key("STATIC_ROOT_PATH").MustString("@data@")
 	AppDataPath = sec.Key("APP_DATA_PATH").MustString("data")
 	EnableGzip = sec.Key("ENABLE_GZIP").MustBool()
 
