diff --git a/src/common/gsettingshintprovider.cpp b/src/common/gsettingshintprovider.cpp
index ed71c56..1475df6 100644
--- a/src/common/gsettingshintprovider.cpp
+++ b/src/common/gsettingshintprovider.cpp
@@ -28,7 +28,7 @@ Q_LOGGING_CATEGORY(QGnomePlatformGSettingsHintProvider, "qt.qpa.qgnomeplatform.g
 
 static GSettings *loadGSettingsSchema(const QString &schema)
 {
-    GSettingsSchemaSource *source = g_settings_schema_source_get_default();
+    GSettingsSchemaSource *source = g_settings_schema_source_new_from_directory("@gds_gsettings_path@", g_settings_schema_source_get_default(), true, nullptr);
     GSettingsSchema *gschema = nullptr;
 
     gschema = g_settings_schema_source_lookup(source, schema.toLatin1(), TRUE);
