diff --git a/cint/ROOT/CMakeLists.txt b/cint/ROOT/CMakeLists.txt
--- a/cint/ROOT/CMakeLists.txt
+++ b/cint/ROOT/CMakeLists.txt
@@ -47,21 +47,13 @@ set(CINTSTLDLLHEADERS
     ${CMAKE_SOURCE_DIR}/cint/cint/lib/stdstrct/stdcxxfunc.h
   )
 set(CINTINCDLLNAMES 
-    stdfunc 
+#    stdfunc 
 #    stdcxxfunc
    )
 set(CINTINCDLLHEADERS
-    ${CMAKE_SOURCE_DIR}/cint/cint/lib/stdstrct/stdfunc.h 
+#    ${CMAKE_SOURCE_DIR}/cint/cint/lib/stdstrct/stdfunc.h 
 #    ${CMAKE_SOURCE_DIR}/cint/cint/lib/stdstrct/stdcxxfunc.h
    )
-if(NOT WIN32)
-  set(CINTSTLDLLNAMES ${CINTSTLDLLNAMES}  valarray)
-  set(CINTSTLDLLHEADERS ${CINTSTLDLLHEADERS} ${CINTDLLDIR}/vary.h)
-  set(CINTINCDLLNAMES ${CINTINCDLLNAMES} posix ipc)
-  set(CINTINCDLLHEADERS ${CINTINCDLLHEADERS} 
-    ${CMAKE_SOURCE_DIR}/cint/cint/lib/posix/exten.h 
-    ${CMAKE_SOURCE_DIR}/cint/cint/lib/ipc/ipcif.h)
-endif()
 
 set(CINTBUILDLOADER
     vector 
