--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -458,12 +458,12 @@ if (BUILD_STATIC_LIB)
   # Build a static library (without -fPIC).
   add_library (phonenumber STATIC ${SOURCES})
   target_link_libraries (phonenumber ${LIBRARY_DEPS})
-  target_include_directories(phonenumber PUBLIC $<INSTALL_INTERFACE:include>)
+  target_include_directories(phonenumber PUBLIC $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)

   if (BUILD_GEOCODER)
     add_library (geocoding STATIC ${GEOCODING_SOURCES})
     target_link_libraries (geocoding ${LIBRARY_DEPS})
-    target_include_directories(geocoding PUBLIC $<INSTALL_INTERFACE:include>)
+    target_include_directories(geocoding PUBLIC $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
     add_dependencies (geocoding generate_geocoding_data)
     add_dependencies (phonenumber generate_geocoding_data)
   endif ()
@@ -478,7 +478,7 @@ if (BUILD_SHARED_LIBS)
   # Build a shared library (with -fPIC).
   add_library (phonenumber-shared SHARED ${SOURCES})
   target_link_libraries (phonenumber-shared ${LIBRARY_DEPS})
-  target_include_directories(phonenumber-shared PUBLIC $<INSTALL_INTERFACE:include>)
+  target_include_directories(phonenumber-shared PUBLIC $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)

   set_target_properties (phonenumber-shared
     PROPERTIES
@@ -494,7 +494,7 @@ if (BUILD_SHARED_LIBS)
   if (BUILD_GEOCODER)
     add_library (geocoding-shared SHARED ${GEOCODING_SOURCES})
     target_link_libraries (geocoding-shared ${LIBRARY_DEPS})
-    target_include_directories(geocoding-shared PUBLIC $<INSTALL_INTERFACE:include>)
+    target_include_directories(geocoding-shared PUBLIC $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
     add_dependencies (geocoding-shared generate_geocoding_data)
     add_dependencies (phonenumber-shared generate_geocoding_data)