diff --git a/common/Jackdmp.cpp b/common/Jackdmp.cpp
index 7eea281..4b8d75d 100644
--- a/common/Jackdmp.cpp
+++ b/common/Jackdmp.cpp
@@ -50,43 +50,11 @@ are "hard-coded" in the source. A much better approach would be to use the contr
 - get available drivers and their possible parameters, then prepare to parse them.
 */

-#ifdef __APPLE__
-#include <CoreFoundation/CFNotificationCenter.h>
-#include <CoreFoundation/CoreFoundation.h>
-
-static void notify_server_start(const char* server_name)
-{
-    // Send notification to be used in the JackRouter plugin
-    CFStringRef ref = CFStringCreateWithCString(NULL, server_name, kCFStringEncodingMacRoman);
-    CFNotificationCenterPostNotificationWithOptions(CFNotificationCenterGetDistributedCenter(),
-            CFSTR("com.grame.jackserver.start"),
-            ref,
-            NULL,
-            kCFNotificationDeliverImmediately | kCFNotificationPostToAllSessions);
-    CFRelease(ref);
-}
-
-static void notify_server_stop(const char* server_name)
-{
-    // Send notification to be used in the JackRouter plugin
-    CFStringRef ref1 = CFStringCreateWithCString(NULL, server_name, kCFStringEncodingMacRoman);
-    CFNotificationCenterPostNotificationWithOptions(CFNotificationCenterGetDistributedCenter(),
-            CFSTR("com.grame.jackserver.stop"),
-            ref1,
-            NULL,
-            kCFNotificationDeliverImmediately | kCFNotificationPostToAllSessions);
-    CFRelease(ref1);
-}
-
-#else
-
 static void notify_server_start(const char* server_name)
 {}
 static void notify_server_stop(const char* server_name)
 {}

-#endif
-
 static void copyright(FILE* file)
 {
     fprintf(file, "jackdmp " VERSION "\n"

