From 552fcf4c45deadbf6b51128d32dd23536af31e9c Mon Sep 17 00:00:00 2001
From: Christoph Heiss <c.heiss@proxmox.com>
Date: Tue, 12 Aug 2025 13:48:48 +0200
Subject: [PATCH proxmox-backup 1/2] cargo: re-route dependencies not available
 on crates.io

Signed-off-by: Christoph Heiss <christoph@c8h4.io>
---
 Cargo.toml | 83 ++++++++++++++++++++++++++++--------------------------
 1 file changed, 43 insertions(+), 40 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 337bb081..0aed1103 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -263,47 +263,50 @@ proxmox-rrd-api-types.workspace = true
 # Local path overrides
 # NOTE: You must run `cargo update` after changing this for it to take effect!
 [patch.crates-io]
-#pbs-api-types = { path = "../proxmox/pbs-api-types" }
-#proxmox-acme = { path = "../proxmox/proxmox-acme" }
-#proxmox-apt = { path = "../proxmox/proxmox-apt" }
-#proxmox-apt-api-types = { path = "../proxmox/proxmox-apt-api-types" }
-#proxmox-async = { path = "../proxmox/proxmox-async" }
-#proxmox-auth-api = { path = "../proxmox/proxmox-auth-api" }
-#proxmox-base64 = { path = "../proxmox/proxmox-base64" }
-#proxmox-borrow = { path = "../proxmox/proxmox-borrow" }
-#proxmox-compression = { path = "../proxmox/proxmox-compression" }
-#proxmox-config-digest = { path = "../proxmox/proxmox-config-digest" }
-#proxmox-daemon = { path = "../proxmox/proxmox-daemon" }
-#proxmox-fuse = { path = "../proxmox-fuse" }
-#proxmox-http = { path = "../proxmox/proxmox-http" }
-#proxmox-human-byte = { path = "../proxmox/proxmox-human-byte" }
-#proxmox-io = { path = "../proxmox/proxmox-io" }
-#proxmox-lang = { path = "../proxmox/proxmox-lang" }
-#proxmox-log = { path = "../proxmox/proxmox-log" }
-#proxmox-ldap = { path = "../proxmox/proxmox-ldap" }
-#proxmox-metrics = { path = "../proxmox/proxmox-metrics" }
-#proxmox-network-api = { path = "../proxmox/proxmox-network-api" }
-#proxmox-notify = { path = "../proxmox/proxmox-notify" }
-#proxmox-openid = { path = "../proxmox/proxmox-openid" }
-#proxmox-rest-server = { path = "../proxmox/proxmox-rest-server" }
-#proxmox-router = { path = "../proxmox/proxmox-router" }
-#proxmox-rrd = { path = "../proxmox/proxmox-rrd" }
-#proxmox-rrd-api-types = { path = "../proxmox/proxmox-rrd-api-types" }
-#proxmox-schema = { path = "../proxmox/proxmox-schema" }
-#proxmox-section-config = { path = "../proxmox/proxmox-section-config" }
-#proxmox-serde = { path = "../proxmox/proxmox-serde" }
-#proxmox-shared-memory = { path = "../proxmox/proxmox-shared-memory" }
-#proxmox-sortable-macro = { path = "../proxmox/proxmox-sortable-macro" }
-#proxmox-subscription = { path = "../proxmox/proxmox-subscription" }
-#proxmox-sys = { path = "../proxmox/proxmox-sys" }
-#proxmox-systemd = { path = "../proxmox/proxmox-systemd" }
-#proxmox-tfa = { path = "../proxmox/proxmox-tfa" }
-#proxmox-time = { path = "../proxmox/proxmox-time" }
-#proxmox-uuid = { path = "../proxmox/proxmox-uuid" }
-#proxmox-worker-task = { path = "../proxmox/proxmox-worker-task" }
+pbs-api-types = { path = "../proxmox/pbs-api-types" }
+proxmox-acme = { path = "../proxmox/proxmox-acme" }
+proxmox-apt = { path = "../proxmox/proxmox-apt" }
+proxmox-apt-api-types = { path = "../proxmox/proxmox-apt-api-types" }
+proxmox-async = { path = "../proxmox/proxmox-async" }
+proxmox-auth-api = { path = "../proxmox/proxmox-auth-api" }
+proxmox-base64 = { path = "../proxmox/proxmox-base64" }
+proxmox-borrow = { path = "../proxmox/proxmox-borrow" }
+proxmox-compression = { path = "../proxmox/proxmox-compression" }
+proxmox-config-digest = { path = "../proxmox/proxmox-config-digest" }
+proxmox-daemon = { path = "../proxmox/proxmox-daemon" }
+proxmox-fuse = { path = "../proxmox-fuse" }
+proxmox-http = { path = "../proxmox/proxmox-http" }
+proxmox-human-byte = { path = "../proxmox/proxmox-human-byte" }
+proxmox-io = { path = "../proxmox/proxmox-io" }
+proxmox-lang = { path = "../proxmox/proxmox-lang" }
+proxmox-log = { path = "../proxmox/proxmox-log" }
+proxmox-ldap = { path = "../proxmox/proxmox-ldap" }
+proxmox-metrics = { path = "../proxmox/proxmox-metrics" }
+proxmox-network-api = { path = "../proxmox/proxmox-network-api" }
+proxmox-notify = { path = "../proxmox/proxmox-notify" }
+proxmox-openid = { path = "../proxmox/proxmox-openid" }
+proxmox-rest-server = { path = "../proxmox/proxmox-rest-server" }
+proxmox-router = { path = "../proxmox/proxmox-router" }
+proxmox-rrd = { path = "../proxmox/proxmox-rrd" }
+proxmox-rrd-api-types = { path = "../proxmox/proxmox-rrd-api-types" }
+proxmox-schema = { path = "../proxmox/proxmox-schema" }
+proxmox-section-config = { path = "../proxmox/proxmox-section-config" }
+proxmox-serde = { path = "../proxmox/proxmox-serde" }
+proxmox-shared-memory = { path = "../proxmox/proxmox-shared-memory" }
+proxmox-sortable-macro = { path = "../proxmox/proxmox-sortable-macro" }
+proxmox-subscription = { path = "../proxmox/proxmox-subscription" }
+proxmox-sys = { path = "../proxmox/proxmox-sys" }
+proxmox-systemd = { path = "../proxmox/proxmox-systemd" }
+proxmox-tfa = { path = "../proxmox/proxmox-tfa" }
+proxmox-time = { path = "../proxmox/proxmox-time" }
+proxmox-uuid = { path = "../proxmox/proxmox-uuid" }
+proxmox-worker-task = { path = "../proxmox/proxmox-worker-task" }
+proxmox-s3-client = { path = "../proxmox/proxmox-s3-client" }
+proxmox-product-config = { path = "../proxmox/proxmox-product-config" }
+proxmox-shared-cache = { path = "../proxmox/proxmox-shared-cache" }
 
-#pathpatterns = {path = "../pathpatterns" }
-#pxar = { path = "../pxar" }
+pathpatterns = {path = "../pathpatterns" }
+pxar = { path = "../pxar" }
 
 [features]
 default = []
-- 
2.50.1

