From b71b5bab3fadc663d322e3ef2faa8f098423fb03 Mon Sep 17 00:00:00 2001
From: Christoph Heiss <christoph@c8h4.io>
Date: Tue, 24 Dec 2024 17:22:35 +0100
Subject: [PATCH 2/5] docs: add target path fixup variable

Signed-off-by: Christoph Heiss <christoph@c8h4.io>
---
 docs/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/docs/Makefile b/docs/Makefile
index c57cbbc2..fa00729e 100644
--- a/docs/Makefile
+++ b/docs/Makefile
@@ -94,6 +94,7 @@ API_VIEWER_FILES :=							\
 SPHINXOPTS    = -E
 SPHINXBUILD   = sphinx-build
 BUILDDIR      = output
+DEB_HOST_RUST_TYPE ?= $(RUSTC_TARGET)
 
 ifeq ($(BUILD_MODE), release)
 COMPILEDIR := ../target/$(DEB_HOST_RUST_TYPE)/release
-- 
2.49.0

