diff --git a/Makefile b/Makefile
index 85862fb8..ed52c5ec 100644
--- a/Makefile
+++ b/Makefile
@@ -841,7 +841,7 @@ endif
 
 ICON_SIZES=16x16 24x24 32x32 48x48 64x64 128x128 256x256 512x512
 
-install: all $(if $(BUILD_DOCS),install-doc) install-datadir install-localstatedir \
+install: all $(if $(BUILD_DOCS),install-doc) install-datadir \
 	$(if $(INSTALL_BLOBS),$(edk2-decompressed)) \
 	recurse-install
 ifneq ($(TOOLS),)
