diff -ru -x '*~' man-1.6g-orig/src/man-config.c man-1.6g/src/man-config.c
--- man-1.6g-orig/src/man-config.c	2005-08-21 01:26:06.000000000 +0200
+++ man-1.6g/src/man-config.c	2012-03-06 11:11:28.159050524 +0100
@@ -214,8 +214,9 @@
 const char *configuration_file = "[no configuration file]";
 
 char *default_config_files[] = {
+    "/etc/man.conf",
      CONFIG_FILE,		/* compiled-in default */
-     "/etc/man.conf", "/etc/man.config",
+     "/etc/man.config",
      "/usr/lib/man.conf", "/usr/lib/man.config",
      "/usr/share/misc/man.conf", "/usr/share/misc/man.config"
 };
