diff --git a/requests/sessions.py b/requests/sessions.py
index e8e2d609..3d728851 100644
--- a/requests/sessions.py
+++ b/requests/sessions.py
@@ -704,7 +704,9 @@ class Session(SessionRedirectMixin):
             # with cURL.
             if verify is True or verify is None:
                 verify = (os.environ.get('REQUESTS_CA_BUNDLE') or
-                          os.environ.get('CURL_CA_BUNDLE'))
+                          os.environ.get('CURL_CA_BUNDLE') or
+                          "/etc/ssl/certs/ca-bundle.crt"
+                          )
 
         # Merge all the kwargs.
         proxies = merge_setting(proxies, self.proxies)
