diff -Naur openjdk-orig/jdk/make/javax/crypto/Makefile openjdk/jdk/make/javax/crypto/Makefile
--- openjdk-orig/jdk/make/javax/crypto/Makefile	2011-06-27 13:18:45.000000000 -0400
+++ openjdk/jdk/make/javax/crypto/Makefile	2012-11-08 14:12:01.360718616 -0500
@@ -156,7 +156,7 @@
 #
 
 ifdef OPENJDK
-all: build-jar install-jar build-policy install-limited
+all: build-jar install-jar build-policy install-unlimited
 else  # OPENJDK
 ifeq ($(strip $(FILES_java)),)
 all:
