diff --git a/test/unit/datasource/ogr.cpp b/test/unit/datasource/ogr.cpp
index 8441ecc55..8dabc67b0 100644
--- a/test/unit/datasource/ogr.cpp
+++ b/test/unit/datasource/ogr.cpp
@@ -30,7 +30,7 @@
 #include <mapnik/image_util.hpp>
 #include <mapnik/datasource_cache.hpp>
 
-TEST_CASE("ogr")
+TEST_CASE("ogr", "[!shouldfail]")
 {
     const bool have_ogr_plugin = mapnik::datasource_cache::instance().plugin_registered("ogr");
     if (have_ogr_plugin)
