diff --git a/packages/flutter_tools/lib/src/base/os.dart b/packages/flutter_tools/lib/src/base/os.dart
index 9134a014f8d..0410f328c66 100644
--- a/packages/flutter_tools/lib/src/base/os.dart
+++ b/packages/flutter_tools/lib/src/base/os.dart
@@ -316,7 +316,6 @@ class _LinuxUtils extends _PosixUtils {
         final String osRelease = _fileSystem.file(osReleasePath).readAsStringSync();
         prettyName = _getOsReleaseValueForKey(osRelease, prettyNameKey);
       } on Exception catch (e) {
-        _logger.printTrace('Failed obtaining PRETTY_NAME for Linux: $e');
         prettyName = '';
       }
       try {
