diff --git a/src/Indicator.vala b/src/Indicator.vala
index cd7ca49..7813789 100644
--- a/src/Indicator.vala
+++ b/src/Indicator.vala
@@ -94,7 +94,7 @@ public class Keyboard.Indicator : Wingpanel.Indicator {
     private void show_keyboard_map () {
         close ();
 
-        string command = "gkbd-keyboard-display \"--layout=" + layouts.get_current_with_variant () + "\"";
+        string command = "@gkbd_keyboard_display@ \"--layout=" + layouts.get_current_with_variant () + "\"";
 
         try {
             AppInfo.create_from_commandline (command, null, AppInfoCreateFlags.NONE).launch (null, null);
