diff -ru -x '*~' git-1.9.2-orig/git-send-email.perl git-1.9.2/git-send-email.perl
--- git-1.9.2-orig/git-send-email.perl	2014-04-09 21:09:34.000000000 +0200
+++ git-1.9.2/git-send-email.perl	2014-04-16 18:35:05.861132282 +0200
@@ -1094,6 +1094,8 @@
 		return;
 	}
 
+	$smtp_ssl_cert_path //= $ENV{'OPENSSL_X509_CERT_FILE'};
+
 	if (!defined $smtp_ssl_cert_path) {
 		# use the OpenSSL defaults
 		return (SSL_verify_mode => SSL_VERIFY_PEER());
