This patch uses the right name for `docbook2texi'.

diff --git a/Documentation/Makefile b/Documentation/Makefile
index 26a2342bea..ceccd67ebb 100644
--- a/Documentation/Makefile
+++ b/Documentation/Makefile
@@ -132,7 +132,7 @@ HTML_REPO = ../../git-htmldocs
 
 MAKEINFO = makeinfo
 INSTALL_INFO = install-info
-DOCBOOK2X_TEXI = docbook2x-texi
+DOCBOOK2X_TEXI = docbook2texi
 DBLATEX = dblatex
 ASCIIDOC_DBLATEX_DIR = /etc/asciidoc/dblatex
 DBLATEX_COMMON = -p $(ASCIIDOC_DBLATEX_DIR)/asciidoc-dblatex.xsl -s $(ASCIIDOC_DBLATEX_DIR)/asciidoc-dblatex.sty
