diff --git a/connect.c b/connect.c
index fd7ffe1..20cd992 100644
--- a/connect.c
+++ b/connect.c
@@ -768,7 +768,7 @@

 				ssh = getenv("GIT_SSH");
 				if (!ssh)
-					ssh = "ssh";
+					ssh = "@ssh@";
				else
					handle_ssh_variant(ssh, 0,
								&port_option,
diff --git a/git-gui/lib/remote_add.tcl b/git-gui/lib/remote_add.tcl
index 50029d0..17b9594 100644
--- a/git-gui/lib/remote_add.tcl
+++ b/git-gui/lib/remote_add.tcl
@@ -139,7 +139,7 @@
 		# Parse the location
 		if { [regexp {(?:git\+)?ssh://([^/]+)(/.+)} $location xx host path]
 		     || [regexp {([^:][^:]+):(.+)} $location xx host path]} {
-			set ssh ssh
+			set ssh @ssh@
 			if {[info exists env(GIT_SSH)]} {
 				set ssh $env(GIT_SSH)
 			}
