diff --git a/swig/python/GNUmakefile.old b/swig/python/GNUmakefile
index b4d5e90..2160b83 100644
--- a/swig/python/GNUmakefile
+++ b/swig/python/GNUmakefile
@@ -67,11 +67,7 @@ egg:
 	
 install:
 
-ifeq ($(PY_HAVE_SETUPTOOLS),1)
-	$(PYTHON) setup.py install 
-else
 	$(PYTHON) setup.py install --prefix=$(DESTDIR)$(prefix)
-endif
 
 	for f in $(SCRIPTS) ; do $(INSTALL) ./scripts/$$f $(DESTDIR)$(INST_BIN) ; done
 

