diff --git a/build/common.mk b/build/common.mk
index ac8fb0841..97ea9bc4f 100644
--- a/build/common.mk
+++ b/build/common.mk
@@ -32,7 +32,7 @@ PKG_ROOT := $(REPO_ROOT)/pkg
 UI_ROOT  := $(PKG_ROOT)/ui

 # Ensure we have an unambiguous GOPATH.
-export GOPATH := $(realpath $(ORG_ROOT)/../../..)
+#export GOPATH := $(realpath $(ORG_ROOT)/../../..)
 #                                       ^  ^  ^~ GOPATH
 #                                       |  |~ GOPATH/src
 #                                       |~ GOPATH/src/github.com
