diff --git a/cmake/LMDBXX.cmake b/cmake/LMDBXX.cmake
index 3b9817d..e69de29 100644
--- a/cmake/LMDBXX.cmake
+++ b/cmake/LMDBXX.cmake
@@ -1,23 +0,0 @@
-include(ExternalProject)
-
-#
-# Build lmdbxx.
-#
-
-set(THIRD_PARTY_ROOT ${CMAKE_SOURCE_DIR}/.third-party)
-set(LMDBXX_ROOT ${THIRD_PARTY_ROOT}/lmdbxx)
-
-set(LMDBXX_INCLUDE_DIR ${LMDBXX_ROOT})
-
-ExternalProject_Add(
-  lmdbxx
-
-  GIT_REPOSITORY https://github.com/bendiken/lmdbxx
-  GIT_TAG 0b43ca87d8cfabba392dfe884eb1edb83874de02
-
-  BUILD_IN_SOURCE 1
-  SOURCE_DIR ${LMDBXX_ROOT}
-  CONFIGURE_COMMAND ""
-  BUILD_COMMAND ""
-  INSTALL_COMMAND ""
-)
diff --git a/cmake/MatrixStructs.cmake b/cmake/MatrixStructs.cmake
index cef00f6..e69de29 100644
--- a/cmake/MatrixStructs.cmake
+++ b/cmake/MatrixStructs.cmake
@@ -1,33 +0,0 @@
-include(ExternalProject)
-
-#
-# Build matrix-structs.
-#
-
-set(THIRD_PARTY_ROOT ${CMAKE_SOURCE_DIR}/.third-party)
-set(MATRIX_STRUCTS_ROOT ${THIRD_PARTY_ROOT}/matrix_structs)
-set(MATRIX_STRUCTS_INCLUDE_DIR ${MATRIX_STRUCTS_ROOT}/include)
-set(MATRIX_STRUCTS_LIBRARY matrix_structs)
-
-link_directories(${MATRIX_STRUCTS_ROOT})
-
-set(WINDOWS_FLAGS "")
-
-if(MSVC)
-    set(WINDOWS_FLAGS "-DCMAKE_GENERATOR_PLATFORM=x64")
-endif()
-
-ExternalProject_Add(
-  MatrixStructs
-
-  GIT_REPOSITORY https://github.com/mujx/matrix-structs
-  GIT_TAG 5e57c2385a79b6629d1998fec4a7c0baee23555e
-
-  BUILD_IN_SOURCE 1
-  SOURCE_DIR ${MATRIX_STRUCTS_ROOT}
-  CONFIGURE_COMMAND ${CMAKE_COMMAND}
-    -DCMAKE_BUILD_TYPE=Release ${MATRIX_STRUCTS_ROOT}
-    ${WINDOWS_FLAGS}
-  BUILD_COMMAND ${CMAKE_COMMAND} --build ${MATRIX_STRUCTS_ROOT} --config Release
-  INSTALL_COMMAND ""
-)
diff --git a/cmake/Tweeny.cmake b/cmake/Tweeny.cmake
index 537ac92..e69de29 100644
--- a/cmake/Tweeny.cmake
+++ b/cmake/Tweeny.cmake
@@ -1,23 +0,0 @@
-include(ExternalProject)
-
-#
-# Build tweeny
-#
-
-set(THIRD_PARTY_ROOT ${CMAKE_SOURCE_DIR}/.third-party)
-set(TWEENY_ROOT ${THIRD_PARTY_ROOT}/tweeny)
-
-set(TWEENY_INCLUDE_DIR ${TWEENY_ROOT}/include)
-
-ExternalProject_Add(
-  Tweeny
-
-  GIT_REPOSITORY https://github.com/mobius3/tweeny
-  GIT_TAG b94ce07cfb02a0eb8ac8aaf66137dabdaea857cf
-
-  BUILD_IN_SOURCE 1
-  SOURCE_DIR ${TWEENY_ROOT}
-  CONFIGURE_COMMAND ""
-  BUILD_COMMAND ""
-  INSTALL_COMMAND ""
-)
