diff --git a/CMakeLists.txt b/CMakeLists.txt
index 077ac37..c639d71 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,16 +18,6 @@ include(Doxygen)
 #
 include(CompilerFlags)
 
-file(DOWNLOAD
-    "https://github.com/nlohmann/json/releases/download/v3.1.2/json.hpp"
-    ${PROJECT_SOURCE_DIR}/include/json.hpp
-    EXPECTED_HASH SHA256=fbdfec4b4cf63b3b565d09f87e6c3c183bdd45c5be1864d3fcb338f6f02c1733)
-
-file(DOWNLOAD
-    "https://github.com/mpark/variant/releases/download/v1.3.0/variant.hpp"
-    ${PROJECT_SOURCE_DIR}/include/variant.hpp
-    EXPECTED_MD5 "be0ce322cdd408e1b347b9f1d59ea67a")
-
 include_directories(include)
 
 set(SRC
