diff --git a/fsck/main.c b/fsck/main.c
index ab3342a..79a0473 100644
--- a/fsck/main.c
+++ b/fsck/main.c
@@ -37,6 +37,7 @@ fsck_progress ("Usage: %s [mode] [options] "					\
 "  -z | --adjust-size\t\tfix file sizes to real size\n"				\
 "  -q | --quiet\t\t\tno speed info\n"						\
 "  -y | --yes\t\t\tno confirmations\n"						\
+"  -v         \t\t ignore -v option for compatibility to fsck.* tools\n"						\
 "  -V\t\t\t\tprints version and exits\n"					\
 "  -a and -p\t\t\tsome light-weight auto checks for bootup\n"			\
 "  -f and -r\t\t\tignored\n"							\
@@ -131,7 +132,7 @@ static char * parse_options (struct fsck_data * data, int argc, char * argv [])
 	};
 	int option_index;
       
-	c = getopt_long (argc, argv, "iql:nb:Szd:R:h:j:gafVrpyt:B:",
+	c = getopt_long (argc, argv, "iql:nb:Szd:R:h:j:gafVrpyvt:B:",
 			 options, &option_index);
 	if (c == -1)
 	    break;
@@ -158,6 +159,10 @@ static char * parse_options (struct fsck_data * data, int argc, char * argv [])
 	    data->options |= OPT_YES;
 	    break;
 
+	case 'v': /* --quiet */
+            reiserfs_warning(stdout, "\nignoring -v option for compatibility\n");
+	    break;
+
 	case 'l': /* --logfile */
 	    data->log_file_name = optarg;
 	    /*asprintf (&data->log_file_name, "%s", optarg);*/
