diff --git a/database/DBio.c b/database/DBio.c
index 93c4b0b..292ea5f 100644
--- a/database/DBio.c
+++ b/database/DBio.c
@@ -2378,12 +2378,12 @@ DBCellWriteFile(cellDef, f)
 
 #define FPRINTF(f,s)\
 {\
-     if (fprintf(f,s) == EOF) goto ioerror;\
+     if (fprintf(f,"%s",s) == EOF) goto ioerror;\
      DBFileOffset += strlen(s);\
 }
 #define FPRINTR(f,s)\
 {\
-     if (fprintf(f,s) == EOF) return 1;\
+     if (fprintf(f,"%s",s) == EOF) return 1;\
      DBFileOffset += strlen(s);\
 }
 
