commit 486964fad93334a52fb05e7d0497ecac3eb684fe
Author: Ismaël Bouya <ismael.bouya@normalesup.org>
Date:   Wed Feb 13 12:16:27 2019 +0100

    Add sendmail

diff --git a/config/production.yaml.example b/config/production.yaml.example
index c56691bf4..8abdfb2a7 100644
--- a/config/production.yaml.example
+++ b/config/production.yaml.example
@@ -66,6 +66,8 @@ auth:
 
 # SMTP server to send emails
 smtp:
+  transport: smtp
+  sendmail: null
   hostname: null
   port: 465 # If you use StartTLS: 587
   username: null
diff --git a/server/initializers/config.ts b/server/initializers/config.ts
index 45a667826..c1c15f05b 100644
--- a/server/initializers/config.ts
+++ b/server/initializers/config.ts
@@ -50,6 +50,8 @@ const CONFIG = {
     },
   },
   SMTP: {
+    TRANSPORT: config.has('smtp.transport') ? config.get<string>('smtp.transport') : 'smtp',
+    SENDMAIL: config.has('smtp.sendmail') ? config.get<string>('smtp.sendmail') : null,
     HOSTNAME: config.get<string>('smtp.hostname'),
     PORT: config.get<number>('smtp.port'),
     USERNAME: config.get<string>('smtp.username'),
diff --git a/server/lib/emailer.ts b/server/lib/emailer.ts
index 7484524a4..512c5c068 100644
--- a/server/lib/emailer.ts
+++ b/server/lib/emailer.ts
@@ -40,33 +40,41 @@ class Emailer {
     this.initialized = true
 
     if (Emailer.isEnabled()) {
-      logger.info('Using %s:%s as SMTP server.', CONFIG.SMTP.HOSTNAME, CONFIG.SMTP.PORT)
-
-      let tls
-      if (CONFIG.SMTP.CA_FILE) {
-        tls = {
-          ca: [ readFileSync(CONFIG.SMTP.CA_FILE) ]
+      if (CONFIG.SMTP.TRANSPORT === 'smtp') {
+        logger.info('Using %s:%s as SMTP server.', CONFIG.SMTP.HOSTNAME, CONFIG.SMTP.PORT)
+
+        let tls
+        if (CONFIG.SMTP.CA_FILE) {
+          tls = {
+            ca: [ readFileSync(CONFIG.SMTP.CA_FILE) ]
+          }
         }
-      }
 
-      let auth
-      if (CONFIG.SMTP.USERNAME && CONFIG.SMTP.PASSWORD) {
-        auth = {
-          user: CONFIG.SMTP.USERNAME,
-          pass: CONFIG.SMTP.PASSWORD
+        let auth
+        if (CONFIG.SMTP.USERNAME && CONFIG.SMTP.PASSWORD) {
+          auth = {
+            user: CONFIG.SMTP.USERNAME,
+            pass: CONFIG.SMTP.PASSWORD
+          }
         }
-      }
 
-      this.transporter = createTransport({
-        host: CONFIG.SMTP.HOSTNAME,
-        port: CONFIG.SMTP.PORT,
-        secure: CONFIG.SMTP.TLS,
-        debug: CONFIG.LOG.LEVEL === 'debug',
-        logger: bunyanLogger as any,
-        ignoreTLS: CONFIG.SMTP.DISABLE_STARTTLS,
-        tls,
-        auth
-      })
+        this.transporter = createTransport({
+          host: CONFIG.SMTP.HOSTNAME,
+          port: CONFIG.SMTP.PORT,
+          secure: CONFIG.SMTP.TLS,
+          debug: CONFIG.LOG.LEVEL === 'debug',
+          logger: bunyanLogger as any,
+          ignoreTLS: CONFIG.SMTP.DISABLE_STARTTLS,
+          tls,
+          auth
+        })
+      } else { // sendmail
+        this.transporter = createTransport({
+          sendmail: true,
+          newline: 'unix',
+          path: CONFIG.SMTP.SENDMAIL,
+        })
+      }
     } else {
       if (!isTestInstance()) {
         logger.error('Cannot use SMTP server because of lack of configuration. PeerTube will not be able to send mails!')
@@ -75,11 +83,17 @@ class Emailer {
   }
 
   static isEnabled () {
-    return !!CONFIG.SMTP.HOSTNAME && !!CONFIG.SMTP.PORT
+    if (CONFIG.SMTP.TRANSPORT === 'sendmail') {
+      return !!CONFIG.SMTP.SENDMAIL
+    } else if (CONFIG.SMTP.TRANSPORT === 'smtp') {
+      return !!CONFIG.SMTP.HOSTNAME && !!CONFIG.SMTP.PORT
+    } else {
+      return false
+    }
   }
 
   async checkConnectionOrDie () {
-    if (!this.transporter) return
+    if (!this.transporter || CONFIG.SMTP.TRANSPORT !== 'smtp') return
 
     logger.info('Testing SMTP server...')
 
