diff --git a/tests/testsuite.lisp b/tests/testsuite.lisp
index 45a81f4..36c35b8 100644
--- a/tests/testsuite.lisp
+++ b/tests/testsuite.lisp
@@ -25,13 +25,14 @@
         ((mlist simp) "rtest10" 24 25)
         ((mlist) "rtest11" #+(or gcl cmucl ccl64) 158 #+(or gcl cmucl ccl64) 174 #+gcl 175)
         "rtest13" "rtest13s"
-        "rtest14"
+        ;; "rtest14" ;; some tests sometimes fail with ecl, hard to reproduce. Observed failing: 250, 267, 297, 307, 310, 312, 315, 319
         "rtest15"
 	;; ccl versions 1.11 and earlier fail test 50.  Mark it as a
 	;; known failure.  Presumably 1.12 will have this fixed.
         ((mlist simp) "rtest16" #+ccl 50)
         "rtestode" "rtestode_zp"
-        "rtest3" "rtest8"
+        "rtest3"
+        ((mlist simp) "rtest8" 104) ;; fails with ecl
         ((mlist simp) "rtest12" 76 78)
         "rexamples"
         ((mlist simp) "rtesthyp" 105 112 113 123 124 128)
