diff --git a/cmake/FindJsoncpp.cmake b/cmake/FindJsoncpp.cmake
index 908dba4..fedfb94 100644
--- a/cmake/FindJsoncpp.cmake
+++ b/cmake/FindJsoncpp.cmake
@@ -16,9 +16,9 @@
 
 find_package(jsoncpp REQUIRED CONFIG)
 
-if (TARGET jsoncpp_static)
+if (TARGET jsoncpp)
     set(Jsoncpp_FOUND TRUE)
-    set(JSONCPP_STATIC_LIBS jsoncpp_static)
+    set(JSONCPP_STATIC_LIBS jsoncpp)
     set(JSONCPP_INCLUDE_PATH $<TARGET_PROPERTY:jsoncpp_static,INTERFACE_INCLUDE_DIRECTORIES>)
     mark_as_advanced(JSONCPP_INCLUDE_PATH JSONCPP_STATIC_LIBS)
 else()
diff --git a/cmake/FindLibevent.cmake b/cmake/FindLibevent.cmake
index 93ef27c..b594567 100644
--- a/cmake/FindLibevent.cmake
+++ b/cmake/FindLibevent.cmake
@@ -19,7 +19,13 @@
 # LIBEVENT_STATIC_LIBS, Libevent static libraries
 # Libevent_FOUND, If false, do not try to use libevent
 
-find_path(Libevent_INCLUDE_DIR "event2/event.h")
+set(Libevent_EXTRA_PREFIXES / /lib /lib64 /usr/local /opt/local "$ENV{HOME}" "${Libevent_ROOT}")
+foreach(prefix ${Libevent_EXTRA_PREFIXES})
+  list(APPEND Libevent_INCLUDE_PATHS "${prefix}/include")
+  list(APPEND Libevent_LIB_PATHS "${prefix}/lib" "${prefix}/lib64")
+endforeach()
+
+find_path(Libevent_INCLUDE_DIR event.h PATHS ${Libevent_INCLUDE_PATHS})
 mark_as_advanced(Libevent_INCLUDE_DIR)
 
 if (Libevent_INCLUDE_DIR)
@@ -31,16 +37,16 @@ if (Libevent_INCLUDE_DIR)
     set(_Libevent_LIBRARY_SUFFIX ${CMAKE_STATIC_LIBRARY_SUFFIX})
 
     find_library(Libevent_core_LIBRARY
-        NAMES ${_Libevent_LIBRARY_PREFIX}event_core${_Libevent_LIBRARY_SUFFIX}
-        HINTS "${_Libevent_INSTALL_PREFIX}")
+        NAMES libevent_core.so
+        PATHS ${Libevent_INCLUDE_PATHS})
 
     find_library(Libevent_extra_LIBRARY
-        NAMES ${_Libevent_LIBRARY_PREFIX}event_extra${_Libevent_LIBRARY_SUFFIX}
-        HINTS "${_Libevent_INSTALL_PREFIX}")
+        NAMES libevent_extra.so
+        PATHS ${Libevent_INCLUDE_PATHS})
 
     find_library(Libevent_pthreads_LIBRARY
-        NAMES ${_Libevent_LIBRARY_PREFIX}event_pthreads${_Libevent_LIBRARY_SUFFIX}
-        HINTS "${_Libevent_INSTALL_PREFIX}")
+        NAMES libevent_pthreads.so
+        PATHS ${Libevent_INCLUDE_PATHS})
 
     mark_as_advanced(
         Libevent_core_LIBRARY
