diff --git a/include/Rivet/Tools/osdir.hh b/include/Rivet/Tools/osdir.hh
index 05f06ca..59af7de 100644
--- a/include/Rivet/Tools/osdir.hh
+++ b/include/Rivet/Tools/osdir.hh
@@ -21,7 +21,7 @@
 
 /// @cond OSDIR
 
-#if defined(unix) || defined(__unix) || defined(__unix__)
+#if defined(unix) || defined(__unix) || defined(__unix__) || defined(__APPLE__)
 #define OSLINK_OSDIR_POSIX
 #elif defined(_WIN32)
 #define OSLINK_OSDIR_WINDOWS
@@ -32,18 +32,7 @@
 #include <string>
 
 #if defined(OSLINK_OSDIR_NOTSUPPORTED)
-
-namespace oslink
-{
-	class directory
-	{
-		public:
-			directory(const std::string&)		{ }
-			operator void*() const				{ return (void*)0; }
-			std::string next()					{ return ""; }
-	};
-}
-
+#error Platform misdetected or oslink is not implemented
 #elif defined(OSLINK_OSDIR_POSIX)
 
 #include <sys/types.h>
