diff --git a/bin/make-plots b/bin/make-plots
index abb024d..f59293d 100755
--- a/bin/make-plots
+++ b/bin/make-plots
@@ -1152,8 +1152,8 @@ class ColorScale(Described):
 
         zcustommajortickmarks = int(self.description.get('ZMajorTickMarks', -1))
         zcustomminortickmarks = int(self.description.get('ZMinorTickMarks', -1))
-        zcustommajorticks=[]
-        zcustomminorticks=[]
+        zcustommajorticks=None
+        zcustomminorticks=None
         if self.description.has_key('ZCustomMajorTicks') and self.description['ZCustomMajorTicks']!='':
             # TODO: Would be nice to have less invisible separation of the custom ticks than split on tabs
             ticks = self.description['ZCustomMajorTicks'].strip().split('\t')
@@ -2214,7 +2214,7 @@ class Ticks(object):
 
 class XTicks(Ticks):
 
-    def draw(self, custommajorticks=[], customminorticks=[], custommajortickmarks=-1, customminortickmarks=-1,drawlabels=True):
+    def draw(self, custommajorticks=None, customminorticks=None, custommajortickmarks=-1, customminortickmarks=-1,drawlabels=True):
         twosided = bool(int(self.description.get('XTwosidedTicks', '0')))
         out = ""
         out += ('\n%\n% X-Ticks\n%\n')
@@ -2265,7 +2265,7 @@ class XTicks(Ticks):
 
 class YTicks(Ticks):
 
-    def draw(self, custommajorticks=[], customminorticks=[], custommajortickmarks=-1, customminortickmarks=-1, drawlabels=True):
+    def draw(self, custommajorticks=None, customminorticks=None, custommajortickmarks=-1, customminortickmarks=-1, drawlabels=True):
         twosided = bool(int(self.description.get('YTwosidedTicks', '0')))
         out = ""
         out += ('\n%\n% Y-Ticks\n%\n')
@@ -2320,7 +2320,7 @@ class ZTicks(Ticks):
         self.description = description
         self.coors = coors
 
-    def draw(self, custommajorticks=[], customminorticks=[],
+    def draw(self, custommajorticks=None, customminorticks=None,
             custommajortickmarks=-1, customminortickmarks=-1,
             drawlabels=True):
         out = ""
