diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e7b71a..9db85b1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -107,6 +107,8 @@ if(USE_SYSTEM_NCNN)

         find_package(Threads)

+        include("${GLSLANG_TARGET_DIR}/SPIRV-Tools/SPIRV-ToolsTarget.cmake")
+        include("${GLSLANG_TARGET_DIR}/SPIRV-Tools-opt/SPIRV-Tools-optTargets.cmake")
         include("${GLSLANG_TARGET_DIR}/OSDependentTargets.cmake")
         include("${GLSLANG_TARGET_DIR}/OGLCompilerTargets.cmake")
         if(EXISTS "${GLSLANG_TARGET_DIR}/HLSLTargets.cmake")
