diff --git a/setup.py b/setup.py
index 59a75b0..0a73268 100644
--- a/setup.py
+++ b/setup.py
@@ -169,6 +169,7 @@ class build_ext(build_ext_base):
                              'Please specify via --pyqt-sip-flags=')
 
         self.include_dirs += (self.qt_include_dir,
+                              '@poppler_include_dir@',
                               os.path.join(self.qt_include_dir, 'QtCore'),
                               os.path.join(self.qt_include_dir, 'QtGui'),
                               os.path.join(self.qt_include_dir, 'QtXml'))
