--- a/gio/unix-types.defs
+++ b/gio/unix-types.defs
@@ -7,18 +7,6 @@
   (gtype-id "G_TYPE_UNIX_CONNECTION")
 )
 
-(define-object DesktopAppInfo
-  (docstring
-  "DesktopAppInfo(desktop_id) -> gio.unix.DesktopAppInfo\n\n"
-  "gio.Unix.DesktopAppInfo is an implementation of gio.AppInfo\n"
-  "based on desktop files."
-  )
-  (in-module "giounix")
-  (parent "GObject")
-  (c-name "GDesktopAppInfo")
-  (gtype-id "G_TYPE_DESKTOP_APP_INFO")
-)
-
 (define-object FDMessage
   (in-module "giounix")
   (parent "GSocketControlMessage")
--- a/gio/unix.defs
+++ b/gio/unix.defs
@@ -32,54 +32,6 @@
 
 
 
-;; From gdesktopappinfo.h
-
-(define-function desktop_app_info_get_type
-  (c-name "g_desktop_app_info_get_type")
-  (return-type "GType")
-)
-
-(define-function desktop_app_info_new_from_filename
-  (c-name "g_desktop_app_info_new_from_filename")
-  (return-type "GDesktopAppInfo*")
-  (parameters
-    '("const-char*" "filename")
-  )
-)
-
-(define-function g_desktop_app_info_new_from_keyfile
-  (c-name "g_desktop_app_info_new_from_keyfile")
-  (return-type "GDesktopAppInfo*")
-  (parameters
-    '("GKeyFile*" "key_file")
-  )
-)
-
-(define-function desktop_app_info_new
-  (c-name "g_desktop_app_info_new")
-  (is-constructor-of "GDesktopAppInfo")
-  (return-type "GDesktopAppInfo*")
-  (parameters
-    '("const-char*" "desktop_id")
-  )
-)
-
-(define-method get_is_hidden
-  (of-object "GDesktopAppInfo")
-  (c-name "g_desktop_app_info_get_is_hidden")
-  (return-type "gboolean")
-)
-
-(define-function desktop_app_info_set_desktop_env
-  (c-name "g_desktop_app_info_set_desktop_env")
-  (return-type "none")
-  (parameters
-    '("const-char*" "desktop_env")
-  )
-)
-
-
-
 ;; From gunixfdmessage.h
 
 (define-function g_unix_fd_message_get_type
--- a/gio/unix.override
+++ b/gio/unix.override
@@ -24,7 +24,6 @@
 #define NO_IMPORT_PYGOBJECT
 #include <pygobject.h>
 #include <gio/gio.h>
-#include <gio/gdesktopappinfo.h>
 #include <gio/gunixinputstream.h>
 #include <gio/gunixmounts.h>
 #include <gio/gunixoutputstream.h>
