diff --git a/configure.pri b/configure.pri
index 897bea540..6f834c202 100644
--- a/configure.pri
+++ b/configure.pri
@@ -269,7 +269,7 @@ defineReplace(webEngineGetMacOSVersion) {
 }
 
 defineReplace(webEngineGetMacOSSDKVersion) {
-    value = $$system("/usr/bin/xcodebuild -sdk $$QMAKE_MAC_SDK -version ProductVersion 2>/dev/null")
+    value = $$system("xcrun --show-sdk-version")
     return($$value)
 }
 
diff --git a/mkspecs/features/platform.prf b/mkspecs/features/platform.prf
index 35eb6b89c..7eed640a5 100644
--- a/mkspecs/features/platform.prf
+++ b/mkspecs/features/platform.prf
@@ -40,8 +40,6 @@ defineTest(isPlatformSupported) {
   } else:osx {
     # FIXME: Try to get it back down to 8.2 for building on OS X 10.11
     !isMinXcodeVersion(8, 3, 3) {
-      skipBuild("Using Xcode version $$QMAKE_XCODE_VERSION, but at least version 8.3.3 is required to build Qt WebEngine.")
-      return(false)
     }
     !clang|intel_icc {
         skipBuild("Qt WebEngine on macOS requires Clang.")
@@ -54,8 +52,6 @@ defineTest(isPlatformSupported) {
       return(false)
     }
     !isMinOSXSDKVersion(10, 12): {
-      skipBuild("Building Qt WebEngine requires a macOS SDK version of 10.12 or newer. Current version is $${WEBENGINE_OSX_SDK_PRODUCT_VERSION}.")
-      return(false)
     }
   } else {
     skipBuild("Unknown platform. Qt WebEngine only supports Linux, Windows, and macOS.")
@@ -111,7 +107,7 @@ defineTest(isMinOSXSDKVersion) {
     requested_minor = $$2
     requested_patch = $$3
     isEmpty(requested_patch): requested_patch = 0
-    WEBENGINE_OSX_SDK_PRODUCT_VERSION = $$system("/usr/bin/xcodebuild -sdk $$QMAKE_MAC_SDK -version ProductVersion 2>/dev/null")
+    WEBENGINE_OSX_SDK_PRODUCT_VERSION = $$system("xcrun --show-sdk-version")
     export(WEBENGINE_OSX_SDK_PRODUCT_VERSION)
     isEmpty(WEBENGINE_OSX_SDK_PRODUCT_VERSION) {
         skipBuild("Could not resolve SDK product version for \'$$QMAKE_MAC_SDK\'.")
diff --git a/src/core/config/mac_osx.pri b/src/core/config/mac_osx.pri
index 7b77a8bf7..0e1284ee4 100644
--- a/src/core/config/mac_osx.pri
+++ b/src/core/config/mac_osx.pri
@@ -5,7 +5,7 @@ load(functions)
 # otherwise query for it.
 QMAKE_MAC_SDK_VERSION = $$eval(QMAKE_MAC_SDK.$${QMAKE_MAC_SDK}.SDKVersion)
 isEmpty(QMAKE_MAC_SDK_VERSION) {
-     QMAKE_MAC_SDK_VERSION = $$system("/usr/bin/xcodebuild -sdk $${QMAKE_MAC_SDK} -version SDKVersion 2>/dev/null")
+     QMAKE_MAC_SDK_VERSION = $$system("xcrun --show-sdk-version")
      isEmpty(QMAKE_MAC_SDK_VERSION): error("Could not resolve SDK version for \'$${QMAKE_MAC_SDK}\'")
 }
 
@@ -14,11 +14,6 @@ isEmpty(QMAKE_MAC_SDK_VERSION) {
 QMAKE_MAC_SDK_VERSION_MAJOR_MINOR = $$section(QMAKE_MAC_SDK_VERSION, ".", 0, 1)
 
 QMAKE_CLANG_DIR = "/usr"
-QMAKE_CLANG_PATH = $$eval(QMAKE_MAC_SDK.macx-clang.$${QMAKE_MAC_SDK}.QMAKE_CXX)
-!isEmpty(QMAKE_CLANG_PATH) {
-    clang_dir = $$clean_path("$$dirname(QMAKE_CLANG_PATH)/../")
-    exists($$clang_dir): QMAKE_CLANG_DIR = $$clang_dir
-}
 
 QMAKE_CLANG_PATH = "$${QMAKE_CLANG_DIR}/bin/clang++"
 message("Using clang++ from $${QMAKE_CLANG_PATH}")
