diff --git a/CMakeLists.txt b/CMakeLists.txt
index 70d89efb5e..47ba750d7a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -131,7 +131,7 @@ set(Octopus_ext_libs)
 find_package(GSL)
 if (OCTOPUS_MPI)
 	set(MPI_DETERMINE_LIBRARY_VERSION ON)
-	find_package(MPI 3 COMPONENTS Fortran)
+	find_package(MPI 3)
 	# Further set MKL variables
 	if (MPI_Fortran_LIBRARY_VERSION_STRING MATCHES Intel)
 		set(MKL_MPI intelmpi CACHE STRING "Octopus: Overloaded")
