diff --git a/src/MainView.vala b/src/MainView.vala
index 1654e68..ad8fed9 100644
--- a/src/MainView.vala
+++ b/src/MainView.vala
@@ -46,8 +46,13 @@ public class Power.MainView : Gtk.Grid {

         var label_size = new Gtk.SizeGroup (Gtk.SizeGroupMode.HORIZONTAL);

-        settings = new GLib.Settings ("org.gnome.settings-daemon.plugins.power");
-        elementary_dpms_settings = new GLib.Settings ("io.elementary.dpms");
+        SettingsSchemaSource gsd_sss = new SettingsSchemaSource.from_directory ("@GSD_GSETTINGS_PATH@", null, true);
+        SettingsSchema gsd_schema = gsd_sss.lookup ("org.gnome.settings-daemon.plugins.power", false);
+        settings = new GLib.Settings.full (gsd_schema, null, null);
+
+        SettingsSchemaSource dpms_sss = new SettingsSchemaSource.from_directory ("@DPMS_HELPER_GSETTINGS_PATH@", null, true);
+        SettingsSchema elementary_dpms_schema = dpms_sss.lookup ("io.elementary.dpms", false);
+        elementary_dpms_settings = new GLib.Settings.full (elementary_dpms_schema, null, null);

         battery = new Battery ();
         power_supply = new PowerSupply ();
