--- a/cmake/FindFFmpeg.cmake
+++ b/cmake/FindFFmpeg.cmake
@@ -276,7 +276,7 @@ function(__ffmpeg_internal_set_dependencies _component)
     if (WIN32)
         set(PC_FILE ${${_component}_LIBRARY_DIR}/../lib/pkgconfig/lib${lib}.pc)
     else()
-        set(PC_FILE ${${_component}_LIBRARY_DIR}/pkgconfig/lib${lib}.pc)
+        set(PC_FILE @ffmpeg-lib@/lib/pkgconfig/lib${lib}.pc)
     endif()
 
     if(EXISTS ${PC_FILE})
