From c26c7670869e02c39c0d0f6dee9094a621f42252 Mon Sep 17 00:00:00 2001
From: "Luke A. Guest" <laguest@archeia.com>
Date: Sun, 25 Sep 2016 15:31:49 +0100
Subject: [PATCH 03/14] Add Gentoo as build option

---
 amd/backport/Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/amd/backport/Makefile b/amd/backport/Makefile
index 0c285ef..dcc94d5 100644
--- a/amd/backport/Makefile
+++ b/amd/backport/Makefile
@@ -17,8 +17,13 @@ else ifeq ("ubuntu",$(OS_NAME))
 ccflags-y += -DOS_NAME_UBUNTU
 else ifeq ("steamos",$(OS_NAME))
 ccflags-y += -DOS_NAME_STEAMOS
+else ifeq ("gentoo",$(OS_NAME))
+ccflags-y += -DOS_NAME_GENTOO
+# We don't have a version inside /etc/os-release.
+OS_VERSION = "0.0"
 else
 ccflags-y += -DOS_NAME_UNKNOWN
+OS_VERSION = "0.0"
 endif
 
 ccflags-y += \
-- 
2.10.1

