diff --git a/src/lib_java/CMakeLists.txt b/src/lib_java/CMakeLists.txt
index 2249adb..00395c1 100644
--- a/src/lib_java/CMakeLists.txt
+++ b/src/lib_java/CMakeLists.txt
@@ -61,29 +61,3 @@ endif()
 if (NOT BUILD_UNIT_TESTS_PACKAGE)
 	set(skipTestsSwitch "-Dmaven.test.skip=true")
 endif()
-
-add_custom_command(
-	COMMAND ${CMAKE_COMMAND} -E echo "Maven: Building Java indexer"
-	
-	TARGET Sourcetrail_lib_java
-	PRE_BUILD
-	WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}/java_indexer/"
-	COMMAND "${MVN_COMMAND}" package ${quietSwitch} ${skipTestsSwitch}
-	BYPRODUCTS "./target/java-indexer.jar"
-)
-# Copy java indexer jars
-# TODO: This also copies unneeded maven directories (classes, generated-sources, ...)
-add_custom_command(
-	TARGET Sourcetrail_lib_java
-	POST_BUILD
-	WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-	COMMAND ${CMAKE_COMMAND} -E copy_directory_if_different "java_indexer/target/"  "${CMAKE_BINARY_DIR}/app/data/java/lib/"
-)
-
-# Delete unneeded/unwanted maven directories as a workaround for the above copy problem:
-add_custom_command(
-	TARGET Sourcetrail_lib_java
-	POST_BUILD
-	WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/app/data/java/lib/"
-	COMMAND ${CMAKE_COMMAND} -E rm -rf "classes" "generated-sources" "maven-archiver" "maven-status" "test-classes"
-)
