--- a/src/procmime.c	2015-10-01 23:02:16.629908590 -0700
+++ b/src/procmime.c	2015-10-01 23:02:46.932001337 -0700
@@ -1196,11 +1196,7 @@
 	if (mime_type_list) 
 		return mime_type_list;
 	
-#if defined(__NetBSD__) || defined(__OpenBSD__) || defined(__FreeBSD__)
-	if ((fp = procmime_fopen(DATAROOTDIR "/mime/globs", "rb")) == NULL) 
-#else
-	if ((fp = procmime_fopen("/usr/share/mime/globs", "rb")) == NULL) 
-#endif
+	if ((fp = procmime_fopen("@MIMEROOTDIR@/mime/globs", "rb")) == NULL) 
 	{
 		fp_is_glob_file = FALSE;
 		if ((fp = procmime_fopen("/etc/mime.types", "rb")) == NULL) {
