diff --git a/green/test/test_integration.py b/green/test/test_integration.py
index 0cf8cbb..40d9f6b 100644
--- a/green/test/test_integration.py
+++ b/green/test/test_integration.py
@@ -41,9 +41,7 @@ class TestFinalizer(unittest.TestCase):
             )
             fh.close()
         args = [
-            sys.executable,
-            "-m",
-            "green.cmdline",
+            "@green@",
             "--finalizer=test_finalizer0.msg",
             "--maxtasksperchild=1",
         ]
@@ -58,6 +56,5 @@ class TestFinalizer(unittest.TestCase):
             stdout=subprocess.PIPE,
             stderr=subprocess.STDOUT,
             env=env,
-            timeout=10,
         ).stdout.decode("utf-8")
         self.assertIn("finalizer worked", output)
