diff -Naur clang-3.4-orig/lib/Driver/Tools.cpp clang-3.4/lib/Driver/Tools.cpp
--- clang-3.4-orig/lib/Driver/Tools.cpp	2013-12-08 21:59:27.000000000 -0500
+++ clang-3.4/lib/Driver/Tools.cpp	2014-01-27 12:48:29.421872597 -0500
@@ -6477,15 +6477,6 @@
     }
   }
 
-  if (ToolChain.getArch() == llvm::Triple::arm ||
-      ToolChain.getArch() == llvm::Triple::thumb ||
-      (!Args.hasArg(options::OPT_static) &&
-       !Args.hasArg(options::OPT_shared))) {
-    CmdArgs.push_back("-dynamic-linker");
-    CmdArgs.push_back(Args.MakeArgString(
-        D.DyldPrefix + getLinuxDynamicLinker(Args, ToolChain)));
-  }
-
   CmdArgs.push_back("-o");
   CmdArgs.push_back(Output.getFilename());
 
