diff --git a/CMakeLists.txt b/CMakeLists.txt
index 12fef27a..39ff1974 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,12 +58,8 @@ list(GET VTK_INCLUDE_DIR 0 VTK_INCLUDE_DIR)
 message( STATUS "Include dirs: ${VTK_INCLUDE_DIR}")
 message( STATUS "Include dirs: ${CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES}")
 
-# OSX
-if(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
-    list( APPEND CXX_INCLUDES -i /opt/usr/local/include/c++/v1/ -i /opt/usr/local/include/ )
-
-# Linux-like
-elseif(NOT PLATFORM STREQUAL "Windows")
+# Unix-like
+if(NOT PLATFORM STREQUAL "Windows")
     foreach( inc IN LISTS CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES)
         list( APPEND CXX_INCLUDES -i ${inc}/ )
     endforeach()
