diff --git a/CMakeLists.txt b/CMakeLists.txt
index 50e712fa..d39c2ce4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -326,9 +326,6 @@ endif()
 # Apple has to use static libraries because
 # "Use of the Apple-provided OpenSSL libraries by apps is strongly discouraged."
 # https://developer.apple.com/library/archive/documentation/Security/Conceptual/cryptoservices/SecureNetworkCommunicationAPIs/SecureNetworkCommunicationAPIs.html
-if(APPLE)
-    set(OPENSSL_USE_STATIC_LIBS TRUE)
-endif()
 find_package(OpenSSL REQUIRED)
 
 #
diff --git a/src/gui/src/OSXHelpers.mm b/src/gui/src/OSXHelpers.mm
index 0c98afc1..38c190a6 100644
--- a/src/gui/src/OSXHelpers.mm
+++ b/src/gui/src/OSXHelpers.mm
@@ -20,10 +20,6 @@
 #import <Foundation/Foundation.h>
 #import <CoreData/CoreData.h>
 #import <Cocoa/Cocoa.h>
-#import <UserNotifications/UNNotification.h>
-#import <UserNotifications/UNUserNotificationCenter.h>
-#import <UserNotifications/UNNotificationContent.h>
-#import <UserNotifications/UNNotificationTrigger.h>
 
 #import <QtGlobal>
 
