--- a/src/lib/net/SecureSocket.cpp	2017-07-22 19:33:22.442645291 +0200
+++ b/src/lib/net/SecureSocket.cpp	2017-07-22 19:36:25.632595581 +0200
@@ -805,9 +805,14 @@
 		showCipherStackDesc(sStack);
 	}
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
 	// m_ssl->m_ssl->session->ciphers is not forward compatable, In future release
-	// of OpenSSL, it's not visible, need to use SSL_get_client_ciphers() instead
+	// of OpenSSL, it's not visible
 	STACK_OF(SSL_CIPHER) * cStack = m_ssl->m_ssl->session->ciphers;
+#else
+	// Use SSL_get_client_ciphers() for newer versions
+	STACK_OF(SSL_CIPHER) * cStack = SSL_get_client_ciphers(m_ssl->m_ssl);
+#endif
 		if (cStack == NULL) {
 		LOG((CLOG_DEBUG1 "remote cipher list not available"));
 	}
