diff --git a/data/meson.build b/data/meson.build
index f10d4ff6..2dc66fb6 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -21,7 +21,7 @@ endif
 
 if build_standalone
   install_data(['daemon.conf'],
-    install_dir : join_paths(sysconfdir, 'fwupd')
+    install_dir : join_paths(sysconfdir_install, 'fwupd')
   )
   install_data(['power.quirk'],
     install_dir: join_paths(datadir, 'fwupd', 'quirks.d'))
diff --git a/data/pki/meson.build b/data/pki/meson.build
index dcd9e128..ff137f3a 100644
--- a/data/pki/meson.build
+++ b/data/pki/meson.build
@@ -2,24 +2,23 @@ install_data([
     'GPG-KEY-Linux-Foundation-Firmware',
     'GPG-KEY-Linux-Vendor-Firmware-Service',
   ],
-  install_dir : join_paths(sysconfdir, 'pki', 'fwupd')
+  install_dir : join_paths(sysconfdir_install, 'pki', 'fwupd')
 )
 
 install_data([
     'GPG-KEY-Linux-Foundation-Metadata',
     'GPG-KEY-Linux-Vendor-Firmware-Service',
   ],
-  install_dir : join_paths(sysconfdir, 'pki', 'fwupd-metadata')
+  install_dir : join_paths(sysconfdir_install, 'pki', 'fwupd-metadata')
 )
 
 install_data([
     'LVFS-CA.pem',
   ],
-  install_dir : join_paths(sysconfdir, 'pki', 'fwupd')
+  install_dir : join_paths(sysconfdir_install, 'pki', 'fwupd')
 )
 install_data([
     'LVFS-CA.pem',
   ],
-  install_dir : join_paths(sysconfdir, 'pki', 'fwupd-metadata')
+  install_dir : join_paths(sysconfdir_install, 'pki', 'fwupd-metadata')
 )
-
diff --git a/data/remotes.d/meson.build b/data/remotes.d/meson.build
index 02d8777b..2c89d593 100644
--- a/data/remotes.d/meson.build
+++ b/data/remotes.d/meson.build
@@ -2,7 +2,7 @@ if build_standalone and get_option('lvfs') != 'false'
   install_data([
       'lvfs-testing.conf',
     ],
-    install_dir : join_paths(sysconfdir, 'fwupd', 'remotes.d')
+    install_dir : join_paths(sysconfdir_install, 'fwupd', 'remotes.d')
   )
   con3 = configuration_data()
   if get_option('lvfs') == 'disabled'
@@ -15,7 +15,7 @@ if build_standalone and get_option('lvfs') != 'false'
     output : 'lvfs.conf',
     configuration : con3,
     install: true,
-    install_dir: join_paths(sysconfdir, 'fwupd', 'remotes.d'),
+    install_dir: join_paths(sysconfdir_install, 'fwupd', 'remotes.d'),
   )
   i18n.merge_file(
     input: 'lvfs.metainfo.xml',
@@ -49,12 +49,12 @@ configure_file(
   output : 'vendor.conf',
   configuration : con2,
   install: true,
-  install_dir: join_paths(sysconfdir, 'fwupd', 'remotes.d'),
+  install_dir: join_paths(sysconfdir_install, 'fwupd', 'remotes.d'),
 )
 configure_file(
   input : 'vendor-directory.conf',
   output : 'vendor-directory.conf',
   configuration : con2,
   install: true,
-  install_dir: join_paths(sysconfdir, 'fwupd', 'remotes.d'),
+  install_dir: join_paths(sysconfdir_install, 'fwupd', 'remotes.d'),
 )
diff --git a/meson.build b/meson.build
index 7557cd4a..4c98de6d 100644
--- a/meson.build
+++ b/meson.build
@@ -190,6 +190,12 @@ endif
 mandir = join_paths(prefix, get_option('mandir'))
 localedir = join_paths(prefix, get_option('localedir'))
 
+if get_option('sysconfdir_install') != ''
+  sysconfdir_install = join_paths(prefix, get_option('sysconfdir_install'))
+else
+  sysconfdir_install = sysconfdir
+endif
+
 diffcmd = find_program('diff')
 gio = dependency('gio-2.0', version : '>= 2.45.8')
 giounix = dependency('gio-unix-2.0', version : '>= 2.45.8', required: false)
diff --git a/meson_options.txt b/meson_options.txt
index 94f73898..5de6cc7b 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,3 +1,4 @@
+option('sysconfdir_install', type: 'string', value: '', description: 'sysconfdir to use during installation')
 option('build', type : 'combo', choices : ['all', 'standalone', 'library'], value : 'all', description : 'build type')
 option('consolekit', type : 'boolean', value : true, description : 'enable ConsoleKit support')
 option('firmware-packager', type : 'boolean', value : true, description : 'enable firmware-packager installation')
diff --git a/plugins/dell-esrt/meson.build b/plugins/dell-esrt/meson.build
index e9f12879..a0126dbb 100644
--- a/plugins/dell-esrt/meson.build
+++ b/plugins/dell-esrt/meson.build
@@ -38,6 +38,6 @@ configure_file(
   output : 'dell-esrt.conf',
   configuration : con2,
   install: true,
-  install_dir: join_paths(sysconfdir, 'fwupd', 'remotes.d'),
+  install_dir: join_paths(sysconfdir_install, 'fwupd', 'remotes.d'),
 )
 endif
diff --git a/plugins/redfish/meson.build b/plugins/redfish/meson.build
index 5263048c..c46fac64 100644
--- a/plugins/redfish/meson.build
+++ b/plugins/redfish/meson.build
@@ -53,7 +53,7 @@ shared_module('fu_plugin_redfish',
 )
 
 install_data(['redfish.conf'],
-  install_dir:  join_paths(sysconfdir, 'fwupd'),
+  install_dir:  join_paths(sysconfdir_install, 'fwupd'),
   install_mode: 'rw-r-----',
 )
 
diff --git a/plugins/thunderbolt/meson.build b/plugins/thunderbolt/meson.build
index 646ed6cb..8d7c59aa 100644
--- a/plugins/thunderbolt/meson.build
+++ b/plugins/thunderbolt/meson.build
@@ -35,7 +35,7 @@ fu_plugin_thunderbolt = shared_module('fu_plugin_thunderbolt',
 )
 
 install_data(['thunderbolt.conf'],
-  install_dir:  join_paths(sysconfdir, 'fwupd')
+  install_dir:  join_paths(sysconfdir_install, 'fwupd')
 )
 # we use functions from 2.52 in the tests
 if get_option('tests') and umockdev.found() and gio.version().version_compare('>= 2.52')
diff --git a/plugins/uefi-capsule/meson.build b/plugins/uefi-capsule/meson.build
index 708586b0..f29536b1 100644
--- a/plugins/uefi-capsule/meson.build
+++ b/plugins/uefi-capsule/meson.build
@@ -21,7 +21,7 @@ if host_machine.system() == 'linux'
     output : '35_fwupd',
     configuration : con2,
     install: true,
-    install_dir: join_paths(sysconfdir, 'grub.d')
+    install_dir: join_paths(sysconfdir_install, 'grub.d')
   )
 elif host_machine.system() == 'freebsd'
   backend_srcs += 'fu-uefi-backend-freebsd.c'
@@ -116,7 +116,7 @@ if get_option('man')
 endif
 
 install_data(['uefi_capsule.conf'],
-  install_dir:  join_paths(sysconfdir, 'fwupd')
+  install_dir:  join_paths(sysconfdir_install, 'fwupd')
 )
 
 # add all the .po files as inputs to watch
