diff --git a/meson.build b/meson.build
index 4330512e..e53b70ab 100644
--- a/meson.build
+++ b/meson.build
@@ -403,7 +403,7 @@ endif
 if build_standalone and get_option('plugin_uefi_capsule')
   efiboot = dependency('efiboot')
 
-  efi_app_location = join_paths(libexecdir, 'fwupd', 'efi')
+  efi_app_location = join_paths(dependency('fwupd-efi').get_pkgconfig_variable('prefix'), 'libexec', 'fwupd', 'efi')
   conf.set_quoted('EFI_APP_LOCATION', efi_app_location)
 
   if host_cpu == 'x86'
