--- a/data/builder/meson.build
+++ b/data/builder/meson.build
@@ -1,3 +0,0 @@
-install_data('README.md',
-  install_dir : join_paths(localstatedir, 'lib', 'fwupd', 'builder')
-)
--- a/data/meson.build
+++ b/data/meson.build
@@ -7,16 +7,12 @@
   subdir('installed-tests')
 endif
 
-install_data(['daemon.conf'],
-  install_dir : join_paths(sysconfdir, 'fwupd')
-)
-
 install_data(['org.freedesktop.fwupd.metainfo.xml'],
   install_dir: join_paths(datadir, 'metainfo')
 )
 
 install_data(['org.freedesktop.fwupd.conf'],
-  install_dir : join_paths(sysconfdir, 'dbus-1', 'system.d')
+  install_dir : join_paths(get_option('prefix'), 'etc', 'dbus-1', 'system.d')
 )
 
 install_data(['metadata.xml'],
--- a/data/pki/meson.build
+++ b/data/pki/meson.build
@@ -3,13 +3,13 @@
       'GPG-KEY-Hughski-Limited',
       'GPG-KEY-Linux-Vendor-Firmware-Service',
     ],
-    install_dir : join_paths(sysconfdir, 'pki', 'fwupd')
+    install_dir : join_paths(get_option('prefix'), 'etc', 'pki', 'fwupd')
   )
 
   install_data([
       'GPG-KEY-Linux-Vendor-Firmware-Service',
     ],
-    install_dir : join_paths(sysconfdir, 'pki', 'fwupd-metadata')
+    install_dir : join_paths(get_option('prefix'), 'etc', 'pki', 'fwupd-metadata')
   )
 endif
 
@@ -17,12 +17,12 @@
   install_data([
       'LVFS-CA.pem',
     ],
-    install_dir : join_paths(sysconfdir, 'pki', 'fwupd')
+    install_dir : join_paths(get_option('prefix'), 'etc', 'pki', 'fwupd')
   )
   install_data([
       'LVFS-CA.pem',
     ],
-    install_dir : join_paths(sysconfdir, 'pki', 'fwupd-metadata')
+    install_dir : join_paths(get_option('prefix'), 'etc', 'pki', 'fwupd-metadata')
   )
 endif
 
--- a/data/remotes.d/meson.build
+++ b/data/remotes.d/meson.build
@@ -3,7 +3,7 @@
       'lvfs.conf',
       'lvfs-testing.conf',
     ],
-    install_dir : join_paths(sysconfdir, 'fwupd', 'remotes.d')
+    install_dir : join_paths(get_option('prefix'), 'etc', 'fwupd', 'remotes.d')
   )
 endif
 
@@ -19,12 +19,12 @@
   output : 'fwupd.conf',
   configuration : con2,
   install: true,
-  install_dir: join_paths(sysconfdir, 'fwupd', 'remotes.d'),
+  install_dir: join_paths(get_option('prefix'), 'etc', 'fwupd', 'remotes.d'),
 )
 configure_file(
   input : 'vendor.conf',
   output : 'vendor.conf',
   configuration : con2,
   install: true,
-  install_dir: join_paths(sysconfdir, 'fwupd', 'remotes.d'),
+  install_dir: join_paths(get_option('prefix'), 'etc', 'fwupd', 'remotes.d'),
 )
--- a/meson_post_install.sh
+++ b/meson_post_install.sh
@@ -11,6 +11,4 @@
     echo 'Updating systemd deps'
     mkdir -p ${DESTDIR}${SYSTEMDUNITDIR}/system-update.target.wants
     ln -sf ../fwupd-offline-update.service ${DESTDIR}${SYSTEMDUNITDIR}/system-update.target.wants/fwupd-offline-update.service
-    echo 'Creating stateful directory'
-    mkdir -p ${DESTDIR}${LOCALSTATEDIR}/lib/fwupd
 #fi
--- a/po/make-images.sh
+++ b/po/make-images.sh
@@ -7,6 +7,7 @@
 #
 install -m 0755 -d ${MESON_INSTALL_DESTDIR_PREFIX}/share/locale/
 ${MESON_SOURCE_ROOT}/po/make-images "Installing firmware update…" ${MESON_INSTALL_DESTDIR_PREFIX}/share/locale/ ${MESON_SOURCE_ROOT}/po/LINGUAS
+shopt -s nullglob
 for x in ${MESON_INSTALL_DESTDIR_PREFIX}/share/locale/*/LC_IMAGES/*.bmp ; do
     gzip -f ${x}
 done
