--- a/data/device-tests/hardware.py
+++ b/data/device-tests/hardware.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!@installedTestsPython@/bin/python3
 # pylint: disable=wrong-import-position,too-many-locals,unused-argument,wrong-import-order
 #
 # Copyright (C) 2017 Richard Hughes <richard@hughsie.com>
--- a/data/installed-tests/meson.build
+++ b/data/installed-tests/meson.build
@@ -1,4 +1,4 @@
-installed_test_datadir = join_paths(datadir, 'installed-tests', 'fwupd')
+installed_test_datadir = join_paths(get_option('installed_test_prefix'), 'share', 'installed-tests', 'fwupd')
 
 con2 = configuration_data()
 con2.set('installedtestsdir', installed_test_datadir)
@@ -52,5 +52,5 @@ configure_file(
   output : 'fwupd-tests.conf',
   configuration : con2,
   install: true,
-  install_dir: join_paths(sysconfdir, 'fwupd', 'remotes.d'),
+  install_dir: join_paths(get_option('installed_test_prefix'), 'etc', 'fwupd', 'remotes.d'),
 )
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,3 +1,4 @@
+option('installed_test_prefix', type: 'string', value: '', description: 'Prefix for installed tests')
 option('build', type : 'combo', choices : ['all', 'standalone', 'library'], value : 'all', description : 'build type')
 option('agent', type : 'boolean', value : true, description : 'enable the fwupd agent')
 option('consolekit', type : 'boolean', value : true, description : 'enable ConsoleKit support')
