diff --git a/kded/engine/backends/encfs/encfsbackend.cpp b/kded/engine/backends/encfs/encfsbackend.cpp
index 628af7b..6edd38e 100644
--- a/kded/engine/backends/encfs/encfsbackend.cpp
+++ b/kded/engine/backends/encfs/encfsbackend.cpp
@@ -137,7 +137,7 @@ QProcess *EncFsBackend::encfs(const QStringList &arguments) const
     auto config = KSharedConfig::openConfig(PLASMAVAULT_CONFIG_FILE);
     KConfigGroup backendConfig(config, "EncfsBackend");
 
-    return process("encfs",
+    return process(NIXPKGS_ENCFS,
                    arguments + backendConfig.readEntry("extraMountOptions", QStringList{}),
                    {});
 }
@@ -146,7 +146,7 @@ QProcess *EncFsBackend::encfs(const QStringList &arguments) const
 
 QProcess *EncFsBackend::encfsctl(const QStringList &arguments) const
 {
-    return process("encfsctl", arguments, {});
+    return process(NIXPKGS_ENCFSCTL, arguments, {});
 }
 
 
