diff --git a/kded/engine/fusebackend_p.cpp b/kded/engine/fusebackend_p.cpp
index d07e313..ea7d81c 100644
--- a/kded/engine/fusebackend_p.cpp
+++ b/kded/engine/fusebackend_p.cpp
@@ -106,7 +106,7 @@ QProcess *FuseBackend::process(const QString &executable,
 
 QProcess *FuseBackend::fusermount(const QStringList &arguments) const
 {
-    return process("fusermount", arguments, {});
+    return process(NIXPKGS_FUSERMOUNT, arguments, {});
 }
 
 
@@ -279,4 +279,3 @@ bool FuseBackend::isOpened(const MountPoint &mountPoint) const
 }
 
 } // namespace PlasmaVault
-
