diff --git a/cli/src/lib.rs b/cli/src/lib.rs
index 33a9904f..633032d7 100644
--- a/cli/src/lib.rs
+++ b/cli/src/lib.rs
@@ -8,7 +8,7 @@ pub mod parse;
 pub mod test;
 pub mod util;
 pub mod wasm;
-pub mod web_ui;
+// pub mod web_ui;
 
 #[cfg(test)]
 mod tests;
diff --git a/cli/src/main.rs b/cli/src/main.rs
index 23e7fc1a..9d784c8a 100644
--- a/cli/src/main.rs
+++ b/cli/src/main.rs
@@ -4,7 +4,7 @@ use std::{env, fs, u64};
 use std::path::Path;
 use std::process::exit;
 use tree_sitter_cli::{
-    config, error, generate, highlight, loader, logger, parse, test, wasm, web_ui,
+    config, error, generate, highlight, loader, logger, parse, test, wasm,
 };
 
 const BUILD_VERSION: &'static str = env!("CARGO_PKG_VERSION");
@@ -250,7 +250,9 @@ fn run() -> error::Result<()> {
         let grammar_path = current_dir.join(matches.value_of("path").unwrap_or(""));
         wasm::compile_language_to_wasm(&grammar_path, matches.is_present("docker"))?;
     } else if matches.subcommand_matches("web-ui").is_some() {
-        web_ui::serve(&current_dir);
+        print!("ERROR: web-ui is not available in the nixpkgs tree-sitter-cli at the moment.");
+        std::process::exit(1);
+        // web_ui::serve(&current_dir);
     }
 
     Ok(())
