diff --git a/cli/loader/src/lib.rs b/cli/loader/src/lib.rs
index 9c1d8dfc..a5cfc74c 100644
--- a/cli/loader/src/lib.rs
+++ b/cli/loader/src/lib.rs
@@ -747,7 +747,7 @@ impl Loader {
             Podman,
         }
 
-        let emcc_name = if cfg!(windows) { "emcc.bat" } else { "emcc" };
+        let emcc_name = if cfg!(windows) { "emcc.bat" } else { "@emcc@" };
 
         // Order of preference: emscripten > docker > podman > error
         let source = if !force_docker && Command::new(emcc_name).output().is_ok() {
