diff --git a/crates/cli/tests/ui_with_jinja.rs b/crates/cli/tests/ui_with_jinja.rs
index 7c39f8a1..e4c96d41 100644
--- a/crates/cli/tests/ui_with_jinja.rs
+++ b/crates/cli/tests/ui_with_jinja.rs
@@ -4,6 +4,10 @@ use std::path::PathBuf;
 use assert_cmd::Command;
 use expect_test::expect_file;
 
+#[cfg(not(feature = "python"))]
+fn main() {}
+
+#[cfg(feature = "python")]
 fn main() {
     let mut test_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
     test_dir.push("tests/jinja");
