diff --git a/CMakeLists.txt b/CMakeLists.txt
index a3816f7..81116dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,8 +19,8 @@ set(VERSION 1.2.2)
 set(QTVERSION 5.12)
 set(CMAKE_AUTOMOC ON)
 
-cmake_minimum_required(VERSION 2.8.12)
-cmake_policy(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.10)
+cmake_policy(VERSION 3.10)
 
 find_package(Qt5 ${QTVERSION} REQUIRED COMPONENTS Core Widgets PrintSupport Network LinguistTools)
 
