diff --git a/pkgs/development/libraries/cogl/default.nix b/pkgs/development/libraries/cogl/default.nix
index abb92f7..c1c57a6 100644
--- a/pkgs/development/libraries/cogl/default.nix
+++ b/pkgs/development/libraries/cogl/default.nix
@@ -3,11 +3,11 @@
 , pangoSupport ? true, pango, cairo }:
 
 stdenv.mkDerivation rec {
-  name = "cogl-1.8.2";
+  name = "cogl-1.9.6";
 
   src = fetchurl {
-    url = mirror://gnome/sources/cogl/1.8/cogl-1.8.2.tar.xz;
-    sha256 = "1ix87hz3qxqysqwx58wbc46lzchlmfs08fjzbf3l6mmsqj8gs9pc";
+    url = mirror://gnome/sources/cogl/1.9/cogl-1.9.6.tar.xz;
+    sha256 = "0pq9ah8yqyradinrgcfihw4cb06k1a5p7al9na4rrx63vgmin8s3";
   };
 
