diff --git a/setup.py b/setup.py
index e3871a7..930bd6a 100755
--- a/setup.py
+++ b/setup.py
@@ -5,7 +5,7 @@ import yowsup
 import platform
 import sys
 
-deps = ['consonance==0.1.2', 'argparse', 'python-axolotl==0.2.2', 'six==1.10', 'appdirs', 'protobuf>=3.6.0']
+deps = ['consonance>=0.1', 'python-axolotl>=0.2', 'six>=1.10', 'appdirs', 'protobuf>=3.6.0']
 
 if sys.version_info < (2, 7):
     deps.append('importlib')
