Index: kdepim-apps-libs-17.04.0/kaddressbookgrantlee/src/formatter/grantleecontactgroupformatter.cpp
===================================================================
--- kdepim-apps-libs-17.04.0.orig/kaddressbookgrantlee/src/formatter/grantleecontactgroupformatter.cpp
+++ kdepim-apps-libs-17.04.0/kaddressbookgrantlee/src/formatter/grantleecontactgroupformatter.cpp
@@ -50,9 +50,9 @@ public:
         mTemplateLoader.clear();
     }
 
-    void changeGrantleePath(const QString &path)
+    void changeGrantleePath(const QStringList &paths)
     {
-        mTemplateLoader->setTemplateDirs(QStringList() << path);
+        mTemplateLoader->setTemplateDirs(paths);
         mEngine->addTemplateLoader(mTemplateLoader);
 
         mSelfcontainedTemplate = mEngine->loadByName(QStringLiteral("contactgroup.html"));
@@ -86,12 +86,12 @@ GrantleeContactGroupFormatter::~Grantlee
 
 void GrantleeContactGroupFormatter::setAbsoluteThemePath(const QString &path)
 {
-    d->changeGrantleePath(path);
+    d->changeGrantleePath(QStringList(path));
 }
 
 void GrantleeContactGroupFormatter::setGrantleeTheme(const GrantleeTheme::Theme &theme)
 {
-    d->changeGrantleePath(theme.absolutePath());
+    d->changeGrantleePath(theme.absolutePaths());
 }
 
 inline static void setHashField(QVariantHash &hash, const QString &name, const QString &value)
Index: kdepim-apps-libs-17.04.0/kaddressbookgrantlee/src/formatter/grantleecontactformatter.cpp
===================================================================
--- kdepim-apps-libs-17.04.0.orig/kaddressbookgrantlee/src/formatter/grantleecontactformatter.cpp
+++ kdepim-apps-libs-17.04.0/kaddressbookgrantlee/src/formatter/grantleecontactformatter.cpp
@@ -74,9 +74,9 @@ public:
         mTemplateLoader.clear();
     }
 
-    void changeGrantleePath(const QString &path)
+    void changeGrantleePath(const QStringList &paths)
     {
-        mTemplateLoader->setTemplateDirs(QStringList() << path);
+        mTemplateLoader->setTemplateDirs(paths);
         mEngine->addTemplateLoader(mTemplateLoader);
 
         mSelfcontainedTemplate = mEngine->loadByName(QStringLiteral("contact.html"));
@@ -112,12 +112,12 @@ GrantleeContactFormatter::~GrantleeConta
 
 void GrantleeContactFormatter::setAbsoluteThemePath(const QString &path)
 {
-    d->changeGrantleePath(path);
+    d->changeGrantleePath(QStringList(path));
 }
 
 void GrantleeContactFormatter::setGrantleeTheme(const GrantleeTheme::Theme &theme)
 {
-    d->changeGrantleePath(theme.absolutePath());
+    d->changeGrantleePath(theme.absolutePaths());
 }
 
 void GrantleeContactFormatter::setForceDisableQRCode(bool b)
