diff --git a/cmake/MujocoDependencies.cmake b/cmake/MujocoDependencies.cmake
<<<<<<< HEAD
index 3e322ea..980aace 100644
--- a/cmake/MujocoDependencies.cmake
+++ b/cmake/MujocoDependencies.cmake
@@ -87,8 +87,6 @@ set(BUILD_SHARED_LIBS
=======
index 99e4a7a..cf9a901 100644
--- a/cmake/MujocoDependencies.cmake
+++ b/cmake/MujocoDependencies.cmake
@@ -82,8 +82,6 @@ set(BUILD_SHARED_LIBS
>>>>>>> 903308adb4b (Improved error handling, differentiate nix/non-nix networks)
 if(NOT TARGET lodepng)
   FetchContent_Declare(
     lodepng
-    GIT_REPOSITORY https://github.com/lvandeve/lodepng.git
-    GIT_TAG ${MUJOCO_DEP_VERSION_lodepng}
   )
 
   FetchContent_GetProperties(lodepng)
<<<<<<< HEAD
@@ -127,10 +125,6 @@ findorfetch(
=======
@@ -111,10 +109,6 @@ findorfetch(
>>>>>>> 903308adb4b (Improved error handling, differentiate nix/non-nix networks)
   qhull
   LIBRARY_NAME
   qhull
-  GIT_REPO
-  https://github.com/qhull/qhull.git
-  GIT_TAG
-  ${MUJOCO_DEP_VERSION_qhull}
   TARGETS
   qhull
<<<<<<< HEAD
   EXCLUDE_FROM_ALL
@@ -151,10 +145,6 @@ findorfetch(
=======
   # TODO(fraromano) Remove when https://github.com/qhull/qhull/pull/112 is merged.
@@ -146,10 +140,6 @@ findorfetch(
>>>>>>> 903308adb4b (Improved error handling, differentiate nix/non-nix networks)
   tinyxml2
   LIBRARY_NAME
   tinyxml2
-  GIT_REPO
-  https://github.com/leethomason/tinyxml2.git
-  GIT_TAG
-  ${MUJOCO_DEP_VERSION_tinyxml2}
   TARGETS
   tinyxml2
   EXCLUDE_FROM_ALL
<<<<<<< HEAD
@@ -169,10 +159,6 @@ findorfetch(
=======
@@ -164,10 +154,6 @@ findorfetch(
>>>>>>> 903308adb4b (Improved error handling, differentiate nix/non-nix networks)
   tinyobjloader
   LIBRARY_NAME
   tinyobjloader
-  GIT_REPO
-  https://github.com/tinyobjloader/tinyobjloader.git
-  GIT_TAG
-  ${MUJOCO_DEP_VERSION_tinyobjloader}
   TARGETS
   tinyobjloader
   EXCLUDE_FROM_ALL
<<<<<<< HEAD
@@ -187,10 +173,6 @@ findorfetch(
=======
@@ -182,10 +168,6 @@ findorfetch(
>>>>>>> 903308adb4b (Improved error handling, differentiate nix/non-nix networks)
   ccd
   LIBRARY_NAME
   ccd
-  GIT_REPO
-  https://github.com/danfis/libccd.git
-  GIT_TAG
-  ${MUJOCO_DEP_VERSION_ccd}
   TARGETS
   ccd
   EXCLUDE_FROM_ALL
<<<<<<< HEAD
@@ -227,10 +209,6 @@ if(MUJOCO_BUILD_TESTS)
=======
@@ -222,10 +204,6 @@ if(MUJOCO_BUILD_TESTS)
>>>>>>> 903308adb4b (Improved error handling, differentiate nix/non-nix networks)
     absl
     LIBRARY_NAME
     abseil-cpp
-    GIT_REPO
-    https://github.com/abseil/abseil-cpp.git
-    GIT_TAG
-    ${MUJOCO_DEP_VERSION_abseil}
     TARGETS
     absl::core_headers
     EXCLUDE_FROM_ALL
<<<<<<< HEAD
@@ -254,10 +232,6 @@ if(MUJOCO_BUILD_TESTS)
=======
@@ -249,10 +227,6 @@ if(MUJOCO_BUILD_TESTS)
>>>>>>> 903308adb4b (Improved error handling, differentiate nix/non-nix networks)
     GTest
     LIBRARY_NAME
     googletest
-    GIT_REPO
-    https://github.com/google/googletest.git
-    GIT_TAG
-    ${MUJOCO_DEP_VERSION_gtest}
     TARGETS
     gtest
     gmock
<<<<<<< HEAD
@@ -288,10 +262,6 @@ if(MUJOCO_BUILD_TESTS)
=======
@@ -283,10 +257,6 @@ if(MUJOCO_BUILD_TESTS)
>>>>>>> 903308adb4b (Improved error handling, differentiate nix/non-nix networks)
     benchmark
     LIBRARY_NAME
     benchmark
-    GIT_REPO
-    https://github.com/google/benchmark.git
-    GIT_TAG
-    ${MUJOCO_DEP_VERSION_benchmark}
     TARGETS
     benchmark::benchmark
     benchmark::benchmark_main
<<<<<<< HEAD
@@ -308,8 +278,6 @@ if(MUJOCO_TEST_PYTHON_UTIL)
=======
@@ -303,8 +273,6 @@ if(MUJOCO_TEST_PYTHON_UTIL)
>>>>>>> 903308adb4b (Improved error handling, differentiate nix/non-nix networks)
 
     FetchContent_Declare(
       Eigen3
-      GIT_REPOSITORY https://gitlab.com/libeigen/eigen.git
-      GIT_TAG ${MUJOCO_DEP_VERSION_Eigen3}
     )
 
     FetchContent_GetProperties(Eigen3)
diff --git a/simulate/cmake/SimulateDependencies.cmake b/simulate/cmake/SimulateDependencies.cmake
<<<<<<< HEAD
index fa539c2..5985d5a 100644
--- a/simulate/cmake/SimulateDependencies.cmake
+++ b/simulate/cmake/SimulateDependencies.cmake
@@ -81,10 +81,6 @@ findorfetch(
   glfw3
   LIBRARY_NAME
   glfw3
-  GIT_REPO
-  https://github.com/glfw/glfw.git
-  GIT_TAG
-  ${MUJOCO_DEP_VERSION_glfw3}
=======
index 6616d6f..5d5a889 100644
--- a/simulate/cmake/SimulateDependencies.cmake
+++ b/simulate/cmake/SimulateDependencies.cmake
@@ -81,10 +81,6 @@ findorfetch(
   glfw
   LIBRARY_NAME
   glfw
-  GIT_REPO
-  https://github.com/glfw/glfw.git
-  GIT_TAG
-  ${MUJOCO_DEP_VERSION_glfw}
>>>>>>> 903308adb4b (Improved error handling, differentiate nix/non-nix networks)
   TARGETS
   glfw
   EXCLUDE_FROM_ALL
