diff --git a/src/toolset/workflow/ArcanistShellCompleteWorkflow.php b/src/toolset/workflow/ArcanistShellCompleteWorkflow.php
index 9c2fcf9a..307231c8 100644
--- a/src/toolset/workflow/ArcanistShellCompleteWorkflow.php
+++ b/src/toolset/workflow/ArcanistShellCompleteWorkflow.php
@@ -92,7 +92,7 @@ EOTEXT
     $argv = $this->getArgument('argv');
 
     $is_generate = $this->getArgument('generate');
-    $is_shell = (bool)strlen($this->getArgument('shell'));
+    $is_shell = phutil_nonempty_string($this->getArgument('shell'));
     $is_current = $this->getArgument('current');
 
     if ($argv) {
