diff --git a/build.gradle b/build.gradle
index 6f3bc6f..16fc51b 100644
--- a/build.gradle
+++ b/build.gradle
@@ -55,16 +55,8 @@ else {
 repositories {
 	mavenLocal()
 	mavenCentral()
-	maven {
-		name = "boricj's binary-file-toolkit"
-		url = uri("https://maven.pkg.github.com/boricj/binary-file-toolkit")
-		credentials(HttpHeaderCredentials) {
-            name = "Authorization"
-            value = "Bearer " + project.findProperty("githubToken") as String ?: System.getenv("GITHUB_TOKEN")
-		}
-        authentication {
-            header(HttpHeaderAuthentication)
-        }
+	flatDir {
+		dirs "@binary-file-toolkit@"
 	}
 }
 
