--- a/src/simutrans/network/network.cc    (revision 11272)
+++ b/src/simutrans/network/network.cc    (revision 11273)
@@ -981,7 +981,12 @@
 		struct UPNPUrls urls;
 		struct IGDdatas data;
 
+#if MINIUPNPC_API_VERSION <= 17
 		UPNP_GetValidIGD( devlist, &urls, &data, lanaddr, sizeof(lanaddr) );
+#else
+       char wanaddr[64] = "uset";
+       UPNP_GetValidIGD(devlist, &urls, &data, lanaddr, sizeof(lanaddr), wanaddr, sizeof(lanaddr));
+#endif
 		// we must know our IP address first
 		if(  UPNP_GetExternalIPAddress(urls.controlURL, data.first.servicetype, externalIPAddress) ==  UPNPCOMMAND_SUCCESS  ) {
 			// this is our ID (at least the routes tells us this)
@@ -1040,7 +1045,13 @@
 		struct UPNPUrls urls;
 		struct IGDdatas data;
 
-		UPNP_GetValidIGD( devlist, &urls, &data, lanaddr, sizeof(lanaddr) );
+#if MINIUPNPC_API_VERSION <= 17
+       UPNP_GetValidIGD(devlist, &urls, &data, lanaddr, sizeof(lanaddr));
+#else
+       char wanaddr[64] = "uset";
+       UPNP_GetValidIGD(devlist, &urls, &data, lanaddr, sizeof(lanaddr), wanaddr, sizeof(lanaddr));
+#endif
+
 		// we must know our IP address first
 		if(  UPNP_GetExternalIPAddress(urls.controlURL, data.first.servicetype, externalIPAddress) ==  UPNPCOMMAND_SUCCESS  ) {
 			// this is our ID (at least the routes tells us this)
