--- a/src/simutrans/sys/simsys.cc	(revision 11203)
+++ b/src/simutrans/sys/simsys.cc	(revision 11204)
@@ -4,8 +4,9 @@
  */
 
 #include <ctype.h>
+#include <filesystem>
+#include <stdio.h>
 #include <stdlib.h>
-#include <stdio.h>
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/types.h>
@@ -399,7 +400,11 @@
 {
 	if(  path  &&  *path  ) {
 		bool ok = !dr_chdir(path);
-		if(testfile) {
+		// Attempt to create it (if we aren't testing for an existing directory containing a file).
+		if(!ok && !testfile) {
+			dr_mkdir(path);
+			ok = !dr_chdir(path);
+		} else if(testfile) {
 			FILE* testf = fopen(testfile,"r");
 			ok = ok && testf;
 			if(testf) {
--- a/src/simutrans/sys/simsys.h	(revision 11203)
+++ b/src/simutrans/sys/simsys.h	(revision 11204)
@@ -151,6 +151,7 @@
 
 /**
 * Check if the directory exists and if so set the result variable to it
+* If the directory doesn't exist previously, it will attempt to create it if testfile is not provided
 * @param path : Path to directory to check
 * @param info : String to provide extra log info
 * @param result : Variable to store the path to directory if it exists
