--- a/src/client.cc
+++ b/src/client.cc
@@ -208,8 +208,8 @@ void Client::handle_events() {
         case SDL_MOUSEBUTTONUP: on_mousebutton(e); break;
         case SDL_ACTIVEEVENT: {
             update_mouse_button_state();
-            if (e.active.gain == 0 && !video::IsFullScreen())
-                show_menu(false);
+//            if (e.active.gain == 0 && !video::IsFullScreen())
+//                show_menu(false);
             break;
         }
