From 8f739318029105415da01bf8723d2f89d3188860 Mon Sep 17 00:00:00 2001
From: Markus Theil <theil.markus@gmail.com>
Date: Mon, 1 Jun 2026 10:45:47 +0200
Subject: [PATCH] configure: fix fuse cflags override for fuse 3

Signed-off-by: Markus Theil <theil.markus@gmail.com>
---
 configure.ac | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 72fa487..6a934d7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -85,7 +85,12 @@ AS_IF([test "x$build_fuse" = "xyes"], [
   if test $backend = freebsd; then
     AC_CHECK_LIB(cuse, cuse_dev_create)
   else
-    PKG_CHECK_MODULES(fuse, fuse, [], PKG_CHECK_MODULES(fuse, fuse3, [use_fuse3=yes], [AC_MSG_ERROR([Can't find fuse])]))
+    PKG_CHECK_EXISTS([fuse], [
+      PKG_CHECK_MODULES(fuse, fuse)
+    ], [PKG_CHECK_EXISTS([fuse3], [
+      PKG_CHECK_MODULES(fuse, fuse3)
+      use_fuse3=yes
+    ], [AC_MSG_ERROR([Can't find fuse])])])
   fi
   if test "x$use_fuse3" = "xyes"; then
     AC_SUBST(CPPFLAGS, "$CPPFLAGS -DFUSE_USE_VERSION=30")
