diff --git a/src/synapse-plugins/calculator-plugin.vala b/src/synapse-plugins/calculator-plugin.vala
index 886a44cd..a4601da6 100644
--- a/src/synapse-plugins/calculator-plugin.vala
+++ b/src/synapse-plugins/calculator-plugin.vala
@@ -50,9 +50,7 @@ namespace Synapse {
                 _("Calculator"),
                 _("Calculate basic expressions."),
                 "accessories-calculator",
-                register_plugin,
-                Environment.find_program_in_path ("bc") != null,
-                _("bc is not installed")
+                register_plugin
             );
         }
 
@@ -94,7 +92,7 @@ namespace Synapse {
                 Pid pid;
                 int read_fd, write_fd;
                 /* Must include math library to get non-integer results and to access standard math functions */
-                string[] argv = {"bc", "-l"};
+                string[] argv = {"@bc@", "-l"};
                 string? solution = null;
 
                 try {
