From 2fe7a445bafedee2c43050e40697d8b0fd7f7b30 Mon Sep 17 00:00:00 2001
From: Maximilian Bosch <maximilian@mbosch.me>
Date: Fri, 19 Mar 2021 19:37:34 +0100
Subject: [PATCH] Only build unit-tests

---
 setup.cfg | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/setup.cfg b/setup.cfg
index 2c2f49f..a5ec152 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -83,13 +83,6 @@ force_sort_within_sections = true
 junit_family = xunit2
 addopts =
     --verbose
-    --doctest-modules
-    --no-cov-on-fail
-    --cov-fail-under=100
-    --cov-report=term-missing
-    --cov-report=xml:build/coverage/coverage.xml
-    --cov-report=html:build/coverage
-    --junitxml=build/testresults/junit.xml
 
 [coverage:run]
 omit =
-- 
2.29.3

