diff --git a/Makefile b/Makefile
index 6a55b06..84a6000 100644
--- a/make/Makefile
+++ b/make/Makefile
@@ -387,6 +387,7 @@ configure-seam-linux64:
 	 ../sources/configure \
 		--prefix='$(PREFIX)' \
 		--with-warnings=yes \
+		--with-zlib='$(zlib)' \
 		--disable-lightning)
 
 .PHONY: configure-seam-darwin64
@@ -434,6 +435,7 @@ configure-alice-ll-linux:
 	(cd $(PWD)/alice/build && \
 	 ../sources/vm-seam/configure \
 		--prefix='$(PREFIX)' \
+		--with-gmp='$(gmp)' \
 		--with-warnings=yes)
 
 .PHONY:	rebuild-alice-ll
