Index: qtbase-opensource-src-5.6.2/src/network/kernel/qdnslookup_unix.cpp
===================================================================
--- qtbase-opensource-src-5.6.2.orig/src/network/kernel/qdnslookup_unix.cpp
+++ qtbase-opensource-src-5.6.2/src/network/kernel/qdnslookup_unix.cpp
@@ -83,7 +83,7 @@ static bool resolveLibraryInternal()
     if (!lib.load())
 #endif
     {
-        lib.setFileName(QLatin1String("resolv"));
+        lib.setFileName(QLatin1String("@glibc@/lib/resolv"));
         if (!lib.load())
             return false;
     }
Index: qtbase-opensource-src-5.6.2/src/network/kernel/qhostinfo_unix.cpp
===================================================================
--- qtbase-opensource-src-5.6.2.orig/src/network/kernel/qhostinfo_unix.cpp
+++ qtbase-opensource-src-5.6.2/src/network/kernel/qhostinfo_unix.cpp
@@ -94,7 +94,7 @@ static bool resolveLibraryInternal()
     if (!lib.load())
 #endif
     {
-        lib.setFileName(QLatin1String("resolv"));
+        lib.setFileName(QLatin1String("@glibc@/lib/libresolv"));
         if (!lib.load())
             return false;
     }
