Index: twinkle-1.4.2/configure.in
===================================================================
--- twinkle-1.4.2.orig/configure.in	2013-07-25 11:09:16.000000000 -0400
+++ twinkle-1.4.2/configure.in	2013-07-25 11:11:59.512418187 -0400
@@ -66,7 +66,7 @@
 
 export PKG_CONFIG_PATH
 
-PKG_CHECK_MODULES(CCRTP, libccrtp1 >= 1.6.0)
+PKG_CHECK_MODULES(CCRTP, libccrtp >= 1.6.0)
 
 PKG_CHECK_MODULES(XML2, libxml-2.0)
 # AC_CHECK_HEADER(libxml/tree.h, [],
@@ -136,7 +136,7 @@
 #echo "INCLUDEPATH += `$CCGNU2_CONFIG --includes`" | sed -e s/-I//g > $QT_INCL_PRO
 # libccrtp1(ccrtp) depend from libccgnu2(commoncpp2) and
 # should include above flags !
-echo "INCLUDEPATH += `$PKG_CONFIG --cflags-only-I libccrtp1`" | sed -e s/-I//g >> $QT_INCL_PRO
+echo "INCLUDEPATH += `$PKG_CONFIG --cflags-only-I libccrtp`" | sed -e s/-I//g >> $QT_INCL_PRO
 echo "INCLUDEPATH += `$PKG_CONFIG --cflags-only-I libxml-2.0`" | sed -e s/-I//g >> $QT_INCL_PRO
 
 # get libraries specified on command line
