diff -ru -x '*~' ./result/src/syscall/creds_test.go go-go1.7.4-src/src/syscall/creds_test.go
--- ./result/src/syscall/creds_test.go	1970-01-01 01:00:01.000000000 +0100
+++ go-go1.7.4-src/src/syscall/creds_test.go	2016-12-21 14:06:39.559932164 +0100
@@ -62,8 +62,8 @@
 		if sys, ok := err.(*os.SyscallError); ok {
 			err = sys.Err
 		}
-		if err != syscall.EPERM {
-			t.Fatalf("WriteMsgUnix failed with %v, want EPERM", err)
+		if err != syscall.EPERM && err != syscall.EINVAL {
+			t.Fatalf("WriteMsgUnix failed with %v, want EPERM or EINVAL", err)
 		}
 	}
 
