diff -r cb140bac9ab0 src/pkg/Makefile
--- a/src/pkg/Makefile	Thu Nov 12 14:55:26 2009 -0800
+++ b/src/pkg/Makefile	Mon Nov 16 11:50:34 2009 +0100
@@ -100,12 +100,15 @@
 
 NOTEST=\
 	debug/proc\
+	exec\
 	go/ast\
 	go/doc\
 	go/token\
 	hash\
 	image\
+	log\
 	malloc\
+	os\
 	rand\
 	runtime\
 	syscall\
diff -r cb140bac9ab0 src/run.bash
--- a/src/run.bash	Thu Nov 12 14:55:26 2009 -0800
+++ b/src/run.bash	Mon Nov 16 11:50:34 2009 +0100
@@ -69,7 +69,3 @@
 ./timing.sh -test
 ) || exit $?
 
-(xcd ../test
-./run
-) || exit $?
-
