diff --git a/src/sync/waitgroup_test.go b/src/sync/waitgroup_test.go
index e3e3096..f80d1e2 100644
--- a/src/sync/waitgroup_test.go
+++ b/src/sync/waitgroup_test.go
@@ -6,6 +6,7 @@ package sync_test
 
 import (
 	"internal/race"
+	"internal/testenv"
 	"runtime"
 	. "sync"
 	"sync/atomic"
@@ -73,6 +74,7 @@ func TestWaitGroupMisuse2(t *testing.T) {
 	if runtime.NumCPU() <= 4 {
 		t.Skip("NumCPU<=4, skipping: this test requires parallelism")
 	}
+	testenv.SkipFlaky(t, 20072)
 	defer func() {
 		err := recover()
 		if err != "sync: negative WaitGroup counter" &&
