diff --git a/src/net/http/serve_test.go b/src/net/http/serve_test.go
index 7137599..43cca2e 100644
--- a/src/net/http/serve_test.go
+++ b/src/net/http/serve_test.go
@@ -5361,6 +5361,7 @@ func testServerKeepAlivesEnabled(t *testing.T, h2 bool) {
 // the server's doing its 1-byte background read between requests,
 // waiting for the connection to maybe close.
 func TestServerCancelsReadTimeoutWhenIdle(t *testing.T) {
+	testenv.SkipFlaky(t, 19608)
 	setParallel(t)
 	defer afterTest(t)
 	const timeout = 250 * time.Millisecond
