diff -r b51fd2d6c160 src/pkg/log/log_test.go
--- a/src/pkg/log/log_test.go	Tue Nov 10 20:05:24 2009 -0800
+++ b/src/pkg/log/log_test.go	Sat Nov 14 20:45:04 2009 +0100
@@ -18,8 +18,8 @@
 	Rtime		= `[0-9][0-9]:[0-9][0-9]:[0-9][0-9]`;
 	Rmicroseconds	= `\.[0-9][0-9][0-9][0-9][0-9][0-9]`;
 	Rline		= `[0-9]+:`;
-	Rlongfile	= `/[A-Za-z0-9_/\-]+\.go:` + Rline;
-	Rshortfile	= `[A-Za-z0-9_\-]+\.go:` + Rline;
+	Rlongfile	= `/[A-Za-z0-9_./\-]+\.go:` + Rline;
+	Rshortfile	= `[A-Za-z0-9_.\-]+\.go:` + Rline;
 )
 
 type tester struct {
