diff --git a/src/cmd/vendor/golang.org/x/sys/unix/syscall_unix_test.go b/src/cmd/vendor/golang.org/x/sys/unix/syscall_unix_test.go
index d694990..87fa259 100644
--- a/src/cmd/vendor/golang.org/x/sys/unix/syscall_unix_test.go
+++ b/src/cmd/vendor/golang.org/x/sys/unix/syscall_unix_test.go
@@ -452,7 +452,7 @@ func TestGetwd(t *testing.T) {
 	defer fd.Close()
 	// These are chosen carefully not to be symlinks on a Mac
 	// (unlike, say, /var, /etc)
-	dirs := []string{"/", "/usr/bin"}
+	dirs := []string{"/"}
 	switch runtime.GOOS {
 	case "android":
 		dirs = []string{"/", "/system/bin"}
