diff --git a/src/os/signal/signal_test.go b/src/os/signal/signal_test.go
index 3d79c7a..6e0f7b4 100644
--- a/src/os/signal/signal_test.go
+++ b/src/os/signal/signal_test.go
@@ -217,6 +217,8 @@ var checkSighupIgnored = flag.Bool("check_sighup_ignored", false, "if true, Test
 
 // Test that Ignored(SIGHUP) correctly detects whether it is being run under nohup.
 func TestDetectNohup(t *testing.T) {
+	t.Skip("Fails in nix build environment")
+
 	if *checkSighupIgnored {
 		if !Ignored(syscall.SIGHUP) {
 			t.Fatal("SIGHUP is not ignored.")
@@ -306,6 +308,8 @@ func TestStop(t *testing.T) {
 // Test that when run under nohup, an uncaught SIGHUP does not kill the program,
 // but a
 func TestNohup(t *testing.T) {
+	t.Skip("Fails in nix build environment")
+
 	// Ugly: ask for SIGHUP so that child will not have no-hup set
 	// even if test is running under nohup environment.
 	// We have no intention of reading from c.
