diff --git a/src/os/exec/exec_test.go b/src/os/exec/exec_test.go
index dce66c5c2e..2532224376 100644
--- a/src/os/exec/exec_test.go
+++ b/src/os/exec/exec_test.go
@@ -627,6 +627,10 @@ func TestExtraFiles(t *testing.T) {
 		t.Skipf("skipping test on %q", runtime.GOOS)
 	}
 
+	if runtime.GOOS == "linux" && runtime.GOARCH  == "arm" {
+		t.Skipf("skipping test on %q %q", runtime.GOARCH, runtime.GOOS)
+	}
+
 	// Force network usage, to verify the epoll (or whatever) fd
 	// doesn't leak to the child,
 	ln, err := net.Listen("tcp", "127.0.0.1:0")
