diff -rupN ulex-1.1.orig/Makefile ulex-1.1/Makefile
--- ulex-1.1.orig/Makefile	2015-06-10 21:31:17.605000000 +0200
+++ ulex-1.1/Makefile	2015-06-10 21:33:57.077000000 +0200
@@ -2,7 +2,7 @@ VERSION=1.1
 # Don't forget to change META file as well
 
 ALL=pa_ulex.cma ulexing.cma
-OCAMLBUILD=ocamlbuild -byte-plugin
+OCAMLBUILD=ocamlbuild -use-ocamlfind -byte-plugin
 
 all::
 	$(OCAMLBUILD) $(ALL)
diff -rupN ulex-1.1.orig/_tags ulex-1.1/_tags
--- ulex-1.1.orig/_tags	2015-06-10 21:31:17.605000000 +0200
+++ ulex-1.1/_tags	2015-06-10 21:32:09.380000000 +0200
@@ -1,3 +1,4 @@
+true: package(camlp4)
 "pa_ulex.ml": camlp4orf, use_camlp4
 <test.*> or "custom_ulexing.ml": use_ulex
 # true: use_ulex
