diff --git a/src/vteconv.cc b/src/vteconv.cc
index b78d3928..5cb63e7e 100644
--- a/src/vteconv.cc
+++ b/src/vteconv.cc
@@ -771,7 +771,7 @@ int
 main (int argc,
       char *argv[])
 {
-        g_test_init (&argc, &argv, NULL);
+        g_test_init (&argc, &argv, (char *)NULL);
 
         g_test_add_func ("/vte/conv/utf8/strlen", test_utf8_strlen);
         g_test_add_func ("/vte/conv/utf8/validate", test_utf8_validate);
diff --git a/src/vtetypes.cc b/src/vtetypes.cc
index 1365a295..8f38c9d9 100644
--- a/src/vtetypes.cc
+++ b/src/vtetypes.cc
@@ -407,7 +407,7 @@ test_util_smart_fd(void)
 int
 main(int argc, char *argv[])
 {
-        g_test_init (&argc, &argv, NULL);
+        g_test_init (&argc, &argv, (char *)NULL);
 
         g_test_add_func("/vte/c++/grid/coords", test_grid_coords);
         g_test_add_func("/vte/c++/grid/span", test_grid_span);
