diff --git a/src/toolchains.rs b/src/toolchains.rs
index 4d85e7c..b1353c6 100644
--- a/src/toolchains.rs
+++ b/src/toolchains.rs
@@ -259,6 +259,8 @@ impl Toolchain {
             .map_err(InstallError::Download)?;
         }

+        nix_patchelf(tmpdir.path().to_path_buf())
+            .expect("failed to patch toolchain for NixOS");
         fs::rename(tmpdir.into_path(), dest).map_err(InstallError::Move)?;

         Ok(())
@@ -557,3 +559,42 @@ pub(crate) fn download_tarball(
     }
     download_tar_gz(client, name, &format!("{}.gz", url,), strip_prefix, dest)
 }
+
+fn nix_patchelf(mut toolchain_path: PathBuf) -> Result<(), Error> {
+    toolchain_path.push("bin");
+
+    for entry in toolchain_path.read_dir()? {
+        let entry = entry?;
+        if !entry.file_type()?.is_file() {
+            continue;
+        }
+
+        eprintln!("info: you seem to be running NixOS. Attempting to patch {}",
+                  entry.path().to_str().unwrap());
+        let _ = ::std::process::Command::new("@patchelf@/bin/patchelf")
+            .arg("--set-interpreter")
+            .arg("@dynamicLinker@")
+            .arg(entry.path())
+            .output();
+    }
+
+    toolchain_path.pop();
+    toolchain_path.push("lib");
+
+    for entry in toolchain_path.read_dir()? {
+        let entry = entry?;
+        if !entry.file_type()?.is_file() {
+            continue;
+        }
+
+        eprintln!("info: you seem to be running NixOS. Attempting to patch {}",
+                  entry.path().to_str().unwrap());
+        let _ = ::std::process::Command::new("@patchelf@/bin/patchelf")
+            .arg("--set-rpath")
+            .arg("@libPath@")
+            .arg(entry.path())
+            .output();
+    }
+
+    Ok(())
+}
