diff -ubr graphviz-2.28.0-orig/lib/cdt/cdt.h graphviz-2.28.0-new/lib/cdt/cdt.h
--- graphviz-2.28.0-orig/lib/cdt/cdt.h	2012-11-11 20:05:55.674397100 +0100
+++ graphviz-2.28.0-new/lib/cdt/cdt.h	2012-11-11 20:06:28.069511092 +0100
@@ -238,11 +238,6 @@
 extern int		dtstat _ARG_((Dt_t*, Dtstat_t*, int));
 extern unsigned int	dtstrhash _ARG_((unsigned int, Void_t*, int));
 
-#if !_PACKAGE_ast
-extern int		memcmp _ARG_((const Void_t*, const Void_t*, size_t));
-extern int		strcmp _ARG_((const char*, const char*));
-#endif
-
 #undef extern
 _END_EXTERNS_
 
