diff --git a/gcc/vhdl/Makefile.in b/gcc/vhdl/Makefile.in
index b8d2ed0..0e8bd55 100644
--- a/gcc/vhdl/Makefile.in
+++ b/gcc/vhdl/Makefile.in
@@ -682,7 +682,7 @@ install-ghdllib: ghdllib grt.lst $(STD93_SRCS) $(STD87_SRCS) \
 	PDIR=`pwd` && cd $(DESTDIR)$(VHDL_LIB_DIR) && \
 	$(MAKE) -f $$PDIR/Makefile REL_DIR=../../.. \
 	 LIBSRC_DIR="src" LIB93_DIR=lib/v93 LIB87_DIR=lib/v87 \
-	 ANALYZE="$$PDIR/../ghdl -a --GHDL1=$$PDIR/../ghdl1 --ieee=none" \
+	 ANALYZE="$$PDIR/../ghdl -a --GHDL1=$$PDIR/../ghdl1 --ieee=none -Wc,-O2" \
 	 std.v93 std.v87 ieee.v93 ieee.v87 synopsys.v93 synopsys.v87 mentor.v93
 # Copy std_standard (this is done after libraries, since they remove dirs).
 	$(INSTALL_DATA) std87_standard.o \
