diff --git a/daemon/gdm-server.c.orig b/daemon/gdm-server.c
index 6357d34..5bef5c8 100644
--- a/daemon/gdm-server.c.orig
+++ b/daemon/gdm-server.c
@@ -264,7 +264,11 @@ gdm_server_init_command (GdmServer *server)
         return;
 
 fallback:
-        server->priv->command = g_strdup_printf (X_SERVER X_SERVER_ARG_FORMAT, verbosity, debug_options);
+        if (g_getenv("GDM_X_SERVER") != NULL) {
+            server->priv->command = g_strdup (g_getenv("GDM_X_SERVER"));
+        } else {
+            server->priv->command = g_strdup_printf (X_SERVER X_SERVER_ARG_FORMAT, verbosity, debug_options);
+        }
 
 }

