From 1a1ab986245e8b74648a1a0adb5d1c7019561d18 Mon Sep 17 00:00:00 2001
From: Attila Lendvai <attila@lendvai.name>
Date: Sat, 30 Jan 2021 15:24:57 +0100
Subject: [PATCH 2/2] nix diff for substituteAll

---
 packaging/bee-clef-service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/packaging/bee-clef-service b/packaging/bee-clef-service
index 34c7edd..31e9d95 100755
--- a/packaging/bee-clef-service
+++ b/packaging/bee-clef-service
@@ -15,7 +15,7 @@ start() {
     ( sleep 2; cat << EOF
 { "jsonrpc": "2.0", "id":1, "result": { "text":"$SECRET" } }
 EOF
-) | clef --stdio-ui --keystore $KEYSTORE --configdir $CONFIGDIR --chainid $CHAINID --rules /etc/bee-clef/rules.js --nousb --4bytedb-custom /etc/bee-clef/4byte.json --pcscdpath "" --auditlog "" --loglevel 3 --ipcpath ${CONFIGDIR}
+) | @clefBinary@ --stdio-ui --keystore $KEYSTORE --configdir $CONFIGDIR --chainid $CHAINID --rules @out@/share/bee-clef/rules.js --nousb --4bytedb-custom @out@/share/bee-clef/4byte.json --pcscdpath "" --auditlog "" --loglevel 3 --ipcpath ${CONFIGDIR}
 }
 
 stop() {
-- 
2.29.2

