diff --git a/3party/CMakeLists.txt b/3party/CMakeLists.txt
index 5178ae0..abe103f 100644
--- a/3party/CMakeLists.txt
+++ b/3party/CMakeLists.txt
@@ -41,9 +41,6 @@ if (NOT WITH_SYSTEM_PROVIDED_3PARTY)
   # Add pugixml library.
   add_subdirectory(pugixml)

-  # Add protobuf library.
-  add_subdirectory(protobuf)
-
   if (NOT PLATFORM_LINUX)
     add_subdirectory(freetype)
     add_subdirectory(icu)
@@ -55,6 +52,9 @@ if (NOT WITH_SYSTEM_PROVIDED_3PARTY)
   target_include_directories(utf8cpp INTERFACE "${OMIM_ROOT}/3party/utfcpp/source")
 endif()

+# Add protobuf library.
+add_subdirectory(protobuf)
+
 add_subdirectory(agg)
 add_subdirectory(bsdiff-courgette)
 add_subdirectory(minizip)
