With optimizations on, python will overwrite the nix store.

--- wicd-1.7.0/in/scripts=wicd.in  2011-04-05 14:31:09.733096865 -0400
+++ wicd-1.7.0/in/scripts=wicd.in  2011-04-05 14:31:16.397096864 -0400
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-exec %PYTHON% -O %SHARE%daemon/wicd-daemon.py $@
+exec %PYTHON% -B %SHARE%daemon/wicd-daemon.py $@

--- wicd-1.7.1b2/wicd/wicd-daemon.py.old     2011-04-07 08:28:38.668311226 -0400
+++ wicd-1.7.1b2/wicd/wicd-daemon.py 2011-04-07 08:28:59.111311223 -0400
@@ -1806,7 +1806,7 @@
     wicd_bus = dbus.service.BusName('org.wicd.daemon', bus=bus)
     daemon = WicdDaemon(wicd_bus, auto_connect=auto_connect)
     if not no_poll:
-        child_pid = Popen([misc.find_path("python2"), "-O", 
+        child_pid = Popen([misc.find_path("python2"), "-B",
                           os.path.join(wpath.daemon, "monitor.py")],
                           shell=False, close_fds=True).pid
     atexit.register(on_exit, child_pid)
