diff --git a/src/@dir@/lp/tail_matrix.h b/src/@dir@/lp/tail_matrix.h
index 2047e8c..c84340e 100644
--- a/src/@dir@/lp/tail_matrix.h
+++ b/src/@dir@/lp/tail_matrix.h
@@ -43,7 +43,6 @@ public:
         const tail_matrix & m_A;
         unsigned m_row;
         ref_row(const tail_matrix& m, unsigned row): m_A(m), m_row(row) {}
-        T operator[](unsigned j) const { return m_A.get_elem(m_row, j);}
     };
     ref_row operator[](unsigned i) const { return ref_row(*this, i);}
 };
