--- jbsrc/jb.c.orig	2014-01-19 20:06:48.525462981 +0100
+++ jbsrc/jb.c	2014-01-19 20:07:36.087934897 +0100
@@ -425,7 +425,6 @@
        */
       jb_compile_options_add_cflags(object->compile_options, "-std=c99");
       jb_compile_options_add_cppflags(object->compile_options, "-D_BSD_SOURCE -D_POSIX_C_SOURCE=199309L");
-      jb_compile_options_add_libs(object->compile_options, "-lbsd-compat");
     }
 
   jb_compile_options_add_string_defines(object->compile_options,
