--- jbsrc/lib/src/core/jb-feature.c.~1~	2008-04-27 16:47:27.000000000 +0200
+++ jbsrc/lib/src/core/jb-feature.c	2008-07-22 11:40:50.856886210 +0200
@@ -164,8 +164,6 @@
 static void
 gconf_configure (void)
 {
-  JBVariable *variable;
-
   jb_require_program("gconftool-2");
 
   if (! strcmp(jb_variable_get_string("gconf-config-source"), "autodetect"))
@@ -178,21 +176,6 @@
       jb_variable_set_string("gconf-config-source", config_source);
       g_free(config_source);
     }
-
-  /* fix the default schemas dir on Ubuntu */
-  variable = jb_variable_get_variable_or_error("gconf-schemas-dir");
-  if (! variable->user_set)
-    {
-      static const char *ubuntu_dir = "$datadir/gconf/schemas";
-      char *expanded;
-
-      expanded = jb_variable_expand(ubuntu_dir, NULL);
-
-      if (g_file_test(expanded, G_FILE_TEST_IS_DIR))
-	jb_variable_set_string("gconf-schemas-dir", ubuntu_dir);
-
-      g_free(expanded);
-    }
 }
 
 static void
