--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -303,27 +303,6 @@
 endforeach()
 
 #
-## Add a rule to download the catch unit test framework
-#
-include(ExternalProject)
-ExternalProject_Add(catch
-  GIT_REPOSITORY "https://lab.louiz.org/louiz/Catch.git"
-  PREFIX "external"
-  UPDATE_COMMAND ""
-  CONFIGURE_COMMAND ""
-  BUILD_COMMAND ""
-  INSTALL_COMMAND ""
-  )
-set_target_properties(catch PROPERTIES EXCLUDE_FROM_ALL TRUE)
-ExternalProject_Get_Property(catch SOURCE_DIR)
-if(NOT EXISTS ${CMAKE_SOURCE_DIR}/tests/catch.hpp)
-  target_include_directories(test_suite
-    PUBLIC "${SOURCE_DIR}/single_include/"
-    )
-  add_dependencies(test_suite catch)
-endif()
-
-#
 ## Add some custom rules to launch the tests
 #
 add_custom_target(check COMMAND "test_suite"
