diff --git a/src/translations.cpp b/src/translations.cpp
index fa0ee479b2..0e470098dc 100644
--- a/src/translations.cpp
+++ b/src/translations.cpp
@@ -141,15 +141,11 @@ void select_language()
 std::string locale_dir()
 {
     std::string loc_dir;
-#if !defined(__ANDROID__) && ((defined(__linux__) || defined(BSD) || (defined(MACOSX) && !defined(TILES))))
     if( !PATH_INFO::base_path().empty() ) {
         loc_dir = PATH_INFO::base_path() + "share/locale";
     } else {
         loc_dir = PATH_INFO::langdir();
     }
-#else
-    loc_dir = PATH_INFO::langdir();
-#endif
     return loc_dir;
 }
 
