diff --git a/configure.ac b/configure.ac
index 5068913..3d4271e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -429,6 +429,20 @@ AC_SUBST([GLESv2_LIB_GLOB])
 AC_SUBST([VG_LIB_GLOB])
 AC_SUBST([GLAPI_LIB_GLOB])
 
+
+dnl readonly text segment on x86 hardened platforms
+AC_ARG_ENABLE([glx_rts],
+    [AS_HELP_STRING([--enable-glx-rts],
+        [on x86, use a readonly text segment for libGL @<:@default=disabled@:>@])],
+    [enable_glx_rts="$enableval"],
+    [enable_glx_rts=no])
+if test "x$enable_glx_rts" = xyes; then
+    DEFINES="$DEFINES -DGLX_X86_READONLY_TEXT"
+else
+    enable_glx_rts=no
+fi
+
+
 dnl
 dnl Arch/platform-specific settings
 dnl
