--- ./src/gallium/winsys/svga/drm/vmw_screen.h.orig
+++ ./src/gallium/winsys/svga/drm/vmw_screen.h
@@ -34,7 +34,7 @@
 #ifndef VMW_SCREEN_H_
 #define VMW_SCREEN_H_
 
-
+#include <sys/stat.h>
 #include "pipe/p_compiler.h"
 #include "pipe/p_state.h"
 
--- a/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_winsys.h
+++ b/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_winsys.h
@@ -28,6 +28,8 @@
 #ifndef RADV_AMDGPU_WINSYS_H
 #define RADV_AMDGPU_WINSYS_H

+#include <sys/types.h>
+
 #include <amdgpu.h>
 #include <pthread.h>
 #include "util/list.h"
