--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,7 +47,7 @@ function(findLib libName libDir libOut)
         find_library(TEMP NAMES ${libName}.lib ${ARGV3}.lib ${ARGV4}.lib ${ARGV5}.lib ${ARGV6}.lib
             PATHS ${${libDir}} NO_DEFAULT_PATH)
     else()
-        find_library(TEMP NAMES lib${libName}.a lib${ARGV3}.a lib${ARGV4}.a lib${ARGV5}.a lib${ARGV6}.a
+        find_library(TEMP NAMES lib${libName}.a lib${ARGV3}.a lib${ARGV4}.a lib${ARGV5}.a lib${ARGV6}.a lib${libName}.so lib${ARGV3}.so lib${ARGV4}.so lib${ARGV5}.so lib${ARGV6}.so
             PATHS ${${libDir}} NO_DEFAULT_PATH)
     endif()
     
diff --git a/FindDependencies.cmake b/FindDependencies.cmake
index 2d644005f..7261ba923 100644
--- a/FindDependencies.cmake
+++ b/FindDependencies.cmake
@@ -46,8 +46,8 @@ if(MKLROOT)
             NO_DEFAULT_PATH)
             
         find_library(MKL_OMP_LIB 
-            NAMES iomp5 iomp5md libiomp5md.lib
-            PATHS ${MKLROOT}/../lib ${MKLROOT}/../compiler/lib
+            NAMES libiomp5.so libiomp5 iomp5 iomp5md libiomp5md.lib
+            PATHS ${MKLROOT}/lib ${MKLROOT}/../lib ${MKLROOT}/../compiler/lib
             PATH_SUFFIXES "intel64" "intel32"
             NO_DEFAULT_PATH
             DOC "MKL OMP Library")
