diff -ru3 brainfuck-0.1-old/Language/Brainfuck.hs brainfuck-0.1/Language/Brainfuck.hs
--- brainfuck-0.1-old/Language/Brainfuck.hs	2015-04-17 21:17:34.568721144 +0300
+++ brainfuck-0.1/Language/Brainfuck.hs	2015-04-17 21:19:17.065872395 +0300
@@ -19,6 +19,8 @@
 along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA -}
 
+{-# LANGUAGE FlexibleContexts #-}
+
 module Language.Brainfuck where
 
 import Data.Array.IO
@@ -287,4 +289,4 @@
 
 halt = if debug
          then putStrLn "Machine Halted.\n"
-         else putStrLn "\n"
\ No newline at end of file
+         else putStrLn "\n"
