diff -ru3 edit-distance-0.2.1.2-old/edit-distance.cabal edit-distance-0.2.1.2/edit-distance.cabal
--- edit-distance-0.2.1.2-old/edit-distance.cabal	2015-04-17 22:46:50.964116064 +0300
+++ edit-distance-0.2.1.2/edit-distance.cabal	2015-04-17 22:41:31.216027373 +0300
@@ -36,7 +36,7 @@
                                 Text.EditDistance.ArrayUtilities
         
         if flag(splitBase)
-                Build-Depends:  base >= 3 && < 4.8, array >= 0.1, random >= 1.0, containers >= 0.1.0.1
+                Build-Depends:  base >= 3 && < 5, array >= 0.1, random >= 1.0, containers >= 0.1.0.1
         else
                 Build-Depends:  base < 3
         
@@ -54,7 +54,7 @@
         else            
                 Build-Depends:          test-framework >= 0.1.1, QuickCheck >= 1.1 && < 2.0, test-framework-quickcheck
                 if flag(splitBase)
-                        Build-Depends:  base >= 3 && < 4.8, array >= 0.1, random >= 1.0
+                        Build-Depends:  base >= 3 && < 5, array >= 0.1, random >= 1.0
                 else
                         Build-Depends:  base < 3
 
@@ -65,7 +65,7 @@
                 Buildable:      False
         else
                 if flag(splitBase)
-                        Build-Depends:  base >= 3 && < 4.8, array >= 0.1, random >= 1.0, time >= 1.0, process >= 1.0,
+                        Build-Depends:  base >= 3 && < 5, array >= 0.1, random >= 1.0, time >= 1.0, process >= 1.0,
                                         deepseq >= 1.2, unix >= 2.3, criterion >= 0.6
                 else
                         Build-Depends:  base < 3,
