diff -ru3 lambdabot-5.0.1.old/State/L.hs lambdabot-5.0.1/State/L.hs
--- lambdabot-5.0.1.old/State/L.hs	2015-04-15 03:26:04.818947594 +0300
+++ lambdabot-5.0.1/State/L.hs	2015-04-15 03:26:20.386099365 +0300
@@ -34,7 +34,6 @@
 import Control.Monad
 import Control.Monad.Cont
 import Control.Monad.Identity
-import Control.Monad.Instances
 import Control.Monad.Reader
 import Control.Monad.ST.Safe
 import Control.Monad.State
diff -ru3 lambdabot-5.0.1.old/State/Pristine.hs lambdabot-5.0.1/State/Pristine.hs
--- lambdabot-5.0.1.old/State/Pristine.hs	2015-04-15 03:26:04.818947594 +0300
+++ lambdabot-5.0.1/State/Pristine.hs	2015-04-15 03:26:20.386099365 +0300
@@ -34,7 +34,6 @@
 import Control.Monad
 import Control.Monad.Cont
 import Control.Monad.Identity
-import Control.Monad.Instances
 import Control.Monad.Reader
 import Control.Monad.ST.Safe
 import Control.Monad.State
