diff --git a/Telegram/SourceFiles/media/view/media_view_pip.cpp b/Telegram/SourceFiles/media/view/media_view_pip.cpp
index 4e1d7c136..7c8f95888 100644
--- a/Telegram/SourceFiles/media/view/media_view_pip.cpp
+++ b/Telegram/SourceFiles/media/view/media_view_pip.cpp
@@ -362,6 +362,10 @@ void PipPanel::init() {
 	) | rpl::filter(rpl::mappers::_1) | rpl::start_with_next([=] {
 		// Workaround Qt's forced transient parent.
 		Ui::Platform::ClearTransientParent(widget());
+	}, rp()->lifetime());
+
+	rp()->shownValue(
+	) | rpl::filter(rpl::mappers::_1) | rpl::start_with_next([=] {
 		Ui::Platform::SetWindowMargins(widget(), _padding);
 	}, rp()->lifetime());
 
Submodule Telegram/lib_base contains modified content
diff --git a/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp b/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp
index 8aca67c..6b781fb 100644
--- a/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp
+++ b/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp
@@ -16,7 +16,11 @@
 #if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
 #include <qpa/qplatformintegration.h>
 #include <private/qguiapplication_p.h>
+#if QT_VERSION >= QT_VERSION_CHECK(6, 9, 0)
+#include <private/qdesktopunixservices_p.h>
+#else // Qt >= 6.9.0
 #include <private/qgenericunixservices_p.h>
+#endif // Qt < 6.9.0
 #endif // Qt >= 6.5.0
 
 #include <sstream>
@@ -39,7 +43,10 @@ std::string ParentWindowID(QWindow *window) {
 	}
 
 #if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
-	if (const auto services = dynamic_cast<QGenericUnixServices*>(
+#if QT_VERSION < QT_VERSION_CHECK(6, 9, 0)
+	using QDesktopUnixServices = QGenericUnixServices;
+#endif // Qt < 6.9.0
+	if (const auto services = dynamic_cast<QDesktopUnixServices*>(
 			QGuiApplicationPrivate::platformIntegration()->services())) {
 		return services->portalWindowIdentifier(window).toStdString();
 	}
