diff --git a/cmake/espeak-ng-for-piper.cmake b/cmake/espeak-ng-for-piper.cmake
index b54a0a6..3576360 100644
--- a/cmake/espeak-ng-for-piper.cmake
+++ b/cmake/espeak-ng-for-piper.cmake
@@ -108,11 +108,13 @@ function(download_espeak_ng_for_piper)
       -Wno-format-truncation
       -Wno-uninitialized
       -Wno-format
+      -Wno-error=format-security
     )

     if(TARGET espeak-ng-bin)
       target_compile_options(espeak-ng-bin PRIVATE
         -Wno-unused-result
+        -Wno-error=format-security
       )
     endif()
   endif()
