diff --git a/setup.py b/setup.py
index 7625074..b3aa2fc 100644
--- a/setup.py
+++ b/setup.py
@@ -249,7 +249,7 @@ class inc_lib_dirs:
             aDir(L, os.path.join("/usr/lib", "python%s" % sys.version[:3], "config"))
         elif platform == "darwin":
             machine = sysconfig_platform.split('-')[-1]
-            if machine=='arm64' or os.environ.get('ARCHFLAGS','')=='-arch arm64':
+            if False and machine=='arm64' or os.environ.get('ARCHFLAGS','')=='-arch arm64':
                 #print('!!!!! detected darwin arm64 build')
                 #probably an M1
                 target = pjoin(
