--- a/helix-term/src/commands.rs
+++ b/helix-term/src/commands.rs
@@ -424,9 +424,9 @@ impl MappableCommand {
         add_newline_below, "Add newline below",
         goto_type_definition, "Goto type definition",
         goto_implementation, "Goto implementation",
-        goto_file_start, "Goto line number <n> else file start",
+        goto_file_start, "Goto line number &lt;n&gt; else file start",
         goto_file_end, "Goto file end",
-        extend_to_file_start, "Extend to line number<n> else file start",
+        extend_to_file_start, "Extend to line number &lt;n&gt; else file start",
         extend_to_file_end, "Extend to file end",
         goto_file, "Goto files/URLs in selections",
         goto_file_hsplit, "Goto files in selections (hsplit)",
