diff --git a/test/surface_test.py b/test/surface_test.py
index c2c91f4f5..58d916de8 100644
--- a/test/surface_test.py
+++ b/test/surface_test.py
@@ -404,6 +404,7 @@ class SurfaceTypeTest(unittest.TestCase):
         finally:
             pygame.display.quit()
 
+    @unittest.skipIf(True, "https://github.com/libsdl-org/SDL/pull/14429")
     def test_solarwolf_rle_usage_2(self):
         """Test for RLE status after setting alpha"""
 
@@ -435,6 +436,7 @@ class SurfaceTypeTest(unittest.TestCase):
         finally:
             pygame.display.quit()
 
+    @unittest.skipIf(True, "https://github.com/libsdl-org/SDL/issues/14424")
     def test_set_alpha__set_colorkey_rle(self):
         pygame.display.init()
         try:
