From 2384bee55a46eac44eb9d329be4c2a097e053ae1 Mon Sep 17 00:00:00 2001
From: worldofpeace <worldofpeace@users.noreply.github.com>
Date: Tue, 17 Jul 2018 07:04:18 -0400
Subject: [PATCH 1/1] 'sysconfdir' will be etc not /etc for install

---
 data/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/meson.build b/data/meson.build
index 7621b03..7c08eaf 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -20,7 +20,7 @@ i18n.merge_file (
 
 install_data(
     meson.project_name() + '.conf',
-    install_dir: join_paths(get_option('sysconfdir'), 'lightdm')
+    install_dir: join_paths(get_option('prefix'), 'etc', 'lightdm')
 )
 
 install_data(
-- 
2.17.1

