diff --git a/src/meson.build b/src/meson.build
index 2450c1a..a908d11 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -1,7 +1,7 @@
 conf_data = configuration_data()
 conf_data.set('CONF_DIR', join_paths(get_option('sysconfdir'), 'lightdm'))
 conf_data.set('GETTEXT_PACKAGE', meson.project_name())
-conf_data.set('GSD_DIR', '/usr/lib/gnome-settings-daemon/')
+conf_data.set('GSD_DIR', '@elementary_settings_daemon@')
 conf_data.set('VERSION', meson.project_version())
 config_header = configure_file (
     input: 'config.vala.in',
