diff -ruN xawtv-3.95/console/fs.h xawtv-3.95.new/console/fs.h
--- xawtv-3.95/console/fs.h	2003-02-14 15:14:04.000000000 +0100
+++ xawtv-3.95.new/console/fs.h	2006-01-03 22:16:06.000000000 +0100
@@ -1,5 +1,5 @@
 #ifndef X_DISPLAY_MISSING
-# include <FSlib.h>
+# include <X11/fonts/FSlib.h>
 
 struct fs_font {
     Font               font;
diff -ruN xawtv-3.95/console/Subdir.mk xawtv-3.95.new/console/Subdir.mk
--- xawtv-3.95/console/Subdir.mk	2003-02-14 15:14:04.000000000 +0100
+++ xawtv-3.95.new/console/Subdir.mk	2006-01-03 22:32:38.000000000 +0100
@@ -74,7 +74,7 @@
 # libraries to link
 console/fbtv     : LDLIBS  += \
 	$(THREAD_LIBS) $(CURSES_LIBS) $(LIRC_LIBS) $(ALSA_LIBS) \
-	$(FS_LIBS) -ljpeg -lm
+	$(FS_LIBS) -ljpeg -lm -lFS
 console/ttv      : LDLIBS  += $(THREAD_LIBS) $(AA_LIBS) -ljpeg -lm
 console/scantv   : LDLIBS  += $(THREAD_LIBS) $(VBI_LIBS) -ljpeg
 console/streamer : LDLIBS  += $(THREAD_LIBS) -ljpeg -lm
