diff -u -r wireguard-linux-compat-1.0.20200908/src/compat/compat-asm.h wireguard-linux-compat-1.0.20200908-lts/src/compat/compat-asm.h
--- wireguard-linux-compat-1.0.20200908/src/compat/compat-asm.h	2020-09-08 16:22:40.000000000 +0000
+++ wireguard-linux-compat-1.0.20200908-lts/src/compat/compat-asm.h	2020-11-10 15:05:43.720093522 +0000
@@ -40,7 +40,7 @@
 #undef pull
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 5, 0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 76)
 #define SYM_FUNC_START ENTRY
 #define SYM_FUNC_END ENDPROC
 #endif
