diff --git a/libzsync/zsync.c b/libzsync/zsync.c
index 793a426..783c349 100644
--- a/libzsync/zsync.c
+++ b/libzsync/zsync.c
@@ -116,7 +116,7 @@ struct zsync_state {
 };
 
 static int zsync_read_blocksums(struct zsync_state *zs, FILE * f,
-                                int rsum_bytes, int checksum_bytes,
+                                int rsum_bytes, unsigned int checksum_bytes,
                                 int seq_matches);
 static int zsync_sha1(struct zsync_state *zs, int fh);
 static int zsync_recompress(struct zsync_state *zs);
