diff --git a/src/librustc_back/archive.rs b/src/librustc_back/archive.rs
index 0bd4265..bbdab9b 100644
--- a/src/librustc_back/archive.rs
+++ b/src/librustc_back/archive.rs
@@ -54,7 +54,7 @@ fn run_ar(handler: &ErrorHandler, maybe_ar_prog: &Option<String>,
           paths: &[&Path]) -> ProcessOutput {
     let ar = match *maybe_ar_prog {
         Some(ref ar) => ar[],
-        None => "ar"
+        None => "@arPath@"
     };
     let mut cmd = Command::new(ar);
 
diff --git a/src/librustc_trans/back/link.rs b/src/librustc_trans/back/link.rs
index ec61d3a..8252939 100644
--- a/src/librustc_trans/back/link.rs
+++ b/src/librustc_trans/back/link.rs
@@ -346,8 +346,8 @@ pub fn mangle_internal_name_by_path_and_seq(path: PathElems, flav: &str) -> Stri
 
 pub fn get_cc_prog(sess: &Session) -> String {
     match sess.opts.cg.linker {
-        Some(ref linker) => return linker.to_string(),
-        None => sess.target.target.options.linker.clone(),
+        Some(ref linker) => linker.to_string(),
+        None => "@ccPath@".to_string(),
     }
 }
 
