diff -uNr fatsort-1.6.2.605.orig/Makefile fatsort-1.6.2.605.new/Makefile
--- fatsort-1.6.2.605.orig/Makefile	2019-11-16 16:40:27.000000000 +0100
+++ fatsort-1.6.2.605.new/Makefile	2020-05-10 21:34:34.820874026 +0200
@@ -1,4 +1,5 @@
-MANDIR=/usr/local/share/man/man1
+PREFIX?=/usr/local
+MANDIR=$(PREFIX)/share/man/man1
 
 INSTALL_FLAGS=-m 0755 -p -D
 
diff -uNr fatsort-1.6.2.605.orig/src/Makefile fatsort-1.6.2.605.new/src/Makefile
--- fatsort-1.6.2.605.orig/src/Makefile	2018-11-17 00:40:59.000000000 +0100
+++ fatsort-1.6.2.605.new/src/Makefile	2020-05-10 21:33:52.053391027 +0200
@@ -30,7 +30,7 @@
 		override CFLAGS += -D __CYGWIN__
 		override CFLAGS += -D __LINUX__
 		override LDFLAGS += -liconv
-		SBINDIR=/usr/local/sbin
+		SBINDIR=$(PREFIX)/sbin
 	endif
 else
 	ifdef MINGW
@@ -60,9 +60,9 @@
 			# OS X's install does not support the '-D' flag.
 			INSTALL_FLAGS=-m 0755 -p
 			# Mac OS X does not have a "/usr/local/sbin"
-			SBINDIR=/usr/local/bin
+			SBINDIR=$(PREFIX)/bin
 		else
-			SBINDIR=/usr/local/sbin
+			SBINDIR=$(PREFIX)/sbin
 		endif
 	endif
 endif
