diff -Naur bc-1.06.orig/bc/scan.l bc-1.06/bc/scan.l
--- bc-1.06.orig/bc/scan.l      2000-09-13 13:25:47.000000000 -0500
+++ bc-1.06/bc/scan.l   2002-07-09 14:36:34.000000000 -0500
@@ -143,7 +143,6 @@
 
 /* Definitions for readline access. */
 extern FILE *rl_instream;
-_PROTOTYPE(char *readline, (char *));
 
 /* rl_input puts upto MAX characters into BUF with the number put in
    BUF placed in *RESULT.  If the yy input file is the same as
