diff --git a/graalvm/native-unix-compile.sh b/graalvm/native-unix-compile.sh
index ec0e49f..e19a120 100755
--- a/graalvm/native-unix-compile.sh
+++ b/graalvm/native-unix-compile.sh
@@ -17,6 +17,7 @@ CLOJURE_LSP_XMX=${CLOJURE_LSP_XMX:-"-J-Xmx4g"}
 
 args=("-jar" "$CLOJURE_LSP_JAR"
       "-H:+ReportExceptionStackTraces"
+      "-H:-CheckToolchain"
       "--verbose"
       "--no-fallback"
       "--native-image-info"
