diff -ru android-ndk-r9d.old/build/tools/make-standalone-toolchain.sh android-ndk-r9d/build/tools/make-standalone-toolchain.sh
--- android-ndk-r9d.old/build/tools/make-standalone-toolchain.sh	2014-09-25 11:42:09.990500975 +0200
+++ android-ndk-r9d/build/tools/make-standalone-toolchain.sh	2014-09-25 11:43:06.097501636 +0200
@@ -252,6 +252,9 @@
 # Now copy the GCC toolchain prebuilt binaries
 run copy_directory "$TOOLCHAIN_PATH" "$TMPDIR"
 
+# Making it writable again
+chmod -R +w "$TMPDIR"
+
 # Replace soft-link mcld by real file
 ALL_LDS=`find $TMPDIR -name "*mcld"`
 for LD in $ALL_LDS; do
