diff -ru android-ndk-r10c.old/build/tools/make-standalone-toolchain.sh android-ndk-r10c/build/tools/make-standalone-toolchain.sh
--- android-ndk-r10c.old/build/tools/make-standalone-toolchain.sh	2014-10-16 03:46:32.000000000 +0200
+++ android-ndk-r10c/build/tools/make-standalone-toolchain.sh	2014-10-24 23:46:22.544928306 +0200
@@ -310,6 +310,9 @@
 # Now copy the GCC toolchain prebuilt binaries
 run copy_directory "$TOOLCHAIN_PATH" "$TMPDIR"
 
+# Making it writable again
+chmod -R +w "$TMPDIR"
+
 # Replace soft-link mcld by real file
 ALL_LDS=`find $TMPDIR -name "*mcld"`
 for LD in $ALL_LDS; do
