diff -r 8833d4892dfc Zope-3.2.1/configure
--- a/configure	Mon Aug 18 14:55:39 2008 +0200
+++ b/configure	Mon Aug 18 14:57:39 2008 +0200
@@ -21,7 +21,7 @@ prefix="$DEFAULT_PREFIX"
 
 # Place the optimal target version number (as returned by sys.version)
 # below
-TARGET="2.4.2"
+TARGET="2.4.4"
 
 # Order a list of "acceptable" python version numbers (as returned by
 # sys.version) below in "best" to "worst" order, not including the
