diff -Nrc SuiteSparse/UMFPACK/Demo/Makefile SuiteSparse-new/UMFPACK/Demo/Makefile
*** SuiteSparse/UMFPACK/Demo/Makefile	2009-11-11 21:09:45.000000000 +0100
--- SuiteSparse-new/UMFPACK/Demo/Makefile	2010-08-02 12:53:16.000000000 +0200
***************
*** 40,46 ****
  ../../CAMD/Lib/libcamd.a:
  	( cd ../../CAMD ; $(MAKE) library )
  
- $(METIS):
- 	( cd $(METIS_PATH) && $(MAKE) )
  
  UMFPACK = ../Lib/libumfpack.a ../../AMD/Lib/libamd.a \
--- 40,44 ----
  ../../CAMD/Lib/libcamd.a:
  	( cd ../../CAMD ; $(MAKE) library )
  
  
  UMFPACK = ../Lib/libumfpack.a ../../AMD/Lib/libamd.a \
