*** cdparanoia-III-alpha9.8/interface/utils.h	Thu Apr 20 00:41:04 2000
--- cdparanoia-III-alpha9.8-old/interface/utils.h	Wed Jan 19 21:44:08 2005
***************
*** 110,117 ****
      case CDDA_MESSAGE_LOGIT:
        d->errorbuf=catstring(d->errorbuf,s);
        break;
-     case CDDA_MESSAGE_FORGETIT:
-     default:
      }
    }
  }
--- 110,115 ----
***************
*** 125,132 ****
      case CDDA_MESSAGE_LOGIT:
        d->messagebuf=catstring(d->messagebuf,s);
        break;
-     case CDDA_MESSAGE_FORGETIT:
-     default:
      }
    }
  }
--- 123,128 ----
***************
*** 167,174 ****
  	}
        }
        break;
-     case CDDA_MESSAGE_FORGETIT:
-     default:
      }
    }
    if(malloced)free(buffer);
--- 163,168 ----
***************
*** 203,210 ****
  	if(!malloced)*messages=catstring(*messages,"\n");
  	}
        break;
-     case CDDA_MESSAGE_FORGETIT:
-     default:
      }
    }
    if(malloced)free(buffer);
--- 197,202 ----
