diff --git a/CMakeLists.txt b/CMakeLists.txt
index 01db0fb1..06319c54 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -166,12 +166,12 @@ if(NOT WITHOUT_X11)
 endif()
 
 if (QT_MAJOR_VERSION STREQUAL "5")
-    find_package(kImageAnnotator)
-    set_package_properties(kImageAnnotator PROPERTIES URL "https://github.com/ksnip/kImageAnnotator" DESCRIPTION "The kImageAnnotator library provides tools to annotate" TYPE REQUIRED)
-    if(kImageAnnotator_FOUND)
+    find_package(kImageAnnotator-Qt5)
+    set_package_properties(kImageAnnotator-Qt5 PROPERTIES URL "https://github.com/ksnip/kImageAnnotator" DESCRIPTION "The kImageAnnotator library provides tools to annotate" TYPE REQUIRED)
+    if(kImageAnnotator-Qt5_FOUND)
         set(KIMAGEANNOTATOR_FOUND 1)
-        find_package(kColorPicker REQUIRED)
-        if(NOT kImageAnnotator_VERSION VERSION_LESS 0.5.0)
+        find_package(kColorPicker-Qt5 REQUIRED)
+        if(NOT kImageAnnotator-Qt5_VERSION VERSION_LESS 0.5.0)
             set(KIMAGEANNOTATOR_CAN_LOAD_TRANSLATIONS 1)
         endif()
     endif()
diff --git a/app/CMakeLists.txt b/app/CMakeLists.txt
index 8c136835..ef4cff74 100644
--- a/app/CMakeLists.txt
+++ b/app/CMakeLists.txt
@@ -157,6 +157,6 @@ target_link_libraries(slideshowfileitemaction
     KF${QT_MAJOR_VERSION}::KIOWidgets
     KF${QT_MAJOR_VERSION}::Notifications)
 
-if(kImageAnnotator_FOUND)
+if(kImageAnnotator-Qt5_FOUND)
     target_link_libraries(gwenview kImageAnnotator::kImageAnnotator)
 endif()
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 05a2ea67..4167a1bb 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -157,7 +157,7 @@ set(gwenviewlib_SRCS
     touch/touch_helper.cpp
     ${GV_JPEG_DIR}/transupp.c
     )
-if (kImageAnnotator_FOUND)
+if (kImageAnnotator-Qt5_FOUND)
     set(gwenviewlib_SRCS ${gwenviewlib_SRCS}
         annotate/annotatedialog.cpp
         annotate/annotateoperation.cpp
@@ -338,7 +338,7 @@ if (GWENVIEW_SEMANTICINFO_BACKEND_BALOO)
         )
 endif()
 
-if(kImageAnnotator_FOUND)
+if(kImageAnnotator-Qt5_FOUND)
     target_link_libraries(gwenviewlib kImageAnnotator::kImageAnnotator)
 endif()
 
