diff --git a/build.rs b/build.rs
index ef46b47..eefbb17 100644
--- a/build.rs
+++ b/build.rs
@@ -1,14 +1,7 @@
 use std::{io::Result, process::Command};
 
 fn main() -> Result<()> {
-    let output = Command::new("git")
-        .args(&["rev-parse", "--short", "HEAD"])
-        .output()
-        .expect("Failed to get git commit hash");
-    let git_hash = String::from_utf8(output.stdout)
-        .expect("Invalid UTF-8")
-        .trim()
-        .to_string();
+    let git_hash = std::fs::read_to_string("COMMIT_HASH").unwrap();
     println!("cargo:rustc-env=GIT_HASH={}", git_hash);
     Ok(())
 }
