diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1007d38..c945de2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1057,7 +1057,7 @@ if( KICAD_BUILD_I18N )
 endif()
 
 if( APPLE )
-    set( KICAD_OSX_CODESIGN ON
+    set( KICAD_OSX_CODESIGN OFF
             CACHE BOOL "Sign KiCad.app on macOS" FORCE )
     set( KICAD_OSX_SIGNING_ID "-"
             CACHE STRING "macOS Signing ID, defaults to ad-hoc" FORCE )
