diff --git a/scwx-qt/scwx-qt.cmake b/scwx-qt/scwx-qt.cmake
index 8743b2ab..afa7c407 100644
--- a/scwx-qt/scwx-qt.cmake
+++ b/scwx-qt/scwx-qt.cmake
@@ -23,7 +23,7 @@ find_package(OpenGL REQUIRED)
 find_package(PNG)
 find_package(Python COMPONENTS Interpreter)
 find_package(SQLite3)
-find_package(TIFF)
+find_package(Tiff CONFIG REQUIRED)
 
 find_package(QT NAMES Qt6
              COMPONENTS Gui
@@ -875,7 +875,7 @@ target_link_libraries(scwx-qt PUBLIC Qt${QT_VERSION_MAJOR}::Widgets
                                      qt6ct-widgets
                                      SQLite::SQLite3
                                      PNG::PNG
-                                     TIFF::TIFF
+                                     TIFF::tiffxx
                                      wxdata)
 
 target_link_libraries(scwx-qt INTERFACE Boost::program_options)
