diff --git a/module/spl/spl-proc.c b/module/spl/spl-proc.c
index f25239a..b731123 100644
--- a/module/spl/spl-proc.c
+++ b/module/spl/spl-proc.c
@@ -38,7 +38,7 @@
 
 #define SS_DEBUG_SUBSYS SS_PROC
 
-#if defined(CONSTIFY_PLUGIN) && LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0)
+#if defined(CONSTIFY_PLUGIN)
 typedef struct ctl_table __no_const spl_ctl_table;
 #else
 typedef struct ctl_table spl_ctl_table;
