diff --git a/tools/shell/CMakeLists.txt b/tools/shell/CMakeLists.txt
index 4e0352d79d..4f0441a090 100644
--- a/tools/shell/CMakeLists.txt
+++ b/tools/shell/CMakeLists.txt
@@ -28,4 +28,4 @@ set_target_properties(shell PROPERTIES OUTPUT_NAME duckdb)
 set_target_properties(shell PROPERTIES RUNTIME_OUTPUT_DIRECTORY
                                        ${PROJECT_BINARY_DIR})
 
-install(TARGETS shell RUNTIME DESTINATION "${PROJECT_BINARY_DIR}")
+install(TARGETS shell RUNTIME DESTINATION "${INSTALL_BIN_DIR}")
