diff -Naur source-old/ruby/GNUmakefile source-new/ruby/GNUmakefile
--- source-old/ruby/GNUmakefile    2024-01-23 16:12:41.009951705 +0000
+++ source-new/ruby/GNUmakefile    2024-01-23 16:13:54.619174062 +0000
@@ -29,20 +29,9 @@
       ruby += input.sdl
     endif
   else ifeq ($(platform),linux)
-    pkg_check = $(if $(shell $(pkg_config) $1 && echo 1),$2)
-    ruby += video.glx video.glx2 video.xshm
-    ruby += $(call pkg_check,xv,video.xvideo)
-    ruby += audio.oss audio.alsa
-    ruby += $(call pkg_check,openal,audio.openal)
-    ruby += $(call pkg_check,libpulse,audio.pulseaudio)
-    ruby += $(call pkg_check,libpulse-simple,audio.pulseaudiosimple)
-    ruby += $(call pkg_check,ao,audio.ao)
-    ruby += input.xlib
-    ruby += $(call pkg_check,libudev,input.udev)
-    ifeq ($(sdl2),true)
-      ruby += $(call pkg_check,sdl2,input.sdl)
-      ruby += $(call pkg_check,sdl2,audio.sdl)
-    endif
+    ruby += video.glx video.glx2 video.xshm video.xvideo
+    ruby += audio.oss audio.alsa audio.openal audio.pulseaudio audio.pulseaudiosimple audio.ao
+    ruby += input.xlib input.udev input.sdl
   else ifeq ($(platform),bsd)
     pkg_check = $(if $(shell $(pkg_config) $1 && echo 1),$2)
     ruby += video.glx video.glx2 video.xshm
