diff -rc apg-2.2.3/Makefile apg-2.2.3-new/Makefile
*** apg-2.2.3/Makefile	2003-08-07 17:40:30.000000000 +0200
--- apg-2.2.3-new/Makefile	2013-07-24 12:25:31.159938436 +0200
***************
*** 113,131 ****
  	if test -x ./apg; then \
  ./mkinstalldirs ${INSTALL_PREFIX}${APG_BIN_DIR}; \
  ./mkinstalldirs ${INSTALL_PREFIX}${APG_MAN_DIR}; \
! ./install-sh -c -m 0755 -o root -g ${FIND_GROUP} ./apg ${INSTALL_PREFIX}${APG_BIN_DIR}; \
  ./install-sh -c -m 0444 ./doc/man/apg.1 ${INSTALL_PREFIX}${APG_MAN_DIR}; \
  fi
  	if test -x ./apgd; then \
  ./mkinstalldirs ${INSTALL_PREFIX}${APGD_BIN_DIR}; \
  ./mkinstalldirs ${INSTALL_PREFIX}${APGD_MAN_DIR}; \
! ./install-sh -c -m 0755 -o root -g ${FIND_GROUP} ./apgd ${INSTALL_PREFIX}${APGD_BIN_DIR}; \
  ./install-sh -c -m 0444 ./doc/man/apgd.8 ${INSTALL_PREFIX}${APGD_MAN_DIR}; \
  fi
  	if test -x ./apgbfm; then \
  ./mkinstalldirs ${INSTALL_PREFIX}${APG_BIN_DIR}; \
  ./mkinstalldirs ${INSTALL_PREFIX}${APG_MAN_DIR}; \
! ./install-sh -c -m 0755 -o root -g ${FIND_GROUP} ./apgbfm ${INSTALL_PREFIX}${APG_BIN_DIR}; \
  ./install-sh -c -m 0444 ./doc/man/apgbfm.1 ${INSTALL_PREFIX}${APG_MAN_DIR}; \
  fi
  
--- 113,131 ----
  	if test -x ./apg; then \
  ./mkinstalldirs ${INSTALL_PREFIX}${APG_BIN_DIR}; \
  ./mkinstalldirs ${INSTALL_PREFIX}${APG_MAN_DIR}; \
! ./install-sh -c -m 0755 ./apg ${INSTALL_PREFIX}${APG_BIN_DIR}; \
  ./install-sh -c -m 0444 ./doc/man/apg.1 ${INSTALL_PREFIX}${APG_MAN_DIR}; \
  fi
  	if test -x ./apgd; then \
  ./mkinstalldirs ${INSTALL_PREFIX}${APGD_BIN_DIR}; \
  ./mkinstalldirs ${INSTALL_PREFIX}${APGD_MAN_DIR}; \
! ./install-sh -c -m 0755 ./apgd ${INSTALL_PREFIX}${APGD_BIN_DIR}; \
  ./install-sh -c -m 0444 ./doc/man/apgd.8 ${INSTALL_PREFIX}${APGD_MAN_DIR}; \
  fi
  	if test -x ./apgbfm; then \
  ./mkinstalldirs ${INSTALL_PREFIX}${APG_BIN_DIR}; \
  ./mkinstalldirs ${INSTALL_PREFIX}${APG_MAN_DIR}; \
! ./install-sh -c -m 0755 ./apgbfm ${INSTALL_PREFIX}${APG_BIN_DIR}; \
  ./install-sh -c -m 0444 ./doc/man/apgbfm.1 ${INSTALL_PREFIX}${APG_MAN_DIR}; \
  fi
  
