--- a/trashcli/list_mount_points.py	2014-12-23 10:10:43.808470486 +0100
+++ a/trashcli/list_mount_points.py	2014-12-23 10:19:04.954796457 +0100
@@ -12,7 +12,7 @@ def mount_points_from_getmnt():
 
 def mount_points_from_df():
     import subprocess
-    df_output = subprocess.Popen(["df", "-P"], stdout=subprocess.PIPE).stdout
+    df_output = subprocess.Popen(["@df@", "-P"], stdout=subprocess.PIPE).stdout
     return list(_mount_points_from_df_output(df_output))
 
 def _mount_points_from_df_output(df_output):
@@ -46,13 +46,7 @@ def _mounted_filesystems_from_getmnt() :
                     ("mnt_freq", c_int),       # Dump frequency (in days).
                     ("mnt_passno", c_int)]     # Pass number for `fsck'.
 
-    if sys.platform == "cygwin":
-        libc_name = "cygwin1.dll"
-    else:
-        libc_name = find_library("c")
-
-    if libc_name == None :
-        libc_name="/lib/libc.so.6" # fix for my Gentoo 4.0
+    libc_name = "@libc@"
 
     libc = cdll.LoadLibrary(libc_name)
     libc.getmntent.restype = POINTER(mntent_struct)
