diff --git a/misaki/espeak.py b/misaki/espeak.py
index e5d880c..42c2596 100644
--- a/misaki/espeak.py
+++ b/misaki/espeak.py
@@ -1,13 +1,12 @@
 from phonemizer.backend.espeak.wrapper import EspeakWrapper
 from typing import Tuple
-import espeakng_loader
 import phonemizer
 import re
 
 # Set espeak-ng library path and espeak-ng-data
-EspeakWrapper.set_library(espeakng_loader.get_library_path())
+EspeakWrapper.set_library("@espeak-library-path@")
 # Change data_path as needed when editing espeak-ng phonemes
-EspeakWrapper.set_data_path(espeakng_loader.get_data_path())
+EspeakWrapper.set_data_path("@espeak-data-path@")
 
 # EspeakFallback is used as a last resort for English
 class EspeakFallback:
