diff -ubr git-1.5.6.1-orig/git-submodule.sh git-1.5.6.1/git-submodule.sh
--- git-1.5.6.1-orig/git-submodule.sh	2008-06-28 16:28:54.000000000 +0200
+++ git-1.5.6.1/git-submodule.sh	2008-06-28 16:29:55.000000000 +0200
@@ -30,7 +30,7 @@
 # NEEDSWORK: identical function exists in get_repo_base in clone.sh
 get_repo_base() {
 	(
-		cd "`/bin/pwd`" &&
+		cd "`pwd`" &&
 		cd "$1" || cd "$1.git" &&
 		{
 			cd .git
