diff -aur k2pdfopt_v2.21/willuslib/array.c k2pdfopt_v2.21.new/willuslib/array.c
--- k2pdfopt_v2.21/willuslib/array.c	2014-05-23 16:29:58.000000000 -0300
+++ k2pdfopt_v2.21.new/willuslib/array.c	2014-07-26 11:35:49.829825567 -0300
@@ -1055,7 +1055,7 @@
 void arrayf_sort(float *a,int n)
 
     {
-    sort(a,(long)n);
+    willus_sort(a,(long)n);
     }
 
 
diff -aur k2pdfopt_v2.21/willuslib/math.c k2pdfopt_v2.21.new/willuslib/math.c
--- k2pdfopt_v2.21/willuslib/math.c	2013-08-15 21:33:50.000000000 -0300
+++ k2pdfopt_v2.21.new/willuslib/math.c	2014-07-26 11:36:02.853170659 -0300
@@ -532,7 +532,7 @@
 
 
 
-void sort(float *x,int n)
+void willus_sort(float *x,int n)
 
     {
     int top,n1;
diff -aur k2pdfopt_v2.21/willuslib/ocrjocr.c k2pdfopt_v2.21.new/willuslib/ocrjocr.c
--- k2pdfopt_v2.21/willuslib/ocrjocr.c	2012-11-12 13:09:42.000000000 -0300
+++ k2pdfopt_v2.21.new/willuslib/ocrjocr.c	2014-07-26 11:36:46.699837185 -0300
@@ -29,6 +29,8 @@
 #ifdef HAVE_GOCR_LIB
 #include <gocr.h>
 
+job_t *JOB;
+
 /*
 ** bmp8 must be grayscale
 ** (x1,y1) and (x2,y2) from top left of bitmap
@@ -66,6 +68,7 @@
     h=y2-y1+1;
     dh=h+bw*2;
     job=&_job;
+    JOB=job;
     job_init(job);
     job_init_image(job);
     // willus_mem_alloc_warn((void **)&job->src.p.p,w*h,funcname,10);
diff -aur k2pdfopt_v2.21/willuslib/string.c k2pdfopt_v2.21.new/willuslib/string.c
--- k2pdfopt_v2.21/willuslib/string.c	2014-02-03 00:37:44.000000000 -0300
+++ k2pdfopt_v2.21.new/willuslib/string.c	2014-07-26 11:37:01.766506277 -0300
@@ -81,7 +81,7 @@
 ** Returns NULL if EOF, otherwise returns pointer to the string.
 **
 */
-char *get_line(char *buf,int max,FILE *f)
+char *willus_get_line(char *buf,int max,FILE *f)
 
     {
     int     i;
diff -aur k2pdfopt_v2.21/willuslib/willus.h k2pdfopt_v2.21.new/willuslib/willus.h
--- k2pdfopt_v2.21/willuslib/willus.h	2014-07-25 15:03:51.000000000 -0300
+++ k2pdfopt_v2.21.new/willuslib/willus.h	2014-07-26 11:37:56.316506038 -0300
@@ -214,9 +214,6 @@
 ** CMAKE handles the defines, not this source
 ** (Mod from Dirk Thierbach, 31-Dec-2013)
 */
-#ifdef USE_CMAKE
-#include "config.h"
-#else /* USE_CMAKE */
 
 #ifndef HAVE_Z_LIB
 #define HAVE_Z_LIB
@@ -268,7 +265,6 @@
 #undef HAVE_GSL_LIB
 #endif
 
-#endif /* USE_CMAKE */
 /*
 ** Consistency check
 */
@@ -533,7 +529,7 @@
                       int *n,FILE *err);
 int    readxyz_ex    (char *filename,double **x,double **y,double **z,
                       int *n,FILE *err,int ignore_after_semicolon);
-void   sort          (float *x,int n);
+void   willus_sort          (float *x,int n);
 void   sortd         (double *x,int n);
 void   sorti         (int *x,int n);
 void   sortxy        (float *x,float *y,int n);
@@ -602,7 +598,7 @@
 /* string.c */
 void   clean_line    (char *buf);
 void   clean_line_end(char *buf);
-char  *get_line      (char *buf,int max,FILE *f);
+char  *willus_get_line      (char *buf,int max,FILE *f);
 char  *get_line_cf   (char *buf,int max,FILE *f);
 int   mem_get_line_cf(char *buf,int maxlen,char *cptr,long *cindex,long csize);
 int    in_string     (char *buffer,char *pattern);
