From d9df7aaaaf9c758499f569376a041045d99e4015 Mon Sep 17 00:00:00 2001
From: Bob van der Linden <bobvanderlinden@gmail.com>
Date: Thu, 9 Feb 2023 16:17:46 +0100
Subject: [PATCH 1/2] cargo: lock: update version

---
 Cargo.lock | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.lock b/Cargo.lock
index 79a81d1..374c10a 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -642,7 +642,7 @@ dependencies = [
 
 [[package]]
 name = "rubyfmt-main"
-version = "0.8.0-pre"
+version = "0.8.1"
 dependencies = [
  "atty",
  "clap",
-- 
2.39.1

