diff --git a/src/transformers/integrations/fsdp.py b/src/transformers/integrations/fsdp.py
index 7bcb11fe7..332231c04 100644
--- a/src/transformers/integrations/fsdp.py
+++ b/src/transformers/integrations/fsdp.py
@@ -26,6 +26,11 @@ def is_fsdp_managed_module(module: nn.Module) -> bool:
     if not is_torch_available():
         return False
 
+    import torch
+
+    if not torch.distributed.is_available():
+        return False
+
     import torch.distributed.fsdp
 
     return isinstance(module, torch.distributed.fsdp.FullyShardedDataParallel) or getattr(
