--- a/common/common.c
+++ b/common/common.c
@@ -991,6 +991,12 @@ ssize_t select_write(const int fd, const void *buf, const size_t buflen, const t
  * communications media and/or vendor protocol.
  */
 static const char * search_paths[] = {
+	"@avahi@",
+	"@freeipmi@",
+	"@libusb@",
+	"@neon@",
+	"@libmodbus@",
+	"@netsnmp@",
 	/* Use the library path (and bitness) provided during ./configure first */
 	LIBDIR,
 	"/usr"LIBDIR,
