diff --git a/src/pikepdf/_methods.py b/src/pikepdf/_methods.py
index da40043f..4f566f01 100644
--- a/src/pikepdf/_methods.py
+++ b/src/pikepdf/_methods.py
@@ -74,7 +74,7 @@ def _mudraw(buffer, fmt) -> bytes:
         tmp_in.flush()
 
         proc = run(
-            ['mutool', 'draw', '-F', fmt, '-o', '-', tmp_in.name],
+            ['@mutool@', 'draw', '-F', fmt, '-o', '-', tmp_in.name],
             capture_output=True,
             check=True,
         )
diff --git a/src/pikepdf/jbig2.py b/src/pikepdf/jbig2.py
index f89b4f90..f187ebdf 100644
--- a/src/pikepdf/jbig2.py
+++ b/src/pikepdf/jbig2.py
@@ -63,7 +63,7 @@ class JBIG2Decoder(JBIG2DecoderInterface):
             output_path = Path(tmpdir) / "outfile"
 
             args = [
-                "jbig2dec",
+                "@jbig2dec@",
                 "--embedded",
                 "--format",
                 "png",
@@ -90,7 +90,7 @@ class JBIG2Decoder(JBIG2DecoderInterface):
     def _version(self) -> Version:
         try:
             proc = self._run(
-                ['jbig2dec', '--version'], stdout=PIPE, check=True, encoding='ascii'
+                ['@jbig2dec@', '--version'], stdout=PIPE, check=True, encoding='ascii'
             )
         except (CalledProcessError, FileNotFoundError) as e:
             raise DependencyError("jbig2dec - not installed or not found") from e
