From 542baf59f3b2eada13bde48e8a89e484c2cfeeef Mon Sep 17 00:00:00 2001
From: Hanaasagi <ambiguous404@gmail.com>
Date: Wed, 17 Apr 2019 06:46:03 +0000
Subject: [PATCH] Support Python 3.7 Generator (PEP 479)

---
 kernprof.py      | 2 ++
 line_profiler.py | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/kernprof.py b/kernprof.py
index 108d36e..6461cf6 100755
--- a/kernprof.py
+++ b/kernprof.py
@@ -102,6 +102,8 @@ def wrapper(*args, **kwds):
                 self.enable_by_count()
                 try:
                     item = g.send(input)
+                except StopIteration:
+                    return
                 finally:
                     self.disable_by_count()
                 input = (yield item)
diff --git a/line_profiler.py b/line_profiler.py
index a481dd2..5744d05 100755
--- a/line_profiler.py
+++ b/line_profiler.py
@@ -100,6 +100,8 @@ def wrapper(*args, **kwds):
                 self.enable_by_count()
                 try:
                     item = g.send(input)
+                except StopIteration:
+                    return
                 finally:
                     self.disable_by_count()
                 input = (yield item)
