diff -ru -x '*~' BayesLogit_orig/src/Makevars BayesLogit/src/Makevars
--- BayesLogit_orig/src/Makevars	2014-04-24 23:31:13.000000000 +0900
+++ BayesLogit/src/Makevars	2014-10-25 18:33:32.398572641 +0900
@@ -6,7 +6,7 @@
 ## W/OUT Dynamic Stuff
 OBJECTS = Matrix.o MatrixFrame.o RRNG.o RNG.o FSF_nmix.o LogitWrapper.o \
 	PolyaGamma.o PolyaGammaAlt.o PolyaGammaSP.o InvertY.o
-PKG_LIBS = $(BLAS_LIBS) $(FLIBS) $(LAPACK_LIBS) 
+PKG_LIBS = $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)
 # PKG_CPPFLAGS = -DUSE_R -DNDEBUG -DDISABLE_SINGLE -DNTHROW -Wall -pedantic -Wextra
 PKG_CPPFLAGS = -DUSE_R -DNDEBUG -DDISABLE_SINGLE -DNTHROW 
 
