diff -ru -x '*~' pbdZMQ_orig/R/zzz.r.in pbdZMQ/R/zzz.r.in
--- pbdZMQ_orig/R/zzz.r.in	2017-07-12 23:46:22.000000000 +0900
+++ pbdZMQ/R/zzz.r.in	2017-07-12 23:58:24.000000000 +0900
@@ -31,7 +31,7 @@
   ### Overwrite RPATH from the shared library installed to the destination.
   if(Sys.info()[['sysname']] == "Darwin"){
     dest <- gsub("/$", "", dn)
-    cmd.int <- system("which install_name_tool", intern = TRUE)
+    cmd.int <- "install_name_tool"
     fn.pbdZMQ.so <- file.path(dest, "pbdZMQ.so")
     fn.libzmq.4.dylib <- file.path(dest, "libzmq.4.dylib")
 
@@ -40,7 +40,7 @@
        file.exists(fn.libzmq.4.dylib)){
 
       ### Check orginal path.
-      cmd.ot <- system("which otool", intern = TRUE)
+      cmd.ot <- "otool"
       # if(length(grep("otool", cmd.ot)) > 0){
         rpath <- system(paste(cmd.ot, " -L ", fn.pbdZMQ.so, sep = ""),
                         intern = TRUE)
diff -ru -x '*~' pbdZMQ_orig/src/install.libs.R pbdZMQ/src/install.libs.R
--- pbdZMQ_orig/src/install.libs.R	2017-07-12 23:46:45.000000000 +0900
+++ pbdZMQ/src/install.libs.R	2017-07-12 23:40:44.000000000 +0900
@@ -14,7 +14,7 @@
   ### For Mac OSX 10.10 Yosemite and when "internal ZMQ" is asked.
   ### Overwrite RPATH from the shared library installed to the destination.
   if(Sys.info()[['sysname']] == "Darwin" && ("libzmq.4.dylib" %in% files)){
-    cmd.int <- system("which install_name_tool", intern = TRUE)
+    cmd.int <- "install_name_tool"
     fn.pbdZMQ.so <- file.path(dest, "pbdZMQ.so")
     fn.libzmq.4.dylib <- file.path(dest, "libzmq.4.dylib")
 
@@ -22,7 +22,7 @@
        file.exists(fn.pbdZMQ.so) &&
        file.exists(fn.libzmq.4.dylib)){
 
-      cmd.ot <- system("which otool", intern = TRUE) 
+      cmd.ot <- "otool"
       if(length(grep("otool", cmd.ot)) > 0){
         rpath <- system(paste(cmd.ot, " -L ", fn.pbdZMQ.so, sep = ""),
                         intern = TRUE)
