diff -ru -x '*~' qtbase_orig/src/Makefile qtbase/src/Makefile
--- qtbase_orig/src/Makefile	2012-03-03 03:57:47.000000000 +0900
+++ qtbase/src/Makefile	2014-11-01 23:06:51.383876323 +0900
@@ -10,6 +10,9 @@
 	      -DR_INCLUDES=${R_INCLUDE_DIR} \
 	      -DCMAKE_INSTALL_PREFIX=../src; \
 	make install
+# I could not control RPATH via CMake. HELP WANTED!
+	patchelf --set-rpath ${CMAKE_LIBRARY_PATH} qtbase.so
+	find ../inst/local/ -executable -type f -print0 | xargs -0 patchelf --set-rpath ${CMAKE_LIBRARY_PATH} 
 
 clean:
 	rm -rf ../kdebindings-build
