diff -ru -x '*~' rpud_orig/configure rpud/configure
--- rpud_orig/configure	2010-09-08 02:14:55.000000000 +0900
+++ rpud/configure	2014-10-25 16:46:39.479098648 +0900
@@ -1794,7 +1794,7 @@
 fi
 
 LIBS="-L${CUDA_HOME}${CUDA_LIB_DIR} -lcublas -lcuda"
-RPATHFLAG="-Wl,-rpath,${CUDA_HOME}${CUDA_LIB_DIR}"
+RPATHFLAG="-Xlinker -rpath=${CUDA_HOME}${CUDA_LIB_DIR}"
 
 
 
diff -ru -x '*~' rpud_orig/src/Makefile.in rpud/src/Makefile.in
--- rpud_orig/src/Makefile.in	2010-08-31 01:53:50.000000000 +0900
+++ rpud/src/Makefile.in	2014-10-25 16:45:30.248109883 +0900
@@ -3,11 +3,11 @@
 OBJS := rpud.o rpudist.o
 
 #compiler/preprocessor options
-INCS := -I@CUDA_HOME@/include
+INCS := -I@CUDA_HOME@/usr_include
 PARAMS := -Xcompiler "@R_INCLUDE@ @CPICFLAGS@"
 
 #linker options
-LD_PARAMS := -Xlinker "@R_LIB@ @RPATHFLAG@"
+LD_PARAMS := -Xlinker "--export-dynamic -fopenmp -L${R_HOME}/lib -lR @RPATHFLAG@" 
 LIBS :=  @LIBS@
 
 TARGETS := rpud.so
