diff -ru -x '*~' WideLM_orig/src/Makefile.in WideLM/src/Makefile.in
--- WideLM_orig/src/Makefile.in	2012-02-17 04:28:05.000000000 +0900
+++ WideLM/src/Makefile.in	2014-10-25 18:54:49.110011921 +0900
@@ -12,12 +12,12 @@
 #compiler/preprocessor options
 R_INCS := @R_INCL@ @RCPP_INCL@
 CC_ARGS := @CU_PARAMS@ -Xcompiler @R_CPIC@
-CU_INCS := -I@CUDA_HOME@/include
+CU_INCS := -I@CUDA_HOME@/usr_include
 CU_ARCH := -gencode arch=compute_20,code=sm_20
 
 #linker options
-LD_PARAMS := -Xlinker "@RCPP_LDFLAGS@"
-RCU_LIBS :=  @R_LIB@ -L@CU_LIBDIR@ -lcublas
+LD_PARAMS := -Xlinker "--export-dynamic -fopenmp -L${R_HOME}/lib -lR"
+RCU_LIBS := -Xlinker "-L@CU_LIBDIR@ -lcublas"
 
 all : WideLM.so
 
