diff -Naur imake-1.0.7-orig/imake.c imake-1.0.7/imake.c
--- imake-1.0.7-orig/imake.c	2013-08-17 10:11:50.000000000 +0000
+++ imake-1.0.7/imake.c	2015-02-04 09:41:59.000000000 +0000
@@ -1045,6 +1045,7 @@
 static void
 get_distrib(FILE *inFile)
 {
+#if 0
   struct stat sb;
 
   static const char*   suse = "/etc/SuSE-release";
@@ -1090,6 +1091,7 @@
      * at the content of /etc/debian_version */
     return;
   }
+#endif  
   /* what's the definitive way to tell what any particular distribution is? */
 
   fprintf (inFile, "%s\n", "#define DefaultLinuxDistribution LinuxUnknown");
@@ -1337,6 +1339,7 @@
 static boolean
 get_gcc(char *cmd)
 {
+#if 0
   struct stat sb;
     static const char* gcc_path[] = {
 #if defined(linux) || \
@@ -1385,6 +1388,9 @@
 	}
       }
     return FALSE;
+#endif
+    strcpy (cmd, IMAKE_COMPILETIME_CPP);
+    return TRUE;
 }
 
 #ifdef CROSSCOMPILE
