--- a/src/ocb_stubblr.ml	1970-01-01 00:00:01.000000000 +0000
+++ b/src/ocb_stubblr.ml	2016-12-04 11:10:10.000000000 +0000
@@ -31,20 +31,9 @@
 
   (* XXX Would be nice to move pkg-config results to a build artefact. *)
 
-  let opam_prefix =
-    let cmd = "opam config var prefix" in
-    lazy ( try run_and_read cmd with Failure _ ->
-            error_msgf "error running opam")
-
-  let var = "PKG_CONFIG_PATH"
-
-  let path () =
-    Lazy.force opam_prefix / "lib" / "pkgconfig" ::
-      (try [Sys.getenv var] with Not_found -> []) |> String.concat ~sep:":"
-
   let run ~flags package =
-    let cmd = strf "%s=%s pkg-config %s %s 2>/dev/null"
-              var (path ()) package (String.concat ~sep:" " flags) in
+    let cmd = strf "pkg-config %s %s 2>/dev/null"
+              package (String.concat ~sep:" " flags) in
     try `Res (run_and_read cmd) with Failure _ -> `Nonexistent
 end
 
