diff -puNr prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs/inode.c prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs/inode.c
--- prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs/inode.c
+++ prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs/inode.c
@@ -851,7 +851,7 @@ ssize_t prlfs_rw(struct inode *inode, char *buf, size_t size,
 		                loff_t *off, unsigned int rw, int user, int flags);
 
 
-int prlfs_readpage(struct file *file, struct page *page) {
+int prlfs_read_folio(struct file *file, struct folio *folio) {
 	char *buf;
 	ssize_t ret;
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 9, 0)
@@ -859,6 +859,7 @@ int prlfs_readpage(struct file *file, struct page *page) {
 #else
 	struct inode *inode = file->f_dentry->d_inode;
 #endif
+	struct page *page = &folio->page;
 	loff_t off = page->index << PAGE_SHIFT;
 
 	if (!file) {
@@ -950,7 +951,7 @@ out:
 }
 
 static const struct address_space_operations prlfs_aops = {
-	.readpage		= prlfs_readpage,
+	.read_folio		= prlfs_read_folio,
 	.writepage		= prlfs_writepage,
 	.write_begin    = simple_write_begin,
 	.write_end      = prlfs_write_end,
