diff --git a/script/rofi-theme-selector b/script/rofi-theme-selector
index 0646e4bc..f82e5ad9 100755
--- a/script/rofi-theme-selector
+++ b/script/rofi-theme-selector
@@ -42,34 +42,8 @@ function find_themes()
     DIRS=${XDG_DATA_DIRS}
     OLDIFS=${IFS}
     IFS=:
-    if [ -z "${XDG_DATA_DIRS}" ]
-    then
-        echo "XDG_DATA_DIRS needs to be set for this script to function correctly."
-        echo -n "Using dirs from \$PATH: "
-        DIRS=
-        # Iterate over items in $PATH
-        for p in ${PATH}; do
-            # Remove trailing / if exists.
-            x=${p%/}
-            # remove both /bin and /sbin and /games from end
-            x=${x%/bin}
-            x=${x%/sbin}
-            x=${x%/games}
-            # Add /share
-            x=${x}/share
-            # Check if entry exists Prepend : so :${x}: matches nicely
-            case ":${DIRS}" in
-                *$x:*);;
-                *) DIRS+="$x:";;
-            esac
-        done
-        # Remove trailing :
-        DIRS=${DIRS%:}
-        echo "${DIRS}"
-    fi
-    # Add user dir.
-    DIRS+=":${HOME}/.local/share/"
-    DIRS+=":${HOME}/.config/"
+
+    DIRS+=":%ROFIOUT%/"
     for p in ${DIRS}; do 
     	p=${p%/}
         TD=${p}/rofi/themes
