--- a/wrappers/python/Makefile.am
+++ b/wrappers/python/Makefile.am
@@ -25,7 +25,7 @@ fix-out-of-source: $(FIXSOURCES)
 
 if WITH_PYTHON
 
-PYEXT_ENV = CC=$(CC) CXX=$(CXX)
+PYEXT_ENV = CC=$(CXX) CXX=$(CXX)
 
 ## Always force setup.py, it's not good at guessing what needs to rebuild
 all-local: fix-out-of-source
diff --git a/wrappers/python/Makefile.in b/wrappers/python/Makefile.in
index 925460c..fdc8888 100644
--- a/wrappers/python/Makefile.in
+++ b/wrappers/python/Makefile.in
@@ -266,7 +266,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 EXTRA_DIST = lhapdf.cpp lhapdf.pyx clhapdf.pxd
 FIXSOURCES = $(EXTRA_DIST) 
-@WITH_PYTHON_TRUE@PYEXT_ENV = CC=$(CC) CXX=$(CXX)
+@WITH_PYTHON_TRUE@PYEXT_ENV = CC=$(CXX) CXX=$(CXX)
 all: all-am
 
 .SUFFIXES:
