diff -upr a/third_party/angle/BUILD.gn b/third_party/angle/BUILD.gn
--- a/third_party/angle/BUILD.gn	2022-08-17 19:38:11.000000000 +0000
+++ b/third_party/angle/BUILD.gn	2022-08-18 11:04:09.061751111 +0000
@@ -489,6 +489,12 @@ config("angle_vulkan_wayland_config") {
   if (angle_enable_vulkan && angle_use_wayland &&
       defined(vulkan_wayland_include_dirs)) {
     include_dirs = vulkan_wayland_include_dirs
+  } else if (angle_enable_vulkan && angle_use_wayland) {
+    include_dirs = [
+      "$wayland_gn_dir/src/src",
+      "$wayland_gn_dir/include/src",
+      "$wayland_gn_dir/include/protocol",
+    ]
   }
 }
 
@@ -1073,6 +1079,7 @@ if (angle_use_wayland) {
     include_dirs = [
       "$wayland_dir/egl",
       "$wayland_dir/src",
+      "$wayland_gn_dir/include/protocol",
     ]
   }
 
diff -upr a/third_party/angle/src/third_party/volk/BUILD.gn b/third_party/angle/src/third_party/volk/BUILD.gn
--- a/third_party/angle/src/third_party/volk/BUILD.gn	2022-08-17 19:38:12.000000000 +0000
+++ b/third_party/angle/src/third_party/volk/BUILD.gn	2022-08-18 11:04:36.499828006 +0000
@@ -21,6 +21,9 @@ source_set("volk") {
   configs += [ "$angle_root:angle_no_cfi_icall" ]
   public_deps = [ "$angle_vulkan_headers_dir:vulkan_headers" ]
   if (angle_use_wayland) {
-    include_dirs = [ "$wayland_dir/src" ]
+    include_dirs = [
+      "$wayland_dir/src",
+      "$wayland_gn_dir/include/protocol",
+    ]
   }
 }
