diff --git a/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc b/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
index a8c86e2..9e81df4 100644
--- a/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
+++ b/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
@@ -650,7 +650,7 @@
     const struct spa_meta_cursor* cursor =
         static_cast<struct spa_meta_cursor*>(spa_buffer_find_meta_data(
             spa_buffer, SPA_META_Cursor, sizeof(*cursor)));
-    if (spa_meta_cursor_is_valid(cursor)) {
+    if (cursor && spa_meta_cursor_is_valid(cursor)) {
       struct spa_meta_bitmap* bitmap = nullptr;
 
       if (cursor->bitmap_offset)
