--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -576,7 +576,6 @@ def write_gn_ninja(path, root_gen_dir, options):
       'base/trace_event/trace_log.cc',
       'base/trace_event/trace_log_constants.cc',
       'base/trace_event/tracing_agent.cc',
-      'base/tracked_objects.cc',
       'base/unguessable_token.cc',
       'base/value_iterators.cc',
       'base/values.cc',
