--- a/tools/gn/bootstrap/bootstrap.py   2017-11-07 23:06:09.000000000 +0000
+++ b/tools/gn/bootstrap/bootstrap.py   2017-11-08 12:17:16.569216182 +0000
@@ -481,6 +481,7 @@
       'base/metrics/sample_vector.cc',
       'base/metrics/sparse_histogram.cc',
       'base/metrics/statistics_recorder.cc',
+      'base/observer_list_threadsafe.cc',
       'base/path_service.cc',
       'base/pending_task.cc',
       'base/pickle.cc',

