--- chromium-70.0.3538.67/build/config/merge_for_jumbo.py.old	2018-10-25 19:15:54.578222230 +0300
+++ chromium-70.0.3538.67/build/config/merge_for_jumbo.py	2018-10-25 19:20:44.397613032 +0300
@@ -132,6 +132,8 @@
       assert not inputs
       continue
 
+    inputs.sort()
+    outputs.sort()
     write_jumbo_files(inputs, outputs, written_input_set, written_output_set)
 
   assert set(args.outputs) == written_output_set, "Did not fill all outputs"
