diff --git a/bin/dde-system-daemon/wallpaper.go b/bin/dde-system-daemon/wallpaper.go
index 6ee26e27..67dc77dc 100644
--- a/bin/dde-system-daemon/wallpaper.go
+++ b/bin/dde-system-daemon/wallpaper.go
@@ -24,7 +24,7 @@ import (
 
 const maxCount = 5
 const maxSize = 32 * 1024 * 1024
-const wallPaperDir = "/usr/share/wallpapers/custom-wallpapers/"
+const wallPaperDir = "/var/lib/dde-daemon/wallpapers/custom-wallpapers/"
 
 func GetUserDir(username string) (string, error) {
 	dir := filepath.Join(wallPaperDir, username)
@@ -136,7 +136,7 @@ func (d *Daemon) SaveCustomWallPaper(sender dbus.Sender, username string, file s
 		"-u",
 		username,
 		"--",
-		"head",
+		"@coreutils@/bin/head",
 		"-c",
 		"0",
 		file,
-- 
2.40.1

