diff --git a/libcomps/src/python/src/CMakeLists.txt b/libcomps/src/python/src/CMakeLists.txt
index d22b84e..57bd1c2 100644
--- a/libcomps/src/python/src/CMakeLists.txt
+++ b/libcomps/src/python/src/CMakeLists.txt
@@ -85,7 +85,7 @@ IF (SKBUILD)
     INSTALL(FILES libcomps/__init__.py DESTINATION libcomps/src/python/src/libcomps)
     INSTALL(TARGETS pycomps LIBRARY DESTINATION libcomps/src/python/src/libcomps)
 ELSE ()
-    EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c "from sys import stdout; from sysconfig import get_path; stdout.write(get_path('platlib'))" OUTPUT_VARIABLE PYTHON_INSTALL_DIR)
+    SET(PYTHON_INSTALL_DIR "@PYTHON_INSTALL_DIR@")

     INSTALL(FILES ${pycomps_SRCDIR}/libcomps/__init__.py DESTINATION ${PYTHON_INSTALL_DIR}/libcomps)
     #INSTALL(FILES ${pycomps_SRCDIR}/tests/__test.py DESTINATION
