diff --git a/opuslib/api/__init__.py b/opuslib/api/__init__.py
index 323a2a4..4c8a8fe 100644
--- a/opuslib/api/__init__.py
+++ b/opuslib/api/__init__.py
@@ -7,20 +7,12 @@

 import ctypes  # type: ignore

-from ctypes.util import find_library  # type: ignore
-
 __author__ = 'Никита Кузнецов <self@svartalf.info>'
 __copyright__ = 'Copyright (c) 2012, SvartalF'
 __license__ = 'BSD 3-Clause License'


-lib_location = find_library('opus')
-
-if lib_location is None:
-    raise Exception(
-        'Could not find Opus library. Make sure it is installed.')
-
-libopus = ctypes.CDLL(lib_location)
+libopus = ctypes.CDLL('@opusLibPath@')

 c_int_pointer = ctypes.POINTER(ctypes.c_int)
 c_int16_pointer = ctypes.POINTER(ctypes.c_int16)
