diff --git a/font-patcher b/font-patcher
index 99ae4f1..42e3d9a 100755
--- a/font-patcher
+++ b/font-patcher
@@ -39,7 +39,7 @@ except ImportError:
         )
     )
 
-sys.path.insert(0, os.path.join(os.path.abspath(os.path.dirname(sys.argv[0])), 'bin', 'scripts', 'name_parser'))
+sys.path.insert(0, os.path.join(os.path.abspath(os.path.dirname(sys.argv[0])), '..', 'lib', 'name_parser'))
 try:
     from FontnameParser import FontnameParser
     from FontnameTools import FontnameTools
@@ -47,7 +47,7 @@ try:
 except ImportError:
     FontnameParserOK = False
 
-sys.path.insert(0, os.path.join(os.path.abspath(os.path.dirname(sys.argv[0])), 'bin', 'scripts', 'braille'))
+sys.path.insert(0, os.path.join(os.path.abspath(os.path.dirname(sys.argv[0])), '..', 'lib', 'braille'))
 try:
     from Braille import get_braille_font
     BrailleGenerator = True
@@ -2241,7 +2241,7 @@ def setup_arguments():
     expert_group.add_argument('--custom',                                  dest='custom',           default=False, type=str,            help='Specify a custom symbol font, all glyphs will be copied; absolute path suggested')
     expert_group.add_argument('--dry',                                     dest='dry_run',          default=False, action='store_true', help='Do neither patch nor store the font, to check naming')
     expert_group.add_argument('--experimental',                            dest='experimental',     default=[], action='extend', nargs=1, help='Enable experimental/unstable feature, get hints with "?" or "help"')
-    expert_group.add_argument('--glyphdir',                                dest='glyphdir',         default=__dir__ + "/src/glyphs/", type=str, help='Path to glyphs to be used for patching')
+    expert_group.add_argument('--glyphdir',                                dest='glyphdir',         default=__dir__ + "/../share/glyphs/", type=str, help='Path to glyphs to be used for patching')
     expert_group.add_argument('--has-no-italic',                           dest='noitalic',         default=False, action='store_true', help='Font family does not have Italic (but Oblique/Slanted), to help create correct RIBBI set')
     expert_group.add_argument('--metrics',                                 dest='metrics',          default=None, choices=get_metrics_names(), help='Select vertical metrics source (for problematic cases)')
     expert_group.add_argument('--name',                                    dest='force_name',       default=None, type=str,             help='Specify naming source (\'full\', \'postscript\', \'filename\', or concrete free name-string)')
