diff --git a/SConstruct b/SConstruct
index bebc82e..3e0127b 100644
--- a/SConstruct
+++ b/SConstruct
@@ -59,6 +59,8 @@ srcs = Split('''\
 ''')
 libxsettingsd = env.Library('xsettingsd', srcs)
 env['LIBS'] = libxsettingsd
+env['ENV']['PATH'] = os.environ.get('PATH', '')
+env['ENV']['PKG_CONFIG_PATH'] = os.environ.get('PKG_CONFIG_PATH', '')
 env.ParseConfig('pkg-config --cflags --libs x11')
 
 xsettingsd     = env.Program('xsettingsd', 'xsettingsd.cc')
