diff --git a/CMakeLists.txt b/CMakeLists.txt
index e5892d4..604534e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,31 +4,28 @@
 cmake_minimum_required(VERSION 2.8.11)
 project(efa_nv_peermem C)
 
-set(KERNEL_VER "" CACHE STRING "Kernel version to build for")
 if(NOT KERNEL_VER)
-  execute_process(COMMAND uname -r OUTPUT_VARIABLE uname_r
-                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-  set(KERNEL_DIR "/lib/modules/${uname_r}/build")
-else()
-  set(KERNEL_DIR "/lib/modules/${KERNEL_VER}/build")
+  message(FATAL_ERROR "No kernel version argument")
+endif()
+if(NOT KERNEL_DIR)
+  message(FATAL_ERROR "No kernel dir argument")
+endif()
+if(NOT NVIDIA_DIR)
+  message(FATAL_ERROR "No nvidia kernel dir argument")
 endif()
 
-unset(KERNEL_MAKEFILE CACHE)
 find_file(KERNEL_MAKEFILE Makefile PATHS ${KERNEL_DIR} NO_DEFAULT_PATH)
 if(NOT KERNEL_MAKEFILE)
   message(FATAL_ERROR "No kernel Makefile")
 endif()
 message("-- Kernel directory - ${KERNEL_DIR}")
 
-set(NVIDIA_DIR "" CACHE PATH "Path to NVIDIA directory")
-if(NVIDIA_DIR)
-  unset(NVIDIA_MODULE_SYMVERS CACHE)
-  find_file(NVIDIA_MODULE_SYMVERS Module.symvers PATHS ${NVIDIA_DIR} REQUIRED NO_DEFAULT_PATH)
-  if(NOT NVIDIA_MODULE_SYMVERS)
-    message(FATAL_ERROR "No NVIDIA Module.symvers")
-  else()
-    message("-- NVIDIA directory - ${NVIDIA_DIR}")
-  endif()
+unset(NVIDIA_MODULE_SYMVERS CACHE)
+find_file(NVIDIA_MODULE_SYMVERS Module.symvers PATHS ${NVIDIA_DIR} REQUIRED NO_DEFAULT_PATH)
+if(NOT NVIDIA_MODULE_SYMVERS)
+  message(FATAL_ERROR "No NVIDIA Module.symvers")
+else()
+  message("-- NVIDIA directory - ${NVIDIA_DIR}")
 endif()
 
 set(GCOV_PROFILE OFF CACHE BOOL "Enable GCOV profiling")
