diff --git a/src/platform/nm-platform.c b/src/platform/nm-platform.c
index 8803377..14e5726 100644
--- a/src/platform/nm-platform.c
+++ b/src/platform/nm-platform.c
@@ -39,6 +39,12 @@
 #include "nm-enum-types.h"
 #include "nm-core-internal.h"

+#if HAVE_LIBNL_INET6_ADDR_GEN_MODE && HAVE_KERNEL_INET6_ADDR_GEN_MODE
+#include <linux/if_link.h>
+#else
+#define IN6_ADDR_GEN_MODE_NONE  1
+#endif
+
 #define ADDRESS_LIFETIME_PADDING 5

 G_STATIC_ASSERT (sizeof ( ((NMPlatformLink *) NULL)->addr.data ) == NM_UTILS_HWADDR_LEN_MAX);
