--- NetworkManager-1.6.2.org/src/ppp/nm-ppp-manager.c	2017-02-15 13:10:27.000000000 +0100
+++ NetworkManager-1.6.2/./src/ppp/nm-ppp-manager.c	2017-04-03 11:45:39.891653216 +0200
@@ -724,7 +724,7 @@
 
 	g_return_val_if_fail (setting != NULL, NULL);
 
-	pppd_binary = nm_utils_find_helper ("pppd", NULL, err);
+	pppd_binary = nm_utils_find_helper ("pppd", PPPD_PATH, err);
 	if (!pppd_binary)
 		return NULL;
 
