diff -Naur gpgme.old/configure.ac gpgme.new/configure.ac
--- gpgme.old/configure.ac	2020-11-12 04:19:50.000000000 -0500
+++ gpgme.new/configure.ac	2021-01-08 03:04:38.000000000 -0500
@@ -166,6 +166,16 @@
                           mym4_minor mym4_micro)
 AC_SUBST(VERSION_NUMBER)
 
+# Try to find a thread-safe version of ttyname().
+gnupg_REPLACE_TTYNAME_R
+if test "$ac_cv_func_ttyname_r" != yes; then
+  AC_MSG_WARN([
+***
+*** ttyname() is not thread-safe and ttyname_r() does not exist
+***])
+fi
+
+
 # We need to compile and run a program on the build machine.  A
 # comment in libgpg-error says that the AC_PROG_CC_FOR_BUILD macro in
 # the AC archive is broken for autoconf 2.57.  Given that there is no
@@ -658,15 +668,6 @@
 
 AC_FUNC_FSEEKO
 
-# Try to find a thread-safe version of ttyname().
-gnupg_REPLACE_TTYNAME_R
-if test "$ac_cv_func_ttyname_r" != yes; then
-  AC_MSG_WARN([
-***
-*** ttyname() is not thread-safe and ttyname_r() does not exist
-***])
-fi
-
 # Try to find a thread-safe version of getenv().
 have_thread_safe_getenv=no
 jm_GLIBC21
