From abdfbb94df98fe88be4dd92ca587500126558411 Mon Sep 17 00:00:00 2001
From: Victor Gaydov <victor@enise.org>
Date: Sun, 26 Jul 2020 11:54:52 +0300
Subject: [PATCH] Remove deprecated scons call

---
 SConstruct | 1 -
 1 file changed, 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index 407025d8..04afa91f 100644
--- a/SConstruct
+++ b/SConstruct
@@ -49,7 +49,6 @@ env = Environment(ENV=os.environ, tools=[
 # performance tuning
 env.Decider('MD5-timestamp')
 env.SetOption('implicit_cache', 1)
-env.SourceCode('.', None)
 
 # provide absolute path to force single sconsign file
 # per-directory sconsign files seems to be buggy with generated sources
-- 
2.34.1

