diff --git a/configure.ac b/configure.ac
index c214ccd..ab0cdd7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,5 +1,5 @@
 AC_PREREQ([2.69])
-AC_INIT([libgeneral], m4_esyscmd([git rev-list --count HEAD | tr -d '\n']), [tihmstar@gmail.com])
+AC_INIT([libgeneral], [tihmstar@gmail.com])
 
 AC_CANONICAL_SYSTEM
 AC_CANONICAL_HOST
@@ -9,10 +9,10 @@ AM_INIT_AUTOMAKE([subdir-objects])
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIRS([m4])
 
-AC_DEFINE([VERSION_COMMIT_COUNT], "m4_esyscmd([git rev-list --count HEAD | tr -d '\n'])", [Git commit count])
-AC_DEFINE([VERSION_COMMIT_SHA], "m4_esyscmd([git rev-parse HEAD | tr -d '\n'])", [Git commit sha])
-AC_SUBST([VERSION_COMMIT_COUNT], ["m4_esyscmd([git rev-list --count HEAD | tr -d '\n'])"])
-AC_SUBST([VERSION_COMMIT_SHA], ["m4_esyscmd([git rev-parse HEAD | tr -d '\n'])"])
+AC_ARG_WITH([version-commit-count], [],
+  [VERSION_COMMIT_COUNT="$withval"])
+AC_DEFINE([VERSION_COMMIT_COUNT], ["$VERSION_COMMIT_COUNT"], [Git commit count])
+AC_SUBST([VERSION_COMMIT_COUNT], ["$VERSION_COMMIT_COUNT"])
 
 # Checks for programs.
 AC_PROG_CXX([clang++])
