diff --git a/support/shobj-conf b/support/shobj-conf
index 5a63e80..4b2a741 100644
--- support/shobj-conf
+++ support/shobj-conf
@@ -189,7 +189,7 @@ darwin*|macosx*)
 	darwin[789]*|darwin10*)	SHOBJ_LDFLAGS=''
 			SHLIB_XLDFLAGS='-dynamiclib -arch_only `/usr/bin/arch` -install_name $(libdir)/$@ -current_version $(SHLIB_MAJOR)$(SHLIB_MINOR) -compatibility_version $(SHLIB_MAJOR) -v'
 			;;
-	*)		SHOBJ_LDFLAGS='-dynamic'
+	*)		SHOBJ_LDFLAGS='-dynamiclib'
 			SHLIB_XLDFLAGS='-arch_only `/usr/bin/arch` -install_name $(libdir)/$@ -current_version $(SHLIB_MAJOR)$(SHLIB_MINOR) -compatibility_version $(SHLIB_MAJOR) -v'
 			;;
 	esac
