diff --git a/src/meson.build b/src/meson.build
index c9a44e418..cd9e466d6 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -35,7 +35,7 @@ foreach test_typelib_dep: test_typelib_deps
   test_gi_typelib_path += [join_paths(test_typelib_dep.get_variable('libdir'), 'girepository-1.0')]
 endforeach
 test_env = [
-  'GI_TYPELIB_PATH=@0@'.format(':'.join(test_gi_typelib_path)),
+  'GI_TYPELIB_PATH=@0@:@1@'.format(':'.join(test_gi_typelib_path), run_command('sh', ['-c', 'echo "$GI_TYPELIB_PATH"']).stdout().strip()),
   'G_TEST_SRCDIR=@0@/tests'.format(meson.current_source_dir()),
   'G_TEST_BUILDDIR=@0@/tests'.format(meson.current_build_dir()),
   'G_DEBUG=gc-friendly',
