diff --git a/src/kpluginselector.cpp b/src/kpluginselector.cpp
index 137c865..097ab75 100644
--- a/src/kpluginselector.cpp
+++ b/src/kpluginselector.cpp
@@ -303,7 +303,7 @@ void KPluginSelector::addPlugins(const QString &componentName,
     QStringList desktopFileNames;
     const QStringList dirs = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, componentName + QStringLiteral("/kpartplugins"), QStandardPaths::LocateDirectory);
     for (const QString &dir : dirs) {
-        QDirIterator it(dir, QStringList() << QStringLiteral("*.desktop"), QDir::NoFilter, QDirIterator::Subdirectories);
+        QDirIterator it(dir, QStringList() << QStringLiteral("*.desktop"), QDir::NoFilter, QDirIterator::Subdirectories | QDirIterator::FollowSymlinks);
         while (it.hasNext()) {
             desktopFileNames.append(it.next());
         }
