commit 48b92b74bc6fd270c33a726257e2879203cf5064
Author: Yury G. Kudryashov [diff odt] <urkud.urkud@gmail.com>
Date:   Wed Mar 21 00:47:43 2012 +0400

    Include SMOKE_QTCORE_INCLUDE_DIR

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 48020a1..2263a73 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,7 +14,7 @@ find_package(Smoke COMPONENTS QtCore QtGui QtNetwork Qt3Support QtDeclarative Qt
                               QtOpenGl QtScript QtSql QtSvg QtTest QtUiTools QtWebKit QtXml QtXmlPatterns
                               Phonon Qwt QSci QImageBlitz)
                               
-include_directories(${SMOKE_INCLUDE_DIR} ${QT_INCLUDES} ${CMAKE_CURRENT_SOURCE_DIR}/src)
+include_directories(${SMOKE_INCLUDE_DIR} ${SMOKE_QTCORE_INCLUDE_DIR} ${QT_INCLUDES} ${CMAKE_CURRENT_SOURCE_DIR}/src)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake ${SMOKE_CMAKE_MODULE_DIR})
 include(MacroOptionalFindPackage)
 include(MacroOptionalAddBindings)
