commit c78779fcaff587818ee37bec3ded5e0617625b95
Author: Yury G. Kudryashov [diff odt] <urkud.urkud@gmail.com>
Date:   Wed Mar 21 01:01:27 2012 +0400

    Install to PERL_SITE_ARCH_INSTALL_DIR

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 48020a1..16188df 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,7 +36,6 @@ macro_log_feature(Qwt5_Qt4_FOUND "Qwt5 for Qt4" "Qwt5 libraries for Qt4" "http:/
 add_definitions(-DDEBUG)
 
 include (FindPerlMore)
-set(CUSTOM_PERL_SITE_ARCH_DIR ${PERL_SITE_ARCH_DIR} CACHE DIR "Custom installation directory for perl binary extension")
 
 # the RPATH to be used when installing, but only if it's not a system directory
 GET_FILENAME_COMPONENT(SMOKE_LIB_DIR ${SMOKE_BASE_LIBRARY} PATH)
diff --git a/INSTALL b/INSTALL
index d19f97e..97cc9f1 100644
--- a/INSTALL
+++ b/INSTALL
@@ -36,7 +36,6 @@ ccmake step.
 The standard options are:
     CMAKE_BUILD_TYPE = The type of build ('Debug', 'Release', etc)
     CMAKE_INSTALL_PREFIX = The location for any executables ( e.g. puic4 )
-    CUSTOM_PERL_SITE_ARCH_DIR = The location for the perl modules themselves.
     QT_QMAKE_EXECUTABLE = The path to your system's qmake.
 
 cmake looks in your path for a qmake executable.  If it can't find it, it will
diff --git a/Makefile.PL b/Makefile.PL
index df9a13c..31dd912 100755
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -3,7 +3,7 @@
 use strict;
 use Config;
 
-my ($prefix, $sitearch, $qmake) = ($Config{prefix}, $Config{sitearch});
+my ($prefix, $qmake) = ($Config{prefix});
 my @cmakeArgs;
 foreach my $arg (@ARGV) {
     my $key = $arg;
@@ -12,7 +12,6 @@ foreach my $arg (@ARGV) {
     $value =~ s/^[^=]*=//g;
     if ($key eq 'PREFIX' or $key eq 'INSTALL_BASE') {
         $prefix = $value;
-        $sitearch = "$prefix";
     }
     elsif ($key eq 'QMAKE') {
         $qmake = $value;
@@ -34,7 +33,6 @@ if($^O =~ /win/i){
 }
 push @args, "-DCMAKE_INSTALL_PREFIX=$prefix" if $prefix;
 push @args, "-DQT_QMAKE_EXECUTABLE=$qmake" if $qmake;
-push @args, "-DCUSTOM_PERL_SITE_ARCH_DIR=$sitearch" if $sitearch;
 push @args, @cmakeArgs;
 
 if ( eval "require Alien::SmokeQt" ) {
diff --git a/phonon/lib/CMakeLists.txt b/phonon/lib/CMakeLists.txt
index f2857c3..78674ea 100644
--- a/phonon/lib/CMakeLists.txt
+++ b/phonon/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(phononpm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/Phonon.pm ${CMAKE_BINARY_DIR}/blib/lib/Phonon.pm)
-install(FILES Phonon.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES Phonon.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/phonon/src/CMakeLists.txt b/phonon/src/CMakeLists.txt
index a04db11..9933dfd 100644
--- a/phonon/src/CMakeLists.txt
+++ b/phonon/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_phonon
 set_target_properties(perl_phonon PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_phonon PROPERTIES PREFIX "")
 
-install(TARGETS perl_phonon DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_phonon DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qimageblitz/lib/CMakeLists.txt b/qimageblitz/lib/CMakeLists.txt
index 0809ba8..22fc1c0 100644
--- a/qimageblitz/lib/CMakeLists.txt
+++ b/qimageblitz/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qimageblitzpm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QImageBlitz.pm ${CMAKE_BINARY_DIR}/blib/lib/QImageBlitz.pm)
-install(FILES QImageBlitz.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QImageBlitz.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qimageblitz/src/CMakeLists.txt b/qimageblitz/src/CMakeLists.txt
index 02ef494..4f02c1b 100644
--- a/qimageblitz/src/CMakeLists.txt
+++ b/qimageblitz/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qimageblitz
 set_target_properties(perl_qimageblitz PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qimageblitz PROPERTIES PREFIX "")
 
-install(TARGETS perl_qimageblitz DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qimageblitz DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qsci/lib/CMakeLists.txt b/qsci/lib/CMakeLists.txt
index 63b451f..d22869f 100644
--- a/qsci/lib/CMakeLists.txt
+++ b/qsci/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qscipm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/Qsci.pm ${CMAKE_BINARY_DIR}/blib/lib/Qsci.pm)
-install(FILES Qsci.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES Qsci.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qsci/src/CMakeLists.txt b/qsci/src/CMakeLists.txt
index 8c1659b..bdcb5a8 100644
--- a/qsci/src/CMakeLists.txt
+++ b/qsci/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qsci
 set_target_properties(perl_qsci PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qsci PROPERTIES PREFIX "")
 
-install(TARGETS perl_qsci DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qsci DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qt3support/lib/CMakeLists.txt b/qt3support/lib/CMakeLists.txt
index 2f04cfa..dda5afa 100644
--- a/qt3support/lib/CMakeLists.txt
+++ b/qt3support/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qt3support4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/Qt3Support4.pm ${CMAKE_BINARY_DIR}/blib/lib/Qt3Support4.pm)
-install(FILES Qt3Support4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES Qt3Support4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qt3support/src/CMakeLists.txt b/qt3support/src/CMakeLists.txt
index b24532e..bed99aa 100644
--- a/qt3support/src/CMakeLists.txt
+++ b/qt3support/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qt3support4
 set_target_properties(perl_qt3support4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qt3support4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qt3support4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qt3support4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qtcore/lib/CMakeLists.txt b/qtcore/lib/CMakeLists.txt
index ef07d10..1d78196 100644
--- a/qtcore/lib/CMakeLists.txt
+++ b/qtcore/lib/CMakeLists.txt
@@ -1,3 +1,3 @@
 add_subdirectory( QtCore4 )
 add_custom_target(perlqtcore4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtCore4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtCore4.pm)
-install( FILES QtCore4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR} )
+install( FILES QtCore4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR} )
diff --git a/qtcore/lib/QtCore4/CMakeLists.txt b/qtcore/lib/QtCore4/CMakeLists.txt
index 44c7893..06aef20 100644
--- a/qtcore/lib/QtCore4/CMakeLists.txt
+++ b/qtcore/lib/QtCore4/CMakeLists.txt
@@ -1,3 +1,3 @@
-install( FILES signals.pm slots.pm isa.pm debug.pm classinfo.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/QtCore4 )
+install( FILES signals.pm slots.pm isa.pm debug.pm classinfo.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/QtCore4 )
 add_custom_target(perlqt4pmlibmkdir ALL ${CMAKE_COMMAND} -E make_directory ${CMAKE_BINARY_DIR}/blib/lib/QtCore4)
 add_custom_target(perlqt4pmlibsubdir ALL ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_BINARY_DIR}/blib/lib/QtCore4)
diff --git a/qtcore/src/CMakeLists.txt b/qtcore/src/CMakeLists.txt
index 3910636..b5e645c 100644
--- a/qtcore/src/CMakeLists.txt
+++ b/qtcore/src/CMakeLists.txt
@@ -53,10 +53,10 @@ target_link_libraries(perlqtcore4
 set_target_properties(perlqtcore4 PROPERTIES
     OUTPUT_NAME ${libraryName}
     PREFIX ""
-    INSTALL_NAME_DIR ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/
+    INSTALL_NAME_DIR ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/
 )
 
-install(TARGETS perlqtcore4 EXPORT PerlQtExport DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perlqtcore4 EXPORT PerlQtExport DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
 install(FILES binding.h handlers.h listclass_macros.h marshall_basetypes.h marshall_complex.h
               marshall.h marshall_macros.h marshall_primitives.h marshall_types.h perlqt.h ppport.h
               QtCore4.h smokehelp.h smokeperl.h util.h
diff --git a/qtdbus/lib/CMakeLists.txt b/qtdbus/lib/CMakeLists.txt
index d03a672..a60d603 100644
--- a/qtdbus/lib/CMakeLists.txt
+++ b/qtdbus/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qtdbus4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtDBus4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtDBus4.pm)
-install(FILES QtDBus4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtDBus4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qtdbus/src/CMakeLists.txt b/qtdbus/src/CMakeLists.txt
index 9aa05a0..5786ea2 100644
--- a/qtdbus/src/CMakeLists.txt
+++ b/qtdbus/src/CMakeLists.txt
@@ -31,4 +31,4 @@ target_link_libraries(perl_qtdbus4
 set_target_properties(perl_qtdbus4 PROPERTIES OUTPUT_NAME "QtDBus4")
 set_target_properties(perl_qtdbus4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qtdbus4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/QtDBus4/)
+install(TARGETS perl_qtdbus4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/QtDBus4/)
diff --git a/qtdeclarative/lib/CMakeLists.txt b/qtdeclarative/lib/CMakeLists.txt
index e8d2847..b458858 100644
--- a/qtdeclarative/lib/CMakeLists.txt
+++ b/qtdeclarative/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qtdeclarative4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtDeclarative4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtDeclarative4.pm)
-install(FILES QtDeclarative4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtDeclarative4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qtdeclarative/src/CMakeLists.txt b/qtdeclarative/src/CMakeLists.txt
index 1662167..ec2a8fc 100644
--- a/qtdeclarative/src/CMakeLists.txt
+++ b/qtdeclarative/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qtdeclarative4
 set_target_properties(perl_qtdeclarative4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qtdeclarative4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qtdeclarative4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qtdeclarative4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qtgui/lib/CMakeLists.txt b/qtgui/lib/CMakeLists.txt
index 82626c8..e62a4ac 100644
--- a/qtgui/lib/CMakeLists.txt
+++ b/qtgui/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qtgui4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtGui4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtGui4.pm)
-install(FILES QtGui4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtGui4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qtgui/src/CMakeLists.txt b/qtgui/src/CMakeLists.txt
index 4cc8b36..44041b6 100644
--- a/qtgui/src/CMakeLists.txt
+++ b/qtgui/src/CMakeLists.txt
@@ -38,4 +38,4 @@ target_link_libraries(perl_qtgui4
 set_target_properties(perl_qtgui4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qtgui4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qtgui4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qtgui4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qthelp/lib/CMakeLists.txt b/qthelp/lib/CMakeLists.txt
index dcf5ebe..da817d4 100644
--- a/qthelp/lib/CMakeLists.txt
+++ b/qthelp/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qthelp4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtHelp4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtHelp4.pm)
-install(FILES QtHelp4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtHelp4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qthelp/src/CMakeLists.txt b/qthelp/src/CMakeLists.txt
index c00359d..689cb29 100644
--- a/qthelp/src/CMakeLists.txt
+++ b/qthelp/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qthelp4
 set_target_properties(perl_qthelp4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qthelp4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qthelp4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qthelp4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qtmultimedia/lib/CMakeLists.txt b/qtmultimedia/lib/CMakeLists.txt
index e55f697..5384539 100644
--- a/qtmultimedia/lib/CMakeLists.txt
+++ b/qtmultimedia/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qtmultimedia4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtMultimedia4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtMultimedia4.pm)
-install(FILES QtMultimedia4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtMultimedia4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qtmultimedia/src/CMakeLists.txt b/qtmultimedia/src/CMakeLists.txt
index 0728aba..df8e552 100644
--- a/qtmultimedia/src/CMakeLists.txt
+++ b/qtmultimedia/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qtmultimedia4
 set_target_properties(perl_qtmultimedia4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qtmultimedia4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qtmultimedia4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qtmultimedia4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qtnetwork/lib/CMakeLists.txt b/qtnetwork/lib/CMakeLists.txt
index 12cd5dd..d1fb0e6 100644
--- a/qtnetwork/lib/CMakeLists.txt
+++ b/qtnetwork/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qtnetwork4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtNetwork4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtNetwork4.pm)
-install(FILES QtNetwork4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtNetwork4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qtnetwork/src/CMakeLists.txt b/qtnetwork/src/CMakeLists.txt
index caf8327..0994d57 100644
--- a/qtnetwork/src/CMakeLists.txt
+++ b/qtnetwork/src/CMakeLists.txt
@@ -39,4 +39,4 @@ target_link_libraries(perl_qtnetwork4
 set_target_properties(perl_qtnetwork4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qtnetwork4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qtnetwork4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qtnetwork4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qtopengl/lib/CMakeLists.txt b/qtopengl/lib/CMakeLists.txt
index f3b5640..d8dc9e4 100644
--- a/qtopengl/lib/CMakeLists.txt
+++ b/qtopengl/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qtopengl4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtOpenGL4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtOpenGL4.pm)
-install(FILES QtOpenGL4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtOpenGL4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qtopengl/src/CMakeLists.txt b/qtopengl/src/CMakeLists.txt
index 20493a9..777af89 100644
--- a/qtopengl/src/CMakeLists.txt
+++ b/qtopengl/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qtopengl4
 set_target_properties(perl_qtopengl4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qtopengl4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qtopengl4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qtopengl4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qtscript/lib/CMakeLists.txt b/qtscript/lib/CMakeLists.txt
index d21dbe5..320ff93 100644
--- a/qtscript/lib/CMakeLists.txt
+++ b/qtscript/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qtscript4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtScript4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtScript4.pm)
-install(FILES QtScript4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtScript4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qtscript/src/CMakeLists.txt b/qtscript/src/CMakeLists.txt
index dd395be..6ff47d8 100644
--- a/qtscript/src/CMakeLists.txt
+++ b/qtscript/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qtscript4
 set_target_properties(perl_qtscript4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qtscript4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qtscript4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qtscript4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qtsql/lib/CMakeLists.txt b/qtsql/lib/CMakeLists.txt
index 59336d3..d9dd4d5 100644
--- a/qtsql/lib/CMakeLists.txt
+++ b/qtsql/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qtsql4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtSql4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtSql4.pm)
-install(FILES QtSql4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtSql4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qtsql/src/CMakeLists.txt b/qtsql/src/CMakeLists.txt
index 3ec2028..59096ba 100644
--- a/qtsql/src/CMakeLists.txt
+++ b/qtsql/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qtsql4
 set_target_properties(perl_qtsql4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qtsql4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qtsql4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qtsql4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qtsvg/lib/CMakeLists.txt b/qtsvg/lib/CMakeLists.txt
index 33f6deb..bf1dc1f 100644
--- a/qtsvg/lib/CMakeLists.txt
+++ b/qtsvg/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qtsvg4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtSvg4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtSvg4.pm)
-install(FILES QtSvg4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtSvg4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qtsvg/src/CMakeLists.txt b/qtsvg/src/CMakeLists.txt
index 11eccd2..449bf0f 100644
--- a/qtsvg/src/CMakeLists.txt
+++ b/qtsvg/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qtsvg4
 set_target_properties(perl_qtsvg4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qtsvg4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qtsvg4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qtsvg4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qttest/lib/CMakeLists.txt b/qttest/lib/CMakeLists.txt
index 3bfa78c..5a8d8da 100644
--- a/qttest/lib/CMakeLists.txt
+++ b/qttest/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qttest4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtTest4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtTest4.pm)
-install(FILES QtTest4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtTest4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qttest/src/CMakeLists.txt b/qttest/src/CMakeLists.txt
index 5492e55..d4662b1 100644
--- a/qttest/src/CMakeLists.txt
+++ b/qttest/src/CMakeLists.txt
@@ -38,4 +38,4 @@ target_link_libraries(perl_qttest4
 set_target_properties(perl_qttest4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qttest4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qttest4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qttest4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qtuitools/lib/CMakeLists.txt b/qtuitools/lib/CMakeLists.txt
index 119e40e..3a5f472 100644
--- a/qtuitools/lib/CMakeLists.txt
+++ b/qtuitools/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qtuitools4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtUiTools4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtUiTools4.pm)
-install(FILES QtUiTools4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtUiTools4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qtuitools/src/CMakeLists.txt b/qtuitools/src/CMakeLists.txt
index a8ae4a2..ecc079f 100644
--- a/qtuitools/src/CMakeLists.txt
+++ b/qtuitools/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qtuitools4
 set_target_properties(perl_qtuitools4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qtuitools4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qtuitools4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qtuitools4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qtwebkit/lib/CMakeLists.txt b/qtwebkit/lib/CMakeLists.txt
index a02f7ee..3846227 100644
--- a/qtwebkit/lib/CMakeLists.txt
+++ b/qtwebkit/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qtwebkit4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtWebKit4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtWebKit4.pm)
-install(FILES QtWebKit4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtWebKit4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qtwebkit/src/CMakeLists.txt b/qtwebkit/src/CMakeLists.txt
index a6e00f8..dbebc44 100644
--- a/qtwebkit/src/CMakeLists.txt
+++ b/qtwebkit/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qtwebkit4
 set_target_properties(perl_qtwebkit4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qtwebkit4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qtwebkit4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qtwebkit4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qtxml/lib/CMakeLists.txt b/qtxml/lib/CMakeLists.txt
index 5505bc4..7db743c 100644
--- a/qtxml/lib/CMakeLists.txt
+++ b/qtxml/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qtxml4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtXml4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtXml4.pm)
-install(FILES QtXml4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtXml4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qtxml/src/CMakeLists.txt b/qtxml/src/CMakeLists.txt
index 018508c..a351609 100644
--- a/qtxml/src/CMakeLists.txt
+++ b/qtxml/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qtxml4
 set_target_properties(perl_qtxml4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qtxml4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qtxml4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qtxml4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qtxmlpatterns/lib/CMakeLists.txt b/qtxmlpatterns/lib/CMakeLists.txt
index a35f3df..3d86103 100644
--- a/qtxmlpatterns/lib/CMakeLists.txt
+++ b/qtxmlpatterns/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qtxmlpatterns4pm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/QtXmlPatterns4.pm ${CMAKE_BINARY_DIR}/blib/lib/QtXmlPatterns4.pm)
-install(FILES QtXmlPatterns4.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES QtXmlPatterns4.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qtxmlpatterns/src/CMakeLists.txt b/qtxmlpatterns/src/CMakeLists.txt
index 9970a98..563e922 100644
--- a/qtxmlpatterns/src/CMakeLists.txt
+++ b/qtxmlpatterns/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qtxmlpatterns4
 set_target_properties(perl_qtxmlpatterns4 PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qtxmlpatterns4 PROPERTIES PREFIX "")
 
-install(TARGETS perl_qtxmlpatterns4 DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qtxmlpatterns4 DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
diff --git a/qwt/lib/CMakeLists.txt b/qwt/lib/CMakeLists.txt
index 0013c4d..d67ffbf 100644
--- a/qwt/lib/CMakeLists.txt
+++ b/qwt/lib/CMakeLists.txt
@@ -1,2 +1,2 @@
 add_custom_target(qwtpm ALL ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/Qwt.pm ${CMAKE_BINARY_DIR}/blib/lib/Qwt.pm)
-install(FILES Qwt.pm DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/)
+install(FILES Qwt.pm DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/)
diff --git a/qwt/src/CMakeLists.txt b/qwt/src/CMakeLists.txt
index 869d818..b644e80 100644
--- a/qwt/src/CMakeLists.txt
+++ b/qwt/src/CMakeLists.txt
@@ -37,4 +37,4 @@ target_link_libraries(perl_qwt
 set_target_properties(perl_qwt PROPERTIES OUTPUT_NAME ${libraryName})
 set_target_properties(perl_qwt PROPERTIES PREFIX "")
 
-install(TARGETS perl_qwt DESTINATION ${CUSTOM_PERL_SITE_ARCH_DIR}/auto/${libraryName}/)
+install(TARGETS perl_qwt DESTINATION ${PERL_SITE_ARCH_INSTALL_DIR}/auto/${libraryName}/)
