diff --git a/CMakeLists.txt b/CMakeLists.txt
index f6d8fa3..5f0657d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,17 @@
+if (POLICY CMP0048)
+	cmake_policy(SET CMP0048 NEW)
+endif (POLICY CMP0048)
+
 project(move-transition VERSION 2.4.3)
 set(PROJECT_FULL_NAME "Move Transition")
 
+include(FindLibobs.cmake)
+find_package(LibObs REQUIRED)
+
+include_directories(
+	"${LIBOBS_INCLUDE_DIR}/../plugins/obs-transitions"
+	"${LIBOBS_INCLUDE_DIR}/../UI/obs-frontend-api")
+
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/version.h.in ${CMAKE_CURRENT_SOURCE_DIR}/version.h)

 set(move-transition_HEADERS
@@ -38,4 +49,10 @@ target_link_libraries(move-transition
 	libobs)
 
 set_target_properties(move-transition PROPERTIES FOLDER "plugins/exeldro")
-install_obs_plugin_with_data(move-transition data)
+set_target_properties(move-transition PROPERTIES PREFIX "")
+
+install(TARGETS move-transition
+	LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/obs-plugins)
+
+install(DIRECTORY data/locale/
+	DESTINATION "${CMAKE_INSTALL_PREFIX}/share/obs/obs-plugins/move-transition/locale")
