From 7a2bbb0d0098d7a33ad3794d0199879ad50e755a Mon Sep 17 00:00:00 2001
From: Austin Seipp <aseipp@pobox.com>
Date: Tue, 8 Apr 2014 06:21:41 -0500
Subject: [PATCH] Remove use of /dev/stderr

Signed-off-by: Austin Seipp <aseipp@pobox.com>
---
 POSIX/posix-l.sh | 2 --
 1 file changed, 2 deletions(-)

diff --git a/POSIX/posix-l.sh b/POSIX/posix-l.sh
index b3f0e31..71d6fe8 100755
--- a/POSIX/posix-l.sh
+++ b/POSIX/posix-l.sh
@@ -7,8 +7,6 @@ for LIB in rt xnet; do
 		fi
 		printf "%s" "-l${LIB}";
 		FIRST=NO;
-	else
-		echo "WARNING: POSIX violation: make's CC doesn't understand -l${LIB}" >/dev/stderr
 	fi
 	rm -f a.out
 done
-- 
1.8.3.2

