diff --git a/CMakeLists.txt b/CMakeLists.txt
index 08d30a4..7df8190 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -386,6 +386,12 @@ else()
 	set_property(TARGET FEBioStudio PROPERTY AUTOGEN_BUILD_DIR ${CMAKE_BINARY_DIR}/CMakeFiles/AutoGen/FEBioStudio_autogen)
 endif()
 
+include(GNUInstallDirs)
+install(TARGETS FEBioStudio
+	BUNDLE DESTINATION Applications
+	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+	)
+
 macro(addLib name)
 	add_library(${name} ${HDR_${name}} ${SRC_${name}})
 	set_property(TARGET ${name} PROPERTY AUTOGEN_BUILD_DIR ${CMAKE_BINARY_DIR}/CMakeFiles/AutoGen/${name}_autogen)
