--- src/contacts-view.vala.orig	2014-04-08 11:35:36.302252460 +0200
+++ src/contacts-view.vala	2014-04-08 11:37:37.045343221 +0200
@@ -265,7 +265,7 @@
     data.destroy ();
   }
 
-  public override void row_selected (ListBoxRow row) {
+  public override void row_selected (ListBoxRow? row) {
     var data = row as ContactDataRow;
     var contact = data != null ? data.contact : null;
     selection_changed (contact);
