diff -up boost_1_53_0/boost/python/to_python_value.hpp\~ boost_1_53_0/boost/python/to_python_value.hpp
--- boost_1_53_0/boost/python/to_python_value.hpp~	2007-12-16 11:12:07.000000000 +0100
+++ boost_1_53_0/boost/python/to_python_value.hpp	2013-07-23 16:19:02.518904596 +0200
@@ -147,8 +147,8 @@ namespace detail
   template <class T>
   inline PyObject* registry_to_python_value<T>::operator()(argument_type x) const
   {
-      typedef converter::registered<argument_type> r;
 # if BOOST_WORKAROUND(__GNUC__, < 3)
+      typedef converter::registered<argument_type> r;
       // suppresses an ICE, somehow
       (void)r::converters;
 # endif 

Diff finished.  Tue Jul 23 16:19:05 2013
