--- boost_1_57_0/boost/system/api_config.hpp	2014-08-03 15:44:11.000000000 -0500
+++ boost_1_57_0/boost/system/api_config.hpp	2015-05-04 17:51:31.189701800 -0500
@@ -33,7 +33,7 @@
 //    Standalone MinGW and all other known Windows compilers do predefine _WIN32
 //    Compilers that predefine _WIN32 or __MINGW32__ do so for Windows 64-bit builds too.
 
-# if defined(_WIN32) || defined(__CYGWIN__) // Windows default, including MinGW and Cygwin
+# if defined(_WIN32) // Windows default, including MinGW and Cygwin
 #   define BOOST_WINDOWS_API
 # else
 #   define BOOST_POSIX_API 
--- boost_1_57_0/boost/system/detail/error_code.ipp	2014-08-03 15:44:11.000000000 -0500
+++ boost_1_57_0/boost/system/detail/error_code.ipp	2015-05-04 17:51:02.925112700 -0500
@@ -97,7 +97,7 @@ namespace
       char buf[64];
       char * bp = buf;
       std::size_t sz = sizeof(buf);
-  #  if defined(__CYGWIN__) || defined(__USE_GNU)
+  #  if defined(__GNU_VISIBLE) || defined(__USE_GNU)
       // Oddball version of strerror_r
       const char * c_str = strerror_r( ev, bp, sz );
       return  c_str
