--- boost_1_40_0/boost/python/detail/config.hpp	2007-11-25 12:07:19.000000000 -0600
+++ boost_1_40_0/boost/python/detail/config.hpp	2009-08-27 23:41:34.092545400 -0500
@@ -83,7 +83,7 @@
 #     endif
 #     define BOOST_PYTHON_DECL_FORWARD
 #     define BOOST_PYTHON_DECL_EXCEPTION __attribute__ ((__visibility__("default")))
-#  elif (defined(_WIN32) || defined(__CYGWIN__))
+#  elif defined(_WIN32)
 #     if defined(BOOST_PYTHON_SOURCE)
 #        define BOOST_PYTHON_DECL __declspec(dllexport)
 #        define BOOST_PYTHON_BUILD_DLL
--- boost_1_40_0/boost/python/detail/wrap_python.hpp	2007-04-27 17:16:47.000000000 -0500
+++ boost_1_40_0/boost/python/detail/wrap_python.hpp	2009-08-27 23:41:34.096545600 -0500
@@ -82,8 +82,8 @@
 // Some things we need in order to get Python.h to work with compilers other
 // than MSVC on Win32
 //
-#if defined(_WIN32) || defined(__CYGWIN__)
-# if defined(__GNUC__) && defined(__CYGWIN__)
+#if defined(_WIN32)
+# if defined(__GNUC__)
 
 #  define SIZEOF_LONG 4
 
--- boost_1_40_0/boost/python/module_init.hpp	2007-06-07 13:08:54.000000000 -0500
+++ boost_1_40_0/boost/python/module_init.hpp	2009-08-27 23:41:34.101545900 -0500
@@ -15,7 +15,7 @@ BOOST_PYTHON_DECL void init_module(char 
 
 }}}
 
-#  if (defined(_WIN32) || defined(__CYGWIN__)) && !defined(BOOST_PYTHON_STATIC_MODULE)
+#  if defined(_WIN32) && !defined(BOOST_PYTHON_STATIC_MODULE)
 
 #   define BOOST_PYTHON_MODULE_INIT(name)               \
 void init_module_##name();                              \
