diff -up ./move/core.hpp~ ./move/core.hpp
--- a/boost/move/core.hpp~	2015-02-09 17:33:35.000000000 +0100
+++ b/boost/move/core.hpp	2015-02-13 13:54:52.012130813 +0100
@@ -43,6 +43,7 @@
 #if defined(BOOST_NO_CXX11_RVALUE_REFERENCES) && !defined(BOOST_MOVE_DOXYGEN_INVOKED)
 
    #include <boost/move/detail/meta_utils.hpp>
+   #include <boost/type_traits/is_class.hpp>
 
    //Move emulation rv breaks standard aliasing rules so add workarounds for some compilers
    #if defined(__GNUC__) && (__GNUC__ >= 4) && \
@@ -65,7 +66,7 @@
    template <class T>
    class rv
       : public ::boost::move_detail::if_c
-         < ::boost::move_detail::is_class_or_union<T>::value
+         < ::boost::is_class<T>::value
          , T
          , ::boost::move_detail::nat
          >::type
