diff -up ./boost/pool/pool.hpp~ ./boost/pool/pool.hpp
--- a/boost/pool/pool.hpp~	2013-08-21 17:49:56.023296922 +0200
+++ b/boost/pool/pool.hpp	2013-08-22 11:38:01.133912638 +0200
@@ -361,9 +361,7 @@ class pool: protected simple_segregated_
     { //! Calculated maximum number of memory chunks that can be allocated in a single call by this Pool.
       size_type partition_size = alloc_size();
       size_type POD_size = math::static_lcm<sizeof(size_type), sizeof(void *)>::value + sizeof(size_type);
-      size_type max_chunks = (std::numeric_limits<size_type>::max() - POD_size) / alloc_size();
-    
-      return max_chunks;
+      return (std::numeric_limits<size_type>::max() - POD_size) / alloc_size();
     }
 
     static void * & nextof(void * const ptr)
