diff --git a/tools/build/src/tools/python.jam b/tools/build/src/tools/python.jam
index 273b28a..2d2031e 100644
--- a/tools/build/v2/tools/python.jam
+++ b/tools/build/v2/tools/python.jam
@@ -428,13 +428,7 @@ local rule windows-installed-pythons ( version ? )
 
 local rule darwin-installed-pythons ( version ? )
 {
-    version ?= $(.version-countdown) ;
-
-    local prefix
-      = [ GLOB /System/Library/Frameworks /Library/Frameworks
-          : Python.framework ] ;
-
-    return $(prefix)/Versions/$(version)/bin/python ;
+    return ;
 }
 
 
@@ -890,25 +884,6 @@ local rule configure ( version ? : cmd-or-prefix ? : includes * : libraries ? :
 
     # See if we can find a framework directory on darwin.
     local framework-directory ;
-    if $(target-os) = darwin
-    {
-        # Search upward for the framework directory.
-        local framework-directory = $(libraries[-1]) ;
-        while $(framework-directory:D=) && $(framework-directory:D=) != Python.framework
-        {
-            framework-directory = $(framework-directory:D) ;
-        }
-
-        if $(framework-directory:D=) = Python.framework
-        {
-            debug-message framework directory is \"$(framework-directory)\" ;
-        }
-        else
-        {
-            debug-message "no framework directory found; using library path" ;
-            framework-directory = ;
-        }
-    }
 
     local dll-path = $(libraries) ;
 
