diff --git a/daemon/src/audio/audiortp/Makefile.am b/daemon/src/audio/audiortp/Makefile.am
index c27eedd..fe64077 100644
--- a/daemon/src/audio/audiortp/Makefile.am
+++ b/daemon/src/audio/audiortp/Makefile.am
@@ -4,6 +4,10 @@ noinst_LTLIBRARIES = libaudiortp.la
 
 if BUILD_ZRTP
 SFL_ZRTP_SRC=audio_zrtp_session.h audio_zrtp_session.cpp zrtp_session_callback.cpp zrtp_session_callback.h
+libaudiortp_la_CXXFLAGS = \
+		@CCGNU2_CFLAGS@ \
+		@ZRTPCPP_CFLAGS@ \
+		@CCRTP_CFLAGS@
 endif
 
 libaudiortp_la_SOURCES = \
