diff --git a/common/paths.cpp b/common/paths.cpp
--- a/common/paths.cpp
+++ b/common/paths.cpp
@@ -151,6 +151,10 @@ wxString PATHS::GetStockDataPath( bool aRespectRunFromBuildDir )
 {
     wxString path;

+    if( wxGetEnv( wxT( "NIX_KICAD10_STOCK_DATA_PATH" ), &path ) ) {
+        return path;
+    }
+
     if( aRespectRunFromBuildDir && wxGetEnv( wxT( "KICAD_RUN_FROM_BUILD_DIR" ), nullptr ) )
     {
         // Allow debugging from build dir by placing relevant files/folders in the build root
@@ -198,6 +202,10 @@ wxString PATHS::GetStockEDALibraryPath()
 {
     wxString path;

+    if( wxGetEnv( wxT( "NIX_KICAD10_STOCK_DATA_PATH" ), &path ) ) {
+        return path;
+    }
+
 #if defined( __WXMAC__ )
     path = GetOSXKicadMachineDataDir();
 #elif defined( __WXMSW__ )
