diff --git a/graph/x11/rules.mk b/graph/x11/rules.mk
index 9aa219a..319d9e7 100644
--- a/graph/x11/rules.mk
+++ b/graph/x11/rules.mk
@@ -48,30 +48,8 @@
 #
 FT_PATH := $(subst ;, ,$(subst :, ,$(subst $(SEP),/,$(PATH))))
 
-ifndef X11_PATH
-  ifneq ($(findstring X11/bin,$(FT_PATH)),)
-    xversion := X11
-  else
-    ifneq ($(findstring X11R6/bin,$(FT_PATH)),)
-      xversion := X11R6
-    else
-      ifneq ($(findstring X11R5/bin,$(FT_PATH)),)
-        xversion := X11R5
-      endif
-    endif
-  endif
-
-  ifdef xversion
-    X11_PATH := $(filter %$(xversion)/bin,$(FT_PATH))
-    X11_PATH := $(X11_PATH:%/bin=%)
-  else
-    X11_DIRS := /usr /usr/X11R6 /usr/local/X11R6
-    X11_XLIB := include/X11/Xlib.h
-    X11_PATH := $(foreach dir,$(X11_DIRS),$(wildcard $(dir)/$(X11_XLIB)))
-    X11_PATH := $(X11_PATH:%/$(X11_XLIB)=%)
-  endif
-endif
-
+X11_INCLUDE := $(shell pkg-config --cflags x11)
+GRAPH_LINK  += $(shell pkg-config --libs x11)
 
 ##########################################################################
 #
@@ -80,19 +58,7 @@ endif
 # gcc on the latter platform, which is why it is safe to use the flags
 # `-L' and `-l' in GRAPH_LINK.
 #
-ifneq ($(X11_PATH),)
-
-  X11_INCLUDE := $(subst /,$(COMPILER_SEP),$(X11_PATH:%=%/include))
-  X11_LIB     := $(subst /,$(COMPILER_SEP),$(X11_PATH:%=%/lib64) \
-                                           $(X11_PATH:%=%/lib))
-
-  # The GRAPH_LINK variable is expanded each time an executable is linked
-  # against the graphics library.
-  #
-  ifeq ($(PLATFORM),unix)
-    GRAPH_LINK += $(X11_LIB:%=-R%)
-  endif
-  GRAPH_LINK += $(X11_LIB:%=-L%) -lX11
+ifneq ($(X11_INCLUDE),)
 
   # Solaris needs a -lsocket in GRAPH_LINK.
   #
@@ -117,12 +83,12 @@ ifneq ($(X11_PATH),)
 	  $(LIBTOOL) --mode=compile $(CC) -static $(CFLAGS) \
                      $(GRAPH_INCLUDES:%=$I%) \
                      $I$(subst /,$(COMPILER_SEP),$(GR_X11)) \
-                     $(X11_INCLUDE:%=$I%) \
+                     $(X11_INCLUDE) \
                      $T$(subst /,$(COMPILER_SEP),$@ $<)
   else
 	  $(CC) $(CFLAGS) $(GRAPH_INCLUDES:%=$I%) \
                 $I$(subst /,$(COMPILER_SEP),$(GR_X11)) \
-                $(X11_INCLUDE:%=$I%) \
+                $(X11_INCLUDE) \
                 $T$(subst /,$(COMPILER_SEP),$@ $<)
   endif
 endif
