diff --git a/common/flatpak-run.c b/common/flatpak-run.c
index 6df992d7..d6018e8f 100644
--- a/common/flatpak-run.c
+++ b/common/flatpak-run.c
@@ -2256,11 +2256,55 @@ out:
   return res;
 }
 
+static void
+get_nix_closure (GHashTable *closure, const gchar *source_path)
+{
+  if (g_file_test (source_path, G_FILE_TEST_IS_SYMLINK))
+    {
+      g_autofree gchar *path = g_malloc(PATH_MAX);
+      realpath(source_path, path);
+      if (g_str_has_prefix(path, "/nix/store/"))
+        {
+          *strchr(path + strlen("/nix/store/"), '/') = 0;
+          g_hash_table_add(closure, g_steal_pointer (&path));
+        }
+    }
+  else if (g_file_test (source_path, G_FILE_TEST_IS_DIR))
+    {
+      g_autoptr(GDir) dir = g_dir_open(source_path, 0, NULL);
+      const gchar *file_name;
+      while ((file_name = g_dir_read_name(dir)))
+        {
+          g_autofree gchar *path = g_build_filename (source_path, file_name, NULL);
+          get_nix_closure (closure, path);
+        }
+    }
+}
+
+static void
+add_nix_store_symlink_targets (FlatpakBwrap *bwrap, const gchar *source_path)
+{
+  GHashTable *closure = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
+
+  get_nix_closure(closure, source_path);
+
+  GHashTableIter iter;
+  gpointer path;
+  g_hash_table_iter_init(&iter, closure);
+  while (g_hash_table_iter_next(&iter, &path, NULL))
+    {
+      flatpak_bwrap_add_args (bwrap, "--ro-bind", path, path, NULL);
+    }
+
+  g_hash_table_destroy(closure);
+}
+
 static void
 add_font_path_args (FlatpakBwrap *bwrap)
 {
   g_autoptr(GString) xml_snippet = g_string_new ("");
   gchar *path_build_tmp = NULL;
+  g_autoptr(GFile) user_font_nix = NULL;
   g_autoptr(GFile) user_font1 = NULL;
   g_autoptr(GFile) user_font2 = NULL;
   g_autoptr(GFile) user_font_cache = NULL;
@@ -2283,6 +2327,18 @@ add_font_path_args (FlatpakBwrap *bwrap)
                               "\t<remap-dir as-path=\"%s\">/run/host/fonts</remap-dir>\n",
                               SYSTEM_FONTS_DIR);
     }
+  else if (g_file_test ("/run/current-system/sw/share/X11/fonts", G_FILE_TEST_EXISTS))
+    {
+      add_nix_store_symlink_targets (bwrap, "/run/current-system/sw/share/X11/fonts");
+      flatpak_bwrap_add_args (bwrap,
+                              "--ro-bind",
+                              "/run/current-system/sw/share/X11/fonts",
+                              "/run/host/fonts",
+                              NULL);
+      g_string_append_printf (xml_snippet,
+                              "\t<remap-dir as-path=\"%s\">/run/host/fonts</remap-dir>\n",
+                              "/run/current-system/sw/share/X11/fonts");
+    }
 
   if (g_file_test ("/usr/local/share/fonts", G_FILE_TEST_EXISTS))
     {
@@ -2317,6 +2373,10 @@ add_font_path_args (FlatpakBwrap *bwrap)
                               NULL);
     }
 
+  path_build_tmp = g_build_filename (g_get_home_dir (), ".nix-profile/share/fonts", NULL);
+  user_font_nix = g_file_new_for_path (path_build_tmp);
+  g_clear_pointer (&path_build_tmp, g_free);
+
   path_build_tmp = g_build_filename (g_get_user_data_dir (), "fonts", NULL);
   user_font1 = g_file_new_for_path (path_build_tmp);
   g_clear_pointer (&path_build_tmp, g_free);
@@ -2325,7 +2385,19 @@ add_font_path_args (FlatpakBwrap *bwrap)
   user_font2 = g_file_new_for_path (path_build_tmp);
   g_clear_pointer (&path_build_tmp, g_free);
 
-  if (g_file_query_exists (user_font1, NULL))
+  if (g_file_query_exists (user_font_nix, NULL))
+    {
+      add_nix_store_symlink_targets (bwrap, g_file_get_path (user_font_nix));
+      flatpak_bwrap_add_args (bwrap,
+                              "--ro-bind",
+                              flatpak_file_get_path_cached (user_font_nix),
+                              "/run/host/user-fonts",
+                              NULL);
+      g_string_append_printf (xml_snippet,
+                              "\t<remap-dir as-path=\"%s\">/run/host/user-fonts</remap-dir>\n",
+                              flatpak_file_get_path_cached (user_font_nix));
+    }
+  else if (g_file_query_exists (user_font1, NULL))
     {
       flatpak_bwrap_add_args (bwrap,
                               "--ro-bind", flatpak_file_get_path_cached (user_font1), "/run/host/user-fonts",
@@ -2374,7 +2446,8 @@ add_font_path_args (FlatpakBwrap *bwrap)
 static void
 add_icon_path_args (FlatpakBwrap *bwrap)
 {
-  g_autofree gchar *user_icons_path = NULL;
+  gchar *path_build_tmp = NULL;
+  g_autoptr(GFile) user_icons_nix = NULL;
   g_autoptr(GFile) user_icons = NULL;
 
   if (g_file_test ("/usr/share/icons", G_FILE_TEST_IS_DIR))
@@ -2383,10 +2456,30 @@ add_icon_path_args (FlatpakBwrap *bwrap)
                               "--ro-bind", "/usr/share/icons", "/run/host/share/icons",
                               NULL);
     }
+  else if (g_file_test ("/run/current-system/sw/share/icons", G_FILE_TEST_IS_DIR))
+    {
+      add_nix_store_symlink_targets (bwrap, "/run/current-system/sw/share/icons");
+      flatpak_bwrap_add_args (bwrap,
+                              "--ro-bind", "/run/current-system/sw/share/icons", "/run/host/share/icons",
+                              NULL);
+    }
 
-  user_icons_path = g_build_filename (g_get_user_data_dir (), "icons", NULL);
-  user_icons = g_file_new_for_path (user_icons_path);
-  if (g_file_query_exists (user_icons, NULL))
+  path_build_tmp = g_build_filename (g_get_home_dir (), ".nix-profile/share/icons", NULL);
+  user_icons_nix = g_file_new_for_path (path_build_tmp);
+  g_clear_pointer (&path_build_tmp, g_free);
+
+  path_build_tmp = g_build_filename (g_get_user_data_dir (), "icons", NULL);
+  user_icons = g_file_new_for_path (path_build_tmp);
+  g_clear_pointer (&path_build_tmp, g_free);
+
+  if (g_file_query_exists (user_icons_nix, NULL))
+    {
+      add_nix_store_symlink_targets(bwrap, flatpak_file_get_path_cached (user_icons_nix));
+      flatpak_bwrap_add_args(bwrap,
+                             "--ro-bind", flatpak_file_get_path_cached (user_icons_nix), "/run/host/user-share/icons",
+                             NULL);
+    }
+  else if (g_file_query_exists (user_icons, NULL))
     {
       flatpak_bwrap_add_args (bwrap,
                               "--ro-bind", flatpak_file_get_path_cached (user_icons), "/run/host/user-share/icons",
