--- a/setup.py	2014-02-04 16:12:37.021993713 +0100
+++ b/setup.py	2014-02-04 16:11:42.653995607 +0100
@@ -13,8 +13,8 @@
     package_dir = {'': 'src'},
     packages = ['fedpkg'],
     scripts = ['src/bin/fedpkg'],
-    data_files = [('/etc/bash_completion.d', ['src/fedpkg.bash']),
-                  ('/etc/rpkg', ['src/fedpkg.conf']),
-                  ('/usr/libexec/', ['src/fedpkg-fixbranches.py']),
+    data_files = [('etc/bash_completion.d', ['src/fedpkg.bash']),
+                  ('etc/rpkg', ['src/fedpkg.conf']),
+                  ('libexec/', ['src/fedpkg-fixbranches.py']),
                  ]
 )
