diff -Nur meshlab.orig/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/Raytracer/RayTracer.cpp meshlab/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/Raytracer/RayTracer.cpp
--- meshlab.orig/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/Raytracer/RayTracer.cpp	2011-01-25 18:51:28.000000000 +1100
+++ meshlab/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/Raytracer/RayTracer.cpp	2012-01-07 11:47:12.836800072 +1100
@@ -7,6 +7,12 @@
 #include "SyntopiaCore/Logging/Logging.h"
 #include "SyntopiaCore/Misc/MiniParser.h"
 
+#ifdef Q_WS_MAC
+#include <OpenGL/glu.h>
+#else
+#include <GL/glu.h>
+#endif
+
 using namespace SyntopiaCore::Math;
 using namespace SyntopiaCore::Misc;
 
diff -Nur meshlab.orig/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/Sphere.h meshlab/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/Sphere.h
--- meshlab.orig/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/Sphere.h	2011-01-25 18:51:28.000000000 +1100
+++ meshlab/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/Sphere.h	2012-01-07 11:45:26.795586149 +1100
@@ -3,6 +3,12 @@
 #include "SyntopiaCore/Math/Vector3.h"
 #include "Object3D.h"
 
+#ifdef Q_WS_MAC
+#include <OpenGL/glu.h>
+#else
+#include <GL/glu.h>
+#endif
+
 namespace SyntopiaCore {
 	namespace GLEngine {	
 
diff -Nur meshlab.orig/src/external/structuresynth/structuresynth.pro meshlab/src/external/structuresynth/structuresynth.pro
--- meshlab.orig/src/external/structuresynth/structuresynth.pro	2011-01-25 18:49:34.000000000 +1100
+++ meshlab/src/external/structuresynth/structuresynth.pro	2012-01-07 11:51:46.695981995 +1100
@@ -113,6 +113,7 @@
            ssynth/StructureSynth/Model/Rendering/TemplateRenderer.cpp
 CONFIG+=opengl
 QT+=xml opengl script
+unix:LIBS += -lGLU
 macx:DESTDIR       = ../lib/macx
 win32-g++:DESTDIR       = ../lib/win32-gcc
 win32-msvc2005:DESTDIR       = ../lib/win32-msvc2005
diff -Nur meshlab.orig/src/meshlab/meshlab.pro meshlab/src/meshlab/meshlab.pro
--- meshlab.orig/src/meshlab/meshlab.pro	2011-02-15 20:39:47.000000000 +1100
+++ meshlab/src/meshlab/meshlab.pro	2012-01-07 11:57:18.055586086 +1100
@@ -85,6 +85,7 @@
 QT += xmlpatterns
 QT += network
 QT += script
+unix:LIBS += -lGLU
 
 
 # the following line is needed to avoid mismatch between
