diff --git a/nautilus_open_any_terminal/nautilus_open_any_terminal.py b/nautilus_open_any_terminal/nautilus_open_any_terminal.py
index b02a995..a616399 100644
--- a/nautilus_open_any_terminal/nautilus_open_any_terminal.py
+++ b/nautilus_open_any_terminal/nautilus_open_any_terminal.py
@@ -228,9 +228,10 @@ def set_terminal_args(*args):
         """Provide keyboard shortcuts for opening terminals in Nautilus."""

         def __init__(self):
-            gsettings_source = Gio.SettingsSchemaSource.get_default()
-            if gsettings_source.lookup(GSETTINGS_PATH, True):
-                self._gsettings = Gio.Settings.new(GSETTINGS_PATH)
+            gsettings_source = Gio.SettingsSchemaSource.new_from_directory("@gsettings_path@", Gio.SettingsSchemaSource.get_default(), True)
+            if True:
+                _schema = gsettings_source.lookup(GSETTINGS_PATH, False)
+                self._gsettings = Gio.Settings.new_full(_schema, None, None);
                 self._gsettings.connect("changed", self._bind_shortcut)
                 self._create_accel_group()
             self._window = None
@@ -326,9 +327,10 @@ class OpenAnyTerminalExtension(GObject.GObject, Nautilus.MenuProvider):
         return items
 
 
-source = Gio.SettingsSchemaSource.get_default()
-if source is not None and source.lookup(GSETTINGS_PATH, True):
-    _gsettings = Gio.Settings.new(GSETTINGS_PATH)
+source = Gio.SettingsSchemaSource.new_from_directory("@gsettings_path@", Gio.SettingsSchemaSource.get_default(), True)
+if True:
+    _schema = source.lookup(GSETTINGS_PATH, False)
+    _gsettings = Gio.Settings.new_full(_schema, None, None);
     _gsettings.connect("changed", set_terminal_args)
     set_terminal_args()
