diff --git a/src/launch/launcher.c b/src/launch/launcher.c
index 7b1fb19..6bc2c46 100644
--- a/src/launch/launcher.c
+++ b/src/launch/launcher.c
@@ -945,9 +945,7 @@ static int launcher_load_standard_system_services(Launcher *launcher, NSSCache *
         static const char *default_data_dirs[] = {
                 "/etc",
                 "/run",
-                "/usr/local/share",
-                "/usr/share",
-                "/lib",
+                "/run/current-system/sw/share",
                 NULL,
         };
         const char *suffix = "dbus-1/system-services";
@@ -1033,9 +1031,9 @@ static int launcher_parse_config(Launcher *launcher, ConfigRoot **rootp, NSSCach
         if (launcher->configfile)
                 configfile = launcher->configfile;
         else if (launcher->user_scope)
-                configfile = "/usr/share/dbus-1/session.conf";
+                configfile = "/etc/dbus-1/session.conf";
         else
-                configfile = "/usr/share/dbus-1/system.conf";
+                configfile = "/etc/dbus-1/system.conf";
 
         config_parser_init(&parser);
 
