diff --git a/hepmc/hepmcwrap.i b/hepmc/hepmcwrap.i
index cf35c1b..b94fbe2 100644
--- a/hepmc/hepmcwrap.i
+++ b/hepmc/hepmcwrap.i
@@ -1,6 +1,7 @@
 %module hepmcwrap
 
 %{
+  #include "HepMC/Flow.h"
   #include "HepMC/GenEvent.h"
   #include "HepMC/GenVertex.h"
   #include "HepMC/GenParticle.h"
@@ -93,6 +94,7 @@ namespace HepMC {
 // headers before importing headers that use those classes.
 // Result is that headers should probably be %included in an order
 // which sees "contents before containers"
+%include "HepMC/Flow.h"
 %include "HepMC/HepMCDefs.h"
 %include "HepMC/SimpleVector.h"
 // #ifdef HEPMC_HAS_ITERATOR_RANGES
