diff --git a/application/src/main/java/nl/jixxed/eliteodysseymaterials/Main.java b/application/src/main/java/nl/jixxed/eliteodysseymaterials/Main.java
index 04b101f0..2e589e12 100644
--- a/application/src/main/java/nl/jixxed/eliteodysseymaterials/Main.java
+++ b/application/src/main/java/nl/jixxed/eliteodysseymaterials/Main.java
@@ -51,48 +51,6 @@ public class Main {
                 System.exit(0);
             }
             log.info("launching app with java version: " + getVersion());
-            if (System.getProperty("sentry.dsn") == null || System.getProperty("sentry.dsn").isBlank() || System.getProperty("sentry.dsn").equals("null")) {
-                log.error("Sentry DSN is not set. Please set the DSN.");
-            }
-            Sentry.init(options -> {
-
-                final String buildVersion = getBuildVersion();
-                options.setDsn(System.getProperty("sentry.dsn"));
-                options.setEnvironment(getEnvironment(buildVersion));
-                options.setRelease("edomh-app@" + buildVersion);
-                options.setEnabled(!buildVersion.equals("dev"));
-                options.setBeforeSend((event, hint) -> {
-                    if (!VersionService.isLatestVersion()) {
-                        return null;  // Returning null prevents the event from being sent to Sentry
-                    }
-                    if (Duration.between(lastSentTime, Instant.now()).getSeconds() < 30) {
-                        return null;  // Throttle if less than 30 seconds have passed
-                    }
-                    lastSentTime = Instant.now();
-                    String supportFile = "";
-                    try {
-                        supportFile = SupportService.createSupportPackage();
-                    } catch (Exception e) {
-                        log.error("Failed to create support package", e);
-                    }
-                    if (!supportFile.isBlank()) {
-                        Attachment attachment = new Attachment(supportFile);
-                        hint.addAttachment(attachment);
-                    }
-                    OperatingSystem os = new OperatingSystem();
-                    os.setName(System.getProperty("os.name"));
-                    os.setVersion(System.getProperty("os.version"));
-                    os.setBuild(System.getProperty("os.arch"));
-                    event.getContexts().setOperatingSystem(os);
-                    if (APPLICATION_STATE.getFileheader() != null) {
-                        event.setTag("game.version", APPLICATION_STATE.getFileheader().getGameversion());
-                        event.setTag("game.build", APPLICATION_STATE.getFileheader().getBuild());
-                        event.setTag("game.language", APPLICATION_STATE.getFileheader().getLanguage());
-                        event.setTag("game.odyssey", String.valueOf(APPLICATION_STATE.getFileheader().getOdyssey()));
-                    }
-                    return event;
-                });
-            });
             FXApplication.launchFx(args);
         }
     }
