diff --git i/src/main/index.ts w/src/main/index.ts
index 23ea0d6..1ef465f 100644
--- i/src/main/index.ts
+++ w/src/main/index.ts
@@ -22,7 +22,9 @@ import { isDeckGameMode } from "./utils/steamOS";
 
 if (!IS_DEV) {
-    autoUpdater.checkForUpdatesAndNotify();
+    console.log("Update checking is disabled. Skipping...");
+    // autoUpdater.checkForUpdatesAndNotify();
+
 }
 
 console.log("Equibop v" + app.getVersion());

