diff --git a/src/libunicode/CMakeLists.txt b/src/libunicode/CMakeLists.txt
index 25abbd1..99f8b8d 100644
--- a/src/libunicode/CMakeLists.txt
+++ b/src/libunicode/CMakeLists.txt
@@ -317,10 +317,10 @@ if(LIBUNICODE_TESTING)
     if(NOT Catch2_FOUND)
         # Suppress all warnings for embedded third-party Catch2.
         # https://github.com/catchorg/Catch2/issues/2583
-        suppress_thirdparty_warnings(Catch2)
-        suppress_thirdparty_warnings(Catch2WithMain)
-        get_target_property(CATCH2_INCLUDE_DIRS Catch2 INTERFACE_INCLUDE_DIRECTORIES)
-        target_include_directories(Catch2 SYSTEM INTERFACE ${CATCH2_INCLUDE_DIRS})
+        # suppress_thirdparty_warnings(Catch2)
+        # suppress_thirdparty_warnings(Catch2WithMain)
+        # get_target_property(CATCH2_INCLUDE_DIRS Catch2 INTERFACE_INCLUDE_DIRECTORIES)
+        # target_include_directories(Catch2 SYSTEM INTERFACE ${CATCH2_INCLUDE_DIRS})
     endif()
 
     target_link_libraries(unicode_test unicode Catch2::Catch2WithMain)
