diff --git a/SConstruct b/SConstruct
index 8ba8bbd..95eab3b 100644
--- a/SConstruct
+++ b/SConstruct
@@ -24,6 +24,8 @@ USING_CLANG = OSX or os.environ.get('CC', None) == 'clang'
 #
 BOOST_HOME = os.environ.get("RIPPLED_BOOST_HOME", None)
 
+ZLIB_HOME = os.environ.get("RIPPLED_ZLIB_HOME", None)
+
 
 if OSX or Ubuntu or Debian or Archlinux:
     CTAGS = 'ctags'
@@ -36,7 +38,7 @@ else:
 # scons tools
 #
 
-HONOR_ENVS = ['CC', 'CXX', 'PATH']
+HONOR_ENVS = ['CC', 'CXX', 'PATH', 'PKG_CONFIG_PATH']
 
 env = Environment(
     tools = ['default', 'protoc'],
@@ -156,8 +158,8 @@ INCLUDE_PATHS = [
     'build/proto'
     ]
 
-# if BOOST_HOME:
-#     INCLUDE_PATHS.append(BOOST_HOME)
+if BOOST_HOME:
+    INCLUDE_PATHS.append("%s/include" % BOOST_HOME)
 
 #-------------------------------------------------------------------------------
 #
@@ -261,7 +263,11 @@ env.Append(
 # such, as installed into `/usr/lib/`
 if BOOST_HOME is not None:
     env.Prepend(
-        LIBPATH = ["%s/stage/lib" % BOOST_HOME])
+        LIBPATH = ["%s/lib" % BOOST_HOME])
+
+if ZLIB_HOME is not None:
+    env.Prepend(
+        LIBPATH = ["%s/lib" % ZLIB_HOME])
 
 if not OSX:
     env.Append(LINKFLAGS = [
