diff --git a/kcms/kwinsupportinfo/kcm_kwinsupportinfo.json.in b/kcms/kwinsupportinfo/kcm_kwinsupportinfo.json.in
index 0de6973e..30035768 100644
--- a/kcms/kwinsupportinfo/kcm_kwinsupportinfo.json.in
+++ b/kcms/kwinsupportinfo/kcm_kwinsupportinfo.json.in
@@ -85,6 +85,6 @@
         "Name[zh_CN]": "窗口管理器",
         "Name[zh_TW]": "視窗管理員"
     },
-    "TryExec": "@QtBinariesDir@/qdbus",
+    "TryExec": "@qdbus@",
     "X-KDE-KInfoCenter-Category": "graphical_information"
 }
diff --git a/kcms/kwinsupportinfo/main.cpp b/kcms/kwinsupportinfo/main.cpp
index 6ae168b5..89d0a2ff 100644
--- a/kcms/kwinsupportinfo/main.cpp
+++ b/kcms/kwinsupportinfo/main.cpp
@@ -18,7 +18,7 @@ public:
     explicit KCMKWinSupportInfo(QObject *parent, const KPluginMetaData &data)
         : KQuickConfigModule(parent, data)
     {
-        m_outputContext = new CommandOutputContext(QLibraryInfo::path(QLibraryInfo::BinariesPath) + QStringLiteral("/qdbus"),
+        m_outputContext = new CommandOutputContext(QStringLiteral("@qdbus@"),
                                                    {QStringLiteral("org.kde.KWin"), QStringLiteral("/KWin"), QStringLiteral("supportInformation")},
                                                    parent);
     }
diff --git a/kcms/xserver/kcm_xserver.json b/kcms/xserver/kcm_xserver.json
index b3f1ce24..dc610932 100644
--- a/kcms/xserver/kcm_xserver.json
+++ b/kcms/xserver/kcm_xserver.json
@@ -141,7 +141,7 @@
         "Name[zh_CN]": "X 服务器",
         "Name[zh_TW]": "X 伺服器"
     },
-    "TryExec": "xdpyinfo",
+    "TryExec": "@xdpyinfo@",
     "X-DocPath": "kinfocenter/graphical.html#xserver",
     "X-KDE-KInfoCenter-Category": "graphical_information",
     "X-KDE-Keywords": "X,X-Server,XServer,XFree86,Display,VideoCard,System Information",
diff --git a/kcms/xserver/main.cpp b/kcms/xserver/main.cpp
index 6d4d1e25..d39fe19e 100644
--- a/kcms/xserver/main.cpp
+++ b/kcms/xserver/main.cpp
@@ -16,7 +16,7 @@ public:
     explicit KCMXServer(QObject *parent, const KPluginMetaData &data)
         : KQuickConfigModule(parent, data)
     {
-        m_outputContext = new CommandOutputContext(QStringLiteral("xdpyinfo"), {}, parent);
+        m_outputContext = new CommandOutputContext(QStringLiteral("@xdpyinfo@"), {}, parent);
     }
     CommandOutputContext *outputContext() const
     {
