From d141939b3d6c51f5b4e166ba3a0c569e31e65321 Mon Sep 17 00:00:00 2001
From: MilesCranmer <miles.cranmer@gmail.com>
Date: Fri, 7 Jun 2024 03:04:20 +0100
Subject: [PATCH] hardcode macos SDK

---
 base/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/base/Makefile b/base/Makefile
index ad2bb6a63c..666eefac36 100644
--- a/base/Makefile
+++ b/base/Makefile
@@ -89,8 +89,8 @@ else
 	@echo "const DARWIN_FRAMEWORK = false" >> $@
 endif
 ifeq ($(OS), Darwin)
-	@echo "const MACOS_PRODUCT_VERSION = \"$(shell sw_vers -productVersion)\"" >> $@
-	@echo "const MACOS_PLATFORM_VERSION = \"$(shell xcrun --show-sdk-version)\"" >> $@
+	@echo "const MACOS_PRODUCT_VERSION = \"$(MACOS_PRODUCT_VERSION)\"" >> $@
+	@echo "const MACOS_PLATFORM_VERSION = \"$(MACOS_PLATFORM_VERSION)\"" >> $@
 endif
 	@echo "const BUILD_TRIPLET = \"$(BB_TRIPLET_LIBGFORTRAN_CXXABI)\"" >> $@
 
-- 
2.39.0

