diff --git a/pyscf/lib/misc.py b/pyscf/lib/misc.py
index ed43689ff..a8a0d0e20 100644
--- a/pyscf/lib/misc.py
+++ b/pyscf/lib/misc.py
@@ -42,8 +42,6 @@ if h5py.version.version[:4] == '2.2.':
     sys.stderr.write('h5py-%s is found in your environment. '
                      'h5py-%s has bug in threading mode.\n'
                      'Async-IO is disabled.\n' % ((h5py.version.version,)*2))
-if h5py.version.version[:2] == '3.':
-    h5py.get_config().default_file_mode = 'a'
 
 c_double_p = ctypes.POINTER(ctypes.c_double)
 c_int_p = ctypes.POINTER(ctypes.c_int)
