diff --git i/src/mkimage.cc w/src/mkimage.cc
index 02e65b1..b263796 100755
--- i/src/mkimage.cc
+++ w/src/mkimage.cc
@@ -285,27 +285,27 @@ bostream& JigdoDescVec::put(bostream& file, MD5Sum* md) const {
 //______________________________________________________________________
 
 namespace {
-  const int SIZE_WIDTH = 12;
+  const int MKIMAGE_SIZE_WIDTH = 12;
 }
 
 ostream& JigdoDesc::ImageInfo::put(ostream& s) const {
-  s << "image-info  " << setw(SIZE_WIDTH) << size() << "              "
+  s << "image-info  " << setw(MKIMAGE_SIZE_WIDTH) << size() << "              "
     << md5() << ' ' << blockLength() << '\n';
   return s;
 }
 ostream& JigdoDesc::UnmatchedData::put(ostream& s) const {
-  s << "in-template " << setw(SIZE_WIDTH) << offset() << ' '
-    << setw(SIZE_WIDTH) << size() << '\n';
+  s << "in-template " << setw(MKIMAGE_SIZE_WIDTH) << offset() << ' '
+    << setw(MKIMAGE_SIZE_WIDTH) << size() << '\n';
   return s;
 }
 ostream& JigdoDesc::MatchedFile::put(ostream& s) const {
-  s << "need-file   " << setw(SIZE_WIDTH) << offset() << ' '
-    << setw(SIZE_WIDTH) << size() << ' ' << md5() << ' ' << rsync() << '\n';
+  s << "need-file   " << setw(MKIMAGE_SIZE_WIDTH) << offset() << ' '
+    << setw(MKIMAGE_SIZE_WIDTH) << size() << ' ' << md5() << ' ' << rsync() << '\n';
   return s;
 }
 ostream& JigdoDesc::WrittenFile::put(ostream& s) const {
-  s << "have-file   " << setw(SIZE_WIDTH) << offset() << ' '
-    << setw(SIZE_WIDTH) << size() << ' ' << md5() << ' ' << rsync() << '\n';
+  s << "have-file   " << setw(MKIMAGE_SIZE_WIDTH) << offset() << ' '
+    << setw(MKIMAGE_SIZE_WIDTH) << size() << ' ' << md5() << ' ' << rsync() << '\n';
   return s;
 }
 
