From a164f7a2b662831ba26f5cebce23f97f1b5afccd Mon Sep 17 00:00:00 2001
From: Tom Hunger <tehunger@gmail.com>
Date: Thu, 8 Jan 2015 14:07:17 +0000
Subject: [PATCH] remove-custom-build-step

---
 Setup.hs | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Setup.hs b/Setup.hs
index 6f80fd8..a45ffc1 100644
--- a/Setup.hs
+++ b/Setup.hs
@@ -19,8 +19,6 @@ main =
 myPostBuild :: Args -> BuildFlags -> PackageDescription -> LocalBuildInfo -> IO ()
 myPostBuild args flags pd lbi =
   do  putStrLn "Custom build step: compiling debuggerInterface.elm"
-      buildInterface
-      concatJS lbi
       postBuild simpleUserHooks args flags pd lbi
 
 
-- 
1.9.4

