commit 9911b327ac299ec7eeae81b98cb520f4153071f2
Author: Christian Grothoff <christian@grothoff.org>
Date:   Wed Dec 9 07:25:26 2020 +0100

    fix #6665

diff --git a/src/testing/testing_api_cmd_rewind.c b/src/testing/testing_api_cmd_rewind.c
index e1b17fa9..979607cd 100644
--- a/src/testing/testing_api_cmd_rewind.c
+++ b/src/testing/testing_api_cmd_rewind.c
@@ -24,7 +24,6 @@
  */
 #include "platform.h"
 #include "taler_exchange_service.h"
-#include "taler/taler_testing_lib.h"
 #include "taler_testing_lib.h"


diff --git a/src/testing/testing_api_cmd_twister_exec_client.c b/src/testing/testing_api_cmd_twister_exec_client.c
index 2cb92a8f..b3903f29 100644
--- a/src/testing/testing_api_cmd_twister_exec_client.c
+++ b/src/testing/testing_api_cmd_twister_exec_client.c
@@ -26,7 +26,7 @@
  */

 #include "platform.h"
-#include <taler/taler_testing_lib.h>
+#include "taler_testing_lib.h"
 #include "taler_twister_testing_lib.h"


diff --git a/src/testing/testing_api_trait_uuid.c b/src/testing/testing_api_trait_uuid.c
index c9b73a5b..4d5003ec 100644
--- a/src/testing/testing_api_trait_uuid.c
+++ b/src/testing/testing_api_trait_uuid.c
@@ -22,9 +22,9 @@
  * @author Jonathan Buchanan
  */
 #include "platform.h"
-#include <taler/taler_signatures.h>
-#include <taler/taler_exchange_service.h>
-#include <taler/taler_testing_lib.h>
+#include "taler_signatures.h"
+#include "taler_exchange_service.h"
+#include "taler_testing_lib.h"


 #define TALER_TESTING_TRAIT_UUID "uuid"
