From 2ea60bdf4d7b0344fc6ff5c97c675842fedccfa8 Mon Sep 17 00:00:00 2001
From: Cole Helbling <cole.e.helbling@outlook.com>
Date: Fri, 23 Apr 2021 09:02:22 -0700
Subject: [PATCH] setup.py: remove impure tuning

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index 7840022..3922aa5 100644
--- a/setup.py
+++ b/setup.py
@@ -48,7 +48,6 @@ setup(
     ext_modules=[
         Extension("intbitset",
                   ["intbitset/intbitset.c", "intbitset/intbitset_impl.c"],
-                  extra_compile_args=['-O3', '-march=core2', '-mtune=native']
                   # For debug -> '-ftree-vectorizer-verbose=2'
                   )
     ],
-- 
2.30.1

