diff --git a/src/ui.zig b/src/ui.zig
index 8401910..50171a7 100644
--- a/src/ui.zig
+++ b/src/ui.zig
@@ -8,11 +8,11 @@ const main = @import("main.zig");
 const util = @import("util.zig");

 pub const c = @cImport({
+    @cDefine("_XOPEN_SOURCE", "1");
     @cInclude("stdio.h");
     @cInclude("string.h");
     @cInclude("curses.h");
     @cInclude("time.h");
-    @cDefine("_X_OPEN_SOURCE", "1");
     @cInclude("wchar.h");
     @cInclude("locale.h");
 });
