diff -rc kdelibs-3.5.10-orig/kio/kio/kdirwatch.cpp kdelibs-3.5.10/kio/kio/kdirwatch.cpp
*** kdelibs-3.5.10-orig/kio/kio/kdirwatch.cpp	2006-07-22 10:16:37.000000000 +0200
--- kdelibs-3.5.10/kio/kio/kdirwatch.cpp	2009-04-01 13:26:48.000000000 +0200
***************
*** 64,74 ****
  // debug
  #include <sys/ioctl.h>
  
! #ifdef HAVE_INOTIFY
  #include <unistd.h>
  #include <fcntl.h>
  #include <sys/syscall.h>
- #include <linux/types.h>
  // Linux kernel headers are documented to not compile
  #define _S390_BITOPS_H
  #include <linux/inotify.h>
--- 64,76 ----
  // debug
  #include <sys/ioctl.h>
  
! #if 1
! #include <sys/inotify.h>
! #include <fcntl.h>
! #elif HAVE_INOTIFY
  #include <unistd.h>
  #include <fcntl.h>
  #include <sys/syscall.h>
  // Linux kernel headers are documented to not compile
  #define _S390_BITOPS_H
  #include <linux/inotify.h>
