--- a/common/mlstdutils/std_utils.ml	2019-02-07 15:45:56.516955598 +0100
+++ b/common/mlstdutils/std_utils.ml	2019-02-07 15:45:56.516955598 +0100
@@ -305,7 +305,7 @@
       | x::xs, y::ys, z::zs -> (x, y, z) :: combine3 xs ys zs
       | _ -> invalid_arg "combine3"
 
-    let rec assoc_lbl ?(cmp = compare) ~default x = function
+    let rec assoc_lbl ?(cmp = Pervasives.compare) ~default x = function
       | [] -> default
       | (y, y') :: _ when cmp x y = 0 -> y'
       | _ :: ys -> assoc_lbl ~cmp ~default x ys
