diff --git a/setup.py b/setup.py
index 1b5f513..d660b9a 100644
--- a/setup.py
+++ b/setup.py
@@ -199,6 +199,7 @@
             pdata = preprocess_file(
                 header_path,
                 cpp_args=["-std=c99", "-D__extension__=", "-D__attribute__(x)="],
+                cpp_path="@crossPrefix@-cpp",
             )
         parser = c_parser.CParser()
         ast = parser.parse(pdata, "tss2_tpm2_types.h")
@@ -238,6 +239,7 @@
                             "-D__float128=long double",
                             "-D_FORTIFY_SOURCE=0",
                         ],
+                        cpp_path="@crossPrefix@-cpp",
                     )
                 parser = c_parser.CParser()
                 past = parser.parse(pdata, "tss2_policy.h")
