diff --git c/src/backup/k9dvdbackup.cpp i/src/backup/k9dvdbackup.cpp
index f5e4859..82fa392 100755
--- c/src/backup/k9dvdbackup.cpp
+++ i/src/backup/k9dvdbackup.cpp
@@ -907,7 +907,7 @@ k9Vobu * k9DVDBackup::remapOffset(uint32_t _sector,uint32_t *_offset,int _dir) {
 
 
         if ((vobu1 !=NULL) && (vobu2!=NULL)) {
-            *_offset = abs(vobu1->newSector - vobu2->newSector)  | maskOffset1 ;
+            *_offset = (vobu1->newSector - vobu2->newSector)  | maskOffset1 ;
             *_offset |= maskOffset2;
             return vobu2;
         }
diff --git c/src/backup/k9execcopy.cpp i/src/backup/k9execcopy.cpp
index d59222c..35de923 100644
--- c/src/backup/k9execcopy.cpp
+++ i/src/backup/k9execcopy.cpp
@@ -306,7 +306,7 @@ void k9ExecCopy::createMkv(k9DVDTitle *_title,const QString &_filename,QMultiMap
 
 #if QT_VERSION >= 0x050000
     m_progressDialog=new QProgressDialog(k9Dialogs::getMainWidget() );
-    m_progressDialog->setCancelButton(false);
+    m_progressDialog->setCancelButton(0);
 
 #else
     m_progressDialog=new KProgressDialog(k9Dialogs::getMainWidget() );
