diff --git a/src/test/run-make/relocation-model/Makefile b/src/test/run-make/relocation-model/Makefile
index b22f34f..c6489bd 100644
--- a/src/test/run-make/relocation-model/Makefile
+++ b/src/test/run-make/relocation-model/Makefile
@@ -2,9 +2,11 @@
 
 all: others
 	$(RUSTC) -C relocation-model=dynamic-no-pic foo.rs
+	paxctl -czexm $(TMPDIR)/foo
 	$(call RUN,foo)
 
 	$(RUSTC) -C relocation-model=default foo.rs
+	paxctl -czexm $(TMPDIR)/foo
 	$(call RUN,foo)
 
 	$(RUSTC) -C relocation-model=default --crate-type=dylib foo.rs
@@ -16,6 +18,7 @@ others:
 else
 others:
 	$(RUSTC) -C relocation-model=static foo.rs
+	paxctl -czexm $(TMPDIR)/foo
 	$(call RUN,foo)
 	$(RUSTC) -C relocation-model=static --crate-type=dylib foo.rs
 endif
