diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6dcb690..d71ccaf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -345,8 +345,9 @@ if(MINGW OR (NOT WIN32))
     find_package(directxmath CONFIG REQUIRED)
     target_link_libraries(${PROJECT_NAME} PUBLIC Microsoft::DirectXMath)
 
-    find_package(directx-headers CONFIG REQUIRED)
-    target_link_libraries(${PROJECT_NAME} PUBLIC Microsoft::DirectX-Headers)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(DIRECTXHEADERS REQUIRED DirectX-Headers)
+    target_link_libraries(${PROJECT_NAME} PUBLIC ${DIRECTXHEADERS_LIBRARIES})
     target_compile_definitions(${PROJECT_NAME} PUBLIC USING_DIRECTX_HEADERS)
 else()
     find_package(directxmath CONFIG QUIET)
