diff --git a/plugin.video.yatp/server.py b/plugin.video.yatp/server.py
index 1adcbb5..488b72c 100644
--- a/plugin.video.yatp/server.py
+++ b/plugin.video.yatp/server.py
@@ -20,24 +20,8 @@ addon = Addon()
 _ = addon.initialize_gettext()
 addon.log_notice('Starting Torrent Server...')
 
-# A monkey-patch to set the necessary librorrent version
-librorrent_addon = Addon('script.module.libtorrent')
-orig_custom_version = librorrent_addon.get_setting('custom_version', False)
-orig_set_version = librorrent_addon.get_setting('set_version', False)
-librorrent_addon.set_setting('custom_version', 'true')
-if addon.libtorrent_version == '1.0.9':
-    librorrent_addon.set_setting('set_version', '4')
-elif addon.libtorrent_version == '1.1.0':
-    librorrent_addon.set_setting('set_version', '5')
-elif addon.libtorrent_version == '1.1.1':
-    librorrent_addon.set_setting('set_version', '6')
-else:
-    librorrent_addon.set_setting('set_version', '0')
-
 from libs.server import wsgi_app
 
-librorrent_addon.set_setting('custom_version', orig_custom_version)
-librorrent_addon.set_setting('set_version', orig_set_version)
 # ======
 
 if addon.enable_limits:
