diff --git a/build.gradle b/build.gradle
index 11d7a99..66805bb 100644
--- a/build.gradle
+++ b/build.gradle
@@ -18,9 +18,12 @@ import org.gradle.nativeplatform.platform.internal.OperatingSystemInternal
 import org.gradle.nativeplatform.platform.internal.DefaultNativePlatform
 import org.xml.sax.SAXParseException
 
-plugins {
-   id 'de.fuerstenau.buildconfig' version '1.1.8'
+buildscript {
+  dependencies {
+    classpath files ("BuildConfig.jar")
+  }
 }
+apply plugin: 'de.fuerstenau.buildconfig'
 
 apply plugin: 'java'
 apply plugin: 'application'
