diff --git a/tools/cpanfile b/tools/cpanfile
index 359c61fe..ca3b7ec7 100755
--- a/tools/cpanfile
+++ b/tools/cpanfile
@@ -20,7 +20,7 @@ requires 'Sort::Naturally',     1.03;
 requires 'Authen::Passphrase',  0.008;
 requires 'File::ReadBackwards', 1.05;
 requires 'URI::Escape',         1.74;
-requires 'URI',                 5.09;
+requires 'URI',                 5.05;
 
 # Used by Installer
 requires 'IPC::Cmd', 1.02;
diff --git a/tools/install.pl b/tools/install.pl
index 0cbb847d..1bd61fa0 100755
--- a/tools/install.pl
+++ b/tools/install.pl
@@ -91,32 +91,6 @@ if ( $ENV{HOMEBREW_FORMULA_PREFIX} ) {
     $cpanopt = " -l " . $ENV{HOMEBREW_FORMULA_PREFIX} . "/libexec";
 }
 
-#Load IPC::Cmd
-install_package( "IPC::Cmd",         $cpanopt );
-install_package( "Config::AutoConf", $cpanopt );
-IPC::Cmd->import('can_run');
-require Config::AutoConf;
-
-say("\r\nWill now check if all LRR software dependencies are met. \r\n");
-
-#Check for Redis
-say("Checking for Redis...");
-can_run('redis-server')
-  or die 'NOT FOUND! Please install a Redis server before proceeding.';
-say("OK!");
-
-#Check for GhostScript
-say("Checking for GhostScript...");
-can_run('gs')
-  or warn 'NOT FOUND! PDF support will not work properly. Please install the "gs" tool.';
-say("OK!");
-
-#Check for libarchive
-say("Checking for libarchive...");
-Config::AutoConf->new()->check_header("archive.h")
-  or die 'NOT FOUND! Please install libarchive and ensure its headers are present.';
-say("OK!");
-
 #Check for PerlMagick
 say("Checking for ImageMagick/PerlMagick...");
 my $imgk;
@@ -154,12 +128,6 @@ if ( $back || $full ) {
 #Clientside Dependencies with Provisioning
 if ( $front || $full ) {
 
-    say("\r\nObtaining remote Web dependencies...\r\n");
-
-    if ( system("npm install") != 0 ) {
-        die "Something went wrong while obtaining node modules - Bailing out.";
-    }
-
     say("\r\nProvisioning...\r\n");
 
     #Load File::Copy
