diff --git a/meson.build b/meson.build
index 47c436f..536c60b 100644
--- a/meson.build
+++ b/meson.build
@@ -440,13 +440,12 @@ if build_tools
     if get_option('enable-wayland')
         wayland_client_dep = dependency('wayland-client', version: '>=1.2.0', required: false)
         wayland_protocols_dep = dependency('wayland-protocols', version: '>=1.12', required: false)
-        wayland_scanner_dep = dependency('wayland-scanner', required: false, native: true)
-        if not wayland_client_dep.found() or not wayland_protocols_dep.found() or not wayland_scanner_dep.found()
+        if not wayland_client_dep.found() or not wayland_protocols_dep.found()
             error('''The Wayland xkbcli programs require wayland-client >= 1.2.0, wayland-protocols >= 1.7 which were not found.
 You can disable the Wayland xkbcli programs with -Denable-wayland=false.''')
         endif
 
-        wayland_scanner = find_program(wayland_scanner_dep.get_pkgconfig_variable('wayland_scanner'))
+        wayland_scanner = find_program('wayland-scanner', native: true)
         wayland_scanner_code_gen = generator(
             wayland_scanner,
             output: '@BASENAME@-protocol.c',
