diff --git i/CMakeLists.txt w/CMakeLists.txt
index 0ee7695..86cba7f 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -46,6 +46,7 @@ string(REPLACE "Version: " "" ICINGA2_VERSION ${VERSION_LINE})
 include(GNUInstallDirs)
 include(InstallConfig)
 
+set(ICINGA2_SYSCONFDIR "/etc/icinga2" CACHE STRING "Icinga 2 config dir")
 set(ICINGA2_USER "icinga" CACHE STRING "Icinga 2 user")
 set(ICINGA2_GROUP "icinga" CACHE STRING "Icinga 2 group")
 set(ICINGA2_COMMAND_GROUP "icingacmd" CACHE STRING "Icinga 2 command group")
diff --git i/config.h.cmake w/config.h.cmake
index 1903b28..de12cc5 100644
--- i/config.h.cmake
+++ w/config.h.cmake
@@ -13,7 +13,7 @@
 #cmakedefine ICINGA2_UNITY_BUILD
 
 #define ICINGA_PREFIX "${CMAKE_INSTALL_PREFIX}"
-#define ICINGA_SYSCONFDIR "${CMAKE_INSTALL_FULL_SYSCONFDIR}"
+#define ICINGA_SYSCONFDIR "${ICINGA2_SYSCONFDIR}"
 #define ICINGA_RUNDIR "${ICINGA2_RUNDIR}"
 #define ICINGA_LOCALSTATEDIR "${CMAKE_INSTALL_FULL_LOCALSTATEDIR}"
 #define ICINGA_PKGDATADIR "${CMAKE_INSTALL_FULL_DATADIR}/icinga2"
diff --git i/icinga-app/CMakeLists.txt w/icinga-app/CMakeLists.txt
index 44fb877..6ea339d 100644
--- i/icinga-app/CMakeLists.txt
+++ w/icinga-app/CMakeLists.txt
@@ -52,6 +52,3 @@ install(
   RUNTIME DESTINATION ${InstallPath}
 )
 
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/log/icinga2\")")
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/lib/icinga2\")")
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${ICINGA2_RUNDIR}/icinga2\")")
diff --git i/lib/base/CMakeLists.txt w/lib/base/CMakeLists.txt
index 858943f..9b41ebc 100644
--- i/lib/base/CMakeLists.txt
+++ w/lib/base/CMakeLists.txt
@@ -82,9 +82,6 @@ set_target_properties (
   VERSION ${SPEC_VERSION}
 )
 
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/cache/icinga2\")")
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/log/icinga2/crash\")")
-
 install(
   TARGETS base
   RUNTIME DESTINATION ${CMAKE_INSTALL_SBINDIR}
diff --git i/lib/compat/CMakeLists.txt w/lib/compat/CMakeLists.txt
index b8b4852..8f47199 100644
--- i/lib/compat/CMakeLists.txt
+++ w/lib/compat/CMakeLists.txt
@@ -59,8 +59,4 @@ install_if_not_exists(
 
 install(TARGETS compat RUNTIME DESTINATION ${CMAKE_INSTALL_SBINDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/icinga2)
 
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/log/icinga2/compat/archives\")")
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/spool/icinga2\")")
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${ICINGA2_RUNDIR}/icinga2/cmd\")")
-
 set(CPACK_NSIS_EXTRA_INSTALL_COMMANDS "${CPACK_NSIS_EXTRA_INSTALL_COMMANDS}" PARENT_SCOPE)
diff --git i/lib/livestatus/CMakeLists.txt w/lib/livestatus/CMakeLists.txt
index ac20b44..0d1d824 100644
--- i/lib/livestatus/CMakeLists.txt
+++ w/lib/livestatus/CMakeLists.txt
@@ -52,8 +52,6 @@ install_if_not_exists(
   ${CMAKE_INSTALL_SYSCONFDIR}/icinga2/features-available
 )
 
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${ICINGA2_RUNDIR}/icinga2/cmd\")")
-
 install(TARGETS livestatus RUNTIME DESTINATION ${CMAKE_INSTALL_SBINDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/icinga2)
 
 set(CPACK_NSIS_EXTRA_INSTALL_COMMANDS "${CPACK_NSIS_EXTRA_INSTALL_COMMANDS}" PARENT_SCOPE)
diff --git i/lib/perfdata/CMakeLists.txt w/lib/perfdata/CMakeLists.txt
index 08920dd..8bbe53c 100644
--- i/lib/perfdata/CMakeLists.txt
+++ w/lib/perfdata/CMakeLists.txt
@@ -72,7 +72,4 @@ install(
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/icinga2
 )
 
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/spool/icinga2/perfdata\")")
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/spool/icinga2/tmp\")")
-
 set(CPACK_NSIS_EXTRA_INSTALL_COMMANDS "${CPACK_NSIS_EXTRA_INSTALL_COMMANDS}" PARENT_SCOPE)
diff --git i/lib/remote/CMakeLists.txt w/lib/remote/CMakeLists.txt
index eb98673..72614b8 100644
--- i/lib/remote/CMakeLists.txt
+++ w/lib/remote/CMakeLists.txt
@@ -62,10 +62,3 @@ if(APPLE)
     LIBRARY DESTINATION ${CMAKE_INSTALL_BINDIR}/icinga-studio.app/Contents
   )
 endif()
-
-#install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/lib/icinga2/api\")")
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/lib/icinga2/api/log\")")
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/lib/icinga2/api/repository\")")
-install(CODE "file(MAKE_DIRECTORY \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/lib/icinga2/api/zones\")")
-
-
