diff --git a/meson_options.txt b/meson_options.txt
index 961ae2a76..692b84dfd 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -179,6 +179,9 @@ option('udev',
 option('udevrulesdir',
        type : 'string',
        description : 'Directory for udev rules (defaults to /lib/udev/rules.d)')
+option('pipewire_pulse_prefix',
+       type : 'string',
+       description: 'Install directory for the pipewire-pulse daemon')
 option('systemd-system-unit-dir',
        type : 'string',
        description : 'Directory for system systemd units (defaults to /usr/lib/systemd/system)')
diff --git a/src/daemon/systemd/user/meson.build b/src/daemon/systemd/user/meson.build
index d17f3794f..34afe4f1a 100644
--- a/src/daemon/systemd/user/meson.build
+++ b/src/daemon/systemd/user/meson.build
@@ -9,7 +9,7 @@ install_data(
 
 systemd_config = configuration_data()
 systemd_config.set('PW_BINARY', pipewire_bindir / 'pipewire')
-systemd_config.set('PW_PULSE_BINARY', pipewire_bindir / 'pipewire-pulse')
+systemd_config.set('PW_PULSE_BINARY', get_option('pipewire_pulse_prefix') / 'bin/pipewire-pulse')
 
 configure_file(input : 'pipewire.service.in',
                output : 'pipewire.service',
