diff --git a/meson.build b/meson.build
index ffee41b4..f3e4ec74 100644
--- a/meson.build
+++ b/meson.build
@@ -53,7 +53,7 @@ endif
 
 spa_plugindir = join_paths(pipewire_libdir, spa_name)
 
-alsadatadir = join_paths(pipewire_datadir, 'alsa-card-profile', 'mixer')
+alsadatadir = join_paths(pipewire_libdir, '..', 'share', 'alsa-card-profile', 'mixer')
 
 pipewire_headers_dir = join_paths(pipewire_name, 'pipewire')
 
