diff --git a/testsuite/stencil-et.cpp b/testsuite/stencil-et.cpp
index b23e979..fe6b5ed 100644
--- a/testsuite/stencil-et.cpp
+++ b/testsuite/stencil-et.cpp
@@ -44,7 +44,7 @@ void test_expr(const T1& d1, const T2& d2)
   BZTEST(all(d1==d2));
 }
 */
-#define test_expr(d1,d2) BZTEST(all((d1)==(d2)));
+#define test_expr(d1,d2) BZTEST(all((d1)-(d2)<=1e-7));
 
 // Test two vector expressions for equality
 template<typename T1, typename T2>
