diff --git a/sshuttle/client.py b/sshuttle/client.py
index cab5b1c..e89f8a6 100644
--- a/sshuttle/client.py
+++ b/sshuttle/client.py
@@ -192,7 +192,7 @@ class FirewallClient:
 
         self.auto_nets = []

-        argvbase = ([sys.executable, sys.argv[0]] +
+        argvbase = ([sys.argv[0]] +
                     ['-v'] * (helpers.verbose or 0) +
                     ['--method', method_name] +
                     ['--firewall'])
