diff --git a/lua/sniprun.lua b/lua/sniprun.lua
index c9b811f..459cf07 100644
--- a/lua/sniprun.lua
+++ b/lua/sniprun.lua
@@ -4,9 +4,7 @@ M.custom_highlight=false
 M.info_floatwin = {}
 
 -- See https://github.com/tjdevries/rofl.nvim/blob/632c10f2ec7c56882a3f7eda8849904bcac6e8af/lua/rofl.lua
-local binary_path = vim.fn.fnamemodify(
-  vim.api.nvim_get_runtime_file("lua/sniprun.lua", false)[1], ":h:h")
-  .. "/target/release/sniprun"
+local binary_path = "@sniprun_bin@/bin/sniprun"
 
 local sniprun_path = vim.fn.fnamemodify( vim.api.nvim_get_runtime_file("lua/sniprun.lua", false)[1], ":p:h") .. "/.."
 
@@ -145,7 +143,7 @@ function M.setup_highlights()
       highlight(group, styles)
     end
     vim.cmd('augroup END')
-  else 
+  else
     for group, styles in pairs(colors_table) do
       local gui = styles.gui and 'gui='..styles.gui or 'gui=NONE'
       local sp = styles.sp and 'guisp='..styles.sp or 'guisp=NONE'
@@ -354,7 +352,7 @@ function M.health()
   -- check if the log is recreated
   if pcall(M.ping) then health_ok("Sent a ping to the sniprun binary")
   else health_warn("Could not send a ping to the sniprun binary - is it present, executable and compatible with your CPU architecture?") end
-    
+
 
   os.execute("sleep 0.2")
   if not M.file_exists(path_log_file) and not M.file_exists(path_log_file_mac)  then health_error("sniprun binary incompatible or crash at start", {"Compile sniprun locally, with a clean reinstall and 'bash ./install.sh 1' as post-install command."})
diff --git a/ressources/init_repl.sh b/ressources/init_repl.sh
index eb51dbe..1382b5c 100755
--- a/ressources/init_repl.sh
+++ b/ressources/init_repl.sh
@@ -35,7 +35,7 @@ mkfifo $working_dir/$pipe
 touch $working_dir/$out
 sleep 36000 > $working_dir/$pipe &
 
-echo "/bin/cat " $working_dir/$pipe " | " $repl  > $working_dir/real_launcher.sh
+echo "cat " $working_dir/$pipe " | " $repl  > $working_dir/real_launcher.sh
 chmod +x $working_dir/real_launcher.sh
 
 echo $repl " process started at $(date +"%F %T")." >> $log
diff --git a/ressources/launcher_repl.sh b/ressources/launcher_repl.sh
index feaa91e..749c55e 100755
--- a/ressources/launcher_repl.sh
+++ b/ressources/launcher_repl.sh
@@ -1,2 +1,2 @@
 #!/bin/bash
-/bin/cat $1 > $2
+cat $1 > $2
