diff --git a/daemon/gdm-session.c.orig b/daemon/gdm-session.c
index b839fea..053d4ac 100644
--- a/daemon/gdm-session.c.orig
+++ b/daemon/gdm-session.c
@@ -344,12 +344,16 @@ get_system_session_dirs (GdmSession *self)
 #ifdef ENABLE_WAYLAND_SUPPORT
                 DATADIR "/wayland-sessions/",
 #endif
+                "/var/empty",
                 "/etc/X11/sessions/",
                 DMCONFDIR "/Sessions/",
                 DATADIR "/gdm/BuiltInSessions/",
                 DATADIR "/xsessions/",
                 NULL
         };
+        if (getenv("GDM_SESSIONS_DIR") != NULL) {
+                search_dirs[1] = getenv("GDM_SESSIONS_DIR");
+        };
 
 #ifdef ENABLE_WAYLAND_SUPPORT
         if (self->priv->ignore_wayland) {
