diff --git a/src/server/ConfigInterface.ts b/src/server/ConfigInterface.ts
index d23badd..1dd2b98 100644
--- a/src/server/ConfigInterface.ts
+++ b/src/server/ConfigInterface.ts
@@ -11,9 +11,10 @@ interface ConfigInterface {
     dbPath: string;
     dbInfoPath: string;
     mysql: {
-        host: string;
+        host?: string;
+        socketPath?: string;
         user: string;
-        password: string;
+        password?: string;
         database: string;
         connectTimeout: number;
         connectionLimit: number;
