Fix broken datadir parameter.

--- akonadi-1.3.85/server/src/storage/dbconfigmysql.cpp	2010-06-09 03:41:30.000000000 -0430
+++ akonadi-local-1.3.85/server/src/storage/dbconfigmysql.cpp	2010-08-11 00:21:20.547181479 -0430
@@ -250,11 +250,10 @@
   // synthesize the mysqld command
   QStringList arguments;
   arguments << QString::fromLatin1( "--defaults-file=%1/mysql.conf" ).arg( akDir );
+  arguments << QString::fromLatin1( "--datadir=%1/" ).arg( dataDir );
 #ifndef Q_WS_WIN
-  arguments << QString::fromLatin1( "--datadir" ) << QString::fromLatin1( "%1/" ).arg( dataDir );
   arguments << QString::fromLatin1( "--socket=%1/mysql.socket" ).arg( miscDir );
 #else
-  arguments << QString::fromLatin1( "--datadir=%1/" ).arg( dataDir );
   arguments << QString::fromLatin1( "--shared-memory" );
 #endif
 
