diff --git a/whisper/audio.py b/whisper/audio.py
index 4f5b6e0..bfe7924 100644
--- a/whisper/audio.py
+++ b/whisper/audio.py
@@ -44,7 +44,7 @@ def load_audio(file: str, sr: int = SAMPLE_RATE):
     # and resampling as necessary.  Requires the ffmpeg CLI in PATH.
     # fmt: off
     cmd = [
-        "ffmpeg",
+        "@ffmpeg@/bin/ffmpeg",
         "-nostdin",
         "-threads", "0",
         "-i", file,
