diff --git a/io.elementary.contracter.gnome-bluetooth.contract b/io.elementary.contracter.gnome-bluetooth.contract
index 3eb2334..ce0e5f5 100644
--- a/io.elementary.contracter.gnome-bluetooth.contract
+++ b/io.elementary.contracter.gnome-bluetooth.contract
@@ -3,5 +3,5 @@ Name=Send Files via Bluetooth
 Icon=bluetooth
 Description=Send files to device...
 MimeType=!inode;
-Exec=bluetooth-sendto %F
+Exec=@gnome_bluetooth@/bin/bluetooth-sendto %F
 Gettext-Domain=gnome-bluetooth2
