diff -r b58e759f84db lib/rq/jobrunner.rb
--- a/lib/rq/jobrunner.rb	Sun Sep 28 13:33:06 2008 +0200
+++ b/lib/rq/jobrunner.rb	Sun Sep 28 13:35:09 2008 +0200
@@ -85,7 +85,7 @@ unless defined? $__rq_jobrunner__
 
         command =
           if @sh_like 
-            sin = "0<#{ @stdin }" if @stdin
+            sin = "0<#{ @stdin }" if @stdin and File.exist?(@stdin)
             sout = "1>#{ @stdout }" if @stdout
             serr = "2>#{ @stderr }" if @stderr
             "( PATH=#{ path }:$PATH #{ command } ;) #{ sin } #{ sout } #{ serr }"
