diff --git a/macros/gmp-check.m4 b/macros/gmp-check.m4
index 72eba8c..25af64e 100644
--- a/macros/gmp-check.m4
+++ b/macros/gmp-check.m4
@@ -105,21 +105,6 @@ AC_DEFUN([GIV_CHECK_GMP], [
 		exit 1
 	])
 
-	AC_MSG_CHECKING([whether gmp version is at least $min_gmp_release])
-	AC_TRY_RUN(
-		[ 
-			#include <cstddef>
-			#include <gmp.h>
-			int main () {
-				return (__GNU_MP_RELEASE < $min_gmp_release);
-			}
-		],
-		[ AC_MSG_RESULT(yes)
-		],
-		[ AC_MSG_RESULT(no)
-		  AC_MSG_ERROR(your GMP is too old. GMP release >= $min_gmp_release needed)
-		  exit 1]
-	)
 	AC_LANG_POP([C++])
 	
 	AC_SUBST(GMP_CFLAGS)
