diff --git a/script/generate-licenses b/script/generate-licenses
index 51749a91e2..d3fd7ee351 100755
--- a/script/generate-licenses
+++ b/script/generate-licenses
@@ -16,28 +16,17 @@ cat assets/icons/LICENSES >> $OUTPUT_FILE
 
 echo -e "\n# ###### CODE LICENSES ######\n" >> $OUTPUT_FILE
 
-if ! cargo install --list | grep "cargo-about v$CARGO_ABOUT_VERSION" > /dev/null; then
-  echo "Installing cargo-about@$CARGO_ABOUT_VERSION..."
-  cargo install "cargo-about@$CARGO_ABOUT_VERSION"
-else
-  echo "cargo-about@$CARGO_ABOUT_VERSION is already installed."
-fi
 
 echo "Generating cargo licenses"
 
 stderr_file=$(mktemp)
 
 cargo about generate \
-    --fail \
     -c script/licenses/zed-licenses.toml \
     "${TEMPLATE_FILE}" \
     2> >(tee "$stderr_file") \
     >> $OUTPUT_FILE
 
-if cat "$stderr_file" | grep -v "\[WARN\]" > /dev/null; then
-    echo "Error: License check failed - warnings found" >&2
-    exit 1
-fi
 
 sed -i.bak 's/&quot;/"/g' $OUTPUT_FILE
 sed -i.bak 's/&#x27;/'\''/g' $OUTPUT_FILE # The ` '\'' ` thing ends the string, appends a single quote, and re-opens the string
