diff --git a/config/unicorn.conf.rb b/config/unicorn.conf.rb
index b8c549bc0f..a7d3592013 100644
--- a/config/unicorn.conf.rb
+++ b/config/unicorn.conf.rb
@@ -35,19 +35,7 @@ FileUtils.mkdir_p("#{discourse_path}/tmp/pids") if !File.exist?("#{discourse_pat
 # feel free to point this anywhere accessible on the filesystem
 pid(ENV["UNICORN_PID_PATH"] || "#{discourse_path}/tmp/pids/unicorn.pid")

-if ENV["RAILS_ENV"] == "production"
-  # By default, the Unicorn logger will write to stderr.
-  # Additionally, some applications/frameworks log to stderr or stdout,
-  # so prevent them from going to /dev/null when daemonized here:
-  stderr_path unicorn_stderr_path
-  stdout_path unicorn_stdout_path
-
-  # nuke workers after 30 seconds instead of 60 seconds (the default)
-  timeout 30
-else
-  # we want a longer timeout in dev cause first request can be really slow
-  timeout(ENV["UNICORN_TIMEOUT"] && ENV["UNICORN_TIMEOUT"].to_i || 60)
-end
+timeout(ENV["UNICORN_TIMEOUT"] && ENV["UNICORN_TIMEOUT"].to_i || 60)

 # important for Ruby 2.0
 preload_app true
