--- a/properties/nm-openvpn-editor.c
+++ b/properties/nm-openvpn-editor.c
@@ -775,8 +775,7 @@
 nm_find_openvpn (void)
 {
 	static const char *openvpn_binary_paths[] = {
-		"/usr/sbin/openvpn",
-		"/sbin/openvpn",
+		"@openvpn@/bin/openvpn",
 		NULL
 	};
 	const char  **openvpn_binary = openvpn_binary_paths;
--- a/src/nm-openvpn-service.c
+++ b/src/nm-openvpn-service.c
@@ -522,9 +522,7 @@
 openvpn_binary_find_exepath (void)
 {
 	static const char *paths[] = {
-		"/usr/sbin/openvpn",
-		"/sbin/openvpn",
-		"/usr/local/sbin/openvpn",
+		"@openvpn@/bin/openvpn",
 	};
 	int i;
 
@@ -2326,7 +2324,7 @@
 	_LOGD ("nm-openvpn-service (version " DIST_VERSION ") starting...");
 
 	if (   !g_file_test ("/sys/class/misc/tun", G_FILE_TEST_EXISTS)
-	    && (system ("/sbin/modprobe tun") == -1))
+	    && (system ("@kmod@/bin/modprobe tun") == -1))
 		return EXIT_FAILURE;
 
 	plugin = nm_openvpn_plugin_new (bus_name);
