diff --git a/source/creator/core/i18n.d b/source/creator/core/i18n.d
index 38761dd..f276ca1 100644
--- a/source/creator/core/i18n.d
+++ b/source/creator/core/i18n.d
@@ -132,7 +132,7 @@ void markDups(TLEntry[] entries) {
         // If prevEntry has same humanName as entry before prevEntry, or as this entry,
         // disambiguate with the source folder
         if (prevIsDup || entryIsDup) {
-            prevEntry.humanName ~= " (" ~ prevEntry.path ~ ")";
+            prevEntry.humanName ~= " (" ~ prevEntry.code ~ ")";
             prevEntry.humanNameC = prevEntry.humanName.toStringz;
         }
         prevIsDup = entryIsDup;
@@ -140,7 +140,7 @@ void markDups(TLEntry[] entries) {
     }
 
     if (prevIsDup) {
-        prevEntry.humanName ~= " (" ~ prevEntry.path ~ ")";
+        prevEntry.humanName ~= " (" ~ prevEntry.code ~ ")";
         prevEntry.humanNameC = prevEntry.humanName.toStringz;
     }
 }
