Already reported uptream

diff --git a/src/giv-data.h b/src/giv-data.h
index 64e7696..d34bfe4 100644
--- a/src/giv-data.h
+++ b/src/giv-data.h
@@ -88,7 +88,7 @@ typedef struct
 typedef struct
 {
   gint op;
-  union
+  struct
   {
     struct
     {
diff --git a/src/giv_types.h b/src/giv_types.h
index 02abebe..c3cfb78 100644
--- a/src/giv_types.h
+++ b/src/giv_types.h
@@ -11,13 +11,11 @@ typedef struct {
 
 typedef struct {
   gint op;
-  union {
-    struct {
-      gdouble x,y;
-    } point;
-    double arc_dev;
-    text_mark_t *text_object;
-  } data;
+  struct {
+    gdouble x,y;
+  } point;
+  double arc_dev;
+  text_mark_t *text_object;
 } point_t;
 
 typedef struct {
