diff --git a/src/ext/CMakeLists.txt b/src/ext/CMakeLists.txt
index 6436f1e..6c844cb 100644
--- a/src/ext/CMakeLists.txt
+++ b/src/ext/CMakeLists.txt
@@ -113,15 +113,10 @@ set (QOI_INCLUDE ${CMAKE_CURRENT_SOURCE_DIR}/qoi PARENT_SCOPE)
 ###########################################################################
 # glfw / glad
 
-set (GLFW_INCLUDE ${CMAKE_CURRENT_SOURCE_DIR}/glfw/include PARENT_SCOPE)
+find_package(glfw3 REQUIRED)
+set (GLFW_INCLUDE GLFW3_INCLUDE PARENT_SCOPE)
 set (GLAD_INCLUDE ${CMAKE_CURRENT_SOURCE_DIR}/glad/include PARENT_SCOPE)
 
-set (GLFW_LIBRARY_TYPE STATIC CACHE STRING "" FORCE)
-set (GLFW_BUILD_DOCS OFF CACHE BOOL "" FORCE)
-set (GLFW_BUILD_TESTS OFF CACHE BOOL "" FORCE)
-set (GLFW_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
-
-add_subdirectory (glfw)
 add_subdirectory (glad)
 
 set_property (TARGET glfw PROPERTY FOLDER "ext")
