diff --git c/Sources/ptImage.cpp i/Sources/ptImage.cpp
index 9c95093..623c157 100755
--- c/Sources/ptImage.cpp
+++ i/Sources/ptImage.cpp
@@ -5291,7 +5291,7 @@ ptImage* ptImage::Box(const uint16_t MaxRadius, float* Mask) {
         NewRow = NewRow < 0? -NewRow : NewRow > Height1? Height1_2-NewRow : NewRow ;
         NewRow *= m_Width;
         for(j = -IntRadius; j <= IntRadius; j++) {
-          if (Dist[abs(i)][abs(j)] < Radius) {
+          if (Dist[int16_t(abs(i))][int16_t(abs(j))] < Radius) {
             NewCol = Col+j;
             NewCol = NewCol < 0? -NewCol : NewCol > Width1? Width1_2-NewCol : NewCol ;
 
