From 9ba51df4258d0dc2fe72d4f621d29073eeadc011 Mon Sep 17 00:00:00 2001
From: David McFarland <corngood@gmail.com>
Date: Sun, 11 Aug 2024 00:03:36 -0300
Subject: [PATCH 2/2] disable parallel compile

---
 nukebuild/Build.cs | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/nukebuild/Build.cs b/nukebuild/Build.cs
index e3d5139bf..ce50db574 100644
--- a/nukebuild/Build.cs
+++ b/nukebuild/Build.cs
@@ -136,12 +136,20 @@ DotNetConfigHelper ApplySettingCore(DotNetConfigHelper c)
             ProcessTasks.StartProcess("xcodebuild", args).AssertZeroExitCode();
         });
 
+    [Serializable]
+    public class SerialBuildSettings : DotNetBuildSettings
+    {
+        protected override Arguments ConfigureProcessArguments(Arguments arguments) =>
+            base.ConfigureProcessArguments(arguments)
+                .Add("-m:1");
+    }
+
     Target Compile => _ => _
         .DependsOn(Clean, CompileNative)
         .DependsOn(CompileHtmlPreviewer)
         .Executes(() =>
         {
-            DotNetBuild(c => ApplySetting(c)
+            DotNetBuild(ApplySetting(new SerialBuildSettings())
                 .SetProjectFile(Parameters.MSBuildSolution)
             );
         });
-- 
2.42.2

