diff -Naur pakcs-1.11.3-upstream/Makefile pakcs-1.11.3/Makefile
--- pakcs-1.11.3-upstream/Makefile	2014-01-31 09:10:03.000000000 -0430
+++ pakcs-1.11.3/Makefile	2014-02-03 20:27:41.558334480 -0430
@@ -66,7 +66,6 @@
 #
 .PHONY: install
 install: installscripts
-	$(MAKE) frontend
 	# pre-compile all libraries:
 	@cd lib && $(MAKE) fcy
 	# install the Curry2Prolog compiler as a saved system:
@@ -105,10 +104,6 @@
 # compile the tools:
 .PHONY: tools
 tools:
-	# compile the Curry Port Name Server demon:
-	@if [ -r bin/pakcs ] ; then cd cpns       && $(MAKE) ; fi
-	# compile the event handler demon for dynamic web pages:
-	@if [ -r bin/pakcs ] ; then cd www        && $(MAKE) ; fi
 	@if [ -r bin/pakcs ] ; then cd currytools && $(MAKE) ; fi
 	@if [ -r bin/pakcs ] ; then cd tools      && $(MAKE) ; fi
 
diff -Naur pakcs-1.11.3-upstream/scripts/pakcs.sh pakcs-1.11.3/scripts/pakcs.sh
--- pakcs-1.11.3-upstream/scripts/pakcs.sh	2014-01-31 09:04:19.000000000 -0430
+++ pakcs-1.11.3/scripts/pakcs.sh	2014-02-03 20:20:40.775350116 -0430
@@ -16,7 +16,7 @@
 # use readline wrapper rlwrap if it is installed and we have tty as stdin:
 USERLWRAP=no
 if tty -s ; then
-  RLWRAP=`which rlwrap`
+  RLWRAP=`type -P rlwrap`
   if [ -x "$RLWRAP" ] ; then
     USERLWRAP=yes
   fi
@@ -29,7 +29,7 @@
 done
 
 if [ $USERLWRAP = yes ] ; then
-  exec rlwrap -c -f "$PAKCSHOME/tools/rlwrap" "$REPL" ${1+"$@"}
+  exec rlwrap -a -c -f "$PAKCSHOME/tools/rlwrap" "$REPL" ${1+"$@"}
 else
   exec "$REPL" ${1+"$@"}
 fi
