--- www/Makefile.orig	2016-10-10 21:04:36.000000000 +0300
+++ pakcs-1.14.0/www/Makefile	2016-10-10 21:07:56.000000000 +0300
@@ -6,7 +6,7 @@ all: submitform Registry
 submitform: SubmitForm.curry $(LIBDIR)/HtmlCgi.curry \
 	    $(LIBDIR)/NamedSocket.curry $(LIBDIR)/CPNS.curry
 	$(REPL) $(REPL_OPTS) :load SubmitForm :save :q
-	mv SubmitForm submitform
+	mv SubmitForm submitform.orig && mv submitform.orig submitform
 
 Registry: Registry.curry $(LIBDIR)/HtmlCgi.curry
 	$(REPL) $(REPL_OPTS) :load Registry :save :q
--- currytools/erd2curry/Makefile.orig	2016-10-10 21:13:49.000000000 +0300
+++ pakcs-1.14.0/currytools/erd2curry/Makefile	2016-10-10 21:21:14.000000000 +0300
@@ -32,4 +32,4 @@ uninstall: clean
 erd2curry: $(DEPS)
 	# create saved state for top-level function "main":
 	$(REPL) $(REPL_OPTS) :load ERD2Curry :save "main \"$(CURDIR)\"" :q
-	mv ERD2Curry $@
+	mv ERD2Curry $@.orig && mv $@.orig $@
