--- a/i18npool/qa/cppunit/test_breakiterator.cxx
+++ b/i18npool/qa/cppunit/test_breakiterator.cxx
@@ -35,7 +35,7 @@ public:
     void testWeak();
     void testAsian();
     void testThai();
-#if (U_ICU_VERSION_MAJOR_NUM > 51)
+#if (U_ICU_VERSION_MAJOR_NUM > 51 && U_ICU_VERSION_MAJOR_NUM < 70)
     void testLao();
 #ifdef TODO
     void testNorthernThai();
@@ -52,7 +52,7 @@ public:
     CPPUNIT_TEST(testWeak);
     CPPUNIT_TEST(testAsian);
     CPPUNIT_TEST(testThai);
-#if (U_ICU_VERSION_MAJOR_NUM > 51)
+#if (U_ICU_VERSION_MAJOR_NUM > 51 && U_ICU_VERSION_MAJOR_NUM < 70)
     CPPUNIT_TEST(testLao);
 #ifdef TODO
     CPPUNIT_TEST(testKhmer);
@@ -843,7 +843,7 @@ void TestBreakIterator::testAsian()
     }
 }
 
-#if (U_ICU_VERSION_MAJOR_NUM > 51)
+#if (U_ICU_VERSION_MAJOR_NUM > 51 && U_ICU_VERSION_MAJOR_NUM < 70)
 //A test to ensure that our Lao word boundary detection is useful
 void TestBreakIterator::testLao()
 {
