diff --git a/library/src/device/CMakeLists.txt b/library/src/device/CMakeLists.txt
index 73a8ec9..9bfd4b8 100644
--- a/library/src/device/CMakeLists.txt
+++ b/library/src/device/CMakeLists.txt
@@ -255,4 +255,10 @@ foreach( sub ${AMDGPU_TARGETS} )
   if( NOT BUILD_SHARED_LIBS )
     set_target_properties( rocfft-device-${sub} PROPERTIES PREFIX "lib" )
   endif( )
+
+  rocm_install_targets(
+    TARGETS
+    rocfft-device-${sub}
+    COMPONENT device
+  )
 endforeach()
