diff --git i/GPXLab/main.cpp w/GPXLab/main.cpp
index b12d2dd..58d37c5 100644
--- i/GPXLab/main.cpp
+++ w/GPXLab/main.cpp
@@ -19,10 +19,10 @@ int main(int argc, char *argv[])
     app.installTranslator(&gpxlab);
 
     QTranslator qt;
-#if defined(Q_OS_WIN32) || defined(Q_OS_MAC)
+#if defined(Q_OS_WIN32)
     qt.load(QLocale::system(), "qt", "_", TRANSLATIONS_DIR);
 #else
-    qt.load(QLocale::system(), "qt", "_", QLibraryInfo::location(QLibraryInfo::TranslationsPath));
+    qt.load(QLocale::system(), "qt", "_", QLatin1String("@qttranslations@/translations"));
 #endif
     app.installTranslator(&qt);
 
