Patch from the fedora repository.

--- sexplib-7.0.5.old/myocamlbuild.ml	2012-03-20 16:41:54.000000000 +0000
+++ sexplib-7.0.5/myocamlbuild.ml	2012-07-30 11:38:21.829869635 +0100
@@ -438,7 +438,13 @@
                    BaseEnvLight.var_choose cond_specs env
                  in
                    flag tags & spec)
-              t.flags
+              t.flags;
+
+            flag ["ocaml"; "compile"; "byte"; "use_toploop"] (S[A"-I"; A"+compiler-libs"]);
+            flag ["ocaml"; "link"; "byte"; "use_toploop"] (S[A"-I"; A"+compiler-libs"; A"+compiler-libs/ocamltoplevel.cma"]);
+            flag ["ocaml"; "compile"; "native"; "use_toploop"] (S[A"-I"; A"+compiler-libs"]);
+            flag ["ocaml"; "link"; "native"; "use_toploop"] (S[A"-I"; A"+compiler-libs"; A"+compiler-libs/ocamltoplevel.cmxa"]);
+
         | _ -> 
             ()
   
--- sexplib-7.0.5.old/_tags	2012-03-20 16:41:54.000000000 +0000
+++ sexplib-7.0.5/_tags	2012-07-30 11:37:01.749255839 +0100
@@ -49,6 +49,7 @@
 <lib_test/*.ml{,i}>: pkg_camlp4.extend
 # Library sexplib_top
 "top": include
+<top/*.ml{,i}>: use_toploop
 # Executable sexp_test
 "lib_test/sexp_test.byte": use_sexplib
 "lib_test/sexp_test.byte": pkg_unix
