diff --git a/src/arch.rs b/src/arch.rs
index a998697..45ec2cb 100644
--- a/src/arch.rs
+++ b/src/arch.rs
@@ -140,7 +140,7 @@ cfg_match! {
     target_arch = "x86_64" => {
         pub const ENTRY_TRAMPOLINE: Option<unsafe extern "C" fn() -> !> = Some(entry_trampoline);
 
-        #[naked]
+        #[unsafe(naked)]
         unsafe extern "C" fn entry_trampoline() -> ! {
             core::arch::naked_asm!(
                 "lea r10, [rip + {context}]",
@@ -159,7 +159,7 @@ cfg_match! {
     target_arch = "aarch64" => {
         pub const ENTRY_TRAMPOLINE: Option<unsafe extern "C" fn() -> !> = Some(entry_trampoline);
 
-        #[naked]
+        #[unsafe(naked)]
         unsafe extern "C" fn entry_trampoline() -> ! {
             core::arch::naked_asm!(
                 "adrp x8, {context}",
diff --git a/src/sys.rs b/src/sys.rs
index 109d809..bf085d9 100644
--- a/src/sys.rs
+++ b/src/sys.rs
@@ -181,6 +181,5 @@ pub fn new_slice_leak(size: usize) -> Option<&'static mut [u8]> {
 
 #[cfg(not(test))]
 #[lang = "eh_personality"]
-#[no_mangle]
 pub extern fn rust_eh_personality() {
 }
