diff --git a/src/core/device.c b/src/core/device.c
index 5307341..7bbe99a 100644
--- a/src/core/device.c
+++ b/src/core/device.c
@@ -255,8 +255,7 @@ static int device_update_unit(Manager *m, struct udev_device *dev, const char *p
                         if (!is_path(alias))
                                 log_warning("SYSTEMD_ALIAS for %s is not a path, ignoring: %s", sysfs, alias);
                         else {
-                                if ((r = device_add_escaped_name(u, alias)) < 0)
-                                        goto fail;
+                                device_update_unit(m, dev, alias, false);
                         }
                 }
 
