diff -ruN systemd-203/units/getty@.service.m4 systemd-203-patched/units/getty@.service.m4
--- systemd-203/units/getty@.service.m4	2013-01-07 22:50:49.083315575 +0100
+++ systemd-203-patched/units/getty@.service.m4	2014-03-18 09:54:40.002476232 +0100
@@ -23,7 +23,9 @@
 # On systems without virtual consoles, don't start any getty. (Note
 # that serial gettys are covered by serial-getty@.service, not this
 # unit
-ConditionPathExists=/dev/tty0
+ConditionPathExists=|/dev/tty0
+ConditionVirtualization=|lxc
+ConditionVirtualization=|lxc-libvirt
 
 [Service]
 # the VT is cleared by TTYVTDisallocate
