From 172bf8a7d8e232d31a85ef61d21ee54b6c2eab7c Mon Sep 17 00:00:00 2001
From: fliiiix <hi@l33t.name>
Date: Tue, 9 Jun 2026 15:33:29 +0200
Subject: [PATCH 3/3] Fix cppcheck warning has no initializer

cppcheck 2.21.0
---
 src/service.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/service.cpp b/src/service.cpp
index ced370c..251df5b 100644
--- a/src/service.cpp
+++ b/src/service.cpp
@@ -50,9 +50,9 @@ extern "C"
 
     typedef struct
     {
-       guint nTempLow;
-       guint nTempHigh;
-       const gchar *sName;
+       guint nTempLow{};
+       guint nTempHigh{};
+       const gchar *sName = nullptr;
     } TempProfile;
 
     TempProfile m_lTempProfiles[] =
-- 
2.43.0

