From 8d9ab4b6ed24a97e8af0cc338a52aacdcf438b8c Mon Sep 17 00:00:00 2001
From: Pavel Sobolev <paveloom@riseup.net>
Date: Tue, 21 Nov 2023 20:53:33 +0300
Subject: [PATCH] Force-unwrap file handles.

---
 Sources/TSCBasic/FileSystem.swift | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Sources/TSCBasic/FileSystem.swift b/Sources/TSCBasic/FileSystem.swift
index 3a63bdf..a1f3d9d 100644
--- a/Sources/TSCBasic/FileSystem.swift
+++ b/Sources/TSCBasic/FileSystem.swift
@@ -425,7 +425,7 @@ private class LocalFileSystem: FileSystem {
         if fp == nil {
             throw FileSystemError(errno: errno, path)
         }
-        defer { fclose(fp) }
+        defer { fclose(fp!) }

         // Read the data one block at a time.
         let data = BufferedOutputByteStream()
@@ -455,7 +455,7 @@ private class LocalFileSystem: FileSystem {
         if fp == nil {
             throw FileSystemError(errno: errno, path)
         }
-        defer { fclose(fp) }
+        defer { fclose(fp!) }

         // Write the data in one chunk.
         var contents = bytes.contents
--
2.42.0
