--- a/libbuild2/cc/pkgconfig-libpkgconf.cxx
+++ b/libbuild2/cc/pkgconfig-libpkgconf.cxx
@@ -84,7 +84,7 @@ namespace build2
     static bool
     pkgconf_error_handler (const char* msg,
                            const pkgconf_client_t*,
-                           const void*)
+                           void*)
     {
       error << runtime_error (msg); // Sanitize the message (trailing dot).
       return true;
