--- a/libbuild2/buildfile
+++ b/libbuild2/buildfile
@@ -83,9 +83,13 @@ config/cxx{host-config}: config/in{host-config}
   # want it).
   #
   build2_config = $regex.replace_lines(                                          \
+    $regex.replace_lines(                                                        \
     $config.save(),                                                              \
     '^( *(#|(config\.(test[. ]|dist\.|install\.chroot|config\.hermetic))).*|)$', \
     [null],                                                                      \
+    return_lines),                                                               \
+    '^.*'$getenv(NIX_STORE)'/[a-z0-9]{32}-.*$',                                  \
+    [null],                                                                      \
     return_lines)
 
   # Also preserve config.version.
