diff --git a/meson.build b/meson.build
index b25f9ef..7975f85 100644
--- a/meson.build
+++ b/meson.build
@@ -40,8 +40,8 @@ if dataroot_dir == ''
     dataroot_dir = datadir
 endif
 
-installed_tests_dir = prefix / libexecdir / 'installed-tests' / meson.project_name()
-installed_tests_data_dir = prefix / datadir / 'installed-tests' / meson.project_name()
+installed_tests_dir = get_option('installed_test_prefix') / 'libexec' / 'installed-tests' / meson.project_name()
+installed_tests_data_dir = get_option('installed_test_prefix') / 'share' / 'installed-tests' / meson.project_name()
 docs_dir = datadir / 'doc' / meson.project_name()
 
 summary({
diff --git a/meson_options.txt b/meson_options.txt
index fccada3..ca87600 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -38,6 +38,10 @@ option('installed-tests',
         type: 'boolean',
         value: false,
         description: 'Enable installation of some test cases')
+option('installed_test_prefix',
+        type: 'string',
+        value: '',
+        description: 'Prefix for installed tests')
 option('pytest',
         type: 'feature',
         value: 'auto',
