diff --git a/meson.build b/meson.build
index 26da8a4..d4f95ec 100644
--- a/meson.build
+++ b/meson.build
@@ -43,8 +43,8 @@ if dataroot_dir == ''
     dataroot_dir = datadir
 endif
 
-installed_tests_dir = prefix / libexecdir / 'installed-tests' / meson.project_name()
-installed_tests_data_dir = prefix / datadir / 'installed-tests' / meson.project_name()
+installed_tests_dir = get_option('installed_test_prefix') / 'libexec' / 'installed-tests' / meson.project_name()
+installed_tests_data_dir = get_option('installed_test_prefix') / 'share' / 'installed-tests' / meson.project_name()
 docs_dir = datadir / 'doc' / meson.project_name()
 
 summary({
diff --git a/meson_options.txt b/meson_options.txt
index 64d9bd3..34675f6 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -42,6 +42,10 @@ option('installed-tests',
         type: 'boolean',
         value: false,
         description: 'Enable installation of some test cases')
+option('installed_test_prefix',
+        type: 'string',
+        value: '',
+        description: 'Prefix for installed tests')
 option('pytest',
         type: 'feature',
         value: 'enabled',
