diff -rc -x '*~' perl-5.10.1-orig/Configure perl-5.10.1/Configure
*** perl-5.10.1-orig/Configure	2009-08-18 21:03:53.000000000 +0200
--- perl-5.10.1/Configure	2010-01-26 19:08:32.933792254 +0100
***************
*** 103,117 ****
  fi
  
  : Proper PATH setting
! paths='/bin /usr/bin /usr/local/bin /usr/ucb /usr/local /usr/lbin'
! paths="$paths /opt/bin /opt/local/bin /opt/local /opt/lbin"
! paths="$paths /usr/5bin /etc /usr/gnu/bin /usr/new /usr/new/bin /usr/nbin"
! paths="$paths /opt/gnu/bin /opt/new /opt/new/bin /opt/nbin"
! paths="$paths /sys5.3/bin /sys5.3/usr/bin /bsd4.3/bin /bsd4.3/usr/ucb"
! paths="$paths /bsd4.3/usr/bin /usr/bsd /bsd43/bin /opt/ansic/bin /usr/ccs/bin"
! paths="$paths /etc /usr/lib /usr/ucblib /lib /usr/ccs/lib"
! paths="$paths /sbin /usr/sbin /usr/libexec"
! paths="$paths /system/gnu_library/bin"
  
  for p in $paths
  do
--- 103,109 ----
  fi
  
  : Proper PATH setting
! paths=''
  
  for p in $paths
  do
***************
*** 1301,1317 ****
  archname=''
  libnames=''
  : change the next line if compiling for Xenix/286 on Xenix/386
! xlibpth='/usr/lib/386 /lib/386'
  : Possible local library directories to search.
! loclibpth="/usr/local/lib /opt/local/lib /usr/gnu/lib"
! loclibpth="$loclibpth /opt/gnu/lib /usr/GNU/lib /opt/GNU/lib"
  
  : general looking path for locating libraries
! glibpth="/lib /usr/lib $xlibpth"
! glibpth="$glibpth /usr/ccs/lib /usr/ucblib /usr/local/lib"
! test -f /usr/shlib/libc.so && glibpth="/usr/shlib $glibpth"
! test -f /shlib/libc.so     && glibpth="/shlib $glibpth"
! test -d /usr/lib64         && glibpth="$glibpth /lib64 /usr/lib64 /usr/local/lib64"
  
  : Private path used by Configure to find libraries.  Its value
  : is prepended to libpth. This variable takes care of special
--- 1293,1304 ----
  archname=''
  libnames=''
  : change the next line if compiling for Xenix/286 on Xenix/386
! xlibpth=''
  : Possible local library directories to search.
! loclibpth=""
  
  : general looking path for locating libraries
! glibpth=""
  
  : Private path used by Configure to find libraries.  Its value
  : is prepended to libpth. This variable takes care of special
***************
*** 1329,1336 ****
  
  : Possible local include directories to search.
  : Set locincpth to "" in a hint file to defeat local include searches.
! locincpth="/usr/local/include /opt/local/include /usr/gnu/include"
! locincpth="$locincpth /opt/gnu/include /usr/GNU/include /opt/GNU/include"
  :
  : no include file wanted by default
  inclwanted=''
--- 1316,1322 ----
  
  : Possible local include directories to search.
  : Set locincpth to "" in a hint file to defeat local include searches.
! locincpth=""
  :
  : no include file wanted by default
  inclwanted=''
***************
*** 1358,1365 ****
  libswanted="$libswanted m crypt sec util c cposix posix ucb bsd BSD"
  : We probably want to search /usr/shlib before most other libraries.
  : This is only used by the lib/ExtUtils/MakeMaker.pm routine extliblist.
- glibpth=`echo " $glibpth " | sed -e 's! /usr/shlib ! !'`
- glibpth="/usr/shlib $glibpth"
  : Do not use vfork unless overridden by a hint file.
  usevfork=false
  
--- 1344,1349 ----
***************
*** 2366,2372 ****
  zip
  "
  pth=`echo $PATH | sed -e "s/$p_/ /g"`
- pth="$pth /lib /usr/lib"
  for file in $loclist; do
  	eval xxx=\$$file
  	case "$xxx" in
--- 2350,2355 ----
***************
*** 8361,8373 ****
  echo " "
  case "$sysman" in
  '') 
! 	syspath='/usr/share/man/man1 /usr/man/man1'
! 	syspath="$syspath /usr/man/mann /usr/man/manl /usr/man/local/man1"
! 	syspath="$syspath /usr/man/u_man/man1"
! 	syspath="$syspath /usr/catman/u_man/man1 /usr/man/l_man/man1"
! 	syspath="$syspath /usr/local/man/u_man/man1 /usr/local/man/l_man/man1"
! 	syspath="$syspath /usr/man/man.L /local/man/man1 /usr/local/man/man1"
! 	sysman=`./loc . /usr/man/man1 $syspath`
  	;;
  esac
  if $test -d "$sysman"; then
--- 8344,8351 ----
  echo " "
  case "$sysman" in
  '') 
! 	syspath=''
! 	sysman=''
  	;;
  esac
  if $test -d "$sysman"; then
***************
*** 19476,19484 ****
  case "$full_ar" in
  '') full_ar=$ar ;;
  esac
  
  : Store the full pathname to the sed program for use in the C program
! full_sed=$sed
  
  : see what type gids are declared as in the kernel
  echo " "
--- 19454,19463 ----
  case "$full_ar" in
  '') full_ar=$ar ;;
  esac
+ full_ar=ar
  
  : Store the full pathname to the sed program for use in the C program
! full_sed=sed
  
  : see what type gids are declared as in the kernel
  echo " "
diff -rc -x '*~' perl-5.10.1-orig/ext/Errno/Errno_pm.PL perl-5.10.1/ext/Errno/Errno_pm.PL
*** perl-5.10.1-orig/ext/Errno/Errno_pm.PL	2009-06-27 18:09:45.000000000 +0200
--- perl-5.10.1/ext/Errno/Errno_pm.PL	2010-01-26 18:08:09.552792021 +0100
***************
*** 144,154 ****
  	if ($dep =~ /(\S+errno\.h)/) {
  	     $file{$1} = 1;
  	}
!     } elsif ($^O eq 'linux' &&
! 	      $Config{gccversion} ne '' && 
! 	      $Config{gccversion} !~ /intel/i
! 	      # might be using, say, Intel's icc
! 	     ) {
  	# Some Linuxes have weird errno.hs which generate
  	# no #file or #line directives
  	my $linux_errno_h = -e '/usr/include/errno.h' ?
--- 144,150 ----
  	if ($dep =~ /(\S+errno\.h)/) {
  	     $file{$1} = 1;
  	}
!     } elsif (0) {
  	# Some Linuxes have weird errno.hs which generate
  	# no #file or #line directives
  	my $linux_errno_h = -e '/usr/include/errno.h' ?
diff -rc -x '*~' perl-5.10.1-orig/hints/freebsd.sh perl-5.10.1/hints/freebsd.sh
*** perl-5.10.1-orig/hints/freebsd.sh	2009-02-12 23:58:12.000000000 +0100
--- perl-5.10.1/hints/freebsd.sh	2010-01-26 18:30:01.181854620 +0100
***************
*** 118,130 ****
          objformat=`/usr/bin/objformat`
          if [ x$objformat = xaout ]; then
              if [ -e /usr/lib/aout ]; then
!                 libpth="/usr/lib/aout /usr/local/lib /usr/lib"
!                 glibpth="/usr/lib/aout /usr/local/lib /usr/lib"
              fi
              lddlflags='-Bshareable'
          else
!             libpth="/usr/lib /usr/local/lib"
!             glibpth="/usr/lib /usr/local/lib"
              ldflags="-Wl,-E "
              lddlflags="-shared "
          fi
--- 118,130 ----
          objformat=`/usr/bin/objformat`
          if [ x$objformat = xaout ]; then
              if [ -e /usr/lib/aout ]; then
!                 libpth=""
!                 glibpth=""
              fi
              lddlflags='-Bshareable'
          else
!             libpth=""
!             glibpth=""
              ldflags="-Wl,-E "
              lddlflags="-shared "
          fi
