--- Source/WebKit/WebProcess/gtk/WebProcessMainGtk.cpp.orig	2020-07-21 13:09:21.000000000 +0200
+++ Source/WebKit/WebProcess/gtk/WebProcessMainGtk.cpp	2020-07-21 15:27:56.000000000 +0200
@@ -33,6 +33,7 @@
 #include <gtk/gtk.h>
 #include <libintl.h>
 #include <libsoup/soup.h>
+#include "NSApplicationActivationPolicy.h"
 
 #if PLATFORM(X11)
 #include <X11/Xlib.h>
@@ -57,6 +58,7 @@
 
         bindtextdomain(GETTEXT_PACKAGE, LOCALEDIR);
         bind_textdomain_codeset(GETTEXT_PACKAGE, "UTF-8");
+        SetActivationPolicyProhibited ();
 
         return true;
     }
