diff -rc hdf5-1.8.5/configure hdf5-1.8.5-new/configure
*** hdf5-1.8.5/configure	2010-06-04 20:26:04.000000000 +0200
--- hdf5-1.8.5-new/configure	2010-08-02 10:30:26.000000000 +0200
***************
*** 30587,30598 ****
    sed 's/#define /#define H5_/' <src/H5config.h |\
      sed 's/#undef /#undef H5_/' >pubconf
    if test ! -f src/H5pubconf.h; then
!     /bin/mv -f pubconf src/H5pubconf.h
    elif (diff pubconf src/H5pubconf.h >/dev/null); then
      rm -f pubconf
      echo "src/H5pubconf.h is unchanged"
    else
!     /bin/mv -f pubconf src/H5pubconf.h
    fi
    echo "Post process src/libhdf5.settings"
    sed '/^#/d' < src/libhdf5.settings > libhdf5.settings.TMP
--- 30587,30598 ----
    sed 's/#define /#define H5_/' <src/H5config.h |\
      sed 's/#undef /#undef H5_/' >pubconf
    if test ! -f src/H5pubconf.h; then
!     mv -f pubconf src/H5pubconf.h
    elif (diff pubconf src/H5pubconf.h >/dev/null); then
      rm -f pubconf
      echo "src/H5pubconf.h is unchanged"
    else
!     mv -f pubconf src/H5pubconf.h
    fi
    echo "Post process src/libhdf5.settings"
    sed '/^#/d' < src/libhdf5.settings > libhdf5.settings.TMP
