diff --git a/src/libhdf5.settings.in b/src/libhdf5.settings.in
index a4d4af6..70f1909 100644
--- a/src/libhdf5.settings.in
+++ b/src/libhdf5.settings.in
@@ -4,10 +4,7 @@
 General Information:
 -------------------
                    HDF5 Version: @H5_VERSION@
-                  Configured on: @CONFIG_DATE@
-                  Configured by: @CONFIG_USER@
                     Host system: @host_cpu@-@host_vendor@-@host_os@
-              Uname information: @UNAME_INFO@
                        Byte sex: @BYTESEX@
              Installation point: @prefix@
 
