diff --git a/tests/test_lbfgs.py b/tests/test_lbfgs.py
index 1af91ef..65e2da9 100644
--- a/tests/test_lbfgs.py
+++ b/tests/test_lbfgs.py
@@ -28,16 +28,16 @@ class TestOWLQN:
 
     def test_owl_line_search_default(self):
         def f(x, g, *args):
-            g[0] = 2 * x
-            return x ** 2
+            g[0] = 2 * x[0]
+            return x[0] ** 2
 
         with pytest.warns(UserWarning, match="OWL-QN"):
             xmin = fmin_lbfgs(f, 100., orthantwise_c=1)
     
     def test_owl_line_search_warning_explicit(self):
         def f(x, g, *args):
-            g[0] = 2 * x
-            return x ** 2
+            g[0] = 2 * x[0]
+            return x[0] ** 2
 
         with pytest.warns(UserWarning, match="OWL-QN"):
             xmin = fmin_lbfgs(f, 100., orthantwise_c=1, line_search='default')
