diff --git a/makedefs b/makedefs
index b8b98c8..94443c0 100644
--- a/makedefs
+++ b/makedefs
@@ -341,20 +341,20 @@ EOF
 		esac
 		;;
     Linux.3*)	SYSTYPE=LINUX3
-		if [ -f /usr/include/db.h ]
-		then
-		    : we are all set
-		elif [ -f /usr/include/db/db.h ]
-		then
-		    CCARGS="$CCARGS -I/usr/include/db"
-		else
-		    # On a properly installed system, Postfix builds
-		    # by including <db.h> and by linking with -ldb
-		    echo "No <db.h> include file found." 1>&2
-		    echo "Install the appropriate db*-devel package first." 1>&2
-		    echo "See the RELEASE_NOTES file for more information." 1>&2
-		    exit 1
-		fi
+		#if [ -f /usr/include/db.h ]
+		#then
+		    #: we are all set
+		#elif [ -f /usr/include/db/db.h ]
+		#then
+		    #CCARGS="$CCARGS -I/usr/include/db"
+		#else
+		    ## On a properly installed system, Postfix builds
+		    ## by including <db.h> and by linking with -ldb
+		    #echo "No <db.h> include file found." 1>&2
+		    #echo "Install the appropriate db*-devel package first." 1>&2
+		    #echo "See the RELEASE_NOTES file for more information." 1>&2
+		    #exit 1
+		#fi
 		SYSLIBS="-ldb"
 		for name in nsl resolv
 		do
