Submodule deps/freetype/zlib contains modified content
diff --git a/deps/freetype/zlib/zutil.h b/deps/freetype/zlib/zutil.h
index b079ea6..4e94f2f 100644
--- a/deps/freetype/zlib/zutil.h
+++ b/deps/freetype/zlib/zutil.h
@@ -130,17 +130,8 @@ extern z_const char * const z_errmsg[10]; /* indexed by 2-zlib_error */
 #  endif
 #endif
 
-#if defined(MACOS) || defined(TARGET_OS_MAC)
+#if defined(MACOS)
 #  define OS_CODE  7
-#  ifndef Z_SOLO
-#    if defined(__MWERKS__) && __dest_os != __be_os && __dest_os != __win32_os
-#      include <unix.h> /* for fdopen */
-#    else
-#      ifndef fdopen
-#        define fdopen(fd,mode) NULL /* No fdopen() */
-#      endif
-#    endif
-#  endif
 #endif
 
 #ifdef __acorn
