diff --git a/src/ipepresenter/Makefile b/src/ipepresenter/Makefile
index ae3664e..f4f74ff 100644
--- a/src/ipepresenter/Makefile
+++ b/src/ipepresenter/Makefile
@@ -8,7 +8,7 @@ include ../common.mak
 
 TARGET = $(call exe_target,ipepresenter)
 
-CXXFLAGS += -I../include -I../ipecanvas -I../ipecairo \
+CPPFLAGS += -I../include -I../ipecanvas -I../ipecairo \
 	$(UI_CFLAGS) $(CAIRO_CFLAGS) $(ZLIB_CFLAGS)
 LIBS += -L$(buildlib) -lipecanvas -lipecairo -lipe \
 	$(UI_LIBS) $(CAIRO_LIBS) $(ZLIB_LIBS)
