diff --git a/conftest.py b/conftest.py
index 4426e02..1f84096 100644
--- a/conftest.py
+++ b/conftest.py
@@ -73,17 +73,12 @@ def mysql_db_user():
 @pytest.fixture
 def postgresql_dsn(postgresql_db_user, postgresql_db_password, postgresql_db_host,
                    db_name):
-    return 'postgresql://{}:{}@{}/{}'.format(
-        postgresql_db_user,
-        postgresql_db_password,
-        postgresql_db_host,
-        db_name
-    )
+    pytest.skip()
 
 
 @pytest.fixture
 def mysql_dsn(mysql_db_user, db_name):
-    return f'mysql+pymysql://{mysql_db_user}@localhost/{db_name}'
+    pytest.skip()
 
 
 @pytest.fixture
@@ -121,8 +116,7 @@ def mssql_db_driver():
 
 @pytest.fixture
 def mssql_dsn(mssql_db_user, mssql_db_password, mssql_db_driver, db_name):
-    return 'mssql+pyodbc://{}:{}@localhost/{}?driver={}'\
-        .format(mssql_db_user, mssql_db_password, db_name, mssql_db_driver)
+    pytest.skip()
 
 
 @pytest.fixture
