diff --git i/src/util/translation_util.cpp w/src/util/translation_util.cpp
index da03d216..c1f12751 100644
--- i/src/util/translation_util.cpp
+++ w/src/util/translation_util.cpp
@@ -103,7 +103,7 @@ TranslationUtil::TranslationUtil(const QString& code, QString translation_file)
 	}
 	
 	QString translation_name = QLatin1String("qt_") + language.code;
-	if (!qt_translator.load(translation_name, QLibraryInfo::location(QLibraryInfo::TranslationsPath)))
+	if (!qt_translator.load(translation_name, QLatin1String("@qttranslations@/translations")))
 		load(qt_translator, translation_name);
 	
 	load(app_translator, translation_file);
