diff --exclude '*~' -rc mdadm-3.1.2-orig/udev-md-raid.rules mdadm-3.1.2/udev-md-raid.rules
*** mdadm-3.1.2-orig/udev-md-raid.rules	2010-02-22 21:14:58.000000000 +0100
--- mdadm-3.1.2/udev-md-raid.rules	2010-05-16 20:45:22.000000000 +0200
***************
*** 5,11 ****
  ACTION=="change", GOTO="md_no_incr"
  
  # import data from a raid member and activate it
! #ENV{ID_FS_TYPE}=="linux_raid_member", IMPORT{program}="/sbin/mdadm --examine --export $tempnode", RUN+="/sbin/mdadm --incremental $env{DEVNAME}"
  # import data from a raid set
  LABEL="md_no_incr"
  KERNEL!="md*", GOTO="md_end"
--- 5,11 ----
  ACTION=="change", GOTO="md_no_incr"
  
  # import data from a raid member and activate it
! ENV{ID_FS_TYPE}=="linux_raid_member", IMPORT{program}="/sbin/mdadm --examine --export $tempnode", RUN+="/sbin/mdadm --incremental $env{DEVNAME}"
  # import data from a raid set
  LABEL="md_no_incr"
  KERNEL!="md*", GOTO="md_end"
