diff --git a/pkg/utils/downloader_test.go b/pkg/utils/downloader_test.go
index b227760..b72dee9 100644
--- a/pkg/utils/downloader_test.go
+++ b/pkg/utils/downloader_test.go
@@ -7,6 +7,7 @@ import (
 )
 
 func TestDownloadSwaggerFile(t *testing.T) {
+    t.Skipf("Nix sandbox does not have networking")
 	var tmpdir string
 	// Github actions does not have a temporary dir :/
 	tmpdir = os.Getenv("RUNNER_TEMP")
