diff --git a/ext/soup/gstsouploader.c b/ext/soup/gstsouploader.c
index 85048ce303..d7d818cf95 100644
--- a/ext/soup/gstsouploader.c
+++ b/ext/soup/gstsouploader.c
@@ -181,7 +181,7 @@ gst_soup_load_library (void)
       GST_DEBUG ("LibSoup 2 found");
     } else {
       GST_DEBUG ("Trying all libsoups");
-      libsoup_sonames[0] = LIBSOUP_3_SONAME;
+      libsoup_sonames[0] = "@nixLibSoup3Path@/" LIBSOUP_3_SONAME;
       libsoup_sonames[1] = LIBSOUP_2_SONAME;
     }
 
