diff -rc glibc-orig/csu/Makefile glibc-2.3.3/csu/Makefile
*** glibc-orig/csu/Makefile	2003-09-25 22:29:39.000000000 +0200
--- glibc-2.3.3/csu/Makefile	2004-09-17 11:06:56.000000000 +0200
***************
*** 104,110 ****
  $(crtstuff:%=$(objpfx)%.o): %.o: %.S $(objpfx)defs.h
  	$(compile.S) -g0 $(ASFLAGS-.os) -o $@
  
! CFLAGS-initfini.s = -g0 -fPIC -fno-inline-functions
  
  vpath initfini.c $(full_config_sysdirs)
  
--- 104,110 ----
  $(crtstuff:%=$(objpfx)%.o): %.o: %.S $(objpfx)defs.h
  	$(compile.S) -g0 $(ASFLAGS-.os) -o $@
  
! CFLAGS-initfini.s = -g0 -fPIC -fno-inline-functions -fno-unit-at-a-time
  
  vpath initfini.c $(full_config_sysdirs)
