diff -ru glibc-2.18-orig/sunrpc/rpc_main.c glibc-2.18/sunrpc/rpc_main.c
--- glibc-2.18-orig/sunrpc/rpc_main.c	2013-08-11 00:52:55.000000000 +0200
+++ glibc-2.18/sunrpc/rpc_main.c	2013-11-15 12:04:48.041006977 +0100
@@ -78,7 +78,7 @@
 
 static const char *svcclosetime = "120";
 static int cppDefined;	/* explicit path for C preprocessor */
-static const char *CPP = "/lib/cpp";
+static const char *CPP = "cpp";
 static const char CPPFLAGS[] = "-C";
 static char *pathbuf;
 static int cpp_pid;
@@ -107,7 +107,6 @@
 static void open_output (const char *infile, const char *outfile);
 static void add_warning (void);
 static void clear_args (void);
-static void find_cpp (void);
 static void open_input (const char *infile, const char *define);
 static int check_nettype (const char *name, const char *list_to_check[]);
 static void c_output (const char *infile, const char *define,
@@ -322,25 +321,6 @@
   argcount = FIXEDARGS;
 }
 
-/* make sure that a CPP exists */
-static void
-find_cpp (void)
-{
-  struct stat64 buf;
-
-  if (stat64 (CPP, &buf) == 0)
-    return;
-
-  if (cppDefined) /* user specified cpp but it does not exist */
-    {
-      fprintf (stderr, _ ("cannot find C preprocessor: %s\n"), CPP);
-      crash ();
-    }
-
-  /* fall back to system CPP */
-  CPP = "cpp";
-}
-
 /*
  * Open input file with given define for C-preprocessor
  */
@@ -359,7 +339,6 @@
   switch (cpp_pid)
     {
     case 0:
-      find_cpp ();
       putarg (0, CPP);
       putarg (1, CPPFLAGS);
       addarg (define);
