diff --git a/cmake/fido2.cmake b/cmake/fido2.cmake
index c20e6e75c0d..f2d5cbd8430 100644
--- a/cmake/fido2.cmake
+++ b/cmake/fido2.cmake
@@ -30,19 +30,8 @@ MACRO(FIND_FIDO_VERSION)
   IF(WITH_FIDO STREQUAL "bundled")
     SET(FIDO_VERSION "1.7.0")
   ELSE()
-    # This does not set any version information:
-    # PKG_CHECK_MODULES(SYSTEM_FIDO fido2)
-
     MYSQL_CHECK_PKGCONFIG()
-    EXECUTE_PROCESS(
-      COMMAND ${MY_PKG_CONFIG_EXECUTABLE} --modversion libfido2
-      OUTPUT_VARIABLE MY_FIDO_MODVERSION
-      OUTPUT_STRIP_TRAILING_WHITESPACE
-      RESULT_VARIABLE MY_MODVERSION_RESULT
-      )
-    IF(MY_MODVERSION_RESULT EQUAL 0)
-      SET(FIDO_VERSION ${MY_FIDO_MODVERSION})
-    ENDIF()
+    PKG_CHECK_MODULES(FIDO libfido2)
   ENDIF()
   MESSAGE(STATUS "FIDO_VERSION (${WITH_FIDO}) is ${FIDO_VERSION}")
 ENDMACRO(FIND_FIDO_VERSION)
