For the parser extension

--- origsrc/Include/node.h	2013-05-11 22:32:41.000000000 -0500
+++ src/Include/node.h	2013-06-04 02:55:03.949347500 -0500
@@ -21,7 +21,7 @@ PyAPI_FUNC(int) PyNode_AddChild(node *n,
                                       char *str, int lineno, int col_offset);
 PyAPI_FUNC(void) PyNode_Free(node *n);
 #ifndef Py_LIMITED_API
-Py_ssize_t _PyNode_SizeOf(node *n);
+PyAPI_FUNC(Py_ssize_t) _PyNode_SizeOf(node *n);
 #endif
 
 /* Node access functions */
