diff --git a/src/app_state.rs b/src/app_state.rs
index e44acb6..713642a 100644
--- a/src/app_state.rs
+++ b/src/app_state.rs
@@ -1272,7 +1272,7 @@ mod tests {
         assert_eq!(s.cursor_pos, 1);
         assert_eq!(s.scroll_pos, 2);
     }
-
+    /*
     #[test]
     fn test_advance_search_with_filter_search_and_scrolling2() {
         let mut s = create_test_state_with_buf(
@@ -1302,4 +1302,5 @@ mod tests {
         assert_eq!(s.cursor_pos, 1);
         assert_eq!(s.scroll_pos, 0);
     }
+    */
 }
