diff --git a/sip/QtWebEngineWidgets/qwebenginecertificateerror.sip b/sip/QtWebEngineWidgets/qwebenginecertificateerror.sip
index 5fb508f..7a4654a 100644
--- a/sip/QtWebEngineWidgets/qwebenginecertificateerror.sip
+++ b/sip/QtWebEngineWidgets/qwebenginecertificateerror.sip
@@ -48,7 +48,7 @@ public:
 %If (QtWebEngine_5_8_0 -)
         CertificateTransparencyRequired,
 %End
-%If (QtWebEngine_5_14_0 -)
+%If (QtWebEngine_5_15_0 -)
         CertificateKnownInterceptionBlocked,
 %End
     };
diff --git a/sip/QtWebEngineWidgets/qwebenginepage.sip b/sip/QtWebEngineWidgets/qwebenginepage.sip
index 0dcbed2..9aa3443 100644
--- a/sip/QtWebEngineWidgets/qwebenginepage.sip
+++ b/sip/QtWebEngineWidgets/qwebenginepage.sip
@@ -663,10 +663,10 @@ signals:
 %End

 public:
-%If (QtWebEngine_5_14_0 -)
+%If (QtWebEngine_5_15_0 -)
     qint64 renderProcessPid() const;
 %End
-%If (QtWebEngine_5_14_0 -)
+%If (QtWebEngine_5_15_0 -)
     void renderProcessPidChanged(qint64 pid);
 %End
 };

