commit 6ea5c039671de2547249c36ca3e1fb51fc4a7e06
Author: Sandro Jäckel <sandro.jaeckel@gmail.com>
Date:   Thu Feb 6 18:33:28 2025 +0100

    Devendor argcomplete

diff --git a/pyproject.toml b/pyproject.toml
index 5ac5cab..6b60188 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -21,6 +21,7 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
+argcomplete = "*"
 python = ">=3.9.0,<4"
 windows-curses = {version = "^2.3.3", markers = "sys_platform == 'win32'"}
 pyreadline3 = {version = "^3.4.1", markers = "sys_platform == 'win32'"}
diff --git a/recline/repl/completer.py b/recline/repl/completer.py
index ff35583..1a05ae3 100644
--- a/recline/repl/completer.py
+++ b/recline/repl/completer.py
@@ -5,6 +5,7 @@
 as argument names and values.
 """
 
+import argcomplete
 import argparse
 import re
 import readline
@@ -12,7 +13,6 @@
 
 import recline
 from recline.arg_types.recline_type import UniqueParam
-from recline.vendor import argcomplete
 
 
 def match_command_hook(substitution, matches, *_):
