diff -rc callgrind-0.10.1-orig/callgrind.in callgrind-0.10.1/callgrind.in
*** callgrind-0.10.1-orig/callgrind.in	2005-11-14 03:21:14.000000000 +0100
--- callgrind-0.10.1/callgrind.in	2006-01-31 13:35:59.000000000 +0100
***************
*** 11,14 ****
          esac
  esac
  
! exec @VG_BINDIR@/valgrind --tool=callgrind $*
--- 11,14 ----
          esac
  esac
  
! VALGRIND_LIB=@prefix@/lib/valgrind/ exec @VG_BINDIR@/valgrind --tool=callgrind $*
diff -rc callgrind-0.10.1-orig/configure callgrind-0.10.1/configure
*** callgrind-0.10.1-orig/configure	2005-11-25 02:00:26.000000000 +0100
--- callgrind-0.10.1/configure	2006-01-31 13:41:48.000000000 +0100
***************
*** 3461,3467 ****
     { (exit 1); exit 1; }; }
    fi
  else
!   vg_exec=`which $prefix/bin/valgrind`
    if test "x$vg_exec" = "x"; then
      echo "$as_me:$LINENO: result: not found in $prefix/bin" >&5
  echo "${ECHO_T}not found in $prefix/bin" >&6
--- 3461,3467 ----
     { (exit 1); exit 1; }; }
    fi
  else
!   vg_exec=`which valgrind`
    if test "x$vg_exec" = "x"; then
      echo "$as_me:$LINENO: result: not found in $prefix/bin" >&5
  echo "${ECHO_T}not found in $prefix/bin" >&6
***************
*** 3484,3493 ****
  
  
  
! VG_BINDIR="$prefix/bin"
  
  cat >>confdefs.h <<_ACEOF
! #define VG_PREFIX "$prefix"
  _ACEOF
  
  
--- 3484,3493 ----
  
  
  
! VG_BINDIR="$vg_binpath"
  
  cat >>confdefs.h <<_ACEOF
! #define VG_PREFIX "$ac_default_prefix"
  _ACEOF
  
  
***************
*** 3495,3501 ****
  use_pkg_config="yes"
  echo "$as_me:$LINENO: checking for valgrind version" >&5
  echo $ECHO_N "checking for valgrind version... $ECHO_C" >&6
! vg_version=`$prefix/bin/valgrind --version`
  case "${vg_version}" in
      valgrind-3.1.*)
  	CG_SRCDIR=build31
--- 3495,3501 ----
  use_pkg_config="yes"
  echo "$as_me:$LINENO: checking for valgrind version" >&5
  echo $ECHO_N "checking for valgrind version... $ECHO_C" >&6
! vg_version=`$vg_binpath/valgrind --version`
  case "${vg_version}" in
      valgrind-3.1.*)
  	CG_SRCDIR=build31
***************
*** 3522,3528 ****
  echo "${ECHO_T}${vg_version}" >&6
  
  # get architecture from .pc file
! eval "pkgdir=$libdir/pkgconfig"
  echo "$as_me:$LINENO: checking for arch/os used in valgrind installation" >&5
  echo $ECHO_N "checking for arch/os used in valgrind installation... $ECHO_C" >&6
  
--- 3522,3528 ----
  echo "${ECHO_T}${vg_version}" >&6
  
  # get architecture from .pc file
! eval "pkgdir=$ac_default_prefix/lib/pkgconfig"
  echo "$as_me:$LINENO: checking for arch/os used in valgrind installation" >&5
  echo $ECHO_N "checking for arch/os used in valgrind installation... $ECHO_C" >&6
  
