--- libnl-1.1/lib/Makefile.orig	2008-01-14 16:48:45.000000000 +0100
+++ libnl-1.1/lib/Makefile	2009-05-26 09:47:42.000000000 +0200
@@ -48,7 +48,7 @@
 
 $(OUT_SLIB): ../Makefile.opts $(OBJ)
 	@echo "  LD $(OUT_SLIB)"; \
-	$(CC) -shared -Wl,-soname,libnl.so.1 -o $(OUT_SLIB) $(OBJ) $(LIBNL_LIB) -lc
+	$(CC) $(CFLAGS) $(LDFLAGS) -shared -Wl,-soname,libnl.so.1 -o $(OUT_SLIB) $(OBJ) $(LIBNL_LIB) -lc
 	@echo "  LN $(OUT_SLIB) $(LN1_SLIB)"; \
 	rm -f $(LN1_SLIB) ; $(LN) -s $(OUT_SLIB) $(LN1_SLIB)
 	@echo "  LN $(LN1_SLIB) $(LN_SLIB)"; \
