diff -Naur a/src/read.c b/src/read.c
--- a/src/read.c	2017-11-19 15:17:47.000000000 -0500
+++ b/src/read.c	2018-02-19 08:53:51.548755213 -0500
@@ -109,10 +109,12 @@
 #endif
     INCLUDEDIR,
 #ifndef _AMIGA
+#if 0
     "/usr/gnu/include",
     "/usr/local/include",
     "/usr/include",
 #endif
+#endif
     0
   };
 
diff -Naur a/src/remake.c b/src/remake.c
--- a/src/remake.c	2017-11-19 15:17:47.000000000 -0500
+++ b/src/remake.c	2018-02-19 08:54:08.304101943 -0500
@@ -1601,9 +1601,11 @@
   static const char *dirs[] =
     {
 #ifndef _AMIGA
+#if 0
       "/lib",
       "/usr/lib",
 #endif
+#endif
 #if defined(WINDOWS32) && !defined(LIBDIR)
 /*
  * This is completely up to the user at product install time. Just define
