diff --git a/auth.go b/auth.go
index 942feae..42c4381 100644
--- a/auth.go
+++ b/auth.go
@@ -69,7 +69,7 @@ func login(ctx context.Context, retryWithApiKey bool) error {
 	}
 
 	var preLogin preLoginResponse
-	if err := jsonPOST(ctx, apiURL+"/accounts/prelogin", &preLogin, preLoginRequest{
+	if err := jsonPOST(ctx, idtURL+"/accounts/prelogin", &preLogin, preLoginRequest{
 		Email: email,
 	}); err != nil {
 		return fmt.Errorf("could not pre-login: %v", err)
