diff --git a/howdy/src/meson.build b/howdy/src/meson.build
index 6533951..ef60e0a 100644
--- a/howdy/src/meson.build
+++ b/howdy/src/meson.build
@@ -101,11 +101,6 @@ install_data(
     rename: 'howdy',
 )
 
-fs = import('fs')
-if not fs.exists(config_path)
-    install_data('config.ini', install_dir: confdir, install_mode: 'rwxr--r--', install_tag: 'config')
-endif
-
 install_data('dlib-data/install.sh', 'dlib-data/Readme.md', install_dir: dlibdatadir, install_mode: 'rwxr--r--')
 
 install_man('../howdy.1')
