diff --git openjdk-orig/jdk/make/sun/awt/mawt.gmk openjdk/jdk/make/sun/awt/mawt.gmk
index c6ab06d..23a14da 100644
--- openjdk-orig/jdk/make/sun/awt/mawt.gmk
+++ openjdk/jdk/make/sun/awt/mawt.gmk
@@ -270,12 +270,6 @@ LDFLAGS  += -L$(MOTIF_LIB) -L$(OPENWIN_LIB)
 endif # !HEADLESS
 endif # PLATFORM
 
-ifeq ($(PLATFORM), linux)
-  # Checking for the X11/extensions headers at the additional location
-  CPPFLAGS += -I$(firstword $(wildcard $(OPENWIN_HOME)/include/X11/extensions) \
-                        $(wildcard /usr/include/X11/extensions))
-endif
-
 ifeq ($(PLATFORM), macosx)
   CPPFLAGS += -I$(OPENWIN_HOME)/include/X11/extensions \
               -I$(OPENWIN_HOME)/include 
