--- a/repl.k
+++ b/repl.k
@@ -1,6 +1,6 @@
 #!k
 `1:"ngn/k, (c) 2019-2021 ngn, GNU AGPLv3. type \\ for more info\n",repl.prompt:," "
-repl.cmds:(,"a")!{`1:1:repl.joinpath[repl.dirname`argv 0]"LICENSE";}
+repl.cmds:(,"a")!{`1:1:repl.joinpath[repl.dirname`argv 0]"../share/ngn-k/LICENSE";}
 repl.dirname:{$[#x:"/"/-1_"/"\x;x;,"."]}
 repl.joinpath:{$[x~,".";y;"/"~*|x;x,y;x,"/",y]}
 repl.fmt:{$[x~(::);"";(`A~@x)&1<#x;"(",("\n "/`k'x),")\n";`k[x],"\n"]}
