diff -ruN initscripts-8.18/src/Makefile initscripts-8.18.new/src/Makefile
--- initscripts-8.18/src/Makefile	2005-03-31 22:52:04.000000000 +0200
+++ initscripts-8.18.new/src/Makefile	2005-12-12 17:54:14.000000000 +0100
@@ -1,12 +1,13 @@
 CFLAGS+=$(RPM_OPT_FLAGS) -Wall -D_GNU_SOURCE
 
 PROGS=usernetctl doexec netreport testd usleep ipcalc initlog \
-	getkey ppp-watch consoletype genhostid kmodule
+	getkey consoletype genhostid
+	#getkey ppp-watch consoletype genhostid kmodule
 PPPWATCH_OBJS=ppp-watch.o shvar.o
 INITLOG_OBJS=initlog.o process.o
 USLEEP_OBJS=usleep.o
 
-mandir=/usr/share/man
+mandir=
 
 all:	$(PROGS)
 
@@ -14,9 +15,9 @@
 	rm -f $(PROGS) *.o
 
 install:
-	mkdir -p $(ROOT)/bin $(ROOT)/usr/sbin $(ROOT)$(mandir)/man{1,8} $(ROOT)/etc
+	mkdir -p $(ROOT)/bin $(ROOT)/sbin $(ROOT)/usr/sbin $(ROOT)$(mandir)/man{1,8} $(ROOT)/etc
 	install -m 755 doexec $(ROOT)/bin/doexec
-	install -m 755 kmodule $(ROOT)/sbin/kmodule
+	#install -m 755 kmodule $(ROOT)/sbin/kmodule
 	install -m 755 usleep $(ROOT)/bin/usleep
 	install -m 4755 usernetctl $(ROOT)/usr/sbin/usernetctl
 	install -m 2755 netreport $(ROOT)/sbin/netreport
@@ -24,7 +25,7 @@
 	install -m 755 genhostid $(ROOT)/sbin/genhostid
 	install -m 755 initlog $(ROOT)/sbin/initlog
 	install -m 755 getkey $(ROOT)/sbin/getkey
-	install -m 755 ppp-watch $(ROOT)/sbin/ppp-watch
+	#install -m 755 ppp-watch $(ROOT)/sbin/ppp-watch
 	install -m 755 consoletype $(ROOT)/sbin/consoletype
 	install -m 644 initlog.1 $(ROOT)$(mandir)/man1
 	install -m 644 genhostid.1 $(ROOT)$(mandir)/man1
@@ -32,7 +33,7 @@
 	install -m 644 netreport.1 $(ROOT)$(mandir)/man1
 	install -m 644 usleep.1 $(ROOT)$(mandir)/man1
 	install -m 644 usernetctl.8 $(ROOT)$(mandir)/man8
-	install -m 644 ppp-watch.8 $(ROOT)$(mandir)/man8
+	#install -m 644 ppp-watch.8 $(ROOT)$(mandir)/man8
 	install -m 644 ipcalc.1 $(ROOT)$(mandir)/man1
 	install -m 644 consoletype.1 $(ROOT)$(mandir)/man1
 	install -m 644 initlog.conf $(ROOT)/etc
