diff --git a/tools/cling/cmake/modules/CMakeLists.txt b/tools/cling/cmake/modules/CMakeLists.txt
--- a/tools/cling/cmake/modules/CMakeLists.txt
+++ b/tools/cling/cmake/modules/CMakeLists.txt
@@ -54,10 +54,8 @@ set(CLING_CONFIG_EXPORTS_FILE)

 if (NOT LLVM_INSTALL_TOOLCHAIN_ONLY)
   get_property(cling_has_exports GLOBAL PROPERTY CLING_HAS_EXPORTS)
-  if(cling_has_exports)
-    install(EXPORT ClingTargets DESTINATION ${CLING_INSTALL_PACKAGE_DIR}
-            COMPONENT cling-cmake-exports)
-  endif()
+  install(EXPORT ClingTargets DESTINATION ${CLING_INSTALL_PACKAGE_DIR}
+          COMPONENT cling-cmake-exports)

   install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/ClingConfig.cmake
