Index: a/src/image/tim_png.cpp
===================================================================
--- a/src/image/tim_png.cpp	(revision 3401)
+++ b/src/image/tim_png.cpp	(revision 3402)
@@ -25,7 +25,8 @@
 #include <stdio.h>
 #include <string.h>
 #include "tim.h"
-#include "png.h"
+#include <png.h>
+#include <zlib.h>
 
 /*==============================================================================
 Types
Index: ChangeLog
===================================================================
--- a/ChangeLog	(revision 3401)
+++ b/ChangeLog	(revision 3402)
@@ -1,3 +1,7 @@
+X-Moto 0.5.10 2012-
+--------------------------------
+ * libpng patch by Kelly Anderson
+
 X-Moto 0.5.9 2011-11-13
 --------------------------------
  * fix a compilation error with libxml
