diff --git a/kcms/dateandtime/helper.cpp b/kcms/dateandtime/helper.cpp
index cec5ab8..fc4a6b9 100644
--- a/kcms/dateandtime/helper.cpp
+++ b/kcms/dateandtime/helper.cpp
@@ -48,10 +48,6 @@
 #include <sys/stat.h>
 #endif
 
-// We cannot rely on the $PATH environment variable, because D-Bus activation
-// clears it. So we have to use a reasonable default.
-static const QString exePath = QLatin1String("/usr/sbin:/usr/bin:/sbin:/bin");
-
 int ClockHelper::ntp( const QStringList& ntpServers, bool ntpEnabled )
 {
     int ret = 0;
@@ -227,7 +223,7 @@ int ClockHelper::tzreset()
 
 void ClockHelper::toHwclock()
 {
-    QString hwclock = KStandardDirs::findExe("hwclock", exePath);
+    QString hwclock = "@hwclock@";
     if (!hwclock.isEmpty()) {
         KProcess::execute(hwclock, QStringList() << "--systohc");
     }
