diff --git a/src/utils/dumplibs.cpp b/src/utils/dumplibs.cpp
index 4215d6183..2d410f150 100644
--- a/src/utils/dumplibs.cpp
+++ b/src/utils/dumplibs.cpp
@@ -140,7 +140,7 @@ void dumpLibs()
     LIBXML_TEST_VERSION
 #endif  // LIBXML_TEST_VERSION
 #ifdef ENABLE_LIBXML
-    const char **xmlVersion = __xmlParserVersion();
+    const char * const *xmlVersion = __xmlParserVersion();
     if (xmlVersion != nullptr)
         logger->log(" libxml2: %s", *xmlVersion);
 #endif  // ENABLE_LIBXML
diff --git a/src/utils/xml/libxml.inc b/src/utils/xml/libxml.inc
index c60abd095..cf4c845a9 100644
--- a/src/utils/xml/libxml.inc
+++ b/src/utils/xml/libxml.inc
@@ -24,6 +24,7 @@
 
 #ifdef ENABLE_LIBXML
 
+#include <libxml/parser.h>
 #include <libxml/xmlwriter.h>
 
 __XML_XMLWRITER_H__
