diff --git a/cmake/AddDependencies.cmake b/cmake/AddDependencies.cmake
index 8272d14..054c079 100644
--- a/cmake/AddDependencies.cmake
+++ b/cmake/AddDependencies.cmake
@@ -51,7 +51,6 @@ endif()
 include(FetchContent)
 FetchContent_Declare(
   qlementine
-  GIT_REPOSITORY https://github.com/oclero/qlementine.git
-  GIT_TAG        v1.2.0
+  SOURCE_DIR "@qlementine-src@"
 )
 FetchContent_MakeAvailable(qlementine)
