diff -rupN ocaml-bitstring-2.0.4/META.in ocaml-bitstring-2.0.4.new/META.in
--- ocaml-bitstring-2.0.4/META.in	2013-05-14 17:42:32.000000000 +0200
+++ ocaml-bitstring-2.0.4.new/META.in	2015-06-11 17:26:11.674000000 +0200
@@ -15,8 +15,8 @@ package "persistent" (
 
 package "syntax" (
   version = "@PACKAGE_VERSION@"
-  requires = "camlp4"
+  requires = "camlp4, bitstring.persistent, bitstring"
   description = "Syntax extension: bitstring operators"
-  archive(syntax,preprocessor) = "-parser o -parser op -printer p unix.cma bitstring.cma bitstring_persistent.cma pa_bitstring.cmo"
-  archive(syntax,toploop) = "unix.cma bitstring.cma bitstring_persistent.cma pa_bitstring.cmo"
+  archive(syntax,preprocessor) = "-parser o -parser op -printer p pa_bitstring.cmo"
+  archive(syntax,toploop) = "pa_bitstring.cmo"
 )
