diff --git a/tests/mediastream.c b/tests/mediastream.c
index 12e1605..cdc8cd1 100644
--- a/tests/mediastream.c
+++ b/tests/mediastream.c
@@ -144,7 +144,6 @@ int main(int argc, char * argv[])
 	rtp_profile_set_payload(&av_profile,97,&payload_type_theora);
 	rtp_profile_set_payload(&av_profile,99,&payload_type_mp4v);
 	rtp_profile_set_payload(&av_profile,100,&payload_type_x_snow);
-	rtp_profile_set_payload(&av_profile,102,&payload_type_h264);
 #endif
 	if (argc<4) {
 		printf(usage);
