diff --git a/cmake/deps.cmake b/cmake/deps.cmake
index b63ced21..2cd1727b 100644
--- a/cmake/deps.cmake
+++ b/cmake/deps.cmake
@@ -13,20 +13,22 @@ endif(PTHREAD_LIB)
 if (MBROLA_BIN)
   set(HAVE_MBROLA ON)
 endif(MBROLA_BIN)
-if (SONIC_LIB AND SONIC_INC)
-  set(HAVE_LIBSONIC ON)
-else()
-  FetchContent_Declare(sonic-git
-    GIT_REPOSITORY https://github.com/waywardgeek/sonic.git
-    GIT_TAG fbf75c3d6d846bad3bb3d456cbc5d07d9fd8c104
-  )
-  FetchContent_MakeAvailable(sonic-git)
-  FetchContent_GetProperties(sonic-git)
-  add_library(sonic OBJECT ${sonic-git_SOURCE_DIR}/sonic.c)
-  target_include_directories(sonic PUBLIC ${sonic-git_SOURCE_DIR})
-  set(HAVE_LIBSONIC ON)
-  set(SONIC_LIB sonic)
-  set(SONIC_INC ${sonic-git_SOURCE_DIR})
+if (USE_LIBSONIC)
+  if (SONIC_LIB AND SONIC_INC)
+    set(HAVE_LIBSONIC ON)
+  else()
+    FetchContent_Declare(sonic-git
+      GIT_REPOSITORY https://github.com/waywardgeek/sonic.git
+      GIT_TAG fbf75c3d6d846bad3bb3d456cbc5d07d9fd8c104
+    )
+    FetchContent_MakeAvailable(sonic-git)
+    FetchContent_GetProperties(sonic-git)
+    add_library(sonic OBJECT ${sonic-git_SOURCE_DIR}/sonic.c)
+    target_include_directories(sonic PUBLIC ${sonic-git_SOURCE_DIR})
+    set(HAVE_LIBSONIC ON)
+    set(SONIC_LIB sonic)
+    set(SONIC_INC ${sonic-git_SOURCE_DIR})
+  endif()
 endif()
 if (PCAUDIO_LIB AND PCAUDIO_INC)
   set(HAVE_LIBPCAUDIO ON)
