--- a/src/libespeak-ng/mbrowrap.c
+++ b/src/libespeak-ng/mbrowrap.c
@@ -205,7 +205,7 @@
 		signal(SIGTERM, SIG_IGN);
 
 		snprintf(charbuf, sizeof(charbuf), "%g", mbr_volume);
-		execlp("mbrola", "mbrola", "-e", "-v", charbuf,
+		execlp("@mbrola@/bin/mbrola", "mbrola", "-e", "-v", charbuf,
 		       voice_path, "-", "-.wav", (char *)NULL);
 		/* if execution reaches this point then the exec() failed */
 		snprintf(mbr_errorbuf, sizeof(mbr_errorbuf),
