diff --git a/src/sas/_config.py b/src/sas/_config.py
index ece08fd4c..926768593 100644
--- a/src/sas/_config.py
+++ b/src/sas/_config.py
@@ -67,8 +67,8 @@ def load_local_config(app_dir):
         logger.info("GuiManager loaded %s", path)
         return module
     except Exception as exc:
-        logger.critical("Error loading %s: %s", path, exc)
-        sys.exit()
+        import sas.sasview.local_config
+        return sas.sasview.local_config
 
 def make_custom_config_path(user_dir):
     """
@@ -116,4 +116,4 @@ def load_custom_config(path):
 
     from sas.sasview import custom_config
     logger.info("GuiManager custom_config defaults to sas.sasview.custom_config")
-    return custom_config
\ No newline at end of file
+    return custom_config
