diff --git a/Source/JavaScriptCore/shell/CMakeLists.txt b/Source/JavaScriptCore/shell/CMakeLists.txt
--- a/Source/JavaScriptCore/shell/CMakeLists.txt
+++ b/Source/JavaScriptCore/shell/CMakeLists.txt
@@ -9,7 +9,6 @@ set(JSC_LIBRARIES
 )
 
 if (WTF_OS_MAC_OS_X)
-    list(APPEND JSC_LIBRARIES edit)
 endif ()
 
 if ("${JavaScriptCore_LIBRARY_TYPE}" MATCHES "STATIC")
diff --git a/Source/WTF/wtf/Platform.h b/Source/WTF/wtf/Platform.h
--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -563,7 +563,6 @@
 #if PLATFORM(IOS)
 
 #define HAVE_NETWORK_EXTENSION 1
-#define HAVE_READLINE 1
 #if USE(APPLE_INTERNAL_SDK)
 #define USE_CFNETWORK 1
 #endif
@@ -650,7 +649,6 @@
 #define HAVE_MADV_DONTNEED 1
 #define HAVE_MERGESORT 1
 #define HAVE_PTHREAD_SETNAME_NP 1
-#define HAVE_READLINE 1
 #define HAVE_SYS_TIMEB_H 1
 
 #if !PLATFORM(GTK) && !PLATFORM(QT)
