diff -Naur --strip-trailing-cr source.org/src/racer/nameres.rs source/src/racer/nameres.rs
--- source.org/src/racer/nameres.rs	2017-11-15 20:37:38.571644733 +0000
+++ source/src/racer/nameres.rs	2017-11-15 20:23:20.521324031 +0000
@@ -577,6 +577,7 @@
     out.into_iter()
 }
 
+#[ignore]
 #[test]
 fn test_do_file_search() {
     let cache = core::FileCache::default();
diff -Naur --strip-trailing-cr source.org/src/racer/util.rs source/src/racer/util.rs
--- source.org/src/racer/util.rs	2017-11-15 19:37:55.095344120 +0000
+++ source/src/racer/util.rs	2017-11-15 20:22:53.746624158 +0000
@@ -475,6 +475,7 @@
 
 }
 
+#[ignore]
 #[test]
 fn test_get_rust_src_path_missing() {
     use std::env;
