diff -urN iniparser3.0b.orig/Makefile iniparser3.0b/Makefile
--- iniparser3.0b.orig/Makefile	2008-01-16 19:56:08.000000000 +0100
+++ iniparser3.0b/Makefile	2008-01-16 19:56:49.000000000 +0100
@@ -11,8 +11,8 @@
 ARFLAGS = rcv
 
 SHLD = ${CC} ${CFLAGS}
-LDSHFLAGS = -shared -Wl,-Bsymbolic  -Wl,-rpath -Wl,/usr/lib -Wl,-rpath,/usr/lib
-LDFLAGS = -Wl,-rpath -Wl,/usr/lib -Wl,-rpath,/usr/lib
+LDSHFLAGS = -shared
+LDFLAGS =
 
 # Set RANLIB to ranlib on systems that require it (Sun OS < 4, Mac OSX)
 # RANLIB  = ranlib
