diff --git a/tests/debug/session.py b/tests/debug/session.py
index af242877..30b21a1e 100644
--- a/tests/debug/session.py
+++ b/tests/debug/session.py
@@ -622,6 +622,7 @@ class Session(object):
         if "PYTHONPATH" in self.config.env:
             # If specified, launcher will use it in lieu of PYTHONPATH it inherited
             # from the adapter when spawning debuggee, so we need to adjust again.
+            self.config.env.prepend_to("PYTHONPATH", os.environ["PYTHONPATH"])
             self.config.env.prepend_to("PYTHONPATH", DEBUGGEE_PYTHONPATH.strpath)
         return self._request_start("launch")
 
