Debian's patch to build with fresher GCC

Index: wvstreams-4.6.1/crypto/wvx509.cc
===================================================================
--- wvstreams-4.6.1.orig/crypto/wvx509.cc	2011-05-20 00:02:38.119136584 +0200
+++ wvstreams-4.6.1/crypto/wvx509.cc	2011-05-20 00:02:26.035136589 +0200
@@ -1157,7 +1157,7 @@
         
         if (ext)
         {
-            X509V3_EXT_METHOD *method = X509V3_EXT_get(ext);
+            X509V3_EXT_METHOD *method = (X509V3_EXT_METHOD *)X509V3_EXT_get(ext);
             if (!method)
             {
                 WvDynBuf buf;
Index: wvstreams-4.6.1/ipstreams/wvunixdgsocket.cc
===================================================================
--- wvstreams-4.6.1.orig/ipstreams/wvunixdgsocket.cc	2011-05-20 00:02:38.391136584 +0200
+++ wvstreams-4.6.1/ipstreams/wvunixdgsocket.cc	2011-05-20 00:02:35.283136585 +0200
@@ -1,8 +1,6 @@
 #include "wvunixdgsocket.h"
-#ifdef MACOS
 #include <sys/types.h>
 #include <sys/stat.h>
-#endif
 
 WvUnixDGSocket::WvUnixDGSocket(WvStringParm filename, bool _server, int perms)
     : socketfile(filename)
Index: wvstreams-4.6.1/streams/wvatomicfile.cc
===================================================================
--- wvstreams-4.6.1.orig/streams/wvatomicfile.cc	2011-05-20 00:02:38.223136584 +0200
+++ wvstreams-4.6.1/streams/wvatomicfile.cc	2011-05-20 00:02:31.619136587 +0200
@@ -10,10 +10,7 @@
 #include "wvatomicfile.h"
 #include "wvfileutils.h"
 #include "wvstrutils.h"
-
-#ifdef MACOS
 #include <sys/stat.h>
-#endif
 
 WvAtomicFile::WvAtomicFile(WvStringParm filename, int flags, mode_t create_mode)
     : tmp_file(WvString::null)
