diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index a923cadce53..95a4d3a807f 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -186,7 +186,9 @@ if(BUILD_INTEGRATION_TESTS)
                            WORKING_DIRECTORY
                            ${TEST_BIN_DIR}
                            FIXTURES_REQUIRED
-                           unpack_data)
+                           unpack_data
+                           DEPENDS
+                           ${DATA}_execute)
         endif()
     endforeach(DATA)
 endif()
