--- a/backend/src/Makefile.am	2012-12-22 11:09:21.674631729 +0100
+++ b/backend/src/Makefile.am	2012-12-22 14:20:50.879566578 +0100
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-backend_bindir = /usr/lib/cups/backend
+backend_bindir = /usr/libexec/cups/backend
 backend_bin_PROGRAMS = cnijusb
 transform = 
 
--- a/backendnet/backend/Makefile.am	2012-12-22 11:09:21.673631729 +0100
+++ b/backendnet/backend/Makefile.am	2012-12-22 14:20:20.176569425 +0100
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-backendnet_bindir = /usr/lib/cups/backend
+backendnet_bindir = /usr/libexec/cups/backend
 backendnet_bin_PROGRAMS = cnijnet
 transform = 
 

