diff --git a/Makefile.comm b/Makefile.comm
index 75efc22..b757000 100644
--- a/Makefile.comm
+++ b/Makefile.comm
@@ -155,7 +155,7 @@ extraclean: distclean
 	     -e "s|@MAN1EXT@|$(man1ext)|g" \
 	     -e "s|@MAN5EXT@|$(man5ext)|g" \
 	     -e "s|@MAN7EXT@|$(man7ext)|g" \
-	     -e "s|@MDATE@|`$(SHELL) $(top_srcdir)/mdate.sh $<`|g" \
+	     -e "s|@MDATE@|`date +'%-d %B %Y' -r $(top_srcdir)/ChangeLog`|g" \
 	     -e "s|@OLDFONTDIR@|$(oldfontdir)|g" \
 	     -e "s|@PDFDOCDIR@|$(pdfdocdir)|g" \
 	     -e "s|@SYSTEMMACRODIR@|$(systemtmacdir)|g" \
