diff --git a/src/Makefile b/src/Makefile
index 3a1cd680..576353fe 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -189,7 +189,7 @@ endif
 # brain and xxHash
 ifeq ($(ENABLE_BRAIN),1)
 CFLAGS                  += -DWITH_BRAIN
-CFLAGS                  += -Ideps/git/xxHash
+LFLAGS                  += -lxxhash
 endif
 
 ##
@@ -313,7 +313,7 @@ WIN_32_OBJS             := $(foreach OBJ,$(OBJS_ALL),obj/$(OBJ).WIN.32.o)
 WIN_64_OBJS             := $(foreach OBJ,$(OBJS_ALL),obj/$(OBJ).WIN.64.o)
 
 ifeq ($(ENABLE_BRAIN),1)
-OBJS_XXHASH             := xxhash
+OBJS_XXHASH             :=
 
 NATIVE_STATIC_OBJS      += $(foreach OBJ,$(OBJS_XXHASH),obj/$(OBJ).NATIVE.STATIC.o)
 NATIVE_SHARED_OBJS      += $(foreach OBJ,$(OBJS_XXHASH),obj/$(OBJ).NATIVE.SHARED.o)
