diff --git a/mk/support/pkg/v8.sh b/mk/support/pkg/v8.sh
index a1810cc..da2b645 100644
--- a/mk/support/pkg/v8.sh
+++ b/mk/support/pkg/v8.sh
@@ -155,7 +155,7 @@ pkg_install () {
         if [[ "$arch" = "ppc64" ]]; then
             arch_gypflags="$arch_gypflags -Duse_system_icu=1"
         fi
-        pkg_make $arch.$mode CXX=$CXX LINK=$CXX LINK.target=$CXX GYPFLAGS="-Dwerror= $arch_gypflags" V=1
+        pkg_make $arch.$mode CXX=$CXX LINK=$CXX LINK.target=$CXX GYPFLAGS="-Dwerror= -Dv8_use_snapshot=false $arch_gypflags" V=1
         for lib in `find "$build_dir/out/$arch.$mode" -maxdepth 1 -name \*.a` `find "$build_dir/out/$arch.$mode/obj.target" -name \*.a`; do
             name=`basename $lib`
             cp $lib "$install_dir/lib/${name/.$arch/}"
@@ -168,7 +168,7 @@ pkg_link-flags () {
     # These are the necessary libraries recommended by the docs:
     # https://developers.google.com/v8/get_started#hello
      if [[ "$ARCH" != "ppc64le" ]]; then
-	 for lib in libv8_{base,libbase,snapshot,libplatform}; do
+	 for lib in libv8_{base,libbase,nosnapshot,libplatform}; do
              echo "$install_dir/lib/$lib.a"
     	 done
     	 for lib in libicu{i18n,uc,data}; do
