--- a/usage.c
+++ b/usage.c
@@ -931,24 +931,6 @@ usage(err, fh, version)
 		(void) fprintf(stderr, "    configuration info: %s\n", cp);
 #endif	/* defined(LSOF_CINFO) */
 
-	    cp = isnullstr(LSOF_HOST);
-	    if (!(cp1 = isnullstr(LSOF_LOGNAME)))
-		cp1 = isnullstr(LSOF_USER);
-	    if (cp || cp1) {
-		if (cp && cp1)
-		    cp2 = "by and on";
-		else if (cp)
-		    cp2 = "on";
-		else
-		    cp2 = "by";
-		(void) fprintf(stderr, "    constructed %s: %s%s%s\n",
-		    cp2,
-		    cp1 ? cp1 : "",
-		    (cp && cp1) ? "@" : "",
-		    cp  ? cp  : ""
-		    );
-	    }
-
 #if	defined(LSOF_BLDCMT)
 	    if ((cp = isnullstr(LSOF_BLDCMT)))
 		(void) fprintf(stderr, "    builder's comment: %s\n", cp);
@@ -958,12 +940,6 @@ usage(err, fh, version)
 		(void) fprintf(stderr, "    compiler: %s\n", cp);
 	    if ((cp = isnullstr(LSOF_CCV)))
 		(void) fprintf(stderr, "    compiler version: %s\n", cp);
-	    if ((cp = isnullstr(LSOF_CCFLAGS)))
-		(void) fprintf(stderr, "    compiler flags: %s\n", cp);
-	    if ((cp = isnullstr(LSOF_LDFLAGS)))
-		(void) fprintf(stderr, "    loader flags: %s\n", cp);
-	    if ((cp = isnullstr(LSOF_SYSINFO)))
-		(void) fprintf(stderr, "    system info: %s\n", cp);
 	    // display configurations that might affect output
 	    char *features[] = {
 #if	defined(HASEFFNLINK)
