diff --git a/packages/net/_test.pony b/packages/net/_test.pony
index baf29e7c7..ae6ce6702 100644
--- a/packages/net/_test.pony
+++ b/packages/net/_test.pony
@@ -6,13 +6,12 @@ actor Main is TestList
 
   fun tag tests(test: PonyTest) =>
     ifdef not osx then
-      test(_TestBroadcast)
+      test(_TestTCPWritev)
+      test(_TestTCPExpect)
+      test(_TestTCPMute)
+      test(_TestTCPUnmute)
     end
-    test(_TestTCPWritev)
-    test(_TestTCPExpect)
-    test(_TestTCPMute)
-    test(_TestTCPUnmute)
-    ifdef not windows then
+    ifdef linux then
       test(_TestTCPThrottle)
     end
 
diff --git a/packages/net/http/_test.pony b/packages/net/http/_test.pony
index e55d5a789..40a4cb618 100644
--- a/packages/net/http/_test.pony
+++ b/packages/net/http/_test.pony
@@ -29,8 +29,6 @@ actor Main is TestList
     test(_Valid)
     test(_ToStringFun)
 
-    test(_HTTPConnTest)
-
 class iso _Encode is UnitTest
   fun name(): String => "net/http/URLEncode.encode"
 
