--- a/Makefile.am.old	2014-08-12 10:07:32.000000000 +0530
+++ b/Makefile.am	2014-09-08 16:23:56.046996941 +0530
@@ -43,7 +43,6 @@
 AM_CPPFLAGS =
 AM_LDFLAGS =
 
-BUILT_SOURCES = uthash/src
 EXTRA_DIST =
 CLEANFILES =
 MOSTLYCLEANFILES =
@@ -113,8 +112,7 @@
 	Packaging/FontForge-doc.spec \
 	Packaging/FontForge.spec \
 	Packaging/FontForge.static.spec \
 	README \
-	uthash/src \
 	$(NULL)
 
 #--------------------------------------------------------------------------
@@ -129,11 +127,6 @@
 
 
 #--------------------------------------------------------------------------
-uthash/src:
-	if [ ! -e uthash/src ]; then \
-	if [ -e uthash ] ; then rm -r uthash ; fi ; \
-	git clone https://github.com/troydhanson/uthash ; \
-	fi ;
 
 # We import a selection of targets from Frank's standard packaging Makefile.
 
