See https://github.com/krb5/krb5/pull/467
diff --git a/src/include/Makefile.in b/src/include/Makefile.in
index 4bb11e4..cb4b44b 100644
--- a/src/include/Makefile.in
+++ b/src/include/Makefile.in
@@ -57,19 +57,19 @@ SBINDIR = @sbindir@
 LIBDIR  = @libdir@
 SYSCONFCONF = @SYSCONFCONF@
 
-PROCESS_REPLACE = -e "s+@KRB5RCTMPDIR+$(KRB5RCTMPDIR)+" \
-		  -e "s+@PREFIX+$(INSTALL_PREFIX)+" \
-		  -e "s+@EXEC_PREFIX+$(INSTALL_EXEC_PREFIX)+" \
-		  -e "s+@BINDIR+$(BINDIR)+" \
-		  -e "s+@LIBDIR+$(LIBDIR)+" \
-		  -e "s+@SBINDIR+$(SBINDIR)+" \
-		  -e "s+@MODULEDIR+$(MODULE_DIR)+" \
-		  -e "s+@GSSMODULEDIR+$(GSS_MODULE_DIR)+" \
-		  -e 's+@LOCALSTATEDIR+$(LOCALSTATEDIR)+' \
-		  -e 's+@RUNSTATEDIR+$(RUNSTATEDIR)+' \
-		  -e 's+@SYSCONFDIR+$(SYSCONFDIR)+' \
-		  -e 's+@DYNOBJEXT+$(DYNOBJEXT)+' \
-		  -e 's+@SYSCONFCONF+$(SYSCONFCONF)+'
+PROCESS_REPLACE = -e "s\"@KRB5RCTMPDIR\"$(KRB5RCTMPDIR)\"" \
+		  -e "s\"@PREFIX\"$(INSTALL_PREFIX)\"" \
+		  -e "s\"@EXEC_PREFIX\"$(INSTALL_EXEC_PREFIX)\"" \
+		  -e "s\"@BINDIR\"$(BINDIR)\"" \
+		  -e "s\"@LIBDIR\"$(LIBDIR)\"" \
+		  -e "s\"@SBINDIR\"$(SBINDIR)\"" \
+		  -e "s\"@MODULEDIR\"$(MODULE_DIR)\"" \
+		  -e "s\"@GSSMODULEDIR\"$(GSS_MODULE_DIR)\"" \
+		  -e "s\"@LOCALSTATEDIR\"$(LOCALSTATEDIR)\"" \
+		  -e "s\"@RUNSTATEDIR\"$(RUNSTATEDIR)\"" \
+		  -e "s\"@SYSCONFDIR\"$(SYSCONFDIR)\"" \
+		  -e "s\"@DYNOBJEXT\"$(DYNOBJEXT)\"" \
+		  -e "s\"@SYSCONFCONF\"$(SYSCONFCONF)\""
 
 OSCONFSRC = $(srcdir)/osconf.hin
 
