--- a/src/nm-openconnect-service.c
+++ b/src/nm-openconnect-service.c
@@ -60,7 +60,7 @@
 
 static const char *openconnect_binary_paths[] =
 {
-	"/usr/bin/openconnect",
+	"@openconnect@/bin/openconnect",
 	"/usr/sbin/openconnect",
 	"/usr/local/bin/openconnect",
 	"/usr/local/sbin/openconnect",
@@ -734,7 +734,7 @@
 
 	_LOGD ("nm-openconnect-service (version " DIST_VERSION ") starting...");
 
-	if (system ("/sbin/modprobe tun") == -1)
+	if (system ("@kmod@/bin/modprobe tun") == -1)
 		exit (EXIT_FAILURE);
 
 	if (bus_name)
