--- a/src/Test/L0/TestUtil.cs
+++ b/src/Test/L0/TestUtil.cs
@@ -29,7 +29,7 @@
         {
             string L0dir = Path.GetDirectoryName(GetTestFilePath());
             string testDir = Path.GetDirectoryName(L0dir);
-            string srcDir = Path.GetDirectoryName(testDir);
+            string srcDir = "/build/src" + Path.GetDirectoryName(testDir).Substring(2);
             ArgUtil.Directory(srcDir, nameof(srcDir));
             Assert.Equal(Src, Path.GetFileName(srcDir));
             return srcDir;
