diff --git a/scheduler/main.c b/scheduler/main.c
index 8925c8373..acf031684 100644
--- a/scheduler/main.c
+++ b/scheduler/main.c
@@ -893,7 +893,7 @@ main(int  argc,				/* I - Number of command-line args */
     * Write dirty config/state files...
     */
 
-    if (DirtyCleanTime && current_time >= DirtyCleanTime && cupsArrayCount(Clients) == 0)
+    if (DirtyCleanTime && current_time >= DirtyCleanTime)
       cupsdCleanDirty();
 
 #ifdef __APPLE__
