diff --git a/SConstruct b/SConstruct
index a80bca3..ed2cd6d 100644
--- a/SConstruct
+++ b/SConstruct
@@ -5,7 +5,7 @@ from subprocess import *
 def getGitDesc():
   return Popen('git describe --abbrev=8 --tags --always', stdout=PIPE, shell=True).stdout.read ().strip ()
 
-env = Environment ()
+env = Environment(ENV = os.environ)
 
 AddOption ("--release", action="store", dest="release", default="git", help="Make a release (default: git describe output)")
 AddOption ("--enable-debug", action="store", dest="debug", default=None, help="Enable the -g flag for debugging (default: true when release is git)")
