diff --git a/modules/input/Makefile.am b/modules/input/Makefile.am
index c8610fac8e..b73305c56a 100644
--- a/modules/input/Makefile.am
+++ b/modules/input/Makefile.am
@@ -167,7 +167,7 @@ QUARTZ_MODULE = im-quartz.la
 endif
 endif
 
-multipress_defs = -DMULTIPRESS_LOCALEDIR=\""$(mplocaledir)"\" -DMULTIPRESS_CONFDIR=\""$(sysconfdir)/gtk-2.0"\"
+multipress_defs = -DMULTIPRESS_LOCALEDIR=\""$(mplocaledir)"\" -DMULTIPRESS_CONFDIR=\""$(prefix)/etc/gtk-2.0"\"
 im_multipress_la_CPPFLAGS = $(multipress_defs)
 libstatic_im_multipress_la_CPPFLAGS = $(im_multipress_la_CPPFLAGS)
 im_multipress_la_LDFLAGS = -rpath $(moduledir) -avoid-version -module $(no_undefined)
@@ -183,7 +183,7 @@ else
 MULTIPRESS_MODULE = im-multipress.la
 endif
 
-imconffiledir = $(sysconfdir)/gtk-2.0
+imconffiledir = $(prefix)/etc/gtk-2.0
 dist_imconffile_DATA = im-multipress.conf
 
 if CROSS_COMPILING
