Only in btrfs-progs-v3.16-orig: btrfs-progs-v3.16
diff -U3 -r btrfs-progs-v3.16-orig/utils.c btrfs-progs-v3.16/utils.c
--- btrfs-progs-v3.16-orig/utils.c	2014-09-03 09:57:45.644858545 +0400
+++ btrfs-progs-v3.16/utils.c	2014-09-03 09:59:26.968855663 +0400
@@ -2360,7 +2360,7 @@
 		len = strlen(ent->mnt_dir);
 		if (strncmp(ent->mnt_dir, path, len) == 0) {
 			/* match found */
-			if (longest_matchlen < len) {
+			if ((longest_matchlen < len) || (longest_matchlen==len && not_btrfs)) {
 				free(longest_match);
 				longest_matchlen = len;
 				longest_match = strdup(ent->mnt_dir);
