diff --git a/src/terminal.rs b/src/terminal.rs
--- a/src/terminal.rs
+++ b/src/terminal.rs
@@ -803,6 +803,7 @@ mod tests {
         SgrTerminalCommand::ExitAttributeMode,
     };

+    #[ignore]
     #[test]
     fn sgr_combining() {
         // No style, no content
@@ -838,6 +839,7 @@ mod tests {
         assert_eq!(String::from_utf8_lossy(outp.contents()), "\u{1b}[30;40m");
     }

+    #[ignore]
     #[test]
     fn sgr_max_length() {
         // Cut at max length
@@ -892,6 +894,7 @@ mod tests {
         );
     }

+    #[ignore]
     #[test]
     fn resettable_style_attribute() {
         type RS = crate::text_face::ResettableStyle<()>;
