diff --git a/CMakeLists.txt b/CMakeLists.txt
index cda04d1..9d42fc9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,31 +38,16 @@ set(CMAKE_MODULE_PATH
 include(FetchContent)
 
 FetchContent_Declare(Sodium
-  GIT_REPOSITORY https://github.com/AmineKhaldi/libsodium-cmake.git
-
-  # Latest commit at the moment this was added here
-  # Anchored to libsodium v1.0.18
-  GIT_TAG f73a3fe1afdc4e37ac5fe0ddd401bf521f6bba65
+  URL @sodium_src@
 )
 set(SODIUM_PCH "on" CACHE STRING "")
 set(SODIUM_DISABLE_TESTS "on" CACHE STRING "")
 set(SODIUM_CHIA_MINIMAL "on" CACHE STRING "")
 FetchContent_MakeAvailable(Sodium)
 
-if (DEFINED ENV{BLST_MAIN})
-  set(BLST_GIT_TAG "origin/master")
-else ()
-  # This is currently anchored to upstream 6b837a0921cf41e501faaee1976a4035ae29d893 dated 2023-06-16
-  set(BLST_GIT_TAG "6b837a0921cf41e501faaee1976a4035ae29d893")
-endif ()
-set(BLST_REPOSITORY "https://github.com/supranational/blst.git")
-
-message(STATUS "blst will be built from: ${BLST_GIT_TAG} and repository ${BLST_REPOSITORY}")
-
 FetchContent_Declare(
   blst
-  GIT_REPOSITORY ${BLST_REPOSITORY}
-  GIT_TAG ${BLST_GIT_TAG}
+  URL @blst_src@
 )
 FetchContent_MakeAvailable(blst)
 
diff --git a/python-bindings/CMakeLists.txt b/python-bindings/CMakeLists.txt
index 0e77050..3cb6187 100644
--- a/python-bindings/CMakeLists.txt
+++ b/python-bindings/CMakeLists.txt
@@ -1,8 +1,7 @@
 
 FetchContent_Declare(
   pybind11
-  GIT_REPOSITORY https://github.com/pybind/pybind11.git
-  GIT_TAG v2.10.0
+  URL @pybind11_src@
 )
 FetchContent_MakeAvailable(pybind11)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5d35c78..34192d5 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -60,8 +60,7 @@ install(FILES $<TARGET_FILE:bls> DESTINATION lib)
 if(BUILD_BLS_TESTS)
   FetchContent_Declare(
     Catch2
-    GIT_REPOSITORY https://github.com/catchorg/Catch2.git
-    GIT_TAG v3.3.2
+    URL @catch2_src@
   )
   FetchContent_MakeAvailable(Catch2)
   find_package(Threads REQUIRED)
