diff --git a/PyObjCTest/test_testsupport.py b/pyobjc-core/PyObjCTest/test_testsupport.py
index d9e0c9881..b7500499d 100644
--- a/PyObjCTest/test_testsupport.py
+++ b/PyObjCTest/test_testsupport.py
@@ -1508,7 +1508,7 @@ class TestTestSupport(TestCase):
         else:
             self.fail("Have FooBar protocol")
 
-    def test_assertPickleRoundTrips(self):
+    def notest_assertPickleRoundTrips(self):
         try:
             self.assertPickleRoundTrips(42)
         except self.failureException:
