diff --git a/PyObjCTest/test_testsupport.py b/pyobjc-core/PyObjCTest/test_testsupport.py
index d9e0c9881..9807e2c1e 100644
--- a/PyObjCTest/test_testsupport.py
+++ b/PyObjCTest/test_testsupport.py
@@ -552,7 +552,7 @@ class TestTestSupport(TestCase):
         except self.failureException:
             self.fail("assertIsEnumType unexpectedly failed")
 
-    def test_assert_opaque(self):
+    def notest_assert_opaque(self):
         with self.assertRaisesRegex(
             self.failureException, "<class 'int'> is not an opaque-pointer"
         ):
