---
 setup.py | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/setup.py b/setup.py
index 985d28d..933f3df 100755
--- a/setup.py
+++ b/setup.py
@@ -24,13 +24,9 @@ import shutil
 # finding os platform
 os_type = platform.system()
 
-if os_type == 'Linux' or os_type == 'FreeBSD' or os_type == 'OpenBSD':
-    from setuptools import setup, Command, find_packages
-    setuptools_available = True
-    print(os_type + " detected!")
-else:
-    print('This script is only work for GNU/Linux or BSD!')
-    sys.exit(1)
+from setuptools import setup, Command, find_packages
+setuptools_available = True
+print(os_type + " detected!")
 
 # Checking dependencies!
 not_installed = ''
@@ -100,6 +96,7 @@ else:
     print('paplay is found!')
 
 # sound-theme-freedesktop
+notifications_path = ''
 if os_type == 'Linux':
     notifications_path = '/usr/share/sounds/freedesktop/stereo/'
 elif os_type == 'FreeBSD' or os_type == 'OpenBSD':
@@ -139,7 +136,7 @@ if sys.argv[1] == "test":
 
 DESCRIPTION = 'Persepolis Download Manager'
 
-if os_type == 'Linux':
+if os_type in ['Linux', 'Darwin']:
     DATA_FILES = [
         ('/usr/share/man/man1/', ['man/persepolis.1.gz']),
         ('/usr/share/applications/', ['xdg/com.github.persepolisdm.persepolis.desktop']),
-- 
2.39.3 (Apple Git-145)

