diff -ru qt-everywhere-opensource-src-4.8.3-orig/src/corelib/tools/qlocale_icu.cpp qt-everywhere-opensource-src-4.8.3/src/corelib/tools/qlocale_icu.cpp
--- qt-everywhere-opensource-src-4.8.3-orig/src/corelib/tools/qlocale_icu.cpp	2012-09-10 21:36:50.000000000 -0400
+++ qt-everywhere-opensource-src-4.8.3/src/corelib/tools/qlocale_icu.cpp	2012-09-28 22:57:40.764153828 -0400
@@ -81,7 +81,7 @@
     if (status == NotLoaded) {
 
         // resolve libicui18n
-        QLibrary lib(QLatin1String("icui18n"), QLatin1String(U_ICU_VERSION_SHORT));
+        QLibrary lib(QLatin1String("@icu@/lib/libicui18n"), QLatin1String(U_ICU_VERSION_SHORT));
         lib.setLoadHints(QLibrary::ImprovedSearchHeuristics);
         if (!lib.load()) {
             qWarning() << "Unable to load library icui18n" << lib.errorString();
@@ -111,7 +111,7 @@
         }
 
         // resolve libicuuc
-        QLibrary ucLib(QLatin1String("icuuc"), QLatin1String(U_ICU_VERSION_SHORT));
+        QLibrary ucLib(QLatin1String("@icu@/lib/libicuuc"), QLatin1String(U_ICU_VERSION_SHORT));
         ucLib.setLoadHints(QLibrary::ImprovedSearchHeuristics);
         if (!ucLib.load()) {
             qWarning() << "Unable to load library icuuc" << ucLib.errorString();
diff -ru qt-everywhere-opensource-src-4.8.3-orig/src/gui/painting/qcups.cpp qt-everywhere-opensource-src-4.8.3/src/gui/painting/qcups.cpp
--- qt-everywhere-opensource-src-4.8.3-orig/src/gui/painting/qcups.cpp	2012-09-10 21:36:51.000000000 -0400
+++ qt-everywhere-opensource-src-4.8.3/src/gui/painting/qcups.cpp	2012-09-28 18:08:26.600720491 -0400
@@ -87,7 +87,7 @@
 
 static void resolveCups()
 {
-    QLibrary cupsLib(QLatin1String("cups"), 2);
+    QLibrary cupsLib(QLatin1String("@cups@/lib/libcups"), 2);
     if(cupsLib.load()) {
         _cupsGetDests = (CupsGetDests) cupsLib.resolve("cupsGetDests");
         _cupsFreeDests = (CupsFreeDests) cupsLib.resolve("cupsFreeDests");
diff -ru qt-everywhere-opensource-src-4.8.3-orig/src/gui/painting/qprinterinfo_unix.cpp qt-everywhere-opensource-src-4.8.3/src/gui/painting/qprinterinfo_unix.cpp
--- qt-everywhere-opensource-src-4.8.3-orig/src/gui/painting/qprinterinfo_unix.cpp	2012-09-10 21:36:51.000000000 -0400
+++ qt-everywhere-opensource-src-4.8.3/src/gui/painting/qprinterinfo_unix.cpp	2012-09-28 18:08:26.601720488 -0400
@@ -454,7 +454,7 @@
     char *domain;
     int err;
 
-    QLibrary lib(QLatin1String("nsl"));
+    QLibrary lib(QLatin1String("@glibc@/lib/libnsl"));
     typedef int (*ypGetDefaultDomain)(char **);
     ypGetDefaultDomain _ypGetDefaultDomain = (ypGetDefaultDomain)lib.resolve("yp_get_default_domain");
     typedef int (*ypAll)(const char *, const char *, const struct ypall_callback *);
diff -ru qt-everywhere-opensource-src-4.8.3-orig/src/network/kernel/qhostinfo_unix.cpp qt-everywhere-opensource-src-4.8.3/src/network/kernel/qhostinfo_unix.cpp
--- qt-everywhere-opensource-src-4.8.3-orig/src/network/kernel/qhostinfo_unix.cpp	2012-09-10 21:36:51.000000000 -0400
+++ qt-everywhere-opensource-src-4.8.3/src/network/kernel/qhostinfo_unix.cpp	2012-09-28 18:08:53.186643755 -0400
@@ -95,7 +95,7 @@
 static void resolveLibrary()
 {
 #ifndef QT_NO_LIBRARY
-    QLibrary lib(QLatin1String("resolv"));
+    QLibrary lib(QLatin1String("@glibc@/lib/libresolv"));
     lib.setLoadHints(QLibrary::ImprovedSearchHeuristics);
     if (!lib.load())
         return;
diff -ru qt-everywhere-opensource-src-4.8.3-orig/src/plugins/platforms/xlib/qxlibstatic.cpp qt-everywhere-opensource-src-4.8.3/src/plugins/platforms/xlib/qxlibstatic.cpp
--- qt-everywhere-opensource-src-4.8.3-orig/src/plugins/platforms/xlib/qxlibstatic.cpp	2012-09-10 21:36:51.000000000 -0400
+++ qt-everywhere-opensource-src-4.8.3/src/plugins/platforms/xlib/qxlibstatic.cpp	2012-09-28 18:08:26.601720488 -0400
@@ -242,7 +242,7 @@
 }
 
 #  define XFIXES_LOAD_RUNTIME(vernum, symbol, symbol_type) \
-    (symbol_type)qt_load_library_runtime("libXfixes", vernum, 4, #symbol);
+    (symbol_type)qt_load_library_runtime("@libXfixes@/lib/libXfixes", vernum, 4, #symbol);
 #  define XFIXES_LOAD_V1(symbol) \
     XFIXES_LOAD_RUNTIME(1, symbol, Ptr##symbol)
 #  define XFIXES_LOAD_V2(symbol) \
