--- a/qmake/generators/unix/unixmake.cpp
+++ b/qmake/generators/unix/unixmake.cpp
@@ -831,7 +831,7 @@ UnixMakefileGenerator::defaultInstall(const QString &t)
             else if(project->first("TEMPLATE") == "app" && !project->isEmpty("QMAKE_STRIPFLAGS_APP"))
                 ret += " " + var("QMAKE_STRIPFLAGS_APP");
             if(bundle)
-                ret = " \"" + dst_targ + "/Contents/MacOS/$(QMAKE_TARGET)\"";
+                ret += " \"" + dst_targ + "/Contents/MacOS/$(QMAKE_TARGET)\"";
             else
                 ret += " \"" + dst_targ + "\"";
         }
