diff --git a/src/tools/install.neko b/src/tools/install.neko
index 4cffa5e..5702add 100644
--- a/src/tools/install.neko
+++ b/src/tools/install.neko
@@ -39,26 +39,6 @@ buffer_string = $loader.loadprim("std@buffer_string",1);
 // LIBS DATAS
 
 libs = {
-	mod_neko => {
-		src => $array("../../vm/stats","mod_neko","cgi"),
-		inc => "httpd.h",
-		incname => "Apache 1.3.x"
-	},
-	mod_neko2 => {
-		src => $array("../../vm/stats","mod_neko","cgi"),
-		inc => $array("httpd.h","apr.h"),
-		incname => "Apache 2.2.x",
-		cflags => "-D_LARGEFILE64_SOURCE",
-		realdir => "mod_neko",
-		apache2 => true,
-	},
-	mysql => {
-		src => $array("mysql"),
-		inc => "mysql.h",
-		incname => "MySQL 4.+"
-		lib => "libmysqlclient_r.a",
-		lparams => "-lz -lssl"
-	},
 	mysql5 => {
 		src => $array("../common/sha1","../common/socket","my_proto/my_proto","my_proto/my_api","mysql"),
 		inc => $array(),
@@ -83,27 +63,6 @@ libs = {
 		incname => "Sqlite 3",
 		lparams => "-lsqlite3",
 	},
-	ui => {
-		src => $array("ui"),
-		inc => switch system { "Mac" => "Carbon.h" default => "gtk/gtk.h" },
-		incname => switch system { "Mac" => "Carbon" default => "GTK+2.0" },
-		cflags => switch system { "Mac" => "" default => "`pkg-config --cflags gtk+-2.0`" },
-		lparams => switch system { "Mac" => "-framework Carbon" default => "`pkg-config --libs gtk+-2.0` -lgthread-2.0" },
-	},
-	mod_tora => {
-		src => $array("../common/socket","protocol","mod_tora"),
-		inc => "httpd.h",
-		incname => "Apache 1.3.x",
-		cflags => "-I../common",
-	},
-	mod_tora2 => {
-		src => $array("../common/socket","protocol","mod_tora"),
-		inc => $array("httpd.h","apr.h"),
-		incname => "Apache 2.2.x",
-		cflags => "-D_LARGEFILE64_SOURCE -I../common",
-		realdir => "mod_tora",
-		apache2 => true,
-	},
 }
 
 // PLATFORM
