diff --git a/config.h b/config.h
index 112c9ac..f592b29 100644
--- a/config.h
+++ b/config.h
@@ -73,13 +73,13 @@
 
 /* Size in bits of a bmp_t.  Not necessarily a power of two. */
 
-#define BMP_BIT   32
+#define BMP_BIT   64
 
 /* The highest power of two that is strictly less than BMP_BIT.
  * Initialises the index of a binary search for set bits in a bmp_t.
  */
 
-#define BMP_SUB   16
+#define BMP_SUB   32
 
 /*****************************************
  *					 *
