diff --git a/Modules/kwinsupportinfo/kcm_kwinsupportinfo.json.in b/Modules/kwinsupportinfo/kcm_kwinsupportinfo.json.in
index f591b9c..e883212 100644
--- a/Modules/kwinsupportinfo/kcm_kwinsupportinfo.json.in
+++ b/Modules/kwinsupportinfo/kcm_kwinsupportinfo.json.in
@@ -63,6 +63,6 @@
         "Name[x-test]": "xxWindow Managerxx",
         "Name[zh_CN]": "窗口管理器"
     },
-    "TryExec": "@QtBinariesDir@/qdbus",
+    "TryExec": "@qdbus@",
     "X-KDE-KInfoCenter-Category": "graphical_information"
 }
diff --git a/Modules/kwinsupportinfo/main.cpp b/Modules/kwinsupportinfo/main.cpp
index 667c079..b727b67 100644
--- a/Modules/kwinsupportinfo/main.cpp
+++ b/Modules/kwinsupportinfo/main.cpp
@@ -19,7 +19,7 @@ public:
     explicit KCMKWinSupportInfo(QObject *parent, const KPluginMetaData &data, const QVariantList &args)
         : ConfigModule(parent, data, args)
     {
-        auto outputContext = new CommandOutputContext(QLibraryInfo::location(QLibraryInfo::BinariesPath) + QStringLiteral("/qdbus"),
+        auto outputContext = new CommandOutputContext(QStringLiteral("@qdbus@"),
                                                       {QStringLiteral("org.kde.KWin"), QStringLiteral("/KWin"), QStringLiteral("supportInformation")},
                                                       parent);
         qmlRegisterSingletonInstance("org.kde.kinfocenter.kwinsupportinfo.private", 1, 0, "InfoOutputContext", outputContext);
diff --git a/Modules/xserver/kcm_xserver.json b/Modules/xserver/kcm_xserver.json
index 04acd6b..24b8f36 100644
--- a/Modules/xserver/kcm_xserver.json
+++ b/Modules/xserver/kcm_xserver.json
@@ -130,7 +130,7 @@
         "Name[zh_CN]": "X 服务器",
         "Name[zh_TW]": "X 伺服器"
     },
-    "TryExec": "xdpyinfo",
+    "TryExec": "@xdpyinfo@",
     "X-DocPath": "kinfocenter/graphical.html#xserver",
     "X-KDE-KInfoCenter-Category": "graphical_information",
     "X-KDE-Keywords": "X,X-Server,XServer,XFree86,Display,VideoCard,System Information",
diff --git a/Modules/xserver/main.cpp b/Modules/xserver/main.cpp
index c406ff7..a261b90 100644
--- a/Modules/xserver/main.cpp
+++ b/Modules/xserver/main.cpp
@@ -17,7 +17,7 @@ public:
     explicit KCMXServer(QObject *parent, const KPluginMetaData &data, const QVariantList &args)
         : ConfigModule(parent, data, args)
     {
-        auto outputContext = new CommandOutputContext(QStringLiteral("xdpyinfo"), {}, parent);
+        auto outputContext = new CommandOutputContext(QStringLiteral("@xdpyinfo@"), {}, parent);
         qmlRegisterSingletonInstance("org.kde.kinfocenter.xserver.private", 1, 0, "InfoOutputContext", outputContext);
 
         auto *about = new KAboutData(QStringLiteral("kcm_xserver"), i18nc("@label kcm name", "X-Server"), QStringLiteral("1.0"), QString(), KAboutLicense::GPL);
