diff --git a/include/my_global.h b/include/my_global.h
index cb31ae2..2866f87 100644
--- a/include/my_global.h
+++ b/include/my_global.h
@@ -803,12 +803,6 @@ inline unsigned long long my_double2ulonglong(double d)
 #endif
 
 #ifndef isfinite
-#ifdef HAVE_FINITE
-#define isfinite(x) finite(x)
-#else
-#define finite(x) (1.0 / fabs(x) > 0.0)
-#endif /* HAVE_FINITE */
-#elif (__cplusplus >= 201103L)
 #include <cmath>
 static inline bool isfinite(double x) { return std::isfinite(x); }
 #endif /* isfinite */
