--- a/mysys/my_context.c
+++ b/mysys/my_context.c
@@ -206,15 +206,6 @@ my_context_spawn(struct my_context *c, void (*f)(void *), void *d)
     (
      "movq %%rsp, (%[save])\n\t"
      "movq %[stack], %%rsp\n\t"
-#if __GNUC__ >= 4 && __GNUC_MINOR__ >= 4 && !defined(__INTEL_COMPILER)
-     /*
-       This emits a DWARF DW_CFA_undefined directive to make the return address
-       undefined. This indicates that this is the top of the stack frame, and
-       helps tools that use DWARF stack unwinding to obtain stack traces.
-       (I use numeric constant to avoid a dependency on libdwarf includes).
-     */
-     ".cfi_escape 0x07, 16\n\t"
-#endif
      "movq %%rbp, 8(%[save])\n\t"
      "movq %%rbx, 16(%[save])\n\t"
      "movq %%r12, 24(%[save])\n\t"
