diff -rc findutils-4.2.27-orig/locate/code.c findutils-4.2.27/locate/code.c
*** findutils-4.2.27-orig/locate/code.c	2005-06-07 22:24:56.000000000 +0000
--- findutils-4.2.27/locate/code.c	2006-10-20 15:53:53.000000000 +0000
***************
*** 210,216 ****
--- 210,218 ----
        if (diffcount < -LOCATEDB_OLD_OFFSET || diffcount > LOCATEDB_OLD_OFFSET)
  	{
  	  putc (LOCATEDB_OLD_ESCAPE, stdout);
+ #if 0          
  	  putw (diffcount + LOCATEDB_OLD_OFFSET, stdout);
+ #endif          
  	}
        else
  	putc (diffcount + LOCATEDB_OLD_OFFSET, stdout);
diff -rc findutils-4.2.27-orig/locate/locate.c findutils-4.2.27/locate/locate.c
*** findutils-4.2.27-orig/locate/locate.c	2005-09-01 20:42:43.000000000 +0000
--- findutils-4.2.27/locate/locate.c	2006-10-20 15:53:11.000000000 +0000
***************
*** 468,473 ****
--- 468,474 ----
    return VISIT_CONTINUE;
  }
  
+ #if 0
  static int
  visit_old_format(struct process_data *procdata, void *context)
  {
***************
*** 498,503 ****
--- 499,505 ----
    
    return VISIT_CONTINUE;
  }
+ #endif
  
  
  static int
***************
*** 911,920 ****
    lastinspector = NULL;
    past_pat_inspector = NULL;
  
!   if (old_format)
!     add_visitor(visit_old_format, NULL);
!   else
!     add_visitor(visit_locate02_format, NULL);
  
    if (basename_only)
      add_visitor(visit_basename, NULL);
--- 913,919 ----
    lastinspector = NULL;
    past_pat_inspector = NULL;
  
!   add_visitor(visit_locate02_format, NULL);
  
    if (basename_only)
      add_visitor(visit_basename, NULL);
