diff --git a/gst-libs/gst/cuda/gstcudaloader.cpp b/gst-libs/gst/cuda/gstcudaloader.cpp
index 11718b8..d4144c1 100644
--- a/gst-libs/gst/cuda/gstcudaloader.cpp
+++ b/gst-libs/gst/cuda/gstcudaloader.cpp
@@ -229,6 +229,11 @@ gst_cuda_load_library_once_func (void)
       "CUDA plugin loader");
 
   module = g_module_open (filename, G_MODULE_BIND_LAZY);
+
+  if (module == nullptr) {
+    module = g_module_open ("@driverLink@/lib/" CUDA_LIBNAME, G_MODULE_BIND_LAZY);
+  }
+
   if (module == nullptr) {
     GST_WARNING ("Could not open library %s, %s", filename, g_module_error ());
     return;
diff --git a/sys/nvcodec/gstcuvidloader.c b/sys/nvcodec/gstcuvidloader.c
index c51a428..ea0e1b5 100644
--- a/sys/nvcodec/gstcuvidloader.c
+++ b/sys/nvcodec/gstcuvidloader.c
@@ -87,6 +87,11 @@ gst_cuvid_load_library (guint api_major_ver, guint api_minor_ver)
     return TRUE;
 
   module = g_module_open (filename, G_MODULE_BIND_LAZY);
+
+  if (module == NULL) {
+    module = g_module_open ("@driverLink@/lib/" NVCUVID_LIBNAME, G_MODULE_BIND_LAZY);
+  }
+
   if (module == NULL) {
     GST_WARNING ("Could not open library %s, %s", filename, g_module_error ());
     return FALSE;
diff --git a/sys/nvcodec/gstnvenc.c b/sys/nvcodec/gstnvenc.c
index c65c85a..57232bb 100644
--- a/sys/nvcodec/gstnvenc.c
+++ b/sys/nvcodec/gstnvenc.c
@@ -919,6 +919,11 @@ gst_nvenc_load_library (guint * api_major_ver, guint * api_minor_ver)
   };
 
   module = g_module_open (NVENC_LIBRARY_NAME, G_MODULE_BIND_LAZY);
+
+  if (module == NULL) {
+    module = g_module_open ("@driverLink@/lib/" NVENC_LIBRARY_NAME, G_MODULE_BIND_LAZY);
+  }
+
   if (module == NULL) {
     GST_WARNING ("Could not open library %s, %s",
         NVENC_LIBRARY_NAME, g_module_error ());
