diff --git a/scripts/ssr-glinject b/scripts/ssr-glinject
index 48be48d..5038d4c 100755
--- a/scripts/ssr-glinject
+++ b/scripts/ssr-glinject
@@ -59,6 +59,6 @@ do
 	fi
 done
 
-echo "ssr-glinject: LD_PRELOAD = $LD_PRELOAD:libssr-glinject.so"
+echo "ssr-glinject: LD_PRELOAD = $LD_PRELOAD:@out@/lib/libssr-glinject.so"
 echo "ssr-glinject: command = $@"
-LD_PRELOAD="$LD_PRELOAD:libssr-glinject.so" "$@"
+LD_PRELOAD="$LD_PRELOAD:@out@/lib/libssr-glinject.so" "$@"
diff --git a/src/AV/Input/GLInjectInput.cpp b/src/AV/Input/GLInjectInput.cpp
index fc98f31..18f5196 100644
--- a/src/AV/Input/GLInjectInput.cpp
+++ b/src/AV/Input/GLInjectInput.cpp
@@ -113,7 +113,7 @@ bool ExecuteDetached(const char* command, const char* working_directory) {
 
 			// try to execute command
 			do {
-				res = execl("/bin/sh", "/bin/sh", "-c", command, (char*) NULL);
+				res = execl("@sh@", "@sh@", "-c", command, (char*) NULL);
 			} while(res == -1 and errno == EINTR);
 
 			// failed, send feedback
@@ -207,7 +207,7 @@ void GLInjectInput::SetCapturing(bool capturing) {
 bool GLInjectInput::LaunchApplication(const QString& channel, bool relax_permissions, const QString& command, const QString& working_directory) {
 
 	// prepare command
-	QString full_command = "LD_PRELOAD=\"libssr-glinject.so\" ";
+	QString full_command = "LD_PRELOAD=\"@out@/lib/libssr-glinject.so\" ";
 	full_command += "SSR_CHANNEL=\"" + ShellEscape(channel) + "\" ";
 	if(relax_permissions)
 		full_command += "SSR_STREAM_RELAX_PERMISSIONS=1 ";
