Index: kde-gtk-config-5.12.4/src/gtkconfigkcmodule.cpp
===================================================================
--- kde-gtk-config-5.12.4.orig/src/gtkconfigkcmodule.cpp
+++ kde-gtk-config-5.12.4/src/gtkconfigkcmodule.cpp
@@ -78,6 +78,16 @@ GTKConfigKCModule::GTKConfigKCModule(QWi
     m_iconsModel = new IconThemesModel(false, this);
     ui->cb_icon->setModel(m_iconsModel);
     ui->cb_icon_fallback->setModel(m_iconsModel);
+
+    // Add GSETTINGS_SCHEMAS_PATH to the front of XDG_DATA_DIRS.
+    // Normally this would be done by wrapGAppsHook, but this plugin
+    // (shared object) cannot be wrapped.
+    QByteArray xdgdata = qgetenv("XDG_DATA_DIRS");
+    if (!xdgdata.isEmpty()) {
+        xdgdata.push_front(":");
+    }
+    xdgdata.push_front(QByteArray(GSETTINGS_SCHEMAS_PATH));
+    qputenv("XDG_DATA_DIRS", xdgdata);
     
     m_tempGtk2Preview = QStandardPaths::writableLocation(QStandardPaths::TempLocation)+ "/gtkrc-2.0";
     m_tempGtk3Preview = QStandardPaths::writableLocation(QStandardPaths::TempLocation)+ "/.config/gtk-3.0/settings.ini";
