diff --git a/pyproject.toml b/pyproject.toml
index 9a83e90..0cdffaf 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,16 +1,12 @@
-[tool.poetry]
+[project]
 name = "django-q"
 version = "1.3.9"
 description = "A multiprocessing distributed task queue for Django"
-authors = ["Ilan Steemers <koed00@gmail.com>"]
-maintainers = ["Ilan Steemers <koed00@gmail.com>"]
-license = "MIT"
+authors = [ { name = "Ilan Steemers", email = "koed00@gmail.com"} ]
+maintainers = [ { name = "Ilan Steemers", email = "koed00@gmail.com"} ]
+license.text = "MIT"
 readme = 'README.rst'

-repository = "https://github.com/koed00/django-q"
-homepage = "https://django-q.readthedocs.org"
-documentation = "https://django-q.readthedocs.org"
-
 keywords = ["django", "distributed", "multiprocessing", "queue", "scheduler"]

 classifiers = [
@@ -31,7 +27,6 @@ classifiers = [
     'Topic :: System :: Distributed Computing',
     'Topic :: Software Development :: Libraries :: Python Modules',
 ]
-include = ['CHANGELOG.md']

 [tool.poetry.plugins] # Optional super table
