diff -Naur tvtime-1.0.2.org/src/xmltv.c tvtime-1.0.2/src/xmltv.c
--- tvtime-1.0.2.org/src/xmltv.c	2009-07-02 21:48:49.426191088 +0200
+++ tvtime-1.0.2/src/xmltv.c	2009-07-02 21:50:20.842066085 +0200
@@ -118,9 +118,9 @@
 typedef struct {
     const char *code;
     const char *name;
-} locale_t;
+} tvtime_locale_t;
 
-static locale_t locale_table[] = {
+static tvtime_locale_t locale_table[] = {
     {"AA", "Afar"},           {"AB", "Abkhazian"},      {"AF", "Afrikaans"},
     {"AM", "Amharic"},        {"AR", "Arabic"},         {"AS", "Assamese"},
     {"AY", "Aymara"},         {"AZ", "Azerbaijani"},    {"BA", "Bashkir"},
@@ -168,7 +168,7 @@
     {"XH", "Xhosa"},          {"YO", "Yoruba"},         {"ZH", "Chinese"},
     {"ZU", "Zulu"} };
 
-const int num_locales = sizeof( locale_table ) / sizeof( locale_t );
+const int num_locales = sizeof( locale_table ) / sizeof( tvtime_locale_t );
 
 /**
  * Timezone parsing code based loosely on the algorithm in
