diff --git a/c/syscalls.c b/c/syscalls.c
index 2be25a8..311f02b 100644
--- a/c/syscalls.c
+++ b/c/syscalls.c
@@ -670,7 +670,7 @@ s48_ref_t scm_gethostname(s48_call_t call)
 {
    char hostname[MAXHOSTNAMELEN+1];
     /* different OS's declare differently, so punt the prototype. */
-    int gethostname();
+    int gethostname(char *name, size_t size);
     int retval = gethostname(hostname, MAXHOSTNAMELEN);
     if (retval == -1) s48_os_error_2(call, "scm_gethostname", errno, 0);
     return s48_enter_byte_string_2(call, hostname);
