--- a/src/simutrans/simmain.cc	(revision 11174)
+++ b/src/simutrans/simmain.cc	(revision 11175)
@@ -589,6 +589,11 @@
 			}
 		}
 	}
+	
+	// create directory and subdirectories
+	dr_mkdir(env_t::install_dir);
+	dr_mkdir(env_t::user_dir);
+    
 	// set portable, if base dir and user dir were same
 	not_portable &= strcmp( env_t::user_dir, env_t::base_dir )!=0;
 
--- a/src/simutrans/sys/simsys.cc	(revision 11174)
+++ b/src/simutrans/sys/simsys.cc	(revision 11175)
@@ -379,13 +379,15 @@
 {
 	if(  path  &&  *path  ) {
 		bool ok = !dr_chdir(path);
-		FILE * testf = NULL;
-		ok &=  ok  &&  testfile  &&  (testf = fopen(testfile,"r"));
+		if(testfile) {
+			FILE* testf = fopen(testfile,"r");
+			ok = ok && testf;
+			fclose(testf);
+		}
 		if(!ok) {
 			printf("WARNING: Objects not found in %s \"%s\"!\n",  info, path);
 		}
 		else {
-			fclose(testf);
 			dr_getcwd( result, PATH_MAX-1 );
 			strcat( result, PATH_SEPARATOR );
 			return true;
@@ -532,8 +534,6 @@
 	}
 #endif
 
-	// create directory and subdirectories
-	dr_mkdir(buffer);
 	strcat(buffer, PATH_SEPARATOR);
 	return buffer;
 }
@@ -578,8 +578,6 @@
 	}
 #endif
 
-	// create directory and subdirectories
-	dr_mkdir(buffer);
 	strcat(buffer, PATH_SEPARATOR);
 	return buffer;
 }
--- a/src/simutrans/sys/simsys.h	(revision 11174)
+++ b/src/simutrans/sys/simsys.h	(revision 11175)
@@ -163,7 +163,7 @@
 * @param base_dir_arg : command line argument "-set_basedir"
 * @param executable_path : Path to executable (as stored in argv[0])
 */
-bool dr_set_basedir(const char * base_dir_arg, char * executable_path);
+bool dr_set_basedir(const char *base_dir_arg, char *executable_path);
 
 /* query home directory */
 char const* dr_query_homedir();
