From 42d3c4e7e95ea0e0dda9b502b2de2c4ed9ed9467 Mon Sep 17 00:00:00 2001
From: "T.M. Abraham" <tabraham@novell.com>
Date: Fri, 6 Jan 2012 21:58:55 -0500
Subject: [PATCH] properly check for HAVE_FALLOC_PH in both occurrences

---
 nbd-server.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/nbd-server.c b/nbd-server.c
index 0ecf0e8..2cd066d 100644
--- a/nbd-server.c
+++ b/nbd-server.c
@@ -1461,7 +1461,7 @@ int expflush(CLIENT *client) {
  * file to resparsify stuff that isn't needed anymore (see NBD_CMD_TRIM)
  */
 int exptrim(struct nbd_request* req, CLIENT* client) {
-#ifdef HAVE_FALLOC_PH
+#if HAVE_FALLOC_PH
 	FILE_INFO prev = g_array_index(client->export, FILE_INFO, 0);
 	FILE_INFO cur = prev;
 	int i = 1;
-- 
1.7.8

