diff --git a/dogtail/distro.py b/dogtail/distro.py
index 9533ed8..d931ab7 100644
--- a/dogtail/distro.py
+++ b/dogtail/distro.py
@@ -340,6 +340,9 @@ class _ArchPackageDb(PackageDb):
         deps = self._pacman("d", packageName)
         return [d for d in deps.split() if d != "None"]
 
+class _NixPackageDb(PackageDb):  # pragma: no cover
+    def __init__(self):
+        PackageDb.__init__(self)
 
 class Distro:
     """
@@ -409,6 +412,9 @@ class Arch(Distro):  # pragma: no cover
     def __init__(self):
         self.packageDb = _ArchPackageDb()
 
+class Nix(Distro):  # pragma: no cover
+    def __init__(self):
+        self.packageDb = _NixPackageDb()
 
 def detectDistro():  # pragma: no cover
     logger.log("Detecting distribution:", newline=False)
@@ -440,6 +446,8 @@ def detectDistro():  # pragma: no cover
     elif os.path.exists("/etc/os-release") and \
             re.match(".*GNOME-Continuous", open("/etc/os-release").readline()):
         distro = GnomeContinuous()
+    elif os.path.exists("/nix/store"):  # pragma: no cover
+        distro = Nix()  # pragma: no cover
     else:
         raise DistributionNotSupportedError("Unknown")
     