diff --git a/src/wallet/CMakeLists.txt b/src/wallet/CMakeLists.txt
index 63908005..f6656d5c 100644
--- a/src/wallet/CMakeLists.txt
+++ b/src/wallet/CMakeLists.txt
@@ -86,43 +86,40 @@ target_link_libraries(wallet
     ${EXTRA_LIBRARIES})
 add_dependencies(wallet version)
 
-if (NOT BUILD_GUI_DEPS)
-  set(wallet_rpc_sources
-    wallet_rpc_server.cpp)
+set(wallet_rpc_sources
+  wallet_rpc_server.cpp)
 
-  set(wallet_rpc_headers)
+set(wallet_rpc_headers)
 
-  set(wallet_rpc_private_headers
-    wallet_rpc_server.h)
+set(wallet_rpc_private_headers
+  wallet_rpc_server.h)
 
-  monero_private_headers(wallet_rpc_server
-    ${wallet_rpc_private_headers})
-  monero_add_executable(wallet_rpc_server
-    ${wallet_rpc_sources}
-    ${wallet_rpc_headers}
-    ${wallet_rpc_private_headers})
-
-  target_link_libraries(wallet_rpc_server
-    PRIVATE
-      wallet
-      epee
-      rpc
-      cryptonote_core
-      cncrypto
-      common
-      ${Boost_CHRONO_LIBRARY}
-      ${Boost_PROGRAM_OPTIONS_LIBRARY}
-      ${Boost_FILESYSTEM_LIBRARY}
-      ${Boost_THREAD_LIBRARY}
-      ${CMAKE_THREAD_LIBS_INIT}
-      ${EXTRA_LIBRARIES})
-  add_dependencies(wallet_rpc_server version)
-  set_property(TARGET wallet_rpc_server
-    PROPERTY
-      OUTPUT_NAME "monero-wallet-rpc")
-  install(TARGETS wallet_rpc_server DESTINATION bin)
-endif()
+monero_private_headers(wallet_rpc_server
+  ${wallet_rpc_private_headers})
+monero_add_executable(wallet_rpc_server
+  ${wallet_rpc_sources}
+  ${wallet_rpc_headers}
+  ${wallet_rpc_private_headers})
 
+target_link_libraries(wallet_rpc_server
+  PRIVATE
+    wallet
+    epee
+    rpc
+    cryptonote_core
+    cncrypto
+    common
+    ${Boost_CHRONO_LIBRARY}
+    ${Boost_PROGRAM_OPTIONS_LIBRARY}
+    ${Boost_FILESYSTEM_LIBRARY}
+    ${Boost_THREAD_LIBRARY}
+    ${CMAKE_THREAD_LIBS_INIT}
+    ${EXTRA_LIBRARIES})
+add_dependencies(wallet_rpc_server version)
+set_property(TARGET wallet_rpc_server
+  PROPERTY
+    OUTPUT_NAME "monero-wallet-rpc")
+install(TARGETS wallet_rpc_server DESTINATION bin)
 
 # build and install libwallet_merged only if we building for GUI
 if (BUILD_GUI_DEPS)
