--- a/src/integrations/navidrome.py
+++ b/src/integrations/navidrome.py
@@ -556,11 +556,6 @@
     process = None
 
     def check_if_ready(self, row) -> bool:
-        if get_navidrome_path():
-            return True
-        else:
-            row.get_root().main_stack.set_visible_child_name('setup')
-            row.get_root().main_stack.get_child_by_name('setup').set_integration(self)
         return False
 
     def start_instance(self) -> bool:
