diff --git a/meson.build b/meson.build
index 4c0cbe280f..d4479741c8 100644
--- a/meson.build
+++ b/meson.build
@@ -68,14 +68,12 @@
 
 version = '@0@'.format(meson.project_version())
 git = find_program('git', native: true, required: false)
-if git.found()
-  git_commit = run_command([git, 'rev-parse', '--short', 'HEAD'], check: false)
-  git_branch = run_command([git, 'rev-parse', '--abbrev-ref', 'HEAD'], check: false)
-  if git_commit.returncode() == 0 and git_branch.returncode() == 0
+if true
+  if true
     version = '@0@-@1@ (branch \'@2@\')'.format(
       meson.project_version(),
-      git_commit.stdout().strip(),
-      git_branch.stdout().strip(),
+      '@gitCommit@'.strip(),
+      '@gitBranch@'.strip(),
     )
   endif
 endif
