diff --git a/bin/dune b/bin/dune
index e0ddc1f..a4e8523 100644
--- a/bin/dune
+++ b/bin/dune
@@ -47,6 +47,7 @@
             notty
             notty.unix
             nottui
+            nottui-unix
             lwd
             oseq
             eio
diff --git a/bin/ui_base.ml b/bin/ui_base.ml
index f56ee8d..8d8ed70 100644
--- a/bin/UI_base.ml
+++ b/bin/UI_base.ml
@@ -587,7 +587,7 @@ let ui_loop ~quit ~term root =
       if term_width <> prev_term_width || term_height <> prev_term_height then (
         Lwd.set Vars.term_width_height (term_width, term_height)
       );
-      Nottui.Ui_loop.step
+      Nottui_unix.step
         ~process_event:true
         ~timeout:0.05
         ~renderer
