diff --git a/Source/kwsys/ProcessUNIX.c b/Source/kwsys/ProcessUNIX.c
index ce6bb7b9cf..9ffc32f039 100644
--- a/Source/kwsys/ProcessUNIX.c
+++ b/Source/kwsys/ProcessUNIX.c
@@ -2501,20 +2501,20 @@ static pid_t kwsysProcessFork(kwsysProcess* cp,
    have two integers to store: the pid and then the ppid.  */
 #if defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__) ||       \
   defined(__OpenBSD__) || defined(__GLIBC__) || defined(__GNU__)
-#  define KWSYSPE_PS_COMMAND "ps axo pid,ppid"
+#  define KWSYSPE_PS_COMMAND "@ps@ axo pid,ppid"
 #  define KWSYSPE_PS_FORMAT "%d %d\n"
 #elif defined(__sun) && (defined(__SVR4) || defined(__svr4__)) /* Solaris */
-#  define KWSYSPE_PS_COMMAND "ps -e -o pid,ppid"
+#  define KWSYSPE_PS_COMMAND "@ps@ -e -o pid,ppid"
 #  define KWSYSPE_PS_FORMAT "%d %d\n"
 #elif defined(__hpux) || defined(__sun__) || defined(__sgi) ||                \
   defined(_AIX) || defined(__sparc)
-#  define KWSYSPE_PS_COMMAND "ps -ef"
+#  define KWSYSPE_PS_COMMAND "@ps@ -ef"
 #  define KWSYSPE_PS_FORMAT "%*s %d %d %*[^\n]\n"
 #elif defined(__QNX__)
-#  define KWSYSPE_PS_COMMAND "ps -Af"
+#  define KWSYSPE_PS_COMMAND "@ps@ -Af"
 #  define KWSYSPE_PS_FORMAT "%*d %d %d %*[^\n]\n"
 #elif defined(__CYGWIN__)
-#  define KWSYSPE_PS_COMMAND "ps aux"
+#  define KWSYSPE_PS_COMMAND "@ps@ aux"
 #  define KWSYSPE_PS_FORMAT "%d %d %*[^\n]\n"
 #endif
 
diff --git a/Source/kwsys/SystemInformation.cxx b/Source/kwsys/SystemInformation.cxx
index aca032459e..093c9aac0c 100644
--- a/Source/kwsys/SystemInformation.cxx
+++ b/Source/kwsys/SystemInformation.cxx
@@ -3850,7 +3850,7 @@ long long SystemInformationImplementation::GetProcMemoryUsed()
   long long memUsed = 0;
   pid_t pid = getpid();
   std::ostringstream oss;
-  oss << "ps -o rss= -p " << pid;
+  oss << "@ps@ -o rss= -p " << pid;
   FILE* file = popen(oss.str().c_str(), "r");
   if (!file) {
     return -1;
