diff --git i/core/util_test.go w/core/util_test.go
index 8979edff..cfae13a4 100644
--- i/core/util_test.go
+++ w/core/util_test.go
@@ -56,10 +56,6 @@ func TestSerialUtils(t *testing.T) {
 	fmt.Println(badSerial)
 }
 
-func TestBuildID(t *testing.T) {
-	test.AssertEquals(t, "Unspecified", GetBuildID())
-}
-
 const JWK1JSON = `{
   "kty": "RSA",
   "n": "vuc785P8lBj3fUxyZchF_uZw6WtbxcorqgTyq-qapF5lrO1U82Tp93rpXlmctj6fyFHBVVB5aXnUHJ7LZeVPod7Wnfl8p5OyhlHQHC8BnzdzCqCMKmWZNX5DtETDId0qzU7dPzh0LP0idt5buU7L9QNaabChw3nnaL47iu_1Di5Wp264p2TwACeedv2hfRDjDlJmaQXuS8Rtv9GnRWyC9JBu7XmGvGDziumnJH7Hyzh3VNu-kSPQD3vuAFgMZS6uUzOztCkT0fpOalZI6hqxtWLvXUMj-crXrn-Maavz8qRhpAyp5kcYk3jiHGgQIi7QSK2JIdRJ8APyX9HlmTN5AQ",
