diff --git i/CMake/config.cmake w/CMake/config.cmake
index 28f8047..c596cf8 100644
--- i/CMake/config.cmake
+++ w/CMake/config.cmake
@@ -95,7 +95,7 @@ else()
     find_package(LibXml2 REQUIRED)
     find_package(CURL REQUIRED)
     set(Boost_NO_BOOST_CMAKE ON)
-    set(Boost_USE_STATIC_LIBS ON)
+    set(Boost_USE_STATIC_LIBS OFF)
     find_package(Boost COMPONENTS program_options filesystem system thread regex date_time REQUIRED)
 
     # add include directories
diff --git i/src/libsrcml/srcml_reader_handler.hpp w/src/libsrcml/srcml_reader_handler.hpp
index 0b23fed..c02dfef 100644
--- i/src/libsrcml/srcml_reader_handler.hpp
+++ w/src/libsrcml/srcml_reader_handler.hpp
@@ -456,7 +456,7 @@ public :
 
             if(uri == SRCML_CPP_NS_URI) {
 
-                if(archive->language != 0) {
+                if(srcml_check_language(archive->language->c_str()) != 0) {
 
                     if(*archive->language == "C++" || *archive->language == "C" || *archive->language == "Objective-C")
                         archive->options |= SRCML_OPTION_CPP | SRCML_OPTION_CPP_NOMACRO;
