diff -rc read.c read.c
*** read.c	2006-03-17 15:24:20.000000000 +0100
--- read.c	2007-05-24 17:16:31.000000000 +0200
***************
*** 99,107 ****
--- 99,109 ----
  #endif
      INCLUDEDIR,
  #ifndef _AMIGA
+ #if 0    
      "/usr/gnu/include",
      "/usr/local/include",
      "/usr/include",
+ #endif    
  #endif
      0
    };
diff -rc reremake.c
*** remake.c	2006-03-20 03:36:37.000000000 +0100
--- remake.c	2007-05-24 17:06:54.000000000 +0200
***************
*** 1452,1460 ****
--- 1452,1462 ----
    static char *dirs[] =
      {
  #ifndef _AMIGA
+ #if 0
        "/lib",
        "/usr/lib",
  #endif
+ #endif
  #if defined(WINDOWS32) && !defined(LIBDIR)
  /*
   * This is completely up to the user at product install time. Just define
