diff --git a/overlayunit.pas b/overlayunit.pas
index 8c2276d..a62e60a 100644
--- a/overlayunit.pas
+++ b/overlayunit.pas
@@ -4913,7 +4913,7 @@ begin
    //Determine Mangohud dependency status
 
           //locate MangoHud and store result in tmp folder
-          RunCommand('bash -c ''find /usr/share/vulkan/implicit_layer.d/MangoHud.json >> /tmp/goverlay/dependency_mangohud''', s);
+          RunCommand('bash -c ''find-xdg-data-files vulkan/implicit_layer.d/MangoHud.json >> /tmp/goverlay/dependency_mangohud''', s);
 
           // Assign Text file dependency_mangohud to variable mangohudVAR
           AssignFile(mangohudVAR, '/tmp/goverlay/dependency_mangohud');
@@ -4922,7 +4922,7 @@ begin
           CloseFile(mangohudVAR);
 
           // Read String and store value on mangohuddependencyVALUE based on result
-          if mangohudSTR = '/usr/share/vulkan/implicit_layer.d/MangoHud.json' then
+          if mangohudSTR <> '' then
           mangohuddependencyVALUE := 1
           else
           mangohuddependencyVALUE := 0;
@@ -4931,7 +4931,7 @@ begin
    //Determine vkBasalt dependency staus
 
            //locate vkBasalt and store result in tmp folder
-           RunCommand('bash -c ''find /usr/share/vulkan/implicit_layer.d/vkBasalt.json >> /tmp/goverlay/dependency_vkbasalt''', s);
+           RunCommand('bash -c ''find-xdg-data-files vulkan/implicit_layer.d/vkBasalt.json >> /tmp/goverlay/dependency_vkbasalt''', s);
 
            // Assign Text file dependency_mangohud to variable mangohudVAR
            AssignFile(vkbasaltVAR, '/tmp/goverlay/dependency_vkbasalt');
@@ -4940,7 +4940,7 @@ begin
            CloseFile(vkbasaltVAR);
 
            // Read String and store value on vkbasaltdependencyVALUE based on result
-           if vkbasaltSTR = '/usr/share/vulkan/implicit_layer.d/vkBasalt.json' then
+           if vkbasaltSTR <> '' then
            vkbasaltdependencyVALUE := 1
            else
            vkbasaltdependencyVALUE := 0;
