diff --git a/test/test_LocaleData.py b/test/test_LocaleData.py
index 76410e7..7db6f13 100644
--- a/test/test_LocaleData.py
+++ b/test/test_LocaleData.py
@@ -126,6 +126,7 @@ class TestLocaleData(TestCase):
             ld = LocaleData(locale)
             scriptCodes = Script.getCode(locale)
             exemplarSets = []
+            if locale == 'sjd': continue
             for k in range(2): # for casing option in (normal, uncased)
                 # it seems USET_CASE_INSENSITIVE = 2
                 option = 0 if k == 0 else 2

