--- createrepo_c-0.10.0-src.orig/src/python/CMakeLists.txt	2017-03-19 10:50:33.796342953 +0100
+++ createrepo_c-0.10.0-src/src/python/CMakeLists.txt	2017-03-19 10:53:51.207580073 +0100
@@ -19,7 +19,7 @@
     FIND_PACKAGE(PythonInterp 3.0 REQUIRED)
 endif()
 
-EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c "from sys import stdout; from distutils import sysconfig; stdout.write(sysconfig.get_python_lib(True))" OUTPUT_VARIABLE PYTHON_INSTALL_DIR)
+SET(PYTHON_INSTALL_DIR "@PYTHON_INSTALL_DIR@")
 INCLUDE_DIRECTORIES (${PYTHON_INCLUDE_PATH})
 
 MESSAGE(STATUS "Python install dir is ${PYTHON_INSTALL_DIR}")
