diff --git a/sqlite/gnatcoll_sqlite.gpr b/sqlite/gnatcoll_sqlite.gpr
index 5bd53d35..580739f8 100644
--- a/sqlite/gnatcoll_sqlite.gpr
+++ b/sqlite/gnatcoll_sqlite.gpr
@@ -69,7 +69,12 @@ project GnatColl_Sqlite is
          for Source_Dirs use (".", "amalgamation");
       when "external" =>
          for Source_Dirs use (".");
-         for Library_Options use ("-lsqlite3") & Thread_Lib;
+         case Library_Type is
+            when "relocatable" =>
+               for Library_Options use ("-lsqlite3") & Thread_Lib;
+            when others =>
+               null;
+         end case;
    end case;
 
    package Compiler is
