--- a/glfw/egl_context.c
+++ b/glfw/egl_context.c
@@ -314,7 +314,7 @@ bool _glfwInitEGL(void)
 #elif defined(__CYGWIN__)
         "libEGL-1.so",
 #else
-        "libEGL.so.1",
+        "@libEGL@",
 #endif
         NULL
     };

--- a/kitty/desktop.c
+++ b/kitty/desktop.c
@@ -34,10 +34,7 @@ init_x11_startup_notification(PyObject UNUSED *self, PyObject *args) {
         done = true;
 
         const char* libnames[] = {
-            "libstartup-notification-1.so",
-            // some installs are missing the .so symlink, so try the full name
-            "libstartup-notification-1.so.0",
-            "libstartup-notification-1.so.0.0.0",
+            "@libstartup_notification@",
             NULL
         };
         for (int i = 0; libnames[i]; i++) {
@@ -113,10 +110,7 @@ load_libcanberra(void) {
     if (done) return;
     done = true;
     const char* libnames[] = {
-        "libcanberra.so",
-        // some installs are missing the .so symlink, so try the full name
-        "libcanberra.so.0",
-        "libcanberra.so.0.2.5",
+        "@libcanberra@",
         NULL
     };
     for (int i = 0; libnames[i]; i++) {
