--- a/setup.py
+++ b/setup.py
@@ -202,7 +202,7 @@ def init_env(
     cflags = os.environ.get(
         'OVERRIDE_CFLAGS', (
             '-Wextra -Wno-missing-field-initializers -Wall -std=c11'
-            ' -pedantic-errors -Werror {} {} -fwrapv {} {} -pipe {} -fvisibility=hidden'
+            ' {} {} -fwrapv {} {} -pipe {} -fvisibility=hidden'
         ).format(
             optimize,
             ' '.join(sanitize_args),
