--- a/setup.py
+++ b/setup.py
@@ -217,7 +217,7 @@ def init_env(
     cflags = os.environ.get(
         'OVERRIDE_CFLAGS', (
             '-Wextra {} -Wno-missing-field-initializers -Wall -Wstrict-prototypes -std=c11'
-            ' -pedantic-errors -Werror {} {} -fwrapv {} {} -pipe {} -fvisibility=hidden'
+            ' {} {} -fwrapv {} {} -pipe {} -fvisibility=hidden'
         ).format(
             float_conversion,
             optimize,
