Tell the elisp code about the right Java directory.

--- jde/lisp/jde.el	(revision 90)
+++ jde/lisp/jde.el	(working copy)
@@ -2448,17 +2448,14 @@ environment variable."
 (defmethod initialize-instance ((this jde-bsh) &rest fields)
   "Constructor for the JDEE BeanShell instance."
   (call-next-method)
-  (let* ((jde-java-directory
-          (concat
-           (jde-find-jde-data-directory)
-           "java/")))
-
-    (oset this bsh-cmd-dir (expand-file-name "bsh-commands" jde-java-directory))
-    (oset this checkstyle-jar  (expand-file-name "lib/checkstyle-all.jar" jde-java-directory))
-    (oset this regexp-jar (expand-file-name "lib/jakarta-regexp.jar" jde-java-directory))
+  (let ((jde-java-directory "@out@/share/java"))
+
+    (oset this bsh-cmd-dir "@datadir@/bsh-commands")
+    (oset this checkstyle-jar  (expand-file-name "checkstyle-all.jar" jde-java-directory))
+    (oset this regexp-jar (expand-file-name "jakarta-regexp.jar" jde-java-directory))
     (oset this jde-classes-dir (expand-file-name "classes" jde-java-directory))
-    (oset this jde-jar (expand-file-name "lib/jde.jar" jde-java-directory))
-    (oset this jar  (expand-file-name "lib/bsh.jar" jde-java-directory))
+    (oset this jde-jar (expand-file-name "jde.jar" jde-java-directory))
+    (oset this jar  (expand-file-name "bsh.jar" jde-java-directory))
     (oset-default 'jde-bsh the-bsh this)))
 
 (defmethod bsh-create-buffer ((this jde-bsh))

--- jde/lisp/jde-checkstyle.el	(revision 90)
+++ jde/lisp/jde-checkstyle.el	(working copy)
@@ -316,10 +316,7 @@ string describing how the compilation fi
 	 (vm-path (oref (jde-run-get-vm) :path))
 	 (source-file 
 	  (concat (file-name-nondirectory buffer-file-name)))
-	 (jde-java-directory
-	  (concat
-	   (jde-find-jde-data-directory)
-	   "java/"))
+	 (jde-java-directory "@out@/share/java")
 	 (args (append
 		(unless jde-checkstyle-expanded-properties-file
 		  (jde-checkstyle-get-property-args this))
@@ -328,12 +325,12 @@ string describing how the compilation fi
 		      (if jde-checkstyle-classpath
 			  (jde-build-classpath jde-checkstyle-classpath)
 			(jde-normalize-path
-			 (expand-file-name "lib/checkstyle-all.jar" jde-java-directory))))
+			 (expand-file-name "checkstyle-all.jar" jde-java-directory))))
 		(list jde-checkstyle-class)		
 		(list "-c" 
 		      (if jde-checkstyle-style
 			  (jde-normalize-path jde-checkstyle-style)
-			(concat (jde-find-jde-data-directory) "java/lib/sun_checks.xml")))
+			"@datadir@/sun_checks.xml"))
 		(if jde-checkstyle-expanded-properties-file
                     (list "-p" (jde-normalize-path jde-checkstyle-expanded-properties-file)))
 		(if jde-checkstyle-module-package-names-file

--- jde/lisp/jde-dbs.el	(revision 90)
+++ jde/lisp/jde-dbs.el	(working copy)
@@ -899,9 +899,7 @@ for the breakpoint."
 		  (jde-normalize-path 'jde-run-working-directory)
 		source-directory))	     
              (vm (oref (jde-run-get-vm) :path))
-	     (jde-java-directory
-	      (expand-file-name "java"
-	       (jde-find-jde-data-directory)))
+	     (jde-java-directory "@out@/share/java")
 	     (vm-args 
 		(let (args)
 		  (setq args 
@@ -912,7 +910,7 @@ for the breakpoint."
 			  (jde-build-classpath
 			       (list
 				 (expand-file-name 
-				  (if jde-bug-debug "classes" "lib/jde.jar")
+				  (if jde-bug-debug "classes" "jde.jar")
 				  jde-java-directory)
 				 (if (jde-bug-vm-includes-jpda-p)
                                    (jde-get-tools-jar)
