diff -rc valgrind-orig/coregrind/vg_unsafe.h valgrind-2.2.0/coregrind/vg_unsafe.h
*** valgrind-orig/coregrind/vg_unsafe.h	2004-08-29 23:02:28.000000000 +0200
--- valgrind-2.2.0/coregrind/vg_unsafe.h	2005-02-23 21:39:25.000000000 +0100
***************
*** 31,36 ****
--- 31,40 ----
  */
  
  
+ #ifndef pgoff_t
+ #define pgoff_t unsigned long
+ #endif
+ 
  /* These includes are only used for making sense of the args for
     system calls. */
  #include "vg_unistd.h"    /* for system call numbers */
***************
*** 47,53 ****
  #include <sys/socket.h>   /* for struct msghdr */
  #include <linux/sockios.h>/* for SIOCOUTQ */
  #include <sys/un.h>       /* for sockaddr_un */
! #include <net/if.h>       /* for struct ifreq et al */
  #include <net/if_arp.h>   /* for struct arpreq */
  #include <net/route.h>    /* for struct rtentry */
  #ifdef HAVE_LINUX_COMPILER_H
--- 51,58 ----
  #include <sys/socket.h>   /* for struct msghdr */
  #include <linux/sockios.h>/* for SIOCOUTQ */
  #include <sys/un.h>       /* for sockaddr_un */
! /* hack - should be fixed in CVS */
! //#include <net/if.h>       /* for struct ifreq et al */
  #include <net/if_arp.h>   /* for struct arpreq */
  #include <net/route.h>    /* for struct rtentry */
  #ifdef HAVE_LINUX_COMPILER_H
