Get protobuf include path from environment

--- a/python/Makefile
+++ b/python/Makefile
@@ -5,7 +5,7 @@ all: ct/proto/client_pb2.py ct/proto/ct_pb2.py ct/proto/tls_options_pb2.py \
 	ct/proto/test_message_pb2.py ct/proto/certificate_pb2.py
 
 ct/proto/%_pb2.py: ct/proto/%.proto
-	$(PROTOC) $^ -I/usr/include/ -I/usr/local/include -I$(INSTALL_DIR)/include -I. --python_out=.
+	$(PROTOC) $^ $(protocFlags) -I. --python_out=.
 
 ct/proto/ct_pb2.py: ../proto/ct.proto
 	$(PROTOC) --python_out=ct/proto -I../proto ../proto/ct.proto

