diff --git a/src/common/ConfigReader.cpp b/src/common/ConfigReader.cpp
index 6618455..5356e76 100644
--- a/src/common/ConfigReader.cpp
+++ b/src/common/ConfigReader.cpp
@@ -136,11 +136,6 @@ namespace SDDM {
         QString currentSection = QStringLiteral(IMPLICIT_SECTION);
 
         QFile in(m_path);
-        QDateTime modificationTime = QFileInfo(in).lastModified();
-        if (modificationTime <= m_fileModificationTime) {
-            return;
-        }
-        m_fileModificationTime = modificationTime;
 
         in.open(QIODevice::ReadOnly);
         while (!in.atEnd()) {
