diff -p1 -aur ../omake-0.9.8.6.ori/lib/build/OCaml.om ./lib/build/OCaml.om
--- ../omake-0.9.8.6.ori/lib/build/OCaml.om	2008-03-05 01:07:25.000000000 +0000
+++ ./lib/build/OCaml.om	2013-06-01 15:52:37.000000000 +0000
@@ -178,3 +178,3 @@ declare OCAMLDEPFLAGS
 public.OCAMLPPFLAGS  =
-public.OCAMLFLAGS    = -warn-error A
+public.OCAMLFLAGS    =
 public.OCAMLCFLAGS   = -g
Seulement dans ./lib/build: OCaml.om~
Seulement dans .: warn.patch
