diff --git a/setup.py b/setup.py
index 4092388..6cb7dac 100644
--- a/setup.py
+++ b/setup.py
@@ -50,19 +50,19 @@ ext_modules = [LateIncludeExtension("friture_extensions.exp_smoothing_conv",
 # these will be installed when calling 'pip install friture'
 # they are also retrieved by 'requirements.txt'
 install_requires = [
-    "sounddevice==0.4.2",
-    "rtmixer==0.1.3",
-    "PyOpenGL==3.1.5",
-    "PyOpenGL-accelerate==3.1.5",
-    "docutils==0.17.1",
-    "numpy==1.21.1",
-    "PyQt5==5.15.4",
-    "appdirs==1.4.4",
-    "pyrr==0.10.3",
+    "sounddevice>=0.4.1",
+    "rtmixer>=0.1.1",
+    "PyOpenGL>=3.1.4",
+    "PyOpenGL-accelerate>=3.1.5",
+    "docutils>=0.17.1",
+    "numpy>=1.20.3",
+    "PyQt5>=5.15.4",
+    "appdirs>=1.4.4",
+    "pyrr>=0.10.3",
 ]
 
 # Cython and numpy are needed when running setup.py, to build extensions
-setup_requires=["numpy==1.21.1", "Cython==0.29.24"]
+setup_requires=["numpy>=1.20.3", "Cython>=0.29.22"]
 
 with open(join(dirname(__file__), 'README.rst')) as f:
     long_description = f.read()
