ndex 518c6e95..6061047b 100644
--- a/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift
+++ b/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift
@@ -10,6 +10,7 @@
 //
 //===----------------------------------------------------------------------===//
 
+import Foundation
 import SwiftOptions
 
 import func TSCBasic.lookupExecutablePath
@@ -126,7 +127,20 @@ extension GenericUnixToolchain {
       }
 
       let clangTool: Tool = cxxCompatEnabled ? .clangxx : .clang
-      var clangPath = try getToolPath(clangTool)
+
+      // For Nix, prefer linking using the wrapped system clang, instead of using
+      // the unwrapped clang packaged with swift. The latter is unable to link, but
+      // we still want to use it for other purposes (clang importer).
+      var clangPath: AbsolutePath
+      let env = ProcessInfo.processInfo.environment
+      if let nixCC = env["NIX_CC"],
+         let binPath = try? AbsolutePath(validating: "\(nixCC)/bin"),
+         let tool = lookupExecutablePath(filename: cxxCompatEnabled
+                                                        ? "clang++" : "clang",
+                                         searchPaths: [binPath]) {
+        clangPath = tool
+      } else {
+      clangPath = try getToolPath(clangTool)
       if let toolsDirPath = parsedOptions.getLastArgument(.toolsDirectory) {
         // FIXME: What if this isn't an absolute path?
         let toolsDir = try AbsolutePath(validating: toolsDirPath.asSingle)
@@ -142,6 +156,7 @@ extension GenericUnixToolchain {
         commandLine.appendFlag("-B")
         commandLine.appendPath(toolsDir)
       }
+      } // nixCC
 
       // Executables on Linux get -pie
       if targetTriple.os == .linux && linkerOutputType == .executable {
