diff --git a/ZConfig/tests/test_schema2html.py b/ZConfig/tests/test_schema2html.py
index 838cf7c..52daf4e 100644
--- a/ZConfig/tests/test_schema2html.py
+++ b/ZConfig/tests/test_schema2html.py
@@ -74,6 +74,7 @@ else:
 
 class TestSchema2HTML(unittest.TestCase):
 
+    @unittest.skip('broken test (https://github.com/zopefoundation/ZConfig/issues/34)')
     def test_no_schema(self):
         self.assertRaises(SystemExit,
                           run_transform)
