diff -rc w3m-0.5.1-orig/configure w3m-0.5.1/configure
*** w3m-0.5.1-orig/configure	2004-04-28 20:19:24.000000000 +0200
--- w3m-0.5.1/configure	2005-09-16 11:33:48.000000000 +0200
***************
*** 5642,5648 ****
   echo "$as_me:$LINENO: checking for -l$lib" >&5
  echo $ECHO_N "checking for -l$lib... $ECHO_C" >&6
   extlib="not found"
!  for dir in /lib /usr/lib /usr/local/lib /usr/ucblib /usr/ccslib /usr/ccs/lib
   do
     if test -f $dir/lib$lib.a -o -f $dir/lib$lib.so ; then
      LIBS="$LIBS -l$lib"
--- 5642,5648 ----
   echo "$as_me:$LINENO: checking for -l$lib" >&5
  echo $ECHO_N "checking for -l$lib... $ECHO_C" >&6
   extlib="not found"
!  for dir in /no-such-path
   do
     if test -f $dir/lib$lib.a -o -f $dir/lib$lib.so ; then
      LIBS="$LIBS -l$lib"
***************
*** 5768,5774 ****
   if test x"$enable_ipv6" = xno; then
      echo "$as_me:$LINENO: checking for libinet6" >&5
  echo $ECHO_N "checking for libinet6... $ECHO_C" >&6
!     for dir in /usr/local/v6/lib /usr/local/lib /usr/lib
      do
  	if test -f $dir/libinet6.a; then
  	  if test $dir != "/usr/lib"; then
--- 5768,5774 ----
   if test x"$enable_ipv6" = xno; then
      echo "$as_me:$LINENO: checking for libinet6" >&5
  echo $ECHO_N "checking for libinet6... $ECHO_C" >&6
!     for dir in /no-such-path
      do
  	if test -f $dir/libinet6.a; then
  	  if test $dir != "/usr/lib"; then
