From 1faa30525c9671ffd3a08901896b521a040d7e5c Mon Sep 17 00:00:00 2001
From: Nick Cao <nickcao@nichi.co>
Date: Tue, 20 Sep 2022 18:42:08 +0800
Subject: [PATCH 1/4] skip symlink system libraries

---
 base/Makefile | 2 --
 1 file changed, 2 deletions(-)

diff --git a/base/Makefile b/base/Makefile
index 23a9c4011..12f92aa05 100644
--- a/base/Makefile
+++ b/base/Makefile
@@ -181,7 +181,6 @@ $$(build_private_libdir)/$$(libname_$2):
 		fi; \
 	fi
 ifneq ($$(USE_SYSTEM_$1),0)
-SYMLINK_SYSTEM_LIBRARIES += symlink_$2
 endif
 endef
 
@@ -265,7 +264,6 @@ $(build_private_libdir)/libLLVM.$(SHLIB_EXT):
 	ln -sf "$$REALPATH" "$@"
 ifneq ($(USE_SYSTEM_LLVM),0)
 ifneq ($(USE_LLVM_SHLIB),0)
-SYMLINK_SYSTEM_LIBRARIES += symlink_libLLVM
 endif
 endif
 
-- 
2.38.1

