diff --git a/Make.inc b/Make.inc
index 24f5964e9d..633fdfaea3 100644
--- a/Make.inc
+++ b/Make.inc
@@ -156,7 +156,7 @@ endif
 .SUFFIXES:
 
 # find out if git repository is available
-ifeq ($(shell [ -e $(JULIAHOME)/.git ] && echo true || echo "Warning: git information unavailable; versioning information limited" >&2), true)
+ifeq ($(shell [ -e $(JULIAHOME)/.git ]), true)
 NO_GIT := 0
 else
 NO_GIT := 1