diff --git a/meson.build b/meson.build
index 155c9e7..9125372 100644
--- a/meson.build
+++ b/meson.build
@@ -60,15 +60,15 @@ sh = find_program('sh')
 sed = find_program('sed')
 test = find_program('test')
 
-ret = run_command ([test, '-e', join_paths(glib_docpath, 'glib/index.html')])
-if ret.returncode() != 0
-  error('Missing documentation for GLib.')
-endif
-
-ret = run_command ([test, '-e', join_paths(glib_docpath, 'gobject/index.html')])
-if ret.returncode() != 0
-  error('Missing documentation for GObject.')
-endif
+# ret = run_command ([test, '-e', join_paths(glib_docpath, 'glib/index.html')])
+# if ret.returncode() != 0
+#   error('Missing documentation for GLib.')
+# endif
+
+# ret = run_command ([test, '-e', join_paths(glib_docpath, 'gobject/index.html')])
+# if ret.returncode() != 0
+#   error('Missing documentation for GObject.')
+# endif
 
 python_name = get_option('python_name')
 
