diff --git a/meson.build b/meson.build
index a8b02b4..dd31a76 100644
--- a/meson.build
+++ b/meson.build
@@ -67,7 +67,7 @@ test = find_program('test')
 with_docs = get_option('with_docs')
 if with_docs
   gtkdoc = dependency('gtk-doc')
-  glib_docpath = join_paths(glib_prefix, 'share', 'gtk-doc', 'html')
+  glib_docpath = join_paths('@glib_devdoc@', 'share', 'gtk-doc', 'html')
 
   ret = run_command ([test, '-e', join_paths(glib_docpath, 'glib/index.html')])
   if ret.returncode() != 0
