commit 4ec09e6f6e00e40622a5207ed24dc657da9a9090
Author: Pavol Rusnak <stick@gk2.sk>
Date:   Tue Dec 4 12:06:22 2018 +0100

    build: add install: true to executable in meson.build

diff --git a/meson.build b/meson.build
index 050e1b1..9224ed5 100644
--- a/meson.build
+++ b/meson.build
@@ -39,4 +39,5 @@ endforeach
 
 # compile the main project
 executable('luppp', luppp_src + [version_hxx],
+    install: true,
     dependencies: deps)
