diff --git a/src/ch/meson.build b/src/ch/meson.build
index e34974d56c..4767763c2c 100644
--- a/src/ch/meson.build
+++ b/src/ch/meson.build
@@ -68,7 +68,7 @@ if conf.has('WITH_CH')
   }
 
   virt_install_dirs += [
-    localstatedir / 'lib' / 'libvirt' / 'ch',
-    runstatedir / 'libvirt' / 'ch',
+    install_prefix + localstatedir / 'lib' / 'libvirt' / 'ch',
+    install_prefix + runstatedir / 'libvirt' / 'ch',
   ]
 endif
