diff --git a/src/util/virpci.c b/src/util/virpci.c
index 780b4f9eec..0edb848d14 100644
--- a/src/util/virpci.c
+++ b/src/util/virpci.c
@@ -1461,7 +1461,9 @@ virPCIDeviceFindBestVFIOVariant(virPCIDevice *dev,
     uname(&unameInfo);
     modulesAliasPath = g_strdup_printf("/lib/modules/%s/modules.alias", unameInfo.release);
     if (virFileReadAll(modulesAliasPath, 8 * 1024 * 1024, &modulesAliasContent) < 0)
-        return -1;
+        modulesAliasPath = g_strdup_printf("/run/booted-system/kernel-modules/lib/modules/%s/modules.alias", unameInfo.release);
+        if (virFileReadAll(modulesAliasPath, 8 * 1024 * 1024, &modulesAliasContent) < 0)
+            return -1;
 
     /* Look for all lines that are aliases for vfio_pci drivers.
      * (The first line is always a comment, so we can be sure "alias"
