--- zathura-ps-0.2.5.orig/config.mk	2017-12-21 14:21:17.000000000 +0100
+++ zathura-ps-0.2.5/config.mk	2017-12-31 01:05:17.507268817 +0100
@@ -16,6 +16,9 @@
 DESKTOPPREFIX ?= ${PREFIX}/share/applications
 
 # libs
+GTK_INC ?= $(shell pkg-config --cflags gtk+-2.0)
+GTK_LIB ?= $(shell pkg-config --libs gtk+-2.0)
+
 CAIRO_INC ?= $(shell pkg-config --cflags cairo)
 CAIRO_LIB ?= $(shell pkg-config --libs cairo)
 
@@ -34,8 +37,8 @@
 PLUGINDIR = ${LIBDIR}/zathura
 endif
 
-INCS = ${GLIB_INC} ${SPECTRE_INC} ${GIRARA_INC} ${CAIRO_INC} ${ZATHURA_INC}
-LIBS = ${GLIB_LIB} ${SPECTRE_LIB} ${GIRARA_LIB} ${CAIRO_LIB}
+INCS = ${GLIB_INC} ${SPECTRE_INC} ${GIRARA_INC} ${CAIRO_INC} ${ZATHURA_INC} ${GTK_INC}
+LIBS = ${GLIB_LIB} ${SPECTRE_LIB} ${GIRARA_LIB} ${CAIRO_LIB } ${GTK_LIB}
 
 # compiler flags
 CFLAGS += -std=c11 -fPIC -pedantic -Wall -Wno-format-zero-length $(INCS)
