diff --git a/dotnetcore2/runtime.py b/dotnetcore2/runtime.py
index 475e2b4..5b578ec 100644
--- a/dotnetcore2/runtime.py
+++ b/dotnetcore2/runtime.py
@@ -41,6 +41,7 @@ def _get_bin_folder() -> str:
 
 
 def get_runtime_path():
+    return "@dotnet@/dotnet"
     search_string = os.path.join(_get_bin_folder(), 'dotnet*')
     matches = [f for f in glob.glob(search_string, recursive=True)]
     return matches[0]
@@ -96,8 +97,7 @@ class _FileLock():
 
 
 def ensure_dependencies() -> Optional[str]:
-    if dist is None:
-        return None
+    return None
 
     bin_folder = _get_bin_folder()
     deps_path = os.path.join(bin_folder, 'deps')
