diff -Naur qcad-2.0.4.0-1.src.orig/qcadlib/src/engine/rs_entity.cpp qcad-2.0.4.0-1.src.patched/qcadlib/src/engine/rs_entity.cpp
--- qcad-2.0.4.0-1.src.orig/qcadlib/src/engine/rs_entity.cpp	2004-09-14 15:13:02.000000000 -0500
+++ qcad-2.0.4.0-1.src.patched/qcadlib/src/engine/rs_entity.cpp	2006-06-23 14:21:40.000000000 -0500
@@ -849,7 +849,7 @@
         os << " layer: NULL ";
     } else {
         os << " layer: " << e.layer->getName().latin1() << " ";
-        os << " layer address: " << (int)(e.layer) << " ";
+        os << " layer address: " << (intptr_t)(e.layer) << " ";
     }
 
     os << e.pen << "\n";
diff -Naur qcad-2.0.4.0-1.src.orig/qcadlib/src/engine/rs_layer.cpp qcad-2.0.4.0-1.src.patched/qcadlib/src/engine/rs_layer.cpp
--- qcad-2.0.4.0-1.src.orig/qcadlib/src/engine/rs_layer.cpp	2004-09-14 15:13:02.000000000 -0500
+++ qcad-2.0.4.0-1.src.patched/qcadlib/src/engine/rs_layer.cpp	2006-06-23 14:21:23.000000000 -0500
@@ -57,7 +57,7 @@
     os << " name: " << l.getName().latin1()
     << " pen: " << l.getPen()
 	<< " frozen: " << (int)l.isFrozen()
-	<< " address: " << (int)(&l)
+	<< " address: " << (intptr_t)(&l)
     << std::endl;
     return os;
 }
