diff -ru a/modules/bindings/glibc/linux.lisp b/modules/bindings/glibc/linux.lisp
--- a/modules/bindings/glibc/linux.lisp	2008-10-10 16:15:49.000000000 +0300
+++ b/modules/bindings/glibc/linux.lisp	2012-12-04 01:01:35.000000000 +0200
@@ -86,7 +86,7 @@
 
 (def-c-type __key_t)            ; int
 
-(c-lines "#include <bits/ipctypes.h>~%")
+(c-lines "#include <sys/ipc.h>~%")
 (def-c-type __ipc_pid_t)        ; ushort
 
 ; --------------------------- <sys/types.h> -----------------------------------
