diff -ru3 graphviz.old/Data/GraphViz/Algorithms.hs graphviz/Data/GraphViz/Algorithms.hs
--- graphviz.old/Data/GraphViz/Algorithms.hs	2015-05-18 15:21:38.379771357 +0300
+++ graphviz/Data/GraphViz/Algorithms.hs	2015-05-18 15:01:01.940122684 +0300
@@ -38,6 +38,7 @@
 import Data.GraphViz.Types.Canonical
 import Data.GraphViz.Types.Internal.Common
 
+import           Prelude                   hiding (traverse)
 import           Control.Arrow             (first, second, (***))
 import           Control.Monad             (unless)
 import           Control.Monad.Trans.State
