From 06b12ab8a733d4de2a39060ba29c06e4ec1c1187 Mon Sep 17 00:00:00 2001
From: Anton Latukha <anton.latukha@gmail.com>
Date: Sun, 16 Jan 2022 18:16:50 +0200
Subject: [PATCH] fix aeson <2 support

---
 src/Nix/Json.hs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Nix/Json.hs b/src/Nix/Json.hs
index 3fe05269..329899e4 100644
--- a/src/Nix/Json.hs
+++ b/src/Nix/Json.hs
@@ -9,6 +9,8 @@ import qualified Data.Text.Lazy.Encoding       as TL
 #if MIN_VERSION_aeson(2,0,0)
 import qualified Data.Aeson.Key                as AKM
 import qualified Data.Aeson.KeyMap             as AKM
+#else
+import           Nix.Expr.Types
 #endif
 import qualified Data.Vector                   as V
 import           Nix.Atoms
-- 
2.34.1
