diff -ru phonetic-languages-phonetics-basics-0.5.1.0/Data/Phonetic/Languages/SpecificationsRead.hs phonetic-languages-phonetics-basics-0.5.1.0/Data/Phonetic/Languages/SpecificationsRead.hs
--- phonetic-languages-phonetics-basics-0.5.1.0/Data/Phonetic/Languages/SpecificationsRead.hs	2021-04-30 17:45:52.000000000 +0200
+++ phonetic-languages-phonetics-basics-0.5.1.0/Data/Phonetic/Languages/SpecificationsRead.hs	2021-05-08 18:16:15.054951952 +0200
@@ -1,11 +1,11 @@
--- |
--- Module      :  Data.Phonetic.Languages.SpecificationsRead
--- Copyright   :  (c) OleksandrZhabenko 2021
--- License     :  MIT
--- Stability   :  Experimental
--- Maintainer  :  olexandr543@yahoo.com
---
-{-| Provides functions to read data specifications for other modules from textual files.
+{-|
+Module      :  Data.Phonetic.Languages.SpecificationsRead
+Copyright   :  (c) OleksandrZhabenko 2021
+License     :  MIT
+Stability   :  Experimental
+Maintainer  :  olexandr543@yahoo.com
+
+Provides functions to read data specifications for other modules from textual files.
 -}
 
 module Data.Phonetic.Languages.SpecificationsRead where
diff -ru phonetic-languages-phonetics-basics-0.5.1.0/Main.hs phonetic-languages-phonetics-basics-0.5.1.0/Main.hs
--- phonetic-languages-phonetics-basics-0.5.1.0/Main.hs	2021-04-30 17:45:52.000000000 +0200
+++ phonetic-languages-phonetics-basics-0.5.1.0/Main.hs	2021-05-08 18:14:06.344145599 +0200
@@ -1,11 +1,11 @@
--- |
--- Module      :  Main
--- Copyright   :  (c) OleksandrZhabenko 2020-2021
--- License     :  MIT
--- Stability   :  Experimental
--- Maintainer  :  olexandr543@yahoo.com
---
-{-| Can be used to calculate the durations of the approximations of the phonemes
+{-|
+Module      :  Main
+Copyright   :  (c) OleksandrZhabenko 2020-2021
+License     :  MIT
+Stability   :  Experimental
+Maintainer  :  olexandr543@yahoo.com
+
+Can be used to calculate the durations of the approximations of the phonemes
 using some prepared text with its correct (at least mostly) pronunciation.
 The prepared text is located in the same directory and contains lines -the
 phonetic language word and its duration in seconds separated with whitespace.
