--- regex-tdfa-text-1.0.0.2/Text/Regex/TDFA/Text/Lazy.orig.hs	2015-08-05 20:30:01.228983428 +0100
+++ regex-tdfa-text-1.0.0.2/Text/Regex/TDFA/Text/Lazy.hs	2015-08-05 20:39:25.682563005 +0100
@@ -26,7 +26,7 @@
 import Data.Array.IArray((!),elems,amap)
 import qualified Data.Text.Lazy as L(Text,empty,take,drop,uncons,unpack)

-import Text.Regex.Base(MatchArray,RegexContext(..),Extract(..),RegexMaker(..),RegexLike(..))
+import Text.Regex.Base(MatchText,MatchArray,RegexContext(..),Extract(..),RegexMaker(..),RegexLike(..))
 import Text.Regex.Base.Impl(polymatch,polymatchM)
 import Text.Regex.TDFA.ReadRegex(parseRegex)
 import Text.Regex.TDFA.String() -- piggyback on RegexMaker for String
@@ -74,7 +74,8 @@
                ,after (o+l) source))
          (matchOnce regex source)
   matchAllText regex source =
-    let go i _ _ | i `seq` False = undefined
+    let go :: Int -> L.Text -> [MatchArray] -> [MatchText L.Text]
+        go i _ _ | i `seq` False = undefined
         go _i _t [] = []
         go i t (x:xs) =
           let (off0,len0) = x!0
