diff -urNZ a/garcon-gtk/garcon-gtk-menu.c b/garcon-gtk/garcon-gtk-menu.c
--- a/garcon-gtk/garcon-gtk-menu.c	2017-11-16 19:22:33.551926068 +0000
+++ b/garcon-gtk/garcon-gtk-menu.c	2017-11-16 19:23:33.055497211 +0000
@@ -676,6 +676,9 @@
                   G_CALLBACK (garcon_gtk_menu_deactivate), menu);
               gtk_widget_show (mi);
 
+	      /* submenu are child items, too. */
+	      has_children = TRUE;
+	      
               if (menu->priv->show_menu_icons)
                 {
                   icon_name = garcon_menu_element_get_icon_name (li->data);
Binary files a/.git/index and b/.git/index differ
