--- a/engine/src/flutter/third_party/flatbuffers/include/flatbuffers/util.h
+++ b/engine/src/flutter/third_party/flatbuffers/include/flatbuffers/util.h
@@ -202,7 +202,7 @@
 
 // clang-format off
 // Use locale independent functions {strtod_l, strtof_l, strtoll_l, strtoull_l}.
-#if defined(FLATBUFFERS_LOCALE_INDEPENDENT) && (FLATBUFFERS_LOCALE_INDEPENDENT > 0)
+#if defined(__GLIBC__) && defined(FLATBUFFERS_LOCALE_INDEPENDENT) && (FLATBUFFERS_LOCALE_INDEPENDENT > 0)
   class ClassicLocale {
     #ifdef _MSC_VER
       typedef _locale_t locale_type;
--- a/engine/src/flutter/third_party/flatbuffers/src/util.cpp
+++ b/engine/src/flutter/third_party/flatbuffers/src/util.cpp
@@ -252,7 +252,7 @@
 }
 
 // Locale-independent code.
-#if defined(FLATBUFFERS_LOCALE_INDEPENDENT) && \
+#if defined(__GLIBC__) && defined(FLATBUFFERS_LOCALE_INDEPENDENT) && \
     (FLATBUFFERS_LOCALE_INDEPENDENT > 0)
 
 // clang-format off
