diff --git a/libvmi/Makefile.am b/libvmi/Makefile.am
index cbf3868..bddd215 100644
--- a/libvmi/Makefile.am
+++ b/libvmi/Makefile.am
@@ -123,7 +123,7 @@ lib_LTLIBRARIES= libvmi.la
 libvmi_la_SOURCES= $(h_public) $(h_private) $(drivers) $(os) $(c_sources)
 libvmi_la_CFLAGS= -fvisibility=hidden $(GLIB_CFLAGS)
 libvmi_la_CFLAGS+= -Wall -Wextra
-libvmi_la_LDFLAGS= $(GLIB_LIBS) -no-undefined
+libvmi_la_LDFLAGS= $(GLIB_LIBS) -lxenctrl -no-undefined
 libvmi_la_LDFLAGS+= -version-info $(VERSION)
 
 if CONFIGFILE
