diff --git a/cmake/modules/HandleLLVMOptions.cmake b/cmake/modules/HandleLLVMOptions.cmake
index 15497d405e0..33f7f18193a 100644
--- a/cmake/modules/HandleLLVMOptions.cmake
+++ b/cmake/modules/HandleLLVMOptions.cmake
@@ -127,7 +127,10 @@ else(WIN32)
       set(LLVM_HAVE_LINK_VERSION_SCRIPT 1)
     endif()
   else(FUCHSIA OR UNIX)
-    MESSAGE(SEND_ERROR "Unable to determine platform")
+    if(${CMAKE_SYSTEM_NAME} MATCHES "Wasi")
+    else()
+      MESSAGE(SEND_ERROR "Unable to determine platform")
+    endif()
   endif(FUCHSIA OR UNIX)
 endif(WIN32)
 
