diff -ur a/configure.ac b/configure.ac
--- a/configure.ac	2022-03-03 09:05:27.000000000 -0800
+++ b/configure.ac	2022-06-04 00:35:32.154279066 -0700
@@ -279,6 +279,10 @@
     AS_IF([test "x$with_java12" = "xyes" -a "$BDJ_TYPE" != "j2me"], [java_code_version=1.7])
 ])

+if test "x$use_bdjava_jar" = "xyes"; then
+  CPPFLAGS="${CPPFLAGS} -DJARDIR='\"\$(datadir)/java\"'"
+fi
+
 AC_DEFINE_UNQUOTED([JAVA_ARCH], ["$java_arch"], ["Defines the architecture of the java vm."])
 AC_DEFINE_UNQUOTED([JDK_HOME], ["$JDK_HOME"], [""])
 AM_CONDITIONAL([USING_BDJAVA_BUILD_JAR], [ test $use_bdjava_jar = "yes" ])
diff -ur a/src/libbluray/bdj/bdj.c b/src/libbluray/bdj/bdj.c
--- a/src/libbluray/bdj/bdj.c	2022-03-03 09:05:27.000000000 -0800
+++ b/src/libbluray/bdj/bdj.c	2022-06-04 00:39:33.259603941 -0700
@@ -533,6 +533,7 @@
 #  ifdef __FreeBSD__
         "/usr/local/share/java/" BDJ_JARFILE,
 #  else
+        JARDIR "/" BDJ_JARFILE,
         "/usr/share/java/" BDJ_JARFILE,
         "/usr/share/libbluray/lib/" BDJ_JARFILE,
 #  endif
