diff --git a/tests/utils.py b/tests/utils.py
index 90ba1d4..e320a58 100644
--- a/tests/utils.py
+++ b/tests/utils.py
@@ -38,6 +38,10 @@ def get_queue_index(name='default'):
         if q.name == name:
             # assert that the connection is correct
             pool_kwargs = q.connection.connection_pool.connection_kwargs
+            pool_kwargs.pop("maint_notifications_config", None)
+            pool_kwargs.pop("maint_notifications_pool_handler", None)
+            connection_kwargs.pop("maint_notifications_config", None)
+            connection_kwargs.pop("maint_notifications_pool_handler", None)
             if not _is_buggy_retry(pool_kwargs) or not _is_buggy_retry(connection_kwargs):
                 assert pool_kwargs == connection_kwargs
             else:
