diff --git a/libqtile/backend/x11/xcursors.py b/libqtile/backend/x11/xcursors.py
index 24454b83..ef37875c 100644
--- a/libqtile/backend/x11/xcursors.py
+++ b/libqtile/backend/x11/xcursors.py
@@ -107,7 +107,7 @@ class Cursors(dict):
 
     def _setup_xcursor_binding(self):
         try:
-            xcursor = ffi.dlopen('libxcb-cursor.so.0')
+            xcursor = ffi.dlopen('@xcb-cursor@/lib/libxcb-cursor.so.0')
         except OSError:
             logger.warning("xcb-cursor not found, fallback to font pointer")
             return False
diff --git a/libqtile/pangocffi.py b/libqtile/pangocffi.py
index dbae27ed..54c2c35f 100644
--- a/libqtile/pangocffi.py
+++ b/libqtile/pangocffi.py
@@ -52,10 +52,9 @@ try:
 except ImportError:
     raise ImportError("No module named libqtile._ffi_pango, be sure to run `./scripts/ffibuild`")
 
-gobject = ffi.dlopen('libgobject-2.0.so.0')
-pango = ffi.dlopen('libpango-1.0.so.0')
-pangocairo = ffi.dlopen('libpangocairo-1.0.so.0')
-
+gobject = ffi.dlopen('@glib@/lib/libgobject-2.0.so.0')
+pango = ffi.dlopen('@pango@/lib/libpango-1.0.so.0')
+pangocairo = ffi.dlopen('@pango@/lib/libpangocairo-1.0.so.0')
 
 def patch_cairo_context(cairo_t):
     def create_layout():
