diff -rc libtunepimp-0.5.3/include/tunepimp-0.5/metadata.h libtunepimp-0.5.3-new/include/tunepimp-0.5/metadata.h
*** libtunepimp-0.5.3/include/tunepimp-0.5/metadata.h	2006-11-18 05:52:08.000000000 -0500
--- libtunepimp-0.5.3-new/include/tunepimp-0.5/metadata.h	2011-11-06 09:00:14.816684749 -0500
***************
*** 29,34 ****
--- 29,35 ----
  
  #include <string>
  #include <cstdio>
+ #include <cstring>
  
  #include "defs.h"
  #include "tp_c.h"
diff -rc libtunepimp-0.5.3/lib/c_wrapper.cpp libtunepimp-0.5.3-new/lib/c_wrapper.cpp
*** libtunepimp-0.5.3/lib/c_wrapper.cpp	2006-11-18 05:52:33.000000000 -0500
--- libtunepimp-0.5.3-new/lib/c_wrapper.cpp	2011-11-06 09:00:14.816684749 -0500
***************
*** 43,48 ****
--- 43,49 ----
                           if (obj == NULL) return;
  
  #include <deque>
+ #include <cstdlib>
  
  class NotifyData
  {
diff -rc libtunepimp-0.5.3/lib/fileio.cpp libtunepimp-0.5.3-new/lib/fileio.cpp
*** libtunepimp-0.5.3/lib/fileio.cpp	2006-11-18 05:52:33.000000000 -0500
--- libtunepimp-0.5.3-new/lib/fileio.cpp	2011-11-06 09:00:14.816684749 -0500
***************
*** 27,32 ****
--- 27,34 ----
  
  #include <assert.h>
  #include <errno.h>
+ #include <string.h>
+ #include <stdlib.h>
  #include <string>
  #ifndef WIN32
  #include <unistd.h>
***************
*** 122,128 ****
  
  void tmktempname(const char *path, char *newPath, int newPathLen)
  {
!     char *ptr, *temp;
  
      temp = (char *)malloc(strlen(path) + 32);
      ptr = strrchr(path, dirSepChar);
--- 124,131 ----
  
  void tmktempname(const char *path, char *newPath, int newPathLen)
  {
!     char *temp;
! 	const char *ptr;
  
      temp = (char *)malloc(strlen(path) + 32);
      ptr = strrchr(path, dirSepChar);
diff -rc libtunepimp-0.5.3/lib/metadata.cpp libtunepimp-0.5.3-new/lib/metadata.cpp
*** libtunepimp-0.5.3/lib/metadata.cpp	2006-11-18 05:52:33.000000000 -0500
--- libtunepimp-0.5.3-new/lib/metadata.cpp	2011-11-06 09:00:14.817684754 -0500
***************
*** 24,31 ****
     $Id: metadata.cpp 8359 2006-08-07 20:34:50Z luks $
  
  ----------------------------------------------------------------------------*/
! #include <math.h>
! #include <stdio.h>
  #include "metadata.h"
  #include "../config.h"
  using namespace std;
--- 24,32 ----
     $Id: metadata.cpp 8359 2006-08-07 20:34:50Z luks $
  
  ----------------------------------------------------------------------------*/
! #include <cmath>
! #include <cstdio>
! #include <cstdlib>
  #include "metadata.h"
  #include "../config.h"
  using namespace std;
diff -rc libtunepimp-0.5.3/lib/protocol.cpp libtunepimp-0.5.3-new/lib/protocol.cpp
*** libtunepimp-0.5.3/lib/protocol.cpp	2006-11-28 15:25:04.000000000 -0500
--- libtunepimp-0.5.3-new/lib/protocol.cpp	2011-11-06 09:00:23.281742454 -0500
***************
*** 8,18 ****
  -------------------------------------------------------------------*/
  #include <stdio.h>
  #include <stdlib.h>
  #include <string>
  #include <map>
  #include <expat.h>
  #include <curl/curl.h>
- #include <curl/types.h>
  #include <curl/easy.h>
  using namespace std;
  
--- 8,18 ----
  -------------------------------------------------------------------*/
  #include <stdio.h>
  #include <stdlib.h>
+ #include <string.h>
  #include <string>
  #include <map>
  #include <expat.h>
  #include <curl/curl.h>
  #include <curl/easy.h>
  using namespace std;
  
diff -rc libtunepimp-0.5.3/lib/readmeta.cpp libtunepimp-0.5.3-new/lib/readmeta.cpp
*** libtunepimp-0.5.3/lib/readmeta.cpp	2006-11-18 05:52:33.000000000 -0500
--- libtunepimp-0.5.3-new/lib/readmeta.cpp	2011-11-06 09:00:14.817684754 -0500
***************
*** 30,36 ****
  #endif
  #endif
  
! #include <stdio.h>
  #ifndef WIN32
  #include <unistd.h>
  #endif
--- 30,37 ----
  #endif
  #endif
  
! #include <cstdio>
! #include <cstdlib>
  #ifndef WIN32
  #include <unistd.h>
  #endif
diff -rc libtunepimp-0.5.3/lib/tunepimp.cpp libtunepimp-0.5.3-new/lib/tunepimp.cpp
*** libtunepimp-0.5.3/lib/tunepimp.cpp	2006-11-18 05:52:33.000000000 -0500
--- libtunepimp-0.5.3-new/lib/tunepimp.cpp	2011-11-06 09:00:14.818684760 -0500
***************
*** 30,36 ****
  #endif
  #endif
  
! #include <stdio.h>
  #include <map>
  using namespace std;
  
--- 30,37 ----
  #endif
  #endif
  
! #include <cstdio>
! #include <cstdlib>
  #include <map>
  using namespace std;
  
diff -rc libtunepimp-0.5.3/lib/utf8/utf8util.cpp libtunepimp-0.5.3-new/lib/utf8/utf8util.cpp
*** libtunepimp-0.5.3/lib/utf8/utf8util.cpp	2006-11-18 05:52:26.000000000 -0500
--- libtunepimp-0.5.3-new/lib/utf8/utf8util.cpp	2011-11-06 09:00:14.818684760 -0500
***************
*** 26,31 ****
--- 26,32 ----
  ----------------------------------------------------------------------------*/
  
  #include <stdio.h>
+ #include <stdlib.h>
  #include "utf8util.h"
  #include "utf8.h"
  #ifdef WIN32
diff -rc libtunepimp-0.5.3/plugins/mp3/id3_2_3_meta.cpp libtunepimp-0.5.3-new/plugins/mp3/id3_2_3_meta.cpp
*** libtunepimp-0.5.3/plugins/mp3/id3_2_3_meta.cpp	2006-11-18 05:51:07.000000000 -0500
--- libtunepimp-0.5.3-new/plugins/mp3/id3_2_3_meta.cpp	2011-11-06 09:00:14.818684760 -0500
***************
*** 26,31 ****
--- 26,32 ----
  ----------------------------------------------------------------------------*/
  
  #include <stdio.h>
+ #include <stdlib.h>
  #include <assert.h>
  #include <ctype.h>
  #include <musicbrainz/mb_c.h>
diff -rc libtunepimp-0.5.3/plugins/mp3/id3_meta.cpp libtunepimp-0.5.3-new/plugins/mp3/id3_meta.cpp
*** libtunepimp-0.5.3/plugins/mp3/id3_meta.cpp	2006-11-18 05:51:07.000000000 -0500
--- libtunepimp-0.5.3-new/plugins/mp3/id3_meta.cpp	2011-11-06 09:00:14.818684760 -0500
***************
*** 26,31 ****
--- 26,32 ----
  ----------------------------------------------------------------------------*/
  
  #include <stdio.h>
+ #include <stdlib.h>
  #include <time.h>
  #include <ctype.h>
  #include <musicbrainz/mb_c.h>
diff -rc libtunepimp-0.5.3/plugins/mp4/mp4.cpp libtunepimp-0.5.3-new/plugins/mp4/mp4.cpp
*** libtunepimp-0.5.3/plugins/mp4/mp4.cpp	2006-11-18 05:51:08.000000000 -0500
--- libtunepimp-0.5.3-new/plugins/mp4/mp4.cpp	2011-11-06 09:00:14.819684766 -0500
***************
*** 27,33 ****
  
  #include <string.h>
  #include <stdio.h>
! #include <mp4.h>
  #include "metadata.h"
  #include "plugin.h"
  #ifndef WIN32
--- 27,34 ----
  
  #include <string.h>
  #include <stdio.h>
! #include <stdlib.h>
! #include <mp4v2/mp4v2.h>
  #include "metadata.h"
  #include "plugin.h"
  #ifndef WIN32
diff -rc libtunepimp-0.5.3/plugins/wma/wma.cpp libtunepimp-0.5.3-new/plugins/wma/wma.cpp
*** libtunepimp-0.5.3/plugins/wma/wma.cpp	2006-11-18 05:51:28.000000000 -0500
--- libtunepimp-0.5.3-new/plugins/wma/wma.cpp	2011-11-06 09:00:14.819684766 -0500
***************
*** 27,32 ****
--- 27,33 ----
  
  #include <string.h>
  #include <stdio.h>
+ #include <stdlib.h>
  #include <wmafile.h>
  #include "metadata.h"
  #include "plugin.h"
diff -rc libtunepimp-0.5.3/plugins/wma/wmafile.cpp libtunepimp-0.5.3-new/plugins/wma/wmafile.cpp
*** libtunepimp-0.5.3/plugins/wma/wmafile.cpp	2006-11-18 05:51:28.000000000 -0500
--- libtunepimp-0.5.3-new/plugins/wma/wmafile.cpp	2011-11-06 09:00:14.819684766 -0500
***************
*** 19,24 ****
--- 19,25 ----
   *   USA                                                                   *
   ***************************************************************************/
  
+ #include <string.h>
  #include <tstring.h>
  #include <wmafile.h>
  #include <wmatag.h>
