Author: Corey Wright <undefined@pobox.com>
Description: Adds support for DVD-ROM medium-type.

Index: dvdisaster/scsi-layer.c
===================================================================
--- dvdisaster.orig/scsi-layer.c	2012-03-06 11:10:17.147044691 +0900
+++ dvdisaster/scsi-layer.c	2012-03-06 11:10:30.927044292 +0900
@@ -913,6 +913,11 @@
 	    break;
 	 }
 
+	 if(layer_type & 0x01)
+	 {  dh->typeDescr = g_strdup("DVD-ROM");
+	    break;
+	 }
+
 	 if(layer_type & 0x06) /* strange thing: (re-)writeable but neither plus nor dash */ 
 	 {  dh->typeDescr = g_strdup("DVD-ROM (fake)");
 	    dh->subType = DVD;
