diff --git a/extrasrc/Sconstruct.libchrome b/extrasrc/Sconstruct.libchrome
index 4feb76d..311fe8a 100644
--- a/extrasrc/Sconstruct.libchrome
+++ b/extrasrc/Sconstruct.libchrome
@@ -103,7 +103,7 @@ base_lib = {
     'pc_libs' : 'glib-2.0',
 }
 
-env = Environment()
+env = Environment(ENV = os.environ)
 
 BASE_VER = os.environ.get('BASE_VER', '0')
 GTEST_DIR = os.environ.get('GTEST_DIR', '0')
diff --git a/extrasrc/Sconstruct.libchromeos b/extrasrc/Sconstruct.libchromeos
index 1da6001..66f9acb 100644
--- a/extrasrc/Sconstruct.libchromeos
+++ b/extrasrc/Sconstruct.libchromeos
@@ -18,7 +18,7 @@ base_lib = {
     'pc_libs' : 'dbus-c++-1',
 }
 
-env = Environment()
+env = Environment(ENV = os.environ)
 
 PKG_CONFIG = os.environ.get('PKG_CONFIG', 'pkg-config')
 BASE_VER = os.environ.get('BASE_VER', '0')
