diff --git a/interface/vmcs_host/linux/vcfiled/CMakeLists.txt b/interface/vmcs_host/linux/vcfiled/CMakeLists.txt
index aed0e83..b325676 100644
--- a/interface/vmcs_host/linux/vcfiled/CMakeLists.txt
+++ b/interface/vmcs_host/linux/vcfiled/CMakeLists.txt
@@ -17,15 +17,6 @@ target_link_libraries(vcfiled
 install(TARGETS vcfiled 
         RUNTIME DESTINATION sbin)
 
-configure_file (etc/init.d/vcfiled ${PROJECT_BINARY_DIR}/etc/init.d/vcfiled)
-
-# script to start up vcfiled at start of day
-install(PROGRAMS ${PROJECT_BINARY_DIR}/etc/init.d/vcfiled
-        DESTINATION /etc/init.d)
-# install locally to the installation directory too
-install(PROGRAMS ${PROJECT_BINARY_DIR}/etc/init.d/vcfiled
-        DESTINATION ${VMCS_INSTALL_PREFIX}/share/install)
-
 # test program for vcfiled_check library
 add_executable(vcfiled_lock_test vcfiled_lock_test.c)
 target_link_libraries(vcfiled_lock_test vcfiled_check)
