diff --git a/setup.py b/setup.py
index 05e42c1..8347acb 100644
--- a/setup.py
+++ b/setup.py
@@ -41,4 +41,9 @@ setup(
         "Programming Language :: Python :: 3.10",
     ],
     keywords="rhasspy wyoming piper",
+    entry_points={
+        'console_scripts': [
+            'wyoming-piper = wyoming_piper:__main__.run'
+        ]
+    }
 )
diff --git a/wyoming_piper/__main__.py b/wyoming_piper/__main__.py
index ab1580b..4c0a143 100755
--- a/wyoming_piper/__main__.py
+++ b/wyoming_piper/__main__.py
@@ -143,8 +143,12 @@ def get_description(voice_info: Dict[str, Any]):
 
 # -----------------------------------------------------------------------------
 
+def run():
+    asyncio.run(main())
+
+
 if __name__ == "__main__":
     try:
-        asyncio.run(main())
+        run()
     except KeyboardInterrupt:
         pass
