diff --git a/src/common.c b/src/common.c
index 1ca612a..7c21377 100644
--- a/src/common.c
+++ b/src/common.c
@@ -37,13 +37,13 @@ char *get_config_path(char* default_path, char* pbuf, size_t bufsize) {
 	if(check_path(path))
 		return path;
 
-	// priority 4: $SYSCONFDIR/proxychains.conf
-	path = SYSCONFDIR "/" PROXYCHAINS_CONF_FILE;
+	// priority 4: /etc/proxychains.conf
+	path = "/etc/" PROXYCHAINS_CONF_FILE;
 	if(check_path(path))
 		return path;
 
-	// priority 5: /etc/proxychains.conf
-	path = "/etc/" PROXYCHAINS_CONF_FILE;
+	// priority 5: $SYSCONFDIR/proxychains.conf
+	path = SYSCONFDIR "/" PROXYCHAINS_CONF_FILE;
 	if(check_path(path))
 		return path;
 
-- 
2.37.2

