--- mysql-5.5.17/libmysql/CMakeLists.txt	2011-10-12 07:10:24.000000000 -0500
+++ mysql-5.5.17/libmysql/CMakeLists.txt	2011-11-08 03:19:31.379219300 -0600
@@ -25,6 +25,11 @@ INCLUDE_DIRECTORIES(
 ADD_DEFINITIONS(${SSL_DEFINES})
 
 SET(CLIENT_API_FUNCTIONS
+dynstr_free
+dynstr_append_mem
+dynstr_append_os_quoted
+dynstr_realloc
+init_dynamic_string
 get_tty_password
 handle_options
 load_defaults
@@ -131,6 +136,7 @@ mysql_server_end
 mysql_set_character_set
 mysql_get_character_set_info
 mysql_stmt_next_result
+strfill
 
 CACHE INTERNAL "Functions exported by client API"
 
