diff --git a/utils.pas b/utils.pas
index eb8b828..1ff2440 100644
--- a/utils.pas
+++ b/utils.pas
@@ -100,7 +100,7 @@ uses
 {$ifdef CALLSTACK}
   lineinfo2,
 {$endif CALLSTACK}
-  LazFileUtils, LazUtf8, StdCtrls, Graphics;
+  LazFileUtils, LazUtf8, StdCtrls, Graphics, FileUtil;
 
 {$ifdef windows}
 function FileOpenUTF8(Const FileName : string; Mode : Integer) : THandle;
@@ -235,7 +235,7 @@ end;
 
 function ParamStrUTF8(Param: Integer): utf8string;
 begin
-  Result:=FileUtil.ParamStrUTF8(Param);
+  Result:=LazUtf8.ParamStrUTF8(Param);
 end;
 
 function ParamCount: integer;
