diff --git a/mons.sh b/mons.sh
index b86ce5c..feb0f33 100755
--- a/mons.sh
+++ b/mons.sh
@@ -151,8 +151,7 @@ main() {
     # =============================
 
     [ -z "$DISPLAY" ]  && { echo 'DISPLAY: no variable set.';  exit 1; }
-    command -vp xrandr >/dev/null 2>&1 || { echo 'xrandr: command not found.'; exit 1; }
-    XRANDR="$(command -pv xrandr)"
+    XRANDR="@xrandr@"
 
     # =============================
     #      Argument Checking
