diff --git c/src/cgraphicsscene.cpp i/src/cgraphicsscene.cpp
index ac2929a..c399706 100644
--- c/src/cgraphicsscene.cpp
+++ i/src/cgraphicsscene.cpp
@@ -1436,7 +1436,7 @@ namespace Caneda
             QPointF newPos = m_currentWiringWire->mapFromScene(pos);
             QPointF refPos = m_currentWiringWire->port1()->pos();
 
-            if( abs(refPos.x()-newPos.x()) > abs(refPos.y()-newPos.y()) ) {
+            if( (refPos.x()-newPos.x()) > (refPos.y()-newPos.y()) ) {
                 m_currentWiringWire->movePort2(QPointF(newPos.x(), refPos.y()));
             }
             else {
