Index: subversion/bindings/swig/perl/native/Makefile.PL.in
===================================================================
--- subversion-1.4.5/subversion/bindings/swig/perl/native/Makefile.PL.in	(revision 27184)
+++ subversion-1.4.5/subversion/bindings/swig/perl/native/Makefile.PL.in	(working copy)
@@ -26,6 +26,7 @@
 my @ldmodules = map {"-lsvn_$_-1"} (@modules, qw/diff subr/);
 
 my $apr_shlib_path_var = '@SVN_APR_SHLIB_PATH_VAR@';
+my $cppflags = '@CPPFLAGS@';
 my $apr_cflags = '@SVN_APR_INCLUDES@';
 my $apu_cflags = '@SVN_APRUTIL_INCLUDES@';
 
@@ -38,7 +39,7 @@
 
 my %config = (
     ABSTRACT => 'Perl bindings for Subversion',
-    INC  => join(' ',$apr_cflags, $apu_cflags, 
+    INC  => join(' ', $cppflags, $apr_cflags, $apu_cflags, 
                  " -I$swig_srcdir/perl/libsvn_swig_perl",
                  " -I$svnlib_srcdir/include",
                  " -I$svnlib_builddir",

