diff --git a/data/meson.build b/data/meson.build
index 1eb3dfb0..623b66ec 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -26,4 +26,4 @@ i18n.merge_file (
 )
 
 install_data ('appstream.conf',
-              install_dir: get_option('sysconfdir'))
+              install_dir: get_option('prefix') / 'etc')
diff --git a/meson.build b/meson.build
index dc1fb1a5..5ee03b73 100644
--- a/meson.build
+++ b/meson.build
@@ -108,12 +108,12 @@ if get_option ('gir')
     dependency('gobject-introspection-1.0', version: '>=1.56')
 endif
 
-stemmer_inc_dirs = include_directories(['/usr/include'])
+stemmer_inc_dirs = include_directories(['@libstemmer_includedir@'])
 if get_option('stemming')
     stemmer_lib = cc.find_library('stemmer', required: true)
     if not cc.has_header('libstemmer.h')
         if cc.has_header('libstemmer/libstemmer.h')
-            stemmer_inc_dirs = include_directories('/usr/include/libstemmer')
+            stemmer_inc_dirs = include_directories('@libstemmer_includedir@/libstemmer')
         else
             error('Unable to find Snowball header "libstemmer.h". Please ensure libstemmer/Snowball is installed properly in order to continue.')
         endif
