--- a/linux/Makefile
+++ b/linux/Makefile
@@ -131,7 +131,6 @@
 		ln -fs $(x) $(patsubst %.so.1.$(VERSION),%.so.1,$(DESTDIR)$(libdir)/$(x)) ;\
 		ln -fs $(x) $(patsubst %.so.1.$(VERSION),%.so,$(DESTDIR)$(libdir)/$(x)) ;\
 		)
-	$(INSTALL) -d -m 755 $(DESTDIR)$(sharedstatedir)/fwupdate/
 	$(INSTALL) -d -m 755 $(DESTDIR)$(libexecdir)/fwupdate/
 	$(INSTALL) -m 755 cleanup $(DESTDIR)$(libexecdir)/fwupdate/cleanup
 	$(INSTALL) -d -m 755 $(DESTDIR)$(libdatadir)/systemd/system
