diff --git a/test_jaconv.py b/test_jaconv.py
index 7e0a169..aaf180d 100644
--- a/test_jaconv.py
+++ b/test_jaconv.py
@@ -1,11 +1,8 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
-from nose.tools import assert_equal, nottest
 import jaconv
 from functools import partial
 
-assert_equal.__self__.maxDiff = None
-
 HIRAGANA = ('ぁあぃいぅうぇえぉおかがきぎくぐけげこごさざしじすずせぜそぞた',
             'だちぢっつづてでとどなにぬねのはばぱひびぴふぶぷへべぺほぼぽま',
             'みむめもゃやゅゆょよらりるれろわをんーゎゐゑゕゖゔゝゞ・「」。、')
@@ -25,13 +22,15 @@ FULL_ASCII = ('！＂＃＄％＆＇（）＊＋，－．／：；＜＝＞？
 FULL_DIGIT = '０１２３４５６７８９'
 
 
-@nottest
+def assert_equal(x, y):
+    assert x == y
+
+
 def _compare(mathod, lhs, rhs):
     for i in range(len(lhs)):
         assert_equal(mathod(lhs[i]), rhs[i])
 
 
-@nottest
 def _concat(*iterables):
     result = ''
     for iterable in iterables:
