diff --git a/scrapy/commands/startproject.py b/scrapy/commands/startproject.py
index 5941066..89f8edb 100644
--- a/scrapy/commands/startproject.py
+++ b/scrapy/commands/startproject.py
@@ -4,7 +4,7 @@ import os
 import string
 from importlib import import_module
 from os.path import join, exists, abspath
-from shutil import ignore_patterns, move, copy2, copystat
+from shutil import ignore_patterns, move, copyfile, copystat
 
 import scrapy
 from scrapy.commands import ScrapyCommand
@@ -76,8 +76,7 @@ class Command(ScrapyCommand):
             if os.path.isdir(srcname):
                 self._copytree(srcname, dstname)
             else:
-                copy2(srcname, dstname)
-        copystat(src, dst)
+                copyfile(srcname, dstname)
 
     def run(self, args, opts):
         if len(args) not in (1, 2):
