+++ b/platform/x11/detect.py
@@ -175,6 +175,11 @@ def configure(env):
     env.ParseConfig('pkg-config xrender --cflags --libs')
     env.ParseConfig('pkg-config xi --cflags --libs')
 
+    env.ParseConfig('pkg-config xext --cflags --libs')
+    env.ParseConfig('pkg-config xfixes --cflags --libs')
+    env.ParseConfig('pkg-config glu --cflags --libs')
+    env.ParseConfig('pkg-config zlib --cflags --libs')
+
     if (env['touch']):
         env.Append(CPPFLAGS=['-DTOUCH_ENABLED'])
 
@@ -264,7 +269,7 @@ def configure(env):
         print("Enabling ALSA")
         env.Append(CPPFLAGS=["-DALSA_ENABLED", "-DALSAMIDI_ENABLED"])
 	# Don't parse --cflags, we don't need to add /usr/include/alsa to include path
-        env.ParseConfig('pkg-config alsa --libs')
+        env.ParseConfig('pkg-config alsa --cflags --libs')
     else:
         print("ALSA libraries not found, disabling driver")
 
