diff --git a/client/out/languageserver.js b/client/out/languageserver.js
index 6c7429c..6f53aa4 100644
--- a/client/out/languageserver.js
+++ b/client/out/languageserver.js
@@ -79,11 +79,9 @@ class LuaClient {
                     break;
                 case "linux":
                     command = this.context.asAbsolutePath(path.join('server', binDir ? binDir : 'bin-Linux', 'lua-language-server'));
-                    yield fs.promises.chmod(command, '777');
                     break;
                 case "darwin":
                     command = this.context.asAbsolutePath(path.join('server', binDir ? binDir : 'bin-macOS', 'lua-language-server'));
-                    yield fs.promises.chmod(command, '777');
                     break;
             }
             let serverOptions = {
