diff --git a/src/source/common/ddesktopservicesthread.h b/src/source/common/ddesktopservicesthread.h
index 49313744..456a5e96 100644
--- a/src/source/common/ddesktopservicesthread.h
+++ b/src/source/common/ddesktopservicesthread.h
@@ -8,10 +8,14 @@
 
 #include <QThread>
 
+#include <dtkwidget_global.h>
+#include <dtkgui_global.h>
 #include <DDesktopServices>
 #include <QDebug>
 #include <QFileInfo>
+
 DWIDGET_USE_NAMESPACE
+DGUI_USE_NAMESPACE
 
 // 文管打开文件目录线程
 class DDesktopServicesThread : public QThread

