From 1c749a3610fb423e50260a2c807b44a94ef4c69d Mon Sep 17 00:00:00 2001
From: Nina Fromm <git@e1mo.de>
Date: Thu, 25 Jun 2026 16:23:00 +0200
Subject: [PATCH] settings: catch unwritable .env

---
 app/Misc/Helper.php | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/app/Misc/Helper.php b/app/Misc/Helper.php
index 2d647b89..3fc2ef84 100644
--- a/app/Misc/Helper.php
+++ b/app/Misc/Helper.php
@@ -1172,7 +1172,12 @@ class Helper
                 $contents = $contents."\n{$key}={$value}\n";
             }
         }
-        \File::put($env_path, $contents);
+        try {
+          \File::put($env_path, $contents);
+        } catch (\Exception $e) {
+          \Helper::logException($e, 'Error updating ' . $key  .' in .env file: ');
+          \Session::flash('flash_error_unescaped', "Unable to write settings to <code>$env_path</code>: " . $e->getMessage());
+        }
     }
 
     /**
-- 
2.53.0

