diff --git a/src/gasman.c b/src/gasman.c
index 417811401..314a74190 100644
--- a/src/gasman.c
+++ b/src/gasman.c
@@ -1861,7 +1861,7 @@ void SparcStackFuncBags( void )
 #endif
 
 
-void GenStackFuncBags ( void )
+NOINLINE void GenStackFuncBags ( void )
 {
     Bag *               top;            /* top of stack                    */
     Bag *               p;              /* loop variable                   */
