diff --git a/Makefile b/Makefile
index dde9f81..5af7cab 100644
--- a/Makefile
+++ b/Makefile
@@ -34,7 +34,6 @@ install: all
 	-mkdir -pv $(DESTDIR)/sbin
 	install -m 755 mstpd $(DESTDIR)/sbin/mstpd
 	install -m 755 mstpctl $(DESTDIR)/sbin/mstpctl
-	install -m 755 bridge-stp /sbin/bridge-stp
 	-mkdir -pv $(DESTDIR)/lib/mstpctl-utils/
 	cp -rv lib/* $(DESTDIR)/lib/mstpctl-utils/
 	gzip -f $(DESTDIR)/lib/mstpctl-utils/mstpctl.8
@@ -42,8 +41,9 @@ install: all
 	if [ -d $(DESTDIR)/etc/network/if-pre-up.d ] ; then ln -sf /lib/mstpctl-utils/ifupdown.sh $(DESTDIR)/etc/network/if-pre-up.d/mstpctl ; fi
 	if [ -d $(DESTDIR)/etc/network/if-pre-up.d ] ; then ln -sf /lib/mstpctl-utils/ifupdown.sh $(DESTDIR)/etc/network/if-post-down.d/mstpctl ; fi
 	if [ -d $(DESTDIR)/etc/bash_completion.d ] ; then ln -sf /lib/mstpctl-utils/bash_completion $(DESTDIR)/etc/bash_completion.d/mstpctl ; fi
-	ln -sf /lib/mstpctl-utils/mstpctl.8.gz $(DESTDIR)/usr/share/man/man8/mstpctl.8.gz
-	ln -sf /lib/mstpctl-utils/mstpctl-utils-interfaces.5.gz $(DESTDIR)/usr/share/man/man5/mstpctl-utils-interfaces.5.gz
+	mkdir -p $(DESTDIR)/share/man/man8 $(DESTDIR)/share/man/man5
+	ln -sf /lib/mstpctl-utils/mstpctl.8.gz $(DESTDIR)/share/man/man8/mstpctl.8.gz
+	ln -sf /lib/mstpctl-utils/mstpctl-utils-interfaces.5.gz $(DESTDIR)/share/man/man5/mstpctl-utils-interfaces.5.gz
 
 romfs: all
 	$(ROMFSINST) /sbin/mstpd
diff --git a/bridge_track.c b/bridge_track.c
index c92fdf6..0c01aec 100644
--- a/bridge_track.c
+++ b/bridge_track.c
@@ -28,6 +28,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <linux/param.h>
+#include <linux/in6.h>
 #include <linux/if_bridge.h>
 #include <asm/byteorder.h>
 
diff --git a/broadcom_xstrata/driver_deps.c b/broadcom_xstrata/driver_deps.c
index e72e9e3..5194253 100644
--- a/broadcom_xstrata/driver_deps.c
+++ b/broadcom_xstrata/driver_deps.c
@@ -14,6 +14,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <asm/byteorder.h>
+#include <linux/in6.h>
 #include <linux/if_bridge.h>
 
 #include "log.h"
diff --git a/ctl_functions.h b/ctl_functions.h
index 9c3b914..df464de 100644
--- a/ctl_functions.h
+++ b/ctl_functions.h
@@ -27,6 +27,7 @@
 #ifndef CTL_SOCKET_H
 #define CTL_SOCKET_H
 
+#include <linux/in6.h>
 #include <linux/if_bridge.h>
 #include <asm/byteorder.h>
 
diff --git a/mstp.c b/mstp.c
index 1c6a2df..b2a1acd 100644
--- a/mstp.c
+++ b/mstp.c
@@ -37,6 +37,7 @@
 
 #include <string.h>
 #include <sys/time.h>
+#include <linux/in6.h>
 #include <linux/if_bridge.h>
 #include <asm/byteorder.h>
 
