From 11ad41c2d3af88b5cf85d4bc2f5388b65e5ff347 Mon Sep 17 00:00:00 2001
From: Randy Eckenrode <randy@largeandhighquality.com>
Date: Mon, 16 Sep 2024 12:42:59 -0400
Subject: [PATCH] Use availability check for __collate_lookup_l

---
 tr/str.c | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/tr/str.c b/tr/str.c
index 6a79df2..8226abe 100644
--- a/tr/str.c
+++ b/tr/str.c
@@ -69,6 +69,7 @@ static void	genseq(STR *);
  * Using libc internal function __collate_lookup_l for character
  * equivalence
  */
+void __collate_lookup(const __darwin_wchar_t *, int *, int *, int *);
 void __collate_lookup_l(const __darwin_wchar_t *, int *, int *, int *,
 locale_t);
 /*
@@ -255,7 +256,13 @@ genequiv(STR *s)
      */
     int tprim, tsec;
     int len;
-    __collate_lookup_l(s->equiv, &len, &tprim, &tsec, LC_GLOBAL_LOCALE);
+    if (__builtin_available(macOS 11.0, *)) {
+        __collate_lookup_l(s->equiv, &len, &tprim, &tsec, LC_GLOBAL_LOCALE);
+    } else {
+        locale_t old = uselocale(LC_GLOBAL_LOCALE);
+        __collate_lookup(s->equiv, &len, &tprim, &tsec);
+        uselocale(old);
+    }
     
     if (tprim != -1) {
         for (p = 1, i = 1; i < NCHARS_SB; i++) {
@@ -270,7 +277,13 @@ genequiv(STR *s)
                  * perform lookup of primary weight and fill cache
                  */
                 int csec;
-                __collate_lookup_l((__darwin_wchar_t *)&i, &len, &cprim, &csec, LC_GLOBAL_LOCALE);
+                if (__builtin_available(macOS 11.0, *)) {
+                    __collate_lookup_l((__darwin_wchar_t *)&i, &len, &cprim, &csec, LC_GLOBAL_LOCALE);
+                } else {
+                    locale_t old = uselocale(LC_GLOBAL_LOCALE);
+                    __collate_lookup((__darwin_wchar_t *)&i, &len, &cprim, &csec);
+                    uselocale(old);
+                }
                 collation_weight_cache[i] = cprim;
             }
             
-- 
2.46.0

