From 46d124da602d84b7611a7ff0ac0862168d451cdb Mon Sep 17 00:00:00 2001
From: Thomas Tuegel <ttuegel@gmail.com>
Date: Wed, 14 Oct 2015 06:31:29 -0500
Subject: [PATCH 2/2] no canonicalize path

---
 src/sycoca/vfolder_menu.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sycoca/vfolder_menu.cpp b/src/sycoca/vfolder_menu.cpp
index d3e31c3..d15d743 100644
--- a/src/sycoca/vfolder_menu.cpp
+++ b/src/sycoca/vfolder_menu.cpp
@@ -415,7 +415,7 @@ VFolderMenu::absoluteDir(const QString &_dir, const QString &baseDir, bool keepR
     }
 
     if (!relative) {
-        QString resolved = QDir(dir).canonicalPath();
+      QString resolved = QDir::cleanPath(dir);
         if (!resolved.isEmpty()) {
             dir = resolved;
         }
-- 
2.5.2

