diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8bd825f..694d9b2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,6 +72,8 @@ if(WITH_JSON)
     endif()
 
     add_definitions("-DJSON")
+    # JSON_CFLAGS is a list, i.e. semicolon-separated, convert it to space-separated
+    string(REPLACE ";" " " JSON_CFLAGS "${JSON_CFLAGS}")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${JSON_CFLAGS}")
 endif()
 
