diff --git a/mesonbuild/environment.py b/mesonbuild/environment.py
index 756dd8193..a5cc6ef8b 100644
--- a/mesonbuild/environment.py
+++ b/mesonbuild/environment.py
@@ -151,7 +151,7 @@ def _get_env_var(for_machine: MachineChoice, is_cross: bool, var_name: str) -> T
         # compiling we fall back on the unprefixed host version. This
         # allows native builds to never need to worry about the 'BUILD_*'
         # ones.
-        ([var_name + '_FOR_BUILD'] if is_cross else [var_name]),
+        [var_name + '_FOR_BUILD'] + ([] if is_cross else [var_name]),
         # Always just the unprefixed host verions
         [var_name]
     )[for_machine]
