diff -aur lsyncd-release-2.1.5.orig/configure.ac lsyncd-release-2.1.5/configure.ac
--- lsyncd-release-2.1.5.orig/configure.ac	2013-06-07 18:24:02.000000000 +0600
+++ lsyncd-release-2.1.5/configure.ac	2014-03-04 13:24:15.390350812 +0600
@@ -15,10 +15,10 @@
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
 
-#AC_PATH_PROG([A2X], [a2x], [no])
-#if test x${A2X} = xno ; then
-#	AC_MSG_ERROR([Program 'a2x' (package asciidoc) is required])
-#fi
+AC_PATH_PROG([A2X], [a2x], [no])
+if test x${A2X} = xno ; then
+	AC_MSG_ERROR([Program 'a2x' (package asciidoc) is required])
+fi
 
 ###
 # Checks for Lua