diff --git i/data/org.gnome.gitlab.YaLTeR.VideoTrimmer.metainfo.xml.in.in w/data/org.gnome.gitlab.YaLTeR.VideoTrimmer.metainfo.xml.in.in
index 9bd25b6..c627369 100644
--- i/data/org.gnome.gitlab.YaLTeR.VideoTrimmer.metainfo.xml.in.in
+++ w/data/org.gnome.gitlab.YaLTeR.VideoTrimmer.metainfo.xml.in.in
@@ -19,12 +19,6 @@
 			Video Trimmer cuts out a fragment of a video given the start and end timestamps. The video is never re-encoded, so the process is very fast and does not reduce the video quality.
 		</p>
 	</description>
-	<screenshots>
-		<screenshot type="default">
-			<image>https://gitlab.gnome.org/YaLTeR/video-trimmer/uploads/e840fa093439348448007197d07c8033/image.png</image>
-			<caption>Main Window</caption>
-		</screenshot>
-	</screenshots>
 	<releases>
 		<release version="0.7.1" date="2022-03-23">
 			<description>
