diff -u aa/CoreFoundation/CMakeLists.txt b/CoreFoundation/CMakeLists.txt
--- a/CoreFoundation/CMakeLists.txt	1969-12-31 19:00:01.000000000 -0500
+++ b/CoreFoundation/CMakeLists.txt	2023-06-29 18:59:19.492601179 -0400

@@ -424,16 +424,11 @@
   set(CMAKE_SHARED_LIBRARY_SUFFIX "")
 endif()
 
-install(TARGETS
-          CoreFoundation
-        DESTINATION
-          "${CMAKE_INSTALL_FULL_LIBDIR}")
 install(DIRECTORY
           ${CoreFoundation_FRAMEWORK_DIRECTORY}
         DESTINATION
-          ${CMAKE_INSTALL_PREFIX}/System/Library/Frameworks
-        USE_SOURCE_PERMISSIONS
-        PATTERN PrivateHeaders EXCLUDE)
+          ${CMAKE_INSTALL_PREFIX}/Library/Frameworks
+        USE_SOURCE_PERMISSIONS)
 
 
 # TODO(compnerd) formalize this
