From f6c446cf6fab2edbd2606b4c6100903e9437362a Mon Sep 17 00:00:00 2001
From: Thomas Tuegel <ttuegel@mailbox.org>
Date: Sun, 25 Apr 2021 08:01:02 -0500
Subject: [PATCH 2/3] akonadi timestamps

---
 src/server/storage/dbconfigmysql.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/server/storage/dbconfigmysql.cpp b/src/server/storage/dbconfigmysql.cpp
index 3550f9d..e9e8887 100644
--- a/src/server/storage/dbconfigmysql.cpp
+++ b/src/server/storage/dbconfigmysql.cpp
@@ -241,8 +241,7 @@ bool DbConfigMysql::startInternalServer()
     bool confUpdate = false;
     QFile actualFile(actualConfig);
     // update conf only if either global (or local) is newer than actual
-    if ((QFileInfo(globalConfig).lastModified() > QFileInfo(actualFile).lastModified())
-        || (QFileInfo(localConfig).lastModified() > QFileInfo(actualFile).lastModified())) {
+    if (true) {
         QFile globalFile(globalConfig);
         QFile localFile(localConfig);
         if (globalFile.open(QFile::ReadOnly) && actualFile.open(QFile::WriteOnly)) {
-- 
2.31.1

