Index: qt-everywhere-opensource-src-5.5.1/qtbase/src/dbus/qdbus_symbols.cpp
===================================================================
--- qt-everywhere-opensource-src-5.5.1.orig/qtbase/src/dbus/qdbus_symbols.cpp
+++ qt-everywhere-opensource-src-5.5.1/qtbase/src/dbus/qdbus_symbols.cpp
@@ -89,7 +89,7 @@ bool qdbus_loadLibDBus()
 #ifdef Q_OS_WIN
         QLatin1String("dbus-1"),
 #endif
-        QLatin1String("libdbus-1")
+        QLatin1String("@dbus_libs@/lib/libdbus-1")
     };
 
     lib->unload();
