diff --git a/weblate/utils/lock.py b/weblate/utils/lock.py
index 53c1486bc9..a0a5fc5a74 100644
--- a/weblate/utils/lock.py
+++ b/weblate/utils/lock.py
@@ -43,8 +43,6 @@ class WeblateLock:
             self._name = self._format_template(cache_template)
             self._lock = cache.lock(
                 key=self._name,
-                expire=3600,
-                auto_renewal=True,
             )
             self._enter_implementation = self._enter_redis
         else:
@@ -62,7 +60,7 @@ class WeblateLock:
 
     def _enter_redis(self):
         try:
-            lock_result = self._lock.acquire(timeout=self._timeout)
+            lock_result = self._lock.acquire()
         except AlreadyAcquired:
             return
 
