diff --git a/src/libLfunction/Lcomplex.h b/src/libLfunction/Lcomplex.h
index 363bbf4..ffecd70 100644
--- a/src/libLfunction/Lcomplex.h
+++ b/src/libLfunction/Lcomplex.h
@@ -56,8 +56,11 @@
 #include <cmath>
 #include <sstream>
 
-namespace std
-{
+#ifdef _LIBCPP_VERSION
+_LIBCPP_BEGIN_NAMESPACE_STD
+#else
+namespace std {
+#endif
   // Forward declarations
   template<typename _Tp> class complex;
   template<> class complex<float>;
@@ -1193,6 +1196,10 @@ namespace std
   inline
   complex<long double>::complex(const complex<double>& __z)
   : _M_value(_ComplexT(__z._M_value)) { }
-} // namespace std
+#ifdef _LIBCPP_VERSION
+_LIBCPP_END_NAMESPACE_STD
+#else
+}
+#endif
 
 #endif	/* _CPP_COMPLEX */
