diff --git a/samcli/commands/_utils/table_print.py b/samcli/commands/_utils/table_print.py
index de63af29..a9d0f2fe 100644
--- a/samcli/commands/_utils/table_print.py
+++ b/samcli/commands/_utils/table_print.py
@@ -7,6 +7,7 @@ from functools import wraps
 from typing import Sized
 
 import click
+import shutil
 
 MIN_OFFSET = 20
 
@@ -30,7 +31,7 @@ def pprint_column_names(
 
     def pprint_wrap(func):
         # Calculate terminal width, number of columns in the table
-        width, _ = click.get_terminal_size()
+        width, _ = shutil.get_terminal_size()
         # For UX purposes, set a minimum width for the table to be usable
         # and usable_width keeps margins in mind.
         width = max(width, min_width)
