diff --git a/cmake/Modules/FindCed.cmake b/cmake/Modules/FindCed.cmake
index d6e2aca..3085adb 100644
--- a/cmake/Modules/FindCed.cmake
+++ b/cmake/Modules/FindCed.cmake
@@ -1,11 +1 @@
-include(LibFetchMacros)
-
-set(Ced_GIT_VERSION "master")
-
-libfetch_git_pkg(Ced
-	REPOSITORY ${SELF_BUILT_GIT_BASE}/compact_enc_det.git
-	#https://github.com/google/compact_enc_det.git
-	REFERENCE  ${Ced_GIT_VERSION}
-	FIND_PATH  compact_enc_det/compact_enc_det.h
-)
-message(STATUS "Found Google CED ${Ced_VERSION}")
+add_subdirectory(../ced-src ced-src)
