diff --git a/CMakeLists.txt b/CMakeLists.txt
index 48af2a89..43786c31 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -75,15 +75,7 @@ else()
 	message(STATUS "Localization disabled: Gettext tools (msgfmt) missing")
 endif()
 
-include(FetchContent)
-FetchContent_Declare(ced-sources
-  GIT_REPOSITORY https://github.com/performous/compact_enc_det.git
-  #https://github.com/google/compact_enc_det.git
-  GIT_TAG        master
-  SOURCE_DIR ced-src
-)
-
-FetchContent_MakeAvailable(ced-sources)
+add_subdirectory(ced-src)
 
 option(USE_SELF_BUILT_AUBIO "Use custom aubio local build instead of using system lib (if available)" FALSE)
 
