diff --git a/Package.swift b/Package.swift
index 52dbba0e47..e3a1fe9c2c 100644
--- a/Package.swift
+++ b/Package.swift
@@ -146,16 +146,16 @@
             exclude: ["CMakeLists.txt"],
             swiftSettings: [.swiftLanguageMode(.v6)]
         ),
-        .testTarget(
-            name: "DocCHTMLTests",
-            dependencies: [
-                .target(name: "DocCHTML"),
-                .target(name: "SwiftDocC"),
-                .product(name: "Markdown", package: "swift-markdown"),
-                .target(name: "DocCTestUtilities"),
-            ],
-            swiftSettings: [.swiftLanguageMode(.v6)]
-        ),
+        // .testTarget(
+        //     name: "DocCHTMLTests",
+        //     dependencies: [
+        //         .target(name: "DocCHTML"),
+        //         .target(name: "SwiftDocC"),
+        //         .product(name: "Markdown", package: "swift-markdown"),
+        //         .target(name: "DocCTestUtilities"),
+        //     ],
+        //     swiftSettings: [.swiftLanguageMode(.v6)]
+        // ),
 
         // Test app for DocCCommandLine
         .executableTarget(
