--- a/cpan/podlators/lib/Pod/Man.pm	2014-04-07 06:25:23.730505243 +0200
+++ b/cpan/podlators/lib/Pod/Man.pm	2014-04-07 06:26:40.816552603 +0200
@@ -768,7 +768,7 @@
     } else {
         ($name, $section) = $self->devise_title;
     }
-    my $date = $$self{date} || $self->devise_date;
+    my $date = "1970-01-01"; # Fixed date for NixOS, orig: $$self{date} || $self->devise_date;
     $self->preamble ($name, $section, $date)
         unless $self->bare_output or DEBUG > 9;
 
