diff --git a/pkl-gradle/src/test/kotlin/org/pkl/gradle/JavaCodeGeneratorsTest.kt b/pkl-gradle/src/test/kotlin/org/pkl/gradle/JavaCodeGeneratorsTest.kt
index 68a5b533..84fe66df 100644
--- a/pkl-gradle/src/test/kotlin/org/pkl/gradle/JavaCodeGeneratorsTest.kt
+++ b/pkl-gradle/src/test/kotlin/org/pkl/gradle/JavaCodeGeneratorsTest.kt
@@ -18,9 +18,11 @@ package org.pkl.gradle
 import kotlin.io.path.listDirectoryEntries
 import kotlin.io.path.readText
 import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test
 
 class JavaCodeGeneratorsTest : AbstractTest() {
+  @Disabled("runTask doesn't use the MITM cache")
   @Test
   fun `generate code`() {
     writeBuildFile()
@@ -68,6 +70,7 @@ class JavaCodeGeneratorsTest : AbstractTest() {
     )
   }
 
+  @Disabled("runTask doesn't use the MITM cache")
   @Test
   fun `compile generated code`() {
     writeBuildFile()
@@ -84,6 +87,7 @@ class JavaCodeGeneratorsTest : AbstractTest() {
     assertThat(addressClassFile).exists()
   }
 
+  @Disabled("runTask doesn't use the MITM cache")
   @Test
   fun `no source modules`() {
     writeFile(
diff --git a/pkl-gradle/src/test/kotlin/org/pkl/gradle/KotlinCodeGeneratorsTest.kt b/pkl-gradle/src/test/kotlin/org/pkl/gradle/KotlinCodeGeneratorsTest.kt
index 2b9ffc88..99b91893 100644
--- a/pkl-gradle/src/test/kotlin/org/pkl/gradle/KotlinCodeGeneratorsTest.kt
+++ b/pkl-gradle/src/test/kotlin/org/pkl/gradle/KotlinCodeGeneratorsTest.kt
@@ -18,9 +18,11 @@ package org.pkl.gradle
 import kotlin.io.path.listDirectoryEntries
 import kotlin.io.path.readText
 import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test
 
 class KotlinCodeGeneratorsTest : AbstractTest() {
+  @Disabled("runTask doesn't use the MITM cache")
   @Test
   fun `generate code`() {
     writeBuildFile()
@@ -69,6 +71,7 @@ class KotlinCodeGeneratorsTest : AbstractTest() {
     )
   }
 
+  @Disabled("runTask doesn't use the MITM cache")
   @Test
   fun `compile generated code`() {
     writeBuildFile()
@@ -84,6 +87,7 @@ class KotlinCodeGeneratorsTest : AbstractTest() {
     assertThat(addressClassFile).exists()
   }
 
+  @Disabled("runTask doesn't use the MITM cache")
   @Test
   fun `no source modules`() {
     writeFile(

