diff --git a/src/Makefile.am b/src/Makefile.am
index 4678ab530..f3c12cb94 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -35,7 +35,6 @@ pgpool_SOURCES = main/main.c \
 	rewrite/pool_timestamp.c \
 	rewrite/pool_lobj.c \
 	utils/pool_select_walker.c \
-	utils/strlcpy.c \
 	utils/psprintf.c \
 	utils/pool_params.c \
 	utils/ps_status.c \
@@ -106,7 +105,7 @@ pgpool_LDADD = -L@PGSQL_LIB_DIR@ -lpq parser/libsql-parser.a \
 if enable_rpath
   pgpool_LDFLAGS = -rpath @PGSQL_LIB_DIR@ -rpath $(libdir)
 else
-  pgpool_LDFLAGS = 
+  pgpool_LDFLAGS =
 endif
 
 AM_YFLAGS = -d
diff --git a/src/libs/pcp/Makefile.am b/src/libs/pcp/Makefile.am
index 31f51233a..bca3f5374 100644
--- a/src/libs/pcp/Makefile.am
+++ b/src/libs/pcp/Makefile.am
@@ -6,7 +6,6 @@ dist_libpcp_la_SOURCES = pcp.c \
 					../../utils/pool_path.c \
 					../../tools/fe_port.c \
 					../../tools/fe_memutils.c \
-					../../utils/strlcpy.c \
 					../../utils/pool_health_check_stats.c \
 					../../utils/sprompt.c
 
diff --git a/src/test/regression/tests/010.rewrite_timestamp/timestamp/Makefile b/src/test/regression/tests/010.rewrite_timestamp/timestamp/Makefile
index f19c88f29..baf6220e5 100644
--- a/src/test/regression/tests/010.rewrite_timestamp/timestamp/Makefile
+++ b/src/test/regression/tests/010.rewrite_timestamp/timestamp/Makefile
@@ -5,7 +5,6 @@ CFLAGS=-Wall -O0 -g -std=gnu99
 CC=gcc
 
 OBJS=main.o \
-	 $(topsrc_dir)/utils/strlcpy.o \
 	 $(topsrc_dir)/utils/pgstrcasecmp.o \
 	 $(topsrc_dir)/utils/psprintf.o \
 	 $(topsrc_dir)/main/pool_globals.o \
@@ -15,7 +14,6 @@ OBJS=main.o \
 all: all-pre $(PROGRAM)
 
 all-pre:
-	$(MAKE) -C $(topsrc_dir)/utils strlcpy.o
 	$(MAKE) -C $(topsrc_dir)/utils pgstrcasecmp.o
 	$(MAKE) -C $(topsrc_dir)/utils psprintf.o
 	$(MAKE) -C $(topsrc_dir)/main pool_globals.o
diff --git a/src/tools/pgenc/Makefile.am b/src/tools/pgenc/Makefile.am
index 643ad51b4..bff5d4a1d 100644
--- a/src/tools/pgenc/Makefile.am
+++ b/src/tools/pgenc/Makefile.am
@@ -7,7 +7,6 @@ nodist_pg_enc_SOURCES = ssl_utils.c \
 		md5.c \
 		base64.c \
 		pool_passwd.c \
-		strlcpy.c \
 		regex_array.c \
 		pool_config_variables.c \
 		pool_config.c \
@@ -33,8 +32,6 @@ base64.h: ../../../src/include/utils/base64.h
 	rm -f $@ && ln -s $< .
 ssl_utils.h: ../../../src/include/utils/ssl_utils.h
 	rm -f $@ && ln -s $< .
-strlcpy.c: ../../../src/utils/strlcpy.c
-	rm -f $@ && ln -s $< .
 regex_array.c: ../../../src/utils/regex_array.c
 	rm -f $@ && ln -s $< .
 pool_config_variables.c: ../../../src/config/pool_config_variables.c
diff --git a/src/tools/pgmd5/Makefile.am b/src/tools/pgmd5/Makefile.am
index 1bf535a2a..f6fd04075 100644
--- a/src/tools/pgmd5/Makefile.am
+++ b/src/tools/pgmd5/Makefile.am
@@ -5,7 +5,6 @@ dist_pg_md5_SOURCES = pg_md5.c \
 		../fe_port.c
 nodist_pg_md5_SOURCES = md5.c \
 		pool_passwd.c \
-		strlcpy.c \
 		regex_array.c \
 		pool_config_variables.c \
 		pool_config.c \
@@ -23,8 +22,6 @@ pool_passwd.c: ../../../src/auth/pool_passwd.c
 	rm -f $@ && ln -s $< .
 pool_path.c: ../../../src/utils/pool_path.c
 	rm -f $@ && ln -s $< .
-strlcpy.c: ../../../src/utils/strlcpy.c
-	rm -f $@ && ln -s $< .
 regex_array.c: ../../../src/utils/regex_array.c
 	rm -f $@ && ln -s $< .
 pool_config_variables.c: ../../../src/config/pool_config_variables.c
diff --git a/src/tools/watchdog/Makefile.am b/src/tools/watchdog/Makefile.am
index 14226aba5..974bd5953 100644
--- a/src/tools/watchdog/Makefile.am
+++ b/src/tools/watchdog/Makefile.am
@@ -1,5 +1,5 @@
 parser_incdir = $(top_srcdir)/src/include/parser
-AM_CPPFLAGS = -D_GNU_SOURCE -DPOOL_PRIVATE -I $(parser_incdir) -I @PGSQL_INCLUDE_DIR@ 
+AM_CPPFLAGS = -D_GNU_SOURCE -DPOOL_PRIVATE -I $(parser_incdir) -I @PGSQL_INCLUDE_DIR@
 bin_PROGRAMS = wd_cli
 
 bin_SCRIPTS = watchdog_setup
@@ -18,7 +18,6 @@ nodist_wd_cli_SOURCES = ssl_utils.c \
 		fe_memutils.c \
 		stringinfo.h \
 		stringinfo.c \
-		strlcpy.c \
 		socket_stream.c \
 		regex_array.c \
 		psprintf.c \
@@ -37,8 +36,6 @@ md5.h: ../../../src/include/auth/md5.h
 	rm -f $@ && ln -s $< .
 socket_stream.c: ../../../src/utils/socket_stream.c
 	rm -f $@ && ln -s $< .
-strlcpy.c: ../../../src/utils/strlcpy.c
-	rm -f $@ && ln -s $< .
 regex_array.c: ../../../src/utils/regex_array.c
 	rm -f $@ && ln -s $< .
 wd_ipc_conn.c: ../../../src/watchdog/wd_ipc_conn.c
