diff --git a/VTK/GUISupport/Qt/CMakeLists.txt b/VTK/GUISupport/Qt/CMakeLists.txt
index 2c35be7..866f1aa 100644
--- a/VTK/GUISupport/Qt/CMakeLists.txt
+++ b/VTK/GUISupport/Qt/CMakeLists.txt
@@ -166,6 +166,7 @@ IF(DESIRED_QT_VERSION MATCHES 4)
      )
   ENDIF(QT_QTGUI_LIBRARY)
   INCLUDE_DIRECTORIES(${QT_QTDESIGNER_INCLUDE_DIR})
+  INCLUDE_DIRECTORIES(${QT_QTCORE_INCLUDE_DIR})
   QT4_WRAP_CPP ( PluginMocSrcs ${PluginMocHeaders} )
 ELSE(DESIRED_QT_VERSION MATCHES 4)
   QT_WRAP_CPP ( QVTKWidgetPlugin PluginMocSrcs ${PluginMocHeaders} )
diff --git a/VTK/GUISupport/Qt/vtkEventQtSlotConnect.h b/VTK/GUISupport/Qt/vtkEventQtSlotConnect.h
index 6782b90..ced3a67 100644
--- a/VTK/GUISupport/Qt/vtkEventQtSlotConnect.h
+++ b/VTK/GUISupport/Qt/vtkEventQtSlotConnect.h
@@ -43,7 +43,7 @@
 #include "vtkObject.h"
 #include "vtkCommand.h"  // for event defines
 #include "QVTKWin32Header.h"  // for export define
-#include "qobject.h"          // for version info
+#include <qobject.h>          // for version info
 
 class QObject;
 class vtkQtConnections;
diff --git a/VTK/GUISupport/Qt/vtkQtConnection.h b/VTK/GUISupport/Qt/vtkQtConnection.h
index 923ec2e..b189ddc 100644
--- a/VTK/GUISupport/Qt/vtkQtConnection.h
+++ b/VTK/GUISupport/Qt/vtkQtConnection.h
@@ -33,7 +33,7 @@
 
 #include "vtkObject.h"
 #include "vtkCommand.h"  // for event defines
-#include "qobject.h"
+#include <qobject.h>
 
 class QObject;
 class vtkCallbackCommand;
