diff --git a/src/mangojuice.vala b/src/mangojuice.vala
index 8c2cbef..46e7f73 100644
--- a/src/mangojuice.vala
+++ b/src/mangojuice.vala
@@ -2566,7 +2566,7 @@ public class MangoJuice : Adw.Application {
     }
 
     async bool check_vkbasalt_installed_async () {
-        string[] paths = { "/usr/lib/libvkbasalt.so", "/usr/lib/x86_64-linux-gnu/libvkbasalt.so", "/usr/local/lib/libvkbasalt.so", "/usr/lib64/vkbasalt/libvkbasalt.so" };
+        string[] paths = { "@vkbasalt@" };
         foreach (var path in paths) {
             if (FileUtils.test (path, FileTest.EXISTS)) {
                 return true;
