diff --git a/cmake/modules/QtVersionAbstraction.cmake b/cmake/modules/QtVersionAbstraction.cmake
index 5bd853c84..93ddf3cf8 100644
--- a/cmake/modules/QtVersionAbstraction.cmake
+++ b/cmake/modules/QtVersionAbstraction.cmake
@@ -17,6 +17,7 @@ if( Qt5Core_FOUND )
     message(STATUS "Found Qt5 core, checking for further dependencies...")
     find_package(Qt5Network REQUIRED)
     find_package(Qt5Xml REQUIRED)
+    find_package(Qt5Sql REQUIRED)
     find_package(Qt5Concurrent REQUIRED)
     if(UNIT_TESTING)
         find_package(Qt5Test REQUIRED)
