Common subdirectories: green.old/debian and green.new/debian
diff -u green.old/green.h green.new/green.h
--- green.old/green.h	2016-06-12 18:11:56.779434416 -0700
+++ green.new/green.h	2016-06-12 18:14:38.830557379 -0700
@@ -19,7 +19,14 @@
 
 
 #include <stdbool.h>
-#include "glib/poppler.h"
+#include "poppler.h"
+#include "gdk-pixbuf/gdk-pixbuf.h"
+#include "gdk-pixbuf/gdk-pixbuf-core.h"
+#include "gdk-pixbuf/gdk-pixbuf-features.h"
+#include "gdk-pixbuf/gdk-pixbuf-enum-types.h"
+ 
+ #define GREEN_FULLSCREEN	0x0001
+ 
 
 
 #define GREEN_FULLSCREEN	0x0001
diff -u green.old/Makefile green.new/Makefile
--- green.old/Makefile	2016-06-12 18:11:56.779434416 -0700
+++ green.new/Makefile	2016-06-12 18:13:09.591974048 -0700
@@ -17,6 +17,12 @@
 SDL_CFLAGS	:=	$$(sdl-config --cflags)
 SDL_LIBS	:=	$$(sdl-config --libs)
 
+GDKPIXBUF_CFLAGS	:=	$$(pkg-config gdk-pixbuf-2.0 --cflags)
+GDKPIXBUF_LIBS	:=	$$(pkg-config gdk-pixbuf-2.0 --libs)
+
+GTK_CFLAGS	:=	$$(pkg-config gtk+-2.0 --cflags)
+GTK_LIBS	:=	$$(pkg-config gtk+-2.0 --libs)
+
 
 all: green
 
@@ -28,13 +34,14 @@
 	$(INSTALL) green.1 $(MANDIR)/man1/
 
 green: main.o green.o sdl.o
-	$(CC) $^ $(POPPLER_LIBS) $(SDL_LIBS) -o $@
+	$(CC) $^ $(POPPLER_LIBS) $(SDL_LIBS) $(GDKPIXBUF_LIBS) $(GTK_LIBS) -o $@
 
 main.o: main.c green.h
-	$(CC) $(CONFIG) $(CFLAGS) -c $< $(POPPLER_CFLAGS) -o $@
+	$(CC) $(CONFIG) $(CFLAGS) $(GDKPIXBUF_CFLAGS) -c $< $(POPPLER_CFLAGS) -o $@
 
 green.o: green.c green.h
-	$(CC) $(CFLAGS) -c $< $(POPPLER_CFLAGS) -o $@
+	$(CC) $(CFLAGS) -c $< $(POPPLER_CFLAGS)  $(GDKPIXBUF_CFLAGS) -o $@
 
 sdl.o: sdl.c green.h
-	$(CC) $(CFLAGS) -c $< $(POPPLER_CFLAGS) $(SDL_CFLAGS) -o $@
+	$(CC) $(CFLAGS) -c $< $(POPPLER_CFLAGS) $(SDL_CFLAGS) $(GDKPIXBUF_CFLAGS) $(GTK_CFLAGS) -o $@
+
