diff --git a/command/init.go b/command/init.go
index 403ca245b..05d98329a 100644
--- a/command/init.go
+++ b/command/init.go
@@ -64,6 +64,11 @@ func (c *InitCommand) Run(args []string) int {
 		return 1
 	}
 
+	val, ok := os.LookupEnv("NIX_TERRAFORM_PLUGIN_DIR")
+	if ok {
+		flagPluginPath = append(flagPluginPath, val)
+	}
+
 	if len(flagPluginPath) > 0 {
 		c.pluginPath = flagPluginPath
 		c.getPlugins = false
