diff -ruN nix-0.9.2/scripts/nix-profile.sh.in nix-0.9.2.new/scripts/nix-profile.sh.in
--- nix-0.9.2/scripts/nix-profile.sh.in	2005-09-21 20:29:43.000000000 +0200
+++ nix-0.9.2.new/scripts/nix-profile.sh.in	2005-10-11 18:44:57.000000000 +0200
@@ -4,7 +4,7 @@
     if ! test -L "$NIX_LINK"; then
         echo "creating $NIX_LINK"
         _NIX_DEF_LINK=@localstatedir@/nix/profiles/default
-        ln -s "$_NIX_DEF_LINK" "$NIX_LINK"
+        @coreutils@/ln -s "$_NIX_DEF_LINK" "$NIX_LINK"
     fi
 
     export PATH=$NIX_LINK/bin:@prefix@/bin:$PATH
