diff --git a/CMakeLists.txt b/CMakeLists.txt
index 763f75b..defa74a 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -212,7 +212,7 @@ CHECK_INCLUDE_FILES(endian.h HAVE_ENDIAN_H)
 CHECK_INCLUDE_FILES(stdint.h HAVE_STDINT_H)
 CHECK_INCLUDE_FILES(byteswap.h HAVE_BYTESWAP_H)
 CHECK_LIBRARY_EXISTS(gypsy gypsy_control_get_default "" GYPSY_FOUND)
-CHECK_INCLUDE_FILES(libspeechd.h HAVE_LIBSPEECHD)
+CHECK_INCLUDE_FILES(speech-dispatcher/libspeechd.h HAVE_LIBSPEECHD)
 CHECK_INCLUDE_FILES(sys/socket.h HAVE_SOCKET)
 CHECK_INCLUDE_FILES(sys/shm.h HAVE_SHMEM)
 CHECK_FUNCTION_EXISTS(snprintf   HAVE_SNPRINTF)
