diff -rc gnome-vfs-2.8.4-orig/configure gnome-vfs-2.8.4/configure
*** gnome-vfs-2.8.4-orig/configure	2005-02-21 10:20:34.000000000 +0100
--- gnome-vfs-2.8.4/configure	2005-03-02 13:00:41.000000000 +0100
***************
*** 128,134 ****
    case $CONFIG_SHELL in
    '')
      as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
! for as_dir in /bin$PATH_SEPARATOR/usr/bin$PATH_SEPARATOR$PATH
  do
    IFS=$as_save_IFS
    test -z "$as_dir" && as_dir=.
--- 128,134 ----
    case $CONFIG_SHELL in
    '')
      as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
! for as_dir in $PATH
  do
    IFS=$as_save_IFS
    test -z "$as_dir" && as_dir=.
***************
*** 306,312 ****
    # So, first we look for a working echo in the user's PATH.
  
    lt_save_ifs="$IFS"; IFS=$PATH_SEPARATOR
!   for dir in $PATH /usr/ucb; do
      IFS="$lt_save_ifs"
      if (test -f $dir/echo || test -f $dir/echo$ac_exeext) &&
         test "X`($dir/echo '\t') 2>/dev/null`" = 'X\t' &&
--- 306,312 ----
    # So, first we look for a working echo in the user's PATH.
  
    lt_save_ifs="$IFS"; IFS=$PATH_SEPARATOR
!   for dir in $PATH; do
      IFS="$lt_save_ifs"
      if (test -f $dir/echo || test -f $dir/echo$ac_exeext) &&
         test "X`($dir/echo '\t') 2>/dev/null`" = 'X\t' &&
***************
*** 27670,27676 ****
    ;;
    *)
    as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
! as_dummy="$PATH:/usr/kerberos/bin"
  for as_dir in $as_dummy
  do
    IFS=$as_save_IFS
--- 27670,27676 ----
    ;;
    *)
    as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
! as_dummy="$PATH"
  for as_dir in $as_dummy
  do
    IFS=$as_save_IFS
