From 1a526e40ffc1d6cb050334e8641d8b90d9858a54 Mon Sep 17 00:00:00 2001
From: Thomas Tuegel <ttuegel@gmail.com>
Date: Sun, 30 Aug 2015 07:05:03 -0500
Subject: [PATCH] qalculate filename string type

---
 src/backends/qalculate/qalculateexpression.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/backends/qalculate/qalculateexpression.cpp b/src/backends/qalculate/qalculateexpression.cpp
index 1891baa..b2a1984 100644
--- a/src/backends/qalculate/qalculateexpression.cpp
+++ b/src/backends/qalculate/qalculateexpression.cpp
@@ -771,9 +771,9 @@ void QalculateExpression::evaluatePlotCommand()
 	if (plotParameters.filetype == PLOT_FILETYPE_EPS ||
 	    plotParameters.filetype == PLOT_FILETYPE_PS  ||
 	    (plotParameters.filetype == PLOT_FILETYPE_AUTO && p >= 4 &&
-	     plotParameters.filename.substr(p-4,4) == QLatin1String(".eps")) ||
+	     plotParameters.filename.substr(p-4,4) == basic_string<char>(".eps")) ||
 	    (plotParameters.filetype == PLOT_FILETYPE_AUTO && p >= 3 &&
-	     plotParameters.filename.substr(p-3,3) == QLatin1String(".ps")))
+	     plotParameters.filename.substr(p-3,3) == basic_string<char>(".ps")))
 	    setResult(new Cantor::EpsResult(KUrl(plotParameters.filename.c_str())));
 	else
 	    setResult(new Cantor::ImageResult(KUrl(plotParameters.filename.c_str())));
-- 
2.5.0

