diff --git a/debian/elementary-default-settings.gsettings-override b/debian/elementary-default-settings.gsettings-override
index 6452c30..899972d 100644
--- a/debian/elementary-default-settings.gsettings-override
+++ b/debian/elementary-default-settings.gsettings-override
@@ -1,5 +1,5 @@
 [net.launchpad.plank.dock.settings]
-dock-items=['gala-multitaskingview.dockitem','org.gnome.Epiphany.dockitem','org.pantheon.mail.dockitem','io.elementary.calendar.dockitem','io.elementary.music.dockitem','io.elementary.videos.dockitem','io.elementary.photos.dockitem','io.elementary.switchboard.dockitem','io.elementary.appcenter.dockitem']
+dock-items=['gala-multitaskingview.dockitem','org.gnome.Epiphany.dockitem','org.gnome.Geary.dockitem','io.elementary.calendar.dockitem','io.elementary.music.dockitem','io.elementary.videos.dockitem','io.elementary.photos.dockitem','io.elementary.switchboard.dockitem']
 hide-delay=250
 hide-mode='window-dodge'
 show-dock-item=false
@@ -8,13 +8,6 @@ theme='Gtk+'
 [org.freedesktop.ibus.general.hotkey]
 triggers=['<Control>space']
 
-[org.gnome.desktop.background]
-draw-background=true
-picture-options='zoom'
-picture-uri='file:///usr/share/backgrounds/elementaryos-default'
-primary-color='#000000'
-show-desktop-icons=false
-
 [org.gnome.desktop.datetime]
 automatic-timezone=true
 
