diff --git a/libs/widgets/KoCsvImportDialog.cpp b/libs/widgets/KoCsvImportDialog.cpp
index 0ffdcf6..cdca006 100644
--- a/libs/widgets/KoCsvImportDialog.cpp
+++ b/libs/widgets/KoCsvImportDialog.cpp
@@ -21,6 +21,7 @@
 #include "KoCsvImportDialog.h"
 
 // Qt
+#include <QButtonGroup>
 #include <QTextCodec>
 #include <QTextStream>
 
diff --git a/libs/widgets/KoPageLayoutWidget.cpp b/libs/widgets/KoPageLayoutWidget.cpp
index f91555c..a3816f9 100644
--- a/libs/widgets/KoPageLayoutWidget.cpp
+++ b/libs/widgets/KoPageLayoutWidget.cpp
@@ -23,6 +23,8 @@
 
 #include <KoUnit.h>
 
+#include <QButtonGroup>
+
 class Q_DECL_HIDDEN KoPageLayoutWidget::Private
 {
 public:
diff --git a/plugins/chartshape/dialogs/TableEditorDialog.cpp b/plugins/chartshape/dialogs/TableEditorDialog.cpp
index c0d5136..d2a772e 100644
--- a/plugins/chartshape/dialogs/TableEditorDialog.cpp
+++ b/plugins/chartshape/dialogs/TableEditorDialog.cpp
@@ -24,6 +24,7 @@
 
 // Qt
 #include <QAbstractItemModel>
+#include <QAction>
 
 // Calligra
 #include <KoIcon.h>
diff --git a/plugins/formulashape/FormulaToolWidget.cpp b/plugins/formulashape/FormulaToolWidget.cpp
index ed10919..8f52177 100644
--- a/plugins/formulashape/FormulaToolWidget.cpp
+++ b/plugins/formulashape/FormulaToolWidget.cpp
@@ -30,6 +30,7 @@
 #include <QWidgetAction>
 #include <QTableWidget>
 #include <QAction>
+#include <QHeaderView>
 #include <QMenu>
 
 FormulaToolWidget::FormulaToolWidget( KoFormulaTool* tool, QWidget* parent )
diff --git a/sheets/dialogs/LayoutDialog.cpp b/sheets/dialogs/LayoutDialog.cpp
index a0a9832..7d7db53 100644
--- a/sheets/dialogs/LayoutDialog.cpp
+++ b/sheets/dialogs/LayoutDialog.cpp
@@ -36,6 +36,7 @@
 #include <math.h>
 
 #include <QIntValidator>
+#include <QButtonGroup>
 #include <QCheckBox>
 #include <QFrame>
 #include <QLabel>
diff --git a/words/part/dialogs/KWAnchoringProperties.cpp b/words/part/dialogs/KWAnchoringProperties.cpp
index d64208c..bfddb3a 100644
--- a/words/part/dialogs/KWAnchoringProperties.cpp
+++ b/words/part/dialogs/KWAnchoringProperties.cpp
@@ -35,6 +35,7 @@
 
 #include <kundo2command.h>
 
+#include <QButtonGroup>
 #include <QComboBox>
 
 const int KWAnchoringProperties::vertRels[4][20] = {
diff --git a/words/part/dialogs/KWRunAroundProperties.cpp b/words/part/dialogs/KWRunAroundProperties.cpp
index e38599a..7e8b2d5 100644
--- a/words/part/dialogs/KWRunAroundProperties.cpp
+++ b/words/part/dialogs/KWRunAroundProperties.cpp
@@ -28,6 +28,8 @@
 
 #include <kundo2command.h>
 
+#include <QButtonGroup>
+
 KWRunAroundProperties::KWRunAroundProperties(FrameConfigSharedState *state)
         : m_state(state)
 {
diff --git a/stage/part/KPrPresentationTool.cpp b/stage/part/KPrPresentationTool.cpp
index ae743da..3007f91 100644
--- a/stage/part/KPrPresentationTool.cpp
+++ b/stage/part/KPrPresentationTool.cpp
@@ -32,6 +32,7 @@
 #include <QDesktopServices>
 #include <QUrl>
 #include <QDBusConnection>
+#include <QFrame>
 
 #include <KoShape.h>
 #include <KoShapeManager.h>
