diff --git a/flake8_plugins/flake8_class_attributes_plugin/tests/conftest.py b/flake8_plugins/flake8_class_attributes_plugin/tests/conftest.py
index 1ad762aed..c91078dcf 100644
--- a/flake8_plugins/flake8_class_attributes_plugin/tests/conftest.py
+++ b/flake8_plugins/flake8_class_attributes_plugin/tests/conftest.py
@@ -1,6 +1,7 @@
 import ast
 import os
 
+import flake8
 from flake8.options.manager import OptionManager
 
 from flake8_plugins.flake8_class_attributes_plugin.flake8_class_attributes.checker import ClassAttributesChecker
@@ -17,7 +18,11 @@ def run_validator_for_test_file(filename, max_annotations_complexity=None,
         raw_content = file_handler.read()
     tree = ast.parse(raw_content)
 
-    options = OptionManager('flake8_class_attributes_order', '0.1.3')
+    options = OptionManager(
+        version=flake8.__version__,
+        plugin_versions='flake8_class_attributes_order: 0.1.3',
+        parents=[],
+        )
     options.use_class_attributes_order_strict_mode = strict_mode
     options.class_attributes_order = attributes_order
     ClassAttributesChecker.parse_options(options)
