From 59d26b68704373d405b70806f890ddd545cc976a Mon Sep 17 00:00:00 2001
From: Thomas Tuegel <ttuegel@mailbox.org>
Date: Sun, 28 Mar 2021 10:31:48 -0500
Subject: [PATCH 2/2] Debug module loader

---
 src/kiod/kiod_main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/kiod/kiod_main.cpp b/src/kiod/kiod_main.cpp
index 1976ee1..eb402bf 100644
--- a/src/kiod/kiod_main.cpp
+++ b/src/kiod/kiod_main.cpp
@@ -50,7 +50,7 @@ void KIOD::loadModule(const QString &name)
         module = factory->create<KDEDModule>();
     }
     if (!module) {
-        qCWarning(KIOD_CATEGORY) << "Error loading plugin:" << loader.errorString();
+        qCWarning(KIOD_CATEGORY) << "Error loading plugin" << name << loader.errorString();
         return;
     }
     module->setModuleName(name); // makes it register to DBus
-- 
2.30.1

