diff --git a/src/core/ksambashare.cpp b/src/core/ksambashare.cpp
index 1895783..9fe7286 100644
--- a/src/core/ksambashare.cpp
+++ b/src/core/ksambashare.cpp
@@ -73,8 +73,7 @@ KSambaSharePrivate::~KSambaSharePrivate()
 bool KSambaSharePrivate::isSambaInstalled()
 {
     const bool daemonExists =
-        !QStandardPaths::findExecutable(QStringLiteral("smbd"),
-                                       {QStringLiteral("/usr/sbin/"), QStringLiteral("/usr/local/sbin/")}).isEmpty();
+        !QStandardPaths::findExecutable(QStringLiteral("smbd")).isEmpty();
     if (!daemonExists) {
         qCDebug(KIO_CORE_SAMBASHARE) << "KSambaShare: Could not find smbd";
     }
