--- a/resources/pulsar.sh	2023-03-16 04:11:14.000000000 +0100
+++ b/resources/pulsar.sh	2023-03-24 14:37:13.468813964 +0100
@@ -123,22 +123,9 @@
 elif [ $OS == 'Linux' ]; then
   SCRIPT=$(readlink -f "$0")
 
-  PULSAR_PATH="/opt/Pulsar/pulsar"
+  # PULSAR_PATH is set-up via `wrapProgram` in the postFixup phase

-  #Will allow user to get context menu on cinnamon desktop enviroment
-  #Add a check to make sure that DESKTOP_SESSION is set before attempting to grep it
-  #expr substr is expecting 3 arguments string, index, length
-  #If grep doesnt find anything is provides an empty string which causes the expr: syntax error: missing argument after '8' error - see pulsar-edit/pulsar#174
-  #Im also not quite sure why they used grep instead of simply [ "${DESKTOP_SESSION}" == "cinnamon" ]
-  if [ -n "${DESKTOP_SESSION}" ] && [ "$(expr substr $(printenv | grep 'DESKTOP_SESSION=') 17 8)" == "cinnamon" ]; then
-    #This local path is almost assuredly wrong as it shouldnt exist in a standard install
-    ACTION_PATH="resources/linux/desktopenviroment/cinnamon/pulsar.nemo_action"
-
-    #Validate the file exists before attempting to copy it
-    if [ -f "${ACTION_PATH}" ]; then
-        cp "${$ACTION_PATH}" "/usr/share/nemo/actions/pulsar.nemo_action"
-    fi
-  fi
+  # We remove the nemo integration. It is handled by the postFixup phase
 
   #Set tmpdir only if tmpdir is unset
   : ${TMPDIR:=/tmp}
