diff --git a/cpp/ycm/CMakeLists.txt b/cpp/ycm/CMakeLists.txt
index 00e4882..8f29797 100644
--- a/cpp/ycm/CMakeLists.txt
+++ b/cpp/ycm/CMakeLists.txt
@@ -310,11 +310,6 @@ if ( EXTERNAL_LIBCLANG_PATH OR USE_SYSTEM_LIBCLANG )
       # our libraries require, in particular the Python one (from pyenv for
       # instance).
       set( CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE )
-      # When loading our library, the dynamic linker will look for
-      # libclang.so.4, not libclang.so.4.x.
-      file( RENAME
-            ${EXTERNAL_LIBCLANG_PATH}.${CLANG_MAJOR_VERSION}.${CLANG_MINOR_VERSION}
-            ${EXTERNAL_LIBCLANG_PATH}.${CLANG_MAJOR_VERSION} )
     endif()
   endif()
 
