diff --git a/src/libstore/unix/build/local-derivation-goal.cc b/src/libstore/unix/build/local-derivation-goal.cc
--- a/src/libstore/unix/build/local-derivation-goal.cc
+++ b/src/libstore/unix/build/local-derivation-goal.cc
@@ -1189,6 +1189,7 @@ void LocalDerivationGoal::initEnv()
        may change that in the future. So tell the builder which file
        descriptor to use for that. */
     env["NIX_LOG_FD"] = "2";
+    env["NIX_DEBUG"] = "@NIX_DEBUG@";
 
     /* Trigger colored output in various tools. */
     env["TERM"] = "xterm-256color";
