--- a/test/tool/plinko/plinko_test.c
+++ b/test/tool/plinko/plinko_test.c
@@ -89,7 +89,7 @@
     sigaction(SIGQUIT, &savequit, 0);
     sigaction(SIGPIPE, &savepipe, 0);
     sigprocmask(SIG_SETMASK, &savemask, 0);
-    execve("bin/plinko", (char *const[]){"bin/plinko", 0}, (char *const[]){0});
+    execve("bin/plinko", (char *const[]){"bin/plinko", 0}, (char *const[]){"PATH=@coreutils@/bin", 0});
     _exit(127);
   }
   close(pfds[0][0]);
