diff --git a/ustr-compiler.h b/ustr-compiler.h
index 9e71276..c5f847a 100644
--- a/ustr-compiler.h
+++ b/ustr-compiler.h
@@ -11,17 +11,11 @@
 #define USTR_CONF_HAVE_ATTR_FMT 1
 #endif
 
+#include <stdarg.h>
 /* We assume this is enough,
  * C99 specifies that va_copy() exists and is a macro */
-#ifdef va_copy
 # define USTR_CONF_HAVE_VA_COPY 1
 # define USTR__VA_COPY(x, y)   va_copy(x, y)
-#elif __va_copy
-# define USTR_CONF_HAVE_VA_COPY 1
-# define USTR__VA_COPY(x, y) __va_copy(x, y)
-#else
-# define USTR_CONF_HAVE_VA_COPY 0
-#endif
 
 #ifndef USTR_CONF_HAVE_ATTR_NONNULL
 #if defined(__GNUC__) && (__GNUC__ > 3) /* not sure */
