diff --git a/build/pkgs/python3/spkg-build b/build/pkgs/python3/spkg-build
index 56db087ae5..b450703c5f 100644
--- a/build/pkgs/python3/spkg-build
+++ b/build/pkgs/python3/spkg-build
@@ -27,6 +27,8 @@ fi
 export EXTRA_CFLAGS="`testcflags.sh -Wno-unused` $CFLAGS"
 unset CFLAGS
 
+export LDFLAGS="$LDFLAGS -lcrypt"
+
 if [ "$UNAME" = Darwin ]; then
     PYTHON_CONFIGURE="--disable-toolbox-glue $PYTHON_CONFIGURE"
 
