diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5338579..a28e58f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,25 +92,7 @@ endif()

 # build tests? Then we need googletest...
 if (BUILD_TESTS)
-    include(ExternalProject)
-
-    ExternalProject_Add(googletest
-        PREFIX "${CMAKE_CURRENT_BINARY_DIR}/googletest"
-        URL https://github.com/google/googletest/releases/download/v1.15.2/googletest-1.15.2.tar.gz
-        URL_HASH SHA256=7b42b4d6ed48810c5362c265a17faebe90dc2373c885e5216439d37927f02926
-        DOWNLOAD_EXTRACT_TIMESTAMP true
-        TIMEOUT 30
-        DOWNLOAD_NO_PROGRESS true
-        INSTALL_COMMAND "")
-
-    ExternalProject_Get_Property(googletest SOURCE_DIR)
-    include_directories(${SOURCE_DIR}/googlemock/include ${SOURCE_DIR}/googletest/include)
-    ExternalProject_Get_Property(googletest BINARY_DIR)
-    link_directories(${BINARY_DIR}/lib)
     set(GMOCK_BOTH_LIBRARIES gmock_main gmock gtest)
-    set_property(DIRECTORY PROPERTY CLEAN_NO_CUSTOM "${CMAKE_CURRENT_BINARY_DIR}/googletest")
-    unset(SOURCE_DIR)
-    unset(BINARY_DIR)

     enable_testing()
     add_subdirectory(tests)
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 859c048..58ab5c2 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -2,7 +2,6 @@ include(GoogleTest)

 function(add_gmock_test target)
     add_executable(${target} ${ARGN})
-    add_dependencies(${target} googletest)
     target_link_libraries(${target} config playlist bookmarks event_bus ${GMOCK_BOTH_LIBRARIES} ${XDG_BASEDIR_LIBRARIES} ${Boost_LIBRARIES} ${CURL_LIBRARIES} ${JSONCPP_LIBRARIES} pthread)
     target_include_directories(${target} PRIVATE ${JSONCPP_INCLUDE_DIRS})
     gtest_discover_tests(${target})
