diff --git a/src/ocrmypdf/__main__.py b/src/ocrmypdf/__main__.py
index cec02d13..f554ab73 100755
--- a/src/ocrmypdf/__main__.py
+++ b/src/ocrmypdf/__main__.py
@@ -77,7 +77,7 @@ def run(args=None):
 
 
 if __name__ == '__main__':
-    multiprocessing.freeze_support()
+    # multiprocessing.freeze_support()
     if os.name == 'posix':
         multiprocessing.set_start_method('forkserver')
     sys.exit(run())
