diff --git a/cmake/defaults/Packages.cmake b/cmake/defaults/Packages.cmake
index ad3453ed3..99df751a7 100644
--- a/cmake/defaults/Packages.cmake
+++ b/cmake/defaults/Packages.cmake
@@ -294,7 +294,11 @@ if (PXR_BUILD_DRACO_PLUGIN)
 endif()
 
 if (PXR_ENABLE_MATERIALX_SUPPORT)
-    find_package(MaterialX REQUIRED)
+    # MaterialX versions before 1.39.5 do not support package components.
+    find_package(MaterialX QUIET COMPONENTS RenderGlsl)
+    if (NOT MaterialX_FOUND)
+        find_package(MaterialX REQUIRED)
+    endif()
     add_definitions(-DPXR_MATERIALX_SUPPORT_ENABLED)
 endif()
 
diff --git a/pxr/pxrConfig.cmake.in b/pxr/pxrConfig.cmake.in
index 1a28e14d6..4403d036c 100644
--- a/pxr/pxrConfig.cmake.in
+++ b/pxr/pxrConfig.cmake.in
@@ -87,7 +87,11 @@ if(@PXR_ENABLE_MATERIALX_SUPPORT@)
             set(MaterialX_DIR [[@MaterialX_DIR@]])
         endif()
     endif()
-    find_dependency(MaterialX)
+    # MaterialX versions before 1.39.5 do not support package components.
+    find_package(MaterialX QUIET COMPONENTS RenderGlsl)
+    if (NOT MaterialX_FOUND)
+        find_dependency(MaterialX)
+    endif()
 endif()
 
 # Similar to MaterialX above, we are using Imath's cmake package config, so set
