diff --git a/CMakeLists.txt b/CMakeLists.txt
index 37745b5..cad11a0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,13 +16,6 @@ endif()
 set(VERSION "${VERSION}${VERSION_SUFFIX}")
 message("Version ${VERSION}")
 
-set(LLDB_PACKAGE $ENV{LLDB_PACKAGE} CACHE PATH "Zip archive containing LLDB files")
-if (LLDB_PACKAGE)
-    message("Using LLDB_PACKAGE=${LLDB_PACKAGE}")
-else()
-    message(FATAL_ERROR "LLDB_PACKAGE not set." )
-endif()
-
 set(TEST_TIMEOUT 5000 CACHE STRING "Test timeout [ms]")
 
 # General OS-specific definitions
@@ -87,16 +80,6 @@ configure_file(package.json ${CMAKE_CURRENT_BINARY_DIR}/package.json @ONLY)
 configure_file(webpack.config.js ${CMAKE_CURRENT_BINARY_DIR}/webpack.config.js @ONLY)
 file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/package-lock.json DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
 
-# Run 'npm install'
-execute_process(
-    COMMAND ${NPM} ci # like install, but actually respects package-lock file.
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-    RESULT_VARIABLE Result
-)
-if (NOT ${Result} EQUAL 0)
-    message(FATAL_ERROR "npm intall failed: ${Result}")
-endif()
-
 # Copy it back, so we can commit the lock file.
 file(COPY ${CMAKE_CURRENT_BINARY_DIR}/package-lock.json DESTINATION ${CMAKE_CURRENT_SOURCE_DIR})
 

