diff --git a/gcc/config/rs6000/sysv4.h b/gcc/config/rs6000/sysv4.h
index cbee89140dd..e1f26b0a096 100644
--- a/gcc/config/rs6000/sysv4.h
+++ b/gcc/config/rs6000/sysv4.h
@@ -996,13 +996,7 @@ ncrtn.o%s"
     { GPLUSPLUS_BACKWARD_INCLUDE_DIR, "G++", 1, 1,	\
       GPLUSPLUS_INCLUDE_DIR_ADD_SYSROOT, 0 },
 
-#ifdef LOCAL_INCLUDE_DIR
-#define INCLUDE_DEFAULTS_MUSL_LOCAL			\
-    { LOCAL_INCLUDE_DIR, 0, 0, 1, 1, 2 },		\
-    { LOCAL_INCLUDE_DIR, 0, 0, 1, 1, 0 },
-#else
 #define INCLUDE_DEFAULTS_MUSL_LOCAL
-#endif
 
 #ifdef PREFIX_INCLUDE_DIR
 #define INCLUDE_DEFAULTS_MUSL_PREFIX			\
