diff --git a/src/sage/repl/configuration.py b/src/sage/repl/configuration.py
index 67d7d2accf..18279581e2 100644
--- a/src/sage/repl/configuration.py
+++ b/src/sage/repl/configuration.py
@@ -9,10 +9,11 @@ the IPython simple prompt is being used::
     sage: cmd = 'print([sys.stdin.isatty(), sys.stdout.isatty()])'
     sage: import pexpect
     sage: output = pexpect.run(
-    ....:     'bash -c \'echo "{0}" | sage\''.format(cmd),
+    ....:     'bash -c \'export SAGE_BANNER=no; echo "{0}" | sage\''.format(cmd),
     ....: ).decode('utf-8', 'surrogateescape')
-    sage: 'sage: [False, True]' in output
-    True
+    sage: print(output)
+    sage...[False, True]
+    ...
 """
 
 #*****************************************************************************
