diff --git a/src/sage/repl/rich_output/display_manager.py b/src/sage/repl/rich_output/display_manager.py
index fb21f7a9c9..f39470777d 100644
--- a/src/sage/repl/rich_output/display_manager.py
+++ b/src/sage/repl/rich_output/display_manager.py
@@ -749,9 +749,9 @@ class DisplayManager(SageObject):
             import sage.env
             import re
             import os
-            with open(os.path.join(sage.env.THREEJS_DIR, 'build', 'three.min.js')) as f:
+            with open(os.path.join(sage.env.THREEJS_DIR, 'build', 'three.js')) as f:
                 text = f.read().replace('\n','')
-            version = re.search(r'REVISION="(\d+)"', text).group(1)
+            version = re.search(r"REVISION = '(\d+)'", text).group(1)
             return """
 <script src="https://cdn.jsdelivr.net/gh/mrdoob/three.js@r{0}/build/three.min.js"></script>
 <script src="https://cdn.jsdelivr.net/gh/mrdoob/three.js@r{0}/examples/js/controls/OrbitControls.js"></script>
