diff --git a/src/doc/en/prep/Advanced-2DPlotting.rst b/src/doc/en/prep/Advanced-2DPlotting.rst
index 337457afef..f7c76f4b56 100644
--- a/src/doc/en/prep/Advanced-2DPlotting.rst
+++ b/src/doc/en/prep/Advanced-2DPlotting.rst
@@ -695,6 +695,8 @@ by the cells.
 
     sage: pdf_savename = name+'.pdf'
     sage: p.save(pdf_savename)
+    ...
+    DeprecationWarning: The py23 module has been deprecated and will be removed in a future release. Please update your code.
 
 Notably, we can export in formats ready for inclusion in web pages.
 
diff --git a/src/sage/plot/disk.py b/src/sage/plot/disk.py
index 8680a1c9b1..e83763b678 100644
--- a/src/sage/plot/disk.py
+++ b/src/sage/plot/disk.py
@@ -156,6 +156,8 @@ class Disk(GraphicPrimitive):
             sage: f = tmp_filename(ext='.pdf')
             sage: p = disk((0,0), 5, (0, pi/4), alpha=0.5)
             sage: p.save(f)
+            ...
+            DeprecationWarning: The py23 module has been deprecated and will be removed in a future release. Please update your code.
 
         """
         import matplotlib.patches as patches
diff --git a/src/sage/plot/text.py b/src/sage/plot/text.py
index 04cbdedf76..a970f97b79 100644
--- a/src/sage/plot/text.py
+++ b/src/sage/plot/text.py
@@ -325,6 +325,8 @@ def text(string, xy, **options):
     You can save text as part of PDF output::
 
         sage: text("sage", (0,0), rgbcolor=(0,0,0)).save(os.path.join(SAGE_TMP, 'a.pdf'))
+        ...
+        DeprecationWarning: The py23 module has been deprecated and will be removed in a future release. Please update your code.
 
     Some examples of bounding box::
 
