diff --git a/src/sage/tests/cmdline.py b/src/sage/tests/cmdline.py
index bd6b76ab82..ccf1203dec 100644
--- a/src/sage/tests/cmdline.py
+++ b/src/sage/tests/cmdline.py
@@ -837,8 +837,6 @@ def test_executable(args, input="", timeout=100.0, **kwds):
         ///
         4
         }}}
-        sage: err # py2
-        ''
         sage: ret
         0
 
@@ -871,8 +869,8 @@ def test_executable(args, input="", timeout=100.0, **kwds):
         sage: output = tmp_filename(ext='.sws')
         sage: with open(input, 'w') as F:
         ....:     _ = F.write(s)
-        sage: test_executable(["sage", "--rst2sws", input, output]) # py2
-        ('', '', 0)
+        sage: test_executable(["sage", "--rst2sws", input, output])[2] # py2
+        0
         sage: import tarfile # py2
         sage: f = tarfile.open(output, 'r') # py2
         sage: print(f.extractfile('sage_worksheet/worksheet.html').read()) # py2
