reverted:
--- b/src/sage/geometry/polyhedron/backend_cdd.py
+++ a/src/sage/geometry/polyhedron/backend_cdd.py
@@ -154,7 +154,9 @@
             ...    [0.62, -1.38, 0.38],[0.144, -1.04, 0.04],
             ...    [0.1309090909, -1.0290909091, 0.04]]
             sage: Polyhedron(point_list)
+            Traceback (most recent call last):
+            ...
+            ValueError: *Error: Numerical inconsistency is found.  Use the GMP exact arithmetic.
-            A 3-dimensional polyhedron in RDF^3 defined as the convex hull of 14 vertices
             sage: Polyhedron(point_list, base_ring=QQ)
             A 3-dimensional polyhedron in QQ^3 defined as the convex hull of 14 vertices
         """
