diff -ruN procps-3.2.6/Makefile procps-3.2.6.new/Makefile
--- procps-3.2.6/Makefile	2005-10-30 07:27:04.000000000 +0100
+++ procps-3.2.6.new/Makefile	2005-12-13 16:11:40.000000000 +0100
@@ -43,7 +43,7 @@
 usr/lib                  := $(DESTDIR)/usr/$(lib64)/
 usr/include              := $(DESTDIR)/usr/include/
 
-#SKIP     := $(bin)kill $(man1)kill.1
+SKIP     := $(bin)kill $(man1)kill.1
 
 BINFILES := $(usr/bin)uptime $(usr/bin)tload $(usr/bin)free $(usr/bin)w \
             $(usr/bin)top $(usr/bin)vmstat $(usr/bin)watch $(usr/bin)skill \
