diff -ruN procps-3.2.6/Makefile procps-3.2.6.new/Makefile
--- procps-3.2.6/Makefile	2005-10-30 07:27:04.000000000 +0100
+++ procps-3.2.6.new/Makefile	2005-12-13 16:52:02.000000000 +0100
@@ -43,7 +43,7 @@
 usr/lib                  := $(DESTDIR)/usr/$(lib64)/
 usr/include              := $(DESTDIR)/usr/include/
 
-#SKIP     := $(bin)kill $(man1)kill.1
+SKIP     := $(bin)kill $(man1)kill.1 $(usr/bin)uptime $(man1)uptime.1
 
 BINFILES := $(usr/bin)uptime $(usr/bin)tload $(usr/bin)free $(usr/bin)w \
             $(usr/bin)top $(usr/bin)vmstat $(usr/bin)watch $(usr/bin)skill \
@@ -127,26 +127,26 @@
 # produce separate executables for ppc and ppc64, s390 and s390x,
 # i386 and x86-64, mips and mips64, sparc and sparc64, and so on.
 # Failure to do so will cause data corruption.
-m64 := $(call check_gcc,-m64,$(call check_gcc,-mabi=64,))
-ALL_CFLAGS += $(m64)
+#m64 := $(call check_gcc,-m64,$(call check_gcc,-mabi=64,))
+#ALL_CFLAGS += $(m64)
 
-ALL_CFLAGS += $(call check_gcc,-Wdeclaration-after-statement,)
-ALL_CFLAGS += $(call check_gcc,-Wpadded,)
-ALL_CFLAGS += $(call check_gcc,-Wstrict-aliasing,)
+#ALL_CFLAGS += $(call check_gcc,-Wdeclaration-after-statement,)
+#ALL_CFLAGS += $(call check_gcc,-Wpadded,)
+#ALL_CFLAGS += $(call check_gcc,-Wstrict-aliasing,)
 
 # Adding -fno-gcse might be good for those files which
 # use computed goto.
 #ALL_CFLAGS += $(call check_gcc,-fno-gcse,)
 
 # if not debugging, enable things that could confuse gdb
-ifeq (,$(findstring -g,$(filter -g%,$(CFLAGS))))
-ALL_CFLAGS += $(call check_gcc,-fweb,)
-ALL_CFLAGS += $(call check_gcc,-frename-registers,)
-ALL_CFLAGS += $(call check_gcc,-fomit-frame-pointer,)
-endif
+#ifeq (,$(findstring -g,$(filter -g%,$(CFLAGS))))
+#ALL_CFLAGS += $(call check_gcc,-fweb,)
+#ALL_CFLAGS += $(call check_gcc,-frename-registers,)
+#ALL_CFLAGS += $(call check_gcc,-fomit-frame-pointer,)
+#endif
 
 # in case -O3 is enabled, avoid bloat
-ALL_CFLAGS += $(call check_gcc,-fno-inline-functions,)
+#ALL_CFLAGS += $(call check_gcc,-fno-inline-functions,)
 
 endif
 endif
