diff -u ddclient-3.8.1/ddclient ddclient-3.8.1.patched/ddclient
--- ddclient-3.8.1/ddclient	2011-07-11 23:04:21.000000000 +0200
+++ ddclient-3.8.1.patched/ddclient	2012-11-08 11:52:31.930647236 +0100
@@ -574,7 +574,7 @@
     "usage: ${program} [options]",
     "options are:",
     [ "daemon",      "=s", "-daemon delay         : run as a daemon, specify delay as an interval." ],
-+     [ "foreground",  "!",  "-foreground           : do not fork" ],
+    [ "foreground",  "!",  "-foreground           : do not fork" ],
     [ "proxy",       "=s", "-proxy host           : use 'host' as the HTTP proxy" ],
     [ "server",      "=s", "-server host          : update DNS information on 'host'" ],
     [ "protocol",    "=s", "-protocol type        : update protocol used" ],
