diff --git a/setup.py b/setup.py
index 81f8727ed6..12200e0b9f 100755
--- a/setup.py
+++ b/setup.py
@@ -43,13 +43,13 @@ REQUIRES = [
     "jinja2>=2.11.1",
     "PyJWT==1.7.1",
     # PyJWT has loose dependency. We want the latest one.
-    "cryptography==2.9.2",
+    "cryptography>=2.9.2",
     "pip>=8.0.3",
     "python-slugify==4.0.1",
     "pytz>=2020.1",
     "pyyaml==5.3.1",
-    "requests==2.24.0",
-    "ruamel.yaml==0.15.100",
+    "requests>=2.23.0",
+    "ruamel.yaml>=0.15.100",
     "voluptuous==0.11.7",
     "voluptuous-serialize==2.4.0",
     "yarl==1.4.2",
