diff --git a/setup.py b/setup.py
index c2042ab245..98f348510f 100755
--- a/setup.py
+++ b/setup.py
@@ -32,7 +32,7 @@ PROJECT_URLS = {
 PACKAGES = find_packages(exclude=["tests", "tests.*"])
 
 REQUIRES = [
-    "aiohttp==3.6.1",
+    "aiohttp>=3.6.1",
     "astral==1.10.1",
     "async_timeout==3.0.1",
     "attrs==19.3.0",
@@ -48,8 +48,8 @@ REQUIRES = [
     "python-slugify==4.0.0",
     "pytz>=2020.1",
     "pyyaml==5.3.1",
-    "requests==2.24.0",
-    "ruamel.yaml==0.15.100",
+    "requests>=2.23.0",
+    "ruamel.yaml>=0.15.100",
     "voluptuous==0.11.7",
     "voluptuous-serialize==2.4.0",
 ]
