--- kklib/mimalloc/src/random.c	1969-12-31 18:00:01.000000000 -0600
+++ kklib/mimalloc/src/random.c.new	2022-01-16 19:43:54.000000000 -0600
@@ -195,6 +195,7 @@
 #elif defined(__APPLE__)
 #include <AvailabilityMacros.h>
 #if defined(MAC_OS_X_VERSION_10_10) && MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_10
+#include <CommonCrypto/CommonCryptoError.h>
 #include <CommonCrypto/CommonRandom.h>
 #endif
 static bool os_random_buf(void* buf, size_t buf_len) {
