diff --git a/3party/CMakeLists.txt b/3party/CMakeLists.txt
index c802251db8..314f4d4543 100644
--- a/3party/CMakeLists.txt
+++ b/3party/CMakeLists.txt
@@ -33,9 +33,6 @@ if (NOT WITH_SYSTEM_PROVIDED_3PARTY)
   # Add pugixml library.
   add_subdirectory(pugixml)
 
-  # Add protobuf library.
-  add_subdirectory(protobuf)
-
   add_subdirectory(freetype)
   add_subdirectory(icu)
   add_subdirectory(harfbuzz)
@@ -45,6 +42,9 @@ if (NOT WITH_SYSTEM_PROVIDED_3PARTY)
   target_include_directories(utf8cpp SYSTEM INTERFACE "${OMIM_ROOT}/3party/utfcpp/source")
 endif()
 
+# Add protobuf library.
+add_subdirectory(protobuf)
+
 add_subdirectory(agg)
 add_subdirectory(bsdiff-courgette)
 add_subdirectory(glaze)
