diff --git a/weasyprint/text/ffi.py b/weasyprint/text/ffi.py
index 0734cbea..22e31a5e 100644
--- a/weasyprint/text/ffi.py
+++ b/weasyprint/text/ffi.py
@@ -387,22 +387,11 @@ def _dlopen(ffi, *names):
     return ffi.dlopen(names[0])  # pragma: no cover
 
 
-gobject = _dlopen(
-    ffi, 'gobject-2.0-0', 'gobject-2.0', 'libgobject-2.0-0',
-    'libgobject-2.0.so.0', 'libgobject-2.0.dylib',  'libgobject-2.0-0.dll')
-pango = _dlopen(
-    ffi, 'pango-1.0-0', 'pango-1.0', 'libpango-1.0-0', 'libpango-1.0.so.0',
-    'libpango-1.0.dylib', 'libpango-1.0-0.dll')
-harfbuzz = _dlopen(
-    ffi, 'harfbuzz', 'harfbuzz-0.0', 'libharfbuzz-0',
-    'libharfbuzz.so.0', 'libharfbuzz.so.0', 'libharfbuzz.0.dylib',
-    'libharfbuzz-0.dll')
-fontconfig = _dlopen(
-    ffi, 'fontconfig-1', 'fontconfig', 'libfontconfig', 'libfontconfig.so.1',
-    'libfontconfig-1.dylib', 'libfontconfig-1.dll')
-pangoft2 = _dlopen(
-    ffi, 'pangoft2-1.0-0', 'pangoft2-1.0', 'libpangoft2-1.0-0',
-    'libpangoft2-1.0.so.0', 'libpangoft2-1.0.dylib', 'libpangoft2-1.0-0.dll')
+gobject = _dlopen(ffi, '@gobject@')
+pango = _dlopen(ffi, '@pango@')
+harfbuzz = _dlopen(ffi, '@harfbuzz@')
+fontconfig = _dlopen(ffi, '@fontconfig@')
+pangoft2 = _dlopen(ffi, '@pangoft2@')
 
 gobject.g_type_init()

