diff -Naurd x/tensorflow/workspace2.bzl y/tensorflow/workspace2.bzl
--- x/tensorflow/workspace2.bzl	2023-09-21 12:55:06.104407343 +0000
+++ y/tensorflow/workspace2.bzl	2023-09-21 12:55:39.732453203 +0000
@@ -872,9 +872,9 @@
 
     tf_http_archive(
         name = "pybind11_protobuf",
-        urls = tf_mirror_urls("https://github.com/pybind/pybind11_protobuf/archive/80f3440cd8fee124e077e2e47a8a17b78b451363.zip"),
-        sha256 = "c7ab64b1ccf9a678694a89035a8c865a693e4e872803778f91f0965c2f281d78",
-        strip_prefix = "pybind11_protobuf-80f3440cd8fee124e077e2e47a8a17b78b451363",
+        urls = tf_mirror_urls("https://github.com/pybind/pybind11_protobuf/archive/c8cc30e2495309e3499b7d76033446236d21c837.zip"),
+        sha256 = "0663f73b34c0b4af55003edbb066e62aceff99bfcf12e47ea9a15d6013d81413",
+        strip_prefix = "pybind11_protobuf-c8cc30e2495309e3499b7d76033446236d21c837",
         patch_file = [
             "//third_party/pybind11_protobuf:remove_license.patch",
             "//third_party/pybind11_protobuf:python_runtime_dep.patch",
