diff --git a/tensorflow/tools/pip_package/setup.py b/tensorflow/tools/pip_package/setup.py
index 594e74f40c0..bfbf010144f 100644
--- a/tensorflow/tools/pip_package/setup.py
+++ b/tensorflow/tools/pip_package/setup.py
@@ -54,9 +54,9 @@ _VERSION = '2.3.1'
 REQUIRED_PACKAGES = [
     'absl-py >= 0.7.0',
     'astunparse == 1.6.3',
-    'gast == 0.3.3',
+    'gast >= 0.3.3',
     'google_pasta >= 0.1.8',
-    'h5py >= 2.10.0, < 2.11.0',
+    'h5py >= 2.10.0',
     'keras_preprocessing >= 1.1.1, < 1.2',
     # TODO(mihaimaruseac): numpy 1.19.0 has ABI breakage
     # https://github.com/numpy/numpy/pull/15355
