diff --git i/source/config.h.in w/source/config.h.in
index ffcce3a..0caba6d 100644
--- i/source/config.h.in
+++ w/source/config.h.in
@@ -107,7 +107,7 @@ namespace cmd {
     }
 
     namespace wrapper {
-        constexpr char DEFAULT_PATH[] = "@ROOT_INSTALL_PREFIX@/@PRIVATE_INSTALLDIR@/wrapper";
+        constexpr char DEFAULT_PATH[] = "@PRIVATE_INSTALLDIR@/wrapper";
         constexpr char DEFAULT_DIR_PATH[] = "@ROOT_INSTALL_PREFIX@/@PRIVATE_INSTALLDIR@/wrapper.d";
 
         constexpr char FLAG_VERBOSE[] = "--verbose";
@@ -120,7 +120,7 @@ namespace cmd {
     }
 
     namespace library {
-        constexpr char DEFAULT_PATH[] = "@ROOT_INSTALL_PREFIX@/@PRIVATE_INSTALLDIR@/@CMAKE_SHARED_LIBRARY_PREFIX@exec@CMAKE_SHARED_LIBRARY_SUFFIX@";
+        constexpr char DEFAULT_PATH[] = "@PRIVATE_INSTALLDIR@/@CMAKE_SHARED_LIBRARY_PREFIX@exec@CMAKE_SHARED_LIBRARY_SUFFIX@";
 
         constexpr char KEY_REPORTER[] = "INTERCEPT_REPORT_COMMAND";
         constexpr char KEY_DESTINATION[] = "INTERCEPT_REPORT_DESTINATION";
