Index: wsjtx/CMakeLists.txt
===================================================================
--- wsjtx/CMakeLists.txt	(784f75)
+++ wsjtx/CMakeLists.txt	(working copy)
@@ -860,6 +860,7 @@
 find_package (Qt5Multimedia 5 REQUIRED)
 find_package (Qt5PrintSupport 5 REQUIRED)
 find_package (Qt5Sql 5 REQUIRED)
+find_package (Qt5SerialPort 5 REQUIRED)
 
 if (WIN32)
   add_definitions (-DQT_NEEDS_QTMAIN)
