diff -rc coreutils-5.97-orig/lib/strftime.c coreutils-5.97/lib/strftime.c
*** coreutils-5.97-orig/lib/strftime.c	2006-10-18 17:10:16.000000000 +0200
--- coreutils-5.97/lib/strftime.c	2006-10-18 17:08:16.000000000 +0200
***************
*** 180,185 ****
--- 180,191 ----
  # define memset_zero(P, Len) (memset (P, '0', Len), (P) += (Len))
  #endif
  
+ #if FPRINTFTIME
+ #define advance(p, n) ;
+ #else
+ #define advance(p, n) p += _n;
+ #endif 
+ 
  #define add(n, f)							      \
    do									      \
      {									      \
***************
*** 198,204 ****
  		memset_space (p, _delta);				      \
  	    }								      \
  	  f;								      \
! 	  p += FPRINTFTIME ? 0 : _n;					      \
  	}								      \
        i += _incr;							      \
      } while (0)
--- 204,210 ----
  		memset_space (p, _delta);				      \
  	    }								      \
  	  f;								      \
! 	  advance(p, n);						      \
  	}								      \
        i += _incr;							      \
      } while (0)
