--- a/src/odb/src/swig/python/dbtypes.i
+++ b/src/odb/src/swig/python/dbtypes.i
@@ -271,7 +271,7 @@ WRAP_OBJECT_RETURN_REF(odb::dbViaParams, params_return)
   swig_type_info *tf = SWIG_TypeQuery("odb::dbShape" "*");
   for(std::vector<odb::dbShape>::iterator it = $1->begin(); it != $1->end(); it++) {
     PyObject *o = SWIG_NewInstanceObj(&(*it), tf, 0);
-    $result = SWIG_Python_AppendOutput($result, o);
+    $result = SWIG_AppendOutput($result, o);
   }
 }
 
@@ -283,14 +283,14 @@ WRAP_OBJECT_RETURN_REF(odb::dbViaParams, params_return)
     auto layer = it->second;
     PyObject *layer_swig = SWIG_NewInstanceObj(layer, tf, 0);
     PyObject *tuple = PyTuple_Pack(2, PyFloat_FromDouble(value), layer_swig);
-    $result = SWIG_Python_AppendOutput($result, tuple);
+    $result = SWIG_AppendOutput($result, tuple);
   }
 }
 
 %typemap(argout) std::vector<int> &OUTPUT {
   for(auto it = $1->begin(); it != $1->end(); it++) {
     PyObject *obj = PyInt_FromLong((long)*it);
-    $result = SWIG_Python_AppendOutput($result, obj);
+    $result = SWIG_AppendOutput($result, obj);
   }
 }
 
