diff --git a/configure.ac b/configure.ac
index 0055a70..fd4a905 100644
--- a/configure.ac
+++ b/configure.ac
@@ -111,18 +111,6 @@ AS_IF([ test "$enable_man" != no ], [
   ])
 ])
 
-AS_IF([test "$enable_man" != no], [
-  dnl check for DocBook XSL stylesheets in the local catalog
-  JH_CHECK_XML_CATALOG([http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl],
-     [DocBook XSL Stylesheets], [have_docbook_style=yes],[have_docbook_style=no])
-  AS_IF([ test "$have_docbook_style" != yes ], [
-    AS_IF([ test "$enable_man" = yes ], [
-      AC_MSG_ERROR([DocBook XSL Stylesheets are required for --enable-man])
-    ])
-    enable_man=no
-  ])
-])
-
 AM_CONDITIONAL(ENABLE_MAN, test "$enable_man" != no)
 
 AC_MSG_CHECKING([whether to generate man pages])
