diff -ur source/src/btf.cpp new/src/btf.cpp
--- source/src/btf.cpp	1970-01-01 09:00:01.000000000 +0900
+++ new/src/btf.cpp	2021-12-04 21:46:59.337023489 +0900
@@ -225,7 +225,7 @@
   char err_buf[256];
   int err;
 
-  dump = btf_dump__new(btf, nullptr, &opts, dump_printf);
+  dump = btf_dump__new_deprecated(btf, nullptr, &opts, dump_printf);
   err = libbpf_get_error(dump);
   if (err)
   {
@@ -496,7 +496,7 @@
   char err_buf[256];
   int err;
 
-  dump = btf_dump__new(btf, nullptr, &opts, dump_printf);
+  dump = btf_dump__new_deprecated(btf, nullptr, &opts, dump_printf);
   err = libbpf_get_error(dump);
   if (err)
   {
@@ -554,7 +554,7 @@
   char err_buf[256];
   int err;
 
-  dump = btf_dump__new(btf, nullptr, &opts, dump_printf);
+  dump = btf_dump__new_deprecated(btf, nullptr, &opts, dump_printf);
   err = libbpf_get_error(dump);
   if (err)
   {
@@ -648,7 +648,7 @@
   char err_buf[256];
   int err;
 
-  dump = btf_dump__new(btf, nullptr, &opts, dump_printf);
+  dump = btf_dump__new_deprecated(btf, nullptr, &opts, dump_printf);
   err = libbpf_get_error(dump);
   if (err)
   {
