diff --git a/daemon/buildinfo.c b/daemon/buildinfo.c
index b64a78f29..ccd1c884d 100644
--- a/daemon/buildinfo.c
+++ b/daemon/buildinfo.c
@@ -60,7 +60,6 @@
 // Optional libraries
 
 #ifdef HAVE_PROTOBUF
-#if defined(ACLK_NG) || defined(ENABLE_PROMETHEUS_REMOTE_WRITE)
 #define FEAT_PROTOBUF 1
 #ifdef BUNDLED_PROTOBUF
 #define FEAT_PROTOBUF_BUNDLED " (bundled)"
@@ -71,10 +70,6 @@
 #define FEAT_PROTOBUF 0
 #define FEAT_PROTOBUF_BUNDLED ""
 #endif
-#else
-#define FEAT_PROTOBUF 0
-#define FEAT_PROTOBUF_BUNDLED ""
-#endif
 
 #ifdef ENABLE_JSONC
 #define FEAT_JSONC 1
