diff --git a/collectors/Makefile.am b/collectors/Makefile.am
index 021e2ff23..115b88277 100644
--- a/collectors/Makefile.am
+++ b/collectors/Makefile.am
@@ -33,7 +33,7 @@ usercustompluginsconfigdir=$(configdir)/custom-plugins.d
 usergoconfigdir=$(configdir)/go.d
 
 # Explicitly install directories to avoid permission issues due to umask
-install-exec-local:
+no-install-exec-local:
 	$(INSTALL) -d $(DESTDIR)$(usercustompluginsconfigdir)
 	$(INSTALL) -d $(DESTDIR)$(usergoconfigdir)
 
diff --git a/collectors/charts.d.plugin/Makefile.am b/collectors/charts.d.plugin/Makefile.am
index 03c7f0a94..01985db01 100644
--- a/collectors/charts.d.plugin/Makefile.am
+++ b/collectors/charts.d.plugin/Makefile.am
@@ -34,7 +34,7 @@ dist_userchartsconfig_DATA = \
     $(NULL)
 
 # Explicitly install directories to avoid permission issues due to umask
-install-exec-local:
+no-install-exec-local:
 	$(INSTALL) -d $(DESTDIR)$(userchartsconfigdir)
 
 chartsconfigdir=$(libconfigdir)/charts.d
diff --git a/collectors/ebpf.plugin/Makefile.am b/collectors/ebpf.plugin/Makefile.am
index 18b1fc6c8..b4b0c7852 100644
--- a/collectors/ebpf.plugin/Makefile.am
+++ b/collectors/ebpf.plugin/Makefile.am
@@ -13,7 +13,7 @@ SUFFIXES = .in
 userebpfconfigdir=$(configdir)/ebpf.d
 
 # Explicitly install directories to avoid permission issues due to umask
-install-exec-local:
+no-install-exec-local:
 	$(INSTALL) -d $(DESTDIR)$(userebpfconfigdir)
 
 dist_plugins_SCRIPTS = \
diff --git a/collectors/node.d.plugin/Makefile.am b/collectors/node.d.plugin/Makefile.am
index c3142d433..95e324455 100644
--- a/collectors/node.d.plugin/Makefile.am
+++ b/collectors/node.d.plugin/Makefile.am
@@ -26,7 +26,7 @@ dist_usernodeconfig_DATA = \
     $(NULL)
 
 # Explicitly install directories to avoid permission issues due to umask
-install-exec-local:
+no-install-exec-local:
 	$(INSTALL) -d $(DESTDIR)$(usernodeconfigdir)
 
 nodeconfigdir=$(libconfigdir)/node.d
diff --git a/collectors/python.d.plugin/Makefile.am b/collectors/python.d.plugin/Makefile.am
index 38eb90f79..ce7079441 100644
--- a/collectors/python.d.plugin/Makefile.am
+++ b/collectors/python.d.plugin/Makefile.am
@@ -32,7 +32,7 @@ dist_userpythonconfig_DATA = \
     $(NULL)
 
 # Explicitly install directories to avoid permission issues due to umask
-install-exec-local:
+no-install-exec-local:
 	$(INSTALL) -d $(DESTDIR)$(userpythonconfigdir)
 
 pythonconfigdir=$(libconfigdir)/python.d
diff --git a/collectors/statsd.plugin/Makefile.am b/collectors/statsd.plugin/Makefile.am
index 71f2d468d..2c9ced2bf 100644
--- a/collectors/statsd.plugin/Makefile.am
+++ b/collectors/statsd.plugin/Makefile.am
@@ -18,5 +18,5 @@ dist_userstatsdconfig_DATA = \
     $(NULL)
 
 # Explicitly install directories to avoid permission issues due to umask
-install-exec-local:
+no-install-exec-local:
 	$(INSTALL) -d $(DESTDIR)$(userstatsdconfigdir)
diff --git a/health/Makefile.am b/health/Makefile.am
index b963ea0cd..6979e69bf 100644
--- a/health/Makefile.am
+++ b/health/Makefile.am
@@ -19,7 +19,7 @@ dist_userhealthconfig_DATA = \
     $(NULL)
 
 # Explicitly install directories to avoid permission issues due to umask
-install-exec-local:
+no-install-exec-local:
 	$(INSTALL) -d $(DESTDIR)$(userhealthconfigdir)
 
 healthconfigdir=$(libconfigdir)/health.d
diff --git a/system/Makefile.am b/system/Makefile.am
index 5323738c9..06e1b6a73 100644
--- a/system/Makefile.am
+++ b/system/Makefile.am
@@ -20,11 +20,10 @@ include $(top_srcdir)/build/subst.inc
 SUFFIXES = .in
 
 dist_config_SCRIPTS = \
-    edit-config \
     $(NULL)
 
 # Explicitly install directories to avoid permission issues due to umask
-install-exec-local:
+no-install-exec-local:
 	$(INSTALL) -d $(DESTDIR)$(configdir)
 
 nodist_noinst_DATA = \
diff --git a/web/Makefile.am b/web/Makefile.am
index ccaccd764..16a2977e5 100644
--- a/web/Makefile.am
+++ b/web/Makefile.am
@@ -12,7 +12,7 @@ SUBDIRS = \
 usersslconfigdir=$(configdir)/ssl
 
 # Explicitly install directories to avoid permission issues due to umask
-install-exec-local:
+no-install-exec-local:
 	$(INSTALL) -d $(DESTDIR)$(usersslconfigdir)
 
 dist_noinst_DATA = \
