diff --git a/tests/metrics_test.py b/tests/metrics_test.py
index 01bb9fa..8555b52 100644
--- a/tests/metrics_test.py
+++ b/tests/metrics_test.py
@@ -4,18 +4,22 @@ from __future__ import annotations
 
 from typesense.metrics import Metrics
 
+import platform
+
 
 def test_actual_retrieve(actual_metrics: Metrics) -> None:
     """Test that the Debug object can retrieve a debug on Typesense server and verify response structure."""
     response = actual_metrics.retrieve()
 
-    assert "system_cpu_active_percentage" in response
+    if platform.system() == "Linux":
+        assert "system_cpu_active_percentage" in response
+        assert "system_network_received_bytes" in response
+        assert "system_network_sent_bytes" in response
+
     assert "system_disk_total_bytes" in response
     assert "system_disk_used_bytes" in response
     assert "system_memory_total_bytes" in response
     assert "system_memory_used_bytes" in response
-    assert "system_network_received_bytes" in response
-    assert "system_network_sent_bytes" in response
     assert "typesense_memory_active_bytes" in response
     assert "typesense_memory_allocated_bytes" in response
     assert "typesense_memory_fragmentation_ratio" in response
