diff --git a/src/qml/qml/qqmltypeloader.cpp b/src/qml/qml/qqmltypeloader.cpp
index 5d09bf21c6..eacc8ccaea 100644
--- a/src/qml/qml/qqmltypeloader.cpp
+++ b/src/qml/qml/qqmltypeloader.cpp
@@ -1235,7 +1235,8 @@ QQmlTypeLoader::QQmlTypeLoader(QQmlEngine *engine)
     // env import paths
     addEnvImportPath("QML_IMPORT_PATH");
     addEnvImportPath("QML2_IMPORT_PATH");
-
+    addEnvImportPath("NIXPKGS_QT6_QML_IMPORT_PATH");
+ 
     addImportPath(QStringLiteral("qrc:/qt/qml"));
     addImportPath(QStringLiteral("qrc:/qt-project.org/imports"));
 
