diff --git a/data/lightdm.conf b/data/lightdm.conf
index 16b80f7..b3af435 100644
--- a/data/lightdm.conf
+++ b/data/lightdm.conf
@@ -28,7 +28,7 @@
 #guest-account-script=guest-account
 #logind-check-graphical=false
 #log-directory=/var/log/lightdm
-#run-directory=/var/run/lightdm
+run-directory=/run/lightdm
 #cache-directory=/var/cache/lightdm
 #sessions-directory=/usr/share/lightdm/sessions:/usr/share/xsessions:/usr/share/wayland-sessions
 #remote-sessions-directory=/usr/share/lightdm/remote-sessions
