diff --git a/config/config.exs b/config/config.exs
index d75b2e10a..d46ebffd2 100644
--- a/config/config.exs
+++ b/config/config.exs
@@ -128,7 +128,9 @@ config :mobilizon, Mobilizon.Web.Email.Mailer,
   tls_options: [
     verify: :verify_peer,
     versions: [:"tlsv1.2", :"tlsv1.3"],
-    cacerts: :public_key.cacerts_get(),
+    cacerts: if System.get_env("SSL_CERT_FILE") != nil and :public_key.cacerts_load(System.get_env("SSL_CERT_FILE")) == :ok do
+        :public_key.cacerts_get()
+      end,
     server_name_indication: ~c"localhost",
     depth: 99
   ],

