diff --git a/src/planner-gantt-row.c b/src/planner-gantt-row.c
index a981d10..c881736 100644
--- a/src/planner-gantt-row.c
+++ b/src/planner-gantt-row.c
@@ -943,7 +943,7 @@ gantt_row_setup_frame_gc (PlannerGanttRow *row, gboolean highlight)
 {
 	gdk_gc_set_line_attributes (row->priv->frame_gc,
 				    0,
-				    highlight ? GDK_LINE_ON_OFF_DASH : GDK_LINE_SOLID,
+				    /*highlight ? GDK_LINE_ON_OFF_DASH :*/ GDK_LINE_SOLID,
 				    GDK_CAP_BUTT,
 				    GDK_JOIN_MITER);
 }
@@ -1789,7 +1789,7 @@ gantt_row_draw (GnomeCanvasItem *item,
 
  				gdk_gc_set_line_attributes (priv->frame_gc,
  								0,
- 								GDK_LINE_ON_OFF_DASH,
+								GDK_LINE_SOLID /*GDK_LINE_ON_OFF_DASH*/,
  								GDK_CAP_BUTT,
  								GDK_JOIN_MITER);
  				draw_cut_line (drawable, priv->frame_gc, cx1, cy1, cx2, cy1);
@@ -1858,7 +1858,7 @@ gantt_row_draw (GnomeCanvasItem *item,
 
 				gdk_gc_set_line_attributes (priv->frame_gc,
 								0,
-								GDK_LINE_ON_OFF_DASH,
+								GDK_LINE_SOLID /*GDK_LINE_ON_OFF_DASH*/,
 								GDK_CAP_BUTT,
 								GDK_JOIN_MITER);
 			}
