diff -Naur kdesdk-4.10.2-upstream/kdesdk-thumbnailers/CMakeLists.txt kdesdk-4.10.2/kdesdk-thumbnailers/CMakeLists.txt
--- kdesdk-4.10.2-upstream/kdesdk-thumbnailers/CMakeLists.txt	2013-05-02 21:05:56.446226846 -0430
+++ kdesdk-4.10.2/kdesdk-thumbnailers/CMakeLists.txt	2013-05-02 21:09:58.504053136 -0430
@@ -9,7 +9,7 @@
 macro_log_feature(GETTEXTPO_FOUND "gettext-po" "A library for processing PO files" "www.gnu.org/software/gettext" FALSE "" "Required to build the PO thumbnailer")
 
 if(GETTEXTPO_FOUND)
-macro_optional_add_subdirectory(po_thumbnailer)
+add_subdirectory(po_thumbnailer)
 endif(GETTEXTPO_FOUND)
 
 macro_display_feature_log()
