Choose the right CDROM reader syntax for `cd-paranoia'.

--- abcde-2.3.99.6/abcde	2006-08-05 21:14:00.000000000 +0200
+++ abcde-2.3.99.6/abcde	2008-10-29 22:55:38.000000000 +0100
@@ -3184,7 +3184,10 @@ if [ -n "$DISCID" ] || [ "$CDROMREADERSY
 if [ "$CDROMREADERSYNTAX" = "" ]; then
 	for DEFAULT_CDROMREADER in $DEFAULT_CDROMREADERS; do
 		if new_checkexec $DEFAULT_CDROMREADER; then
-			CDROMREADERSYNTAX=$DEFAULT_CDROMREADER
+                        case "$DEFAULT_CDROMREADER" in
+			  cd-paranoia)  CDROMREADERSYNTAX=cdparanoia;;
+			  *)            CDROMREADERSYNTAX=$DEFAULT_CDROMREADER;;
+			esac
 			break
 		fi
 	done
