diff --git a/Source/CMakeLists.txt b/Source/CMakeLists.txt
index c268a92111..2e736af7a7 100644
--- a/Source/CMakeLists.txt
+++ b/Source/CMakeLists.txt
@@ -916,7 +916,6 @@ endif()
 # On Apple we need CoreFoundation and CoreServices
 if(APPLE)
   target_link_libraries(CMakeLib PUBLIC "-framework CoreFoundation")
-  target_link_libraries(CMakeLib PUBLIC "-framework CoreServices")
 endif()
 
 if(WIN32 AND NOT UNIX)
diff --git a/Source/cmGlobalXCodeGenerator.cxx b/Source/cmGlobalXCodeGenerator.cxx
index 116e510c6f..69ce510ff7 100644
--- a/Source/cmGlobalXCodeGenerator.cxx
+++ b/Source/cmGlobalXCodeGenerator.cxx
@@ -56,10 +56,6 @@
 
 #if !defined(CMAKE_BOOTSTRAP) && defined(__APPLE__)
 #  include <CoreFoundation/CoreFoundation.h>
-#  if !TARGET_OS_IPHONE
-#    define HAVE_APPLICATION_SERVICES
-#    include <ApplicationServices/ApplicationServices.h>
-#  endif
 #endif
 
 #if !defined(CMAKE_BOOTSTRAP)
diff --git a/Utilities/cmlibarchive/CMakeLists.txt b/Utilities/cmlibarchive/CMakeLists.txt
index b38e6530a3..8aa71d95cd 100644
--- a/Utilities/cmlibarchive/CMakeLists.txt
+++ b/Utilities/cmlibarchive/CMakeLists.txt
@@ -2041,11 +2041,6 @@ IF(ENABLE_TEST)
   ADD_CUSTOM_TARGET(run_all_tests)
 ENDIF(ENABLE_TEST)
 
-# We need CoreServices on Mac OS.
-IF(APPLE)
-  LIST(APPEND ADDITIONAL_LIBS "-framework CoreServices")
-ENDIF(APPLE)
-
 add_subdirectory(libarchive)
 IF(0) # CMake does not build libarchive's command-line tools.
 add_subdirectory(cat)
