diff -ur cmake-3.12.1/Utilities/cmlibuv/CMakeLists.txt cmake-3.12.1-patched/Utilities/cmlibuv/CMakeLists.txt
--- cmake-3.12.1/Utilities/cmlibuv/CMakeLists.txt	2018-08-09 21:14:08.000000000 +0900
+++ cmake-3.12.1-patched/Utilities/cmlibuv/CMakeLists.txt	2018-08-13 10:14:53.000000000 +0900
@@ -173,6 +173,22 @@
     src/unix/kqueue.c
     src/unix/proctitle.c
     )
+
+  include(CheckIncludeFile)
+
+  check_include_file("ApplicationServices/ApplicationServices.h" HAVE_ApplicationServices)
+  if (HAVE_ApplicationServices)
+    list(APPEND uv_defines
+      HAVE_APPLICATIONSERVICES_APPLICATIONSERVICES_H=1
+      )
+  endif()
+
+  check_include_file("CoreServices/CoreServices.h" HAVE_CoreServices)
+  if (HAVE_CoreServices)
+    list(APPEND uv_defines
+      HAVE_CORESERVICES_CORESERVICES_H=1
+      )
+  endif()
 endif()
 
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
diff -ur cmake-3.12.1/Utilities/cmlibuv/src/unix/fsevents.c cmake-3.12.1-patched/Utilities/cmlibuv/src/unix/fsevents.c
--- cmake-3.12.1/Utilities/cmlibuv/src/unix/fsevents.c	2018-08-09 21:14:08.000000000 +0900
+++ cmake-3.12.1-patched/Utilities/cmlibuv/src/unix/fsevents.c	2018-08-13 10:01:29.000000000 +0900
@@ -21,7 +21,7 @@
 #include "uv.h"
 #include "internal.h"
 
-#if TARGET_OS_IPHONE || MAC_OS_X_VERSION_MAX_ALLOWED < 1070
+#if !HAVE_CORESERVICES_CORESERVICES_H || MAC_OS_X_VERSION_MAX_ALLOWED < 1070
 
 /* iOS (currently) doesn't provide the FSEvents-API (nor CoreServices) */
 
@@ -38,7 +38,7 @@
 void uv__fsevents_loop_delete(uv_loop_t* loop) {
 }
 
-#else /* TARGET_OS_IPHONE */
+#else /* !HAVE_CORESERVICES_CORESERVICES_H */
 
 #include <dlfcn.h>
 #include <assert.h>
@@ -916,4 +916,4 @@
   return 0;
 }
 
-#endif /* TARGET_OS_IPHONE */
+#endif /* !HAVE_CORESERVICES_CORESERVICES_H */
