diff --git a/build/standalone.gypi b/build/standalone.gypi
index 7670e5b..230518c 100644
--- a/build/standalone.gypi
+++ b/build/standalone.gypi
@@ -205,7 +205,7 @@
           [ 'visibility=="hidden" and v8_enable_backtrace==0', {
             'cflags': [ '-fvisibility=hidden' ],
           }],
-          [ 'component=="shared_library"', {
+          [ 'component=="shared_library" or component=="static_library" and (v8_target_arch=="x64" or v8_target_arch=="arm64")', {
             'cflags': [ '-fPIC', ],
           }],
         ],
