diff --git a/Lib/fontTools/feaLib/ast.py b/Lib/fontTools/feaLib/ast.py
index efcce8c680..18e5a891d3 100644
--- a/Lib/fontTools/feaLib/ast.py
+++ b/Lib/fontTools/feaLib/ast.py
@@ -719,7 +719,8 @@ def __init__(self, prefix, glyphs, suffix, lookups, location=None):
         for i, lookup in enumerate(lookups):
             if lookup:
                 try:
-                    (_ for _ in lookup)
+                    for _ in lookup:
+                        break
                 except TypeError:
                     self.lookups[i] = [lookup]
 
@@ -777,7 +778,8 @@ def __init__(self, prefix, glyphs, suffix, lookups, location=None):
         for i, lookup in enumerate(lookups):
             if lookup:
                 try:
-                    (_ for _ in lookup)
+                    for _ in lookup:
+                        break
                 except TypeError:
                     self.lookups[i] = [lookup]
 

