diff -ruN iproute2-2.6.15-060110/Makefile iproute2-2.6.15-060110.new/Makefile
--- iproute2-2.6.15-060110/Makefile	2005-07-08 22:08:47.000000000 +0000
+++ iproute2-2.6.15-060110.new/Makefile	2006-02-16 17:03:31.000000000 +0000
@@ -1,11 +1,11 @@
-DESTDIR=
+DESTDIR=$(out)
 SBINDIR=/usr/sbin
 CONFDIR=/etc/iproute2
 DOCDIR=/usr/share/doc/iproute2
 MANDIR=/usr/share/man
 
 # Path to db_185.h include
-DBM_INCLUDE:=/usr/include
+#DBM_INCLUDE:=/usr/include
 
 DEFINES= -DRESOLVE_HOSTNAMES
 
