diff --git a/src/helper/CMakeLists.txt b/src/helper/CMakeLists.txt
index 1b7ed7c..757748c 100644
--- a/src/helper/CMakeLists.txt
+++ b/src/helper/CMakeLists.txt
@@ -22,15 +22,7 @@ message("D-Bus files will be installed into ${DBUS_DATADIR_PREFIX_DIR}/dbus-1")
 
 # Find polkit
 pkg_check_modules(POLKIT REQUIRED polkit-gobject-1)
-execute_process(
-    COMMAND pkg-config --variable=policydir polkit-gobject-1
-    RESULT_VARIABLE POLKIT_POLICY_INSTALL_DIR_RESULT
-    OUTPUT_VARIABLE POLKIT_POLICY_INSTALL_DIR
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-if(NOT POLKIT_POLICY_INSTALL_DIR_RESULT EQUAL "0")
-  message(FATAL_ERROR "Failed to retrieve Polkit `policydir` variable using pkg-config")
-endif()
+option(POLKIT_POLICY_INSTALL_DIR "Polkit policy directory")
 
 list(APPEND PROCESS_MONITOR_SRC
   pmon/processmonitor.cpp
