Submodule libretroshare contains modified content
diff --git a/libretroshare/src/libretroshare.pro b/libretroshare/src/libretroshare.pro
index 923f878c..ef00f3d7 100644
--- a/libretroshare/src/libretroshare.pro
+++ b/libretroshare/src/libretroshare.pro
@@ -930,20 +930,14 @@ rs_jsonapi {
         genrestbedlib.variable_out = PRE_TARGETDEPS
         win32-g++:isEmpty(QMAKE_SH) {
             genrestbedlib.commands = \
-                cd /D $$shell_path($${RS_SRC_PATH}) && git submodule update --init supportlibs/restbed || cd . $$escape_expand(\\n\\t) \
-                cd /D $$shell_path($${RESTBED_SRC_PATH}) && git submodule update --init dependency/asio || cd . $$escape_expand(\\n\\t) \
-                cd /D $$shell_path($${RESTBED_SRC_PATH}) && git submodule update --init dependency/catch || cd . $$escape_expand(\\n\\t )\
-                cd /D $$shell_path($${RESTBED_SRC_PATH}) && git submodule update --init dependency/kashmir || cd . $$escape_expand(\\n\\t) \
+                cd /D $$shell_path($${RS_SRC_PATH}) && cd . $$escape_expand(\\n\\t) \
+                cd /D $$shell_path($${RESTBED_SRC_PATH}) && cd . $$escape_expand(\\n\\t) \
+                cd /D $$shell_path($${RESTBED_SRC_PATH}) && cd . $$escape_expand(\\n\\t )\
+                cd /D $$shell_path($${RESTBED_SRC_PATH}) && cd . $$escape_expand(\\n\\t) \
                 $(CHK_DIR_EXISTS) $$shell_path($$UDP_DISCOVERY_BUILD_PATH) $(MKDIR) $$shell_path($${UDP_DISCOVERY_BUILD_PATH}) $$escape_expand(\\n\\t)
         } else {
             genrestbedlib.commands = \
-                cd $${RS_SRC_PATH} && ( \
-                git submodule update --init supportlibs/restbed ; \
-                cd $${RESTBED_SRC_PATH} ; \
-                git submodule update --init dependency/asio ; \
-                git submodule update --init dependency/catch ; \
-                git submodule update --init dependency/kashmir ; \
-                true ) && \
+                cd $${RS_SRC_PATH} && \
                 mkdir -p $${RESTBED_BUILD_PATH} &&
         }
         genrestbedlib.commands += \
@@ -1063,14 +1057,9 @@ rs_broadcast_discovery {
         udpdiscoverycpplib.variable_out = PRE_TARGETDEPS
         win32-g++:isEmpty(QMAKE_SH) {
             udpdiscoverycpplib.commands = \
-                cd /D $$shell_path($${RS_SRC_PATH}) && git submodule update --init supportlibs/udp-discovery-cpp || cd . $$escape_expand(\\n\\t) \
                 $(CHK_DIR_EXISTS) $$shell_path($$UDP_DISCOVERY_BUILD_PATH) $(MKDIR) $$shell_path($${UDP_DISCOVERY_BUILD_PATH}) $$escape_expand(\\n\\t)
         } else {
-            udpdiscoverycpplib.commands = \
-                cd $${RS_SRC_PATH} && ( \
-                git submodule update --init supportlibs/udp-discovery-cpp || \
-                true ) && \
-                mkdir -p $${UDP_DISCOVERY_BUILD_PATH} &&
+            udpdiscoverycpplib.commands = mkdir -p $${UDP_DISCOVERY_BUILD_PATH} &&
         }
         udpdiscoverycpplib.commands += \
             cd $$shell_path($${UDP_DISCOVERY_BUILD_PATH}) && \
@@ -1106,15 +1095,13 @@ rs_sam3_libsam3 {
     win32-g++:isEmpty(QMAKE_SH) {
         LIBSAM3_MAKE_PARAMS = CC=gcc
         libsam3.commands = \
-            cd /D $$shell_path($${RS_SRC_PATH}) && git submodule update --init supportlibs/libsam3 || cd . $$escape_expand(\\n\\t) \
+            cd /D $$shell_path($${RS_SRC_PATH}) && cd . $$escape_expand(\\n\\t) \
             $(CHK_DIR_EXISTS) $$shell_path($$LIBSAM3_BUILD_PATH) $(MKDIR) $$shell_path($${LIBSAM3_BUILD_PATH}) $$escape_expand(\\n\\t) \
             $(COPY_DIR) $$shell_path($${LIBSAM3_SRC_PATH}) $$shell_path($${LIBSAM3_BUILD_PATH}) || cd . $$escape_expand(\\n\\t)
     } else {
         LIBSAM3_MAKE_PARAMS =
         libsam3.commands = \
-            cd $${RS_SRC_PATH} && ( \
-            git submodule update --init supportlibs/libsam3 || \
-            true ) && \
+            cd $${RS_SRC_PATH} && \
             mkdir -p $${LIBSAM3_BUILD_PATH} && \
             (cp -r $${LIBSAM3_SRC_PATH}/* $${LIBSAM3_BUILD_PATH} || true) &&
     }
diff --git a/retroshare-gui/src/retroshare-gui.pro b/retroshare-gui/src/retroshare-gui.pro
index d73117b84..b2fc6799a 100644
--- a/retroshare-gui/src/retroshare-gui.pro
+++ b/retroshare-gui/src/retroshare-gui.pro
@@ -66,10 +66,7 @@ rs_gui_cmark {
 		gencmarklib.CONFIG += target_predeps combine
 		gencmarklib.variable_out = PRE_TARGETDEPS
 		gencmarklib.commands = \
-		    cd $${RS_SRC_PATH} && ( \
-		    git submodule update --init supportlibs/cmark ; \
-		    cd $${CMARK_SRC_PATH} ; \
-		    true ) && \
+		    cd $${RS_SRC_PATH} && \
 		    mkdir -p $${CMARK_BUILD_PATH} && cd $${CMARK_BUILD_PATH} && \
 		    cmake \
 		        -DCMAKE_CXX_COMPILER=$$QMAKE_CXX \
