diff --git a/widget/gtk/mozwayland/mozwayland.c b/widget/gtk/mozwayland/mozwayland.c
index 7a448e6..7792581 100644
--- a/widget/gtk/mozwayland/mozwayland.c
+++ b/widget/gtk/mozwayland/mozwayland.c
@@ -200,3 +200,8 @@ MOZ_EXPORT int wl_list_empty(const struct wl_list* list) { return -1; }

 MOZ_EXPORT void wl_list_insert_list(struct wl_list* list,
                                     struct wl_list* other) {}
+
+MOZ_EXPORT struct wl_proxy *
+wl_proxy_marshal_flags(struct wl_proxy *proxy, uint32_t opcode,
+                      const struct wl_interface *interface, uint32_t version,
+                      uint32_t flags, ...) { return NULL; }
