diff -rc rpm-4.4.8-orig/lib/poptALL.c rpm-4.4.8/lib/poptALL.c
*** rpm-4.4.8-orig/lib/poptALL.c	2007-03-21 19:46:31.000000000 +0100
--- rpm-4.4.8/lib/poptALL.c	2007-03-21 19:44:45.000000000 +0100
***************
*** 483,489 ****
--- 483,491 ----
      rpmFreeMacros(NULL);
  /*@i@*/	rpmFreeMacros(rpmCLIMacroContext);
      rpmFreeRpmrc();
+ #ifdef	WITH_LUA
      (void) rpmluaFree(NULL);
+ #endif
      rpmFreeFilesystems();
  /*@i@*/	urlFreeCache();
      rpmlogClose();
