diff --git a/src/kbuildsycoca/vfolder_menu.cpp b/src/kbuildsycoca/vfolder_menu.cpp
index 2eb1275..e39a36f 100644
--- a/src/kbuildsycoca/vfolder_menu.cpp
+++ b/src/kbuildsycoca/vfolder_menu.cpp
@@ -412,7 +412,7 @@ VFolderMenu::absoluteDir(const QString &_dir, const QString &baseDir, bool keepR
     }
 
     if (!relative) {
-        QString resolved = QDir(dir).canonicalPath();
+        QString resolved = QDir::cleanPath(dir);
         if (!resolved.isEmpty()) {
             dir = resolved;
         }
