diff --git a/packages/data-schemas/src/config/meiliLogger.ts b/packages/data-schemas/src/config/meiliLogger.ts
index 0d4d39475..c38560d35 100644
--- a/packages/data-schemas/src/config/meiliLogger.ts
+++ b/packages/data-schemas/src/config/meiliLogger.ts
@@ -2,7 +2,7 @@ import path from 'path';
 import winston from 'winston';
 import 'winston-daily-rotate-file';
 
-const logDir = path.join(__dirname, '..', '..', '..', 'api', 'logs');
+const logDir = path.join('.', 'logs');
 
 const { NODE_ENV, DEBUG_LOGGING = 'false' } = process.env;
 
diff --git a/packages/data-schemas/src/config/winston.ts b/packages/data-schemas/src/config/winston.ts
index 7e5287296..dd68614bc 100644
--- a/packages/data-schemas/src/config/winston.ts
+++ b/packages/data-schemas/src/config/winston.ts
@@ -3,7 +3,7 @@ import winston from 'winston';
 import 'winston-daily-rotate-file';
 import { redactFormat, redactMessage, debugTraverse, jsonTruncateFormat } from './parsers';
 
-const logDir = path.join(__dirname, '..', '..', '..', 'api', 'logs');
+const logDir = path.join('.', 'logs');
 
 const { NODE_ENV, DEBUG_LOGGING, CONSOLE_JSON, DEBUG_CONSOLE } = process.env;
 
