From da96f833450db916cd8985669bb8f2069d35bc97 Mon Sep 17 00:00:00 2001
From: Petr Hodina <petr.hodina@luxonis.com>
Date: Fri, 21 Mar 2025 07:28:47 +0100
Subject: [PATCH 02/14] CMakeLists.txt: Add embedded dependencies


diff --git a/CMakeLists.txt b/CMakeLists.txt
index 389f5319d..5b670b87d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -677,72 +677,83 @@ endif()
 if(DEPTHAI_ENABLE_DEVICE_FW)
     # Add device FW
 
-    DepthaiDownload(
-        "${DEPTHAI_SHARED_COMMIT_HASH}" "${DEPTHAI_SHARED_COMMIT_HASH_ENFORCE}"
-        PATCH_ONLY ON
-        "${DEPTHAI_RESOURCES_OUTPUT_DIR}"            # Output folder
-        DEPTHAI_RESOURCE_LIST                       # List of output resources
-        "${DEPTHAI_DEVICE_SIDE_MATURITY}"           # Maturity
-        "${DEPTHAI_DEVICE_SIDE_COMMIT}"             # commit hash
-        "${DEPTHAI_DEVICE_SIDE_VERSION}"            # Optional version
-    )
+    # DepthaiDownload(
+    #     "${DEPTHAI_SHARED_COMMIT_HASH}" "${DEPTHAI_SHARED_COMMIT_HASH_ENFORCE}"
+    #     PATCH_ONLY ON
+    #     "${DEPTHAI_RESOURCES_OUTPUT_DIR}"            # Output folder
+    #     DEPTHAI_RESOURCE_LIST                       # List of output resources
+    #     "${DEPTHAI_DEVICE_SIDE_MATURITY}"           # Maturity
+    #     "${DEPTHAI_DEVICE_SIDE_COMMIT}"             # commit hash
+    #     "${DEPTHAI_DEVICE_SIDE_VERSION}"            # Optional version
+    # )
+    list(APPEND DEPTHAI_RESOURCE_LIST "@NIX_PATH@/depthai-device-fwp-6b1f86173749c96fe42fbc132d0b41ec09e5da6d.tar.xz")
+    list(APPEND DEPTHAI_RESOURCE_LIST "@NIX_PATH@/depthai-bootloader-fwp-0.0.26.tar.xz")
+    list(APPEND DEPTHAI_RESOURCE_LIST "@NIX_PATH@/depthai-device-kb-fwp-0.0.1+462021e2f146d868dfe59cdf9230c3b733bef115.tar.xz")
+    list(APPEND DEPTHAI_RESOURCE_LIST "@NIX_PATH@/depthai-device-rvc4-fwp-0.0.1+c21e47f41f34e3d818fb11cfb44726c5edaf2695.tar.xz")
+    list(APPEND DEPTHAI_RESOURCE_LIST "@NIX_PATH@/resources/depthai-visualizer-0.7.3.tar.gz")
+
     list(APPEND RESOURCE_COMPILED_FILES ${DEPTHAI_RESOURCE_LIST})
 endif()
 
 if(DEPTHAI_ENABLE_DEVICE_BOOTLOADER_FW)
-    # Add bootloader FW
-    DepthaiBootloaderDownload(
-        "${DEPTHAI_BOOTLOADER_SHARED_COMMIT_HASH}" "${DEPTHAI_BOOTLOADER_SHARED_COMMIT_HASH_ENFORCE}"
-        "${DEPTHAI_RESOURCES_OUTPUT_DIR}"                # Output folder
-        DEPTHAI_BOOTLOADER_RESOURCE_LIST                # List of output resources
-        "${DEPTHAI_BOOTLOADER_MATURITY}"                # Maturity
-        "${DEPTHAI_BOOTLOADER_VERSION}"                 # if maturity == snapshot -> hash else version
-    )
+    #    # Add bootloader FW
+    #    DepthaiBootloaderDownload(
+    #        "${DEPTHAI_BOOTLOADER_SHARED_COMMIT_HASH}" "${DEPTHAI_BOOTLOADER_SHARED_COMMIT_HASH_ENFORCE}"
+    #        "${DEPTHAI_RESOURCES_OUTPUT_DIR}"                # Output folder
+    #        DEPTHAI_BOOTLOADER_RESOURCE_LIST                # List of output resources
+    #        "${DEPTHAI_BOOTLOADER_MATURITY}"                # Maturity
+    #        "${DEPTHAI_BOOTLOADER_VERSION}"                 # if maturity == snapshot -> hash else version
+    #    )
+    list(APPEND DEPTHAI_BOOTLOADER_RESOURCE_LIST "")
     list(APPEND RESOURCE_COMPILED_FILES ${DEPTHAI_BOOTLOADER_RESOURCE_LIST})
 endif()
 
 if(DEPTHAI_ENABLE_DEVICE_RVC3_FW)
     # Add device-kb FW
-    DepthaiDeviceDownloader(
-        "depthai-device-kb"
-        "luxonis-keembay-snapshot-local"
-        "luxonis-keembay-release-local"
-        "${DEPTHAI_SHARED_COMMIT_HASH}" "${DEPTHAI_SHARED_COMMIT_HASH_ENFORCE}"
-        "${DEPTHAI_RESOURCES_OUTPUT_DIR}"                # Output folder
-        DEPTHAI_DEVICE_KB_RESOURCE_LIST                 # List of output resources
-        "${DEPTHAI_DEVICE_KB_MATURITY}"                # Maturity
-        "${DEPTHAI_DEVICE_RVC3_VERSION}"
-    )
+    # DepthaiDeviceDownloader(
+    #     "depthai-device-kb"
+    #     "luxonis-keembay-snapshot-local"
+    #     "luxonis-keembay-release-local"
+    #     "${DEPTHAI_SHARED_COMMIT_HASH}" "${DEPTHAI_SHARED_COMMIT_HASH_ENFORCE}"
+    #     "${DEPTHAI_RESOURCES_OUTPUT_DIR}"                # Output folder
+    #     DEPTHAI_DEVICE_KB_RESOURCE_LIST                 # List of output resources
+    #     "${DEPTHAI_DEVICE_KB_MATURITY}"                # Maturity
+    #     "${DEPTHAI_DEVICE_RVC3_VERSION}"
+    # )
+    list(APPEND DEPTHAI_DEVICE_KB_RESOURCE_LIST "")
     list(APPEND RESOURCE_COMPILED_FILES ${DEPTHAI_DEVICE_KB_RESOURCE_LIST})
 endif()
 
 if(DEPTHAI_ENABLE_DEVICE_RVC4_FW)
     # Add device-RVC4 FW
-    DepthaiDeviceDownloader(
-        "depthai-device-rvc4"
-        "luxonis-rvc4-snapshot-local"
-        "luxonis-rvc4-release-local"
-        "${DEPTHAI_SHARED_COMMIT_HASH}" "${DEPTHAI_SHARED_COMMIT_HASH_ENFORCE}"
-        "${DEPTHAI_RESOURCES_OUTPUT_DIR}"                # Output folder
-        DEPTHAI_DEVICE_RVC4_RESOURCE_LIST                 # List of output resources
-        "${DEPTHAI_DEVICE_RVC4_MATURITY}"                # Maturity
-        "${DEPTHAI_DEVICE_RVC4_VERSION}"
-    )
+    # DepthaiDeviceDownloader(
+    #     "depthai-device-rvc4"
+    #     "luxonis-rvc4-snapshot-local"
+    #     "luxonis-rvc4-release-local"
+    #     "${DEPTHAI_SHARED_COMMIT_HASH}" "${DEPTHAI_SHARED_COMMIT_HASH_ENFORCE}"
+    #     "${DEPTHAI_RESOURCES_OUTPUT_DIR}"                # Output folder
+    #     DEPTHAI_DEVICE_RVC4_RESOURCE_LIST                 # List of output resources
+    #     "${DEPTHAI_DEVICE_RVC4_MATURITY}"                # Maturity
+    #     "${DEPTHAI_DEVICE_RVC4_VERSION}"
+    # )
+    list(APPEND DEPTHAI_DEVICE_RVC4_RESOURCE_LIST "")
     list(APPEND RESOURCE_COMPILED_FILES ${DEPTHAI_DEVICE_RVC4_RESOURCE_LIST})
 endif()
 
 
 if(DEPTHAI_EMBED_FRONTEND)
-    DepthaiVisualizerDownloader(
-        "${DEPTHAI_VISUALIZER_COMMIT}"                # Visualizer hash
-        "${DEPTHAI_RESOURCES_OUTPUT_DIR}"
-        DEPTHAI_VISUALIZER_RESOURCE_LIST                 # List of output resources
-    )
+    # DepthaiVisualizerDownloader(
+    #     "${DEPTHAI_VISUALIZER_COMMIT}"                # Visualizer hash
+    #     "${DEPTHAI_RESOURCES_OUTPUT_DIR}"
+    #     DEPTHAI_VISUALIZER_RESOURCE_LIST                 # List of output resources
+    # )
+    list(APPEND DEPTHAI_VISUALIZER_RESOURCE_LIST "")
     list(APPEND RESOURCE_COMPILED_FILES ${DEPTHAI_VISUALIZER_RESOURCE_LIST}) # TODO - Handle the case non debug case
     target_compile_definitions(${TARGET_CORE_NAME} PRIVATE DEPTHAI_VISUALIZER_VERSION="${DEPTHAI_VISUALIZER_COMMIT}")
 endif()
 
 message(STATUS "LIST OF RESOURCE COMPILED FILES: ${RESOURCE_COMPILED_FILES}")
+
 if(DEPTHAI_BINARIES_RESOURCE_COMPILE)
     # Add RC and resource compile the binares
     include(CMakeRC)
-- 
2.48.1

