diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3161667..2f5cd8a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,12 +4,12 @@
 
 cmake_minimum_required(VERSION 3.2)
 
-include("cmake/HunterGate.cmake")
-HunterGate(
-    URL "https://github.com/cpp-pm/hunter/archive/v0.23.320.tar.gz"
-    SHA1 "9b4e732afd22f40482c11ad6342f7d336634226f"
-    LOCAL # Local config for dependencies
-)
+#include("cmake/HunterGate.cmake")
+#HunterGate(
+#    URL "https://github.com/cpp-pm/hunter/archive/v0.23.320.tar.gz"
+#    SHA1 "9b4e732afd22f40482c11ad6342f7d336634226f"
+#    LOCAL # Local config for dependencies
+#)
 
 ### Constants
 set(TARGET_NAME "XLink")
