From d30c2b9b576b06976a9310ea82065167eee90a29 Mon Sep 17 00:00:00 2001
From: Petr Hodina <petr.hodina@luxonis.com>
Date: Fri, 21 Mar 2025 07:39:30 +0100
Subject: [PATCH 12/14] cmake: Handle catch2 dependencies for tests


diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 8de4a92f0..3b177aa78 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -51,9 +51,9 @@ function(dai_add_test test_name test_src)
 
     # Link to core and Catch2 testing framework
     if(NOT DEPTHAI_MERGED_TARGET)
-        target_link_libraries(${test_name} PRIVATE depthai::opencv ${OpenCV_LIBS} Catch2::Catch2WithMain Threads::Threads)
+	    target_link_libraries(${test_name} PRIVATE depthai::opencv ${OpenCV_LIBS} Catch2 Catch2Main Threads::Threads)
     else()
-        target_link_libraries(${test_name} PRIVATE depthai::core ${OpenCV_LIBS} Catch2::Catch2WithMain Threads::Threads)
+	    target_link_libraries(${test_name} PRIVATE depthai::core ${OpenCV_LIBS} Catch2 Catch2Main Threads::Threads)
     endif()
 
     # Add sanitizers for tests as well
-- 
2.48.1

