From 8d76592333577f1973dfb60dda29fa31422aa403 Mon Sep 17 00:00:00 2001
From: Petr Hodina <petr.hodina@luxonis.com>
Date: Fri, 21 Mar 2025 07:43:07 +0100
Subject: [PATCH 14/14] cmake: Install examples after build- WIP


diff --git a/examples/cpp/CMakeLists.txt b/examples/cpp/CMakeLists.txt
index 01c39abf1..dd0b8e10e 100644
--- a/examples/cpp/CMakeLists.txt
+++ b/examples/cpp/CMakeLists.txt
@@ -11,6 +11,9 @@ target_include_directories(utility PUBLIC "utility" "$<BUILD_INTERFACE:${FP16_IN
 add_default_flags(utility LEAN)
 target_link_libraries(utility ${OpenCV_LIBS})
 
+set(EXAMPLES_BIN "")
+set(EXAMPLES_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/share/examples")
+
 set(test_env
     # Misc
     "UBSAN_OPTIONS=halt_on_error=1"
@@ -22,7 +25,8 @@ set(test_env
 # Helper for adding new examples
 function(dai_add_example example_name example_src enable_test use_pcl)
 
-    # Add example
+    # Add example to the list of executables
+    list(APPEND EXAMPLES_BIN ${example_name})
     add_executable(${example_name} ${example_src})
     add_default_flags(${example_name} LEAN)
     set(DEPTHAI_TARGET depthai::core)
@@ -335,3 +339,11 @@ endif()
 # if(DEPTHAI_HAVE_PCL_SUPPORT)
 #     dai_add_example(visualize_pointcloud PointCloud/visualize_pointcloud.cpp ON ON)
 # endif()
+
+# Install the executables
+file(MAKE_DIRECTORY ${EXAMPLES_INSTALL_DIR})  # Ensure the directory exists
+
+foreach(file ${EXAMPLES_BIN})
+    # Install each executable to the specified directory
+    install(TARGETS ${file} DESTINATION ${EXAMPLES_INSTALL_DIR})
+endforeach()
-- 
2.48.1

