diff --git a/tests/test_pipes.py b/tests/test_pipes.py
index d883abf..2e74d7a 100644
--- a/tests/test_pipes.py
+++ b/tests/test_pipes.py
@@ -2,6 +2,7 @@ import asyncio
 import io
 import os
 import socket
+import unittest
 
 from uvloop import _testbase as tb
 
@@ -96,6 +97,7 @@ class _BasePipeTest:
         # extra info is available
         self.assertIsNotNone(proto.transport.get_extra_info('pipe'))
 
+    @unittest.skip("darwin sandbox")
     def test_read_pty_output(self):
         proto = MyReadPipeProto(loop=self.loop)
 
@@ -198,6 +200,7 @@ class _BasePipeTest:
         self.loop.run_until_complete(proto.done)
         self.assertEqual('CLOSED', proto.state)
 
+    @unittest.skip("darwin sandbox")
     def test_write_pty(self):
         master, slave = os.openpty()
         os.set_blocking(master, False)
