diff -rc klibc-1.4-orig/scripts/Kbuild.install klibc-1.4/scripts/Kbuild.install
*** klibc-1.4-orig/scripts/Kbuild.install	2006-06-19 03:17:15.000000000 +0200
--- klibc-1.4/scripts/Kbuild.install	2006-11-25 22:41:09.000000000 +0100
***************
*** 96,107 ****
  	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)lib
  	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)bin
  	$(Q)set -e ; for d in linux scsi asm-$(KLIBCARCH) asm-generic $(ASMKLIBCARCH); do \
! 	  mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KLIBCCROSS)include/$$d ; \
! 	  for r in $(KLIBCKERNELSRC)/include $(KLIBCKERNELOBJ)/include \
! 	           $(KLIBCKERNELOBJ)/include2 ; do \
  	    [ ! -d $$r/$$d ] || \
! 	      cp -rfL $$r/$$d/. \
! 	          $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/$$d/. ; \
  	  done ; \
  	done
  	$(Q)cd $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include && ln -sf asm-$(KLIBCARCH) asm
--- 96,105 ----
  	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)lib
  	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)bin
  	$(Q)set -e ; for d in linux scsi asm-$(KLIBCARCH) asm-generic $(ASMKLIBCARCH); do \
! 	  for r in $(KLIBCKERNELSRC)/include ; do \
  	    [ ! -d $$r/$$d ] || \
! 	      cp -rd $$r/$$d \
! 	          $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/$$d ; \
  	  done ; \
  	done
  	$(Q)cd $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include && ln -sf asm-$(KLIBCARCH) asm
