diff -rc kvm-74-orig/kernel/Makefile kvm-74/kernel/Makefile
*** kvm-74-orig/kernel/Makefile	2008-08-27 10:53:37.000000000 +0200
--- kvm-74/kernel/Makefile	2008-08-29 14:12:00.000000000 +0200
***************
*** 59,64 ****
--- 59,65 ----
  	-rsync -R \
  	     "$(LINUX)"/arch/$(ARCH_DIR)/include/asm/./kvm*.h \
               $T/include/asm-$(ARCH_DIR)/
+ 	chmod -R u+w $T
  
  	set -e && for i in $(find $T -name '*.h'); do \
  		$(call unifdef,$$i); done
Only in kvm-74/kernel: Makefile~
Only in kvm-74/kernel: Makefile.rej
