diff -rc kvm-78-orig/kernel/Makefile kvm-78/kernel/Makefile
*** kvm-78-orig/kernel/Makefile	2008-10-30 11:58:12.000000000 +0100
--- kvm-78/kernel/Makefile	2008-11-07 15:59:34.000000000 +0100
***************
*** 66,71 ****
--- 66,72 ----
  	     "$(LINUX)"/./include/linux/kvm*.h \
  	     $(if $(wildcard $(headers-old)), $(headers-old)) \
               $T/
+ 	chmod -R u+w $T
  	$(if $(wildcard $(headers-new)), \
  	rsync -R \
  	     $(headers-new) \
Only in kvm-78/kernel: Makefile~
