diff --git a/CMakeLists.txt b/CMakeLists.txt
index efeacfc..77bd6bd 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -121,7 +121,11 @@ endif()
 # Looking for external libraries
 #
 
-setup_conan()
+find_package(muparserx REQUIRED)
+find_package(nlohmann_json REQUIRED)
+find_package(spdlog REQUIRED)
+# for tests only
+find_package(catch2)
 
 # If we do not set them with a space CMake fails afterwards if nothing is set for this vars!
 set(AER_LINKER_FLAGS " ")
@@ -269,16 +273,16 @@ endif()
 set(AER_LIBRARIES
 	${AER_LIBRARIES}
 	${BLAS_LIBRARIES}
-	CONAN_PKG::nlohmann_json
+	nlohmann_json
 	Threads::Threads
-	CONAN_PKG::spdlog
+	spdlog
 	${DL_LIB}
 	${THRUST_DEPENDANT_LIBS})
 
 set(AER_COMPILER_DEFINITIONS ${AER_COMPILER_DEFINITIONS} ${CONAN_DEFINES})
 # Cython build is only enabled if building through scikit-build.
 if(SKBUILD) # Terra Addon build
-	set(AER_LIBRARIES ${AER_LIBRARIES} CONAN_PKG::muparserx)
+	set(AER_LIBRARIES ${AER_LIBRARIES} muparserx)
 	add_subdirectory(qiskit/providers/aer/pulse/qutip_extra_lite/cy)
 	add_subdirectory(qiskit/providers/aer/backends/wrappers)
 	add_subdirectory(src/open_pulse)
diff --git a/setup.py b/setup.py
index fd71e9f..1561cc4 100644
--- a/setup.py
+++ b/setup.py
@@ -11,12 +11,6 @@ import inspect
 
 PACKAGE_NAME = os.getenv('QISKIT_AER_PACKAGE_NAME', 'qiskit-aer')
 
-try:
-    from conans import client
-except ImportError:
-    subprocess.call([sys.executable, '-m', 'pip', 'install', 'conan'])
-    from conans import client
-
 try:
     from skbuild import setup
 except ImportError:
@@ -46,8 +40,6 @@ common_requirements = [
 
 setup_requirements = common_requirements + [
     'scikit-build',
-    'cmake!=3.17,!=3.17.0',
-    'conan>=1.22.2'
 ]
 
 requirements = common_requirements + ['qiskit-terra>=0.12.0']
