diff --git a/salt/utils/rsax931.py b/salt/utils/rsax931.py
index b7ecfcd..301c7b5 100644
--- a/salt/utils/rsax931.py
+++ b/salt/utils/rsax931.py
@@ -44,8 +44,11 @@ def _load_libcrypto():
             # two checks below
             lib = glob.glob('/opt/local/lib/libcrypto.so*') + glob.glob('/opt/tools/lib/libcrypto.so*')
             lib = lib[0] if len(lib) > 0 else None
-        if lib:
-            return cdll.LoadLibrary(lib)
+        try:
+            return cdll.LoadLibrary("@libcrypto@")
+        except OSError:
+            if lib:
+                return cdll.LoadLibrary(lib)
         raise OSError('Cannot locate OpenSSL libcrypto')
