diff --git a/mkspecs/features/platform.prf b/mkspecs/features/platform.prf
--- a/mkspecs/features/platform.prf
+++ b/mkspecs/features/platform.prf
@@ -38,10 +38,6 @@ defineTest(isPlatformSupported) {
          return(false)
     }
   } else:osx {
-    !isMinXcodeVersion(7, 3) {
-      skipBuild("Using Xcode version $$QMAKE_XCODE_VERSION, but at least version 7.3 is required to build Qt WebEngine.")
-      return(false)
-    }
     !clang|intel_icc {
         skipBuild("Qt WebEngine on macOS requires Clang.")
         return(false)
@@ -52,10 +48,6 @@ defineTest(isPlatformSupported) {
       skipBuild("Building Qt WebEngine requires macOS version 10.11 or newer.")
       return(false)
     }
-    !isMinOSXSDKVersion(10, 10): {
-      skipBuild("Building Qt WebEngine requires a macOS SDK version of 10.11 or newer. Current version is $${WEBENGINE_OSX_SDK_PRODUCT_VERSION}.")
-      return(false)
-    }
   } else {
     skipBuild("Unknown platform. Qt WebEngine only supports Linux, Windows, and macOS.")
     return(false)
diff --git a/src/core/config/mac_osx.pri b/src/core/config/mac_osx.pri
--- a/src/core/config/mac_osx.pri
+++ b/src/core/config/mac_osx.pri
@@ -5,8 +5,6 @@ load(functions)
 # otherwise query for it.
 QMAKE_MAC_SDK_VERSION = $$eval(QMAKE_MAC_SDK.$${QMAKE_MAC_SDK}.SDKVersion)
 isEmpty(QMAKE_MAC_SDK_VERSION) {
-     QMAKE_MAC_SDK_VERSION = $$system("/usr/bin/xcodebuild -sdk $${QMAKE_MAC_SDK} -version SDKVersion 2>/dev/null")
-     isEmpty(QMAKE_MAC_SDK_VERSION): error("Could not resolve SDK version for \'$${QMAKE_MAC_SDK}\'")
 }
 
 QMAKE_CLANG_DIR = "/usr"
