Correctly handle the `--no-root-login' option.

--- lsh-2.0.4/src/lshd.c	2006-05-01 13:47:44.000000000 +0200
+++ lsh-2.0.4/src/lshd.c	2009-09-08 12:20:36.000000000 +0200
@@ -758,6 +758,10 @@ main_argp_parser(int key, char *arg, str
       self->allow_root = 1;
       break;
 
+    case OPT_NO_ROOT_LOGIN:
+      self->allow_root = 0;
+      break;
+
     case OPT_KERBEROS_PASSWD:
       self->pw_helper = PATH_KERBEROS_HELPER;
       break;

