diff --git a/src/main/index.ts b/src/main/index.ts
index c4a19fa..81a1ee8 100644
--- a/src/main/index.ts
+++ b/src/main/index.ts
@@ -657,7 +657,7 @@ async function launchSeanimeServer(isRestart: boolean): Promise<void> {
             binaryPath = path.join(app.getAppPath(), "binaries", binaryName)
         } else {
             // In production, use the resources path
-            binaryPath = path.join(process.resourcesPath, "binaries", binaryName)
+            binaryPath = "SEANIME_BIN"
         }
 
         logStartupEvent("Using binary", `${binaryPath} (${process.arch})`)
@@ -671,14 +671,14 @@ async function launchSeanimeServer(isRestart: boolean): Promise<void> {
         }
 
         // Make binary executable (for macOS/Linux)
-        if (process.platform !== "win32") {
-            try {
-                fs.chmodSync(binaryPath, "755")
-            }
-            catch (error) {
-                logger.server.error("Failed to make binary executable", formatError(error))
-            }
-        }
+        // if (process.platform !== "win32") {
+        //     try {
+        //         fs.chmodSync(binaryPath, "755")
+        //     }
+        //     catch (error) {
+        //         logger.server.error("Failed to make binary executable", formatError(error))
+        //     }
+        // }
 
         // Arguments
         const args: string[] = []
