diff -Naur valgrind-3.10.1/configure valgrind-3.10.1-patch/configure
--- valgrind-3.10.1/configure	2014-11-25 11:42:25.000000000 -0800
+++ valgrind-3.10.1-patch/configure	2015-04-21 09:45:37.656113841 -0700
@@ -5553,7 +5553,7 @@
         kernel=`uname -r`
 
         case "${kernel}" in
-             2.6.*|3.*)
+             2.6.*|3.*|4.*)
         	    { $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.6.x/3.x family (${kernel})" >&5
 $as_echo "2.6.x/3.x family (${kernel})" >&6; }
 
