diff -Nur pygtksourceview-2.10.1-orig/configure pygtksourceview-2.10.1/configure
--- pygtksourceview-2.10.1-orig/configure	2010-04-18 15:29:55.000000000 +0200
+++ pygtksourceview-2.10.1/configure	2015-01-30 20:36:31.784541887 +0100
@@ -12950,7 +12950,7 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for pygtk codegen" >&5
 $as_echo_n "checking for pygtk codegen... " >&6; }
-CODEGENDIR=`$PKG_CONFIG --variable=codegendir pygtk-2.0`
+CODEGENDIR=`$PKG_CONFIG --variable=codegendir pygobject-2.0`
 echo $CODEGENDIR
 if test -f $CODEGENDIR/codegen.py; then
 	CODEGEN=$CODEGENDIR/codegen.py
diff -Nur pygtksourceview-2.10.1-orig/configure.ac pygtksourceview-2.10.1/configure.ac
--- pygtksourceview-2.10.1-orig/configure.ac	2010-04-18 15:28:39.000000000 +0200
+++ pygtksourceview-2.10.1/configure.ac	2015-01-30 20:36:42.781648830 +0100
@@ -116,7 +116,7 @@
 
 dnl codegen
 AC_MSG_CHECKING(for pygtk codegen)
-CODEGENDIR=`$PKG_CONFIG --variable=codegendir pygtk-2.0`
+CODEGENDIR=`$PKG_CONFIG --variable=codegendir pygobject-2.0`
 echo $CODEGENDIR
 if test -f $CODEGENDIR/codegen.py; then
 	CODEGEN=$CODEGENDIR/codegen.py

