diff --git a/Zend/zend_signal.c b/Zend/zend_signal.c
index 24d454d..6307620 100644
--- a/Zend/zend_signal.c
+++ b/Zend/zend_signal.c
@@ -399,7 +399,7 @@ void zend_signal_init(void) /* {{{ */
 
 /* {{{ zend_signal_startup
  * alloc zend signal globals */
-void zend_signal_startup(void)
+ZEND_API void zend_signal_startup(void)
 {
 
 #ifdef ZTS
diff --git a/Zend/zend_signal.h b/Zend/zend_signal.h
index e8ee7d6..462d06f 100644
--- a/Zend/zend_signal.h
+++ b/Zend/zend_signal.h
@@ -89,7 +89,7 @@ ZEND_API void zend_signal_handler_unblock(void);
 void zend_signal_activate(void);
 void zend_signal_deactivate(void);
 BEGIN_EXTERN_C()
-void zend_signal_startup(void);
+ZEND_API void zend_signal_startup(void);
 END_EXTERN_C()
 void zend_signal_init(void);
