--- php-5.0.4/configure	2005-04-03 11:42:50.000000000 +0200
+++ php-5.0.4-1/configure	2005-07-10 11:06:22.292271176 +0200
@@ -3381,7 +3381,7 @@
 
   case $host_alias in
   *aix*)
-    APXS_LIBEXECDIR=`$APXS -q LIBEXECDIR`
+    APXS_LIBEXECDIR="$prefix/modules"
     EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-brtl -Wl,-bI:$APXS_LIBEXECDIR/httpd.exp"
     PHP_AIX_LDFLAGS="-Wl,-brtl"
     build_type=shared
@@ -3576,7 +3576,7 @@
   if test "$?" != "0"; then
     APACHE_INSTALL="$APXS -i -a -n php5 $SAPI_SHARED" # Old apxs does not have -S option
   else 
-    APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
+    APXS_LIBEXECDIR="$prefix/modules"
     if test -z `$APXS -q SYSCONFDIR`; then
       APACHE_INSTALL="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
                        $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
@@ -4680,7 +4680,7 @@
     { echo "configure: error: Please note that Apache version >= 2.0.40 is required." 1>&2; exit 1; }
   fi
 
-  APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
+  APXS_LIBEXECDIR="$prefix/modules"
   if test -z `$APXS -q SYSCONFDIR`; then
     INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
                  $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
@@ -5510,7 +5510,7 @@
     { echo "configure: error: Please note that Apache version >= 2.0.44 is required." 1>&2; exit 1; }
   fi
 
-  APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
+  APXS_LIBEXECDIR="$prefix/modules"
   if test -z `$APXS -q SYSCONFDIR`; then
     INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
                  $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
@@ -6327,7 +6327,7 @@
 
   case $host_alias in
   *aix*)
-    APXS_LIBEXECDIR=`$APXS -q LIBEXECDIR`
+    APXS_LIBEXECDIR="$prefix/modules"
     EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-brtl -Wl,-bI:$APXS_LIBEXECDIR/httpd.exp"
     PHP_AIX_LDFLAGS="-Wl,-brtl"
     build_type=shared
@@ -6522,7 +6522,7 @@
   if test "$?" != "0"; then
     APACHE_HOOKS_INSTALL="$APXS -i -a -n php5 $SAPI_SHARED" # Old apxs does not have -S option
   else 
-    APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
+    APXS_LIBEXECDIR="$prefix/modules"
     if test -z `$APXS -q SYSCONFDIR`; then
       APACHE_HOOKS_INSTALL="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
                        $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
diff -rc php-5.3.18/configure php-5.3.18-new/configure
*** php-5.3.18/configure	2012-10-17 18:29:10.000000000 +0200
--- php-5.3.18-new/configure	2012-11-13 05:36:24.730136551 +0100
***************
*** 45113,45121 ****
  
  
  if test "$PHP_GETTEXT" != "no"; then
!   for i in $PHP_GETTEXT /usr/local /usr; do
!     test -r $i/include/libintl.h && GETTEXT_DIR=$i && break
!   done
  
    if test -z "$GETTEXT_DIR"; then
      { echo "configure: error: Cannot locate header file libintl.h" 1>&2; exit 1; }
--- 45113,45119 ----
  
  
  if test "$PHP_GETTEXT" != "no"; then
!   GETTEXT_DIR=$PHP_GETTEXT
  
    if test -z "$GETTEXT_DIR"; then
      { echo "configure: error: Cannot locate header file libintl.h" 1>&2; exit 1; }
