diff --git a/retroarch.cfg b/retroarch.cfg
index cdcb199c9f..08b9b1cf10 100644
--- a/retroarch.cfg
+++ b/retroarch.cfg
@@ -681,7 +681,7 @@
 # menu_show_online_updater = true
 
 # If disabled, will hide the ability to update cores (and core info files) inside the menu.
-# menu_show_core_updater = true
+menu_show_core_updater = false
 
 # If disabled, the libretro core will keep running in the background when we
 # are in the menu.
@@ -823,10 +823,10 @@
 # rgui_browser_directory =
 
 # Core directory for libretro core implementations.
-# libretro_directory =
+libretro_directory = @libretro_directory@
 
 # Core info directory for libretro core information.
-# libretro_info_path =
+libretro_info_path = @libretro_info_path@
 
 # Path to content database directory.
 # content_database_path =
