--- gnustep-gui-0.24.0/GNUmakefile.postamble	2010-05-17 22:38:59.000000000 +0600
+++ gnustep-gui-0.24.0.patched/GNUmakefile.postamble	2014-12-01 13:44:05.000000000 +0600
@@ -40,20 +40,20 @@
 # The following rule is important mainly for packaging, because in that case 
 # you install into a fake system tree, and the directory is not there.
 #
-$(DESTDIR)$(GNUSTEP_MAKEFILES)/Additional:
-	$(MKDIRS) $(DESTDIR)$(GNUSTEP_MAKEFILES)/Additional
+$(DESTDIR_GNUSTEP_MAKEFILES)/Additional:
+	$(MKDIRS) $(DESTDIR_GNUSTEP_MAKEFILES)/Additional
 
 # Things to do before installing
-before-install:: $(DESTDIR)$(GNUSTEP_MAKEFILES)/Additional
+before-install:: $(DESTDIR_GNUSTEP_MAKEFILES)/Additional
 	$(INSTALL_DATA) gui.make \
-	   $(DESTDIR)$(GNUSTEP_MAKEFILES)/Additional/gui.make
+	   $(DESTDIR_GNUSTEP_MAKEFILES)/Additional/gui.make
 
 # Things to do after installing
 # after-install::
 
 # Things to do before uninstalling
  before-uninstall::
-	rm -f $(DESTDIR)$(GNUSTEP_MAKEFILES)/Additional/gui.make
+	rm -f $(DESTDIR_GNUSTEP_MAKEFILES)/Additional/gui.make
 
 # Things to do after uninstalling
 # after-uninstall::
