diff -ru gcc-4.3.1-orig/libgfortran/configure gcc-4.3.1/libgfortran/configure
--- gcc-4.3.1-orig/libgfortran/configure	2008-06-06 16:49:11.000000000 +0200
+++ gcc-4.3.1/libgfortran/configure	2008-06-27 08:25:08.000000000 +0200
@@ -35405,6 +35405,11 @@
 # A language specific compiler.
 CC=$lt_compiler
 
+# Ugly hack to get libmudflap (and possibly other libraries) to build.
+# Libtool filters out \`-B' flags when linking (why?), so the \`-B' flag
+# to Glibc gets lost.  Here we forcibly add it to any invocation.
+CC="\$CC $NIX_EXTRA_CFLAGS $NIX_EXTRA_LDFLAGS"
+
 # Is the compiler the GNU compiler?
 with_gcc=$GCC
 
