diff --git a/c++/src/ColumnWriter.cc b/c++/src/ColumnWriter.cc
index 9cdbae0709..d049e91bb1 100644
--- a/c++/src/ColumnWriter.cc
+++ b/c++/src/ColumnWriter.cc
@@ -212,7 +212,9 @@
       }
     }
     // write row index to output stream
-    rowIndex->SerializeToZeroCopyStream(indexStream.get());
+    if (!rowIndex->SerializeToZeroCopyStream(indexStream.get())) {
+      throw std::logic_error("Failed to write index stream.");
+    }
 
     // construct row index stream
     proto::Stream stream;
