--- mlterm-3.7.2/xwindow/x_shortcut.c
+++ mlterm-3.7.2/xwindow/x_shortcut.c
@@ -292,6 +292,11 @@
 	/* ingoring except these masks */
 	state &= (ModMask|ControlMask|ShiftMask|CommandMask|button_mask) ;
 
+	if( state & button_mask)
+	{
+		state &= ~Mod2Mask ;	/* XXX NumLock */
+	}
+
 	if( shortcut->map[func].ksym == ksym &&
 	    shortcut->map[func].state ==
 	      ( state |
@@ -318,6 +323,11 @@
 	/* ingoring except these masks */
 	state &= (ModMask|ControlMask|ShiftMask|CommandMask|button_mask) ;
 
+	if( state & button_mask)
+	{
+		state &= ~Mod2Mask ;	/* XXX NumLock */
+	}
+
 	for( count = 0 ; count < shortcut->str_map_size ; count ++)
 	{
                 if( shortcut->str_map[count].ksym == ksym &&
