diff -rc wis-go7007-linux-0.9.8-orig/apps/gorecord.c wis-go7007-linux-0.9.8/apps/gorecord.c
*** wis-go7007-linux-0.9.8-orig/apps/gorecord.c	2006-04-02 00:35:17.000000000 +0200
--- wis-go7007-linux-0.9.8/apps/gorecord.c	2008-06-20 19:47:48.000000000 +0200
***************
*** 196,202 ****
  		sprintf(sympath, "/sys/class/sound/pcmC%dD0c/device", i);
  		if (realpath(sympath, canonpath) == NULL)
  			continue;
! 		if (!strcmp(gopath, canonpath))
  			break;
  	}
  	if (i == 20) {
--- 196,202 ----
  		sprintf(sympath, "/sys/class/sound/pcmC%dD0c/device", i);
  		if (realpath(sympath, canonpath) == NULL)
  			continue;
! 		if (!strncmp(gopath, canonpath, strlen(gopath)))
  			break;
  	}
  	if (i == 20) {
