diff -ru wis-go7007-linux-0.9.8-orig/udev/wis-ezusb.rules.in wis-go7007-linux-0.9.8/udev/wis-ezusb.rules.in
--- wis-go7007-linux-0.9.8-orig/udev/wis-ezusb.rules.in	2007-04-26 10:32:05.967814203 +0200
+++ wis-go7007-linux-0.9.8/udev/wis-ezusb.rules.in	2007-04-26 10:38:00.372601940 +0200
@@ -1,8 +1,10 @@
-ACTION=="add", BUS=="usb", ENV{PRODUCT}=="93b/a002/*", \
-  RUN+="@FXLOAD@ -t fx2 -I @FIRMWARE_DIR@/ezusb/hpi_PX-M402U.hex"
-ACTION=="add", BUS=="usb", ENV{PRODUCT}=="93b/a004/*", \
-  RUN+="@FXLOAD@ -t fx2 -I @FIRMWARE_DIR@/ezusb/hpi_PX-TV402U.hex"
-ACTION=="add", BUS=="usb", ENV{PRODUCT}=="eb1/6666/*", \
-  RUN+="@FXLOAD@ -t fx2 -I @FIRMWARE_DIR@/ezusb/hpi_LR192.hex"
-ACTION=="add", BUS=="usb", ENV{PRODUCT}=="eb1/6668/*", \
-  RUN+="@FXLOAD@ -t fx2 -I @FIRMWARE_DIR@/ezusb/hpi_StarTrek.hex"
+ACTION=="add", SUBSYSTEM=="usb_device", ATTRS{idVendor}=="093b", ATTRS{idProduct}=="a002", \
+  RUN+="@FXLOAD@ -t fx2 -I @FIRMWARE_DIR@/ezusb/hpi_PX-M402U.hex -D %N"
+ACTION=="add", SUBSYSTEM=="usb_device", ATTRS{idVendor}=="093b", ATTRS{idProduct}=="a004", \
+  RUN+="@FXLOAD@ -t fx2 -I @FIRMWARE_DIR@/ezusb/hpi_PX-TV402U.hex -D %N"
+ACTION=="add", SUBSYSTEM=="usb_device", ATTRS{idVendor}=="093b", ATTRS{idProduct}=="a104", \
+  RUN+="@FXLOAD@ -t fx2 -I @FIRMWARE_DIR@/ezusb/hpi_PX-TV402U.hex -D %N"
+ACTION=="add", SUBSYSTEM=="usb_device", ATTRS{idVendor}=="0eb1", ATTRS{idProduct}=="6666", \
+  RUN+="@FXLOAD@ -t fx2 -I @FIRMWARE_DIR@/ezusb/hpi_LR192.hex -D %N"
+ACTION=="add", SUBSYSTEM=="usb_device", ATTRS{idVendor}=="0eb1", ATTRS{idProduct}=="6668", \
+  RUN+="@FXLOAD@ -t fx2 -I @FIRMWARE_DIR@/ezusb/hpi_StarTrek.hex -D %N"
