diff --git a/build.gradle b/build.gradle
index c206e2f..5f12648 100644
--- a/build.gradle
+++ b/build.gradle
@@ -5,7 +5,6 @@ plugins {
     id 'org.springdoc.openapi-gradle-plugin' version '1.8.0'
     id "io.swagger.swaggerhub" version "1.3.2"
     id 'edu.sc.seis.launch4j' version '3.0.5'
-    id 'com.diffplug.spotless' version '6.25.0'
     id 'com.github.jk1.dependency-license-report' version '2.5'
 }
 
@@ -69,20 +68,6 @@ launch4j {
   messagesInstanceAlreadyExists="Stirling-PDF is already running."
 }
 
-spotless {
-    java {
-        target project.fileTree('src/main/java')
-
-        googleJavaFormat('1.19.1').aosp().reorderImports(false)
-
-        importOrder('java', 'javax', 'org', 'com', 'net', 'io')
-        toggleOffOn()
-        trimTrailingWhitespace()
-        indentWithSpaces()
-        endWithNewline()
-    }
-}
-
 dependencies {
     //security updates
     implementation 'ch.qos.logback:logback-classic:1.5.3'
@@ -163,9 +148,6 @@ dependencies {
     annotationProcessor 'org.projectlombok:lombok:1.18.28'
 }
 
-tasks.withType(JavaCompile) {
-    dependsOn 'spotlessApply'
-}
 compileJava {
     options.compilerArgs << '-parameters'
 }
