diff --git a/build.gradle b/build.gradle
index 3a967d1..f43fec0 100644
--- a/build.gradle
+++ b/build.gradle
@@ -278,6 +278,13 @@ subprojects {
     tasks.withType(Test).configureEach {
         useJUnitPlatform()
         finalizedBy(jacocoReport)
+
+        filter {
+            excludeTestsMatching "*CertSignControllerTest"
+            excludeTestsMatching "*UntrustedSignerTests"
+            excludeTestsMatching "*SignWithKeystore"
+            excludeTestsMatching "*TrustedAnchorTests"
+        }
     }
 
     jacocoReport.configure {
