diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5ed8020c..44ca43e7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,12 +2,20 @@ cmake_minimum_required(VERSION 3.16)
 project(deltachat LANGUAGES C)
 include(GNUInstallDirs)
 
+if(APPLE)
+	set(DYNAMIC_EXT "dylib")
+elseif(UNIX)
+	set(DYNAMIC_EXT "so")
+else()
+	set(DYNAMIC_EXT "dll")
+endif()
+
 find_program(CARGO cargo)
 
 add_custom_command(
 	OUTPUT
 	"target/release/libdeltachat.a"
-	"target/release/libdeltachat.so"
+	"target/release/libdeltachat.${DYNAMIC_EXT}"
 	"target/release/pkgconfig/deltachat.pc"
         COMMAND
         PREFIX=${CMAKE_INSTALL_PREFIX}
@@ -32,11 +40,11 @@ add_custom_target(
 	ALL
 	DEPENDS
 	"target/release/libdeltachat.a"
-	"target/release/libdeltachat.so"
+	"target/release/libdeltachat.${DYNAMIC_EXT}"
 	"target/release/pkgconfig/deltachat.pc"
 )
 
 install(FILES "deltachat-ffi/deltachat.h" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 install(FILES "target/release/libdeltachat.a" DESTINATION ${CMAKE_INSTALL_LIBDIR})
-install(FILES "target/release/libdeltachat.so" DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(FILES "target/release/libdeltachat.${DYNAMIC_EXT}" DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES "target/release/pkgconfig/deltachat.pc" DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
diff --git a/python/install_python_bindings.py b/python/install_python_bindings.py
index c8ed43e2..714c7e30 100755
--- a/python/install_python_bindings.py
+++ b/python/install_python_bindings.py
@@ -24,7 +24,7 @@
 
     print("running:", " ".join(cmd))
     subprocess.check_call(cmd)
-    subprocess.check_call("rm -rf build/ src/deltachat/*.so" , shell=True)
+    subprocess.check_call("rm -rf build/ src/deltachat/*.so src/deltachat/*.dylib" , shell=True)
 
     if len(sys.argv) <= 1 or sys.argv[1] != "onlybuild":
         subprocess.check_call([
