	Fix compilation with _LARGEFILE64_SOURCE

diff -ur zlib-1.2.5.orig/zlib.h zlib-1.2.5/zlib.h
--- zlib-1.2.5.orig/zlib.h	2010-04-20 06:12:48.000000000 +0200
+++ zlib-1.2.5/zlib.h	2010-09-27 21:20:27.398694997 +0200
@@ -1578,7 +1578,7 @@
 #  define gzoffset gzoffset64
 #  define adler32_combine adler32_combine64
 #  define crc32_combine crc32_combine64
-#  ifdef _LARGEFILE64_SOURCE
+#  ifndef _LARGEFILE64_SOURCE
      ZEXTERN gzFile ZEXPORT gzopen64 OF((const char *, const char *));
      ZEXTERN z_off_t ZEXPORT gzseek64 OF((gzFile, z_off_t, int));
      ZEXTERN z_off_t ZEXPORT gztell64 OF((gzFile));
