diff --git a/libpiper/tests/CMakeLists.txt b/libpiper/tests/CMakeLists.txt
index 2f5a6b4..276ca57 100644
--- a/libpiper/tests/CMakeLists.txt
+++ b/libpiper/tests/CMakeLists.txt
@@ -1,23 +1,21 @@
 cmake_minimum_required(VERSION 3.26)
 
 message(STATUS "Building piper tests")
-include(FetchContent)
-FetchContent_Declare(
-    googletest
-    GIT_REPOSITORY https://github.com/google/googletest.git
-    GIT_TAG v1.17.0
-)
-
-# For Windows: Prevent overriding the parent project's compiler/linker settings
-set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-FetchContent_MakeAvailable(googletest)
 
 enable_testing()
 
-include(DownloadModel.cmake)
+find_package(GTest CONFIG QUIET)
 
-download_piper_model(VOICE "en/en_US/lessac/medium/en_US-lessac-medium"
-                     OUTPUT_DIR EN_MODEL_DIR)
+if(NOT GTest_FOUND)
+    message(FATAL_ERROR "No system GTest")
+endif()
+
+if(PIPER_TEST_MODEL_DIR)
+    message(STATUS "Using pre-supplied test model at ${PIPER_TEST_MODEL_DIR}")
+    set(EN_MODEL_DIR ${PIPER_TEST_MODEL_DIR})
+else()
+    message(FATAL_ERROR "No pre-supplied test model")
+endif()
 
 add_executable(piper_test
    test_piper.cpp
@@ -44,19 +42,6 @@ target_link_directories(piper_test PUBLIC
     ${ONNXRUNTIME_DIR}/lib
 )
 
-if(WIN32)
-    add_custom_command(TARGET piper_test POST_BUILD
-        COMMAND ${CMAKE_COMMAND} -E copy_if_different
-        "$<TARGET_FILE:piper>"
-        "$<TARGET_FILE_DIR:piper_test>"
-    )
-    add_custom_command(TARGET piper_test POST_BUILD
-        COMMAND ${CMAKE_COMMAND} -E copy_if_different
-        "${ONNXRUNTIME_DIR}/lib/onnxruntime.dll"
-        "$<TARGET_FILE_DIR:piper_test>"
-    )
-endif()
-
 include(GoogleTest)
 get_target_property(PIPER_BINARY_DIR piper BINARY_DIR)
 get_target_property(PIPER_EXE_BINARY_DIR piper_exe BINARY_DIR)
