diff --git a/cmake/FindQuaZip.cmake b/cmake/FindQuaZip.cmake
index 8d309181..92aa892c 100644
--- a/cmake/FindQuaZip.cmake
+++ b/cmake/FindQuaZip.cmake
@@ -9,7 +9,6 @@ IF(QUAZIP_INCLUDE_DIRS AND QUAZIP_LIBRARIES)
   SET(QUAZIP_FOUND TRUE)
 ELSE(QUAZIP_INCLUDE_DIRS AND QUAZIP_LIBRARIES)
     IF(Qt5Core_FOUND)
-        set(QUAZIP_LIB_VERSION_SUFFIX 5)
     ENDIF()
   IF(WIN32)
     FIND_PATH(QUAZIP_LIBRARY_DIR
diff --git a/src/keeshare/ShareObserver.cpp b/src/keeshare/ShareObserver.cpp
index 33f5ed1f..5fcd9202 100644
--- a/src/keeshare/ShareObserver.cpp
+++ b/src/keeshare/ShareObserver.cpp
@@ -46,8 +46,8 @@
 #include <QStringBuilder>
 
 #if defined(WITH_XC_KEESHARE_SECURE)
-#include <quazip5/quazip.h>
-#include <quazip5/quazipfile.h>
+#include <quazip/quazip.h>
+#include <quazip/quazipfile.h>
 #endif
 
 namespace
