diff --git a/tests/integration/integration_test.go b/tests/integration/integration_test.go
index afbc5330..13745c70 100644
--- a/tests/integration/integration_test.go
+++ b/tests/integration/integration_test.go
@@ -246,8 +246,8 @@ func Test_EventFilters(t *testing.T) {
 			eventFunc:  checkExecve,
 		},
 		{
-			name:       "trace only execve events that starts with /usr/bin",
-			filterArgs: []string{"event=execve", "execve.pathname=/usr/bin*"},
+			name:       "trace only execve events that starts with /run",
+			filterArgs: []string{"event=execve", "execve.pathname=/run*"},
 			eventFunc:  checkExecve,
 		},
 		{
