diff --git i/src/GUI/app.cpp w/src/GUI/app.cpp
index 10e84d5..1e0abbe 100644
--- i/src/GUI/app.cpp
+++ w/src/GUI/app.cpp
@@ -34,11 +34,10 @@ App::App(int &argc, char **argv) : QApplication(argc, argv)
 	installTranslator(gpxsee);
 
 	QTranslator *qt = new QTranslator(this);
-#if defined(Q_OS_WIN32) || defined(Q_OS_MAC)
+#if defined(Q_OS_WIN32)
 	qt->load(QLocale::system(), "qt", "_", ProgramPaths::translationsDir());
 #else // Q_OS_WIN32 || Q_OS_MAC
-	qt->load(QLocale::system(), "qt", "_", QLibraryInfo::location(
-	  QLibraryInfo::TranslationsPath));
+	qt->load(QLocale::system(), "qt", "_", QLatin1String("@qttranslations@/translations"));
 #endif // Q_OS_WIN32 || Q_OS_MAC
 	installTranslator(qt);
 
