--- a/tests/meson.build
+++ b/tests/meson.build
@@ -97,10 +97,10 @@ test_data = [
   'bug775229.pixdata',
   'aero.gif',
 ]
 
-installed_test_bindir = join_paths(gdk_pixbuf_libexecdir, 'installed-tests', meson.project_name())
-installed_test_datadir = join_paths(gdk_pixbuf_datadir, 'installed-tests', meson.project_name())
+installed_test_bindir = join_paths('@installedtestsprefix@', 'libexec', 'installed-tests', meson.project_name())
+installed_test_datadir = join_paths('@installedtestsprefix@', 'share', 'installed-tests', meson.project_name())
 
 if get_option('installed_tests')
   install_data(test_data, install_dir: installed_test_bindir)
   install_subdir('test-images', install_dir: installed_test_bindir)
