diff --git a/src/crosvm.rs b/src/crosvm.rs
index b7055df..5989c87 100644
--- a/src/crosvm.rs
+++ b/src/crosvm.rs
@@ -141,7 +141,9 @@ impl Default for Config {
             x_display: None,
             shared_dirs: Vec::new(),
             sandbox: !cfg!(feature = "default-no-sandbox"),
-            seccomp_policy_dir: PathBuf::from(SECCOMP_POLICY_DIR),
+            seccomp_policy_dir: PathBuf::from(
+                option_env!("DEFAULT_SECCOMP_POLICY_DIR").unwrap_or(SECCOMP_POLICY_DIR),
+            ),
             seccomp_log_failures: false,
             cras_audio: false,
             cras_capture: false,
