diff --git i/src/crosvm.rs w/src/crosvm.rs
index ab7c466b..636dc140 100644
--- i/src/crosvm.rs
+++ w/src/crosvm.rs
@@ -345,7 +345,9 @@ impl Default for JailConfig {
     fn default() -> Self {
         JailConfig {
             pivot_root: PathBuf::from(option_env!("DEFAULT_PIVOT_ROOT").unwrap_or("/var/empty")),
-            seccomp_policy_dir: PathBuf::from(SECCOMP_POLICY_DIR),
+            seccomp_policy_dir: PathBuf::from(
+                option_env!("DEFAULT_SECCOMP_POLICY_DIR").unwrap_or(SECCOMP_POLICY_DIR),
+            ),
             seccomp_log_failures: false,
         }
     }
