diff -Naur --strip-trailing-cr -r platformio-core-4.1.0.org/platformio/proc.py platformio-core-4.1.0/platformio/proc.py
--- platformio-core-4.1.0.org/platformio/proc.py	2019-11-07 14:54:20.000000000 +0000
+++ platformio-core-4.1.0/platformio/proc.py	2019-12-31 09:05:47.693515281 +0000
@@ -167,7 +167,7 @@
             conditions.append(isdir(join(p, "click")) or isdir(join(p, "platformio")))
         if all(conditions):
             _PYTHONPATH.append(p)
-    os.environ["PYTHONPATH"] = os.pathsep.join(_PYTHONPATH)
+    os.environ['PYTHONPATH'] = os.pathsep.join(sys.path)
 
 
 def where_is_program(program, envpath=None):
