diff --git a/bokeh/util/compiler.py b/bokeh/util/compiler.py
index a752aad7d..8af05ff63 100644
--- a/bokeh/util/compiler.py
+++ b/bokeh/util/compiler.py
@@ -442,8 +442,8 @@ def _detect_nodejs():
     raise RuntimeError('node.js v%s or higher is needed to allow compilation of custom models ' % version +
                        '("conda install nodejs" or follow https://nodejs.org/en/download/)')
 
-_nodejs = None
-_npmjs = None
+_nodejs = "@node_bin@"
+_npmjs = "@npm_bin@"
 
 def _nodejs_path():
     global _nodejs
