From a5fc2936e8e13ebb73de4254234ca4d6ead15d1e Mon Sep 17 00:00:00 2001
From: Itamar Turner-Trauring <itamar@pythonspeed.com>
Date: Sun, 18 Jan 2026 14:15:02 -0500
Subject: [PATCH] Pass tests for 3.14.

---
 eliot/tests/test_coroutines.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/eliot/tests/test_coroutines.py b/eliot/tests/test_coroutines.py
index 39c5496c..cb2c1b13 100644
--- a/eliot/tests/test_coroutines.py
+++ b/eliot/tests/test_coroutines.py
@@ -34,7 +34,8 @@ def run_coroutines(*async_functions):
     """
     Run a coroutine until it finishes.
     """
-    loop = asyncio.get_event_loop()
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
     futures = [asyncio.ensure_future(f()) for f in async_functions]
 
     async def wait_for_futures():
