diff -Naur llvm-3.3.src-orig/tools/bugpoint/bugpoint.cpp llvm-3.3.src/tools/bugpoint/bugpoint.cpp
--- llvm-3.3.src-orig/tools/bugpoint/bugpoint.cpp	2013-01-27 20:35:51.000000000 -0500
+++ llvm-3.3.src/tools/bugpoint/bugpoint.cpp	2013-06-21 18:29:47.612731499 -0400
@@ -48,9 +48,9 @@
                       "is killed (default is 300s), 0 disables timeout"));
 
 static cl::opt<int>
-MemoryLimit("mlimit", cl::init(-1), cl::value_desc("MBytes"),
+MemoryLimit("mlimit", cl::init(0), cl::value_desc("MBytes"),
              cl::desc("Maximum amount of memory to use. 0 disables check."
-                      " Defaults to 100MB (800MB under valgrind)."));
+                      " Check disabled by default."));
 
 static cl::opt<bool>
 UseValgrind("enable-valgrind",
