--- bmake/unit-tests/Makefile.orig	2021-05-30 14:24:38.822484317 +0200
+++ bmake/unit-tests/Makefile	2021-05-31 13:25:21.645751428 +0200
@@ -455,7 +455,8 @@
 ENV.varmisc=		FROM_ENV=env
 ENV.varmisc+=		FROM_ENV_BEFORE=env
 ENV.varmisc+=		FROM_ENV_AFTER=env
-ENV.varmod-localtime+=	TZ=Europe/Berlin
+# Set absolute path to tz file since musl doesn't support TZDIR
+ENV.varmod-localtime+=	TZDIR=${TZDIR} TZ=:${TZDIR}/Europe/Berlin
 ENV.varname-vpath+=	VPATH=varname-vpath.dir:varname-vpath.dir2
 
 # Override make flags for some of the tests; default is -k.
--- bmake/unit-tests/varmod-localtime.mk.orig	2021-05-30 14:30:34.397986246 +0200
+++ bmake/unit-tests/varmod-localtime.mk	2021-05-31 13:24:41.430906606 +0200
@@ -3,7 +3,7 @@
 # Tests for the :localtime variable modifier, which formats a timestamp
 # using strftime(3) in local time.
 
-.if ${TZ} != "Europe/Berlin"	# see unit-tests/Makefile
+.if ${TZ} != ":${TZDIR}/Europe/Berlin"	# see unit-tests/Makefile
 .  error
 .endif
 
