--- a/aconfigure
+++ b/aconfigure
@@ -8945,6 +8945,10 @@
                                  ac_webrtc_instset=neon
                                  ac_webrtc_cflags="-DWEBRTC_ARCH_ARMV7 -mfloat-abi=hard -mfpu=neon"
                                  ;;
+                             arm64*|aarch64*)
+                                 ac_webrtc_instset=neon
+                                 ac_webrtc_cflags="-DWEBRTC_ARCH_ARM64"
+                                 ;;
                              *)
                                  ac_webrtc_instset=sse2
                                  ;;
