http://bugs.gentoo.org/331975
https://savannah.gnu.org/bugs/?30723

--- main.c	2010/07/19 07:10:53	1.243
+++ main.c	2010/08/10 07:35:34	1.244
@@ -2093,7 +2093,7 @@
             const char *pv = define_makeflags (1, 1);
             char *p = alloca (sizeof ("MAKEFLAGS=") + strlen (pv) + 1);
             sprintf (p, "MAKEFLAGS=%s", pv);
-            putenv (p);
+            putenv (allocated_variable_expand (p));
           }
 
 	  if (ISDB (DB_BASIC))
