diff --git a/src/v/security/gssapi_principal_mapper.cc b/src/v/security/gssapi_principal_mapper.cc
index 3a8a553cef..2403a0f97c 100644
--- a/src/v/security/gssapi_principal_mapper.cc
+++ b/src/v/security/gssapi_principal_mapper.cc
@@ -17,6 +17,7 @@
 #include <boost/algorithm/string/case_conv.hpp>
 #include <fmt/format.h>
 #include <fmt/ostream.h>
+#include <re2/re2.h>
 
 #include <charconv>
 
@@ -91,7 +92,7 @@ parse_rules(const std::vector<ss::sstring>& unparsed_rules) {
             if (conv_rc.ec != std::errc()) {
                 throw std::runtime_error(
                   "Invalid rule - Invalid value for number of components: "
-                  + num_components_str.as_string());
+                  + std::string(num_components_str));
             }
             gssapi_rule::case_change_operation case_change
               = gssapi_rule::case_change_operation::noop;
diff --git a/src/v/utils/log_hist.h b/src/v/utils/log_hist.h
index 6fb3272de5..abc436f439 100644
--- a/src/v/utils/log_hist.h
+++ b/src/v/utils/log_hist.h
@@ -18,6 +18,7 @@
 #include <bit>
 #include <chrono>
 #include <cstdint>
+#include <optional>
 
 /*
  * A histogram implementation
