Source: upstream rev. 269

=== modified file 'modules/ntrack-libnl.c'
--- old/modules/ntrack-libnl.c	2010-12-26 13:22:32 +0000
+++ new/modules/ntrack-libnl.c	2011-03-26 23:21:49 +0000
@@ -74,6 +74,7 @@
 	nl_cache_mngr_add (mngr, \
 	                   name, \
 	                   0, \
+	                   0, \
 	                   &cache_o)
 #define NL_PRINT_ERROR(text,err) \
 	printf ("%s/%s\n", text, nl_geterror(err))

