diff --git a/src/tauon/__main__.py b/src/tauon/__main__.py
index 04691586..e48afa02 100755
--- a/src/tauon/__main__.py
+++ b/src/tauon/__main__.py
@@ -115,8 +115,8 @@ def transfer_args_and_exit() -> None:
 if "--no-start" in sys.argv:
 	transfer_args_and_exit()
 
-# If we're installed, use home data locations
-install_mode = bool(str(install_directory).startswith(("/opt/", "/usr/", "/app/", "/snap/")) or sys.platform in ("darwin", "win32"))
+# Nixpkgs install, use home data dirs.
+install_mode = True
 
 # Assume that it's a classic Linux install, use standard paths
 if str(install_directory).startswith("/usr/") and Path("/usr/share/TauonMusicBox").is_dir():
