diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2651,7 +2651,7 @@
   )
   if(C_MACRO_PREFIX_MAP AND CXX_MACRO_PREFIX_MAP)
     if(APPLE)
-      if(XCODE AND ${XCODE_VERSION} VERSION_LESS 12.0)
+      if(FALSE)
         # Developers may have say LLVM Clang-10.0.1 toolchain (which supports the flag)
         # with Xcode-11 (the Clang of which doesn't support the flag).
         message(
diff --git a/build_files/cmake/platform/platform_apple.cmake b/build_files/cmake/platform/platform_apple.cmake
--- a/build_files/cmake/platform/platform_apple.cmake
+++ b/build_files/cmake/platform/platform_apple.cmake
@@ -55,7 +55,6 @@
   endif()
 endif()
 if(NOT EXISTS "${LIBDIR}/.git")
-  message(FATAL_ERROR "Mac OSX requires pre-compiled libs at: '${LIBDIR}'")
 endif()
 if(FIRST_RUN)
   message(STATUS "Using pre-compiled LIBDIR: ${LIBDIR}")
@@ -115,10 +114,6 @@
   find_library(_sndfile_VORBIS_LIBRARY NAMES vorbis HINTS ${LIBDIR}/ffmpeg/lib)
   find_library(_sndfile_VORBISENC_LIBRARY NAMES vorbisenc HINTS ${LIBDIR}/ffmpeg/lib)
   list(APPEND LIBSNDFILE_LIBRARIES
-    ${_sndfile_FLAC_LIBRARY}
-    ${_sndfile_OGG_LIBRARY}
-    ${_sndfile_VORBIS_LIBRARY}
-    ${_sndfile_VORBISENC_LIBRARY}
   )
 
   print_found_status("SndFile libraries" "${LIBSNDFILE_LIBRARIES}")
@@ -169,9 +164,7 @@
   set(FFMPEG_ROOT_DIR ${LIBDIR}/ffmpeg)
   set(FFMPEG_FIND_COMPONENTS
     avcodec avdevice avfilter avformat avutil
-    mp3lame ogg opus swresample swscale
-    theora theoradec theoraenc vorbis vorbisenc
-    vorbisfile vpx x264)
+    swresample swscale)
   # Frameworks required by libavfilter, using legacy macOS CGL
   string(APPEND PLATFORM_LINKFLAGS " -framework CoreImage -framework OpenGL")
   if(EXISTS ${LIBDIR}/ffmpeg/lib/libaom.a)
@@ -250,7 +243,6 @@
 endif()
 
 if(WITH_CODEC_FFMPEG)
-  string(APPEND PLATFORM_LINKFLAGS " -liconv") # ffmpeg needs it !
 endif()
 
 if(WITH_PUGIXML)
@@ -411,12 +403,12 @@
   " -Wl,-unexported_symbols_list,'${PLATFORM_SYMBOLS_MAP}'"
 )
 
-if(${XCODE_VERSION} VERSION_EQUAL 15.0)
+if(FALSE)
   # V4.5 specific workaround: Enforce the legacy Xcode linker to avoid incorrect
   # assembly generation caused by known bugs in the modern linker shipped with
   # Xcode 15.0. See issue #148792 for details.
   string(APPEND PLATFORM_LINKFLAGS " -Wl,-ld_classic")
-elseif(${XCODE_VERSION} VERSION_GREATER_EQUAL 15.0)
+elseif(FALSE)
   if("${CMAKE_OSX_ARCHITECTURES}" STREQUAL "x86_64" AND WITH_LEGACY_MACOS_X64_LINKER)
     # Silence "no platform load command found in <static library>, assuming: macOS".
     #
diff --git a/source/blender/blendthumb/CMakeLists.txt b/source/blender/blendthumb/CMakeLists.txt
--- a/source/blender/blendthumb/CMakeLists.txt
+++ b/source/blender/blendthumb/CMakeLists.txt
@@ -81,16 +81,6 @@
     # Prevent Xcode from overwriting the signature.
     XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY ""
   )
-  # CMake needs the target defined in the same file as add_custom_command.
-  # It needs to be code-signed (ad-hoc in this case)
-  # even on developer machine to generate thumbnails.
-  # Command taken from XCode build process.
-  add_custom_command(
-    TARGET blender-thumbnailer POST_BUILD
-    COMMAND codesign --deep --force --sign -
-      --entitlements "${CMAKE_SOURCE_DIR}/release/darwin/thumbnailer_entitlements.plist"
-      --timestamp=none $<TARGET_BUNDLE_DIR:blender-thumbnailer>
-  )
 elseif(UNIX)
   # -----------------------------------------------------------------------------
   # Build `blender-thumbnailer` executable
