diff --git a/invocations/packaging/release.py b/invocations/packaging/release.py
index 54322c3..81ac173 100644
--- a/invocations/packaging/release.py
+++ b/invocations/packaging/release.py
@@ -23,7 +23,7 @@ from shutil import rmtree
 
 from invoke.vendor.lexicon import Lexicon
 
-from blessings import Terminal
+from blessed import Terminal
 from docutils.utils import Reporter
 from enum import Enum
 from invoke import Collection, task, Exit
diff --git a/setup.py b/setup.py
index 78ae28b..c78a74c 100644
--- a/pyproject.toml	2026-04-25 00:07:29.730867343 +0200
+++ b/pyproject.toml	2026-04-25 00:08:23.736953102 +0200
@@ -16,7 +16,7 @@
     # time if missing), but that got hairy fast, and these are all
     # pure-Python packages, so it shouldn't be a huge burden for users to
     # obtain them.
-    "blessings>=1.6",
+    "blessed",
     "build>=1.3",
     # For envs that don't actually have pip - we use some of its tooling atm.
     "pip~=25.2",
