diff --git a/src/vcpkg/vcpkgpaths.cpp b/src/vcpkg/vcpkgpaths.cpp
index bb07a8fe..7dcf9c9b 100644
--- a/src/vcpkg/vcpkgpaths.cpp
+++ b/src/vcpkg/vcpkgpaths.cpp
@@ -607,7 +607,8 @@ namespace vcpkg
                         lock_taking_context = &wdc;
                     }
 
-                    const auto vcpkg_root_file = root / ".vcpkg-root";
+                    fs.create_directories(Path{"/tmp/vcpkg"}, VCPKG_LINE_INFO);
+                    const auto vcpkg_root_file = Path{"/tmp/vcpkg"} / Hash::get_string_sha256(root.c_str());
                     if (args.wait_for_lock.value_or(false))
                     {
                         file_lock_handle = fs.take_exclusive_file_lock(*lock_taking_context, vcpkg_root_file);
