diff --git a/src/Composer/Autoload/AutoloadGenerator.php b/src/Composer/Autoload/AutoloadGenerator.php
--- a/src/Composer/Autoload/AutoloadGenerator.php	(revision 556ca0690674f5e5376b721c2ea4985c7b1d5718)
+++ b/src/Composer/Autoload/AutoloadGenerator.php	(date 1724267672502)
@@ -268,6 +268,7 @@
             foreach ($paths as $path) {
                 $exportedPaths[] = $this->getPathCode($filesystem, $basePath, $vendorPath, $path);
             }
+            sort($exportedPaths);
             $exportedPrefix = var_export($namespace, true);
             $namespacesFile .= "    $exportedPrefix => ";
             $namespacesFile .= "array(".implode(', ', $exportedPaths)."),\n";
@@ -280,6 +281,7 @@
             foreach ($paths as $path) {
                 $exportedPaths[] = $this->getPathCode($filesystem, $basePath, $vendorPath, $path);
             }
+            sort($exportedPaths);
             $exportedPrefix = var_export($namespace, true);
             $psr4File .= "    $exportedPrefix => ";
             $psr4File .= "array(".implode(', ', $exportedPaths)."),\n";
