diff --git a/setup.py b/setup.py
index 990fe4cde3ca7c605b40ff2a9078feed2dedde48..91ce3bba0b2dcddd540aa76ebf2cb4a83c200eb6 100644
--- a/setup.py
+++ b/setup.py
@@ -20,6 +20,12 @@ from setuptools.command.build_ext import build_ext
 from setuptools_scm import get_version
 from torch.utils.cpp_extension import CUDA_HOME, ROCM_HOME
 
+if "NIX_ATTRS_JSON_FILE" in os.environ:
+    with open(os.environ["NIX_ATTRS_JSON_FILE"], "r") as f:
+        NIX_ATTRS = json.load(f)
+else:
+    NIX_ATTRS = {"cmakeFlags": os.environ.get("cmakeFlags", "").split()}
+
 
 def load_module_from_path(module_name, path):
     spec = importlib.util.spec_from_file_location(module_name, path)
@@ -213,6 +219,8 @@ class cmake_build_ext(build_ext):
         if other_cmake_args:
             cmake_args += other_cmake_args.split()
 
+        cmake_args += NIX_ATTRS["cmakeFlags"]
+
         subprocess.check_call(
             ["cmake", ext.cmake_lists_dir, *build_tool, *cmake_args],
             cwd=self.build_temp,
