diff --git a/misc/fuel/fuel-listener.el b/misc/fuel/fuel-listener.el
index 2d1b182a75..bf2e573425 100644
--- a/misc/fuel/fuel-listener.el
+++ b/misc/fuel/fuel-listener.el
@@ -30,13 +30,13 @@
   "Interacting with a Factor listener inside Emacs."
   :group 'fuel)
 
-(defcustom fuel-factor-root-dir nil
+(defcustom fuel-factor-root-dir "@fuel_factor_root_dir@"
   "Full path to the factor root directory when starting a listener."
   :type 'directory
   :group 'fuel-listener)
 
 ;;; Is factor.com still valid on Windows...?
-(defcustom fuel-listener-factor-binary nil
+(defcustom fuel-listener-factor-binary "@fuel_listener_factor_binary@"
   "Full path to the factor executable to use when starting a listener."
   :type '(file :must-match t)
   :group 'fuel-listener)
