diff --git a/osmgpsmap-1.0.pc.in b/osmgpsmap-1.0.pc.in
index 86efb3c..da6d1a9 100644
--- a/osmgpsmap-1.0.pc.in
+++ b/osmgpsmap-1.0.pc.in
@@ -6,6 +6,7 @@ includedir=@includedir@
 Name: @PACKAGE_NAME@
 Description: Moving map widget using openstreet map data
 Version: @PACKAGE_VERSION@
-Requires: gtk+-3.0 libsoup-3.0
+Requires: gtk+-3.0
+Requires.private: libsoup-3.0
 Libs: -L${libdir} -losmgpsmap-1.0
 Cflags: -I${includedir}/osmgpsmap-1.0
