diff --git a/src/crystal/system/unix/time.cr b/src/crystal/system/unix/time.cr
index 333b66075..1c29a0e55 100644
--- a/src/crystal/system/unix/time.cr
+++ b/src/crystal/system/unix/time.cr
@@ -43,6 +43,7 @@ module Crystal::System::Time
   # Many systems use /usr/share/zoneinfo, Solaris 2 has
   # /usr/share/lib/zoneinfo, IRIX 6 has /usr/lib/locale/TZ.
   ZONE_SOURCES = {
+    "@tzdata@/share/zoneinfo/",
     "/usr/share/zoneinfo/",
     "/usr/share/lib/zoneinfo/",
     "/usr/lib/locale/TZ/",
