## fix-ghc-7.8-build-failure.diff [diff]
diff --git a/gameServer/Actions.hs b/gameServer/Actions.hs
index 2cebe4f..355ee26 100644
--- a/gameServer/Actions.hs
+++ b/gameServer/Actions.hs
@@ -562,7 +562,7 @@ processAction (AddClient cl) = do
     si <- gets serverInfo
     newClId <- io $ do
         ci <- addClient rnc cl
-        _ <- Exception.mask (forkIO . clientRecvLoop (clientSocket cl) (coreChan si) (sendChan cl) ci)
+        _ <- Exception.mask (\x -> forkIO $ clientRecvLoop (clientSocket cl) (coreChan si) (sendChan cl) ci x)
 
         infoM "Clients" (show ci ++ ": New client. Time: " ++ show (connectTime cl))
 


