diff --git a/tests/test_strpdatetime.py b/tests/test_strpdatetime.py
index 6c371d6..a3d0232 100644
--- a/tests/test_strpdatetime.py
+++ b/tests/test_strpdatetime.py
@@ -44,5 +44,4 @@ TEST_DATA = [
 @pytest.mark.parametrize("string, format, expected", TEST_DATA)
 def test_datetime_strptime(string, format, expected):
     """Test datetime_strptime"""
-    locale.setlocale(locale.LC_ALL, "en_US.UTF-8")
     assert strpdatetime(string, format) == expected
