diff --git a/configure.ac b/configure.ac
index 347d325..dce95a0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -489,7 +489,8 @@ fi
 ############################################################################
 AM_ICONV
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.19])
+AM_GNU_GETTEXT_VERSION([0.19.6])
+AM_GNU_GETTEXT_REQUIRE_VERSION([0.19.6])
 LDFLAGS="$LDFLAGS $LIBINTL $LIBICONV"
 
 ############################################################################
