diff --git a/lib/builtins/CMakeLists.txt b/lib/builtins/CMakeLists.txt
index 3a66dd9c3fb..7efc85d9f9f 100644
--- a/lib/builtins/CMakeLists.txt
+++ b/lib/builtins/CMakeLists.txt
@@ -301,6 +301,10 @@ if (NOT MSVC)
     i386/umoddi3.S
   )
 
+  set(i486_SOURCES ${i386_SOURCES})
+  set(i586_SOURCES ${i386_SOURCES})
+  set(i686_SOURCES ${i386_SOURCES})
+
   if (WIN32)
     set(i386_SOURCES
       ${i386_SOURCES}
@@ -608,6 +612,7 @@ else ()
   endif()
 
   foreach (arch ${BUILTIN_SUPPORTED_ARCH})
+      message("arch: ${arch}")
     if (CAN_TARGET_${arch})
       # For ARM archs, exclude any VFP builtins if VFP is not supported
       if (${arch} MATCHES "^(arm|armhf|armv7|armv7s|armv7k|armv7m|armv7em)$")
