diff -ruN cdrtools-2.01/DEFAULTS/Defaults.linux cdrtools-2.01.new/DEFAULTS/Defaults.linux
--- cdrtools-2.01/DEFAULTS/Defaults.linux	2003-02-16 01:01:48.000000000 +0100
+++ cdrtools-2.01.new/DEFAULTS/Defaults.linux	2005-08-30 21:13:55.000000000 +0200
@@ -27,7 +27,8 @@
 # Installation config stuff
 #
 ###########################################################################
-INS_BASE=	/opt/schily
+#INS_BASE=	/opt/schily
+INS_BASE=	$(out)
 INS_KBASE=	/
 #
 DEFUMASK=	002
