diff --git a/src/cmake/Config.cmake.in b/src/cmake/Config.cmake.in
index c122b013..04f36bf0 100644
--- a/src/cmake/Config.cmake.in
+++ b/src/cmake/Config.cmake.in
@@ -43,11 +43,7 @@ if (NOT @BUILD_SHARED_LIBS@) # NOT @BUILD_SHARED_LIBS@
         # ZLIB_VERSION_STRING is still available for backward compatibility.
         # See https://cmake.org/cmake/help/git-stage/module/FindZLIB.html
 
-        if (@ZLIB_VERSION@) # @ZLIB_VERSION@
-            find_dependency(ZLIB @ZLIB_VERSION@)
-        else()
-            find_dependency(ZLIB @ZLIB_VERSION_STRING@)
-        endif()
+        find_dependency(ZLIB)
     endif()
 
     if (NOT TARGET MINIZIP::minizip-ng)
