diff --git a/content/common/sandbox_linux.cc b/content/common/sandbox_linux.cc
index d4618e5..108f846 100644
--- a/content/common/sandbox_linux.cc
+++ b/content/common/sandbox_linux.cc
@@ -38,15 +38,9 @@ void LogSandboxStarted(const std::string& sandbox_name) {
 // Implement the command line enabling logic for seccomp-legacy.
 bool IsSeccompLegacyDesired() {
 #if defined(SECCOMP_SANDBOX)
-#if defined(NDEBUG)
-  // Off by default; allow turning on with a switch.
-  return CommandLine::ForCurrentProcess()->HasSwitch(
-      switches::kEnableSeccompSandbox);
-#else
   // On by default; allow turning off with a switch.
   return !CommandLine::ForCurrentProcess()->HasSwitch(
       switches::kDisableSeccompSandbox);
-#endif  // NDEBUG
 #endif  // SECCOMP_SANDBOX
   return false;
 }
