diff --git a/m4/sasl2.m4 b/m4/sasl2.m4
index 098c853a..91d98def 100644
--- a/m4/sasl2.m4
+++ b/m4/sasl2.m4
@@ -350,7 +350,7 @@ int main(void)
 
     return (!have_spnego);  // 0 = success, 1 = failure
 }
-],[ac_cv_gssapi_supports_spnego=yes],[ac_cv_gssapi_supports_spnego=no])
+],[ac_cv_gssapi_supports_spnego=yes],[ac_cv_gssapi_supports_spnego=no],[ac_cv_gssapi_supports_spnego=no])
     LIBS="$cmu_save_LIBS"
   ])
   AS_IF([test "$ac_cv_gssapi_supports_spnego" = yes],[
