diff --git a/include/bitcoin/bitcoin/wallet/dictionary.hpp b/include/bitcoin/bitcoin/wallet/dictionary.hpp
index 632f1afc..63a51764 100644
--- a/include/bitcoin/bitcoin/wallet/dictionary.hpp
+++ b/include/bitcoin/bitcoin/wallet/dictionary.hpp
@@ -19,6 +19,7 @@
 #ifndef LIBBITCOIN_WALLET_DICTIONARY_HPP
 #define LIBBITCOIN_WALLET_DICTIONARY_HPP
 
+#include <cstddef>
 #include <array>
 #include <vector>
 #include <bitcoin/bitcoin/compat.hpp>
