diff --git a/meson.build b/meson.build
index 616ccff..d32b4c5 100644
--- a/meson.build
+++ b/meson.build
@@ -29,7 +29,7 @@ version = meson.project_version()
 prefix = get_option('prefix')
 bindir = prefix / get_option('bindir')
 localedir = prefix / get_option('localedir')
-graphviewsrcdir = prefix / 'src/delineate/graph_view'
+graphviewsrcdir = prefix / 'opt/delineate/graph_view'

 datadir = prefix / get_option('datadir')
 pkgdatadir = datadir / meson.project_name()
