diff --git a/tools/gpgkey2ssh.c b/tools/gpgkey2ssh.c
index f12c5f4..2e3f2ac 100644
--- a/tools/gpgkey2ssh.c
+++ b/tools/gpgkey2ssh.c
@@ -281,7 +281,7 @@ main (int argc, char **argv)
   keyid = argv[1];
 
   asprintf (&command,
-            "gpg2 --list-keys --with-colons --with-key-data '%s'",
+            "@out@/bin/gpg2 --list-keys --with-colons --with-key-data '%s'",
             keyid);
   if (! command)
     {
