diff --git a/module/Makefile b/module/Makefile
index fe573de..c8022c8 100644
--- a/module/Makefile
+++ b/module/Makefile
@@ -50,7 +50,7 @@ ifneq ($(KERNELRELEASE),)
 # inside kbuild
 # Note: this can be removed once it is in kernel tree and Kconfig is properly used
 CONFIG_DRM_EVDI := m
-ccflags-y := -isystem include/uapi/drm include/drm $(CFLAGS) $(EL8FLAG) $(EL9FLAG) $(RPIFLAG)
+ccflags-y := -isystem include/uapi/drm $(CFLAGS) $(EL8FLAG) $(EL9FLAG) $(RPIFLAG)
 evdi-y := evdi_platform_drv.o evdi_platform_dev.o evdi_sysfs.o evdi_modeset.o evdi_connector.o evdi_encoder.o evdi_drm_drv.o evdi_fb.o evdi_gem.o evdi_painter.o evdi_params.o evdi_cursor.o evdi_debug.o evdi_i2c.o
 evdi-$(CONFIG_COMPAT) += evdi_ioc32.o
 obj-$(CONFIG_DRM_EVDI) := evdi.o
diff --git a/module/evdi_drm.h b/module/evdi_drm.h
index 29b8427..5012693 100644
--- a/module/evdi_drm.h
+++ b/module/evdi_drm.h
@@ -12,12 +12,11 @@
 
 #ifdef __KERNEL__
 #include <linux/types.h>
+#include <drm/drm.h>
 #else
 #include <stdint.h>
 #endif
 
-#include "drm.h"
-
 /* Output events sent from driver to evdi lib */
 #define DRM_EVDI_EVENT_UPDATE_READY  0x80000000
 #define DRM_EVDI_EVENT_DPMS          0x80000001
