diff --git a/scwx-qt/tools/generate_versions.py b/scwx-qt/tools/generate_versions.py
index e7c470e..dde3e4f 100644
--- a/scwx-qt/tools/generate_versions.py
+++ b/scwx-qt/tools/generate_versions.py
@@ -74,15 +74,8 @@ def CollectVersionInfo(args):
 
     versionInfo = VersionInfo()
 
-    repo = git.Repo(args.gitRepo_, search_parent_directories = True)
-
-    commitString = str(repo.head.commit)[:10]
-    
-    if not repo.is_dirty(submodules = False):
-        copyrightYear = datetime.datetime.fromtimestamp(repo.head.commit.committed_date).year
-    else:
-        commitString  = commitString + "+dirty"
-        copyrightYear = datetime.date.today().year
+    commitString = "@rev@"
+    copyrightYear = datetime.date.today().year
     
     versionInfo.commitString_  = commitString
     versionInfo.copyrightYear_ = copyrightYear
