diff --git a/libexec/functions b/libexec/functions
index bc68107..6c2211c 100644
--- a/libexec/functions
+++ b/libexec/functions
@@ -29,16 +29,6 @@ if [ -z "${SINGULARITY_MESSAGELEVEL:-}" ]; then
     SINGULARITY_MESSAGELEVEL=5
 fi
 
-if [ -z "${USER:-}" ]; then
-    USER=`id -un`
-    export USER
-fi
-if [ -z "${HOME:-}" ]; then
-    HOME=`getent passwd "$USER" | cut -d : -f 6`
-    export HOME
-fi
-
-
 message() {
     LEVEL="${1:-}"
     MESSAGE="${2:-}"
