diff --git a/script/otbr-firewall b/script/otbr-firewall
index 52cef42dd0b..75840f8ae12 100644
--- a/script/otbr-firewall
+++ b/script/otbr-firewall
@@ -38,12 +38,8 @@
 # Description:       This service sets up firewall for OTBR.
 ### END INIT INFO
 
-THREAD_IF="wpan0"
 OTBR_FORWARD_INGRESS_CHAIN="OTBR_FORWARD_INGRESS"
 
-. /lib/lsb/init-functions
-. /lib/init/vars.sh
-
 set -euxo pipefail
 
 ipset_destroy_if_exist()
