--- a/openj9/runtime/CMakeLists.txt
+++ b/openj9/runtime/CMakeLists.txt
@@ -146,7 +146,7 @@ file(GLOB_RECURSE nls_files "${CMAKE_CURRENT_SOURCE_DIR}/nls/*.nls")
 add_custom_command(
 	OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/nls.stamp"
 	DEPENDS j9nls ${nls_files}
-	COMMAND "${Java_JAVA_EXECUTABLE}" -cp "$<TARGET_PROPERTY:j9nls,JAR_FILE>" com.ibm.oti.NLSTool.J9NLS -source "${CMAKE_CURRENT_SOURCE_DIR}"
+	COMMAND "${Java_JAVA_EXECUTABLE}" -Djava.properties.date=reproducibility -cp "$<TARGET_PROPERTY:j9nls,JAR_FILE>" com.ibm.oti.NLSTool.J9NLS -source "${CMAKE_CURRENT_SOURCE_DIR}"
 	COMMAND "${CMAKE_COMMAND}" -E touch "${CMAKE_CURRENT_BINARY_DIR}/nls.stamp"
 	COMMENT "Generating NLS files"
 	VERBATIM
