--- a/Makefile	2016-06-10 14:37:48.000000000 +0200
+++ b/Makefile	2016-06-23 05:10:26.000000000 +0200
@@ -4,17 +4,13 @@
 ZLIB_LIB=-lz
 
 # The directory containing the Zlib library (libz.a or libz.so)
-ZLIB_LIBDIR=/usr/lib
+ZLIB_LIBDIR=@ZLIB_LIBDIR@
 # ZLIB_LIBDIR=/usr/local/lib
 
 # The directory containing the Zlib header file (zlib.h)
-ZLIB_INCLUDE=/usr/include
+ZLIB_INCLUDE=@ZLIB_INCLUDE@
 # ZLIB_INCLUDE=/usr/local/include
 
-# Where to install the library.  By default: sub-directory 'zip' of
-# OCaml's standard library directory.
-INSTALLDIR=`$(OCAMLC) -where`/zip
-
 ### End of configuration section
 
 OCAMLC=ocamlc -g -safe-string
