diff --git a/core/migrations/0027_auto_20251120_0956.py b/core/migrations/0027_auto_20251120_0956.py
index fe795ff5..0db2090c 100644
--- a/core/migrations/0027_auto_20251120_0956.py
+++ b/core/migrations/0027_auto_20251120_0956.py
@@ -11,11 +11,6 @@ class Migration(migrations.Migration):
     operations = [
         migrations.RunSQL(
             sql="""
-                CREATE OR REPLACE FUNCTION public.immutable_unaccent(regdictionary, text)
-                RETURNS text
-                LANGUAGE c IMMUTABLE PARALLEL SAFE STRICT AS
-                '$libdir/unaccent', 'unaccent_dict';
-
                 CREATE OR REPLACE FUNCTION public.f_unaccent(text)
                 RETURNS text
                 LANGUAGE sql IMMUTABLE PARALLEL SAFE STRICT
