diff --git a/src/repository/opamRepositoryConfig.ml b/src/repository/opamRepositoryConfig.ml
index c2954c1d..528fc621 100644
--- a/src/repository/opamRepositoryConfig.ml
+++ b/src/repository/opamRepositoryConfig.ml
@@ -27,23 +27,7 @@ type 'a options_fun =
   'a
 
 let default = {
-  download_tool = lazy (
-    try
-      let tools =
-        if OpamStd.Sys.(os () = Darwin)
-        then ["wget", `Default; "curl", `Curl]
-        else ["curl", `Curl; "wget", `Default]
-      in
-      let cmd, kind =
-        List.find (fun (c,_) -> OpamSystem.resolve_command c <> None) tools
-      in
-      [ CIdent cmd, None ], kind
-    with Not_found ->
-      OpamConsole.error_and_exit `Configuration_error
-        "Could not find a suitable download command. Please make sure you \
-         have either \"curl\" or \"wget\" installed, or specify a custom \
-         command through variable OPAMFETCH."
-  );
+  download_tool = lazy ([ CIdent SUBSTITUTE_NIXOS_CURL_PATH, None ], `Curl);
   validation_hook = None;
   retries = 3;
   force_checksums = None;
