diff -rc glib-orig/glib.h glib-1.2.10/glib.h
*** glib-orig/glib.h	2001-02-27 04:44:38.000000000 +0100
--- glib-1.2.10/glib.h	2004-09-22 10:22:01.000000000 +0200
***************
*** 272,278 ****
  /* Wrap the gcc __PRETTY_FUNCTION__ and __FUNCTION__ variables with
   * macros, so we can refer to them as strings unconditionally.
   */
! #ifdef	__GNUC__
  #define	G_GNUC_FUNCTION		__FUNCTION__
  #define	G_GNUC_PRETTY_FUNCTION	__PRETTY_FUNCTION__
  #else	/* !__GNUC__ */
--- 272,278 ----
  /* Wrap the gcc __PRETTY_FUNCTION__ and __FUNCTION__ variables with
   * macros, so we can refer to them as strings unconditionally.
   */
! #if   defined(__GNUC__) && (__GNUC__ == 3 && __GNUC_MINOR__ < 4)
  #define	G_GNUC_FUNCTION		__FUNCTION__
  #define	G_GNUC_PRETTY_FUNCTION	__PRETTY_FUNCTION__
  #else	/* !__GNUC__ */
