diff --git a/cmake/FindEnchant.cmake b/cmake/FindEnchant.cmake
index 7c182e6a..5537595e 100644
--- a/cmake/FindEnchant.cmake
+++ b/cmake/FindEnchant.cmake
@@ -16,7 +16,7 @@ if(ENCHANT_INCLUDE_DIR AND ENCHANT_LIBRARIES)
 endif(ENCHANT_INCLUDE_DIR AND ENCHANT_LIBRARIES)
 
 find_package(PkgConfig)
-pkg_check_modules(PC_ENCHANT enchant)
+pkg_check_modules(PC_ENCHANT enchant-2)
 
 find_path(ENCHANT_INCLUDE_DIR
   NAMES enchant.h
@@ -24,7 +24,7 @@ find_path(ENCHANT_INCLUDE_DIR
   PATH_SUFFIXES "enchant")
 
 find_library(ENCHANT_LIBRARIES
-  NAMES enchant
+  NAMES enchant-2
   HINTS ${PC_ENCHANT_LIBRARY_DIRS})
 
 if(ENCHANT_INCLUDE_DIR AND ENCHANT_LIBRARIES)
@@ -39,7 +39,7 @@ if(ENCHANT_INCLUDE_DIR AND ENCHANT_LIBRARIES)
   #include <stdlib.h>
   #include <stddef.h>
   #include <string.h>
-  #include <enchant/enchant.h>
+  #include <enchant-2/enchant.h>
 
   EnchantBroker *enchant_broker_init();
   char **enchant_dict_suggest(EnchantDict *dict, const char *str,
@@ -78,6 +78,10 @@ if(ENCHANT_INCLUDE_DIR AND ENCHANT_LIBRARIES)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS_BACKUP}")
 endif()
 
+if(ENCHANT_API_COMPATIBLE)
+  set(ENCHANT_LIBRARY_FILENAME ${ENCHANT_LIBRARIES})
+endif(ENCHANT_API_COMPATIBLE)
+
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(Enchant DEFAULT_MSG ENCHANT_LIBRARIES
   ENCHANT_INCLUDE_DIR ENCHANT_API_COMPATIBLE)
