diff --git a/psw/ae/aesm_service/source/core/ipc/AEInitQuoteExRequest.cpp b/psw/ae/aesm_service/source/core/ipc/AEInitQuoteExRequest.cpp
index 270f831a..46187b10 100644
--- a/psw/ae/aesm_service/source/core/ipc/AEInitQuoteExRequest.cpp
+++ b/psw/ae/aesm_service/source/core/ipc/AEInitQuoteExRequest.cpp
@@ -107,9 +107,6 @@ bool AEInitQuoteExRequest::check()
         m_request->att_key_id().size() != sizeof(sgx_att_key_id_t))
         return false;

-    if (!m_request->has_buf_size())
-        return false;
-
     if (m_request->b_pub_key_id())
     {
         if (m_request->buf_size() > MAX_PUB_KEY_ID_BUF_SIZE)
diff --git a/psw/ae/aesm_service/source/core/ipc/messages.proto b/psw/ae/aesm_service/source/core/ipc/messages.proto
index 20b79fa9..7632079f 100644
--- a/psw/ae/aesm_service/source/core/ipc/messages.proto
+++ b/psw/ae/aesm_service/source/core/ipc/messages.proto
@@ -45,7 +45,7 @@ message Request{
     message InitQuoteExRequest{
         optional    bytes  att_key_id              = 1;
         required    bool   b_pub_key_id            = 3;
-        required    uint64 buf_size                = 4;
+        optional    uint64 buf_size                = 4;
         optional    uint32 timeout                 = 9;
     }
