From e7446c9bcb47674c9d0ee3b5bab129e9b86eb1c9 Mon Sep 17 00:00:00 2001
From: Walter Franzini <walter.franzini@gmail.com>
Date: Fri, 7 Jun 2019 17:57:11 +0200
Subject: [PATCH] musl does not support rewind pipe, make it build anyway

---
 src/formats.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/formats.c b/src/formats.c
index f3efe764..477bf451 100644
--- a/src/formats.c
+++ b/src/formats.c
@@ -424,7 +424,6 @@ static void UNUSED rewind_pipe(FILE * fp)
   /* To fix this #error, either simply remove the #error line and live without
    * file-type detection with pipes, or add support for your compiler in the
    * lines above.  Test with cat monkey.wav | ./sox --info - */
-  #error FIX NEEDED HERE
   #define NO_REWIND_PIPE
   (void)fp;
 #endif
-- 
2.19.2

