From ae8919eb81abad369e4a26ffcd845b140983398d Mon Sep 17 00:00:00 2001
From: Thomas Tuegel <ttuegel@gmail.com>
Date: Wed, 14 Oct 2015 06:28:57 -0500
Subject: [PATCH 1/2] qdiriterator follow symlinks

---
 src/sycoca/kbuildsycoca.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sycoca/kbuildsycoca.cpp b/src/sycoca/kbuildsycoca.cpp
index 1deae14..250baa8 100644
--- a/src/sycoca/kbuildsycoca.cpp
+++ b/src/sycoca/kbuildsycoca.cpp
@@ -208,7 +208,7 @@ bool KBuildSycoca::build()
         QStringList relFiles;
         const QStringList dirs = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, m_resourceSubdir, QStandardPaths::LocateDirectory);
         Q_FOREACH (const QString &dir, dirs) {
-            QDirIterator it(dir, QDirIterator::Subdirectories);
+            QDirIterator it(dir, QDirIterator::Subdirectories | QDirIterator::FollowSymlinks);
             while (it.hasNext()) {
                 const QString filePath = it.next();
                 Q_ASSERT(filePath.startsWith(dir)); // due to the line below...
-- 
2.5.2

