diff --git a/daemon/gdm-session-worker.c.orig b/daemon/gdm-session-worker.c
index 7bbda49..592691d 100644
--- a/daemon/gdm-session-worker.c.orig
+++ b/daemon/gdm-session-worker.c
@@ -1557,6 +1557,12 @@ gdm_session_worker_accredit_user (GdmSessionWorker  *worker,
                 goto out;
         }
 
+        if (g_getenv ("GDM_X_SERVER_EXTRA_ARGS") != NULL) {
+                g_debug ("forwarding GDM_X_SERVER_EXTRA_ARGS= %s", g_getenv("GDM_X_SERVER_EXTRA_ARGS"));
+                gdm_session_worker_set_environment_variable (worker, "GDM_X_SERVER_EXTRA_ARGS",
+                                                             g_getenv("GDM_X_SERVER_EXTRA_ARGS"));
+        }
+
         gdm_session_worker_update_environment_from_passwd_info (worker,
                                                                 uid,
                                                                 gid,
