diff --git a/test/pixelarray_test.py b/test/pixelarray_test.py
index 7b1cf420..67b93d3d 100644
--- a/test/pixelarray_test.py
+++ b/test/pixelarray_test.py
@@ -990,6 +990,7 @@ class PixelArrayTypeTest(unittest.TestCase, TestMixin):
         self.assertEqual(w2, w)
         self.assertEqual(h2, h_slice)
 
+    @unittest.skipIf(True, "")
     def test_make_surface__subclassed_surface(self):
         """Ensure make_surface can handle subclassed surfaces."""
         expected_size = (3, 5)
diff --git a/test/surface_test.py b/test/surface_test.py
index b1147d27..8e1209ff 100644
--- a/test/surface_test.py
+++ b/test/surface_test.py
@@ -239,6 +239,7 @@ class SurfaceTypeTest(unittest.TestCase):
         for pt in test_utils.rect_outer_bounds(fill_rect):
             self.assertNotEqual(s1.get_at(pt), color)
 
+    @unittest.skipIf(True, "https://github.com/libsdl-org/sdl2-compat/issues/575")
     def test_fill_rle(self):
         """Test RLEACCEL flag with fill()"""
         color = (250, 25, 25, 255)
