diff -ruN gildas-src-jan20a.orig/contrib/imager/pro/define.ima gildas-src-jan20a/contrib/imager/pro/define.ima
--- gildas-src-jan20a.orig/contrib/imager/pro/define.ima	2020-01-01 02:15:16.000000000 +0100
+++ gildas-src-jan20a/contrib/imager/pro/define.ima	2020-01-14 11:18:46.000000000 +0100
@@ -9,7 +9,7 @@
 !
 ! Patch for a Bug on Mac-OS/X where Python blocks if activated first
 ! from a script launched by a widget.
-python print "Starting Python" 
+python print("Starting Python")
 !
 ! INPUT, GO and UVT_CONVERT always defined by GreG
 define command GO "@ p_go.ima" gag_pro:go_greg.hlp  
