diff -ruN gildas-src-oct22b/admin/python-config-ldflags.py gildas-src-oct22b.patched/admin/python-config-ldflags.py
--- gildas-src-oct22b/admin/python-config-ldflags.py	2022-10-03 14:16:33.000000000 +0200
+++ gildas-src-oct22b.patched/admin/python-config-ldflags.py	2022-10-19 22:03:53.000000000 +0200
@@ -32,7 +32,7 @@
   libs.insert(0, '-L' + getvar('LIBDIR'))
 
 # Framework (specific for Mac)
-if not getvar('PYTHONFRAMEWORK'):
-    libs.extend(getvar('LINKFORSHARED').split())
+#if not getvar('PYTHONFRAMEWORK'):
+#    libs.extend(getvar('LINKFORSHARED').split())
 
 print(' '.join(libs))
