diff --git a/meson.build b/meson.build
index e5b33ad..c115ca4 100644
--- a/meson.build
+++ b/meson.build
@@ -175,9 +175,8 @@ add_project_arguments([
   '-DADW_VERSION_MAX_ALLOWED=ADW_ENCODE_VERSION(1,6,0)',
 ], language: 'c')
 
-subdir('accounts-service')
-subdir('help')
 if not get_option('use_system_libmalcontent')
+  subdir('accounts-service')
   subdir('libmalcontent')
 else
   libmalcontent_api_version = '0'
@@ -196,22 +195,27 @@ if get_option('ui').enabled()
   )
   subdir('libmalcontent-ui')
 endif
-subdir('libmalcontent-timer')
-subdir('libmalcontent-web')
-subdir('malcontent-client')
+if not get_option('use_system_libmalcontent')
+  subdir('libmalcontent-timer')
+  subdir('libmalcontent-web')
+  subdir('malcontent-client')
+endif
 if get_option('ui').enabled()
   subdir('malcontent-control')
+  subdir('help')
 
   gnome.post_install(
     gtk_update_icon_cache: true,
     update_desktop_database: true,
   )
 endif
-subdir('malcontent-timerd')
-subdir('malcontent-timer-extension-agent')
-subdir('malcontent-webd')
-subdir('malcontent-webd-update')
-subdir('nss')
-subdir('pam')
-subdir('po')
+if not get_option('use_system_libmalcontent')
+  subdir('malcontent-timerd')
+  subdir('malcontent-timer-extension-agent')
+  subdir('malcontent-webd')
+  subdir('malcontent-webd-update')
+  subdir('nss')
+  subdir('pam')
+  subdir('po')
+endif
 subdir('tests')
