diff --git a/src/common/settings.cpp b/src/common/settings.cpp
index 2c88356888..96198006b7 100644
--- a/src/common/settings.cpp
+++ b/src/common/settings.cpp
@@ -176,7 +176,7 @@ bool IsFastmemEnabled() {
     if (values.cpu_accuracy.GetValue() == CpuAccuracy::Unsafe) {
         return bool(values.cpuopt_unsafe_host_mmu);
     }
-#if !defined(__APPLE__) && !defined(__linux__) && !defined(__ANDROID__) && !defined(_WIN32)
+#if !defined(__APPLE__) && !defined(__linux__) && !defined(__ANDROID__) && !defined(_WIN32) || (defined(__linux__) && defined(__aarch64__))
     return false;
 #else
     return true;
