diff --git a/Makefile b/Makefile
index b2b5f2c..49d4c99 100644
--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@ LUA=lua
 LUA_CFLAGS=`$(PKG_CONFIG) --cflags lua5.2 2>/dev/null || $(PKG_CONFIG) --cflags lua`
 SOCFLAGS=-fPIC
 SOCC=$(CC) -shared $(SOCFLAGS)
-CFLAGS=-fPIC -g -Wall -Werror $(LUA_CFLAGS) -fvisibility=hidden -Wno-unused-function --std=gnu99
+CFLAGS=-fPIC -g -Wall $(LUA_CFLAGS) -fvisibility=hidden -Wno-unused-function --std=gnu99
 
 MODNAME=ffi
 MODSO=$(MODNAME).so
