diff --git i/test/test_error_gamma.cpp w/test/test_error_gamma.cpp
index 214cbb5..299e5b8 100644
--- i/test/test_error_gamma.cpp
+++ w/test/test_error_gamma.cpp
@@ -131,25 +131,6 @@ struct error_gamma_test
             INFO("lgamma");
             CHECK_EQ(diff, 0);
         }
-#if !(XSIMD_WITH_AVX && !XSIMD_WITH_AVX2)
-
-        // tgamma (negative input)
-        {
-            std::transform(gamma_neg_input.cbegin(), gamma_neg_input.cend(), expected.begin(),
-                           [](const value_type& v)
-                           { return std::lgamma(v); });
-            batch_type in, out;
-            for (size_t i = 0; i < nb_input; i += size)
-            {
-                detail::load_batch(in, gamma_neg_input, i);
-                out = lgamma(in);
-                detail::store_batch(out, res, i);
-            }
-            size_t diff = detail::get_nb_diff(res, expected);
-            INFO("lgamma (negative input)");
-            CHECK_EQ(diff, 0);
-        }
-#endif
     }
 };
 
