diff --git a/install/src/vboxguest-5.2.0/vboxvideo/Makefile.include.header b/install/src/vboxguest-5.2.0/vboxvideo/Makefile.include.header
index 8df1eb4d25..5a3e5604e7 100644
--- a/install/src/vboxguest-5.2.0/vboxvideo/Makefile.include.header
+++ b/install/src/vboxguest-5.2.0/vboxvideo/Makefile.include.header
@@ -122,7 +122,6 @@ ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
           Specify KERN_VER=<version> and run Make again)
 endif
 # Kernel include folder
-KERN_INCL := $(KERN_DIR)/include
 # module install folder
 INSTALL_MOD_DIR ?= misc
 MODULE_DIR := $(INSTALL_MOD_PATH)/lib/modules/$(KERN_VER)/$(INSTALL_MOD_DIR)
