diff --git a/plugins/php/php_plugin.c b/plugins/php/php_plugin.c
index ca0ef6c1..00c39b09 100644
--- a/plugins/php/php_plugin.c
+++ b/plugins/php/php_plugin.c
@@ -257,6 +257,9 @@ PHP_MINIT_FUNCTION(uwsgi_php_minit) {
 	return SUCCESS;
 }
 
+ZEND_BEGIN_ARG_INFO_EX(arginfo_void, 0, 0, 0)
+ZEND_END_ARG_INFO()
+
 PHP_FUNCTION(uwsgi_version) {
 	RETURN_STRING(UWSGI_VERSION);
 }
@@ -488,20 +491,20 @@ PHP_FUNCTION(uwsgi_signal) {
 }
 
 zend_function_entry uwsgi_php_functions[] = {
-	PHP_FE(uwsgi_version,   NULL)
-	PHP_FE(uwsgi_setprocname,   NULL)
-	PHP_FE(uwsgi_worker_id,   NULL)
-	PHP_FE(uwsgi_masterpid,   NULL)
-	PHP_FE(uwsgi_signal,   NULL)
-
-	PHP_FE(uwsgi_rpc,   NULL)
-
-	PHP_FE(uwsgi_cache_get,   NULL)
-	PHP_FE(uwsgi_cache_set,   NULL)
-	PHP_FE(uwsgi_cache_update,   NULL)
-	PHP_FE(uwsgi_cache_del,   NULL)
-	PHP_FE(uwsgi_cache_clear,   NULL)
-	PHP_FE(uwsgi_cache_exists,   NULL)
+	PHP_FE(uwsgi_version,   arginfo_void)
+	PHP_FE(uwsgi_setprocname,   arginfo_void)
+	PHP_FE(uwsgi_worker_id,   arginfo_void)
+	PHP_FE(uwsgi_masterpid,   arginfo_void)
+	PHP_FE(uwsgi_signal,   arginfo_void)
+
+	PHP_FE(uwsgi_rpc,   arginfo_void)
+
+	PHP_FE(uwsgi_cache_get,   arginfo_void)
+	PHP_FE(uwsgi_cache_set,   arginfo_void)
+	PHP_FE(uwsgi_cache_update,   arginfo_void)
+	PHP_FE(uwsgi_cache_del,   arginfo_void)
+	PHP_FE(uwsgi_cache_clear,   arginfo_void)
+	PHP_FE(uwsgi_cache_exists,   arginfo_void)
 	{ NULL, NULL, NULL},
 };
 
