diff --git a/tests/test_services.py b/tests/test_services.py
index 1fb07b3..f7a552c 100644
--- a/tests/test_services.py
+++ b/tests/test_services.py
@@ -3,6 +3,7 @@ from dataclasses import asdict
 
 import pytest
 from homeassistant.helpers.entity_registry import DATA_REGISTRY, EntityRegistry
+from homeassistant.helpers.trigger import TRIGGERS, TRIGGER_PLATFORM_SUBSCRIPTIONS
 from homeassistant.loader import (
     DATA_COMPONENTS,
     DATA_INTEGRATIONS,
@@ -31,6 +32,8 @@ def setup_hass_for_service_test(hass):
     hass.data[DATA_PRELOAD_PLATFORMS] = {}
     hass.data[DATA_MISSING_PLATFORMS] = {}
     hass.data[DATA_REGISTRY] = EntityRegistry(hass)
+    hass.data[TRIGGER_PLATFORM_SUBSCRIPTIONS] = []
+    hass.data[TRIGGERS] = {}
     return hass
 
 
