diff --git a/Cargo.lock b/Cargo.lock
index 6cebf66d..b6e40cd9 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -2443,6 +2443,7 @@ dependencies = [
  "rstest",
  "serial_test",
  "tempfile",
+ "zstd-sys",
 ]
 
 [[package]]
@@ -5365,4 +5366,5 @@ checksum = "9fd07cbbc53846d9145dbffdf6dd09a7a0aa52be46741825f5c97bdd4f73f12b"
 dependencies = [
  "cc",
  "libc",
+ "pkg-config",
 ]
diff --git a/Cargo.toml b/Cargo.toml
index 0791d462..d520d9ae 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -58,6 +58,9 @@ rayon = "1.5.1"
 reedline = { version = "0.6.0", features = ["bashisms"]}
 is_executable = "1.0.1"
 
+# Specify that the indirect dependency ztsd-sys should pick up the system zstd C library
+zstd-sys = { version = "2", features = [ "pkg-config" ] }
+
 [dev-dependencies]
 nu-test-support = { path="./crates/nu-test-support", version = "0.63.0"  }
 tempfile = "3.2.0"
