diff --git a/chromadb/test/conftest.py b/chromadb/test/conftest.py
index efde1c382..163f55c57 100644
--- a/chromadb/test/conftest.py
+++ b/chromadb/test/conftest.py
@@ -678,9 +678,6 @@ def sqlite_persistent(request: pytest.FixtureRequest) -> Generator[System, None,
 
 def system_fixtures() -> List[Callable[[], Generator[System, None, None]]]:
     fixtures = [
-        fastapi,
-        async_fastapi,
-        fastapi_persistent,
         sqlite_fixture,
         sqlite_persistent_fixture,
     ]
