From 834cb4f51555f3e027ad0632970e0fdd8ec38a85 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sandro=20J=C3=A4ckel?= <sandro.jaeckel@gmail.com>
Date: Thu, 5 Mar 2026 03:19:19 +0100
Subject: [PATCH] Fix Python 3.14 compatibility

---
 tests/test_monoprice.py | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/tests/test_monoprice.py b/tests/test_monoprice.py
index 60def3b..3039895 100644
--- a/tests/test_monoprice.py
+++ b/tests/test_monoprice.py
@@ -332,7 +332,7 @@ class TestAsyncMonoprice(TestMonoprice):
 
     def setUp(self):
         self.responses = {}
-        loop = asyncio.get_event_loop()
+        loop = asyncio.new_event_loop()
         monoprice = loop.run_until_complete(get_async_monoprice(create_dummy_port(self.responses)))
 
         # Dummy monoprice that converts async to sync
