diff -ru -x '*~' dbus-1.2.24-orig/bus/config-parser.c dbus-1.2.24/bus/config-parser.c
--- dbus-1.2.24-orig/bus/config-parser.c	2010-03-23 20:01:27.000000000 +0100
+++ dbus-1.2.24/bus/config-parser.c	2010-07-20 14:17:20.000000000 +0200
@@ -2159,12 +2159,16 @@
 
   retval = FALSE;
   
-  dir = _dbus_directory_open (dirname, error);
+  dbus_error_init (&tmp_error);
+  
+  dir = _dbus_directory_open (dirname, &tmp_error);
 
   if (dir == NULL)
-    goto failed;
+    {
+      retval = TRUE;
+      goto failed;
+    }
 
-  dbus_error_init (&tmp_error);
   while (_dbus_directory_get_next_file (dir, &filename, &tmp_error))
     {
       DBusString full_path;
