diff --git a/smpeg2-config.in b/smpeg2-config.in
index 5cce954..0e61939 100644
--- a/smpeg2-config.in
+++ b/smpeg2-config.in
@@ -42,7 +42,7 @@ while test $# -gt 0; do
       if test @includedir@ != /usr/include ; then
         includes=-I@includedir@
       fi
-      echo $includes -I@includedir@/smpeg2 `@SDL_CONFIG@ --cflags`
+      echo $includes -I@includedir@/smpeg2 `@SDL2_CONFIG@ --cflags`
       ;;
     --libs)
       if [ "`uname`" = "SunOS" ]; then
@@ -50,7 +50,7 @@ while test $# -gt 0; do
       else
         libdirs="-L@libdir@ @SMPEG_RLD_FLAGS@"
       fi
-      echo $libdirs -lsmpeg2 `@SDL_CONFIG@ --libs`
+      echo $libdirs -lsmpeg2 `@SDL2_CONFIG@ --libs`
       ;;
     *)
       echo "${usage}" 1>&2
