diff -ruN gcc-4.0.2/configure gcc-4.0.2.new/configure
--- gcc-4.0.2/configure	2005-09-13 09:01:28.000000000 +0200
+++ gcc-4.0.2.new/configure	2006-01-20 20:38:09.000000000 +0100
@@ -3471,7 +3471,7 @@
 # being built; programs in there won't even run.
 if test "${build}" = "${host}" && test -d ${srcdir}/gcc; then
   # Search for pre-installed headers if nothing else fits.
-  FLAGS_FOR_TARGET=$FLAGS_FOR_TARGET' -B$(build_tooldir)/bin/ -B$(build_tooldir)/lib/ -isystem $(build_tooldir)/include -isystem $(build_tooldir)/sys-include'
+  FLAGS_FOR_TARGET=$FLAGS_FOR_TARGET' -B$(build_tooldir)/bin/ -B$(build_tooldir)/lib/ -isystem $(build_tooldir)/include -isystem $(build_tooldir)/sys-include $(NIX_EXTRA_CFLAGS)'
 fi
 
 if test "x${use_gnu_ld}" = x &&
