From 6a6a7d1c708a2f568277699c5b605fa03ccb5faa Mon Sep 17 00:00:00 2001
From: Jascha Geerds <jascha@jgeerds.name>
Date: Tue, 19 Sep 2017 03:17:20 +0200
Subject: [PATCH 3/3] Create config dir if it doesn't exist

---
 gtweak/gtksettings.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gtweak/gtksettings.py b/gtweak/gtksettings.py
index a11363b..2871c49 100644
--- a/gtweak/gtksettings.py
+++ b/gtweak/gtksettings.py
@@ -36,6 +36,10 @@ class GtkSettingsManager:
     def _get_keyfile(self):
         keyfile = None
         try:
+            config_dir = os.path.dirname(self._path)
+            if not os.path.isdir(config_dir):
+                os.makedirs(config_dir)
+
             keyfile = GLib.KeyFile()
             keyfile.load_from_file(self._path, 0)
         except MemoryError:
-- 
2.14.1

