diff -ruN qt-x11-free-3.3.3/configure qt-x11-free-3.3.3.new/configure
--- qt-x11-free-3.3.3/configure	2004-06-14 11:18:55.000000000 +0200
+++ qt-x11-free-3.3.3.new/configure	2005-11-12 19:39:43.000000000 +0100
@@ -16,9 +16,9 @@
 relconf=`basename $0`
 # the directory of this script is the "source tree"
 relpath=`dirname $0`
-relpath=`(cd $relpath; /bin/pwd)`
+relpath=`(cd $relpath; pwd)`
 # the current directory is the "build tree" or "object tree"
-outpath=`/bin/pwd`
+outpath=`pwd`
 
 # later cache the command line in config.status
 OPT_CMDLINE=`echo $@ | sed "s,-v ,,g; s,-v$,,g"`
