diff -rc qt-x11-free-3.3.3-orig/mkspecs/linux-g++/qmake.conf qt-x11-free-3.3.3/mkspecs/linux-g++/qmake.conf
*** qt-x11-free-3.3.3-orig/mkspecs/linux-g++/qmake.conf	2004-08-05 16:42:57.000000000 +0200
--- qt-x11-free-3.3.3/mkspecs/linux-g++/qmake.conf	2005-03-02 12:25:55.000000000 +0100
***************
*** 85,90 ****
  QMAKE_DEL_FILE		= rm -f
  QMAKE_DEL_DIR		= rmdir
  QMAKE_STRIP             = strip
! QMAKE_STRIPFLAGS_LIB 	+= --strip-unneeded
  QMAKE_CHK_DIR_EXISTS	= test -d
  QMAKE_MKDIR		= mkdir -p
--- 85,90 ----
  QMAKE_DEL_FILE		= rm -f
  QMAKE_DEL_DIR		= rmdir
  QMAKE_STRIP             = strip
! QMAKE_STRIPFLAGS_LIB 	+= --strip-debug
  QMAKE_CHK_DIR_EXISTS	= test -d
  QMAKE_MKDIR		= mkdir -p
