From 7d833508e3bc4c737834e9edf1c429d36f67a38c Mon Sep 17 00:00:00 2001
From: "M. A" <mak@nyantec.com>
Date: Sat, 25 Jun 2022 13:34:42 +0000
Subject: [PATCH] Remove geo from database.yml

---
 config/database.yml.postgresql | 28 ----------------------------
 1 file changed, 28 deletions(-)

diff --git a/config/database.yml.postgresql b/config/database.yml.postgresql
index c1b1247b5b08..a81437d1e1b5 100644
--- a/config/database.yml.postgresql
+++ b/config/database.yml.postgresql
@@ -26,13 +26,6 @@ production:
   #   username: git
   #   password: "secure password"
   #   host: localhost
-  geo:
-    adapter: postgresql
-    encoding: unicode
-    database: gitlabhq_geo_production
-    username: git
-    password: "secure password"
-    host: localhost
 
 #
 # Development specific
@@ -57,13 +50,6 @@ development:
   #   host: localhost
   #   variables:
   #     statement_timeout: 15s
-  geo:
-    adapter: postgresql
-    encoding: unicode
-    database: gitlabhq_geo_development
-    username: postgres
-    password: "secure password"
-    host: localhost
 
 #
 # Staging specific
@@ -84,13 +70,6 @@ staging:
   #   username: git
   #   password: "secure password"
   #   host: localhost
-  geo:
-    adapter: postgresql
-    encoding: unicode
-    database: gitlabhq_geo_staging
-    username: git
-    password: "secure password"
-    host: localhost
 
 # Warning: The database defined as "test" will be erased and
 # re-generated from your development database when you run "rake".
@@ -117,10 +96,3 @@ test: &test
   #   prepared_statements: false
   #   variables:
   #     statement_timeout: 15s
-  geo:
-    adapter: postgresql
-    encoding: unicode
-    database: gitlabhq_geo_test
-    username: postgres
-    password:
-    host: localhost
-- 
2.36.0

