diff -ru -x '*~' diffoscope-orig/diffoscope/__init__.py diffoscope/diffoscope/__init__.py
--- diffoscope-orig/diffoscope/__init__.py	1970-01-01 01:00:01.000000000 +0100
+++ diffoscope/diffoscope/__init__.py	2016-12-08 17:48:51.732122110 +0100
@@ -81,8 +81,8 @@
     for var in ['LANG', 'LC_NUMERIC', 'LC_TIME', 'LC_COLLATE', 'LC_MONETARY',
                 'LC_MESSAGES', 'LC_PAPER', 'LC_NAME', 'LC_ADDRESS',
                 'LC_TELEPHONE', 'LC_MEASUREMENT', 'LC_IDENTIFICATION']:
-        os.environ[var] = 'C'
-    os.environ['LC_CTYPE'] = 'C.UTF-8'
+        os.environ[var] = 'en_US'
+    os.environ['LC_CTYPE'] = 'en_US.UTF-8'
     os.environ['TZ'] = 'UTC'
 
 
