From 7a42ef048b420f9b939085ea8c6af6deb9c19dac Mon Sep 17 00:00:00 2001
From: Moritz Ulrich <moritz@tarn-vedra.de>
Date: Sun, 30 Oct 2016 22:17:54 +0100
Subject: [PATCH] Foo

---
 icons/Makefile | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/icons/Makefile b/icons/Makefile
index 709514c..9a5fa94 100644
--- a/icons/Makefile
+++ b/icons/Makefile
@@ -9,10 +9,9 @@ all: $(ICONS)
 
 %.png:
 	mkdir -p hicolor/$(word 1, $(subst /, ,$(@)))/apps/
-	inkscape $(subst .png,.svg, $(word 2, $(subst /, , $(@)))) --without-gui \
-		--export-width=$(word 1, $(subst x, , $(@)))  \
-	        --export-height=$(word 2, $(subst x, , $(subst /, , $(@)))) \
-		--export-png=hicolor/$(word 1, $(subst /, ,$(@)))/apps/$(word 2, $(subst /, , $@))
+	convert $(subst .png,.svg, $(word 2, $(subst /, , $(@)))) \
+		-size $(word 1, $(subst x, , $(@)))x$(word 2, $(subst x, , $(subst /, , $(@)))) \
+		hicolor/$(word 1, $(subst /, ,$(@)))/apps/$(word 2, $(subst /, , $@))
 
 %.xpm:
 	mkdir -p pixmaps/
-- 
2.10.1

