diff -wbBur rdanbrook-nestopia-f1dde9b/Makefile rdanbrook-nestopia-f1dde9b.my/Makefile
--- rdanbrook-nestopia-f1dde9b/Makefile	2013-01-20 20:10:25.000000000 +0400
+++ rdanbrook-nestopia-f1dde9b.my/Makefile	2013-01-21 15:18:54.727577673 +0400
@@ -197,11 +197,11 @@
 	install -m 0644 NstDatabase.xml $(DATADIR)
 	install -m 0644 source/unix/icons/*.png $(DATADIR)/icons
 	install -m 0644 source/unix/icons/*.svg $(DATADIR)/icons
-	install -m 0644 source/unix/icons/nestopia.svg $(PREFIX)/share/pixmaps
-	xdg-desktop-menu install --novendor $(DATADIR)/nestopia.desktop
+	install -m 0644 source/unix/icons/nestopia.svg $(PREFIX)/share/pixmaps/nestopia.svg
+	install -Dm0644 $(DATADIR)/nestopia.desktop $(PREFIX)/share/applications/nestopia.desktop
 
 uninstall:
-	xdg-desktop-menu uninstall $(DATADIR)/nestopia.desktop
+	rm $(PREFIX)/share/applications/nestopia.desktop
 	rm $(PREFIX)/share/pixmaps/nestopia.svg
 	rm $(BINDIR)/$(BIN)
 	rm -rf $(DATADIR)
