diff --git a/xtmpl_js.ml b/xtmpl_js.ml
index e0d3894..991d1b3 100644
--- a/xtmpl_js.ml
+++ b/xtmpl_js.ml
@@ -25,6 +25,8 @@
 
 (** *)
 
+open Js_of_ocaml
+
 let log s = Firebug.console##log (Js.string s);;
 
 module X = Xtmpl_rewrite
diff --git a/xtmpl_js.mli b/xtmpl_js.mli
index d709896..5ed471c 100644
--- a/xtmpl_js.mli
+++ b/xtmpl_js.mli
@@ -25,6 +25,8 @@
 
 (** Convenient functions to use in JS code *)
 
+open Js_of_ocaml
+
 (** Create a new tree of DOM nodes from a given XML tree.
   Errors are logged to the firebug console.
   @param doc Default is [Dom_html.document].
