https://bugzilla.redhat.com/show_bug.cgi?id=454629

--- glibc-20090510T1842/localedata/locales/ca_ES.orig	2009-06-26 14:05:18.422043552 +0200
+++ glibc-20090510T1842/localedata/locales/ca_ES	2009-06-26 14:05:38.292042700 +0200
@@ -138,6 +138,8 @@
 date_fmt	"<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U0065>/
 <U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
 <U0025><U005A><U0020><U0025><U0059>"
+first_weekday   2
+first_workday   2
 END LC_TIME
 
 LC_PAPER
