--- hibernate-script-1.98.1/install.sh	2008-03-31 09:40:29.000000000 +0200
+++ hibernate-script-1.98.1/install.sh	2008-04-01 15:50:46.000000000 +0200
@@ -63,7 +63,7 @@ fi
 cp -a blacklisted-modules $BLACKLIST
 
 # Test if they have anything in there, and warn them
-if /bin/ls $OLD_SCRIPTLET_DIR/* > /dev/null 2>&1 ; then
+if ls $OLD_SCRIPTLET_DIR/* > /dev/null 2>&1 ; then
     echo "  **"
     echo "  ** You have scriptlets already installed in $OLD_SCRIPTLET_DIR"
     echo "  ** Since version 0.95, these have moved to $SCRIPTLET_DIR."
