From 63571e28c65935f32567c0b179a096d62726b778 Mon Sep 17 00:00:00 2001
From: Thomas Tuegel <ttuegel@mailbox.org>
Date: Tue, 2 Nov 2021 06:00:32 -0500
Subject: [PATCH 3/3] fusermount path

---
 kded/engine/fusebackend_p.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kded/engine/fusebackend_p.cpp b/kded/engine/fusebackend_p.cpp
index 91f3523..1c19d88 100644
--- a/kded/engine/fusebackend_p.cpp
+++ b/kded/engine/fusebackend_p.cpp
@@ -86,7 +86,7 @@ QProcess *FuseBackend::process(const QString &executable, const QStringList &arg
 
 QProcess *FuseBackend::fusermount(const QStringList &arguments) const
 {
-    return process("fusermount", arguments, {});
+    return process(NIXPKGS_FUSERMOUNT, arguments, {});
 }
 
 FutureResult<> FuseBackend::initialize(const QString &name, const Device &device, const MountPoint &mountPoint, const Vault::Payload &payload)
-- 
2.33.1

