diff --git a/meson.build b/meson.build
index 0b3b8c0..7f6ce09 100644
--- a/meson.build
+++ b/meson.build
@@ -86,7 +86,7 @@ if gnutls_dep.found()
   backends += ['gnutls']
   # test-specific, maybe move to tls/tests
   if cc.has_function('gnutls_pkcs11_init', prefix: '#include <gnutls/pkcs11.h>', dependencies: gnutls_dep)
-    config_h.set10('HAVE_GNUTLS_PKCS11', true)
+    config_h.set10('HAVE_GNUTLS_PKCS11', false)
   endif
 endif
 
