diff --git a/src/app/mainWindow.cpp b/src/app/mainWindow.cpp
index 68ec189..feb0e53 100644
--- a/src/app/mainWindow.cpp
+++ b/src/app/mainWindow.cpp
@@ -222,7 +222,7 @@ inline void MainWindow::slotComboScan()
 
 inline bool MainWindow::slotScanPath(const QString &path)
 {
-    return slotScanUrl(KUrl::KUrl(path));
+    return slotScanUrl(KUrl(path));
 }
 
 bool MainWindow::slotScanUrl(const KUrl &url)
diff --git a/src/part/radialMap/widget.cpp b/src/part/radialMap/widget.cpp
index d48a673..762f74e 100644
--- a/src/part/radialMap/widget.cpp
+++ b/src/part/radialMap/widget.cpp
@@ -64,7 +64,7 @@ RadialMap::Widget::path() const
 KUrl
 RadialMap::Widget::url(File const * const file) const
 {
-    return KUrl::KUrl(file ? file->fullPath() : m_tree->fullPath());
+    return KUrl(file ? file->fullPath() : m_tree->fullPath());
 }
 
 void
