diff --git a/appl_grid/appl_grid.h b/appl_grid/appl_grid.h
index 5059622..a0651c9 100644
--- a/appl_grid/appl_grid.h
+++ b/appl_grid/appl_grid.h
@@ -56,7 +56,7 @@ public:
   class exception : public std::exception { 
   public:
     exception(const std::string& s) { std::cerr << what() << " " << s << std::endl; }; 
-    exception(std::ostream& s)      { std::cerr << what() << " " << s << std::endl; }; 
+    exception(std::ostream& s)      { s << what() << " " << std::endl; }; 
     virtual const char* what() const throw() { return "appl::grid::exception"; }
   };
 
diff --git a/appl_grid/appl_pdf.h b/appl_grid/appl_pdf.h
index c71fd84..2525527 100644
--- a/appl_grid/appl_pdf.h
+++ b/appl_grid/appl_pdf.h
@@ -51,7 +51,7 @@ public:
   class exception : public std::exception { 
   public: 
     exception(const std::string& s="") { std::cerr << what() << " " << s << std::endl; }; 
-    exception(std::ostream& s)         { std::cerr << what() << " " << s << std::endl; }; 
+    exception(std::ostream& s)         { s << " " << std::endl; }; 
     const char* what() const throw() { return "appl::appl_pdf::exception "; }
   };
   
diff --git a/src/appl_igrid.h b/src/appl_igrid.h
index d25288e..be354df 100644
--- a/src/appl_igrid.h
+++ b/src/appl_igrid.h
@@ -52,7 +52,7 @@ private:
   class exception { 
   public:
     exception(const std::string& s) { std::cerr << s << std::endl; }; 
-    exception(std::ostream& s)      { std::cerr << s << std::endl; }; 
+    exception(std::ostream& s)      { s << std::endl; }; 
   };
 
   typedef double (igrid::*transform_t)(double) const;
diff --git a/src/lumi_pdf.cxx b/src/lumi_pdf.cxx
--- a/src/lumi_pdf.cxx
+++ b/src/lumi_pdf.cxx
@@ -235,6 +235,6 @@ void lumi_pdf::write(const std::string& filename) const {
 // std::string lumi_pdf::summary(std::ostream& s=std::cout) const { 
 std::string lumi_pdf::summary() const { 
   std::stringstream s;
-  s << "lumi_pdf::lumi_pdf() " << s << "\tsize " << m_combinations.size() << " lookup size " << m_lookup.size() << " " << this; 
+  s << "lumi_pdf::lumi_pdf() " << "\tsize " << m_combinations.size() << " lookup size " << m_lookup.size() << " " << this; 
   return s.str();
 }
