diff --git a/source/configurator/configurator.cpp b/source/configurator/configurator.cpp
index c058820..5a4b316 100644
--- a/source/configurator/configurator.cpp
+++ b/source/configurator/configurator.cpp
@@ -2155,7 +2155,7 @@ void MyDialog::OnsightrangesliderScroll(wxScrollEvent &e)
 	int v = sightRange->GetValue();
 	if(v == sightRange->GetMax())
 	{
-		s = "Unlimited";
+		s = _("Unlimited");
 	} else
 	{
 		s.Printf(wxT("%i m"), v);
