diff --git a/pixie/ffi-infer.pxi b/pixie/ffi-infer.pxi
index 9f13ac7..74301c2 100644
--- a/pixie/ffi-infer.pxi
+++ b/pixie/ffi-infer.pxi
@@ -1,15 +1,12 @@
 (ns pixie.ffi-infer
   (:require [pixie.io-blocking :as io]))
 
+(defn -add-library-path [p]
+  (swap! load-paths conj (str p "/include"))
+  (swap! load-paths conj (str p "/lib")))
 
-(defn -add-rel-path [rel]
-  (swap! load-paths conj (str (first @load-paths) "/" rel)))
-
-(-add-rel-path "lib")
-(-add-rel-path "include")
-(-add-rel-path "../lib")
-(-add-rel-path "../include")
-
+(doseq [lp @libraryPaths@]
+  (-add-library-path lp))
 
 (def *config* nil)
 (set-dynamic! (var *config*))
