diff --git a/Makefile.build b/Makefile.build
index 2963a3d..876479c 100644
--- a/Makefile.build
+++ b/Makefile.build
@@ -101,13 +101,8 @@ BYTEFLAGS=$(CAMLDEBUG) $(USERFLAGS)
 OPTFLAGS=$(CAMLDEBUGOPT) $(CAMLTIMEPROF) $(USERFLAGS)
 DEPFLAGS= $(LOCALINCLUDES) -I ide -I ide/utils
 
-ifeq ($(ARCH),Darwin)
-LINKMETADATA=-ccopt "-sectcreate __TEXT __info_plist config/Info-$(notdir $@).plist"
-CODESIGN=codesign -s -
-else
 LINKMETADATA=
 CODESIGN=true
-endif
 
 define bestocaml
 $(if $(OPT),\

