diff --git a/src/main/vencordFilesDir.ts b/src/main/vencordFilesDir.ts
index 8ef9ab8..ac376e0 100644
--- a/src/main/vencordFilesDir.ts
+++ b/src/main/vencordFilesDir.ts
@@ -10,4 +10,4 @@ import { SESSION_DATA_DIR } from "./constants";
 import { State } from "./settings";
 
 // this is in a separate file to avoid circular dependencies
-export const VENCORD_FILES_DIR = State.store.vencordDir || join(SESSION_DATA_DIR, "vencordFiles");
+export const VENCORD_FILES_DIR = State.store.vencordDir || "@vencord@";
