diff --git a/lib/CodeGen/AsmPrinter/CMakeLists.txt b/lib/CodeGen/AsmPrinter/CMakeLists.txt
index eb924282..c77c140b 100644
--- a/lib/CodeGen/AsmPrinter/CMakeLists.txt
+++ b/lib/CodeGen/AsmPrinter/CMakeLists.txt
@@ -44,3 +44,8 @@ add_llvm_component_library(LLVMAsmPrinter
   Support
   Target
   )
+
+if (CMAKE_COMPILER_IS_GNUCXX)
+  set_source_files_properties(DwarfCompileUnit.cpp PROPERTIES
+                              COMPILE_FLAGS -fno-strict-aliasing)
+endif()
