diff -rc mythtv-0.21-orig/configure mythtv-0.21/configure
*** mythtv-0.21-orig/configure	2008-03-04 22:23:23.000000000 +0100
--- mythtv-0.21/configure	2008-06-23 11:00:09.000000000 +0200
***************
*** 1687,1693 ****
  
  has_library()
  {
!     if test -f /etc/ld.so.conf ; then
          LIBPATHS=`cat /etc/ld.so.conf | grep -v "#" | grep -v "include"`
      else
          LIBPATHS=''
--- 1687,1693 ----
  
  has_library()
  {
!     if false ; then
          LIBPATHS=`cat /etc/ld.so.conf | grep -v "#" | grep -v "include"`
      else
          LIBPATHS=''
***************
*** 1698,1704 ****
      LIBPATHS="$LIBPATHS `echo $DYLD_LIBRARY_PATH | sed s/':'/' '/g` "
  
      # recurse down just one level, should be enough in most cases
!     if test -f /etc/ld.so.conf ; then
          INCL=`cat /etc/ld.so.conf | grep -v "#" | grep  "^include"`
          SRCH=`echo "$INCL" | sed '{ s/^include \//\//; s/^include /\/etc\//}'`
          for x in $SRCH ; do
--- 1698,1704 ----
      LIBPATHS="$LIBPATHS `echo $DYLD_LIBRARY_PATH | sed s/':'/' '/g` "
  
      # recurse down just one level, should be enough in most cases
!     if false ; then
          INCL=`cat /etc/ld.so.conf | grep -v "#" | grep  "^include"`
          SRCH=`echo "$INCL" | sed '{ s/^include \//\//; s/^include /\/etc\//}'`
          for x in $SRCH ; do
***************
*** 1707,1713 ****
          done
      fi
  
!     LIBPATHS="`echo $LIBPATHS /$libdir_name /usr/$libdir_name /usr/local/$libdir_name /usr/X11R6/$libdir_name | sed s/'\/ '/' '/g` "
  
      HAS_IT="no"
      for LIBPATH in $LIBPATHS ; do
--- 1707,1713 ----
          done
      fi
  
!     LIBPATHS="`echo $LIBPATHS | sed s/'\/ '/' '/g` "
  
      HAS_IT="no"
      for LIBPATH in $LIBPATHS ; do
***************
*** 1720,1728 ****
  
  has_header()
  {
!     HPATHS="/usr/local/include /usr/include /usr/include/g++-v3 /usr/X11R6/include"
      INCL=`echo $DYLD_LIBRARY_PATH $LD_LIBRARY_PATH | sed s/':'/' '/g`
!     HPATHS="$HPATHS "$(echo $INCL | sed "s/$libdir_name/include/g")
  
      HAS_IT="no"
      for HPATH in $HPATHS ; do
--- 1720,1728 ----
  
  has_header()
  {
!     HPATHS=""
      INCL=`echo $DYLD_LIBRARY_PATH $LD_LIBRARY_PATH | sed s/':'/' '/g`
!     HPATHS="$HPATHS "$(echo $INCL | sed "s/$libdir_name\($\|[\/ ]\)/include\1/g")
  
      HAS_IT="no"
      for HPATH in $HPATHS ; do
