diff -ruN old/src/kdenlivesettings.kcfg new/src/kdenlivesettings.kcfg
--- old/src/kdenlivesettings.kcfg	2019-09-10 23:20:27.555392353 -0400
+++ new/src/kdenlivesettings.kcfg	2019-09-10 23:25:47.533964155 -0400
@@ -378,14 +378,14 @@
 </group>
 
   <group name="env">
-    <entry name="mltpath" type="Path">
+    <entry name="mltpath" type="Path" hidden="true">
       <label>Mlt framework install path.</label>
-      <default></default>
+      <default>@mlt@/share/mlt/profiles</default>
     </entry>
 
-    <entry name="rendererpath" type="Path">
+    <entry name="rendererpath" type="Path" hidden="true">
       <label>Mlt melt renderer install path.</label>
-      <default></default>
+      <default>@mlt@/bin/melt</default>
     </entry>
 
     <entry name="ffmpegpath" type="Path">
