diff --git a/setup.py b/setup.py
index bf8badc..570be20 100644
--- a/setup.py
+++ b/setup.py
@@ -10,11 +10,11 @@ setup(name='baselines',
       packages=[package for package in find_packages()
                 if package.startswith('baselines')],
       install_requires=[
-          'gym[mujoco,atari,classic_control,robotics]',
+          'gym',
           'scipy',
           'tqdm',
           'joblib',
-          'zmq',
+          'pyzmq',
           'dill',
           'progressbar2',
           'mpi4py',
