--- a/libgdome/gdomecore/gdome-xml-documentt.c	2003-07-13 14:47:54.000000000 +0400
+++ b/libgdome/gdomecore/gdome-xml-documentt.c	2013-07-02 14:09:18.304613703 +0400
@@ -342,7 +342,7 @@
 	}
 	xmlNodeDumpOutput (out_buff, NULL, (xmlNode *)is, 0, 0, NULL);
 	xmlOutputBufferFlush(out_buff);
-	ret = g_strndup (out_buff->buffer->content, out_buff->buffer->use);
+	ret = g_strndup (xmlBufContent(out_buff), xmlBufUse(out_buff));
 	(void)xmlOutputBufferClose(out_buff);

 	return gdome_xml_str_mkref_own (ret);
