From d3f8111ade372c1eb7f3973031f59198508fb588 Mon Sep 17 00:00:00 2001
From: Kevin Liu <kevin@potatofrom.space>
Date: Thu, 23 Aug 2018 10:31:42 -0400
Subject: [PATCH] Remove download_external for spdlog

---
 cmake/Libraries.cmake | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/cmake/Libraries.cmake b/cmake/Libraries.cmake
index 1f951e59..2134444a 100644
--- a/cmake/Libraries.cmake
+++ b/cmake/Libraries.cmake
@@ -7,11 +7,6 @@ if(ENABLE_TESTS)
                     "ef5e700c8a0f3ee123e2e0209b8b4961")
 endif()
 
-download_external(SPDLOG "spdlog-0.14.0"
-                  "https://github.com/gabime/spdlog/archive/v0.14.0.zip"
-                  "f213d83c466aa7044a132e2488d71b11"
-                  "spdlog-1")
-
 # Find standard libraries
 find_package(Socket REQUIRED)
 find_package(Threads REQUIRED)
