--- a/Cargo.lock
+++ b/Cargo.lock
@@ -242,6 +242,7 @@ dependencies = [
  "toml_edit",
  "twox-hash",
  "url",
+ "zstd",
 ]
 
 [[package]]
@@ -1904,4 +1905,5 @@ checksum = "2141bed8922b427761470e6bbfeff255da94fa20b0bbeab0d9297fcaf71e3aa7"
 dependencies = [
  "cc",
  "libc",
+ "pkg-config",
 ]
diff --git a/Cargo.toml b/Cargo.toml
index 8f24673..b59c350 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -92,6 +92,8 @@ twox-hash = { version = "1.5", default-features = false }
 # Url parsing/manipulation
 url = "2.1"
 
+zstd = { version = "*", features = ["pkg-config"] }
+
 [dev-dependencies]
 # Avoid loading license check many times
 lazy_static = "1.4.0"
