diff -Naur test-patch-orig/C/Util/7zipInstall/resource.rc test-patch/C/Util/7zipInstall/resource.rc
--- test-patch-orig/C/Util/7zipInstall/resource.rc	2026-02-13 10:56:35.641911991 +0100
+++ test-patch/C/Util/7zipInstall/resource.rc	2026-02-13 10:56:37.940846112 +0100
@@ -1,7 +1,7 @@
 #include <windows.h>
 // #include <winnt.h>
 // #include <WinUser.h>
-#include <CommCtrl.h>
+#include <commctrl.h>
 
 #define USE_COPYRIGHT_CR
 #include "../../7zVersion.rc"
diff -Naur test-patch-orig/C/Util/7zipUninstall/resource.rc test-patch/C/Util/7zipUninstall/resource.rc
--- test-patch-orig/C/Util/7zipUninstall/resource.rc	2026-02-13 10:56:35.642411594 +0100
+++ test-patch/C/Util/7zipUninstall/resource.rc	2026-02-13 10:56:37.940994503 +0100
@@ -1,7 +1,7 @@
 #include <windows.h>
 // #include <winnt.h>
 // #include <WinUser.h>
-#include <CommCtrl.h>
+#include <commctrl.h>
 
 #define USE_COPYRIGHT_CR
 #include "../../7zVersion.rc"
diff -Naur test-patch-orig/CPP/7zip/GuiCommon.rc test-patch/CPP/7zip/GuiCommon.rc
--- test-patch-orig/CPP/7zip/GuiCommon.rc	2026-02-13 10:56:35.659762553 +0100
+++ test-patch/CPP/7zip/GuiCommon.rc	2026-02-13 10:56:37.941068290 +0100
@@ -4,7 +4,7 @@
 // #include <WinUser.h>
 
 // for Windows CE:
-#include <CommCtrl.h>
+#include <commctrl.h>
 
 
 LANGUAGE LANG_ENGLISH, SUBLANG_ENGLISH_US
diff -Naur test-patch-orig/CPP/7zip/UI/FileManager/PanelItemOpen.cpp test-patch/CPP/7zip/UI/FileManager/PanelItemOpen.cpp
--- test-patch-orig/CPP/7zip/UI/FileManager/PanelItemOpen.cpp	2026-02-13 10:56:35.668303663 +0100
+++ test-patch/CPP/7zip/UI/FileManager/PanelItemOpen.cpp	2026-02-13 10:56:37.944973625 +0100
@@ -4,7 +4,7 @@
 
 #include "../../../Common/MyWindows.h"
 
-#include <TlHelp32.h>
+#include <tlhelp32.h>
 
 #include "../../../Common/IntToString.h"
 
diff -Naur test-patch-orig/CPP/7zip/UI/FileManager/SysIconUtils.h test-patch/CPP/7zip/UI/FileManager/SysIconUtils.h
--- test-patch-orig/CPP/7zip/UI/FileManager/SysIconUtils.h	2026-02-13 10:56:35.670523698 +0100
+++ test-patch/CPP/7zip/UI/FileManager/SysIconUtils.h	2026-02-13 10:56:37.941136283 +0100
@@ -5,7 +5,7 @@
 
 #include "../../../Common/MyWindows.h"
 
-#include <CommCtrl.h>
+#include <commctrl.h>
 
 #include "../../../Common/MyString.h"
 
diff -Naur test-patch-orig/CPP/Windows/Control/ComboBox.h test-patch/CPP/Windows/Control/ComboBox.h
--- test-patch-orig/CPP/Windows/Control/ComboBox.h	2026-02-13 10:56:35.676322052 +0100
+++ test-patch/CPP/Windows/Control/ComboBox.h	2026-02-13 10:56:37.941211028 +0100
@@ -5,7 +5,7 @@
 
 #include "../../Common/MyWindows.h"
 
-#include <CommCtrl.h>
+#include <commctrl.h>
 
 #include "../Window.h"
 
diff -Naur test-patch-orig/CPP/Windows/Control/ImageList.h test-patch/CPP/Windows/Control/ImageList.h
--- test-patch-orig/CPP/Windows/Control/ImageList.h	2026-02-13 10:56:35.676476306 +0100
+++ test-patch/CPP/Windows/Control/ImageList.h	2026-02-13 10:56:37.941536358 +0100
@@ -3,7 +3,7 @@
 #ifndef ZIP7_INC_WINDOWS_CONTROL_IMAGE_LIST_H
 #define ZIP7_INC_WINDOWS_CONTROL_IMAGE_LIST_H
 
-#include <CommCtrl.h>
+#include <commctrl.h>
 
 #include "../Defs.h"
 
diff -Naur test-patch-orig/CPP/Windows/Control/ListView.h test-patch/CPP/Windows/Control/ListView.h
--- test-patch-orig/CPP/Windows/Control/ListView.h	2026-02-13 10:56:35.676528815 +0100
+++ test-patch/CPP/Windows/Control/ListView.h	2026-02-13 10:56:37.941655415 +0100
@@ -5,7 +5,7 @@
 
 #include "../../Common/MyWindows.h"
 
-#include <CommCtrl.h>
+#include <commctrl.h>
 
 #include "../Window.h"
 
diff -Naur test-patch-orig/CPP/Windows/Control/ProgressBar.h test-patch/CPP/Windows/Control/ProgressBar.h
--- test-patch-orig/CPP/Windows/Control/ProgressBar.h	2026-02-13 10:56:35.676556782 +0100
+++ test-patch/CPP/Windows/Control/ProgressBar.h	2026-02-13 10:56:37.941746329 +0100
@@ -5,7 +5,7 @@
 
 #include "../../Common/MyWindows.h"
 
-#include <CommCtrl.h>
+#include <commctrl.h>
 
 #include "../Window.h"
 
