From c7460a7fc255ef4e8e0e37798605c4d8bb50633d Mon Sep 17 00:00:00 2001
From: Orivej Desh <orivej@gmx.fr>
Date: Sat, 1 Apr 2017 01:30:37 +0000
Subject: [PATCH] path

---
 mainwindow.cpp | 2 +-
 platform.cpp   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/mainwindow.cpp b/mainwindow.cpp
index 9687c4e..234b176 100644
--- a/mainwindow.cpp
+++ b/mainwindow.cpp
@@ -688,7 +688,7 @@ void MainWindow::loadConfig() {
     if ( args.contains( "default-gamespath" ) )
 	defGamesPath = args["default-gamespath"].toString();
 
-    QString insteadPath = conf.value("InsteadPath", defInsteadPath).toString();
+    QString insteadPath = defInsteadPath;
     bool autoRefresh = conf.value("AutoRefresh", "false").toString() == "true";
     bool autoRefreshSW = conf.value("AutoRefreshSW", "false").toString() == "true";
     QString lang = conf.value( "Language", "*" ).toString();
diff --git a/platform.cpp b/platform.cpp
index bda86be..c356f1c 100644
--- a/platform.cpp
+++ b/platform.cpp
@@ -22,7 +22,7 @@ QString getConfigPath() {
 }
 
 QString getDefaultInterpreterPath() {
-    return "/usr/local/bin/sdl-instead";
+    return "@instead@/bin/sdl-instead";
 }
 
 #elif defined(Q_OS_WIN)
-- 
2.12.2

