diff --git a/core/config_reader.c b/core/config_reader.c
index 451fc48..ed45f4d 100644
--- a/core/config_reader.c
+++ b/core/config_reader.c
@@ -1355,7 +1355,7 @@ int read_config_file(const char* file)
   char file_path[PATH_MAX];
 
   snprintf(file_path, sizeof(file_path), "%s%s%s%s", gimx_params.homedir, GIMX_DIR, CONFIG_DIR, file);
-
+  snprintf(file_path, sizeof(file_path), "%s", file);
   if(read_file(file_path) == -1)
   {
     gerror("read_file failed\n");
diff --git a/core/gimx.c b/core/gimx.c
index 700cae9..9143d8b 100755
--- a/core/gimx.c
+++ b/core/gimx.c
@@ -192,7 +192,7 @@ void show_config()
   char file_path[PATH_MAX];
 
   snprintf(file_path, sizeof(file_path), "%s%s%s%s", gimx_params.homedir, GIMX_DIR, CONFIG_DIR, gimx_params.config_file);
-
+  snprintf(file_path, sizeof(file_path), "%s", gimx_params.config_file);
   FILE * fp = gfile_fopen(file_path, "r");
   if (fp == NULL)
   {
