diff --git a/pyproject.toml b/pyproject.toml
index 2677758..5bbe863 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,7 +30,7 @@ scipy = [
 ]
 lark = "^0.11.1"
 rpcq = "^3.10.0"
-pydantic = "^1.10.7"
+pydantic = ">=1.10.7"
 networkx = ">=2.5"
 importlib-metadata = { version = ">=3.7.3,<5", python = "<3.8" }
 qcs-sdk-python = "0.14.3"
diff --git a/pyquil/external/rpcq.py b/pyquil/external/rpcq.py
index 719eca1..d814d7b 100644
--- a/pyquil/external/rpcq.py
+++ b/pyquil/external/rpcq.py
@@ -1,6 +1,9 @@
 from typing import Dict, List, Union, Optional, Any
 from typing_extensions import Literal
-from pydantic import BaseModel, Field
+try:
+    from pydantic.v1 import BaseModel, Field
+except ImportError:
+    from pydantic import BaseModel, Field
 from rpcq.messages import TargetDevice as TargetQuantumProcessor
 
 
