diff --git a/meson.build b/meson.build
index a1ec1f67b6..8ce50973f7 100644
--- a/meson.build
+++ b/meson.build
@@ -864,8 +864,8 @@
 isocodes = dependency('iso-codes', required: false)
 if isocodes.found()
   isocodes_prefix = isocodes.get_pkgconfig_variable('prefix')
-  isocodes_location = isocodes_prefix / get_option('datadir') / 'xml' / 'iso-codes'
-  isocodes_localedir= isocodes_prefix / get_option('datadir') / 'locale'
+  isocodes_location = isocodes_prefix / 'share' / 'xml' / 'iso-codes'
+  isocodes_localedir= isocodes_prefix / 'share' / 'locale'
 endif
 conf.set('HAVE_ISO_CODES', isocodes.found())
 
