From 5366ca35b3d20ef962ccf54399cc44f523d803be Mon Sep 17 00:00:00 2001
From: GGG <gggkiller2@gmail.com>
Date: Mon, 5 Aug 2024 04:19:29 -0300
Subject: [PATCH] Disable building docs
---
 Makefile.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.in b/Makefile.in
index 41d7324..b1f5ec6 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -249,6 +249,7 @@ $(CURL_VERSION)/.chrome: $(chrome_libs)	$(CURL_VERSION).tar.gz $(CURL_VERSION)/.
 	# (for cross compilation), then pass it on to curl.
 	{ \
 	  config_flags="--prefix=@prefix@"; \
+	  config_flags="$$config_flags --disable-manual"; \
 	  config_flags="$$config_flags --with-nghttp2=$(nghttp2_install_dir)"; \
 	  config_flags="$$config_flags --with-brotli=$(brotli_install_dir)"; \
 	  config_flags="$$config_flags --with-openssl=$(boringssl_install_dir)"; \
