diff --git a/nautilus/mat2.py b/nautilus/mat2.py
index 11e6986..5a0e68f 100644
--- a/nautilus/mat2.py
+++ b/nautilus/mat2.py
@@ -22,6 +22,9 @@ import gi
 gi.require_version('Nautilus', '3.0')
 gi.require_version('Gtk', '3.0')
 gi.require_version('GdkPixbuf', '2.0')
+gi.require_version('GIRepository', '2.0')
+from gi.repository import GIRepository
+GIRepository.Repository.prepend_search_path('@poppler_path@')
 from gi.repository import Nautilus, GObject, Gtk, Gio, GLib, GdkPixbuf
 
 from libmat2 import parser_factory
