diff -rc splashutils-1.3-orig/splash.h splashutils-1.3/splash.h
*** splashutils-1.3-orig/splash.h	2006-07-15 00:02:23.000000000 +0200
--- splashutils-1.3/splash.h	2009-01-29 16:41:00.000000000 +0100
***************
*** 21,40 ****
   */
  #ifdef TARGET_KERNEL
  	#include <linux/fb.h>
- 	#include <linux/console_splash.h>
  #else
  	#include <fb.h>
- 	#include <console_splash.h>
  #endif
  
  /*
   * Necessary to avoid compilation errors when fbsplash support is
   * disabled.
   */
- #if !defined(CONFIG_FBSPLASH)
  	#define FB_SPLASH_IO_ORIG_USER		0
  	#define FB_SPLASH_IO_ORIG_KERNEL	1
! #endif
  
  /*
   * Adjustable settings
--- 21,60 ----
   */
  #ifdef TARGET_KERNEL
  	#include <linux/fb.h>
  #else
  	#include <fb.h>
  #endif
  
  /*
   * Necessary to avoid compilation errors when fbsplash support is
   * disabled.
   */
  	#define FB_SPLASH_IO_ORIG_USER		0
  	#define FB_SPLASH_IO_ORIG_KERNEL	1
! 
! #define FBIOSPLASH_SETCFG      _IOWR('F', 0x19, struct fb_splash_iowrapper)
! #define FBIOSPLASH_GETCFG      _IOR('F', 0x1A, struct fb_splash_iowrapper)
! #define FBIOSPLASH_SETSTATE    _IOWR('F', 0x1B, struct fb_splash_iowrapper)
! #define FBIOSPLASH_GETSTATE    _IOR('F', 0x1C, struct fb_splash_iowrapper)
! #define FBIOSPLASH_SETPIC      _IOWR('F', 0x1D, struct fb_splash_iowrapper)
! 
! #define FB_SPLASH_THEME_LEN            128     /* Maximum lenght of a theme name */
! 
! struct fb_splash_iowrapper
! {
!        unsigned short vc;              /* Virtual console */
!        unsigned char origin;           /* Point of origin of the request */
!        void *data;
! };
! 
! /* A structure used by the framebuffer splash code (drivers/video/fbsplash.c) */
! struct vc_splash {
!        __u8 bg_color;                          /* The color that is to be treated as transparent */
!        __u8 state;                             /* Current splash state: 0 = off, 1 = on */
!        __u16 tx, ty;                           /* Top left corner coordinates of the text field */
!        __u16 twidth, theight;                  /* Width and height of the text field */
!        char* theme;
! };
  
  /*
   * Adjustable settings
Only in splashutils-1.3/: splash.h~
