index 7ea81c7..0c19767 100644
--- a/src/utils/iso5426converter.cpp
+++ b/src/utils/iso5426converter.cpp
@@ -1211,7 +1211,11 @@ QChar Iso5426Converter::getCombiningChar(uint c) {
     return 0x1EF1; // SMALL LETTER U WITH HORN AND DOT BELOW
 
   default:
+#if (QT_VERSION < QT_VERSION_CHECK(5, 14, 0))
+    myDebug() << "no match for" << hex << c;
+#else
     myDebug() << "no match for" << Qt::hex << c;
+#endif
     return QChar();
   }
 }
