diff --git a/CMakeLists.txt b/CMakeLists.txt
index 31e8048..be97670 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,12 +5,7 @@ set(CMAKE_CXX_STANDARD 17)
 find_package(PkgConfig)
 include(FetchContent)
 
-FetchContent_Declare(
-        spdlog
-        GIT_REPOSITORY https://github.com/gabime/spdlog.git
-        GIT_TAG v1.10.0
-)
-FetchContent_MakeAvailable(spdlog)
+find_package(spdlog REQUIRED)
 
 pkg_check_modules(GTKMM REQUIRED gtkmm-3.0)
 find_package(X11 REQUIRED)
