From b9070aeab0ab672ffe321089631f9afe263b0caa Mon Sep 17 00:00:00 2001
From: Thomas Tuegel <ttuegel@gmail.com>
Date: Thu, 4 Jun 2015 12:03:32 -0500
Subject: [PATCH] work around buggy wcwidth

---
 test/unicode.jl | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test/unicode.jl b/test/unicode.jl
index 52c3e6a..f1ef698 100644
--- a/test/unicode.jl
+++ b/test/unicode.jl
@@ -103,5 +103,6 @@ end
 
 # handling of embedded NUL chars (#10958)
 @test length("\0w") == length("\0α") == 2
-@test strwidth("\0w") == strwidth("\0α") == 1
+@test strwidth("\0w") == charwidth('\0') + charwidth('w')
+@test strwidth("\0α") == charwidth('\0') + charwidth('α')
 @test normalize_string("\0W", casefold=true) == "\0w"
-- 
2.4.1

