From e2b0ed6664fe4adfd0f9ce8fa14732d47b30ab5c Mon Sep 17 00:00:00 2001
From: Thomas Tuegel <ttuegel@gmail.com>
Date: Sat, 10 Oct 2015 16:18:53 -0500
Subject: [PATCH 2/3] use system suitesparse

---
 base/sparse/cholmod.jl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/sparse/cholmod.jl b/base/sparse/cholmod.jl
index ec7e980..f834cc1 100644
--- a/base/sparse/cholmod.jl
+++ b/base/sparse/cholmod.jl
@@ -151,7 +151,7 @@ function __init__()
 
         # Register gc tracked allocator if CHOLMOD is new enough
         if current_version >= v"3.0.0"
-            cnfg = cglobal((:SuiteSparse_config, :libsuitesparseconfig), Ptr{Void})
+            cnfg = cglobal((:SuiteSparse_config, :libsuitesparse), Ptr{Void})
             unsafe_store!(cnfg, cglobal(:jl_malloc, Ptr{Void}), 1)
             unsafe_store!(cnfg, cglobal(:jl_calloc, Ptr{Void}), 2)
             unsafe_store!(cnfg, cglobal(:jl_realloc, Ptr{Void}), 3)
-- 
2.5.2

