diff --git a/src/daemon.c b/src/daemon.c
index c8b6320..2b74949 100644
--- a/src/daemon.c
+++ b/src/daemon.c
@@ -1102,7 +1102,7 @@ daemon_create_user_authorized_cb (Daemon                *daemon,
 
         sys_log (context, "create user '%s'", cd->user_name);
 
-        argv[0] = "/usr/sbin/useradd";
+        argv[0] = "@shadow@/bin/useradd";
         argv[1] = "-m";
         argv[2] = "-c";
         argv[3] = cd->real_name;
@@ -1335,7 +1335,7 @@ daemon_delete_user_authorized_cb (Daemon                *daemon,
         }
         free (resolved_homedir);
 
-        argv[0] = "/usr/sbin/userdel";
+        argv[0] = "@shadow@/bin/userdel";
         if (ud->remove_files) {
                 argv[1] = "-f";
                 argv[2] = "-r";
diff --git a/src/user.c b/src/user.c
index 189b2c5..5358c02 100644
--- a/src/user.c
+++ b/src/user.c
@@ -1145,7 +1145,7 @@ user_change_real_name_authorized_cb (Daemon                *daemon,
                         new_gecos = g_strdup (name);
                 }
 
-                argv[0] = "/usr/sbin/usermod";
+                argv[0] = "@shadow@/bin/usermod";
                 argv[1] = "-c";
                 argv[2] = new_gecos;
                 argv[3] = "--";
@@ -1218,7 +1218,7 @@ user_change_user_name_authorized_cb (Daemon                *daemon,
                          accounts_user_get_uid (ACCOUNTS_USER (user)),
                          name);
 
-                argv[0] = "/usr/sbin/usermod";
+                argv[0] = "@shadow@/bin/usermod";
                 argv[1] = "-l";
                 argv[2] = name;
                 argv[3] = "--";
@@ -1627,7 +1627,7 @@ user_change_home_dir_authorized_cb (Daemon                *daemon,
                          accounts_user_get_uid (ACCOUNTS_USER (user)),
                          home_dir);
 
-                argv[0] = "/usr/sbin/usermod";
+                argv[0] = "@shadow@/bin/usermod";
                 argv[1] = "-m";
                 argv[2] = "-d";
                 argv[3] = home_dir;
@@ -1683,7 +1683,7 @@ user_change_shell_authorized_cb (Daemon                *daemon,
                          accounts_user_get_uid (ACCOUNTS_USER (user)),
                          shell);
 
-                argv[0] = "/usr/sbin/usermod";
+                argv[0] = "@shadow@/bin/usermod";
                 argv[1] = "-s";
                 argv[2] = shell;
                 argv[3] = "--";
@@ -1824,7 +1824,7 @@ user_change_icon_file_authorized_cb (Daemon                *daemon,
                         return;
                 }
 
-                argv[0] = "/bin/cat";
+                argv[0] = "@coreutils@/bin/cat";
                 argv[1] = filename;
                 argv[2] = NULL;
 
@@ -1904,7 +1904,7 @@ user_change_locked_authorized_cb (Daemon                *daemon,
                          locked ? "locking" : "unlocking",
                          accounts_user_get_user_name (ACCOUNTS_USER (user)),
                          accounts_user_get_uid (ACCOUNTS_USER (user)));
-                argv[0] = "/usr/sbin/usermod";
+                argv[0] = "@shadow@/bin/usermod";
                 argv[1] = locked ? "-L" : "-U";
                 argv[2] = "--";
                 argv[3] = accounts_user_get_user_name (ACCOUNTS_USER (user));
@@ -2026,7 +2026,7 @@ user_change_account_type_authorized_cb (Daemon                *daemon,
 
                 g_free (groups);
 
-                argv[0] = "/usr/sbin/usermod";
+                argv[0] = "@shadow@/bin/usermod";
                 argv[1] = "-G";
                 argv[2] = str->str;
                 argv[3] = "--";
@@ -2093,7 +2093,7 @@ user_change_password_mode_authorized_cb (Daemon                *daemon,
                 if (mode == PASSWORD_MODE_SET_AT_LOGIN ||
                     mode == PASSWORD_MODE_NONE) {
 
-                        argv[0] = "/usr/bin/passwd";
+                        argv[0] = "/run/wrappers/bin/passwd";
                         argv[1] = "-d";
                         argv[2] = "--";
                         argv[3] = accounts_user_get_user_name (ACCOUNTS_USER (user));
@@ -2105,7 +2105,7 @@ user_change_password_mode_authorized_cb (Daemon                *daemon,
                         }
 
                         if (mode == PASSWORD_MODE_SET_AT_LOGIN) {
-                                argv[0] = "/usr/bin/chage";
+                                argv[0] = "@shadow@/bin/chage";
                                 argv[1] = "-d";
                                 argv[2] = "0";
                                 argv[3] = "--";
@@ -2126,7 +2126,7 @@ user_change_password_mode_authorized_cb (Daemon                *daemon,
                         accounts_user_set_locked (ACCOUNTS_USER (user), FALSE);
                 }
                 else if (accounts_user_get_locked (ACCOUNTS_USER (user))) {
-                        argv[0] = "/usr/sbin/usermod";
+                        argv[0] = "@shadow@/bin/usermod";
                         argv[1] = "-U";
                         argv[2] = "--";
                         argv[3] = accounts_user_get_user_name (ACCOUNTS_USER (user));
@@ -2203,7 +2203,7 @@ user_change_password_authorized_cb (Daemon                *daemon,
 
         g_object_freeze_notify (G_OBJECT (user));
 
-        argv[0] = "/usr/sbin/usermod";
+        argv[0] = "@shadow@/bin/usermod";
         argv[1] = "-p";
         argv[2] = strings[0];
         argv[3] = "--";
