diff --git a/src/alcotest/dune b/src/alcotest/dune
index 220a8b3..df1ffe0 100644
--- a/src/alcotest/dune
+++ b/src/alcotest/dune
@@ -3,6 +3,6 @@
   (name qcheck_alcotest)
   (public_name qcheck-alcotest)
   (wrapped false)
-  (libraries unix bytes qcheck-core qcheck-core.runner alcotest)
+  (libraries unix qcheck-core qcheck-core.runner alcotest)
   (flags :standard -w +a-4-42-44-48-50-58-32-60@8 -safe-string)
   )
diff --git a/src/core/dune b/src/core/dune
index ad0939f..42dc8ac 100644
--- a/src/core/dune
+++ b/src/core/dune
@@ -3,6 +3,6 @@
   (name qcheck_core)
   (public_name qcheck-core)
   (wrapped false)
-  (libraries unix bytes)
+  (libraries unix)
   (flags :standard -w +a-4-42-44-48-50-58-32-60@8 -safe-string)
   )
diff --git a/src/ounit/dune b/src/ounit/dune
index 2fadb7a..41f8d4b 100644
--- a/src/ounit/dune
+++ b/src/ounit/dune
@@ -3,6 +3,6 @@
   (name qcheck_ounit)
   (public_name qcheck-ounit)
   (wrapped false)
-  (libraries unix bytes qcheck-core qcheck-core.runner ounit2)
+  (libraries unix qcheck-core qcheck-core.runner ounit2)
   (flags :standard -w +a-4-42-44-48-50-58-32-60@8 -safe-string)
   )
