diff -ruN hotplug-2004_03_29/Makefile hotplug-2004_03_29.new/Makefile
--- hotplug-2004_03_29/Makefile	2003-06-28 02:13:10.000000000 +0200
+++ hotplug-2004_03_29.new/Makefile	2006-01-04 23:48:15.000000000 +0100
@@ -12,7 +12,7 @@
 mandir = ${prefix}/usr/share/man
 srcdir = .
 
-INSTALL = /usr/bin/install -c
+INSTALL = install -c
 INSTALL_PROGRAM = ${INSTALL}
 INSTALL_DATA  = ${INSTALL} -m 644
 INSTALL_SCRIPT = ${INSTALL_PROGRAM}
