diff --git a/vendor/glycin/src/sandbox.rs b/vendor/glycin/src/sandbox.rs
index 08db832..4f44b21 100644
--- a/vendor/glycin/src/sandbox.rs
+++ b/vendor/glycin/src/sandbox.rs
@@ -202,7 +202,7 @@ impl Sandbox {
 
                 args.push(self.exec());
 
-                ("bwrap".into(), args, Some(seccomp_memfd))
+                ("@bwrap@".into(), args, Some(seccomp_memfd))
             }
             SandboxMechanism::FlatpakSpawn => {
                 let memory_limit = Self::memory_limit();
@@ -299,8 +299,8 @@ impl Sandbox {
                 "/",
                 // Make /usr available as read only
                 "--ro-bind",
-                "/usr",
-                "/usr",
+                "/nix/store",
+                "/nix/store",
                 // Make tmpfs dev available
                 "--dev",
                 "/dev",
