From 8b1808e1278d2cb0dc56a4e98781eceeadfb9718 Mon Sep 17 00:00:00 2001
From: adisbladis <adis@blad.is>
Date: Thu, 12 Oct 2017 18:13:28 +0800
Subject: [PATCH] Disable tests using external networking

---
 test/cpp-utils/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/test/cpp-utils/CMakeLists.txt b/test/cpp-utils/CMakeLists.txt
index 5a2bc9f8..d838edba 100644
--- a/test/cpp-utils/CMakeLists.txt
+++ b/test/cpp-utils/CMakeLists.txt
@@ -20,7 +20,6 @@ set(SOURCES
     tempfile/TempFileIncludeTest.cpp
     tempfile/TempDirIncludeTest.cpp
     tempfile/TempDirTest.cpp
-    network/CurlHttpClientTest.cpp
     network/FakeHttpClientTest.cpp
     io/ConsoleIncludeTest.cpp
     io/ConsoleTest_AskYesNo.cpp
-- 
2.14.2

