diff -rc bash-3.1-orig/lib/sh/winsize.c bash-3.1/lib/sh/winsize.c
*** bash-3.1-orig/lib/sh/winsize.c	2006-10-18 18:08:25.000000000 +0200
--- bash-3.1/lib/sh/winsize.c	2006-10-18 18:10:00.000000000 +0200
***************
*** 30,35 ****
--- 30,37 ----
  
  #include <sys/ioctl.h>
  
+ #include <termios.h>
+ 
  #if !defined (STRUCT_WINSIZE_IN_SYS_IOCTL)
  /* For struct winsize on SCO */
  /*   sys/ptem.h has winsize but needs mblk_t from sys/stream.h */
