diff -ruN udev-068/Makefile udev-068.new/Makefile
--- udev-068/Makefile	2005-08-18 19:04:49.000000000 +0200
+++ udev-068.new/Makefile	2005-08-19 16:09:22.000000000 +0200
@@ -55,7 +55,7 @@
 STARTER =	udevstart
 RELEASE_NAME =	$(ROOT)-$(VERSION)
 LOCAL_CFG_DIR =	etc/udev
-DESTDIR =
+DESTDIR = ${out}
 KERNEL_DIR = /lib/modules/${shell uname -r}/build
 
 # override this to make udev look in a different location for it's config files
