diff --git a/src/modules/finished/FinishedPage.cpp b/src/modules/finished/FinishedPage.cpp
index 6c5f9ad16..24d75e07b 100644
--- a/src/modules/finished/FinishedPage.cpp
+++ b/src/modules/finished/FinishedPage.cpp
@@ -71,7 +71,10 @@ FinishedPage::retranslate()
         {
             ui->mainText->setText( tr( "<h1>All done.</h1><br/>"
                                        "%1 has been set up on your computer.<br/>"
-                                       "You may now start using your new system." )
+                                       "You may now start using your new system.<br/>"
+                                       "You can change every setting later except the bootloader.<br/>"
+                                       "Check the <a href=\"https://nixos.org/manual/nixos/stable/\">manual</a> for instructions on how to install software, upgrade the system or enable services.<br/>"
+                                       "You can find ways to get in touch with the <a href=\"https://nixos.org/community/\">community on the website!</a>")
                                        .arg( branding->versionedName() ) );
             ui->restartCheckBox->setToolTip( tr( "<html><head/><body>"
                                                  "<p>When this box is checked, your system will "
@@ -84,7 +87,10 @@ FinishedPage::retranslate()
             ui->mainText->setText( tr( "<h1>All done.</h1><br/>"
                                        "%1 has been installed on your computer.<br/>"
                                        "You may now restart into your new system, or continue "
-                                       "using the %2 Live environment." )
+                                       "using the %2 Live environment.<br/>"
+                                       "You can change every setting later except the bootloader.<br/>"
+                                       "Check the <a href=\"https://nixos.org/manual/nixos/stable/\">manual</a> for instructions on how to install software, upgrade the system or enable services.<br/>"
+                                       "You can find ways to get in touch with the <a href=\"https://nixos.org/community/\">community on the website!</a>")
                                        .arg( branding->versionedName(), branding->productName() ) );
             ui->restartCheckBox->setToolTip( tr( "<html><head/><body>"
                                                  "<p>When this box is checked, your system will "
diff --git a/src/modules/packagechooser/PackageChooserPage.cpp b/src/modules/packagechooser/PackageChooserPage.cpp
index 721329c1b..164b9945e 100644
--- a/src/modules/packagechooser/PackageChooserPage.cpp
+++ b/src/modules/packagechooser/PackageChooserPage.cpp
@@ -52,6 +52,7 @@ PackageChooserPage::currentChanged( const QModelIndex& index )
     if ( !index.isValid() || !ui->products->selectionModel()->hasSelection() )
     {
         ui->productName->setText( m_introduction.name.get() );
+        ui->productName->setStyleSheet("font-weight: bold");
         ui->productScreenshot->setPixmap( m_introduction.screenshot );
         ui->productDescription->setText( m_introduction.description.get() );
     }
diff --git a/src/modules/packagechooser/page_package.ui b/src/modules/packagechooser/page_package.ui
index d021b08b3..fecfa3060 100644
--- a/src/modules/packagechooser/page_package.ui
+++ b/src/modules/packagechooser/page_package.ui
@@ -38,19 +38,6 @@ SPDX-License-Identifier: GPL-3.0-or-later
      </item>
      <item>
-      <layout class="QVBoxLayout" name="verticalLayout" stretch="1,30,1">
+      <layout class="QVBoxLayout" name="verticalLayout" stretch="30,1,1">
-       <item>
-        <widget class="QLabel" name="productName">
-         <property name="sizePolicy">
-          <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
-           <horstretch>0</horstretch>
-           <verstretch>0</verstretch>
-          </sizepolicy>
-         </property>
-         <property name="text">
-          <string>Product Name</string>
-         </property>
-        </widget>
-       </item>
        <item>
         <widget class="FixedAspectRatioLabel" name="productScreenshot">
          <property name="sizePolicy">
@@ -67,6 +54,19 @@ SPDX-License-Identifier: GPL-3.0-or-later
          </property>
         </widget>
        </item>
+       <item>
+        <widget class="QLabel" name="productName">
+         <property name="sizePolicy">
+          <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
+           <horstretch>0</horstretch>
+           <verstretch>0</verstretch>
+          </sizepolicy>
+         </property>
+         <property name="text">
+          <string>Product Name</string>
+         </property>
+        </widget>
+       </item>
        <item>
         <widget class="QLabel" name="productDescription">
          <property name="sizePolicy">
