From: Stefano Zacchiroli <zack@upsilon.cc>
Date: Fri, 11 Dec 2009 13:26:26 +0100
Subject: [PATCH] gcc 4.4 build fixes

---
 src/common/AbstractLogger.cc |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/common/AbstractLogger.cc b/src/common/AbstractLogger.cc
index 6f638fb..c5b7e3d 100644
--- a/src/common/AbstractLogger.cc
+++ b/src/common/AbstractLogger.cc
@@ -19,6 +19,7 @@
 #include <config.h>
 
 #include <stdarg.h>
+#include <stdio.h>
 
 #include "AbstractLogger.hh"
 
-- 
