diff --git a/dynconfig/wscript b/dynconfig/wscript
index aa4e66e..d53f433 100755
--- a/dynconfig/wscript
+++ b/dynconfig/wscript
@@ -379,9 +379,9 @@ def build(bld):
                         cflags=cflags)
 
     # install some extra empty directories
-    bld.INSTALL_DIRS("", "${CONFIGDIR} ${PRIVATE_DIR} ${LOGFILEBASE}");
-    bld.INSTALL_DIRS("", "${PRIVATE_DIR} ${PRIVILEGED_SOCKET_DIR}")
-    bld.INSTALL_DIRS("", "${STATEDIR} ${CACHEDIR}");
+    #bld.INSTALL_DIRS("", "${CONFIGDIR} ${PRIVATE_DIR} ${LOGFILEBASE}");
+    #bld.INSTALL_DIRS("", "${PRIVATE_DIR} ${PRIVILEGED_SOCKET_DIR}")
+    #bld.INSTALL_DIRS("", "${STATEDIR} ${CACHEDIR}");
 
     # these might be on non persistent storage
-    bld.INSTALL_DIRS("", "${LOCKDIR} ${PIDDIR} ${SOCKET_DIR}")
+    #bld.INSTALL_DIRS("", "${LOCKDIR} ${PIDDIR} ${SOCKET_DIR}")
