--- ./test_prctl.py	2018-01-26 16:02:52.000000000 -0500
+++ ./test_prctl.py	2018-06-21 18:26:43.370065009 -0400
@@ -154,6 +154,7 @@
         prctl.set_keepcaps(False)
         self.assertEqual(prctl.get_keepcaps(), False)
 
+    @unittest.skip("No access to /proc in the Nix build sandbox")
     @require('set_mce_kill')
     def test_mce_kill(self):
         """Test the MCE_KILL setting"""
@@ -173,6 +174,7 @@
         prctl.set_name(name)
         self.assertEqual(prctl.get_name(), name[:15])
 
+    @unittest.skip("The Nix build sandbox has no_new_privs already enabled")
     @require('get_no_new_privs')
     def test_no_new_privs(self):
         """Test the no_new_privs function"""
@@ -189,6 +191,7 @@
                 self.assertNotEqual(sp.returncode, 0)
             os._exit(0)
 
+    @unittest.skip("No access to /proc in the Nix build sandbox")
     def test_proctitle(self):
         """Test setting the process title, including too long titles"""
         title = "This is a test!"
@@ -225,6 +228,7 @@
             os._exit(0)
         self.assertRaises(OSError, prctl.set_ptracer, new_pid)
 
+    @unittest.skip("The Nix build sandbox has seccomp already enabled")
     @require('get_seccomp')
     def test_seccomp(self):
         """Test manipulation of the seccomp setting"""
