commit 1770b1871a567dfa78ae1e32094b6007bb7639e9
Author: r-vdp <ramses@well-founded.dev>
Date:   Mon Oct 28 12:08:49 2024 +0100

    Get the efi app from fwupd-efi

diff --git a/meson.build b/meson.build
index e4e764b97..8acaa3d2d 100644
--- a/meson.build
+++ b/meson.build
@@ -482,7 +482,7 @@ endif
 
 # EFI
 if build_standalone
-  efi_app_location = join_paths(libexecdir, 'fwupd', 'efi')
+  efi_app_location = join_paths(dependency('fwupd-efi').get_variable(pkgconfig: 'prefix'), 'libexec', 'fwupd', 'efi')
   conf.set_quoted('EFI_APP_LOCATION', efi_app_location)
   if host_cpu == 'x86'
     EFI_MACHINE_TYPE_NAME = 'ia32'
