diff --git a/meson.build b/meson.build
index 9ae278b66..7cddf1a0d 100644
--- a/meson.build
+++ b/meson.build
@@ -507,7 +507,7 @@ if build_standalone
 if host_machine.system() == 'windows'
   libdir_pkg = 'fwupd-@0@'.format(fwupd_version)
 else
-  libdir_pkg = join_paths(libdir, 'fwupd-@0@'.format(fwupd_version))
+  libdir_pkg = join_paths(prefix, 'lib', 'fwupd-@0@'.format(fwupd_version))
 endif
 conf.set_quoted('FWUPD_LIBDIR_PKG', libdir_pkg)
 endif

