diff -rc junixsocket-1.3/src/main/org/newsclub/net/unix/NativeUnixSocket.java junixsocket-1.3-new/src/main/org/newsclub/net/unix/NativeUnixSocket.java
*** junixsocket-1.3/src/main/org/newsclub/net/unix/NativeUnixSocket.java        Tue Jul 20 14:59:41 2010
--- junixsocket-1.3-new/src/main/org/newsclub/net/unix/NativeUnixSocket.java    Sun May 27 22:26:15 2012
***************
*** 43,49 ****
              String prefix = "lib";
              String suffix = ".so";
              String os = osName.replaceAll("[^A-Za-z0-9]", "").toLowerCase();
!             if ("macosx".equals(os)) {
                  suffix = ".dylib";
              } else if ("linux".equals(os) || "freebsd".equals(os)
                      || "sunos".equals(os)) {
--- 43,49 ----
              String prefix = "lib";
              String suffix = ".so";
              String os = osName.replaceAll("[^A-Za-z0-9]", "").toLowerCase();
!             if ("macosx".equals(os) || "darwin".equals(os)) {
                  suffix = ".dylib";
              } else if ("linux".equals(os) || "freebsd".equals(os)
                      || "sunos".equals(os)) {
