diff -rc sdf-support-2.5/utils/sdf-modules/src/main.c sdf-support-2.5-new/utils/sdf-modules/src/main.c
*** sdf-support-2.5/utils/sdf-modules/src/main.c	2008-11-10 14:20:07.000000000 +0100
--- sdf-support-2.5-new/utils/sdf-modules/src/main.c	2010-08-24 10:53:04.000000000 +0200
***************
*** 19,25 ****
  /*{{{  defines */
  
  #define SEP '/'
! #define PATH_LEN (_POSIX_PATH_MAX)
  
  /*}}}  */
  /*{{{  variables */
--- 19,25 ----
  /*{{{  defines */
  
  #define SEP '/'
! #define PATH_LEN (256)
  
  /*}}}  */
  /*{{{  variables */
