diff --git a/src/orca/debug.py b/src/orca/debug.py
index e79482ed4..cbf3a24ec 100644
--- a/src/orca/debug.py
+++ b/src/orca/debug.py
@@ -502,7 +502,7 @@ def traceit(frame, event, arg):
     return traceit
 
 def getOpenFDCount(pid):
-    procs = subprocess.check_output([ 'lsof', '-w', '-Ff', '-p', str(pid)])
+    procs = subprocess.check_output([ '@lsof@', '-w', '-Ff', '-p', str(pid)])
     procs = procs.decode('UTF-8').split('\n')
     files = list(filter(lambda s: s and s[0] == 'f' and s[1:].isdigit(), procs))
 
@@ -510,7 +510,7 @@ def getOpenFDCount(pid):
 
 def getCmdline(pid):
     try:
-        openFile = os.popen('cat /proc/%s/cmdline' % pid)
+        openFile = os.popen('@cat@ /proc/%s/cmdline' % pid)
         cmdline = openFile.read()
         openFile.close()
     except:
@@ -520,7 +520,7 @@ def getCmdline(pid):
     return cmdline
 
 def pidOf(procName):
-    openFile = subprocess.Popen('pgrep %s' % procName,
+    openFile = subprocess.Popen('@pgrep@ %s' % procName,
                                 shell=True,
                                 stdout=subprocess.PIPE).stdout
     pids = openFile.read()
diff --git a/src/orca/orca.py b/src/orca/orca.py
index 2fe0a0bf2..087526556 100644
--- a/src/orca/orca.py
+++ b/src/orca/orca.py
@@ -285,7 +285,7 @@ def updateKeyMap(keyboardEvent):
 
 def _setXmodmap(xkbmap):
     """Set the keyboard map using xkbcomp."""
-    p = subprocess.Popen(['xkbcomp', '-w0', '-', os.environ['DISPLAY']],
+    p = subprocess.Popen(['@xkbcomp@', '-w0', '-', os.environ['DISPLAY']],
         stdin=subprocess.PIPE, stdout=None, stderr=None)
     p.communicate(xkbmap)
 
@@ -363,7 +363,7 @@ def _storeXmodmap(keyList):
     """
 
     global _originalXmodmap
-    _originalXmodmap = subprocess.check_output(['xkbcomp', os.environ['DISPLAY'], '-'])
+    _originalXmodmap = subprocess.check_output(['@xkbcomp@', os.environ['DISPLAY'], '-'])
 
 def _restoreXmodmap(keyList=[]):
     """Restore the original xmodmap values for the keys in keyList.
@@ -375,7 +375,7 @@ def _restoreXmodmap(keyList=[]):
 
     global _capsLockCleared
     _capsLockCleared = False
-    p = subprocess.Popen(['xkbcomp', '-w0', '-', os.environ['DISPLAY']],
+    p = subprocess.Popen(['@xkbcomp@', '-w0', '-', os.environ['DISPLAY']],
         stdin=subprocess.PIPE, stdout=None, stderr=None)
     p.communicate(_originalXmodmap)
 
diff --git a/src/orca/orca_bin.py.in b/src/orca/orca_bin.py.in
index 8c9d40153..eec0d5437 100644
--- a/src/orca/orca_bin.py.in
+++ b/src/orca/orca_bin.py.in
@@ -62,7 +62,7 @@ class ListApps(argparse.Action):
                 name = "[DEAD]"
 
             try:
-                cmdline = subprocess.getoutput('cat /proc/%s/cmdline' % pid)
+                cmdline = subprocess.getoutput('@cat@ /proc/%s/cmdline' % pid)
             except:
                 cmdline = '(exception encountered)'
             else:
@@ -197,7 +197,7 @@ def inGraphicalDesktop():
 def otherOrcas():
     """Returns the pid of any other instances of Orca owned by this user."""
 
-    openFile = subprocess.Popen('pgrep -u %s -x orca' % os.getuid(),
+    openFile = subprocess.Popen('@pgrep@ -u %s -x orca' % os.getuid(),
                                 shell=True,
                                 stdout=subprocess.PIPE).stdout
     pids = openFile.read()
