diff --git a/cctools/ld64/src/ld/Options.cpp b/cctools/ld64/src/ld/Options.cpp
index 2565518..9250016 100644
--- a/cctools/ld64/src/ld/Options.cpp
+++ b/cctools/ld64/src/ld/Options.cpp
@@ -2522,6 +2522,11 @@ void Options::parse(int argc, const char* argv[])
 					throw "missing argument to -rpath";
 				fRPaths.push_back(path);
 			}
+			else if ( strcmp(arg, "-rpath-link") == 0 ) {
+				const char* path = argv[++i];
+				if ( path == NULL )
+					throw "missing argument to -rpath-link";
+			}
 			else if ( strcmp(arg, "-read_only_stubs") == 0 ) {
 				fReadOnlyx86Stubs = true;
 			}
