--- a/src/meta/prefs.h
+++ b/src/meta/prefs.h
@@ -310,13 +310,13 @@ typedef struct
    */
   GSList *bindings;
 
-  /** for keybindings that can have shift or not like Alt+Tab */
+  /* for keybindings that can have shift or not like Alt+Tab */
   gboolean      add_shift:1;
 
-  /** for keybindings that apply only to a window */
+  /* for keybindings that apply only to a window */
   gboolean      per_window:1;
 
-  /** for keybindings not added with meta_display_add_keybinding() */
+  /* for keybindings not added with meta_display_add_keybinding() */
   gboolean      builtin:1;
 } MetaKeyPref;
 
@@ -339,5 +339,3 @@ CDesktopVisualBellType meta_prefs_get_vi
 MetaPlacementMode meta_prefs_get_placement_mode (void);
 
 #endif
-
-
--- a/src/core/workspace.c
+++ b/src/core/workspace.c
@@ -194,7 +194,7 @@ meta_workspace_new (MetaScreen *screen)
   return workspace;
 }
 
-/** Foreach function for workspace_free_struts() */
+/* Foreach function for workspace_free_struts() */
 static void
 free_this (gpointer candidate, gpointer dummy)
 {
@@ -1390,4 +1390,3 @@ meta_workspace_get_screen (MetaWorkspace
 {
   return workspace->screen;
 }
-
