diff --git a/repos.sexp b/repos.sexp
index 2b2459655..ad346d8e5 100644
--- a/repos.sexp
+++ b/repos.sexp
@@ -7,15 +7,15 @@
 ((:name :ansi-test
   :repository "https://gitlab.common-lisp.net/yitzchak/ansi-test"
   :directory "dependencies/ansi-test/"
-  :branch "add-expected-failures")
+  :commit "346cf2eb1133942054df8ce2125ed7e999b6d82b")
  (:name :cl-bench
   :repository "https://gitlab.common-lisp.net/ansi-test/cl-bench.git"
   :directory "dependencies/cl-bench/"
-  :branch "master")
+  :commit "7d184b4ef2a6272f0e3de88f6c243edb20f7071a")
  (:name :cl-who
   :repository "https://github.com/edicl/cl-who.git"
   :directory "dependencies/cl-who/"
-  :branch "master")
+  :commit "07dafe9b351c32326ce20b5804e798f10d4f273d")
  (:name :quicklisp-client
   :repository "https://github.com/quicklisp/quicklisp-client.git"
   :directory "dependencies/quicklisp-client/"
@@ -23,15 +23,15 @@
  (:name :shasht
   :repository "https://github.com/yitzchak/shasht.git"
   :directory "dependencies/shasht/"
-  :branch "master")
+  :commit "f38e866990c6b5381a854d63f7ea0227c87c2f6d")
  (:name :trivial-do
   :repository "https://github.com/yitzchak/trivial-do.git"
   :directory "dependencies/trivial-do/"
-  :branch "master")
+  :commit "a19f93227cb80a6bec8846655ebcc7998020bd7e")
  (:name :trivial-gray-streams
   :repository "https://github.com/trivial-gray-streams/trivial-gray-streams.git"
   :directory "dependencies/trivial-gray-streams/"
-  :branch "master")
+  :commit "2b3823edbc78a450db4891fd2b566ca0316a7876")
  (:name :acclimation
   :repository "https://github.com/robert-strandh/Acclimation.git"
   :directory "src/lisp/kernel/contrib/Acclimation/"
@@ -128,7 +128,7 @@
  (:name :lparallel
   :repository "https://github.com/yitzchak/lparallel.git"
   :directory "src/lisp/kernel/contrib/lparallel/"
-  :branch "fix-asdf-feature"
+  :commit "9c98bf629328b27a5a3fbb7a637afd1db439c00f"
   :extension :cando)
  (:name :parser.common-rules
   :repository "https://github.com/scymtym/parser.common-rules.git"
@@ -152,7 +152,7 @@
  (:name :trivial-features ; Needed both by the host and eclasp
   :repository "https://github.com/yitzchak/trivial-features.git"
   :directory "src/lisp/kernel/contrib/trivial-features/"
-  :branch "asdf-feature")
+  :commit "0008ef4f3376fb76f63c35ecee6573d0d0f98d57")
  (:name :trivial-garbage
   :repository "https://github.com/trivial-garbage/trivial-garbage.git"
   :directory "src/lisp/kernel/contrib/trivial-garbage/"
