diff --git a/server/database.js b/server/database.js
index d22ceb29..7490be8a 100644
--- a/server/database.js
+++ b/server/database.js
@@ -236,6 +236,7 @@ class Database {
             if (! fs.existsSync(Database.sqlitePath)) {
                 log.info("server", "Copying Database");
                 fs.copyFileSync(Database.templatePath, Database.sqlitePath);
+                fs.chmodSync(Database.sqlitePath, 0o640);
             }

             const Dialect = require("knex/lib/dialects/sqlite3/index.js");
