--- libffado-2.1.0/SConstruct	1970-01-01 01:00:01.000000000 +0100
+++ libffado-2.1.0/SConstruct	2012-09-30 16:50:23.603283095 +0000
@@ -349,11 +349,9 @@
 # Optional checks follow:
 #
 
-# PyQT checks
-build_mixer = False
-if conf.CheckForApp( 'which pyuic4' ) and conf.CheckForPyModule( 'dbus' ) and conf.CheckForPyModule( 'PyQt4' ) and conf.CheckForPyModule( 'dbus.mainloop.qt' ):
-    env['PYUIC4'] = True
-    build_mixer = True
+env['PYUIC4'] = True
+build_mixer = True
+
 
 if conf.CheckForApp( 'xdg-desktop-menu --help' ):
     env['XDG_TOOLS'] = True
@@ -384,7 +382,7 @@
     name2 = pkg.replace("+","").replace(".","").replace("-","").upper()
     env['%s_FLAGS' % name2] = conf.GetPKGFlags( pkg, pkgs[pkg] )
 
-if not env['DBUS1_FLAGS'] or not env['DBUSC1_FLAGS'] or not conf.CheckForApp('which dbusxx-xml2cpp'):
+if False:
     env['DBUS1_FLAGS'] = ""
     env['DBUSC1_FLAGS'] = ""
     print """
