--- libffado-r2117/SConstruct	1970-01-01 01:00:01.000000000 +0100
+++ libffado-r2117/SConstruct	2012-04-06 18:26:45.521100690 +0200
@@ -314,10 +314,8 @@
 #
 
 # PyQT checks
-build_mixer = False
-if conf.CheckForApp( 'which pyuic4' ) and conf.CheckForPyModule( 'dbus' ) and conf.CheckForPyModule( 'PyQt4' ) and conf.CheckForPyModule( 'dbus.mainloop.qt' ):
-    env['PYUIC4'] = True
-    build_mixer = True
+env['PYUIC4'] = True
+build_mixer = True
 
 if conf.CheckForApp( 'xdg-desktop-menu --help' ):
     env['XDG_TOOLS'] = True
@@ -348,7 +346,8 @@
     name2 = pkg.replace("+","").replace(".","").replace("-","").upper()
     env['%s_FLAGS' % name2] = conf.GetPKGFlags( pkg, pkgs[pkg] )
 
-if not env['DBUS1_FLAGS'] or not env['DBUSC1_FLAGS'] or not conf.CheckForApp('which dbusxx-xml2cpp'):
+
+if False:
     env['DBUS1_FLAGS'] = ""
     env['DBUSC1_FLAGS'] = ""
     print """
