Author: Adrian Knoth <adi@drcomp.erfurt.thur.de>
Forwarded: Yes
Applied-Upstream: 2.3.1
Last-Update: 2016-08-11
Description: Fix FTBFS with gcc6

Index: b/src/ffadotypes.h
===================================================================
--- a/src/ffadotypes.h
+++ b/src/ffadotypes.h
@@ -51,6 +51,8 @@ struct ffado_handle {
 };
 
 
+#include <cmath>
+#include <algorithm>
 #include <vector>
 #include <string>
 
