diff --git a/Makefile b/Makefile
index e40dbc3..e198b93 100644
--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,7 @@ ALLCFLAGS = -pipe $(CFLAGS) $(ADDCFLAGS) \
   -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations
 LDLIBRHASH = -Llibrhash -lrhash
 ALLLDFLAGS = $(LDLIBRHASH) $(LDFLAGS) $(ADDLDFLAGS)
-SHAREDLIB  = librhash/librhash.so.0
+SHAREDLIB  = librhash/librhash.0.dylib
 SHRDLFLAGS = $(LDFLAGS) $(ADDLDFLAGS)
 HEADERS = calc_sums.h hash_print.h common_func.h hash_update.h file_mask.h file_set.h find_file.h hash_check.h output.h parse_cmdline.h rhash_main.h win_utils.h version.h
 SOURCES = calc_sums.c hash_print.c common_func.c hash_update.c file_mask.c file_set.c find_file.c hash_check.c output.c parse_cmdline.c rhash_main.c win_utils.c
