Index: qtbase-opensource-src-5.8.0/src/gui/kernel/qguiapplication.cpp
===================================================================
--- qtbase-opensource-src-5.8.0.orig/src/gui/kernel/qguiapplication.cpp
+++ qtbase-opensource-src-5.8.0/src/gui/kernel/qguiapplication.cpp
@@ -1217,6 +1217,10 @@ void QGuiApplicationPrivate::createPlatf
 
     // Load the platform integration
     QString platformPluginPath = QString::fromLocal8Bit(qgetenv("QT_QPA_PLATFORM_PLUGIN_PATH"));
+    if (!platformPluginPath.isEmpty()) {
+        platformPluginPath.append(QStringLiteral(":"));
+    }
+    platformPluginPath.append(QStringLiteral(NIXPKGS_QPA_PLATFORM_PLUGIN_PATH));
 
 
     QByteArray platformName;
