diff --git a/common/MachOFile.cpp b/common/MachOFile.cpp
index 3e7b95bcfe..265ae7c475 100644
--- a/common/MachOFile.cpp
+++ b/common/MachOFile.cpp
@@ -2493,7 +2493,14 @@
 }
 #endif // BUILDING_APP_CACHE_UTIL || BUILDING_DYLDINFO
 
-#if BUILDING_CACHE_BUILDER || BUILDING_CACHE_BUILDER_UNIT_TESTS
+#if BUILDING_DYLDINFO || BUILDING_CACHE_BUILDER || BUILDING_CACHE_BUILDER_UNIT_TESTS
+
+#if BUILDING_DYLDINFO
+static bool startsWith(const char* buffer, const char* valueToFind) {
+    return strncmp(buffer, valueToFind, strlen(valueToFind)) == 0;
+}
+#endif
+
 static bool platformExcludesPrebuiltClosure_macOS(const char* path) {
     // We no longer support ROSP, so skip all paths which start with the special prefix
     if ( startsWith(path, "/System/Library/Templates/Data/") )
