diff --git a/other-tools/dyld_info.cpp b/other-tools/dyld_info.cpp
index 2de4978ba5..2ce27257d9 100644
--- a/other-tools/dyld_info.cpp
+++ b/other-tools/dyld_info.cpp
@@ -991,7 +991,7 @@
     return ((SymbolicatedImage*)di)->lookupSymbol(referencePC, referenceValue, *referenceType, *referenceName);
 }
 
-static int printDumpOpInfoCallback(void* di, uint64_t pc, uint64_t offset, uint64_t opSize, /* uint64_t instSize, */
+static int printDumpOpInfoCallback(void* di, uint64_t pc, uint64_t offset, uint64_t opSize, uint64_t instSize,
                                    int tagType, void* tagBuf)
 {
     return ((SymbolicatedImage*)di)->opInfo(pc, offset, opSize, tagType, tagBuf);
