diff --git a/configure.ac b/configure.ac
index 1e55b1cc..0832a387 100644
--- a/configure.ac
+++ b/configure.ac
@@ -83,12 +81,12 @@ elif test "x$enable_python_plugins" != xno; then
         AM_CONDITIONAL([BUILD_PYTHON_API], [true])
         AC_DEFINE([HAVE_PYTHON], [1], [Python support])
     else
-        if test "x$enable_python_plugins" = xyes; then
-            AC_MSG_ERROR([Python not found, cannot enable Python plugins.])
-        else
-            AM_CONDITIONAL([BUILD_PYTHON_API], [false])
-            AC_MSG_NOTICE([Python development package not found, Python plugin support disabled.])
-        fi
+        AS_IF(
+            [test "x$enable_python_plugins" = xyes],
+            [],
+            [AM_CONDITIONAL([BUILD_PYTHON_API], [false])
+            AC_MSG_NOTICE([Python development package not found, Python plugin support disabled.])]
+        )
     fi
     AS_IF([test "x$PLATFORM" = xosx], [rm -f Python.framework])
 else
@@ -107,7 +105,7 @@ else
             [AM_CONDITIONAL([BUILD_C_API], [true]) LIBS="$LIBS -ldl" AC_DEFINE([HAVE_C], [1], [C support])],
             [AS_IF(
                 [test "x$enable_c_plugins" = xyes],
-                    [AC_MSG_ERROR([dl library needed to run C plugins])],
+                [],
                 [AM_CONDITIONAL([BUILD_C_API], [false])])
             ])
     else
@@ -116,7 +114,6 @@ else
 fi
 
 # threading
-ACX_PTHREAD([], [AC_MSG_ERROR([pthread is required])])
 LIBS="$PTHREAD_LIBS $LIBS"
 CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
 AS_IF([test "x$PTHREAD_CC" != x], [ CC="$PTHREAD_CC" ])
