diff --git a/testsuite/test_api.py b/testsuite/test_api.py
index 8dde32ff..38e34acf 100644
--- a/testsuite/test_api.py
+++ b/testsuite/test_api.py
@@ -329,7 +329,10 @@ def test_check_nullbytes(self):
         count_errors = pep8style.input_file('stdin', lines=['\x00\n'])
 
         stdout = sys.stdout.getvalue()
-        expected = "stdin:1:1: E901 ValueError"
+        if sys.version_info < (3, 11, 4):
+            expected = "stdin:1:1: E901 ValueError"
+        else:
+            expected = "stdin:1:1: E901 SyntaxError: source code string cannot contain null bytes"  # noqa: E501
         self.assertTrue(stdout.startswith(expected),
                         msg='Output %r does not start with %r' %
                         (stdout, expected))
