diff --git a/meson.build b/meson.build
index ad776069..ba8027c2 100644
--- a/meson.build
+++ b/meson.build
@@ -16,8 +16,8 @@ libdl_bio_dep = [
 libc_bio_dep = [
 	cc.find_library('c_bio')
 ]
-if fs.is_file('/usr' / get_option('libdir') / 'java/core-all_classes.jar')
-  bootclasspath_dir = '/usr' / get_option('libdir') / 'java'
+if fs.is_file('@artStandalonePackageDir@' / get_option('libdir') / 'java/core-all_classes.jar')
+  bootclasspath_dir = '@artStandalonePackageDir@' / get_option('libdir') / 'java'
 elif fs.is_file('/usr/local' / get_option('libdir') / 'java/core-all_classes.jar')
   bootclasspath_dir = '/usr/local' / get_option('libdir') / 'java'
 elif fs.is_file(get_option('prefix') / get_option('libdir') / 'java/core-all_classes.jar')
