diff --git a/etebase_server/settings.py b/etebase_server/settings.py
index 9baf8d3..501d9f6 100644
--- a/etebase_server/settings.py
+++ b/etebase_server/settings.py
@@ -23,11 +22,6 @@
 # Quick-start development settings - unsuitable for production
 # See https://docs.djangoproject.com/en/3.0/howto/deployment/checklist/
 
-# SECURITY WARNING: keep the secret key used in production secret!
-# See secret.py for how this is generated; uses a file 'secret.txt' in the root
-# directory
-SECRET_FILE = os.path.join(BASE_DIR, "secret.txt")
-
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
@@ -143,7 +137,7 @@
 
     section = config["global"]
 
-    SECRET_FILE = section.get("secret_file", SECRET_FILE)
+    SECRET_FILE = section.get("secret_file", None)
     STATIC_ROOT = section.get("static_root", STATIC_ROOT)
     STATIC_URL = section.get("static_url", STATIC_URL)
     MEDIA_ROOT = section.get("media_root", MEDIA_ROOT)

