diff -rc Python-2.5.4/configure Python-2.5.4-new/configure
*** Python-2.5.4/configure	Sat Dec 13 15:13:52 2008
--- Python-2.5.4-new/configure	Fri Jul 24 16:38:05 2009
***************
*** 4534,4540 ****
  	    ;;
  	# is there any other compiler on Darwin besides gcc?
  	Darwin*)
! 	    BASECFLAGS="$BASECFLAGS -Wno-long-double -no-cpp-precomp -mno-fused-madd"
  	    if test "${enable_universalsdk}"; then
  		BASECFLAGS="-arch ppc -arch i386 -isysroot ${UNIVERSALSDK} ${BASECFLAGS}"
  	    fi
--- 4534,4540 ----
  	    ;;
  	# is there any other compiler on Darwin besides gcc?
  	Darwin*)
! 	    BASECFLAGS="$BASECFLAGS -no-cpp-precomp -mno-fused-madd"
  	    if test "${enable_universalsdk}"; then
  		BASECFLAGS="-arch ppc -arch i386 -isysroot ${UNIVERSALSDK} ${BASECFLAGS}"
  	    fi
