--- a/contrib/session-helper/meson.build
+++ b/contrib/session-helper/meson.build
@@ -21,7 +21,7 @@
     output : 'colord-session.service',
     configuration : con2,
     install: true,
-    install_dir: systemd.get_pkgconfig_variable('systemduserunitdir'),
+    install_dir: join_paths(get_option('prefix'), 'etc', 'systemd', 'user'),
   )
 endif
 
--- a/data/meson.build
+++ b/data/meson.build
@@ -8,13 +8,13 @@
 
 if get_option('enable-systemd')
   install_data('colord.conf',
-    install_dir: systemd.get_pkgconfig_variable('tmpfilesdir')
+    install_dir: join_paths(get_option('prefix'), 'lib', 'tmpfiles.d')
   )
 endif
 
 if get_option('enable-bash-completion')
   install_data('colormgr',
-    install_dir: bash_completion.get_pkgconfig_variable('completionsdir')
+    install_dir: join_paths(get_option('prefix'), 'etc', 'bash_completion.d')
   )
 endif
 
@@ -31,7 +31,7 @@
     output : 'colord.service',
     configuration : con2,
     install: true,
-    install_dir: systemd.get_pkgconfig_variable('systemdsystemunitdir'),
+    install_dir: join_paths(get_option('prefix'), 'etc', 'systemd', 'system'),
   )
 endif
 
@@ -50,5 +50,5 @@
   output : 'org.freedesktop.ColorManager.conf',
   configuration : con2,
   install: true,
-  install_dir: join_paths(get_option('datadir'), 'dbus-1', 'system.d') ,
+  install_dir: join_paths(get_option('prefix'), 'etc', 'dbus-1', 'system.d') ,
 )
--- a/meson.build
+++ b/meson.build
@@ -258,6 +258,3 @@
 subdir('policy')
 subdir('rules')
 subdir('src')
-
-meson.add_install_script('meson_post_install.sh',
-                         localstatedir, get_option('with-daemon-user'))
--- a/rules/meson.build
+++ b/rules/meson.build
@@ -9,5 +9,5 @@
     sensor_rules_in,
     '95-cd-devices.rules',
   ],
-  install_dir : join_paths(udev.get_pkgconfig_variable('udevdir'), 'rules.d')
+  install_dir : join_paths(get_option('prefix'), 'lib', 'udev', 'rules.d')
 )
--- a/src/plugins/meson.build
+++ b/src/plugins/meson.build
@@ -66,7 +66,7 @@
     install_dir: join_paths(get_option('libdir'), 'colord-plugins'),
     c_args : [
       cargs,
-      '-DCOLORD_SANE_BINARY="' + join_paths(get_option('libexecdir'), 'colord-sane' + '"'),
+      '-DCOLORD_SANE_BINARY="' + join_paths(get_option('prefix'), get_option('libexecdir'), 'colord-sane' + '"'),
     ],
     dependencies : [
       gio,
