From c2e40f7b4e7248e78808af9d9b36bffb7c8f881b Mon Sep 17 00:00:00 2001
From: Ethan Rooke <ethan@roo.ke>
Date: Thu, 20 Jun 2024 09:29:17 -0500
Subject: [PATCH 2/2] tests: remove dependence on parent folder name

---
 tests/test_console.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/test_console.py b/tests/test_console.py
index f8dc416..8167fad 100644
--- a/tests/test_console.py
+++ b/tests/test_console.py
@@ -153,7 +153,7 @@ _ROOT = Path.cwd()
             PathNode.from_path(_ROOT.joinpath("src/module.py")),
             [_ROOT.joinpath("alternative_src")],
             does_not_raise(),
-            Text("pytask/src/module.py"),
+            Text(f"{_ROOT.name}/src/module.py"),
             id="Common path found for PathNode not in 'paths' and 'paths'",
         ),
         pytest.param(
@@ -174,7 +174,7 @@ _ROOT = Path.cwd()
             PythonNode(name=_ROOT.as_posix() + "/task_a.py::task_a::a", value=None),
             [_ROOT],
             does_not_raise(),
-            Text("pytask/task_a.py::task_a::a"),
+            Text(f"{_ROOT.name}/task_a.py::task_a::a"),
             id="PythonNode with automatically assigned name",
         ),
     ],
-- 
2.45.1

