diff --git a/src/pulsecore/core-util.c b/src/pulsecore/core-util.c
index c383a61c0..3d0a68c2e 100644
--- a/src/pulsecore/core-util.c
+++ b/src/pulsecore/core-util.c
@@ -110,7 +110,7 @@
 #include <sys/personality.h>
 #endif
 
-#ifdef HAVE_CPUID_H
+#if defined(HAVE_CPUID_H) && !(defined(__APPLE__) && defined(__aarch64__))
 #include <cpuid.h>
 #endif
 
diff --git a/src/pulsecore/cpu-x86.c b/src/pulsecore/cpu-x86.c
index 317a0101e..d1eb95d2b 100644
--- a/src/pulsecore/cpu-x86.c
+++ b/src/pulsecore/cpu-x86.c
@@ -24,7 +24,7 @@
 
 #include <stdint.h>
 
-#ifdef HAVE_CPUID_H
+#if defined(HAVE_CPUID_H) && !(defined(__APPLE__) && defined(__aarch64__))
 #include <cpuid.h>
 #endif
 
