From 419258112b9d90d149ebbd5c657a36d8532b78a2 Mon Sep 17 00:00:00 2001
From: Andrew Childs <andrew.childs@bibo.com.ph>
Date: Tue, 19 Apr 2022 17:31:36 +0900
Subject: [PATCH 6/8] Only use version-script on GNU-ish linkers

---
 src/pulse/meson.build | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/pulse/meson.build b/src/pulse/meson.build
index c2128e087..a5e47867e 100644
--- a/src/pulse/meson.build
+++ b/src/pulse/meson.build
@@ -74,7 +74,11 @@ run_target('update-map-file',
   command : [ join_paths(meson.source_root(), 'scripts/generate-map-file.sh'), 'map-file',
               [ libpulse_headers, 'simple.h', join_paths(meson.build_root(), 'src', 'pulse', 'version.h') ] ])
 
-versioning_link_args = '-Wl,-version-script=' + join_paths(meson.source_root(), 'src', 'pulse', 'map-file')
+if meson.get_compiler('c').get_linker_id().startswith('ld.')
+  versioning_link_args = '-Wl,-version-script=' + join_paths(meson.source_root(), 'src', 'pulse', 'map-file')
+else
+  versioning_link_args = []
+endif
 
 libpulse = shared_library('pulse',
   libpulse_sources,
-- 
2.35.1

