diff --git a/src/meson.build b/src/meson.build
index 9efb561d8..d181f4867 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -179,6 +179,13 @@ if host_machine.system() == 'windows'
     'pulsecore/semaphore-win32.c',
     'pulsecore/thread-win32.c',
   ]
+elif host_machine.system() == 'darwin'
+  libpulsecommon_sources += [
+    'pulsecore/mutex-posix.c',
+    'pulsecore/poll-posix.c',
+    'pulsecore/semaphore-osx.c',
+    'pulsecore/thread-posix.c'
+  ]
 else
   libpulsecommon_sources += [
     'pulsecore/mutex-posix.c',
