diff --git a/meson.build b/meson.build
index 45df103f0..d57d13172 100644
--- a/meson.build
+++ b/meson.build
@@ -342,7 +342,7 @@ cdata.set('MESON_BUILD', 1)
 # On ELF systems we don't want the libraries to be unloaded since we don't clean them up properly,
 # so we request the nodelete flag to be enabled.
 # On other systems, we don't really know how to do that, but it's welcome if somebody can tell.
-nodelete_link_args = ['-Wl,-z,nodelete']
+nodelete_link_args = cc.get_supported_link_arguments(['-Wl,-z,nodelete'])
 
 # Code coverage
 
