diff --git a/powerdnsadmin/assets.py b/powerdnsadmin/assets.py
index b339c0e..6cf8087 100644
--- a/powerdnsadmin/assets.py
+++ b/powerdnsadmin/assets.py
@@ -50,7 +50,7 @@ class ModernBrowserCssFilter(Filter):
 css_login = Bundle(
     'node_modules/@fortawesome/fontawesome-free/css/all.css',
     'node_modules/admin-lte/dist/css/adminlte.css',
-    filters=(ModernBrowserCssFilter, 'rcssmin', 'cssrewrite'),
+    filters=(ModernBrowserCssFilter, 'rcssmin'),
     output='generated/login.css')
 
 js_login = Bundle(
@@ -66,7 +66,7 @@ css_main = Bundle(
     'node_modules/admin-lte/dist/css/adminlte.css',
     'node_modules/datatables.net-bs5/css/dataTables.bootstrap5.css',
     'custom/css/custom.css',
-    filters=(ModernBrowserCssFilter, 'rcssmin', 'cssrewrite'),
+    filters=(ModernBrowserCssFilter, 'rcssmin'),
     output='generated/main.css')
 
 js_main = Bundle(
-- 
2.43.0

