diff -ubr ccrtp-1.8.0-orig/src/ccrtp/sources.h ccrtp-1.8.0/src/ccrtp/sources.h
--- ccrtp-1.8.0-orig/src/ccrtp/sources.h	2010-04-18 20:51:49.000000000 +0200
+++ ccrtp-1.8.0/src/ccrtp/sources.h	2012-07-07 11:42:50.961179016 +0200
@@ -45,6 +45,7 @@
 #define CCXX_RTP_SOURCES_H_
 
 #include <string>
+#include <cstddef>
 #include <ccrtp/rtcppkt.h>
 
 #ifdef	CCXX_NAMESPACES
@@ -406,7 +407,7 @@
 	public:
 		typedef std::forward_iterator_tag iterator_category;
 		typedef Participant value_type;
-		typedef ptrdiff_t difference_type;
+		typedef std::ptrdiff_t difference_type;
 		typedef const Participant* pointer;
 		typedef const Participant& reference;
 
