diff --git a/ps/gscreator.cpp b/ps/gscreator.cpp
index 5b84e49..cbb7c25 100644
--- a/ps/gscreator.cpp
+++ b/ps/gscreator.cpp
@@ -101,7 +101,7 @@ static const char *epsprolog =
     "[ ] 0 setdash newpath false setoverprint false setstrokeadjust\n";
 
 static const char * gsargs_ps[] = {
-    "gs",
+    "@gs@",
     "-sDEVICE=png16m",
     "-sOutputFile=-",
     "-dSAFER",
@@ -120,7 +120,7 @@ static const char * gsargs_ps[] = {
 };
 
 static const char * gsargs_eps[] = {
-    "gs",
+    "@gs@",
     "-sDEVICE=png16m",
     "-sOutputFile=-",
     "-dSAFER",
