diff --git i/preflight.sh w/preflight.sh
index a2fcd1e..43b9c36 100755
--- i/preflight.sh
+++ w/preflight.sh
@@ -78,19 +78,6 @@ command -v "awk" > /dev/null 2>&1 &&
   test $(awk 'BEGIN{print(123)}') == 123 ||
     __preflightish_die "Failed to find usable 'awk' executable in \$PATH"
 
-# == Jujutsu ==
-__preflightish_require "0.37" jj --version --ignore-working-copy
-
-# == fzf ==
-__preflightish_require "0.67.0" fzf --version
-
-# == python3 ==
-__preflightish_require "3.9" python3 --version
-
-# == column ==
-command -v "column" > /dev/null 2>&1 ||
-  __preflightish_die "Failed to find the 'column' executable in \$PATH"
-
 # == Success ==
 [[ "${BASH_SOURCE[0]}" == "$0" ]] &&
   echo "  OK         All preflight.sh checks passed"
