diff --git a/data/org.gnome.Shell.Extensions.GSConnect.desktop.in b/data/org.gnome.Shell.Extensions.GSConnect.desktop.in
index 3fb887c3..e8cbe1bd 100644
--- a/data/org.gnome.Shell.Extensions.GSConnect.desktop.in
+++ b/data/org.gnome.Shell.Extensions.GSConnect.desktop.in
@@ -5,7 +5,7 @@
 [Desktop Entry]
 Type=Application
 Name=GSConnect
-Exec=gapplication launch org.gnome.Shell.Extensions.GSConnect %U
+Exec=@gapplication@ launch org.gnome.Shell.Extensions.GSConnect %U
 Terminal=false
 NoDisplay=true
 Icon=org.gnome.Shell.Extensions.GSConnect
diff --git a/src/extension.js b/src/extension.js
index 3fae443a..7aa19842 100644
--- a/src/extension.js
+++ b/src/extension.js
@@ -4,6 +4,8 @@
 
 'use strict';
 
+'@typelibPath@'.split(':').forEach(path => imports.gi.GIRepository.Repository.prepend_search_path(path));
+
 const Gio = imports.gi.Gio;
 const GObject = imports.gi.GObject;
 const Gtk = imports.gi.Gtk;
diff --git a/src/prefs.js b/src/prefs.js
index b8860c82..d6292606 100644
--- a/src/prefs.js
+++ b/src/prefs.js
@@ -4,6 +4,8 @@
 
 'use strict';
 
+'@typelibPath@'.split(':').forEach(path => imports.gi.GIRepository.Repository.prepend_search_path(path));
+
 const {Gio, GLib, Adw} = imports.gi;
 
 // Bootstrap
