diff --git a/wavefile/libsndfile.py b/wavefile/libsndfile.py
index 67f0a46..ce066ee 100644
--- a/wavefile/libsndfile.py
+++ b/wavefile/libsndfile.py
@@ -19,11 +19,11 @@ import numpy as np
 if sys.platform == "win32":
     dllName = 'libsndfile-1'
 elif "linux" in sys.platform:
-    dllName = 'libsndfile.so.1'
+    dllName = '@libsndfile@'
 elif "cygwin" in sys.platform:
     dllName = 'libsndfile-1.dll'
 elif "darwin" in sys.platform:
-    dllName = 'libsndfile.dylib'
+    dllName = '@libsndfile@'
 else:
     dllName = 'libsndfile'
 
