diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index 6acfc49..4ab8ab3 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -453,6 +453,7 @@ tasks.register<Copy>("includeJdk") {
     from(jdkHome)
     destinationDir = composeResources("jdk").get().asFile
 
+    dirPermissions { unix("rwx------") };
     fileTree(destinationDir).files.forEach { file ->
         file.setWritable(true, false)
         file.setReadable(true, false)