diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index 6acfc49..5de69b1 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -366,15 +366,8 @@ tasks.register<Exec>("packageFlatpak"){
     )
 }
 tasks.register<Zip>("zipDistributable"){
-    dependsOn("createDistributable", "setExecutablePermissions")
     group = "compose desktop"
 
-    val dir = distributable().destinationDir.get()
-    val packageName = distributable().packageName.get()
-
-    from(dir){ eachFile{ permissions{ unix("755") } } }
-    archiveBaseName.set(packageName)
-    destinationDirectory.set(dir.file("../").asFile)
 }
 
 afterEvaluate{