diff --git a/src/pstoeditoptions.h b/src/pstoeditoptions.h
index 7846883..0fb6a6f 100755
--- a/src/pstoeditoptions.h
+++ b/src/pstoeditoptions.h
@@ -453,7 +453,7 @@ private:
 		"Later versions of Ghostscript will probably support -dDELAYBIND again. "
 		"But also in that case the security risk remains. So be careful with what "
 		"files you process with pstoedit and Ghostscript.",
-		false),	//
+		true),	//
 #endif
 	pagetoextract		(true, "-page","page number",g_t,"extract a specific page: 0 means all pages",
 		"Select a single page from a multi-page PostScript or PDF file.",
