diff --git a/src/pty.c b/src/pty.c
index 3494104..1443f4a 100644
--- a/src/pty.c
+++ b/src/pty.c
@@ -299,7 +299,7 @@ static void setup_child(int master, struct winsize *ws)
 	if (ret)
 		log_warn("cannot reset blocked signals: %m");
 
-	for (i = 1; i < SIGUNUSED; ++i)
+	for (i = 1; i < SIGSYS; ++i)
 		signal(i, SIG_DFL);
 
 	ret = grantpt(master);
diff --git a/src/uterm_vt.c b/src/uterm_vt.c
index af377f5..fbe9e76 100644
--- a/src/uterm_vt.c
+++ b/src/uterm_vt.c
@@ -40,6 +40,7 @@
 #include <sys/ioctl.h>
 #include <sys/signalfd.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <termios.h>
 #include <time.h>
 #include <unistd.h>
