compatibility with newer spdlog/fmt versions

diff --git a/plugins/module_identification/src/api/module_identification.cpp b/plugins/module_identification/src/api/module_identification.cpp
index 808875e8a68..3cf3cf1fb8b 100644
--- a/plugins/module_identification/src/api/module_identification.cpp
+++ b/plugins/module_identification/src/api/module_identification.cpp
@@ -406,7 +406,7 @@ namespace hal
 
                 const u32 num_threads = std::min(config.m_max_thread_count, std::thread::hardware_concurrency() - 1);
 
-                log_info("module_identification", "running with {} threads and {} multithreading priority", num_threads, config.m_multithreading_priority);
+                log_info("module_identification", "running with {} threads and {} multithreading priority", num_threads, fmt::underlying(config.m_multithreading_priority));
 
                 auto stats = Statistics();
 
diff --git a/plugins/z3_utils/src/simplification.cpp b/plugins/z3_utils/src/simplification.cpp
index d64cab665f2..561fa7f1a23 100644
--- a/plugins/z3_utils/src/simplification.cpp
+++ b/plugins/z3_utils/src/simplification.cpp
@@ -617,7 +617,7 @@ namespace hal
                     return false;
 
                 default: {
-                    log_error("z3_utils", "commutative check not implemeted for type {}!", t);
+                    log_error("z3_utils", "commutative check not implemeted for type {}!", fmt::underlying(t));
                     return false;
                 }
             }
