diff --git a/windmill-worker/nsjail/run.go.config.proto b/windmill-worker/nsjail/run.go.config.proto
index 3af548d1..39ff4da7 100644
--- a/windmill-worker/nsjail/run.go.config.proto
+++ b/windmill-worker/nsjail/run.go.config.proto
@@ -25,6 +25,13 @@ mount {
 }
 
 mount {
+	src: "/nix/store"
+	dst: "/nix/store"
+	is_bind: true
+}
+
+mount {
+    mandatory: false
     src: "/lib"
     dst: "/lib"
 	is_bind: true
@@ -32,6 +39,7 @@ mount {
 
 
 mount {
+    mandatory: false
     src: "/lib64"
     dst: "/lib64"
 	is_bind: true
@@ -39,6 +47,7 @@ mount {
 
 
 mount {
+    mandatory: false
     src: "/usr"
     dst: "/usr"
 	is_bind: true
