diff --git a/src/main/services/linux.service.ts b/src/main/services/linux.service.ts
index 7f92bed..5f12896 100644
--- a/src/main/services/linux.service.ts
+++ b/src/main/services/linux.service.ts
@@ -129,12 +129,7 @@ export class LinuxService {
         return fs.pathExistsSync(protonPath) && fs.pathExistsSync(winePath);
     }
 
-    public async getWinePath(): Promise<string> {
-        if (await this.isNixOS()) {
-            // Use system wine for nixos
-            return "wine";
-        }
-
+    public getWinePath(): string {
         if (!this.staticConfig.has("proton-folder")) {
             throw new Error("proton-folder variable not set");
         }
diff --git a/src/main/services/mods/bs-mods-manager.service.ts b/src/main/services/mods/bs-mods-manager.service.ts
index c842fda..60cc190 100644
--- a/src/main/services/mods/bs-mods-manager.service.ts
+++ b/src/main/services/mods/bs-mods-manager.service.ts
@@ -162,12 +162,15 @@ export class BsModsManagerService {
         let winePath: string = "";
         if (process.platform === "linux") {
             const { error: winePathError, result: winePathResult } =
-                await tryit(async () => this.linuxService.getWinePath());
+                tryit(() => this.linuxService.getWinePath());
             if (winePathError) {
                 log.error(winePathError);
                 return false;
             }
-            winePath = `"${winePathResult}"`;
+
+            winePath = await this.linuxService.isNixOS()
+                ? `steam-run "${winePathResult}"`
+                : `"${winePathResult}"`;
 
             const winePrefix = this.linuxService.getWinePrefixPath();
             if (!winePrefix) {
