diff --git a/.cargo/config b/.cargo/config
index 72652ad2f..b21c6aa7b 100644
--- a/.cargo/config
+++ b/.cargo/config
@@ -1,3 +1,113 @@
 [target.x86_64-pc-windows-msvc]
 # Link the C runtime statically ; https://github.com/paritytech/parity/issues/6643
 rustflags = ["-Ctarget-feature=+crt-static"]
+
+[source."https://github.com/alexcrichton/mio-named-pipes"]
+git = "https://github.com/alexcrichton/mio-named-pipes"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/js-dist-paritytech/parity-master-1-10-shell.git"]
+git = "https://github.com/js-dist-paritytech/parity-master-1-10-shell.git"
+rev = "bd25b41cd642c6b822d820dded3aa601a29aa079"
+replace-with = "vendored-sources"
+
+[source."https://github.com/js-dist-paritytech/parity-master-1-10-wallet.git"]
+git = "https://github.com/js-dist-paritytech/parity-master-1-10-wallet.git"
+rev = "4b6f112412716cd05123d32eeb7fda448288a6c6"
+replace-with = "vendored-sources"
+
+[source."https://github.com/nikvolf/parity-tokio-ipc"]
+git = "https://github.com/nikvolf/parity-tokio-ipc"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/nikvolf/tokio-named-pipes"]
+git = "https://github.com/nikvolf/tokio-named-pipes"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/parity-js/dapp-wallet.git"]
+git = "https://github.com/parity-js/dapp-wallet.git"
+rev = "65deb02e7c007a0fd8aab0c089c93e3fd1de6f87"
+replace-with = "vendored-sources"
+
+[source."https://github.com/parity-js/shell.git"]
+git = "https://github.com/parity-js/shell.git"
+rev = "eecaadcb9e421bce31e91680d14a20bbd38f92a2"
+replace-with = "vendored-sources"
+
+[source."https://github.com/paritytech/bn"]
+git = "https://github.com/paritytech/bn"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/paritytech/hidapi-rs"]
+git = "https://github.com/paritytech/hidapi-rs"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/paritytech/hyper"]
+git = "https://github.com/paritytech/hyper"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/paritytech/jsonrpc.git"]
+git = "https://github.com/paritytech/jsonrpc.git"
+branch = "parity-1.10"
+replace-with = "vendored-sources"
+
+[source."https://github.com/paritytech/libusb-rs"]
+git = "https://github.com/paritytech/libusb-rs"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/paritytech/libusb-sys"]
+git = "https://github.com/paritytech/libusb-sys"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/paritytech/rust-ctrlc.git"]
+git = "https://github.com/paritytech/rust-ctrlc.git"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/paritytech/rust-rocksdb"]
+git = "https://github.com/paritytech/rust-rocksdb"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/paritytech/rust-secp256k1"]
+git = "https://github.com/paritytech/rust-secp256k1"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/paritytech/rust-snappy"]
+git = "https://github.com/paritytech/rust-snappy"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/paritytech/trezor-sys"]
+git = "https://github.com/paritytech/trezor-sys"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/paritytech/wasm-utils"]
+git = "https://github.com/paritytech/wasm-utils"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/paritytech/wasmi"]
+git = "https://github.com/paritytech/wasmi"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/tailhook/rotor"]
+git = "https://github.com/tailhook/rotor"
+branch = "master"
+replace-with = "vendored-sources"
+
+[source."https://github.com/tomusdrw/ws-rs"]
+git = "https://github.com/tomusdrw/ws-rs"
+branch = "master"
+replace-with = "vendored-sources"
