diff --git a/plug-ins/icc_examin/icc_examin/icc_helfer_fltk.cpp b/plug-ins/icc_examin/icc_examin/icc_helfer_fltk.cpp
index b507454..85f4441 100644
--- a/plug-ins/icc_examin/icc_examin/icc_helfer_fltk.cpp
+++ b/plug-ins/icc_examin/icc_examin/icc_helfer_fltk.cpp
@@ -226,7 +226,7 @@ namespace icc_examin_ns {
 
   int  awake(void)
   {
-    Fl::awake(0);
+    Fl::awake((void *)0);
     return 0;
   }
   int  leerWait(void) { return 0; }
