diff --git a/CMakeLists.txt b/CMakeLists.txt
index dfb182f..5adaaa5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -138,7 +138,7 @@ STRING(REPLACE "/" "\\/" ESCAPEDPREFIX ${PREFIX})
 # Note that for MacOS this needs to be revised 
 # for the @OSX_ICC...@ variables
 ADD_CUSTOM_COMMAND(
-   OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/gimprc
+   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/gimprc
    COMMAND  sed -e s/\@platform\@//g
 		-e s/\@prefix\@/${ESCAPEDPREFIX}/g 
 	        -e s/\@exec_prefix\@/\${prefix}/g
@@ -155,8 +155,8 @@ ADD_CUSTOM_COMMAND(
 
 ADD_CUSTOM_TARGET(RCFile ALL echo
       ""
-      DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/user_install 
-              ${CMAKE_CURRENT_SOURCE_DIR}/gimprc
+      DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/user_install 
+              ${CMAKE_CURRENT_BINARY_DIR}/gimprc
    )
 
 
