diff --git a/lib/stdint.in.h b/lib/stdint.in.h
index 997e406..e0827f5 100644
--- a/lib/stdint.in.h
+++ b/lib/stdint.in.h
@@ -53,7 +53,8 @@
      in <inttypes.h> would reinclude us, skipping our contents because
      _GL_STDINT_H is defined.
      The include_next requires a split double-inclusion guard.  */
-# @INCLUDE_NEXT@ @NEXT_STDINT_H@
+# include <inttypes.h>
+// # @INCLUDE_NEXT@ @NEXT_STDINT_H@
 #endif
 
 #if ! defined _GL_STDINT_H && ! defined _GL_JUST_INCLUDE_SYSTEM_STDINT_H
