--- a/client/scripts/linux	2010-09-15 00:49:48.000000000 +0200
+++ b/client/scripts/linux	2011-04-01 16:08:10.984372269 +0200
@@ -133,9 +133,7 @@
            [ "$current_hostname" = '(none)' ] ||
            [ "$current_hostname" = 'localhost' ] ||
            [ "$current_hostname" = "$old_host_name" ]; then
-           if [ "$new_host_name" != "$old_host_name" ]; then
-               hostname "$new_host_name"
-           fi
+           hostname "$new_host_name"
         fi
     fi
