diff -ubr '--exclude=*.o' esniper-2-27-0-orig/http.c esniper-2-27-0-patched/http.c
--- esniper-2-27-0-orig/http.c	2012-02-06 22:04:06.000000000 +0100
+++ esniper-2-27-0-patched/http.c	2012-07-27 10:54:20.893054646 +0200
@@ -200,6 +200,9 @@
 int
 initCurlStuff(void)
 {
+	/* Path to OpenSSL bundle file. */
+	const char *ssl_capath=NULL;
+
 	/* list for custom headers */
 	struct curl_slist *slist=NULL;
 
@@ -241,6 +244,12 @@
 	if ((curlrc = curl_easy_setopt(easyhandle, CURLOPT_COOKIEFILE, "")))
 		return initCurlStuffFailed();
 
+	/* If the environment variable CURL_CA_BUNDLE is set, pass through its
+	 * contents to curl. */
+	if ((ssl_capath = getenv("CURL_CA_BUNDLE")))
+		if ((curlrc = curl_easy_setopt(easyhandle, CURLOPT_CAINFO, ssl_capath)))
+			return initCurlStuffFailed();
+
 	slist = curl_slist_append(slist, "Accept: text/*");
 	slist = curl_slist_append(slist, "Accept-Language: en");
 	slist = curl_slist_append(slist, "Accept-Charset: iso-8859-1,*,utf-8");
