From 716be9d1c5ff2fbf7cd9d046ea393920a8690ab3 Mon Sep 17 00:00:00 2001
From: Daniel Hill <daniel@gluo.nz>
Date: Mon, 21 Apr 2025 02:33:18 +1200
Subject: [PATCH] use system libuv for mingw-w64 builds

---
 Makefile        | 10 +++++-----
 build_config.rb |  4 +++-
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index 7bc3788..6506120 100644
--- a/Makefile
+++ b/Makefile
@@ -36,15 +36,15 @@ osx: deps/libuv.a
 		./deps/libuv/.libs/libuv.a  -lm -framework OpenGL -lpthread
 	$(CC) test-libversion.c deps/pugl/build/libpugl-0.a -ldl -o zest -framework OpenGL -framework AppKit -lpthread -I deps/pugl -std=gnu99
 
-windows: buildpuglwin deps/libuv-win.a
+windows: buildpuglwin
 	cd deps/nanovg/src   && $(CC) -mstackrealign nanovg.c -c
 	$(AR) rc deps/libnanovg.a deps/nanovg/src/*.o
-	cd src/osc-bridge    && CFLAGS="-mstackrealign -I ../../deps/libuv/include " make lib
-	cd mruby             && WINDOWS=1 MRUBY_CONFIG=../build_config.rb rake
-	$(CC) -mstackrealign -shared -o libzest.dll -static-libgcc `find mruby/build/w64 -type f | grep -e "\.o$$" | grep -v bin` \
+	cd src/osc-bridge    && CFLAGS="-mstackrealign " make lib
+	cd mruby             && WINDOWS=1 MRUBY_CONFIG=../build_config.rb rake -j
+	$(CC) -mstackrealign -shared -o libzest.dll -static-libstdc++ -static-libgcc `find mruby/build/w64 -type f | grep -e "\.o$$" | grep -v bin` \
         ./deps/libnanovg.a \
         src/osc-bridge/libosc-bridge.a \
-        ./deps/libuv-win.a \
+		`$(PKG_CONFIG) --libs --static libuv` \
         -lm -lpthread -lws2_32 -lkernel32 -lpsapi -luserenv -liphlpapi -lglu32 -lgdi32 -lopengl32
 	$(CC) -mstackrealign -DWIN32 test-libversion.c deps/pugl/build/libpugl-0.a -o zest.exe -lpthread -I deps/pugl -std=c99 -lws2_32 -lkernel32 -lpsapi -luserenv -liphlpapi -lglu32 -lgdi32 -lopengl32
 
diff --git a/build_config.rb b/build_config.rb
index e72b5ab..e9518bf 100644
--- a/build_config.rb
+++ b/build_config.rb
@@ -6,6 +6,8 @@ windows = ENV.include? "WINDOWS"
 mac     = ENV['OS'] == "Mac"
 linux   = false if windows || mac
 
+pkg_config = ENV['PKG_CONFIG'] || "pkg-config"
+
 RUBY_PLATFORM = "mingw" if windows
 
 if(windows)
@@ -156,7 +158,7 @@ build_type.new(build_name) do |conf|
         end
         linker.flags_after_libraries  << "-lpthread -ldl -lm"
       else
-        linker.flags_after_libraries  << "#{`pwd`.strip}/../deps/libuv-win.a"
+        linker.flags_after_libraries << `#{pkg_config} --libs libuv`.strip
         linker.flags_after_libraries  << "-lws2_32 -lkernel32 -lpsapi -luserenv -liphlpapi"
         linker.flags_after_libraries  << "-lglu32 -lgdi32 -lopengl32"
       end
-- 
2.48.1

