--- ./tests/__init__.py	1980-01-02 00:00:00.000000000 +0000
+++ ./tests/__init__.py	1980-01-02 00:00:00.000000000 +0000
@@ -301,6 +301,7 @@
             self.assertEqual(id(base_p.thread_pool), id(p.thread_pool))
             self.assertEqual(id(base_p._session_pool), id(p._session_pool))
 
+    @unittest.skip("no network connection inside the Nix sandbox")
     def test_close(self):
         proc = psutil.Process()
         ip_addresses = {info[4][0] for info in socket.getaddrinfo(
@@ -1257,6 +1258,7 @@
         )
 
 
+@unittest.skip("no network connection inside the Nix sandbox")
 class EWSTest(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
