diff -rc Catalyst-Engine-HTTP-Prefork-0.50-orig/lib/Catalyst/Engine/HTTP/Prefork/Handler.pm Catalyst-Engine-HTTP-Prefork-0.50/lib/Catalyst/Engine/HTTP/Prefork/Handler.pm
*** Catalyst-Engine-HTTP-Prefork-0.50-orig/lib/Catalyst/Engine/HTTP/Prefork/Handler.pm	2008-03-14 18:23:47.000000000 +0100
--- Catalyst-Engine-HTTP-Prefork-0.50/lib/Catalyst/Engine/HTTP/Prefork/Handler.pm	2009-03-11 14:18:40.000000000 +0100
***************
*** 199,206 ****
      
      if ( $self->{_chunked_res} ) {
          if ( !$self->{_chunked_done} ) {
!             # Write the final '0' chunk
!             syswrite STDOUT, "0$CRLF";
          }
          
          delete $self->{_chunked_res};
--- 199,207 ----
      
      if ( $self->{_chunked_res} ) {
          if ( !$self->{_chunked_done} ) {
!             # Write the final '0' chunk and the CRLF that terminates
!             # the chunked body.
!             syswrite STDOUT, "0$CRLF$CRLF";
          }
          
          delete $self->{_chunked_res};
