diff --git a/lib/mhamain.pl b/lib/mhamain.pl
index 80980a2..c1259ce 100644
--- a/lib/mhamain.pl
+++ b/lib/mhamain.pl
@@ -1562,7 +1562,7 @@ sub signal_catch {
 ##
 sub defineIndex2MsgId {
     no warnings qw(deprecated);
-    if (!defined(%Index2MsgId)) {
+    unless (%Index2MsgId) {
 	foreach (keys %MsgId) {
 	    $Index2MsgId{$MsgId{$_}} = $_;
 	}
diff --git a/lib/mhopt.pl b/lib/mhopt.pl
index 02fb05e..939109b 100644
--- a/lib/mhopt.pl
+++ b/lib/mhopt.pl
@@ -865,7 +865,7 @@ sub update_data_1_to_2 {
 sub update_data_2_1_to_later {
     no warnings qw(deprecated);
     # we can preserve filter arguments
-    if (defined(%main::MIMEFiltersArgs)) {
+    if (%main::MIMEFiltersArgs) {
 	warn qq/         preserving MIMEARGS...\n/;
 	%readmail::MIMEFiltersArgs = %main::MIMEFiltersArgs;
 	$IsDefault{'MIMEARGS'} = 0;
