From c0297740c3112878c28cceb2dd7d3924dff1079c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gutyina=20Gerg=C5=91?= <gutyina.gergo.2@gmail.com>
Date: Wed, 3 Sep 2025 22:40:59 +0200
Subject: [PATCH] enable vendored feature for utoipa-swagger-ui

---
 Cargo.lock | 7 +++++++
 Cargo.toml | 2 +-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/Cargo.lock b/Cargo.lock
index 08c5aa5c..db45b6cd 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -5972,9 +5972,16 @@ dependencies = [
  "serde_json",
  "url",
  "utoipa",
+ "utoipa-swagger-ui-vendored",
  "zip",
 ]
 
+[[package]]
+name = "utoipa-swagger-ui-vendored"
+version = "0.1.2"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "e2eebbbfe4093922c2b6734d7c679ebfebd704a0d7e56dfcb0d05818ce28977d"
+
 [[package]]
 name = "uuid"
 version = "1.16.0"
diff --git a/Cargo.toml b/Cargo.toml
index f1ef5502..3d52117f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -115,7 +115,7 @@ tokio = { version = "1", features = ["full"] }
 tokio-postgres = { version = "0.7.10", features = ["with-chrono-0_4"] }
 tokio-postgres-rustls = { version = "0.13", features = [] }
 utoipa = { version = "5", features = ["actix_extras", "chrono", "openapi_extensions"] }
-utoipa-swagger-ui = { version = "9.0.0", features = ["actix-web"] }
+utoipa-swagger-ui = { version = "9.0.0", features = ["actix-web", "vendored"] }
 validator = { version = "0.20.0", features = ["derive"] }
 webauthn-rs = { version = "0.5", features = [
     "danger-allow-state-serialisation", "danger-credential-internals"
-- 
2.50.1

