From f8ad40d368a0732732948b965d3ea9b57b0bfd5f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gutyina=20Gerg=C5=91?= <gutyina.gergo.2@gmail.com>
Date: Mon, 1 Sep 2025 23:39:25 +0200
Subject: [PATCH 2/2] enable vendored feature for utoipa-swagger-ui

---
 Cargo.lock | 7 +++++++
 Cargo.toml | 2 +-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/Cargo.lock b/Cargo.lock
index c9d6fa9a..dcdcf124 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -6340,9 +6340,16 @@ dependencies = [
  "serde_json",
  "url",
  "utoipa",
+ "utoipa-swagger-ui-vendored",
  "zip",
 ]
 
+[[package]]
+name = "utoipa-swagger-ui-vendored"
+version = "0.1.2"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "e2eebbbfe4093922c2b6734d7c679ebfebd704a0d7e56dfcb0d05818ce28977d"
+
 [[package]]
 name = "uuid"
 version = "1.15.1"
diff --git a/Cargo.toml b/Cargo.toml
index d5844524..05892391 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -110,7 +110,7 @@ tracing = { version = "0.1", features = ["attributes"] }
 tracing-subscriber = { version = "0.3", features = ["env-filter", "json", "tracing"] }
 tokio = { version = "1", features = ["full"] }
 utoipa = { version = "5", features = ["actix_extras", "chrono", "openapi_extensions"] }
-utoipa-swagger-ui = { version = "9.0.0", features = ["actix-web"] }
+utoipa-swagger-ui = { version = "9.0.0", features = ["actix-web", "vendored"] }
 validator = { version = "0.20.0", features = ["derive"] }
 webauthn-rs = { version = "0.5", features = [
     "danger-allow-state-serialisation", "danger-credential-internals"
-- 
2.50.1

