diff --git a/src/calibre/gui2/__init__.py b/src/calibre/gui2/__init__.py
index 8b78c6f8cc..19118dd0ce 100644
--- a/src/calibre/gui2/__init__.py
+++ b/src/calibre/gui2/__init__.py
@@ -969,7 +969,7 @@ def file_icon_provider():
     from calibre.gui2.win_file_dialogs import is_ok as has_windows_file_dialog_helper
     has_windows_file_dialog_helper = has_windows_file_dialog_helper()
 has_linux_file_dialog_helper = False
-if not iswindows and not ismacos and 'CALIBRE_NO_NATIVE_FILEDIALOGS' not in os.environ and getattr(sys, 'frozen', False):
+if not iswindows and not ismacos and 'CALIBRE_NO_NATIVE_FILEDIALOGS' not in os.environ:
     has_linux_file_dialog_helper = check_for_linux_native_dialogs()

 if has_windows_file_dialog_helper:
