--- a/raylib/build.py
+++ b/raylib/build.py
@@ -158,7 +158,7 @@ def build_unix():

     if platform.system() == "Darwin":
         print("BUILDING FOR MAC")
-        extra_link_args = [get_the_lib_path() + '/libraylib.a', '-framework', 'OpenGL', '-framework', 'Cocoa',
+        extra_link_args = get_lib_flags() + ['-framework', 'OpenGL', '-framework', 'Cocoa',
                            '-framework', 'IOKit', '-framework', 'CoreFoundation', '-framework',
                            'CoreVideo']
         libraries = []
