diff --git a/tool/tsh/resolve_default_addr_test.go b/tool/tsh/resolve_default_addr_test.go
index d5976f156..aec5199aa 100644
--- a/tool/tsh/resolve_default_addr_test.go
+++ b/tool/tsh/resolve_default_addr_test.go
@@ -237,7 +237,7 @@ func TestResolveDefaultAddrTimeoutBeforeAllRacersLaunched(t *testing.T) {

 	blockingHandler, doneCh := newWaitForeverHandler()

-	servers := make([]*httptest.Server, 1000)
+	servers := make([]*httptest.Server, 100)
 	for i := 0; i < len(servers); i++ {
 		servers[i] = makeTestServer(t, blockingHandler)
 	}
