From 61dd07a27b229d33ceb386e9f909773d5885999c Mon Sep 17 00:00:00 2001
From: Thomas Tuegel <ttuegel@gmail.com>
Date: Tue, 27 Oct 2015 17:14:36 -0500
Subject: [PATCH] default icon theme

---
 src/kicontheme.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/kicontheme.cpp b/src/kicontheme.cpp
index d0ab4b9..dce7773 100644
--- a/src/kicontheme.cpp
+++ b/src/kicontheme.cpp
@@ -542,7 +542,7 @@ void KIconTheme::reconfigure()
 // static
 QString KIconTheme::defaultThemeName()
 {
-    return QLatin1String("oxygen");
+    return QLatin1String("breeze");
 }
 
 void KIconTheme::assignIconsToContextMenu(ContextMenus type,
-- 
2.6.2

