diff --git a/test.sh b/test.sh
index f017b9a..027031e 100755
--- a/test.sh
+++ b/test.sh
@@ -135,7 +135,7 @@ if [ $os = "windows" ]; then
   conan install $src_dir -s compiler="Visual Studio" -s compiler.version=16 -sbuild_type=Debug -g cmake_paths
   cmake_gen="$cmake_gen -DCMAKE_TOOLCHAIN_FILE=conan_paths.cmake -DCMAKE_INSTALL_PREFIX=$src_dir/install"
 fi
-cmake .. $cmake_gen
+cmake .. $cmake_gen -DDrogon_DIR=$out/lib/cmake/Drogon -DTrantor_DIR=$out/lib/cmake/Trantor
 
 if [ $? -ne 0 ]; then
     echo "Failed to run CMake for example project"
