From 3c8fe6e2c595ee019408249c364b3019b6c31a8a Mon Sep 17 00:00:00 2001
From: Mike DePaulo <mikedep333@gmail.com>
Date: Fri, 15 May 2015 22:22:13 -0400
Subject: [PATCH] Reintroduce ssh_forward_listen() (Fixes: #194)

---
 src/channels.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/channels.c b/src/channels.c
index 7a4e71f..db5f83a 100644
--- a/src/channels.c
+++ b/src/channels.c
@@ -2206,6 +2206,11 @@ error:
 }
 
 /* DEPRECATED */
+int ssh_forward_listen(ssh_session session, const char *address, int port, int *bound_port) {
+  return ssh_channel_listen_forward(session, address, port, bound_port);
+}
+
+/* DEPRECATED */
 ssh_channel ssh_forward_accept(ssh_session session, int timeout_ms) {
   return ssh_channel_accept(session, SSH_CHANNEL_FORWARDED_TCPIP, timeout_ms, NULL);
 }
-- 
2.1.4

