--- a/syncthing_gtk/configuration.py
+++ b/syncthing_gtk/configuration.py
@@ -30,7 +30,7 @@
 		"autokill_daemon"			: (int, 2),	# 0 - never kill, 1 - always kill, 2 - ask
 		"daemon_priority"			: (int, 0), # uses nice values
 		"max_cpus"					: (int, 0), # 0 for all cpus
-		"syncthing_binary"			: (str, "/usr/bin/syncthing"),
+		"syncthing_binary"			: (str, "@syncthing@"),
 		"syncthing_arguments"		: (str, ""),
 		"minimize_on_start"			: (bool, False),
 		"folder_as_path"			: (bool, True),
--- a/syncthing_gtk/tools.py
+++ b/syncthing_gtk/tools.py
@@ -303,7 +303,7 @@
 			return False
 		# signal 0 doesn't kill anything, but killall exits with 1 if
 		# named process is not found
-		p = Popen(["killall", "-u", os.environ["USER"], "-q", "-s", "0", "syncthing"])
+		p = Popen(["@killall@", "-u", os.environ["USER"], "-q", "-s", "0", "syncthing"])
 		p.communicate()
 		return p.returncode == 0
 	else:
