diff -ru -x '*~' linux-4.9.31-orig/tools/perf/util/annotate.c linux-4.9.31/tools/perf/util/annotate.c
--- linux-4.9.31-orig/tools/perf/util/annotate.c	2017-06-07 12:08:04.000000000 +0200
+++ linux-4.9.31/tools/perf/util/annotate.c	2017-06-12 13:10:08.811079574 +0200
@@ -1350,7 +1350,7 @@
 		 "%s %s%s --start-address=0x%016" PRIx64
 		 " --stop-address=0x%016" PRIx64
 		 " -l -d %s %s -C %s 2>/dev/null|grep -v %s|expand",
-		 objdump_path ? objdump_path : "objdump",
+		 objdump_path ? objdump_path : OBJDUMP_PATH,
 		 disassembler_style ? "-M " : "",
 		 disassembler_style ? disassembler_style : "",
 		 map__rip_2objdump(map, sym->start),
