--- perf/config/utilities.mak.orig	2014-01-25 14:55:32.573320370 +0000
+++ perf/config/utilities.mak	2014-01-25 15:13:34.174337760 +0000
@@ -186,9 +186,14 @@
 endif
 TRY_CC_MSG=echo "    CHK $(3)" 1>&2;
 
+define newline
+
+
+endef
+
 try-cc = $(shell sh -c						  \
 	'TMP="$(OUTPUT)$(TMPOUT).$$$$";				  \
 	 $(TRY_CC_MSG)						  \
-	 echo "$(1)" |						  \
+	 echo -e "$(subst $(newline),\\n,$(1))" | tee _test.c |   \
 	 $(CC) -x c - $(2) -o "$$TMP" $(TRY_CC_OUTPUT) && echo y; \
 	 rm -f "$$TMP"')
