Index: linux-2.6.24-rc3/security/security.c
===================================================================
RCS file: /ext1/sysadm/transparent/repository/linux-2.6.24-rc3/security/security.c,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -p -r1.1 -r1.2
--- linux-2.6.24-rc3/security/security.c	21 Nov 2007 13:03:11 -0000	1.1
+++ linux-2.6.24-rc3/security/security.c	21 Nov 2007 13:07:55 -0000	1.2
@@ -409,6 +409,7 @@ int security_inode_permission(struct ino
 		return 0;
 	return security_ops->inode_permission(inode, mask, nd);
 }
+EXPORT_SYMBOL(security_inode_permission);
 
 int security_inode_setattr(struct dentry *dentry, struct iattr *attr)
 {
