diff --git a/WORKSPACE b/WORKSPACE
index 696cc7ef..55a5bb8a 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -81,17 +81,6 @@ load("@com_github_google_rules_install//:setup.bzl", "install_rules_setup")
 
 install_rules_setup()
 
-# Need to load before rules_flex/rules_bison to make sure
-# win_flex_bison is the chosen toolchain on Windows
-load("//bazel:win_flex_bison.bzl", "win_flex_configure")
-
-win_flex_configure(
-    name = "win_flex_bison",
-    sha256 = "095cf65cb3f12ee5888022f93109acbe6264e5f18f6ffce0bda77feb31b65bd8",
-    # bison 3.3.2, flex 2.6.4
-    url = "https://github.com/lexxmark/winflexbison/releases/download/v2.5.18/win_flex_bison-2.5.18.zip",
-)
-
 http_archive(
     name = "rules_m4",
     sha256 = "c67fa9891bb19e9e6c1050003ba648d35383b8cb3c9572f397ad24040fb7f0eb",
