diff --git a/client/Application.cpp b/client/Application.cpp
index fce9eee..cec84ae 100644
--- a/client/Application.cpp
+++ b/client/Application.cpp
@@ -54,6 +54,7 @@
 #include <QtGui/QFileOpenEvent>
 #include <QtNetwork/QNetworkProxy>
 #include <QtNetwork/QSslConfiguration>
+#include <QtWidgets/QAction>
 #include <QtWidgets/QMessageBox>
 #include <QtWidgets/QProgressBar>
 #include <QtWidgets/QProgressDialog>
diff --git a/client/MainWindow.cpp b/client/MainWindow.cpp
index 06d561e..b4aa0d5 100644
--- a/client/MainWindow.cpp
+++ b/client/MainWindow.cpp
@@ -44,6 +44,8 @@
 #include <QtPrintSupport/QPrinter>
 #include <QtPrintSupport/QPrinterInfo>
 #include <QtPrintSupport/QPrintPreviewDialog>
+#include <QtWidgets/QActionGroup>
+#include <QtWidgets/QButtonGroup>
 #include <QtWidgets/QCompleter>
 #include <QtWidgets/QMessageBox>
 
diff --git a/client/MainWindow.h b/client/MainWindow.h
index b9515dd..661418a 100644
--- a/client/MainWindow.h
+++ b/client/MainWindow.h
@@ -24,6 +24,7 @@
 #include <QtCore/QStringList>
 
 class DigiDoc;
+class QActionGroup;
 class QPrinter;
 
 class MainWindow: public QWidget, private Ui::MainWindow
diff --git a/crypto/MainWindow.cpp b/crypto/MainWindow.cpp
index c3d3115..f12d984 100644
--- a/crypto/MainWindow.cpp
+++ b/crypto/MainWindow.cpp
@@ -35,6 +35,8 @@
 #include <QtGui/QDesktopServices>
 #include <QtGui/QDragEnterEvent>
 #include <QtNetwork/QSslKey>
+#include <QtWidgets/QActionGroup>
+#include <QtWidgets/QButtonGroup>
 #include <QtWidgets/QMessageBox>
 #include <QtWidgets/QProgressBar>
 #include <QtWidgets/QProgressDialog>
diff --git a/crypto/MainWindow.h b/crypto/MainWindow.h
index 16dbec4..01ec99f 100644
--- a/crypto/MainWindow.h
+++ b/crypto/MainWindow.h
@@ -21,6 +21,7 @@
 
 #include "ui_MainWindow.h"
 
+class QActionGroup;
 class CKey;
 class CryptoDoc;
 
