diff --git a/src/gui/curses/CMakeLists.txt b/src/gui/curses/CMakeLists.txt
index 325c611..a8927bc 100644
--- a/src/gui/curses/CMakeLists.txt
+++ b/src/gui/curses/CMakeLists.txt
@@ -53,9 +53,7 @@ IF(${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
   ENDIF(HAVE_BACKTRACE)
 ENDIF(${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
 
-IF(${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
-  LIST(APPEND EXTRA_LIBS "pthread")
-ENDIF(${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
+LIST(APPEND EXTRA_LIBS "pthread")
 
 IF(ICONV_LIBRARY)
   LIST(APPEND EXTRA_LIBS ${ICONV_LIBRARY})

