diff --git a/frigate/detectors/detection_runners.py b/frigate/detectors/detection_runners.py
index fcbb41e66..5b4d3f310 100644
--- a/frigate/detectors/detection_runners.py
+++ b/frigate/detectors/detection_runners.py
@@ -603,6 +603,7 @@ def get_optimized_runner(
             ),
             providers=providers,
             provider_options=options,
+            disabled_optimizers=["SimplifiedLayerNormFusion"],
         ),
         model_type=model_type,
     )
