diff -aur old/src/wl/sys/wl_linux.c new/src/wl/sys/wl_linux.c
--- old/src/wl/sys/wl_linux.c	2010-12-25 18:17:33.333343048 +0100
+++ new/src/wl/sys/wl_linux.c	2010-12-25 18:21:05.856676376 +0100
@@ -483,7 +483,7 @@
 #ifdef WL_ALL_PASSIVE
 		spin_lock_init(&wl->txq_lock);
 #endif 
-		init_MUTEX(&wl->sem);
+		sema_init(&wl->sem, 1);
 	}
 
 	if (!(wl->wlc = wlc_attach((void *) wl, vendor, device, unit, wl->piomode,
