--- a/Makefile
+++ b/Makefile
@@ -107,6 +107,7 @@ LIBFLAGS += -L/opt/libjpeg-turbo/lib64
 endif
 
 LIBOPENAL = -lopenal
+LIBOPENGL = -lGL
 RELEASE_NAME = linux
 
 endif
--- a/libs/sdl/gl.c
+++ b/libs/sdl/gl.c
@@ -7,7 +7,7 @@
 #	include <OpenGLES/ES3/gl.h>
 #	define HL_GLES
 #elif defined(HL_MAC)
-#	include <SDL.h>
+#	include <SDL2/SDL.h>
 #	include <OpenGL/gl3.h>
 #	define glBindImageTexture(...) hl_error("Not supported on OSX")
 #	define glDispatchCompute(...) hl_error("Not supported on OSX")
--- a/libs/sdl/sdl.c
+++ b/libs/sdl/sdl.c
@@ -7,7 +7,7 @@
 #	include <SDL.h>
 #	include <SDL_syswm.h>
 #elif defined(HL_MAC)
-#	include <SDL.h>
+#	include <SDL2/SDL.h>
 #else
 #	include <SDL2/SDL.h>
 #endif
