From 31b97895b5b974bd08e2cb42ed333a804dbb4103 Mon Sep 17 00:00:00 2001
From: Someone Serge <sergei.kozlukov@aalto.fi>
Date: Tue, 21 Nov 2023 19:33:27 +0000
Subject: [PATCH 1/3] CMakeLists.txt: find_package by default, fallback to the
 vendored deps

---
 CMakeLists.txt | 43 +++++++++++++++++++++++++++++++------------
 1 file changed, 31 insertions(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 02250db..6b5a60c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,11 +70,38 @@ if (NOT APPLE)
 endif()
 
 # ----------------------------------------------------------
-#  Build the nanothread library
+#  Find or build the dependencies
 # ----------------------------------------------------------
 
-add_subdirectory(ext/nanothread)
-mark_as_advanced(NANOTHREAD_ENABLE_TESTS)
+find_package(nanothread)
+if (NOT nanothread_FOUND)
+  add_subdirectory(ext/nanothread)
+  mark_as_advanced(NANOTHREAD_ENABLE_TESTS)
+endif()
+
+find_package(PkgConfig)
+if (PkgConfig_FOUND)
+  pkg_check_modules(lz4 liblz4 IMPORTED_TARGET)
+  add_library(lz4 ALIAS PkgConfig::lz4)
+endif()
+if(NOT lz4_FOUND)
+  add_library(lz4 ext/lz4/lz4.c ext/lz4/xxhash.c)
+  target_include_directories(lz4 PUBLIC ext/lz4)
+endif()
+
+find_package(xxHash)
+if (NOT xxHash_FOUND)
+  add_library(xxhash ext/lz4/xxhash.c)
+  target_include_directories(xxhash PUBLIC ext/lz4)
+
+  add_library(xxHash::xxhash ALIAS xxhash)
+endif()
+
+
+find_package(tsl-robin-map)
+if (NOT tsl-robin-map_FOUND)
+  add_subdirectory(ext/robin_map)
+endif()
 
 # ----------------------------------------------------------
 #  Build Dr.Jit-Core
@@ -135,21 +162,13 @@ add_library(
   src/init.cpp
   src/api.cpp
 
-  # LZ4 compression library & XXHash hash function
-  ext/lz4/lz4.h ext/lz4/lz4.c
-  ext/lz4/xxhash.h ext/lz4/xxh3.h ext/lz4/xxhash.c
-
   # Precompiled kernels in compressed PTX format
   resources/kernels.h resources/kernels.c
 )
 
 target_compile_features(drjit-core PRIVATE cxx_std_17)
 
-target_include_directories(drjit-core PRIVATE
-  ext/nanothread/include
-  ext/robin_map/include
-  ext/lz4
-)
+target_link_libraries(drjit-core PRIVATE nanothread tsl::robin_map lz4 xxHash::xxhash)
 
 if (MSVC)
   # Conditional expression is constant (a few in robin_hash.h)
-- 
2.42.0

