Only in grub-1.97.1-orig/: grub-1.97.1
diff -rc -x '*~' grub-1.97.1-orig/util/getroot.c grub-1.97.1/util/getroot.c
*** grub-1.97.1-orig/util/getroot.c	2009-11-09 16:48:16.000000000 +0100
--- grub-1.97.1/util/getroot.c	2010-01-08 00:26:12.000000000 +0100
***************
*** 217,224 ****
  	continue;
  
        if (S_ISLNK (st.st_mode))
! 	/* Don't follow symbolic links.  */
! 	continue;
  
        if (S_ISDIR (st.st_mode))
  	{
--- 217,229 ----
  	continue;
  
        if (S_ISLNK (st.st_mode))
!         {
!           if (strcmp(dir, "mapper") != 0)
!             /* Don't follow symbolic links.  */
!             continue;
!           if (stat (ent->d_name, &st) < 0)
!             continue;
!         }
  
        if (S_ISDIR (st.st_mode))
  	{
