diff --git a/vboxdrv/Makefile.include.header b/vboxdrv/Makefile.include.header
index 8df1eb4d25..5a3e5604e7 100644
--- a/vboxdrv/Makefile.include.header
+++ b/vboxdrv/Makefile.include.header
@@ -117,7 +117,6 @@ else # neq($(KERNELRELEASE),)
 endif # neq($(KERNELRELEASE),)

 # Kernel build folder
-KERN_DIR := /lib/modules/$(KERN_VER)/build
 ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
  $(error Error: unable to find the headers of the Linux kernel to build against. \
           Specify KERN_VER=<version> and run Make again)
diff --git a/vboxnetadp/Makefile.include.header b/vboxnetadp/Makefile.include.header
index 8df1eb4d25..5a3e5604e7 100644
--- a/vboxnetadp/Makefile.include.header
+++ b/vboxnetadp/Makefile.include.header
@@ -117,7 +117,6 @@ else # neq($(KERNELRELEASE),)
 endif # neq($(KERNELRELEASE),)

 # Kernel build folder
-KERN_DIR := /lib/modules/$(KERN_VER)/build
 ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
  $(error Error: unable to find the headers of the Linux kernel to build against. \
           Specify KERN_VER=<version> and run Make again)
diff --git a/vboxnetflt/Makefile.include.header b/vboxnetflt/Makefile.include.header
index 8df1eb4d25..5a3e5604e7 100644
--- a/vboxnetflt/Makefile.include.header
+++ b/vboxnetflt/Makefile.include.header
@@ -117,7 +117,6 @@ else # neq($(KERNELRELEASE),)
 endif # neq($(KERNELRELEASE),)

 # Kernel build folder
-KERN_DIR := /lib/modules/$(KERN_VER)/build
 ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
  $(error Error: unable to find the headers of the Linux kernel to build against. \
           Specify KERN_VER=<version> and run Make again)
diff --git a/vboxpci/Makefile.include.header b/vboxpci/Makefile.include.header
index 8df1eb4d25..5a3e5604e7 100644
--- a/vboxpci/Makefile.include.header
+++ b/vboxpci/Makefile.include.header
@@ -117,7 +117,6 @@ else # neq($(KERNELRELEASE),)
 endif # neq($(KERNELRELEASE),)

 # Kernel build folder
-KERN_DIR := /lib/modules/$(KERN_VER)/build
 ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
  $(error Error: unable to find the headers of the Linux kernel to build against. \
           Specify KERN_VER=<version> and run Make again)
