diff --git a/src/gstrecord/gstinterface.cpp b/src/gstrecord/gstinterface.cpp
index b269b01..c075295 100644
--- a/src/gstrecord/gstinterface.cpp
+++ b/src/gstrecord/gstinterface.cpp
@@ -48,6 +48,7 @@ gstInterface::gstInterface()
 }
 QString gstInterface::libPath(const QString &sLib)
 {
+    return sLib;
     qInfo() << "gstreamer lib name is " << sLib;
     QDir dir;
     QString path  = QLibraryInfo::location(QLibraryInfo::LibrariesPath);
diff --git a/src/main_window.cpp b/src/main_window.cpp
index 7bfc78e..dcbbc2f 100755
--- a/src/main_window.cpp
+++ b/src/main_window.cpp
@@ -585,6 +585,7 @@ void MainWindow::initDynamicLibPath()
 }
 QString MainWindow::libPath(const QString &strlib)
 {
+    return strlib;
     QDir  dir;
     QString path  = QLibraryInfo::location(QLibraryInfo::LibrariesPath);
     dir.setPath(path);
diff --git a/src/waylandrecord/avlibinterface.cpp b/src/waylandrecord/avlibinterface.cpp
index d7942d6..c5cfdf4 100644
--- a/src/waylandrecord/avlibinterface.cpp
+++ b/src/waylandrecord/avlibinterface.cpp
@@ -105,6 +105,7 @@ avlibInterface::avlibInterface()
 
 QString avlibInterface::libPath(const QString &sLib)
 {
+    return sLib;
     //qDebug() << sLib;
     QDir dir;
     QString path  = QLibraryInfo::location(QLibraryInfo::LibrariesPath);
