diff -rc vsftpd-2.0.3/Makefile vsftpd-2.0.3-new/Makefile
*** vsftpd-2.0.3/Makefile	Fri Mar 11 20:34:15 2005
--- vsftpd-2.0.3-new/Makefile	Mon Sep 12 17:06:31 2005
***************
*** 5,11 ****
  #CFLAGS = -g
  CFLAGS	=	-O2 -Wall -W -Wshadow #-pedantic -Werror -Wconversion
  
- LIBS	=	`./vsf_findlibs.sh`
  LINK	=	-Wl,-s
  
  OBJS	=	main.o utility.o prelogin.o ftpcmdio.o postlogin.o privsock.o \
--- 5,10 ----
***************
*** 24,44 ****
  	$(CC) -o vsftpd $(OBJS) $(LINK) $(LIBS)
  
  install:
! 	if [ -x /usr/local/sbin ]; then \
! 		$(INSTALL) -m 755 vsftpd /usr/local/sbin/vsftpd; \
  	else \
! 		$(INSTALL) -m 755 vsftpd /usr/sbin/vsftpd; fi
! 	if [ -x /usr/local/man ]; then \
! 		$(INSTALL) -m 644 vsftpd.8 /usr/local/man/man8/vsftpd.8; \
! 		$(INSTALL) -m 644 vsftpd.conf.5 /usr/local/man/man5/vsftpd.conf.5; \
! 	elif [ -x /usr/share/man ]; then \
! 		$(INSTALL) -m 644 vsftpd.8 /usr/share/man/man8/vsftpd.8; \
! 		$(INSTALL) -m 644 vsftpd.conf.5 /usr/share/man/man5/vsftpd.conf.5; \
  	else \
! 		$(INSTALL) -m 644 vsftpd.8 /usr/man/man8/vsftpd.8; \
! 		$(INSTALL) -m 644 vsftpd.conf.5 /usr/man/man5/vsftpd.conf.5; fi
! 	if [ -x /etc/xinetd.d ]; then \
! 		$(INSTALL) -m 644 xinetd.d/vsftpd /etc/xinetd.d/vsftpd; fi
  
  clean:
  	rm -f *.o *.swp vsftpd
--- 23,41 ----
  	$(CC) -o vsftpd $(OBJS) $(LINK) $(LIBS)
  
  install:
! 	if [ -x $$out/sbin ]; then \
! 		$(INSTALL) -m 755 vsftpd $$out/sbin/vsftpd; \
  	else \
! 		$(INSTALL) -m 755 vsftpd $$out/sbin/vsftpd; fi
! 	if [ -x $$out/man ]; then \
! 		$(INSTALL) -m 644 vsftpd.8 $$out/man/man8/vsftpd.8; \
! 		$(INSTALL) -m 644 vsftpd.conf.5 $$out/man/man5/vsftpd.conf.5; \
! 	elif [ -x $$out/share/man ]; then \
! 		$(INSTALL) -m 644 vsftpd.8 $$out/man/man8/vsftpd.8; \
! 		$(INSTALL) -m 644 vsftpd.conf.5 $$out/share/man/man5/vsftpd.conf.5; \
  	else \
! 		$(INSTALL) -m 644 vsftpd.8 $$out/man/man8/vsftpd.8; \
! 		$(INSTALL) -m 644 vsftpd.conf.5 $$out/man/man5/vsftpd.conf.5; fi
  
  clean:
  	rm -f *.o *.swp vsftpd
