diff -Naur a/setup.py b/setup.py
--- a/setup.py	2021-04-24 12:53:18.300255006 -0500
+++ b/setup.py	2021-04-24 12:47:51.619052574 -0500
@@ -15,29 +15,6 @@
 from distutils.command.install import install
 import os,sys,shutil
 
-class GurobiClean(Command):
-    description = "remove the build directory"
-    user_options = []
-    def initialize_options(self):
-        self.cwd = None
-    def finalize_options(self):
-        self.cwd = os.path.dirname(os.path.realpath(__file__))
-    def run(self):
-        build_dir = os.path.join(os.getcwd(), "build")
-        if os.path.exists(build_dir):
-            print('removing %s' % build_dir)
-            shutil.rmtree(build_dir)
-
-class GurobiInstall(install):
-
-    # Calls the default run command, then deletes the build area
-    # (equivalent to "setup clean --all").
-    def run(self):
-        install.run(self)
-        c = GurobiClean(self.distribution)
-        c.finalize_options()
-        c.run()
-
 License = """
     This software is covered by the Gurobi End User License Agreement.
     By completing the Gurobi installation process and using the software,
@@ -78,7 +55,5 @@
       packages = ['gurobipy'],
       package_dir={'gurobipy' : srcpath },
       package_data = {'gurobipy' : [srcfile] },
-      cmdclass={'install' : GurobiInstall, 
-                'clean'   : GurobiClean }
       )
 
