diff -Naur a/setup.py b/setup.py
--- a/setup.py	2017-12-18 12:48:02.000000000 -0500
+++ b/setup.py	2017-12-18 12:48:43.000000000 -0500
@@ -54,16 +54,3 @@
       package_dir={'gurobipy' : srcpath },
       package_data = {'gurobipy' : [srcfile] }
       )
-
-if sys.platform == 'darwin':
-  from distutils.sysconfig import get_python_lib
-  import subprocess
-  import os.path
-  sitelib = get_python_lib() + '/gurobipy/gurobipy.so'
-  subprocess.call(('install_name_tool', '-change', 'libgurobi70.so', '/Library/gurobi702/mac64/lib/libgurobi70.so', sitelib)) # version for change
-  default = '/System/Library/Frameworks/Python.framework/Versions/2.7/Python'
-  modified = sys.prefix + '/Python'
-  if default != modified:
-    if not os.path.isfile(modified):
-      modified = sys.prefix + '/lib/libpython2.7.dylib' # For Anaconda
-    subprocess.call(('install_name_tool', '-change', default, modified, sitelib))
