diff -u ./configure ../bash-5.0-fixed/configure
--- ./configure	2019-01-02 15:43:31.000000000 +0100
+++ ../bash-5.0-fixed/configure	2020-01-08 14:18:21.017296179 +0100
@@ -16312,11 +16312,7 @@
 solaris2*)	LOCAL_CFLAGS=-DSOLARIS ;;
 lynxos*)	LOCAL_CFLAGS=-DRECYCLES_PIDS ;;
 linux*)		LOCAL_LDFLAGS=-rdynamic		 # allow dynamic loading
-		case "`uname -r`" in
-		1.*|2.[0123]*)	: ;;
-		*)	$as_echo "#define PGRP_PIPE 1" >>confdefs.h
- ;;
-		esac ;;
+		$as_echo "#define PGRP_PIPE 1" >>confdefs.h ;;
 netbsd*|openbsd*)	LOCAL_CFLAGS="-DDEV_FD_STAT_BROKEN" ;;
 *qnx[67]*)	LOCAL_LIBS="-lncurses" ;;
 *qnx*)		LOCAL_CFLAGS="-Dqnx -F -3s" LOCAL_LDFLAGS="-3s" LOCAL_LIBS="-lunix -lncurses" ;;
diff -u ./configure.ac ../bash-5.0-fixed/configure.ac
--- ./configure.ac	2019-01-02 15:39:11.000000000 +0100
+++ ../bash-5.0-fixed/configure.ac	2019-01-02 15:39:11.000000000 +0100
@@ -1108,10 +1108,7 @@
 solaris2*)	LOCAL_CFLAGS=-DSOLARIS ;;
 lynxos*)	LOCAL_CFLAGS=-DRECYCLES_PIDS ;;
 linux*)		LOCAL_LDFLAGS=-rdynamic		 # allow dynamic loading
-		case "`uname -r`" in
-		1.*|2.[[0123]]*)	: ;;
-		*)	AC_DEFINE(PGRP_PIPE) ;;
-		esac ;;
+		AC_DEFINE(PGRP_PIPE) ;;
 netbsd*|openbsd*)	LOCAL_CFLAGS="-DDEV_FD_STAT_BROKEN" ;;
 *qnx[[67]]*)	LOCAL_LIBS="-lncurses" ;;
 *qnx*)		LOCAL_CFLAGS="-Dqnx -F -3s" LOCAL_LDFLAGS="-3s" LOCAL_LIBS="-lunix -lncurses" ;;
