diff -u ./configure ../bash-4.4-fixed/configure
--- ./configure	2016-09-07 22:57:01.000000000 +0200
+++ ../bash-4.4-fixed/configure	2020-01-08 14:10:26.316858174 +0100
@@ -16064,10 +16064,7 @@
 solaris2*)	LOCAL_CFLAGS=-DSOLARIS ;;
 lynxos*)	LOCAL_CFLAGS=-DRECYCLES_PIDS ;;
 linux*)		LOCAL_LDFLAGS=-rdynamic		 # allow dynamic loading
-		case "`uname -r`" in
-		2.[456789]*|[34]*)	$as_echo "#define PGRP_PIPE 1" >>confdefs.h
- ;;
-		esac ;;
+		$as_echo "#define PGRP_PIPE 1" >>confdefs.h ;;
 *qnx6*)		LOCAL_CFLAGS="-Dqnx -Dqnx6" LOCAL_LIBS="-lncurses" ;;
 *qnx*)		LOCAL_CFLAGS="-Dqnx -F -3s" LOCAL_LDFLAGS="-3s" LOCAL_LIBS="-lunix -lncurses" ;;
 powerux*)	LOCAL_LIBS="-lgen" ;;
diff -u ./configure.ac ../bash-4.4-fixed/configure.ac
--- ./configure.ac	2016-09-07 22:56:28.000000000 +0200
+++ ../bash-4.4-fixed/configure.ac	2016-09-07 22:56:28.000000000 +0200
@@ -1092,9 +1092,7 @@
 solaris2*)	LOCAL_CFLAGS=-DSOLARIS ;;
 lynxos*)	LOCAL_CFLAGS=-DRECYCLES_PIDS ;;
 linux*)		LOCAL_LDFLAGS=-rdynamic		 # allow dynamic loading
-		case "`uname -r`" in
-		2.[[456789]]*|[[34]]*)	AC_DEFINE(PGRP_PIPE) ;;
-		esac ;;
+		AC_DEFINE(PGRP_PIPE) ;;
 *qnx6*)		LOCAL_CFLAGS="-Dqnx -Dqnx6" LOCAL_LIBS="-lncurses" ;;
 *qnx*)		LOCAL_CFLAGS="-Dqnx -F -3s" LOCAL_LDFLAGS="-3s" LOCAL_LIBS="-lunix -lncurses" ;;
 powerux*)	LOCAL_LIBS="-lgen" ;;
