--- a/meson.build
+++ b/meson.build
@@ -167,6 +167,12 @@
 install_headers('ui.h')
 
 # TODO when the API is stable enough to be versioned, create a pkg-config file (https://mesonbuild.com/Pkgconfig-module.html) and a declare_dependency() section too
+pkg = import('pkgconfig')
+pkg.generate(libui_libui,
+	description: 'Simple and portable (but not inflexible) GUI library',
+	filebase: 'libui',
+	name: 'libui',
+	version: '@version@')
 
 libui_binary_deps = []
 if libui_mode == 'static'
