diff --git a/meson.build b/meson.build
index 9675cd8d03..a4562b693c 100644
--- a/meson.build
+++ b/meson.build
@@ -1005,9 +1005,9 @@ meson.add_install_script(
   join_paths('tools', 'meson-post-install.sh'),
   nm_datadir,
   nm_bindir,
-  nm_pkgconfdir,
+  nm_prefix + nm_pkgconfdir,
   nm_pkglibdir,
-  nm_pkgstatedir,
+  nm_prefix + nm_pkgstatedir,
   nm_mandir,
   nm_sysconfdir,
   enable_docs ? '1' : '0',
