diff --git a/src/appearancegtk2.cpp b/src/appearancegtk2.cpp
index 3d89c82..4dcce69 100644
--- a/src/appearancegtk2.cpp
+++ b/src/appearancegtk2.cpp
@@ -56,7 +56,7 @@ QString AppearanceGTK2::themesGtkrcFile(const QString& themeName) const
     QStringList themes=installedThemes();
     themes=themes.filter(QRegExp("/"+themeName+"/?$"));
     if(themes.size()==1) {
-        QDirIterator it(themes.first(), QDirIterator::Subdirectories);
+        QDirIterator it(themes.first(), QDirIterator::Subdirectories | QDirIterator::FollowSymlinks);
         while(it.hasNext()) {
             it.next();
             if(it.fileName()=="gtkrc") {
