diff --git i/CMakeLists.txt w/CMakeLists.txt
index 0cd2140..16e3190 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -97,13 +97,11 @@ qt5_add_translation(QM_FILES
 )
 
 configure_file(src/cmake.h.in cmake.h)
-configure_file(data/icons/flags.qrc ${CircleFlags_SOURCE_DIR}/flags/flags.qrc COPYONLY)
-configure_file(data/icons/we10x.qrc ${We10X_SOURCE_DIR}/src/we10x.qrc COPYONLY)
 
 add_executable(${PROJECT_NAME}
-    ${CircleFlags_SOURCE_DIR}/flags/flags.qrc
+    data/icons/flags.qrc
     ${QM_FILES}
-    ${We10X_SOURCE_DIR}/src/we10x.qrc
+    data/icons/we10x.qrc
     data/icons/engines/engines.qrc
     src/addlanguagedialog.cpp
     src/addlanguagedialog.ui
diff --git i/cmake/ExternalLibraries.cmake w/cmake/ExternalLibraries.cmake
index d738716..fb01f3d 100644
--- i/cmake/ExternalLibraries.cmake
+++ w/cmake/ExternalLibraries.cmake
@@ -2,34 +2,28 @@ include(FetchContent)
 
 set(QAPPLICATION_CLASS QApplication)
 FetchContent_Declare(SingleApplication
-    GIT_REPOSITORY https://github.com/itay-grudev/SingleApplication
-    GIT_TAG v3.2.0
+    SOURCE_DIR @singleapplication@
 )
 
 FetchContent_Declare(QTaskbarControl
-    GIT_REPOSITORY https://github.com/Skycoder42/QTaskbarControl
-    GIT_TAG 2.0.2
+    SOURCE_DIR @qtaskbarcontrol@
 )
 
 option(QHOTKEY_INSTALL OFF)
 FetchContent_Declare(QHotkey
-    GIT_REPOSITORY https://github.com/Skycoder42/QHotkey
-    GIT_TAG 1.4.2
+    SOURCE_DIR @qhotkey@
 )
 
 FetchContent_Declare(QOnlineTranslator
-    GIT_REPOSITORY https://github.com/crow-translate/QOnlineTranslator
-    GIT_TAG 1.4.4
+    SOURCE_DIR @qonlinetranslator@
 )
 
 FetchContent_Declare(CircleFlags
-    GIT_REPOSITORY https://github.com/HatScripts/circle-flags
-    GIT_TAG v2.1.0
+    SOURCE_DIR @circleflags@
 )
 
 FetchContent_Declare(We10X
-    GIT_REPOSITORY https://github.com/yeyushengfan258/We10X-icon-theme
-    GIT_TAG bd2c68482a06d38b2641503af1ca127b9e6540db
+    SOURCE_DIR @we10x@
 )
 
 FetchContent_MakeAvailable(SingleApplication QTaskbarControl QHotkey QOnlineTranslator CircleFlags We10X)
