--- a/Gifcurry.cabal
+++ b/Gifcurry.cabal
@@ -149,9 +149,9 @@
                       , gi-gio >= 2.0 && < 3.0
                       , gi-glib >= 2.0 && < 3.0
                       , gi-pango >= 1.0 && < 2.0
-                      , gi-gdk >= 3.0 && < 4.0
+                      , gi-gdk3 >= 3.0 && < 4.0
                       , gi-gdkpixbuf >= 2.0 && < 3.0
-                      , gi-gtk >= 3.0 && < 4.0
+                      , gi-gtk3 >= 3.0 && < 4.0
                       , gi-cairo >= 1.0 && < 2.0
                       , gi-gstbase >= 1.0 && < 2.0
                       , gi-gst >= 1.0 && < 2.0
