diff --git a/libs/partners/chroma/tests/integration_tests/test_vectorstores.py b/libs/partners/chroma/tests/integration_tests/test_vectorstores.py
index 382b24cb5..f99a34112 100644
--- a/tests/integration_tests/test_vectorstores.py
+++ b/tests/integration_tests/test_vectorstores.py
@@ -36,7 +36,7 @@ def test_chroma() -> None:
 
     assert output == [Document(page_content="foo")]
 
-
+@pytest.mark.asyncio
 async def test_chroma_async() -> None:
     """Test end to end construction and search."""
     texts = ["foo", "bar", "baz"]
