diff --git a/src/Prefs.cpp b/src/Prefs.cpp
index 76385c4..db8d20d 100755
--- a/src/Prefs.cpp
+++ b/src/Prefs.cpp
@@ -795,7 +795,6 @@ int Prefs::checkOptions() {
          ntop->getTrace()->traceEvent(TRACE_ERROR, "Unable to create log %s", path);
     }
 
-  free(data_dir); data_dir = strdup(ntop->get_install_dir());
   docs_dir      = ntop->getValidPath(docs_dir);
   scripts_dir   = ntop->getValidPath(scripts_dir);
   callbacks_dir = ntop->getValidPath(callbacks_dir);
