diff --git a/exp_win.c b/exp_win.c
index 9ffad9f..b46d024 100644
--- a/exp_win.c
+++ b/exp_win.c
@@ -39,8 +39,8 @@ conflicts with sys/ioctl.h
 /* Sigh.  On AIX 2.3, termios.h exists but does not define TIOCGWINSZ */
 /* Instead, it has to come from ioctl.h.  However, As I said above, this */
 /* can't be cavalierly included on all machines, even when it exists. */
-/* Darwin also has termios.h, but it requires ioctl.h for `ioctl`. */
-#if defined(HAVE_TERMIOS) && (defined(__APPLE__) || !defined(HAVE_TIOCGWINSZ_IN_TERMIOS_H))
+/* Darwin and Cygwin also have termios.h, but require ioctl.h for `ioctl`. */
+#if defined(HAVE_TERMIOS) && (defined(__APPLE__) || defined(__CYGWIN__) || !defined(HAVE_TIOCGWINSZ_IN_TERMIOS_H))
 #  include <sys/ioctl.h>
 #endif
 
