diff --git a/src/pytesseract.py b/src/pytesseract.py
index 32713cf..5f9209d 100755
--- a/src/pytesseract.py
+++ b/src/pytesseract.py
@@ -25,7 +25,7 @@ if numpy_installed:
     from numpy import ndarray
 
 # CHANGE THIS IF TESSERACT IS NOT IN YOUR PATH, OR IS NAMED DIFFERENTLY
-tesseract_cmd = 'tesseract'
+tesseract_cmd = '@drv@/bin/tesseract'
 RGB_MODE = 'RGB'
 OSD_KEYS = {
     'Page number': ('page_num', int),
