diff -Naur cherrytree-0.37.1-orig/setup.py cherrytree-0.37.1/setup.py
--- cherrytree-0.37.1-orig/setup.py	2016-01-08 20:50:50.000000000 +0100
+++ cherrytree-0.37.1/setup.py	2016-07-05 20:30:27.768178682 +0200
@@ -205,4 +205,9 @@
           },
        distclass=CherryTreeDist
     )
-    subprocess.call("update-desktop-database")
+    try:
+        subprocess.check_call(['update-desktop-database'])
+    except subprocess.CalledProcessError:
+        pass # handle errors in the called executable
+    except OSError:
+        pass # executable not found
