diff -ruN syslinux-3.09/Makefile syslinux-3.09.new/Makefile
--- syslinux-3.09/Makefile	2005-04-02 23:48:47.000000000 +0200
+++ syslinux-3.09.new/Makefile	2005-07-30 20:49:41.000000000 +0200
@@ -26,11 +26,11 @@
 
 NASM	 = nasm -O99
 NINCLUDE = 
-BINDIR   = /usr/bin
-SBINDIR  = /sbin
-LIBDIR   = /usr/lib
+BINDIR   = $(out)/bin
+SBINDIR  = $(out)/sbin
+LIBDIR   = $(out)/lib
 AUXDIR   = $(LIBDIR)/syslinux
-INCDIR   = /usr/include
+INCDIR   = $(out)/include
 
 PERL     = perl
 
